commit 65173d72f1a662a8596aa0e14232556ba40e452f (origin/master, origin/HEAD, master)
Author: Bill Nottingham <notting@redhat.com>
Date:   Fri Dec 7 16:21:59 2012 -0500

    Fix build.

diff --git a/Makefile b/Makefile
index 1c6bce8..c6fc7a0 100644
--- a/Makefile
+++ b/Makefile
@@ -22,6 +22,7 @@ install:
 	install -m644  adjtime $(ROOT)/etc
 	install -m644 inittab $(ROOT)/etc
 	install -m644  rwtab statetab networks $(ROOT)/etc
+	install -m755  service $(ROOT)/sbin
 	install -m644  lang.csh lang.sh $(ROOT)/etc/profile.d
 	install -m644  term256.csh term256.sh $(ROOT)/etc/profile.d
 	install -m644  debug.csh debug.sh $(ROOT)/etc/profile.d
diff --git a/initscripts.spec b/initscripts.spec
index ce34c7f..c7f3a1c 100644
--- a/initscripts.spec
+++ b/initscripts.spec
@@ -175,7 +175,6 @@ rm -rf $RPM_BUILD_ROOT
 %exclude /etc/profile.d/debug*
 /etc/profile.d/*
 /usr/sbin/sys-unconfig
-/sbin/setsysfont
 /bin/ipcalc
 /bin/usleep
 %attr(4755,root,root) /usr/sbin/usernetctl
