diff -ur initscripts-9.37.orig/Makefile initscripts-9.37/Makefile
--- initscripts-9.37.orig/Makefile	2012-04-12 08:29:18.000000000 -0400
+++ initscripts-9.37/Makefile	2012-04-12 08:34:24.000000000 -0400
@@ -13,7 +13,7 @@
 	make -C po
 
 install:
-	mkdir -p $(ROOT)/etc/profile.d $(ROOT)/sbin $(ROOT)/usr/sbin
+	mkdir -p $(ROOT)/etc/profile.d $(ROOT)/usr/sbin
 	mkdir -p $(ROOT)$(mandir)/man{5,8}
 	mkdir -p $(ROOT)/etc/rwtab.d $(ROOT)/etc/statetab.d
 	mkdir -p $(ROOT)/var/lib/stateless/writable
@@ -27,7 +27,7 @@
 	install -m644 inittab.upstart $(ROOT)/etc/inittab.upstart
 	install -m644 inittab.systemd $(ROOT)/etc/inittab.systemd
 	install -m644  rwtab statetab networks $(ROOT)/etc
-	install -m755  service setsysfont $(ROOT)/sbin
+	install -m755  service setsysfont $(ROOT)/usr/sbin
 	install -m644  lang.csh lang.sh $(ROOT)/etc/profile.d
 	install -m644  debug.csh debug.sh $(ROOT)/etc/profile.d
 	install -m755  sys-unconfig $(ROOT)/usr/sbin
@@ -48,11 +48,11 @@
 	install -m644 sysconfig/debug sysconfig/init sysconfig/netconsole sysconfig/readonly-root $(ROOT)/etc/sysconfig/
 	cp -af sysconfig/network-scripts $(ROOT)/etc/sysconfig/
 	cp -af ppp NetworkManager init $(ROOT)/etc
-	mkdir -p $(ROOT)/lib/systemd/
-	cp -af systemd/* $(ROOT)/lib/systemd/
+	mkdir -p $(ROOT)/usr/lib/systemd/
+	cp -af systemd/* $(ROOT)/usr/lib/systemd/
 	mkdir -p $(ROOT)/etc/ppp/peers
-	mkdir -p $(ROOT)/lib
-	cp -af udev $(ROOT)/lib
+	mkdir -p $(ROOT)/usr/lib
+	cp -af udev $(ROOT)/usr/lib
 	chmod 755 $(ROOT)/etc/rc.d/* $(ROOT)/etc/rc.d/init.d/*
 	chmod 644 $(ROOT)/etc/rc.d/init.d/functions
 	chmod 755 $(ROOT)/etc/ppp/peers
@@ -77,13 +77,13 @@
 	  install -m644 sysconfig/init.s390 $(ROOT)/etc/sysconfig/init ; \
 	fi
 
-	mv $(ROOT)/etc/sysconfig/network-scripts/ifup $(ROOT)/sbin
-	mv $(ROOT)/etc/sysconfig/network-scripts/ifdown $(ROOT)/sbin
+	mv $(ROOT)/etc/sysconfig/network-scripts/ifup $(ROOT)/usr/sbin
+	mv $(ROOT)/etc/sysconfig/network-scripts/ifdown $(ROOT)/usr/sbin
 	(cd $(ROOT)/etc/sysconfig/network-scripts; \
 	  ln -sf ifup-ippp ifup-isdn ; \
 	  ln -sf ifdown-ippp ifdown-isdn ; \
-	  ln -sf ../../../sbin/ifup . ; \
-	  ln -sf ../../../sbin/ifdown . )
+	  ln -sf ../../../usr/sbin/ifup . ; \
+	  ln -sf ../../../usr/sbin/ifdown . )
 	make install ROOT=$(ROOT) mandir=$(mandir) -C src
 	make install PREFIX=$(ROOT) -C po
 
@@ -116,20 +116,20 @@
 
 	ln -s halt $(ROOT)/etc/rc.d/init.d/reboot
 
-	mkdir -p -m 755 $(ROOT)/lib/systemd/system/multi-user.target.wants
-	mkdir -p -m 755 $(ROOT)/lib/systemd/system/graphical.target.wants
-	ln -s reboot.target $(ROOT)/lib/systemd/system/ctrl-alt-del.target
-	mkdir -p -m 755 $(ROOT)/lib/systemd/system/local-fs.target.wants
-	mkdir -p -m 755 $(ROOT)/lib/systemd/system/basic.target.wants
-	mkdir -p -m 755 $(ROOT)/lib/systemd/system/sysinit.target.wants
-	ln -s ../fedora-configure.service $(ROOT)/lib/systemd/system/basic.target.wants
-	ln -s ../fedora-loadmodules.service $(ROOT)/lib/systemd/system/basic.target.wants
-	ln -s ../fedora-autorelabel.service $(ROOT)/lib/systemd/system/basic.target.wants
-	ln -s ../fedora-autorelabel-mark.service $(ROOT)/lib/systemd/system/basic.target.wants
-	ln -s ../fedora-readonly.service $(ROOT)/lib/systemd/system/local-fs.target.wants
-	ln -s ../fedora-import-state.service $(ROOT)/lib/systemd/system/local-fs.target.wants
-	ln -s ../fedora-storage-init.service $(ROOT)/lib/systemd/system/local-fs.target.wants
-	ln -s ../fedora-storage-init-late.service $(ROOT)/lib/systemd/system/local-fs.target.wants
+	mkdir -p -m 755 $(ROOT)/usr/lib/systemd/system/multi-user.target.wants
+	mkdir -p -m 755 $(ROOT)/usr/lib/systemd/system/graphical.target.wants
+	ln -s reboot.target $(ROOT)/usr/lib/systemd/system/ctrl-alt-del.target
+	mkdir -p -m 755 $(ROOT)/usr/lib/systemd/system/local-fs.target.wants
+	mkdir -p -m 755 $(ROOT)/usr/lib/systemd/system/basic.target.wants
+	mkdir -p -m 755 $(ROOT)/usr/lib/systemd/system/sysinit.target.wants
+	ln -s ../fedora-configure.service $(ROOT)/usr/lib/systemd/system/basic.target.wants
+	ln -s ../fedora-loadmodules.service $(ROOT)/usr/lib/systemd/system/basic.target.wants
+	ln -s ../fedora-autorelabel.service $(ROOT)/usr/lib/systemd/system/basic.target.wants
+	ln -s ../fedora-autorelabel-mark.service $(ROOT)/usr/lib/systemd/system/basic.target.wants
+	ln -s ../fedora-readonly.service $(ROOT)/usr/lib/systemd/system/local-fs.target.wants
+	ln -s ../fedora-import-state.service $(ROOT)/usr/lib/systemd/system/local-fs.target.wants
+	ln -s ../fedora-storage-init.service $(ROOT)/usr/lib/systemd/system/local-fs.target.wants
+	ln -s ../fedora-storage-init-late.service $(ROOT)/usr/lib/systemd/system/local-fs.target.wants
 
 	mkdir -p $(ROOT)/etc/tmpfiles.d
 	install -m 644 initscripts.tmpfiles.d $(ROOT)/etc/tmpfiles.d/initscripts.conf
diff -ur initscripts-9.37.orig/src/Makefile initscripts-9.37/src/Makefile
--- initscripts-9.37.orig/src/Makefile	2012-04-12 08:29:18.000000000 -0400
+++ initscripts-9.37/src/Makefile	2012-04-12 10:41:19.000000000 -0400
@@ -18,21 +18,21 @@
 	rm -f $(PROGS) *.o *~
 
 install:
-	mkdir -p $(ROOT)/bin $(ROOT)/usr/sbin $(ROOT)$(mandir)/man{1,8} $(ROOT)/etc $(ROOT)/lib/udev
-	install -m 755 usleep $(ROOT)/bin/usleep
+	mkdir -p $(ROOT)/usr/bin $(ROOT)/usr/sbin $(ROOT)$(mandir)/man{1,8} $(ROOT)/etc $(ROOT)/usr/lib/udev
+	install -m 755 usleep $(ROOT)/usr/bin/usleep
 	install -m 4755 usernetctl $(ROOT)/usr/sbin/usernetctl
-	install -m 2755 netreport $(ROOT)/sbin/netreport
-	install -m 755 ipcalc $(ROOT)/bin/ipcalc
-	install -m 755 fstab-decode $(ROOT)/sbin/fstab-decode
-	install -m 755 genhostid $(ROOT)/sbin/genhostid
-	install -m 755 getkey $(ROOT)/sbin/getkey
-	install -m 755 ppp-watch $(ROOT)/sbin/ppp-watch
-	install -m 755 consoletype $(ROOT)/sbin/consoletype
-	install -m 755 securetty $(ROOT)/sbin/securetty
-	install -m 755 sushell $(ROOT)/sbin/sushell
-	install -m 755 rename_device $(ROOT)/lib/udev/rename_device
-	install -m 755 console_init $(ROOT)/lib/udev/console_init
-	install -m 755 console_check $(ROOT)/lib/udev/console_check
+	install -m 2755 netreport $(ROOT)/usr/sbin/netreport
+	install -m 755 ipcalc $(ROOT)/usr/bin/ipcalc
+	install -m 755 fstab-decode $(ROOT)/usr/sbin/fstab-decode
+	install -m 755 genhostid $(ROOT)/usr/sbin/genhostid
+	install -m 755 getkey $(ROOT)/usr/sbin/getkey
+	install -m 755 ppp-watch $(ROOT)/usr/sbin/ppp-watch
+	install -m 755 consoletype $(ROOT)/usr/sbin/consoletype
+	install -m 755 securetty $(ROOT)/usr/sbin/securetty
+	install -m 755 sushell $(ROOT)/usr/sbin/sushell
+	install -m 755 rename_device $(ROOT)/usr/lib/udev/rename_device
+	install -m 755 console_init $(ROOT)/usr/lib/udev/console_init
+	install -m 755 console_check $(ROOT)/usr/lib/udev/console_check
 	install -m 644 genhostid.1 $(ROOT)$(mandir)/man1
 	install -m 644 getkey.1 $(ROOT)$(mandir)/man1
 	install -m 644 netreport.1 $(ROOT)$(mandir)/man1
