diff -ur kde-workspace-4.10.3.orig/kcontrol/fonts/fonts.cpp kde-workspace-4.10.3/kcontrol/fonts/fonts.cpp
--- kde-workspace-4.10.3.orig/kcontrol/fonts/fonts.cpp	2013-05-31 05:17:22.000000000 +0000
+++ kde-workspace-4.10.3/kcontrol/fonts/fonts.cpp	2013-05-31 06:51:22.000000000 +0000
@@ -525,13 +525,13 @@
   QFont f1("Monospace", 16;
   QFont f2("Sans Serif", 16); // toolbar
 #else
-  QFont f0("Sans Serif", 9); // general/menu/desktop
-  QFont f1("Monospace", 9);
-  QFont f2("Sans Serif", 8); // toolbar
+  QFont f0("Sans Serif", 10); // general/menu/desktop
+  QFont f1("Monospace", 10);
+  QFont f2("Sans Serif", 10); // toolbar
 #endif
-  QFont f3("Sans Serif", 8); // window title
-  QFont f4("Sans Serif", 9); // taskbar
-  QFont f5("Sans Serif", 8); // smallestReadableFont
+  QFont f3("Sans Serif", 10); // window title
+  QFont f4("Sans Serif", 10); // taskbar
+  QFont f5("Sans Serif", 9); // smallestReadableFont
 
   defaultFontList << f0 << f1 << f5 << f2 << f0 << f3 << f4 << f0;
 
