diff -up kdebase-workspace-4.1.80/kdm/kfrontend/kgreeter.cpp.orig kdebase-workspace-4.1.80/kdm/kfrontend/kgreeter.cpp
--- kdebase-workspace-4.1.80/kdm/kfrontend/kgreeter.cpp.orig	2008-11-19 23:44:43.000000000 +0100
+++ kdebase-workspace-4.1.80/kdm/kfrontend/kgreeter.cpp	2008-11-19 23:45:55.000000000 +0100
@@ -520,10 +520,12 @@ KGreeter::setPrevWM( QAction *wm )
 	if (curPrev != wm) {
 		if (curPrev)
 			curPrev->setText( sessionTypes[curPrev->data().toInt()].name );
-		if (wm)
+		if (wm) {
 			wm->setText( i18nc( "@item:inmenu session type",
 			                    "%1 (previous)",
 			                    sessionTypes[wm->data().toInt()].name ) );
+			wm->toggle();
+		}
 		curPrev = wm;
 	}
 }
