Index: kcontrol/colors/colorscm.cpp
===================================================================
--- kcontrol/colors/colorscm.cpp	(revision 807742)
+++ kcontrol/colors/colorscm.cpp	(working copy)
@@ -55,7 +55,7 @@
 {
     // NOTE: keep this in sync with kdelibs/kdeui/kernel/kglobalsettings.cpp
     KConfigGroup group(config, "WM");
-    m_colors[ActiveBackground] = group.readEntry("activeBackground", QColor(48, 174, 232));
+    m_colors[ActiveBackground] = group.readEntry("activeBackground", QColor(99, 210, 112));
     m_colors[ActiveForeground] = group.readEntry("activeForeground", QColor(255, 255, 255));
     m_colors[InactiveBackground] = group.readEntry("inactiveBackground", QColor(224, 223, 222));
     m_colors[InactiveForeground] = group.readEntry("inactiveForeground", QColor(20, 19, 18));
Index: kcontrol/colors/schemes/Oxygen.colors
===================================================================
--- kcontrol/colors/schemes/Oxygen.colors	(revision 807742)
+++ kcontrol/colors/schemes/Oxygen.colors	(working copy)
@@ -89,7 +89,7 @@
 Name=Oxygen
 
 [WM]
-activeBackground=48,174,232
+activeBackground=99,210,112
 activeForeground=255,255,255
 inactiveBackground=224,223,222
 inactiveForeground=20,19,18
