Index: kdebase-workspace-4.4.2/kwin/events.cpp
===================================================================
--- kdebase-workspace-4.4.2.orig/kwin/events.cpp	2010-04-07 23:09:06.190839924 -0400
+++ kdebase-workspace-4.4.2/kwin/events.cpp	2010-04-07 23:10:33.414839796 -0400
@@ -1390,7 +1390,7 @@
             {
             finishMoveResize( false );
             // mouse position is still relative to old Client position, adjust it
-            QPoint mousepos( x_root - x, y_root - y );
+            QPoint mousepos( x_root - x + padding_left, y_root - y + padding_top );
             mode = mousePosition( mousepos );
             }
         else if( workspace()->decorationSupportsClientGrouping() )
@@ -1474,7 +1474,10 @@
         return true; // care only about the whole frame
     if ( !buttonDown ) 
         {
-        Position newmode = mousePosition( QPoint( x, y ));
+        QPoint mousePos( x, y );
+        if( w == frameId() )
+            mousePos += QPoint( padding_left, padding_top );
+        Position newmode = mousePosition( mousePos );
         if( newmode != mode )
             {
             mode = newmode;
