diff -up bash-completion-1.3/completions/perl~ bash-completion-1.3/completions/perl
--- bash-completion-1.3/completions/perl~	2011-01-21 11:36:11.000000000 +0200
+++ bash-completion-1.3/completions/perl	2011-04-15 23:32:36.110480132 +0300
@@ -112,7 +112,7 @@ _perldoc()
         if [[ "$cur" != */* ]]; then
             _perlmodules
             COMPREPLY=( "${COMPREPLY[@]}" $( compgen -W \
-                '$( PAGER=/bin/cat man perl |  \
+                '$( MANPAGER=/bin/cat PAGER=/bin/cat man perl |  \
                 sed -ne "/perl.*Perl overview/,/perlwin32/p" | \
                 awk "\$NF=2 { print \$1}" | command grep perl )' -- "$cur" ) )
         fi
