commit 18ecd51264727c75ab3e17a762d8f95361023f0a
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1340749881 -0500

    Fix FTBFS on png >= 0.15

diff --git a/digikam/kioslave/digikamthumbnail.cpp b/digikam/kioslave/digikamthumbnail.cpp
index 6dba518..97acf12 100644
--- a/digikam/kioslave/digikamthumbnail.cpp
+++ b/digikam/kioslave/digikamthumbnail.cpp
@@ -413,16 +413,16 @@ TQImage kio_digikamthumbnailProtocol::loadPNG(const TQString& path)
     if (color_type == PNG_COLOR_TYPE_PALETTE)
         png_set_expand(png_ptr);
 
-    if (info_ptr->color_type == PNG_COLOR_TYPE_RGB_ALPHA)
+    if (color_type == PNG_COLOR_TYPE_RGB_ALPHA)
         has_alpha = 1;
 
-    if (info_ptr->color_type == PNG_COLOR_TYPE_GRAY_ALPHA)
+    if (color_type == PNG_COLOR_TYPE_GRAY_ALPHA)
     {
         has_alpha = 1;
         has_grey = 1;
     }
 
-    if (info_ptr->color_type == PNG_COLOR_TYPE_GRAY)
+    if (color_type == PNG_COLOR_TYPE_GRAY)
         has_grey = 1;
 
     unsigned char **lines;
