diff -Nuri libyuv-svn20140603-orig/include/libyuv/row.h libyuv-svn20140603/include/libyuv/row.h
--- libyuv-svn20140603-orig/include/libyuv/row.h        2014-07-03 06:23:30.592556220 +0000
+++ libyuv-svn20140603/include/libyuv/row.h     2014-07-03 06:33:21.072350733 +0000
@@ -331,7 +331,7 @@
 #endif

 // The following are available on Mips platforms:
-#if !defined(LIBYUV_DISABLE_MIPS) && defined(__mips__)
+#if !defined(LIBYUV_DISABLE_MIPS) && defined(__mips__) && !defined(__mips64)
 #define HAS_COPYROW_MIPS
 #if defined(__mips_dsp) && (__mips_dsp_rev >= 2)
 #define HAS_I422TOABGRROW_MIPS_DSPR2
