--- trinity-kasablanca-14.0.0~pre22+07aa7379/src/ftpthread.cpp.ORI	2014-02-19 22:52:20.875005485 +0100
+++ trinity-kasablanca-14.0.0~pre22+07aa7379/src/ftpthread.cpp	2014-02-19 22:53:01.488995960 +0100
@@ -1094,8 +1094,8 @@
 
 bool FtpThread::FormatFilelist(const char *filename, TQString current, filist *dirtable, filist *filetable)
 {
-	int i, blocks, space, month_int = 1;
-	unsigned int loc, fileloc, datebegin, sizebegin = 0;
+	int i, blocks, space, month_int = 1, loc;
+	unsigned int fileloc, datebegin, sizebegin = 0;
 	string buffer, filestring;
 
 	FILE* dirfile;
diff --git a/src/Makefile.am b/src/Makefile.am
index 0b6b597..8cf8cb2 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -25,8 +25,7 @@ kasablanca_LDFLAGS = $(TDE_RPATH) $(all_libraries)
 kasablanca_LDADD = -lssl $(LIB_TDEUI) -lcrypto $(LIB_TDECORE) -lDCOP $(LIB_QT)
 
 # this is where the desktop file will go 
-shelldesktopdir = $(kde_appsdir)/Utilities
-shelldesktop_DATA = kasablanca.desktop
+xdg_apps_DATA = kasablanca.desktop
 
 # this is where the shell's XML-GUI resource file goes
 shellrcdir = $(kde_datadir)/kasablanca
