diff -up pyenchant-1.6.5/enchant/tests.py.fix-docstring-test pyenchant-1.6.5/enchant/tests.py
--- pyenchant-1.6.5/enchant/tests.py.fix-docstring-test	2012-11-01 09:39:36.268616326 +0100
+++ pyenchant-1.6.5/enchant/tests.py	2012-11-01 09:40:54.037488976 +0100
@@ -414,7 +414,8 @@ class TestDocStrings(unittest.TestCase):
              "dictwithpwl","skippable","dicts","dict's","filenames",
              "trie","api","ctypes","wxspellcheckerdialog","stateful",
              "cmdlinechecker","spellchecks","callback","clunkier","iterator",
-             "ispell","cor","backends"]
+             "ispell","cor","backends", "incant", "runtime", "subclasses",
+             "initialise"]
 
     def test_docstrings(self):
         """Test that all our docstrings are error-free."""
