diff -up pyenchant-1.6.5/enchant/tests.py.fix-tests-without-X pyenchant-1.6.5/enchant/tests.py
--- pyenchant-1.6.5/enchant/tests.py.fix-tests-without-X	2012-11-01 08:54:43.261512454 +0100
+++ pyenchant-1.6.5/enchant/tests.py	2012-11-01 09:01:12.184938020 +0100
@@ -427,11 +427,11 @@ class TestDocStrings(unittest.TestCase):
         import enchant.checker.CmdLineChecker
         try:
             import enchant.checker.GtkSpellCheckerDialog
-        except ImportError:
+        except (ImportError, RuntimeError):
             pass
         try:
             import enchant.checker.wxSpellCheckerDialog
-        except ImportError:
+        except (ImportError, RuntimeError):
             pass
         errors = []
         #  Naive recursion here would blow the stack, instead we
