diff -u -r mxd.orig/src/rppppoek.sh mxd/src/rppppoek.sh
--- mxd.orig/src/rppppoek.sh	2002-06-19 05:36:21.000000000 +0800
+++ mxd/src/rppppoek.sh	2008-09-09 17:07:33.000000000 +0800
@@ -21,10 +21,10 @@
 SUDO=/usr/bin/sudo
 
 # Path to adsl-start (Roaring Penguin utils)
-RPC=/usr/sbin/adsl-start
+RPC=/usr/sbin/pppoe-start
 
 # Path to adsl-stop (Roaring Penguin utils)
-RPD=/usr/sbin/adsl-stop
+RPD=/usr/sbin/pppoe-stop
 
 #####################################################################
 #  You shouldn't modify anything below this line, if you don't know #
