diff -up chromium-9.0.600.0/native_client/src/trusted/service_runtime/service_runtime.gyp.linking-fix chromium-9.0.600.0/native_client/src/trusted/service_runtime/service_runtime.gyp
--- chromium-9.0.600.0/native_client/src/trusted/service_runtime/service_runtime.gyp.linking-fix	2010-12-03 06:56:29.000000000 -0500
+++ chromium-9.0.600.0/native_client/src/trusted/service_runtime/service_runtime.gyp	2010-12-13 12:56:50.063551554 -0500
@@ -187,6 +187,16 @@
                     'linux/arm/sel_segments.c',
                   ],
                 }],
+                ['target_arch=="ia32"', {
+                  'sources': [
+                    'linux/nacl_signal_32.c',
+                  ],
+                }],
+                ['target_arch=="x64"', {
+                  'sources': [
+                    'linux/nacl_signal_64.c',
+                  ],
+                }],
               ],
             }],
             ['OS=="linux" or OS=="mac" or OS=="FreeBSD"', {
