diff -up chromium-45.0.2454.101/third_party/boringssl/boringssl.gyp.addrfix chromium-45.0.2454.101/third_party/boringssl/boringssl.gyp
--- chromium-45.0.2454.101/third_party/boringssl/boringssl.gyp.addrfix	2015-10-07 10:22:48.843303325 -0400
+++ chromium-45.0.2454.101/third_party/boringssl/boringssl.gyp	2015-10-07 10:23:27.086469277 -0400
@@ -17,6 +17,7 @@
       'defines': [
         'BORINGSSL_IMPLEMENTATION',
         'BORINGSSL_NO_STATIC_INITIALIZER',
+        '_POSIX_C_SOURCE=200112L',
       ],
       # TODO(davidben): Fix size_t truncations in BoringSSL.
       # https://crbug.com/429039
