diff -up chromium-12.0.718.0/chrome/chrome_common.gypi.nss chromium-12.0.718.0/chrome/chrome_common.gypi
--- chromium-12.0.718.0/chrome/chrome_common.gypi.nss	2011-03-31 17:00:34.313855213 -0400
+++ chromium-12.0.718.0/chrome/chrome_common.gypi	2011-03-31 17:06:19.778879575 -0400
@@ -18,6 +18,11 @@
               '<(DEPTH)/third_party/wtl/include',
             ],
           }],
+          ['OS=="linux"', {
+            'include_dirs': [
+              '/usr/include/nspr4',
+            ],
+          }],
         ],
         'sources': [
           # .cc, .h, and .mm files under chrome/common that are used on all
