diff -up chromium-39.0.2171.19/build/linux/system.gyp.link_gio chromium-39.0.2171.19/build/linux/system.gyp
--- chromium-39.0.2171.19/build/linux/system.gyp.link_gio	2014-10-22 16:45:02.597002731 +0200
+++ chromium-39.0.2171.19/build/linux/system.gyp	2014-10-22 16:45:17.403998542 +0200
@@ -651,6 +651,23 @@
             },
           ],
         }],
+        ['use_gio==0 and _toolset=="target"', {
+          'link_settings': {
+            'ldflags': [
+              '<!@(<(pkg-config) --libs-only-L --libs-only-other gio-2.0)',
+            ],
+            'libraries': [
+              '<!@(<(pkg-config) --libs-only-l gio-2.0)',
+            ],
+            'conditions': [
+              ['linux_link_gsettings==0 and OS=="linux"', {
+                'libraries': [
+                  '-ldl',
+                ],
+              }],
+            ],
+          },
+        }],
       ],
     },
     {
