diff -up chromium-23.0.1271.95/base/pr_time_unittest.cc.system-code chromium-23.0.1271.95/base/pr_time_unittest.cc
--- chromium-23.0.1271.95/base/pr_time_unittest.cc.system-code	2012-11-27 21:01:53.000000000 -0500
+++ chromium-23.0.1271.95/base/pr_time_unittest.cc	2012-12-12 15:22:19.695293771 -0500
@@ -5,7 +5,7 @@
 #include <time.h>
 
 #include "base/compiler_specific.h"
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
diff -up chromium-23.0.1271.95/base/time.cc.system-code chromium-23.0.1271.95/base/time.cc
diff -up chromium-23.0.1271.95/chrome/browser/metrics/metrics_log.cc.system-code chromium-23.0.1271.95/chrome/browser/metrics/metrics_log.cc
--- chromium-23.0.1271.95/chrome/browser/metrics/metrics_log.cc.system-code	2012-11-27 21:02:10.000000000 -0500
+++ chromium-23.0.1271.95/chrome/browser/metrics/metrics_log.cc	2012-12-12 15:22:19.699293773 -0500
@@ -16,7 +16,7 @@
 #include "base/string_number_conversions.h"
 #include "base/string_util.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 #include "base/time.h"
 #include "base/tracked_objects.h"
 #include "base/utf_string_conversions.h"
diff -up chromium-23.0.1271.95/chrome/browser/task_manager/task_manager_resource_providers.cc.system-code chromium-23.0.1271.95/chrome/browser/task_manager/task_manager_resource_providers.cc
--- chromium-23.0.1271.95/chrome/browser/task_manager/task_manager_resource_providers.cc.system-code	2012-11-27 21:02:10.000000000 -0500
+++ chromium-23.0.1271.95/chrome/browser/task_manager/task_manager_resource_providers.cc	2012-12-12 15:22:19.699293773 -0500
@@ -63,7 +63,7 @@
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
 #include "ui/gfx/image/image_skia.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #if defined(OS_MACOSX)
 #include "ui/gfx/image/image_skia_util_mac.h"
diff -up chromium-23.0.1271.95/chrome/browser/ui/webui/about_ui.cc.system-code chromium-23.0.1271.95/chrome/browser/ui/webui/about_ui.cc
--- chromium-23.0.1271.95/chrome/browser/ui/webui/about_ui.cc.system-code	2012-12-12 15:31:59.208309604 -0500
+++ chromium-23.0.1271.95/chrome/browser/ui/webui/about_ui.cc	2012-12-12 15:32:10.711309989 -0500
@@ -69,7 +69,7 @@
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/layout.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/webkit_glue.h"
 #include "webkit/plugins/webplugininfo.h"
 #include "webkit/user_agent/user_agent_util.h"
diff -up chromium-23.0.1271.95/chrome/browser/ui/webui/help/help_handler.cc.system-code chromium-23.0.1271.95/chrome/browser/ui/webui/help/help_handler.cc
--- chromium-23.0.1271.95/chrome/browser/ui/webui/help/help_handler.cc.system-code	2012-12-12 15:32:18.596310181 -0500
+++ chromium-23.0.1271.95/chrome/browser/ui/webui/help/help_handler.cc	2012-12-12 15:32:28.670310436 -0500
@@ -31,7 +31,7 @@
 #include "grit/google_chrome_strings.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/webkit_glue.h"
 #include "webkit/user_agent/user_agent_util.h"
 
diff -up chromium-23.0.1271.95/chrome/common/extensions/event_filtering_info.h.system-code chromium-23.0.1271.95/chrome/common/extensions/event_filtering_info.h
--- chromium-23.0.1271.95/chrome/common/extensions/event_filtering_info.h.system-code	2012-12-12 15:42:26.328330244 -0500
+++ chromium-23.0.1271.95/chrome/common/extensions/event_filtering_info.h	2012-12-12 15:42:35.185330461 -0500
@@ -7,7 +7,7 @@
 
 #include "base/memory/scoped_ptr.h"
 #include "googleurl/src/gurl.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace base {
 class Value;
diff -up chromium-23.0.1271.95/chrome/common/metrics/metrics_log_base.cc.system-code chromium-23.0.1271.95/chrome/common/metrics/metrics_log_base.cc
--- chromium-23.0.1271.95/chrome/common/metrics/metrics_log_base.cc.system-code	2012-11-27 21:02:37.000000000 -0500
+++ chromium-23.0.1271.95/chrome/common/metrics/metrics_log_base.cc	2012-12-12 15:22:19.706293777 -0500
@@ -11,7 +11,7 @@
 #include "base/string_number_conversions.h"
 #include "base/sys_byteorder.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 #include "base/utf_string_conversions.h"
 #include "chrome/common/chrome_version_info.h"
 #include "chrome/common/logging_chrome.h"
diff -up chromium-23.0.1271.95/chrome/common/profiling.cc.system-code chromium-23.0.1271.95/chrome/common/profiling.cc
--- chromium-23.0.1271.95/chrome/common/profiling.cc.system-code	2012-11-27 21:02:43.000000000 -0500
+++ chromium-23.0.1271.95/chrome/common/profiling.cc	2012-12-12 15:22:19.708293779 -0500
@@ -13,7 +13,7 @@
 #include "base/string_util.h"
 #include "base/threading/thread.h"
 #include "chrome/common/chrome_switches.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 std::string GetProfileName() {
diff -up chromium-23.0.1271.95/chrome/common/zip.cc.system-code chromium-23.0.1271.95/chrome/common/zip.cc
--- chromium-23.0.1271.95/chrome/common/zip.cc.system-code	2012-11-27 21:02:43.000000000 -0500
+++ chromium-23.0.1271.95/chrome/common/zip.cc	2012-12-12 15:22:19.708293779 -0500
@@ -12,8 +12,8 @@
 #include "chrome/common/zip_internal.h"
 #include "chrome/common/zip_reader.h"
 #include "net/base/file_stream.h"
-#include "third_party/zlib/contrib/minizip/unzip.h"
-#include "third_party/zlib/contrib/minizip/zip.h"
+#include <minizip/unzip.h>
+#include <minizip/zip.h>
 
 namespace {
 
diff -up chromium-23.0.1271.95/chrome/common/zip_internal.cc.system-code chromium-23.0.1271.95/chrome/common/zip_internal.cc
--- chromium-23.0.1271.95/chrome/common/zip_internal.cc.system-code	2012-11-27 21:02:43.000000000 -0500
+++ chromium-23.0.1271.95/chrome/common/zip_internal.cc	2012-12-12 15:22:19.710293781 -0500
@@ -5,12 +5,12 @@
 #include "chrome/common/zip.h"
 
 #include "base/utf_string_conversions.h"
-#include "third_party/zlib/contrib/minizip/unzip.h"
-#include "third_party/zlib/contrib/minizip/zip.h"
+#include <minizip/unzip.h>
+#include <minizip/zip.h>
 #if defined(OS_WIN)
 #include "third_party/zlib/contrib/minizip/iowin32.h"
 #elif defined(OS_POSIX)
-#include "third_party/zlib/contrib/minizip/ioapi.h"
+#include <minizip/ioapi.h>
 #endif
 
 namespace {
diff -up chromium-23.0.1271.95/chrome/common/zip_internal.h.system-code chromium-23.0.1271.95/chrome/common/zip_internal.h
--- chromium-23.0.1271.95/chrome/common/zip_internal.h.system-code	2012-11-27 21:02:43.000000000 -0500
+++ chromium-23.0.1271.95/chrome/common/zip_internal.h	2012-12-12 15:22:19.711293782 -0500
@@ -7,8 +7,8 @@
 
 #include <string>
 
-#include "third_party/zlib/contrib/minizip/unzip.h"
-#include "third_party/zlib/contrib/minizip/zip.h"
+#include <minizip/unzip.h>
+#include <minizip/zip.h>
 
 // Utility functions and constants used internally for the zip file
 // library in the directory. Don't use them outside of the library.
diff -up chromium-23.0.1271.95/chrome/common/zip_reader.cc.system-code chromium-23.0.1271.95/chrome/common/zip_reader.cc
--- chromium-23.0.1271.95/chrome/common/zip_reader.cc.system-code	2012-11-27 21:02:43.000000000 -0500
+++ chromium-23.0.1271.95/chrome/common/zip_reader.cc	2012-12-12 15:22:19.711293782 -0500
@@ -10,7 +10,7 @@
 #include "base/utf_string_conversions.h"
 #include "chrome/common/zip_internal.h"
 #include "net/base/file_stream.h"
-#include "third_party/zlib/contrib/minizip/unzip.h"
+#include <minizip/unzip.h>
 #if defined(OS_WIN)
 #include "third_party/zlib/contrib/minizip/iowin32.h"
 #endif
diff -up chromium-23.0.1271.95/chrome/common/zip_reader.h.system-code chromium-23.0.1271.95/chrome/common/zip_reader.h
--- chromium-23.0.1271.95/chrome/common/zip_reader.h.system-code	2012-11-27 21:02:43.000000000 -0500
+++ chromium-23.0.1271.95/chrome/common/zip_reader.h	2012-12-12 15:22:19.713293784 -0500
@@ -12,7 +12,7 @@
 #include "base/file_util.h"
 #include "base/memory/scoped_ptr.h"
 #include "base/time.h"
-#include "third_party/zlib/contrib/minizip/unzip.h"
+#include <minizip/unzip.h>
 
 namespace zip {
 
diff -up chromium-23.0.1271.95/chrome/renderer/automation/automation_renderer_helper.cc.system-code chromium-23.0.1271.95/chrome/renderer/automation/automation_renderer_helper.cc
--- chromium-23.0.1271.95/chrome/renderer/automation/automation_renderer_helper.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/automation/automation_renderer_helper.cc	2012-12-12 15:22:19.714293785 -0500
@@ -28,7 +28,7 @@
 #include "ui/gfx/codec/png_codec.h"
 #include "ui/gfx/point.h"
 #include "ui/gfx/rect.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/webkit_glue.h"
 
 #if !defined(NO_TCMALLOC) && (defined(OS_LINUX) || defined(OS_CHROMEOS))
diff -up chromium-23.0.1271.95/chrome/renderer/benchmarking_extension.cc.system-code chromium-23.0.1271.95/chrome/renderer/benchmarking_extension.cc
--- chromium-23.0.1271.95/chrome/renderer/benchmarking_extension.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/benchmarking_extension.cc	2012-12-12 15:22:19.715293786 -0500
@@ -11,7 +11,7 @@
 #include "content/public/common/content_switches.h"
 #include "content/public/renderer/render_thread.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebCache.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebCache;
 
diff -up chromium-23.0.1271.95/chrome/renderer/chrome_render_process_observer.cc.system-code chromium-23.0.1271.95/chrome/renderer/chrome_render_process_observer.cc
--- chromium-23.0.1271.95/chrome/renderer/chrome_render_process_observer.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/chrome_render_process_observer.cc	2012-12-12 15:22:19.716293787 -0500
@@ -43,7 +43,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebRuntimeFeatures.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #if defined(OS_ANDROID)
 #include "webkit/media/android/webmediaplayer_android.h"
diff -up chromium-23.0.1271.95/chrome/renderer/chrome_render_view_observer.cc.system-code chromium-23.0.1271.95/chrome/renderer/chrome_render_view_observer.cc
--- chromium-23.0.1271.95/chrome/renderer/chrome_render_view_observer.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/chrome_render_view_observer.cc	2012-12-12 15:41:57.419329327 -0500
@@ -55,7 +55,7 @@
 #include "webkit/glue/image_decoder.h"
 #include "webkit/glue/multi_resolution_image_resource_fetcher.h"
 #include "webkit/glue/webkit_glue.h"
-#include "v8/include/v8-testing.h"
+#include <v8-testing.h>
 
 using WebKit::WebAccessibilityObject;
 using WebKit::WebCString;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/api_definitions_natives.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/api_definitions_natives.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/api_definitions_natives.h.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/api_definitions_natives.h	2012-12-12 15:37:26.870321505 -0500
@@ -8,7 +8,7 @@
 #include "chrome/renderer/extensions/chrome_v8_extension.h"
 #include "chrome/renderer/extensions/dispatcher.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/app_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/app_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/app_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/app_bindings.cc	2012-12-12 15:22:19.721293792 -0500
@@ -21,7 +21,7 @@
 #include "grit/renderer_resources.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using content::V8ValueConverter;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/app_window_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/app_window_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/app_window_custom_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/app_window_custom_bindings.cc	2012-12-12 15:41:41.030328783 -0500
@@ -15,7 +15,7 @@
 #include "content/public/renderer/render_view_visitor.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context.cc.system-code	2012-12-12 15:39:13.107324368 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context.cc	2012-12-12 15:39:21.796324566 -0500
@@ -16,7 +16,7 @@
 #include "content/public/renderer/render_view.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context.h.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context.h	2012-12-12 15:39:34.291324924 -0500
@@ -10,7 +10,7 @@
 #include "base/basictypes.h"
 #include "chrome/common/extensions/features/feature.h"
 #include "chrome/renderer/extensions/module_system.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace WebKit {
 class WebFrame;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set.cc	2012-12-12 15:22:19.730293799 -0500
@@ -14,7 +14,7 @@
 #include "content/public/renderer/render_thread.h"
 #include "content/public/renderer/v8_value_converter.h"
 #include "content/public/renderer/render_view.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set.h.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set.h	2012-12-12 15:40:58.286327431 -0500
@@ -9,7 +9,7 @@
 #include <string>
 
 #include "base/basictypes.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class GURL;
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set_unittest.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set_unittest.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set_unittest.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_context_set_unittest.cc	2012-12-12 15:40:26.042326553 -0500
@@ -9,7 +9,7 @@
 #include "chrome/renderer/extensions/chrome_v8_context_set.h"
 #include "testing/gtest/include/gtest/gtest.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_extension_handler.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_extension_handler.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_extension_handler.h.system-code	2012-12-12 15:33:55.176312823 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_extension_handler.h	2012-12-12 15:34:58.138317308 -0500
@@ -9,7 +9,7 @@
 
 #include "base/basictypes.h"
 #include "ipc/ipc_listener.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 class ChromeV8Context;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_extension.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_extension.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_extension.h.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/chrome_v8_extension.h	2012-12-12 15:40:43.136326987 -0500
@@ -10,7 +10,7 @@
 #include "base/string_piece.h"
 #include "chrome/renderer/extensions/chrome_v8_extension_handler.h"
 #include "chrome/renderer/extensions/native_handler.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <map>
 #include <set>
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/context_menus_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/context_menus_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/context_menus_custom_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/context_menus_custom_bindings.cc	2012-12-12 15:22:19.732293801 -0500
@@ -7,7 +7,7 @@
 #include "chrome/common/extensions/extension_messages.h"
 #include "content/public/renderer/render_thread.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/dispatcher.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/dispatcher.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/dispatcher.cc.system-code	2012-12-12 15:35:39.193318221 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/dispatcher.cc	2012-12-12 15:35:48.953318475 -0500
@@ -64,7 +64,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURLRequest.h"
 #include "ui/base/layout.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDataSource;
 using WebKit::WebDocument;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/dispatcher.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/dispatcher.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/dispatcher.h.system-code	2012-12-12 15:32:55.084311093 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/dispatcher.h	2012-12-12 15:33:05.024311470 -0500
@@ -19,7 +19,7 @@
 #include "chrome/renderer/extensions/chrome_v8_context_set.h"
 #include "chrome/renderer/extensions/v8_schema_registry.h"
 #include "chrome/renderer/resource_bundle_source_map.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class GURL;
 class ModuleSystem;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/event_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/event_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/event_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/event_bindings.cc	2012-12-12 15:22:19.732293801 -0500
@@ -34,7 +34,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURLRequest.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebSecurityOrigin;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/experimental.usb_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/experimental.usb_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/experimental.usb_custom_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/experimental.usb_custom_bindings.cc	2012-12-12 15:22:19.733293802 -0500
@@ -5,7 +5,7 @@
 #include "chrome/renderer/extensions/experimental.usb_custom_bindings.h"
 
 #include "base/logging.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/extension_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/extension_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/extension_custom_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/extension_custom_bindings.cc	2012-12-12 15:22:19.734293803 -0500
@@ -19,7 +19,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "webkit/glue/webkit_glue.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/i18n_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/i18n_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/i18n_custom_bindings.cc.system-code	2012-12-12 15:33:17.448311724 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/i18n_custom_bindings.cc	2012-12-12 15:33:25.880311905 -0500
@@ -8,7 +8,7 @@
 #include "chrome/common/extensions/message_bundle.h"
 #include "content/public/renderer/render_view.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/media_galleries_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/media_galleries_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/media_galleries_custom_bindings.cc.system-code	2012-12-12 15:38:26.629323212 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/media_galleries_custom_bindings.cc	2012-12-12 15:38:35.939323431 -0500
@@ -11,7 +11,7 @@
 #include "chrome/common/extensions/extension_constants.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/fileapi/file_system_util.h"
 
 namespace extensions {
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/miscellaneous_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/miscellaneous_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/miscellaneous_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/miscellaneous_bindings.cc	2012-12-12 15:22:30.816294027 -0500
@@ -21,7 +21,7 @@
 #include "content/public/renderer/render_view.h"
 #include "grit/renderer_resources.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScopedMicrotaskSuppression.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // Message passing API example (in a content script):
 // var extension =
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/module_system.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/module_system.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/module_system.h.system-code	2012-12-12 15:38:06.669322739 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/module_system.h	2012-12-12 15:38:16.475322958 -0500
@@ -9,7 +9,7 @@
 #include "base/memory/linked_ptr.h"
 #include "base/memory/scoped_ptr.h"
 #include "chrome/renderer/extensions/native_handler.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <map>
 #include <set>
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/native_handler.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/native_handler.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/native_handler.cc.system-code	2012-12-12 15:36:50.878320290 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/native_handler.cc	2012-12-12 15:36:59.709320496 -0500
@@ -7,7 +7,7 @@
 #include "base/memory/linked_ptr.h"
 #include "base/logging.h"
 #include "chrome/renderer/extensions/module_system.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/native_handler.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/native_handler.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/native_handler.h.system-code	2012-12-12 15:40:00.959325845 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/native_handler.h	2012-12-12 15:40:09.247326042 -0500
@@ -7,7 +7,7 @@
 
 #include "base/bind.h"
 #include "base/memory/linked_ptr.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <string>
 #include <vector>
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/page_actions_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/page_actions_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/page_actions_custom_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/page_actions_custom_bindings.cc	2012-12-12 15:37:40.459321876 -0500
@@ -9,7 +9,7 @@
 #include "chrome/common/extensions/extension_action.h"
 #include "chrome/renderer/extensions/dispatcher.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/page_capture_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/page_capture_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/page_capture_custom_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/page_capture_custom_bindings.cc	2012-12-12 15:22:30.818294029 -0500
@@ -9,7 +9,7 @@
 #include "content/public/renderer/render_view.h"
 #include "grit/renderer_resources.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebBlob.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/request_sender.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/request_sender.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/request_sender.h.system-code	2012-12-12 15:36:22.616319481 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/request_sender.h	2012-12-12 15:36:32.355319734 -0500
@@ -9,7 +9,7 @@
 #include <map>
 
 #include "base/memory/linked_ptr.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace base {
 class ListValue;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/runtime_custom_bindings.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/runtime_custom_bindings.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/runtime_custom_bindings.h.system-code	2012-12-12 15:33:34.693312112 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/runtime_custom_bindings.h	2012-12-12 15:33:43.279312347 -0500
@@ -7,7 +7,7 @@
 
 #include "base/compiler_specific.h"
 #include "chrome/renderer/extensions/chrome_v8_extension.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class ExtensionDispatcher;
 class ChromeV8Context;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/send_request_natives.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/send_request_natives.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/send_request_natives.h.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/send_request_natives.h	2012-12-12 15:39:00.359324091 -0500
@@ -8,7 +8,7 @@
 #include "chrome/renderer/extensions/chrome_v8_extension.h"
 #include "chrome/renderer/extensions/dispatcher.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 class RequestSender;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/set_icon_natives.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/set_icon_natives.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/set_icon_natives.h.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/set_icon_natives.h	2012-12-12 15:37:57.301322363 -0500
@@ -8,7 +8,7 @@
 #include "chrome/renderer/extensions/chrome_v8_extension.h"
 #include "chrome/renderer/extensions/dispatcher.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace base {
 class Value;
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/tabs_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/tabs_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/tabs_custom_bindings.cc.system-code	2012-12-12 15:36:02.102318901 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/tabs_custom_bindings.cc	2012-12-12 15:36:11.920319131 -0500
@@ -10,7 +10,7 @@
 #include "chrome/common/extensions/extension_messages.h"
 #include "content/public/renderer/render_view.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/tts_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/tts_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/tts_custom_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/tts_custom_bindings.cc	2012-12-12 15:39:48.380325356 -0500
@@ -8,7 +8,7 @@
 
 #include "chrome/common/extensions/extension_action.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/user_script_scheduler.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/user_script_scheduler.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/user_script_scheduler.cc.system-code	2012-12-12 15:41:25.819328364 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/user_script_scheduler.cc	2012-12-12 15:41:11.042327845 -0500
@@ -20,7 +20,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 // The length of time to wait after the DOM is complete to try and run user
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/v8_schema_registry.h.system-code chromium-23.0.1271.95/chrome/renderer/extensions/v8_schema_registry.h
--- chromium-23.0.1271.95/chrome/renderer/extensions/v8_schema_registry.h.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/v8_schema_registry.h	2012-12-12 15:22:30.823294034 -0500
@@ -10,7 +10,7 @@
 #include <string>
 
 #include "base/basictypes.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/web_request_custom_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/web_request_custom_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/web_request_custom_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/web_request_custom_bindings.cc	2012-12-12 15:22:30.824294035 -0500
@@ -8,7 +8,7 @@
 
 #include "base/string_number_conversions.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
diff -up chromium-23.0.1271.95/chrome/renderer/extensions/webstore_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/extensions/webstore_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/extensions/webstore_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/extensions/webstore_bindings.cc	2012-12-12 15:22:30.825294036 -0500
@@ -15,7 +15,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebElement.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebNode.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebNodeList.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDocument;
 using WebKit::WebElement;
diff -up chromium-23.0.1271.95/chrome/renderer/external_extension.cc.system-code chromium-23.0.1271.95/chrome/renderer/external_extension.cc
--- chromium-23.0.1271.95/chrome/renderer/external_extension.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/external_extension.cc	2012-12-12 15:22:30.827294037 -0500
@@ -12,7 +12,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebView;
diff -up chromium-23.0.1271.95/chrome/renderer/loadtimes_extension_bindings.cc.system-code chromium-23.0.1271.95/chrome/renderer/loadtimes_extension_bindings.cc
--- chromium-23.0.1271.95/chrome/renderer/loadtimes_extension_bindings.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/loadtimes_extension_bindings.cc	2012-12-12 15:22:30.828294038 -0500
@@ -9,7 +9,7 @@
 #include "base/time.h"
 #include "content/public/renderer/document_state.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDataSource;
 using WebKit::WebFrame;
diff -up chromium-23.0.1271.95/chrome/renderer/playback_extension.cc.system-code chromium-23.0.1271.95/chrome/renderer/playback_extension.cc
--- chromium-23.0.1271.95/chrome/renderer/playback_extension.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/playback_extension.cc	2012-12-12 15:22:34.759294115 -0500
@@ -4,7 +4,7 @@
 
 #include "chrome/renderer/playback_extension.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 const char kPlaybackExtensionName[] = "v8/PlaybackMode";
 
diff -up chromium-23.0.1271.95/chrome/renderer/resource_bundle_source_map.h.system-code chromium-23.0.1271.95/chrome/renderer/resource_bundle_source_map.h
--- chromium-23.0.1271.95/chrome/renderer/resource_bundle_source_map.h.system-code	2012-12-12 15:32:37.603310660 -0500
+++ chromium-23.0.1271.95/chrome/renderer/resource_bundle_source_map.h	2012-12-12 15:32:46.463310896 -0500
@@ -10,7 +10,7 @@
 #include "base/string_piece.h"
 #include "chrome/renderer/extensions/module_system.h"
 #include "chrome/renderer/static_v8_external_string_resource.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <map>
 #include <string>
diff -up chromium-23.0.1271.95/chrome/renderer/searchbox/searchbox_extension.cc.system-code chromium-23.0.1271.95/chrome/renderer/searchbox/searchbox_extension.cc
--- chromium-23.0.1271.95/chrome/renderer/searchbox/searchbox_extension.cc.system-code	2012-12-12 15:42:08.047329764 -0500
+++ chromium-23.0.1271.95/chrome/renderer/searchbox/searchbox_extension.cc	2012-12-12 15:42:16.597329980 -0500
@@ -16,7 +16,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptSource.h"
 #include "ui/base/keycodes/keyboard_codes.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 
diff -up chromium-23.0.1271.95/chrome/renderer/static_v8_external_string_resource.h.system-code chromium-23.0.1271.95/chrome/renderer/static_v8_external_string_resource.h
--- chromium-23.0.1271.95/chrome/renderer/static_v8_external_string_resource.h.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/static_v8_external_string_resource.h	2012-12-12 15:22:36.549294162 -0500
@@ -7,7 +7,7 @@
 
 #include "base/compiler_specific.h"
 #include "base/string_piece.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // A very simple implementation of v8::ExternalAsciiStringResource that just
 // wraps a buffer. The buffer must outlive the v8 runtime instance this resource
diff -up chromium-23.0.1271.95/chrome/renderer/translate_helper.cc.system-code chromium-23.0.1271.95/chrome/renderer/translate_helper.cc
--- chromium-23.0.1271.95/chrome/renderer/translate_helper.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/renderer/translate_helper.cc	2012-12-12 15:22:36.551294164 -0500
@@ -19,7 +19,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptSource.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "third_party/cld/encodings/compact_lang_det/win/cld_unicodetext.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/dom_operations.h"
 
 using WebKit::WebDocument;
diff -up chromium-23.0.1271.95/chrome/test/base/module_system_test.h.system-code chromium-23.0.1271.95/chrome/test/base/module_system_test.h
--- chromium-23.0.1271.95/chrome/test/base/module_system_test.h.system-code	2012-11-27 21:02:36.000000000 -0500
+++ chromium-23.0.1271.95/chrome/test/base/module_system_test.h	2012-12-12 15:26:27.654297066 -0500
@@ -6,7 +6,7 @@
 #define CHROME_TEST_BASE_MODULE_SYSTEM_TEST_H_
 
 #include "chrome/renderer/extensions/module_system.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "testing/gtest/include/gtest/gtest.h"
 
 class AssertNatives;
diff -up chromium-23.0.1271.95/chrome/test/base/v8_unit_test.h.system-code chromium-23.0.1271.95/chrome/test/base/v8_unit_test.h
--- chromium-23.0.1271.95/chrome/test/base/v8_unit_test.h.system-code	2012-11-27 21:02:36.000000000 -0500
+++ chromium-23.0.1271.95/chrome/test/base/v8_unit_test.h	2012-12-12 15:22:36.554294167 -0500
@@ -11,7 +11,7 @@
 #include "base/file_path.h"
 #include "base/string_piece.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // A superclass for unit tests that involve running JavaScript.  This class
 // sets up V8 context and has methods that make it easy to execute scripts in
diff -up chromium-23.0.1271.95/chrome/test/reliability/page_load_test.cc.system-code chromium-23.0.1271.95/chrome/test/reliability/page_load_test.cc
--- chromium-23.0.1271.95/chrome/test/reliability/page_load_test.cc.system-code	2012-11-27 21:02:17.000000000 -0500
+++ chromium-23.0.1271.95/chrome/test/reliability/page_load_test.cc	2012-12-12 15:22:36.557294170 -0500
@@ -68,7 +68,7 @@
 #include "chrome/test/ui/ui_test.h"
 #include "net/base/net_util.h"
 #include "ui/base/keycodes/keyboard_codes.h"
-#include "v8/include/v8-testing.h"
+#include <v8-testing.h>
 
 namespace {
 
diff -up chromium-23.0.1271.95/content/public/renderer/content_renderer_client.h.system-code chromium-23.0.1271.95/content/public/renderer/content_renderer_client.h
--- chromium-23.0.1271.95/content/public/renderer/content_renderer_client.h.system-code	2012-11-27 21:01:27.000000000 -0500
+++ chromium-23.0.1271.95/content/public/renderer/content_renderer_client.h	2012-12-12 15:26:58.926298002 -0500
@@ -13,7 +13,7 @@
 #include "content/public/common/content_client.h"
 #include "content/public/common/page_transition_types.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebPageVisibilityState.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class FilePath;
 class GURL;
diff -up chromium-23.0.1271.95/content/public/renderer/v8_value_converter.h.system-code chromium-23.0.1271.95/content/public/renderer/v8_value_converter.h
--- chromium-23.0.1271.95/content/public/renderer/v8_value_converter.h.system-code	2012-11-27 21:01:27.000000000 -0500
+++ chromium-23.0.1271.95/content/public/renderer/v8_value_converter.h	2012-12-12 15:22:36.559294172 -0500
@@ -6,7 +6,7 @@
 #define CONTENT_PUBLIC_RENDERER_V8_VALUE_CONVERTER_H_
 
 #include "content/common/content_export.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace base {
 class Value;
diff -up chromium-23.0.1271.95/content/renderer/browser_plugin/browser_plugin_bindings.cc.system-code chromium-23.0.1271.95/content/renderer/browser_plugin/browser_plugin_bindings.cc
--- chromium-23.0.1271.95/content/renderer/browser_plugin/browser_plugin_bindings.cc.system-code	2012-12-12 15:43:03.455331408 -0500
+++ chromium-23.0.1271.95/content/renderer/browser_plugin/browser_plugin_bindings.cc	2012-12-12 15:43:13.656331667 -0500
@@ -24,7 +24,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebPluginContainer.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebSerializedScriptValue.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebBindings;
 using WebKit::WebElement;
diff -up chromium-23.0.1271.95/content/renderer/do_not_track_bindings.cc.system-code chromium-23.0.1271.95/content/renderer/do_not_track_bindings.cc
--- chromium-23.0.1271.95/content/renderer/do_not_track_bindings.cc.system-code	2012-12-12 15:43:22.516331924 -0500
+++ chromium-23.0.1271.95/content/renderer/do_not_track_bindings.cc	2012-12-12 15:43:30.149332113 -0500
@@ -6,7 +6,7 @@
 
 #include "content/renderer/render_view_impl.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 
diff -up chromium-23.0.1271.95/content/renderer/gpu/gpu_benchmarking_extension.cc.system-code chromium-23.0.1271.95/content/renderer/gpu/gpu_benchmarking_extension.cc
--- chromium-23.0.1271.95/content/renderer/gpu/gpu_benchmarking_extension.cc.system-code	2012-11-27 21:01:26.000000000 -0500
+++ chromium-23.0.1271.95/content/renderer/gpu/gpu_benchmarking_extension.cc	2012-12-12 15:27:22.843298852 -0500
@@ -22,7 +22,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebViewBenchmarkSupport.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebCanvas;
 using WebKit::WebFrame;
diff -up chromium-23.0.1271.95/content/renderer/render_thread_impl.cc.system-code chromium-23.0.1271.95/content/renderer/render_thread_impl.cc
--- chromium-23.0.1271.95/content/renderer/render_thread_impl.cc.system-code	2012-11-27 21:01:26.000000000 -0500
+++ chromium-23.0.1271.95/content/renderer/render_thread_impl.cc	2012-12-12 15:27:42.102299425 -0500
@@ -94,7 +94,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "ui/base/layout.h"
 #include "ui/base/ui_base_switches.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/webkit_glue.h"
 
 // TODO(port)
diff -up chromium-23.0.1271.95/content/renderer/render_view_impl.cc.system-code chromium-23.0.1271.95/content/renderer/render_view_impl.cc
--- chromium-23.0.1271.95/content/renderer/render_view_impl.cc.system-code	2012-11-27 21:01:26.000000000 -0500
+++ chromium-23.0.1271.95/content/renderer/render_view_impl.cc	2012-12-12 15:22:36.571294184 -0500
@@ -177,7 +177,7 @@
 #include "ui/gfx/native_widget_types.h"
 #include "ui/gfx/point.h"
 #include "ui/gfx/rect.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/appcache/web_application_cache_host_impl.h"
 #include "webkit/dom_storage/dom_storage_types.h"
 #include "webkit/forms/form_data.h"
diff -up chromium-23.0.1271.95/content/renderer/v8_value_converter_impl.cc.system-code chromium-23.0.1271.95/content/renderer/v8_value_converter_impl.cc
--- chromium-23.0.1271.95/content/renderer/v8_value_converter_impl.cc.system-code	2012-11-27 21:01:26.000000000 -0500
+++ chromium-23.0.1271.95/content/renderer/v8_value_converter_impl.cc	2012-12-12 15:22:36.572294185 -0500
@@ -11,7 +11,7 @@
 #include "base/values.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebArrayBuffer.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebArrayBufferView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using base::BinaryValue;
 using base::DictionaryValue;
diff -up chromium-23.0.1271.95/content/renderer/v8_value_converter_impl_unittest.cc.system-code chromium-23.0.1271.95/content/renderer/v8_value_converter_impl_unittest.cc
--- chromium-23.0.1271.95/content/renderer/v8_value_converter_impl_unittest.cc.system-code	2012-11-27 21:01:26.000000000 -0500
+++ chromium-23.0.1271.95/content/renderer/v8_value_converter_impl_unittest.cc	2012-12-12 15:28:01.096299949 -0500
@@ -8,7 +8,7 @@
 #include "base/values.h"
 #include "content/renderer/v8_value_converter_impl.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using content::V8ValueConverter;
 
diff -up chromium-23.0.1271.95/content/renderer/web_intents_host.cc.system-code chromium-23.0.1271.95/content/renderer/web_intents_host.cc
--- chromium-23.0.1271.95/content/renderer/web_intents_host.cc.system-code	2012-11-27 21:01:26.000000000 -0500
+++ chromium-23.0.1271.95/content/renderer/web_intents_host.cc	2012-12-12 15:28:21.439300335 -0500
@@ -23,7 +23,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebSerializedScriptValue.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebVector.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/fileapi/file_system_util.h"
 #include "webkit/glue/cpp_bound_class.h"
 
diff -up chromium-23.0.1271.95/content/shell/shell_content_renderer_client.cc.system-code chromium-23.0.1271.95/content/shell/shell_content_renderer_client.cc
--- chromium-23.0.1271.95/content/shell/shell_content_renderer_client.cc.system-code	2012-11-27 21:01:26.000000000 -0500
+++ chromium-23.0.1271.95/content/shell/shell_content_renderer_client.cc	2012-12-12 15:28:43.632300939 -0500
@@ -8,7 +8,7 @@
 #include "content/shell/shell_render_process_observer.h"
 #include "content/shell/shell_switches.h"
 #include "content/shell/webkit_test_runner.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace content {
 
diff -up chromium-23.0.1271.95/content/shell/webkit_test_runner_bindings.h.system-code chromium-23.0.1271.95/content/shell/webkit_test_runner_bindings.h
--- chromium-23.0.1271.95/content/shell/webkit_test_runner_bindings.h.system-code	2012-12-12 15:42:46.412330815 -0500
+++ chromium-23.0.1271.95/content/shell/webkit_test_runner_bindings.h	2012-12-12 15:42:54.276331021 -0500
@@ -7,7 +7,7 @@
 
 #include "base/basictypes.h"
 #include "base/compiler_specific.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace content {
 
diff -up chromium-23.0.1271.95/net/proxy/proxy_resolver_v8.cc.system-code chromium-23.0.1271.95/net/proxy/proxy_resolver_v8.cc
--- chromium-23.0.1271.95/net/proxy/proxy_resolver_v8.cc.system-code	2012-11-27 21:01:30.000000000 -0500
+++ chromium-23.0.1271.95/net/proxy/proxy_resolver_v8.cc	2012-12-12 15:22:37.197294211 -0500
@@ -24,7 +24,7 @@
 #include "net/proxy/proxy_resolver_js_bindings.h"
 #include "net/proxy/proxy_resolver_request_context.h"
 #include "net/proxy/proxy_resolver_script.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // Notes on the javascript environment:
 //
diff -up chromium-23.0.1271.95/third_party/v8-i18n/include/extension.h.system-code chromium-23.0.1271.95/third_party/v8-i18n/include/extension.h
--- chromium-23.0.1271.95/third_party/v8-i18n/include/extension.h.system-code	2012-11-27 21:03:52.000000000 -0500
+++ chromium-23.0.1271.95/third_party/v8-i18n/include/extension.h	2012-12-12 15:22:37.197294211 -0500
@@ -15,7 +15,7 @@
 #ifndef V8_I18N_INCLUDE_EXTENSION_H_
 #define V8_I18N_INCLUDE_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace v8_i18n {
 
diff -up chromium-23.0.1271.95/third_party/v8-i18n/src/break-iterator.h.system-code chromium-23.0.1271.95/third_party/v8-i18n/src/break-iterator.h
--- chromium-23.0.1271.95/third_party/v8-i18n/src/break-iterator.h.system-code	2012-11-27 21:03:52.000000000 -0500
+++ chromium-23.0.1271.95/third_party/v8-i18n/src/break-iterator.h	2012-12-12 15:22:37.198294212 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_BREAK_ITERATOR_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class BreakIterator;
diff -up chromium-23.0.1271.95/third_party/v8-i18n/src/collator.h.system-code chromium-23.0.1271.95/third_party/v8-i18n/src/collator.h
--- chromium-23.0.1271.95/third_party/v8-i18n/src/collator.h.system-code	2012-11-27 21:03:52.000000000 -0500
+++ chromium-23.0.1271.95/third_party/v8-i18n/src/collator.h	2012-12-12 15:22:37.198294212 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_COLLATOR_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class Collator;
diff -up chromium-23.0.1271.95/third_party/v8-i18n/src/date-format.h.system-code chromium-23.0.1271.95/third_party/v8-i18n/src/date-format.h
--- chromium-23.0.1271.95/third_party/v8-i18n/src/date-format.h.system-code	2012-12-12 15:44:26.860333873 -0500
+++ chromium-23.0.1271.95/third_party/v8-i18n/src/date-format.h	2012-12-12 15:44:37.598334092 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_DATE_FORMAT_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class SimpleDateFormat;
diff -up chromium-23.0.1271.95/third_party/v8-i18n/src/locale.h.system-code chromium-23.0.1271.95/third_party/v8-i18n/src/locale.h
--- chromium-23.0.1271.95/third_party/v8-i18n/src/locale.h.system-code	2012-11-27 21:03:52.000000000 -0500
+++ chromium-23.0.1271.95/third_party/v8-i18n/src/locale.h	2012-12-12 15:29:39.518302050 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_LOCALE_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace v8_i18n {
 
diff -up chromium-23.0.1271.95/third_party/v8-i18n/src/number-format.h.system-code chromium-23.0.1271.95/third_party/v8-i18n/src/number-format.h
--- chromium-23.0.1271.95/third_party/v8-i18n/src/number-format.h.system-code	2012-11-27 21:03:52.000000000 -0500
+++ chromium-23.0.1271.95/third_party/v8-i18n/src/number-format.h	2012-12-12 15:22:41.924294331 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_NUMBER_FORMAT_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class DecimalFormat;
diff -up chromium-23.0.1271.95/third_party/v8-i18n/src/utils.h.system-code chromium-23.0.1271.95/third_party/v8-i18n/src/utils.h
--- chromium-23.0.1271.95/third_party/v8-i18n/src/utils.h.system-code	2012-11-27 21:03:52.000000000 -0500
+++ chromium-23.0.1271.95/third_party/v8-i18n/src/utils.h	2012-12-12 15:22:41.924294331 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_UTILS_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class UnicodeString;
diff -up chromium-23.0.1271.95/third_party/v8-i18n/tests/test-runner.cc.system-code chromium-23.0.1271.95/third_party/v8-i18n/tests/test-runner.cc
--- chromium-23.0.1271.95/third_party/v8-i18n/tests/test-runner.cc.system-code	2012-11-27 21:03:52.000000000 -0500
+++ chromium-23.0.1271.95/third_party/v8-i18n/tests/test-runner.cc	2012-12-12 15:22:41.925294332 -0500
@@ -22,7 +22,7 @@
 
 #include "include/extension.h"
 #include "unicode/locid.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 v8::Persistent<v8::Context> CreateContext();
 bool ExecuteString(v8::Handle<v8::String> source, v8::Handle<v8::Value> name);
diff -up chromium-23.0.1271.95/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp.system-code chromium-23.0.1271.95/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp
--- chromium-23.0.1271.95/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp.system-code	2012-11-27 21:04:41.000000000 -0500
+++ chromium-23.0.1271.95/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp	2012-12-12 15:22:41.927294334 -0500
@@ -28,7 +28,7 @@
 
 #include "WebCoreTestSupport.h"
 #include "WebFrame.h"
-#include <v8/include/v8.h>
+#include <v8.h>
 
 namespace WebKit {
 
diff -up chromium-23.0.1271.95/third_party/WebKit/Source/WebKit/chromium/tests/ListenerLeakTest.cpp.system-code chromium-23.0.1271.95/third_party/WebKit/Source/WebKit/chromium/tests/ListenerLeakTest.cpp
--- chromium-23.0.1271.95/third_party/WebKit/Source/WebKit/chromium/tests/ListenerLeakTest.cpp.system-code	2012-11-27 21:04:51.000000000 -0500
+++ chromium-23.0.1271.95/third_party/WebKit/Source/WebKit/chromium/tests/ListenerLeakTest.cpp	2012-12-12 15:30:00.610302641 -0500
@@ -34,8 +34,8 @@
 #include "URLTestHelpers.h"
 #include "WebView.h"
 #include <gtest/gtest.h>
-#include <v8/include/v8-profiler.h>
-#include <v8/include/v8.h>
+#include <v8-profiler.h>
+#include <v8.h>
 #include <webkit/support/webkit_support.h>
 
 using namespace WebKit;
diff -up chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/DRTTestRunner.cpp.system-code chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/DRTTestRunner.cpp
--- chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/DRTTestRunner.cpp.system-code	2012-12-12 15:43:46.768332496 -0500
+++ chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/DRTTestRunner.cpp	2012-12-12 15:43:59.666332803 -0500
@@ -65,7 +65,7 @@
 #include "platform/WebSerializedScriptValue.h"
 #include "platform/WebSize.h"
 #include "platform/WebURL.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/support/webkit_support.h"
 #include <algorithm>
 #include <cctype>
diff -up chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp.system-code chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp
--- chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp.system-code	2012-11-27 21:03:06.000000000 -0500
+++ chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp	2012-12-12 15:30:19.151303139 -0500
@@ -35,8 +35,8 @@
 #include "TestShell.h"
 #include "webkit/support/webkit_support.h"
 #include <public/WebCompositorSupport.h>
-#include <v8/include/v8-testing.h>
-#include <v8/include/v8.h>
+#include <v8-testing.h>
+#include <v8.h>
 #include <wtf/OwnPtr.h>
 #include <wtf/Vector.h>
 
diff -up chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/TestRunner/TestRunner.cpp.system-code chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/TestRunner/TestRunner.cpp
--- chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/TestRunner/TestRunner.cpp.system-code	2012-12-12 15:44:10.593333402 -0500
+++ chromium-23.0.1271.95/third_party/WebKit/Tools/DumpRenderTree/chromium/TestRunner/TestRunner.cpp	2012-12-12 15:44:19.095333624 -0500
@@ -59,7 +59,7 @@
 #include "platform/WebSerializedScriptValue.h"
 #include "platform/WebSize.h"
 #include "platform/WebURL.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/support/webkit_support.h"
 #include <algorithm>
 #include <clocale>
diff -up chromium-23.0.1271.95/webkit/glue/webkit_glue.cc.system-code chromium-23.0.1271.95/webkit/glue/webkit_glue.cc
--- chromium-23.0.1271.95/webkit/glue/webkit_glue.cc.system-code	2012-11-27 21:01:16.000000000 -0500
+++ chromium-23.0.1271.95/webkit/glue/webkit_glue.cc	2012-12-12 15:30:39.382303668 -0500
@@ -49,7 +49,7 @@
 #if defined(OS_WIN)
 #include "third_party/WebKit/Source/WebKit/chromium/public/win/WebInputEventFactory.h"
 #endif
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/glue_serialize.h"
 
 using WebKit::WebCanvas;
diff -up chromium-23.0.1271.95/webkit/glue/webkitplatformsupport_impl.cc.system-code chromium-23.0.1271.95/webkit/glue/webkitplatformsupport_impl.cc
--- chromium-23.0.1271.95/webkit/glue/webkitplatformsupport_impl.cc.system-code	2012-11-27 21:01:16.000000000 -0500
+++ chromium-23.0.1271.95/webkit/glue/webkitplatformsupport_impl.cc	2012-12-12 15:30:57.000304270 -0500
@@ -56,7 +56,7 @@
 #endif
 
 #if defined(OS_LINUX)
-#include "v8/include/v8.h"
+#include <v8.h>
 #endif
 
 using WebKit::WebAudioBus;
diff -up chromium-23.0.1271.95/webkit/media/webmediaplayer_impl.cc.system-code chromium-23.0.1271.95/webkit/media/webmediaplayer_impl.cc
--- chromium-23.0.1271.95/webkit/media/webmediaplayer_impl.cc.system-code	2012-11-27 21:01:15.000000000 -0500
+++ chromium-23.0.1271.95/webkit/media/webmediaplayer_impl.cc	2012-12-12 15:31:15.932304968 -0500
@@ -32,7 +32,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebSize.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/media/buffered_data_source.h"
 #include "webkit/media/filter_helpers.h"
 #include "webkit/media/webmediaplayer_delegate.h"
diff -up chromium-23.0.1271.95/webkit/plugins/ppapi/message_channel.cc.system-code chromium-23.0.1271.95/webkit/plugins/ppapi/message_channel.cc
--- chromium-23.0.1271.95/webkit/plugins/ppapi/message_channel.cc.system-code	2012-11-27 21:01:15.000000000 -0500
+++ chromium-23.0.1271.95/webkit/plugins/ppapi/message_channel.cc	2012-12-12 15:22:44.049294394 -0500
@@ -21,7 +21,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebNode.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebPluginContainer.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebSerializedScriptValue.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/plugins/ppapi/host_array_buffer_var.h"
 #include "webkit/plugins/ppapi/npapi_glue.h"
 #include "webkit/plugins/ppapi/ppapi_plugin_instance.h"
diff -up chromium-23.0.1271.95/webkit/support/gc_extension.cc.system-code chromium-23.0.1271.95/webkit/support/gc_extension.cc
--- chromium-23.0.1271.95/webkit/support/gc_extension.cc.system-code	2012-11-27 21:01:17.000000000 -0500
+++ chromium-23.0.1271.95/webkit/support/gc_extension.cc	2012-12-12 15:22:44.051294396 -0500
@@ -4,7 +4,7 @@
 
 #include "webkit/support/gc_extension.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 const char kGCExtensionName[] = "v8/GCController";
 
diff -up chromium-23.0.1271.95/webkit/support/test_webkit_platform_support.cc.system-code chromium-23.0.1271.95/webkit/support/test_webkit_platform_support.cc
--- chromium-23.0.1271.95/webkit/support/test_webkit_platform_support.cc.system-code	2012-11-27 21:01:17.000000000 -0500
+++ chromium-23.0.1271.95/webkit/support/test_webkit_platform_support.cc	2012-12-12 15:22:44.055294400 -0500
@@ -29,7 +29,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebStorageNamespace.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/appcache/web_application_cache_host_impl.h"
 #include "webkit/database/vfs_backend.h"
 #include "webkit/glue/simple_webmimeregistry_impl.h"
diff -up chromium-23.0.1271.95/webkit/tools/test_shell/test_shell_webkit_init.cc.system-code chromium-23.0.1271.95/webkit/tools/test_shell/test_shell_webkit_init.cc
--- chromium-23.0.1271.95/webkit/tools/test_shell/test_shell_webkit_init.cc.system-code	2012-11-27 21:01:16.000000000 -0500
+++ chromium-23.0.1271.95/webkit/tools/test_shell/test_shell_webkit_init.cc	2012-12-12 15:22:44.057294402 -0500
@@ -14,7 +14,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptController.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebSecurityPolicy.h"
 #include "ui/gl/gl_bindings_skia_in_process.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/plugins/npapi/plugin_list.h"
 #include "webkit/plugins/webplugininfo.h"
 #include "webkit/tools/test_shell/simple_socket_stream_bridge.h"
