diff -up chromium-23.0.1271.95/net/spdy/spdy_framer.cc.de08ebc chromium-23.0.1271.95/net/spdy/spdy_framer.cc
--- chromium-23.0.1271.95/net/spdy/spdy_framer.cc.de08ebc	2012-12-13 16:20:50.007266598 -0500
+++ chromium-23.0.1271.95/net/spdy/spdy_framer.cc	2012-12-13 16:23:10.417269086 -0500
@@ -661,6 +661,10 @@ void SpdyFramer::WriteHeaderBlock(SpdyFr
   }
 }
 
+// TODO(phajdan.jr): Clean up after we no longer need
+// to workaround http://crbug.com/139744.
+#if !defined(USE_SYSTEM_ZLIB)
+
 // These constants are used by zlib to differentiate between normal data and
 // cookie data. Cookie data is handled specially by zlib when compressing.
 enum ZDataClass {
@@ -820,6 +824,8 @@ void SpdyFramer::WriteHeaderBlockToZ(con
   z->clas = kZStandardData;
 }
 
+#endif  // !defined(USE_SYSTEM_ZLIB)
+
 size_t SpdyFramer::ProcessControlFrameBeforeHeaderBlock(const char* data,
                                                         size_t len) {
   DCHECK_EQ(SPDY_CONTROL_FRAME_BEFORE_HEADER_BLOCK, state_);
@@ -1476,7 +1482,14 @@ SpdyDataFrame* SpdyFramer::CreateDataFra
 // The following compression setting are based on Brian Olson's analysis. See
 // https://groups.google.com/group/spdy-dev/browse_thread/thread/dfaf498542fac792
 // for more details.
+#if defined(USE_SYSTEM_ZLIB)
+// System zlib is not expected to have workaround for http://crbug.com/139744,
+// so disable compression in that case.
+// TODO(phajdan.jr): Remove the special case when it's no longer necessary.
+static const int kCompressorLevel = 0;
+#else  // !defined(USE_SYSTEM_ZLIB)
 static const int kCompressorLevel = 9;
+#endif // !defined(USE_SYSTEM_ZLIB)
 static const int kCompressorWindowSizeInBits = 11;
 static const int kCompressorMemLevel = 1;
 
@@ -1625,10 +1638,27 @@ SpdyControlFrame* SpdyFramer::CompressCo
   memcpy(new_frame->data(), frame.data(),
          frame.length() + SpdyFrame::kHeaderSize);
 
+  // TODO(phajdan.jr): Clean up after we no longer need
+  // to workaround http://crbug.com/139744.
+#if defined(USE_SYSTEM_ZLIB)
+  compressor->next_in = reinterpret_cast<Bytef*>(const_cast<char*>(payload));
+  compressor->avail_in = payload_length;
+#endif  // defined(USE_SYSTEM_ZLIB)
   compressor->next_out = reinterpret_cast<Bytef*>(new_frame->data()) +
                           header_length;
   compressor->avail_out = compressed_max_size;
+  // TODO(phajdan.jr): Clean up after we no longer need
+  // to workaround http://crbug.com/139744.
+#if defined(USE_SYSTEM_ZLIB)
+  int rv = deflate(compressor, Z_SYNC_FLUSH);
+  if (rv != Z_OK) {  // How can we know that it compressed everything?
+    // This shouldn't happen, right?
+    LOG(WARNING) << "deflate failure: " << rv;
+    return NULL;
+  }
+#else  // !defined(USE_SYSTEM_ZLIB)
   WriteHeaderBlockToZ(headers, compressor);
+#endif  // !defined(USE_SYSTEM_ZLIB)
   int compressed_size = compressed_max_size - compressor->avail_out;
 
   // We trust zlib. Also, we can't do anything about it.
diff -up chromium-23.0.1271.95/net/spdy/spdy_framer_test.cc.de08ebc chromium-23.0.1271.95/net/spdy/spdy_framer_test.cc
--- chromium-23.0.1271.95/net/spdy/spdy_framer_test.cc.de08ebc	2012-12-13 16:23:26.420269864 -0500
+++ chromium-23.0.1271.95/net/spdy/spdy_framer_test.cc	2012-12-13 16:24:41.599271166 -0500
@@ -1669,6 +1669,9 @@ TEST_P(SpdyFramerTest, CreateSynStreamUn
   }
 }
 
+// TODO(phajdan.jr): Clean up after we no longer need
+// to workaround http://crbug.com/139744.
+#if !defined(USE_SYSTEM_ZLIB)
 TEST_P(SpdyFramerTest, CreateSynStreamCompressed) {
   SpdyFramer framer(spdy_version_);
   framer.set_enable_compression(true);
@@ -1732,6 +1735,7 @@ TEST_P(SpdyFramerTest, CreateSynStreamCo
                  IsSpdy2() ? arraysize(kV2FrameData) : arraysize(kV3FrameData));
   }
 }
+#endif  // !defined(USE_SYSTEM_ZLIB)
 
 TEST_P(SpdyFramerTest, CreateSynReplyUncompressed) {
   SpdyFramer framer(spdy_version_);
@@ -1857,6 +1861,9 @@ TEST_P(SpdyFramerTest, CreateSynReplyUnc
   }
 }
 
+// TODO(phajdan.jr): Clean up after we no longer need
+// to workaround http://crbug.com/139744.
+#if !defined(USE_SYSTEM_ZLIB)
 TEST_P(SpdyFramerTest, CreateSynReplyCompressed) {
   SpdyFramer framer(spdy_version_);
   framer.set_enable_compression(true);
@@ -1910,6 +1917,7 @@ TEST_P(SpdyFramerTest, CreateSynReplyCom
                  IsSpdy2() ? arraysize(kV2FrameData) : arraysize(kV3FrameData));
   }
 }
+#endif  // !defined(USE_SYSTEM_ZLIB)
 
 TEST_P(SpdyFramerTest, CreateRstStream) {
   SpdyFramer framer(spdy_version_);
@@ -2249,6 +2257,9 @@ TEST_P(SpdyFramerTest, CreateHeadersUnco
   }
 }
 
+// TODO(phajdan.jr): Clean up after we no longer need
+// to workaround http://crbug.com/139744.
+#if !defined(USE_SYSTEM_ZLIB)
 TEST_P(SpdyFramerTest, CreateHeadersCompressed) {
   SpdyFramer framer(spdy_version_);
   framer.set_enable_compression(true);
@@ -2302,6 +2313,7 @@ TEST_P(SpdyFramerTest, CreateHeadersComp
                  IsSpdy2() ? arraysize(kV2FrameData) : arraysize(kV3FrameData));
   }
 }
+#endif  // !defined(USE_SYSTEM_ZLIB)
 
 TEST_P(SpdyFramerTest, CreateWindowUpdate) {
   SpdyFramer framer(spdy_version_);
