diff -up chromium-14.0.835.186/base/base.gypi.nss chromium-14.0.835.186/base/base.gypi
--- chromium-14.0.835.186/base/base.gypi.nss	2011-09-27 16:08:00.709014665 -0400
+++ chromium-14.0.835.186/base/base.gypi	2011-09-27 16:08:35.931004606 -0400
@@ -480,6 +480,7 @@
             '../build/util/build_util.gyp:lastchange',
             '../build/linux/system.gyp:gtk',
             '../build/linux/system.gyp:x11',
+            '../build/linux/system.gyp:ssl',
             'xdg_mime',
           ],
           'defines': [
@@ -491,6 +492,7 @@
           'export_dependent_settings': [
             '../build/linux/system.gyp:gtk',
             '../build/linux/system.gyp:x11',
+            '../build/linux/system.gyp:ssl',
           ],
         }, {  # toolkit_uses_gtk!=1
             'sources/': [
