--- octave-3.6.4-rc0/scripts/pkg/pkg.m.deps	2012-12-21 20:44:11.309702885 -0700
+++ octave-3.6.4-rc0/scripts/pkg/pkg.m	2012-12-21 20:51:19.529914112 -0700
@@ -633,12 +633,12 @@
   endif
   files(1) = [];
   buildlist = fullfile (builddir, "octave_packages");
-  install (files, handle_deps, autoload, installdir, installdir, verbose,
+  install (files, false, autoload, installdir, installdir, verbose,
            buildlist, "", false);
   unwind_protect
     repackage (builddir, buildlist);
   unwind_protect_cleanup
-    unload_packages ({"all"}, handle_deps, buildlist, "");
+    unload_packages ({"all"}, false, buildlist, "");
     if (exist (installdir, "dir"))
       rm_rf (installdir);
     endif
