diff -ruN Canna37p1.orig/Canna.conf Canna37p1/Canna.conf
--- Canna37p1.orig/Canna.conf	2004-01-02 21:26:27.000000000 +0900
+++ Canna37p1/Canna.conf	2004-04-30 17:45:30.268087408 +0900
@@ -109,12 +109,12 @@
 # define DefErrDir $(cannaPrefix)/log
 # define DefLibCannaDir $(cannaExecPrefix)/lib
 #else 
-# define DefCannaPrefix /usr/local/canna
+# define DefCannaPrefix /usr
 # ifdef ObsoleteInstallDir
 #  define DefErrDir /usr/spool/canna
 #  define DefLibCannaDir /usr/lib
 # else
-#  define DefErrDir /var/log
+#  define DefErrDir /var/log/canna
 #  define DefLibCannaDir $(cannaExecPrefix)/lib
 # endif
 #endif
@@ -123,7 +123,7 @@
 # define DefCannaLibDir $(cannaPrefix)/lib
 #else
 # define DefCannaSrvDir $(cannaExecPrefix)/sbin
-# define DefCannaLibDir $(cannaPrefix)/share/canna
+# define DefCannaLibDir $(cannaPrefix)/lib
 #endif
 
 #if defined(LinuxArchitecture) || defined(FreeBSDArchitecture) \
@@ -146,9 +146,11 @@
 cannaSrvDir = DefCannaSrvDir
 XCOMM cannaLibDir = /var/lib/canna
 XCOMM cannaLibDir = /var/db/canna
-cannaLibDir = DefCannaLibDir
+cannaLibDir = /etc/canna
+cannaEtcDir = /etc/canna
+cannaShareDir = $(cannaPrefix)/share/canna
 XCOMM cannaManDir = $(cannaPrefix)/share/man
-cannaManDir = $(cannaPrefix)/man
+cannaManDir = $(cannaPrefix)/share/man
 cannaIncDir = $(cannaPrefix)/include/canna
 
 libCannaDir = DefLibCannaDir
@@ -158,7 +160,7 @@
 #define SupportOldWchar 1
 
 /* IPv6を用いる場合は1と定義してください */
-#define UseInet6 0 /* requires newer IPv6 implementation */
+#define UseInet6 1 /* requires newer IPv6 implementation */
 
 /* 3.2系から3.6系までの形式の辞書を作る場合はこれを定義してください */
 /* #define DIC_3_6_COMPAT */
@@ -175,10 +177,10 @@
 #ifdef InstallAsUser
 cannaOwnerGroup =
 #else
-cannaOwnerGroup = -o $(cannaOwner) -g $(cannaGroup)
+cannaOwnerGroup =
 #endif
 
-cannaDsoRev = 1.2
+cannaDsoRev = 1.2.0
 
 #if defined(cygwinArchitecture)
 /* import libraryの方を指定しておく。DLL本体は/usr/X11R6/binに入る。 */
@@ -199,8 +201,8 @@
 CHGRP = :
 CHMOD = :
 #else
-CHOWN = chown
-CHGRP = chgrp
+CHOWN = echo
+CHGRP = echo
 CHMOD = chmod
 #endif
 
@@ -237,7 +239,7 @@
 /* #define UseInstalledLibCanna YES */
 #define UseInstalledLibCanna NO
 
-DicDir   = $(cannaLibDir)/dic
+DicDir   = /var/lib/canna/dic
 
 /* ここから下は変更不要です */
 
@@ -339,10 +341,10 @@
 #endif
 
 #ifdef InstallAsUser
-UnixSockDir = $(cannaPrefix)/tmp/.iroha_unix
+UnixSockDir = $(cannaPrefix)/var/run/.iroha_unix
 AccessFile = $(cannaPrefix)/etc/hosts.canna
 #else
-UnixSockDir = /tmp/.iroha_unix
+UnixSockDir = /var/run/.iroha_unix
 AccessFile = /etc/hosts.canna
 #endif
 UnixSockName = IROHA
diff -ruN Canna37p1.orig/Imakefile Canna37p1/Imakefile
--- Canna37p1.orig/Imakefile	2003-12-28 02:15:20.000000000 +0900
+++ Canna37p1/Imakefile	2004-04-30 17:45:01.586886913 +0900
@@ -58,6 +58,7 @@
 	@echo "#define CANNAHOSTFILE '$(cannaLibDir)/cannahost'" >> junkjunk
 	@echo "/* for lib/canna */" >> junkjunk
 	@echo "#define CANNALIBDIR '$(cannaLibDir)'" >> junkjunk
+	@echo "#define CANNASHAREDIR '$(cannaShareDir)'" >> junkjunk
 	@echo "#define SUPPORT_OLD_WCHAR '$(supportOldWchar)'" >> junkjunk
 	@echo "/* for scripts */" >> junkjunk
 	@echo "#define CANNABINDIR $(cannaBinDir)" >> junkjunk
@@ -89,6 +90,7 @@
 	 echo "#define CANNAHOSTFILE \"$(cannaLibDir)/cannahost\"";\
 	 echo "/* for lib/canna */";\
 	 echo "#define CANNALIBDIR \"$(cannaLibDir)\""; \
+	 echo "#define CANNASHAREDIR \"$(cannaShareDir)\""; \
 	 echo "#define SUPPORT_OLD_WCHAR $(supportOldWchar)"; \
 	 echo "/* for scripts */";\
 	 echo "#define CANNABINDIR $(cannaBinDir)"; \
diff -ruN Canna37p1.orig/dic/phono/Imakefile Canna37p1/dic/phono/Imakefile
--- Canna37p1.orig/dic/phono/Imakefile	2002-10-19 17:27:40.000000000 +0900
+++ Canna37p1/dic/phono/Imakefile	2004-04-30 17:45:01.587886780 +0900
@@ -28,6 +28,7 @@
        ROMAJI_DIC_DEF = -DSHIFT
 
                DICDIR = $(cannaLibDir)/dic
+	CANNASHAREDIR = $(cannaShareDir)
             SAMPLEDIR = $(cannaLibDir)/sample
          SAMPLESRCDIR = $(cannaLibDir)/sample/src
 
@@ -69,7 +70,7 @@
 MakeDirectories(install,$(DICDIR))
 MakeDirectories(install,$(SAMPLESRCDIR))
 
-InstallMultiple($(ROMKANA_TABLES),$(DICDIR))
+InstallMultiple($(ROMKANA_TABLES),$(CANNASHAREDIR))
 
 #ifdef USE_OBSOLETE_STYLE_FILENAME
 InstallNonExec(default.kpdef,$(SAMPLESRCDIR))
diff -ruN Canna37p1.orig/misc/initfiles/Imakefile Canna37p1/misc/initfiles/Imakefile
--- Canna37p1.orig/misc/initfiles/Imakefile	2003-02-04 05:13:01.000000000 +0900
+++ Canna37p1/misc/initfiles/Imakefile	2004-04-30 17:45:01.587886780 +0900
@@ -6,7 +6,7 @@
 #endif
 
             CANNAROOT = ../..
-            SAMPLEDIR = $(cannaLibDir)/sample
+            SAMPLEDIR = $(cannaShareDir)
 
 all::
 
