diff -rup a/fedora/nscd.service b/fedora/nscd.service
--- a/fedora/nscd.service	2011-10-19 05:04:41.000000000 -0600
+++ b/fedora/nscd.service	2012-02-03 13:40:37.070063851 -0700
@@ -3,16 +3,15 @@ Description=Name Service Cache Daemon
 After=syslog.target
 
 [Service]
-Type=forking
 EnvironmentFile=-/etc/sysconfig/nscd
-ExecStart=/usr/sbin/nscd $NSCD_OPTIONS
+ExecStart=/usr/sbin/nscd --foreground $NSCD_OPTIONS
 ExecStop=/usr/sbin/nscd --shutdown
 ExecReload=/usr/sbin/nscd -i passwd
 ExecReload=/usr/sbin/nscd -i group
 ExecReload=/usr/sbin/nscd -i hosts
-ExecReload=/usr/sbin/nscd -i service
+ExecReload=/usr/sbin/nscd -i services
+ExecReload=/usr/sbin/nscd -i netgroup
 Restart=always
-PIDFile=/run/nscd/nscd.pid
 
 [Install]
 WantedBy=multi-user.target
