### Eclipse Workspace Patch 1.0
#P org.eclipse.jdt.core
Index: compiler/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java,v
retrieving revision 1.203
diff -u -r1.203 CompilerOptions.java
--- compiler/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java	14 Apr 2008 19:41:33 -0000	1.203
+++ compiler/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java	15 Dec 2008 19:50:02 -0000
@@ -1145,9 +1145,9 @@
 		
 		// by default only lines and source attributes are generated.
 		this.produceDebugAttributes = ClassFileConstants.ATTR_SOURCE | ClassFileConstants.ATTR_LINES;
-		this.complianceLevel = this.originalComplianceLevel = ClassFileConstants.JDK1_4; // by default be compliant with 1.4
-		this.sourceLevel = this.originalSourceLevel = ClassFileConstants.JDK1_3; //1.3 source behavior by default
-		this.targetJDK = ClassFileConstants.JDK1_2; // default generates for JVM1.2
+		this.complianceLevel = this.originalComplianceLevel = ClassFileConstants.JDK1_5;
+		this.sourceLevel = this.originalSourceLevel = ClassFileConstants.JDK1_5;
+		this.targetJDK = ClassFileConstants.JDK1_5;
 
 		this.defaultEncoding = null; // will use the platform default encoding
 
