Description: Fix GCC-5 build issue
Author: Thomas Pierson <contact@thomaspierson.fr>, Timothy Potter <timothy.potter@hp.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=777819
Last-Update: 2015-10-03
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall -Wno-sign-compare -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-private-field -Wno-unknown-warning-option --std=c++0x -U__STRICT_ANSI__")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall -Wno-sign-compare -Wno-deprecated-declarations -Wno-unused-local-typedefs --std=c++0x -U__STRICT_ANSI__")
 
 option(BUILD_WERROR "Build with -Werror" ON)
 
