diff -uNr torque-3.0.1.ORIG/src/include/libpbs.h torque-3.0.1/src/include/libpbs.h
--- torque-3.0.1.ORIG/src/include/libpbs.h	2011-06-17 19:19:32.984380003 +0200
+++ torque-3.0.1/src/include/libpbs.h	2011-06-17 19:23:19.406379620 +0200
@@ -117,7 +117,7 @@
 #define EOF -1
 #endif
 
-#define MUNGE_SIZE 256 /* I do not know what the proper size of this should be. My 
+#define MUNGE_SIZE 1024 /* I do not know what the proper size of this should be. My 
                           testing with munge shows it creates a string of 128 bytes */
 
 /* enums for standard job files (sync w/TJobFileType[]) */
