diff -uNr torque-3.0.2.ORIG/contrib/init.d/pbs_server torque-3.0.2/contrib/init.d/pbs_server
--- torque-3.0.2.ORIG/contrib/init.d/pbs_server	2011-10-09 00:09:29.026651535 +0200
+++ torque-3.0.2/contrib/init.d/pbs_server	2011-10-09 00:12:05.012653474 +0200
@@ -14,13 +14,11 @@
         echo "Configuration already exists.  Please remove $PBS_SERVERDB to create a new one."
         exit 1
     fi
-
     $PBS_DAEMON -d $PBS_HOME -t create &
-    while [ ! -r $PBS_SERVERDB ]; do
-        sleep 1
-    done
+    sleep 5
     killproc pbs_server
     RET=$?
+
 }
 
 start() {
