From c3d2b8f3de5b6161845304cf46982d2c5a9918b6 Mon Sep 17 00:00:00 2001
From: Chris Leech <cleech@redhat.com>
Date: Thu Feb 21 21:05:39 PST 2013
Subject: disable iscsid.startup from iscsiadm, prefer systemd socket activation

---
 etc/iscsid.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/etc/iscsid.conf b/etc/iscsid.conf
index ac1d231..5851fa5 100644
--- a/etc/iscsid.conf
+++ b/etc/iscsid.conf
@@ -17,7 +17,7 @@
 # maintainers.
 #
 # Default for Fedora and RHEL. (uncomment to activate).
-iscsid.startup = /etc/rc.d/init.d/iscsid force-start
+#iscsid.startup = /bin/systemctl start iscsid.service
 # 
 # Default for upstream open-iscsi scripts (uncomment to activate).
 # iscsid.startup = /sbin/iscsid
-- 
1.7.11.7

