Index: kde4libs-4.4.2/kfile/kdiroperator.cpp
===================================================================
--- kde4libs-4.4.2.orig/kfile/kdiroperator.cpp	2010-04-12 20:32:34.618195416 -0400
+++ kde4libs-4.4.2/kfile/kdiroperator.cpp	2010-04-12 20:33:34.509194414 -0400
@@ -176,6 +176,13 @@
     }
 }
 
+void KDirOperator::keyPressEvent(QKeyEvent *e)
+{
+    if (!(e->key() == Qt::Key_Return || e->key() == Qt::Key_Enter )) {
+        QWidget::keyPressEvent(e);
+    }
+}
+
 class KDirOperator::Private
 {
 public:
Index: kde4libs-4.4.2/kfile/kdiroperator.h
===================================================================
--- kde4libs-4.4.2.orig/kfile/kdiroperator.h	2010-04-12 20:32:34.634194029 -0400
+++ kde4libs-4.4.2/kfile/kdiroperator.h	2010-04-12 20:34:07.685194085 -0400
@@ -610,6 +610,8 @@
 
     virtual void resizeEvent(QResizeEvent *event);
 
+    virtual void keyPressEvent(QKeyEvent * event);
+
     /**
      * Sets up all the actions. Called from the constructor, you usually
      * better not call this.
Index: kde4libs-4.4.2/kfile/kfilewidget.cpp
===================================================================
--- kde4libs-4.4.2.orig/kfile/kfilewidget.cpp	2010-04-12 20:32:45.857214271 -0400
+++ kde4libs-4.4.2/kfile/kfilewidget.cpp	2010-04-12 20:34:55.097194976 -0400
@@ -1100,7 +1100,9 @@
 
     // if we are saving, let another chance to the user before accepting the dialog (or trying to
     // accept). This way the user can choose a file and add a "_2" for instance to the filename
-    if (operationMode != KFileWidget::Saving) {
+    if (operationMode == KFileWidget::Saving) {
+        locationEdit->setFocus();
+    } else {
         q->slotOk();
     }
 }
