--- latex2html-2002-2-1/config/config.pl.gsfont	2002-09-28 07:15:01.000000000 +0200
+++ latex2html-2002-2-1/config/config.pl	2005-03-16 10:50:17.392513136 +0100
@@ -1176,7 +1176,7 @@
               push(@right_paths,L2hos->path2os($testpath));
               $gs_lib = 1;
             }
-            $testpath .= "${dd}fonts" unless($testpath =~ /\Q$dd\Efonts$/i);
+            $testpath .= "${dd}fonts" unless($testpath =~ /\Q$dd\Efonts/i);
             if(!$gs_fonts && -d $testpath) {
               push(@right_paths,L2hos->path2os($testpath));
               $gs_fonts = 1;
