--- MIME-Types-1.31/ChangeLog	2010-09-21 14:35:38.000000000 +0100
+++ MIME-Types-1.31/ChangeLog	2011-03-15 10:10:00.994144345 +0000
@@ -136,7 +136,7 @@
 
 version 1.14: Wed Nov 24 18:09:49 CET 2004
 
-	- [Ville Skytt‰] provided some minor fixes and two new types
+	- [Ville Skytt√§] provided some minor fixes and two new types
 
 	- [Ave Wringley] contributed video/x-ms-wmv
 
@@ -145,14 +145,14 @@
 
 version 1.13: Sat Apr 24 00:15:09 CEST 2004
 
-	- [Ville Skytt‰] and [Austing Ziegler] updated the type list to
+	- [Ville Skytt√§] and [Austing Ziegler] updated the type list to
 	  IANA registry dated 2004-04-09.  Many changes to clean-up the
 	  list, trying to find a good balance between IANA's rules and
 	  common practice.
 
 version 1.12: Wed Jan 21 10:09:43 CET 2004
 
-	- [Ville Skytt‰] requested removal of application/excel, because
+	- [Ville Skytt√§] requested removal of application/excel, because
 	  it is not IANA registered.  Applied.
 
 	- Documented that alternative types for a certain extension will
@@ -163,7 +163,7 @@
 
 version 1.11: Tue Jan 13 10:36:41 CET 2004
 
-	- [Ville Skytt‰] contributed a large set of new mime-types, and
+	- [Ville Skytt√§] contributed a large set of new mime-types, and
 	  applied some fixes. New IANA names are now all included.
 
 version 1.10: Wed Dec 17 12:47:00 CET 2003
@@ -203,10 +203,10 @@
 	- [Guangzu Wang] pointed to typos in the Synopsis of both packages:
 	  $mimetype   -->  $mimetypes
 
-	- [Ville Skytt‰] some doc patches and many tweaks and additions
+	- [Ville Skytt√§] some doc patches and many tweaks and additions
 	  to the type list.
 
-	- [Ville Skytt‰] fixed bug, where the third column was taken as
+	- [Ville Skytt√§] fixed bug, where the third column was taken as
 	  second if that one was empty.
 
 version 1.004: Tue Oct 22 19:22:43 CEST 2002
