diff -ruN Log-Dispatch-2.11-orig/t/01-basic.t Log-Dispatch-2.11/t/01-basic.t
--- Log-Dispatch-2.11-orig/t/01-basic.t	2005-07-27 15:47:09.000000000 +0100
+++ Log-Dispatch-2.11/t/01-basic.t	2005-09-10 22:37:37.000000000 +0100
@@ -27,6 +27,14 @@
                     syslog_file   => '/var/log/messages',
                   );
 }
+else
+{
+    # Fedora Extras: mail tests coverage
+
+    %TestConfig = ( email_address => 'root@localhost.localdomain',
+                    syslog_file   => '/var/log/messages',
+                  );
+}
 
 use Log::Dispatch::File;
 use Log::Dispatch::Handle;
