Index: ferret-0.11.8.4/ext/posh.h
===================================================================
--- ferret-0.11.8.4.orig/ext/posh.h
+++ ferret-0.11.8.4/ext/posh.h
@@ -512,6 +512,13 @@ Metrowerks:
 #  define POSH_CPU_STRING "PA-RISC"
 #endif
 
+#if defined __aarch64__ || defined __arm64__ || defined FORCE_DOXYGEN
+#  define POSH_CPU_AARCH64 /**<if defined, target CPU is AArch64 */
+#  if !defined FORCE_DOXYGEN
+#     define POSH_CPU_STRING "AArch64"
+#  endif
+#endif
+
 #if !defined POSH_CPU_STRING
 #  error POSH cannot determine target CPU
 #  define POSH_CPU_STRING "Unknown" /* this is here for Doxygen's benefit */
