diff -up libmikmod-3.2.0-beta2/playercode/mdriver.c~ libmikmod-3.2.0-beta2/playercode/mdriver.c
--- libmikmod-3.2.0-beta2/playercode/mdriver.c~	2008-02-18 22:34:33.000000000 +0100
+++ libmikmod-3.2.0-beta2/playercode/mdriver.c	2008-02-18 22:34:33.000000000 +0100
@@ -185,11 +185,12 @@ MIKMODAPI CHAR* MikMod_InfoDriver(void)
 
 	if(len)
 		if((list=_mm_malloc(len*sizeof(CHAR)))) {
+			char *_list = list;
 			list[0]=0;
 			/* list all registered device drivers : */
 			for(t=1,l=firstdriver;l;l=l->next,t++)
-				sprintf(list,(l->next)?"%s%2d %s\n":"%s%2d %s",
-				    list,t,l->Version);
+				_list += sprintf(_list,(l->next)?"%2d %s\n":"%2d %s",
+						 t,l->Version);
 		}
 	MUTEX_UNLOCK(lists);
 	return list;
diff -up libmikmod-3.2.0-beta2/playercode/mloader.c~ libmikmod-3.2.0-beta2/playercode/mloader.c
--- libmikmod-3.2.0-beta2/playercode/mloader.c~	2008-02-18 22:36:55.000000000 +0100
+++ libmikmod-3.2.0-beta2/playercode/mloader.c	2008-02-18 22:36:55.000000000 +0100
@@ -67,10 +67,11 @@ MIKMODAPI CHAR* MikMod_InfoLoader(void)
 
 	if(len)
 		if((list=_mm_malloc(len*sizeof(CHAR)))) {
+			char *_list = list;
 			list[0]=0;
 			/* list all registered module loders */
 			for(l=firstloader;l;l=l->next)
-				sprintf(list,(l->next)?"%s%s\n":"%s%s",list,l->version);
+				_list += sprintf(_list,(l->next)?"%s\n":"%s",l->version);
 		}
 	MUTEX_UNLOCK(lists);
 	return list;
