--- libmikmod-3.1.11/include/mikmod_internals.h.64bit	2004-08-04 05:08:06.604109384 +0200
+++ libmikmod-3.1.11/include/mikmod_internals.h	2004-08-04 05:09:56.700372208 +0200
@@ -50,7 +50,8 @@
 /*========== More type definitions */
 
 /* SLONGLONG: 64bit, signed */
-#if defined (__arch64__) || defined(__alpha)
+#include <stdint.h>
+#if __WORDSIZE == 64
 typedef long		SLONGLONG;
 #define NATIVE_64BIT_INT
 #elif defined(__WATCOMC__)
--- libmikmod-3.1.11/include/mikmod.h.in.64bit	2004-08-04 05:07:56.092707360 +0200
+++ libmikmod-3.1.11/include/mikmod.h.in	2004-08-04 05:09:06.217046840 +0200
@@ -85,7 +85,8 @@
 
 @DOES_NOT_HAVE_SIGNED@
 
-#if defined(__arch64__) || defined(__alpha)
+#include <stdint.h>
+#if __WORDSIZE == 64
 /* 64 bit architectures */
 
 typedef signed char     SBYTE;      /* 1 byte, signed */
