--- gst-plugins-ugly-0.10.9/gst/asfdemux/gstasfdemux.c	2008/08/11 18:44:35	1.145
+++ gst-plugins-ugly-0.10.9/gst/asfdemux/gstasfdemux.c	2008/08/28 09:57:30	1.146
@@ -1333,7 +1333,7 @@
     if (!demux->activated_streams)
       flow = gst_asf_demux_push_complete_payloads (demux, TRUE);
 
-    if (flow != GST_FLOW_OK && flow != GST_FLOW_UNEXPECTED) {
+    if (GST_FLOW_IS_FATAL (flow) || flow == GST_FLOW_NOT_LINKED) {
       GST_DEBUG_OBJECT (demux, "pushing complete payloads failed");
       goto pause;
     }
@@ -1365,7 +1365,7 @@
     gst_pad_pause_task (demux->sinkpad);
 
     /* For the error cases (not EOS) */
-    if (flow != GST_FLOW_OK && flow != GST_FLOW_UNEXPECTED) {
+    if (GST_FLOW_IS_FATAL (flow) || flow == GST_FLOW_NOT_LINKED) {
       /* Post an error. Hopefully something else already has, but if not... */
       GST_ELEMENT_ERROR (demux, STREAM, FAILED,
           (_("Internal data stream error.")),
