--- libksquirrel/configure.ac.ORI	2011-11-21 15:32:36.479813141 +0100
+++ libksquirrel/configure.ac	2011-11-21 15:32:48.082812826 +0100
@@ -587,15 +587,6 @@
     fi
 fi
 
-sq_codec_pict="yes"
-sq_have_pict="no"
-if test $sq_codec_pict = yes; then
-    AC_PATH_PROG(PICTTOPPM, picttoppm)
-    if [ test -n "$PICTTOPPM" ]; then sq_have_pict="yes" AC_SUBST(PICTTOPPM)
-    else missing_prog PICT netpbm
-    fi
-fi
-
 sq_codec_mac="yes"
 sq_have_mac="no"
 if test $sq_codec_mac = yes; then
