diff -ur gnome-user-docs-3.2.1.orig/gnome-help/Makefile.am gnome-user-docs-3.2.1/gnome-help/Makefile.am
--- gnome-user-docs-3.2.1.orig/gnome-help/Makefile.am	2011-11-18 20:31:13.000000000 -0500
+++ gnome-user-docs-3.2.1/gnome-help/Makefile.am	2011-11-18 20:34:02.000000000 -0500
@@ -4,7 +4,7 @@
 
 HELP_ID = gnome-help
 
-HELP_LINGUAS = ca de fi fr es gl hi hu nl ru sl sv vi
+HELP_LINGUAS =
 
 HELP_MEDIA = \
 	figures/color-average.png \
