diff -up xemacs-packages-extra-20090217/xemacs-packages/auctex/tex-jp.el~ xemacs-packages-extra-20090217/xemacs-packages/auctex/tex-jp.el
--- xemacs-packages-extra-20090217/xemacs-packages/auctex/tex-jp.el~	2008-07-23 10:12:18.000000000 +0300
+++ xemacs-packages-extra-20090217/xemacs-packages/auctex/tex-jp.el	2009-02-22 21:30:17.000000000 +0200
@@ -157,7 +157,7 @@ For detail, see `TeX-command-list', whic
   :type 'string)
   (make-variable-buffer-local 'japanese-TeX-command-default)
 
-(defcustom japanese-LaTeX-command-default "LaTeX"
+(defcustom japanese-LaTeX-command-default "pLaTeX"
   "*The default command for `TeX-command' in the japanese-LaTeX mode."
   :group 'AUCTeX-jp
   :type 'string)
