--- xinetd-2.3.15/Makefile.in	2012-05-14 09:22:22.661617117 +0200
+++ xinetd-2.3.15.new/Makefile.in	2012-05-14 09:32:05.260103054 +0200
@@ -14,7 +14,7 @@ topdir = @top_srcdir@
 
 LIBS = -lsio -lstr -lmisc -lxlog -lportable -lpset @LIBS@
 
-CFLAGS += @CFLAGS@ 
+CFLAGS += @CFLAGS@ -fPIE
 DCFLAGS = -Wall -Wredundant-decls -W -Wfloat-equal -Wundef -Wcast-qual -Wwrite-strings -Wmissing-noreturn -Wmissing-format-attribute -Wshadow -Wpointer-arith -Wno-unused -g
 
 
--- xinetd-2.3.15/xinetd/Makefile.in	2005-03-31 01:15:28.000000000 +0200
+++ xinetd-2.3.15.new/xinetd/Makefile.in	2012-05-14 09:32:24.183659971 +0200
@@ -119,7 +119,7 @@ itox: itox.c
 	$(CC) $(CFLAGS) $(DEBUG) $(SRCDIR)/itox.c -o $@ $(LDFLAGS) $(LIBS)
 
 xinetd: $(OBJS)
-	$(CC) $(CFLAGS) $(DEBUG) -o $@ $(OBJS) $(LDFLAGS) $(LIBS) || rm -f $@
+	$(CC) $(CFLAGS) $(DEBUG) -o $@ -PIE $(OBJS) $(LDFLAGS) $(LIBS) || rm -f $@
 
 clean:
 	rm -f $(OBJS) $(NAME) core itox
