diff -up webkit-qtwebkit/Source/JavaScriptCore/JavaScriptCore.pro.javascriptcore_debuginfo webkit-qtwebkit/Source/JavaScriptCore/JavaScriptCore.pro
--- webkit-qtwebkit/Source/JavaScriptCore/JavaScriptCore.pro.javascriptcore_debuginfo	2011-06-21 12:13:07.000000000 -0500
+++ webkit-qtwebkit/Source/JavaScriptCore/JavaScriptCore.pro	2011-06-22 07:42:40.365535990 -0500
@@ -22,12 +22,20 @@ contains(QT_CONFIG, embedded):CONFIG += 
 contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
 unix:contains(QT_CONFIG, reduce_relocations):CONFIG += bsymbolic_functions
 
-CONFIG(QTDIR_build) {
+!CONFIG(webkit-debug):CONFIG(QTDIR_build) {
     # Remove the following 2 lines if you want debug information in JavaScriptCore
     CONFIG -= separate_debug_info
     CONFIG += no_debug_info
 }
 
+# optimize for low mem arch s390 (disable debugging, etc...)
+equals(QT_ARCH, s390): {
+    CONFIG -= separate_debug_info
+    CONFIG += no_debug_info
+    QMAKE_CXXFLAGS_RELEASE -= -g
+    QMAKE_CFLAGS_RELEASE -= -g
+}
+
 *-g++*:QMAKE_CXXFLAGS_RELEASE -= -O2
 *-g++*:QMAKE_CXXFLAGS_RELEASE += -O3
 
diff -up webkit-qtwebkit/Source/WebCore/WebCore.pri.javascriptcore_debuginfo webkit-qtwebkit/Source/WebCore/WebCore.pri
--- webkit-qtwebkit/Source/WebCore/WebCore.pri.javascriptcore_debuginfo	2011-06-21 12:13:07.000000000 -0500
+++ webkit-qtwebkit/Source/WebCore/WebCore.pri	2011-06-22 08:36:58.159265977 -0500
@@ -289,6 +289,14 @@ contains(DEFINES, ENABLE_WEBGL=1)|contai
     CONFIG += no_debug_info
 }
 
+# optimize for low mem arch s390 (disable debugging, etc...)
+equals(QT_ARCH, s390): {
+    CONFIG -= separate_debug_info
+    CONFIG += no_debug_info
+    QMAKE_CXXFLAGS_RELEASE -= -g
+    QMAKE_CFLAGS_RELEASE -= -g
+}
+
 contains (CONFIG, text_breaking_with_icu) {
     LIBS += -licuuc
 }
