diff -up webkit-qtwebkit/Tools/DumpRenderTree/TestNetscapePlugIn/PluginTest.cpp.me webkit-qtwebkit/Tools/DumpRenderTree/TestNetscapePlugIn/PluginTest.cpp
--- webkit-qtwebkit/Tools/DumpRenderTree/TestNetscapePlugIn/PluginTest.cpp.me	2012-01-24 15:26:33.000000000 +0100
+++ webkit-qtwebkit/Tools/DumpRenderTree/TestNetscapePlugIn/PluginTest.cpp	2012-01-24 15:27:01.000000000 +0100
@@ -28,6 +28,7 @@
 #include "PluginObject.h"
 #include <assert.h>
 #include <string.h>
+#include <unistd.h>
 
 using namespace std;
 extern NPNetscapeFuncs *browser;
