--- gme/CMakeLists.txt~	2012-05-27 22:48:53.000000000 +0200
+++ gme/CMakeLists.txt	2013-08-22 19:03:39.175649327 +0200
@@ -159,4 +159,4 @@
                     ARCHIVE DESTINATION lib) # DLL platforms
 
 install(FILES ${EXPORTED_HEADERS} DESTINATION include/gme)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libgme.pc DESTINATION lib/pkgconfig)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libgme.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
