diff -Nur wine-1.1.4/tools/wineshelllink wine-1.1.4.fix_shelllink_desktop_files_encoding/tools/wineshelllink
--- wine-1.1.4/tools/wineshelllink	2008-09-05 23:28:27.000000000 +0800
+++ wine-1.1.4.fix_shelllink_desktop_files_encoding/tools/wineshelllink	2008-09-14 13:26:09.000000000 +0800
@@ -222,6 +222,7 @@
     fi
 
     desktop_entry > "$xdg_data_dir/applications/wine/$link.desktop"
+    magic_locale_to_utf8.sh "$xdg_data_dir/applications/wine/$link.desktop"
     write_menu_file "wine$linkpath" "wine$linkpath/$linkname.desktop"
 else
     if [ -d "$HOME/Desktop" ]
@@ -231,6 +232,12 @@
       desktop_target="$HOME/$linkname.desktop"
     fi
     desktop_entry > "$desktop_target"
+    if [ -d "$HOME/Desktop" ]
+    then
+      magic_locale_to_utf8.sh "$HOME/Desktop/$linkname.desktop"
+    else
+      magic_locale_to_utf8.sh "$HOME/$linkname.desktop"
+    fi
 fi
 
 exit 0
