--- lib/Alien/SeleniumRC.pm.orig	2010-05-23 22:11:41.000000000 +0200
+++ lib/Alien/SeleniumRC.pm	2010-05-23 22:13:36.000000000 +0200
@@ -11,7 +11,7 @@
     my $args = shift || '';
     my $jarfile = find_jar_location();
 
-    my $cmd = "java -jar $jarfile $args";
+    my $cmd = "/usr/bin/selenium-server $args";
     if ( $ENV{SELENIUMRC_USE_SIC} ) {
         $cmd = "sudo /usr/libexec/StartupItemContext `which java` -jar $jarfile $args";
     }
@@ -27,8 +27,7 @@
 }
 
 sub find_jar_location {
-    my $pm_location = $INC{'Alien/SeleniumRC.pm'};
-    (my $src_location = $pm_location) =~ s#\.pm#/selenium-server.jar#;
+    my $src_location = '/usr/share/java/selenium-server-coreless.jar';
 
     if ($^O eq 'cygwin') {
 	$src_location = `cygpath -m '$src_location'`;
