--- a/src/3rdparty/webkit/Source/WebKit/qt/Api/qwebsettings.cpp	2010-10-13 20:31:41.000000000 +0800
+++ b/src/3rdparty/webkit/Source/WebKit/qt/Api/qwebsettings.cpp	2010-10-13 20:32:19.713334761 +0800
@@ -470,7 +470,7 @@ QWebSettings::QWebSettings()
     d->attributes.insert(QWebSettings::FrameFlatteningEnabled, false);
     d->attributes.insert(QWebSettings::SiteSpecificQuirksEnabled, true);
     d->offlineStorageDefaultQuota = 5 * 1024 * 1024;
-    d->defaultTextEncoding = QLatin1String("iso-8859-1");
+    d->defaultTextEncoding = QLatin1String("gb18030");
 }
 
 /*!
