diff -ur qt-everywhere-opensource-src-4.8.0.orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.cpp qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.cpp
--- qt-everywhere-opensource-src-4.8.0.orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.cpp	2011-12-19 15:56:33.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.cpp	2011-12-19 15:58:59.000000000 -0500
@@ -39,14 +39,14 @@
 
 template <> void freeOwnedGPtr<GCond>(GCond* ptr)
 {
-    if (ptr)
-        g_cond_free(ptr);
+//    if (ptr)
+//        g_cond_free(ptr);
 }
 
 template <> void freeOwnedGPtr<GMutex>(GMutex* ptr)
 {
-    if (ptr)
-        g_mutex_free(ptr);
+//    if (ptr)
+//        g_mutex_free(ptr);
 }
 
 template <> void freeOwnedGPtr<GPatternSpec>(GPatternSpec* ptr)
diff -ur qt-everywhere-opensource-src-4.8.0.orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h
--- qt-everywhere-opensource-src-4.8.0.orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h	2011-12-19 15:56:33.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h	2011-12-19 15:59:16.000000000 -0500
@@ -24,6 +24,7 @@
 
 #if ENABLE(GLIB_SUPPORT)
 
+#include <glib.h>
 #include <algorithm>
 #include <wtf/Assertions.h>
 #include <wtf/Noncopyable.h>
diff -ur qt-everywhere-opensource-src-4.8.0.orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h
--- qt-everywhere-opensource-src-4.8.0.orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h	2011-12-19 15:56:33.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h	2011-12-19 15:59:30.000000000 -0500
@@ -39,7 +39,6 @@
 typedef struct _GAsyncResult GAsyncResult;
 typedef struct _GCancellable GCancellable;
 typedef struct _GCharsetConverter GCharsetConverter;
-typedef struct _GCond GCond;
 typedef struct _GDir GDir;
 typedef struct _GdkAtom* GdkAtom;
 typedef struct _GdkCursor GdkCursor;
@@ -52,7 +51,6 @@
 typedef struct _GHashTable GHashTable;
 typedef struct _GInputStream GInputStream;
 typedef struct _GList GList;
-typedef struct _GMutex GMutex;
 typedef struct _GPatternSpec GPatternSpec;
 typedef struct _GPollableOutputStream GPollableOutputStream;
 typedef struct _GSocketClient GSocketClient;
diff -ur qt-everywhere-opensource-src-4.8.0.orig/src/3rdparty/webkit/Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp
--- qt-everywhere-opensource-src-4.8.0.orig/src/3rdparty/webkit/Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp	2011-12-19 15:56:07.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp	2011-12-20 08:09:11.000000000 -0500
@@ -104,8 +104,8 @@
     WebKitVideoSinkPrivate* priv;
 
     sink->priv = priv = G_TYPE_INSTANCE_GET_PRIVATE(sink, WEBKIT_TYPE_VIDEO_SINK, WebKitVideoSinkPrivate);
-    priv->data_cond = g_cond_new();
-    priv->buffer_mutex = g_mutex_new();
+    //priv->data_cond = g_cond_new();
+    //priv->buffer_mutex = g_mutex_new();
 }
 
 static gboolean
@@ -234,12 +234,12 @@
     WebKitVideoSinkPrivate* priv = sink->priv;
 
     if (priv->data_cond) {
-        g_cond_free(priv->data_cond);
+        //g_cond_free(priv->data_cond);
         priv->data_cond = 0;
     }
 
     if (priv->buffer_mutex) {
-        g_mutex_free(priv->buffer_mutex);
+        //g_mutex_free(priv->buffer_mutex);
         priv->buffer_mutex = 0;
     }
 
