diff -Nuar qt-x11-opensource-src-4.5.0-rc1/src/gui/text/qfontdatabase_x11.cpp qt-x11-opensource-src-4.5.0-rc1n/src/gui/text/qfontdatabase_x11.cpp
--- qt-x11-opensource-src-4.5.0-rc1/src/gui/text/qfontdatabase_x11.cpp	2009-02-05 00:05:29.000000000 +0800
+++ qt-x11-opensource-src-4.5.0-rc1n/src/gui/text/qfontdatabase_x11.cpp	2009-02-06 11:30:49.000000000 +0800
@@ -1323,9 +1323,25 @@
 
                 QtFontStyle::Key key = style->key;
 
+                // does this style have a bold equivalent?
+                key.weight = QFont::Bold;
+                QtFontStyle *equiv = foundry->style(key);
+                if (!equiv) {
+                    // let's fake one...
+                    equiv = foundry->style(key, true);
+                    equiv->smoothScalable = true;
+
+                    QtFontSize *equiv_size = equiv->pixelSize(SMOOTH_SCALABLE, true);
+                    QtFontEncoding *equiv_enc = equiv_size->encodingID(-1, 0, 0, 0, 0, true);
+
+                    // keep the same pitch
+                    equiv_enc->pitch = enc->pitch;
+                   key.weight = QFont::Normal;
+                }
+
                 // does this style have an italic equivalent?
                 key.style = QFont::StyleItalic;
-                QtFontStyle *equiv = foundry->style(key);
+                equiv = foundry->style(key);
                 if (equiv) continue;
 
                 // does this style have an oblique equivalent?
