diff -Nur qt-everywhere-opensource-src-4.8.4-orig/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h qt-everywhere-opensource-src-4.8.4/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h
--- qt-everywhere-opensource-src-4.8.4-orig/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h 2013-05-21 07:25:22.000000000 -0400
+++ qt-everywhere-opensource-src-4.8.4/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h 2013-05-21 07:52:12.000000000 -0400
@@ -884,7 +884,7 @@
 #endif
 
 #if !defined(WTF_USE_JSVALUE64) && !defined(WTF_USE_JSVALUE32) && !defined(WTF_USE_JSVALUE32_64)
-#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS) || OS(SOLARIS) || OS(HPUX))) || (CPU(IA64) && !CPU(IA64_32)) || CPU(ALPHA) || CPU(AIX64) || CPU(SPARC64)
+#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS) || OS(SOLARIS) || OS(HPUX))) || (CPU(IA64) && !CPU(IA64_32)) || CPU(ALPHA) || CPU(AIX64) || CPU(SPARC64) || defined(__mips64)
 #define WTF_USE_JSVALUE64 1
 #elif CPU(ARM) || CPU(PPC64)
 #define WTF_USE_JSVALUE32 1
diff -Nur qt-everywhere-opensource-src-4.8.4-orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h qt-everywhere-opensource-src-4.8.4/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
--- qt-everywhere-opensource-src-4.8.4-orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h 2013-05-21 07:25:18.000000000 -0400
+++ qt-everywhere-opensource-src-4.8.4/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h 2013-05-21 07:52:07.000000000 -0400
@@ -146,8 +146,7 @@
 
 /* CPU(MIPS) - MIPS 32-bit */
 /* Note: Only O32 ABI is tested, so we enable it for O32 ABI for now.  */
-#if (defined(mips) || defined(__mips__) || defined(MIPS) || defined(_MIPS_)) \
-    && defined(_ABIO32)
+#if (defined(mips) || defined(__mips__) || defined(MIPS) || defined(_MIPS_)) 
 #define WTF_CPU_MIPS 1
 #if defined(__MIPSEB__)
 #define WTF_CPU_BIG_ENDIAN 1
@@ -156,7 +155,7 @@
 #define WTF_MIPS_ARCH __mips
 #define WTF_MIPS_ISA(v) (defined WTF_MIPS_ARCH && WTF_MIPS_ARCH == v)
 #define WTF_MIPS_ISA_AT_LEAST(v) (defined WTF_MIPS_ARCH && WTF_MIPS_ARCH >= v)
-#define WTF_MIPS_ARCH_REV __mips_isa_rev
+#define WTF_MIPS_ARCH_REV _MIPS_ISA
 #define WTF_MIPS_ISA_REV(v) (defined WTF_MIPS_ARCH_REV && WTF_MIPS_ARCH_REV == v)
 #define WTF_MIPS_DOUBLE_FLOAT (defined __mips_hard_float && !defined __mips_single_float)
 #define WTF_MIPS_FP64 (defined __mips_fpr && __mips_fpr == 64)
@@ -998,13 +997,17 @@
     || CPU(ALPHA) \
     || CPU(SPARC64) \
     || CPU(S390X) \
-    || CPU(PPC64)
+    || CPU(PPC64) || defined(__mips64)
 #define WTF_USE_JSVALUE64 1
 #else
 #define WTF_USE_JSVALUE32_64 1
 #endif
 #endif /* !defined(WTF_USE_JSVALUE64) && !defined(WTF_USE_JSVALUE32_64) */
 
+#if !defined(ENABLE_JIT) && defined(__mips64)
+#define ENABLE_JIT 0
+#endif
+
 #if !defined(ENABLE_REPAINT_THROTTLING)
 #define ENABLE_REPAINT_THROTTLING 0
 #endif
