--- src/gui/painting/qgraphicssystemfactory.cpp (Revision 922048)
+++ src/gui/painting/qgraphicssystemfactory.cpp (Arbeitskopie)
@@ -65,10 +65,10 @@
     }
 #endif
 
-    if (system == QLatin1String("raster"))
+    if (system == QLatin1String("native"))
+        return 0;
+    else if (system.isEmpty() || system == QLatin1String("raster"))
         return new QRasterGraphicsSystem;
-    else if (system.isEmpty() || system == QLatin1String("native"))
-        return 0;
 
 #if !defined(QT_NO_LIBRARY) && !defined(QT_NO_SETTINGS)
     if (!ret) {
