--- qt-x11-opensource-src-4.2.2/mkspecs/linux-g++-64/qmake.conf.multilib	2006-11-27 11:26:42.000000000 -0600
+++ qt-x11-opensource-src-4.2.2/mkspecs/linux-g++-64/qmake.conf	2007-02-26 15:01:02.000000000 -0600
@@ -14,6 +14,8 @@
 QMAKE_CFLAGS		= -m64
 QMAKE_LFLAGS		= -m64
 
+QMAKE_CFLAGS_RELEASE	+= -O2
+
 include(../common/g++.conf)
 include(../common/linux.conf)
 
--- qt-x11-opensource-src-4.2.2/mkspecs/linux-g++-32/qmake.conf.multilib	2006-11-27 11:26:42.000000000 -0600
+++ qt-x11-opensource-src-4.2.2/mkspecs/linux-g++-32/qmake.conf	2007-02-26 15:01:34.000000000 -0600
@@ -11,6 +11,8 @@
 QMAKE_CFLAGS		= -m32
 QMAKE_LFLAGS		= -m32
 
+QMAKE_CFLAGS_RELEASE	+= -O2
+
 include(../common/g++.conf)
 include(../common/linux.conf)
 load(qt_config)
--- qt-x11-opensource-src-4.2.2/mkspecs/linux-g++/qmake.conf.multilib	2006-11-27 11:26:42.000000000 -0600
+++ qt-x11-opensource-src-4.2.2/mkspecs/linux-g++/qmake.conf	2007-02-26 15:01:26.000000000 -0600
@@ -8,6 +8,8 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
+QMAKE_CFLAGS_RELEASE	+= -O2
+
 include(../common/g++.conf)
 include(../common/linux.conf)
 load(qt_config)
