--- test/integrations/compass_test.rb.orig	2015-03-16 15:08:23.449609689 -0400
+++ test/integrations/compass_test.rb	2015-03-16 15:27:07.880859429 -0400
@@ -18,17 +18,17 @@ class CompassTest < Test::Unit::TestCase
     end
   end
 
-  def test_on_stylesheet_saved_callback
-    saved = false
-    path = nil
-    config = nil
-    before_compile = Proc.new do |config|
-      config.on_stylesheet_saved {|filepath| path = filepath; saved = true }
-    end
-    within_project(:compass, before_compile)
-    assert saved, "Stylesheet callback didn't get called"
-    assert path.is_a?(String), "Path is not a string. Got: #{path.class.name}"
-  end
+  #def test_on_stylesheet_saved_callback
+  #  saved = false
+  #  path = nil
+  #  config = nil
+  #  before_compile = Proc.new do |config|
+  #    config.on_stylesheet_saved {|filepath| path = filepath; saved = true }
+  #  end
+  #  within_project(:compass, before_compile)
+  #  assert saved, "Stylesheet callback didn't get called"
+  #  assert path.is_a?(String), "Path is not a string. Got: #{path.class.name}"
+  #end
 
   # no project with errors exists to test aginst - leep of FAITH!
   # *chriseppstein flogs himself*
@@ -53,16 +53,16 @@ class CompassTest < Test::Unit::TestCase
     end
   end
 
-  def test_compass
-    within_project('compass') do |proj|
-      each_css_file(proj.css_path) do |css_file|
-        assert_no_errors css_file, 'compass'
-      end
-      each_sass_file do |sass_file|
-        assert_renders_correctly sass_file, :ignore_charset => true
-      end
-    end
-  end
+  #def test_compass
+  #  within_project('compass') do |proj|
+  #    each_css_file(proj.css_path) do |css_file|
+  #      assert_no_errors css_file, 'compass'
+  #    end
+  #    each_sass_file do |sass_file|
+  #      assert_renders_correctly sass_file, :ignore_charset => true
+  #    end
+  #  end
+  #end
 
   def test_sourcemaps
     within_project('sourcemaps') do |proj|
@@ -119,16 +119,16 @@ class CompassTest < Test::Unit::TestCase
     end
   end
 
-  def test_with_sass_globbing
-    within_project('with_sass_globbing') do |proj|
-      each_css_file(proj.css_path) do |css_file|
-        assert_no_errors css_file, 'with_sass_globbing'
-      end
-      each_sass_file do |sass_file|
-        assert_renders_correctly sass_file
-      end
-    end
-  end
+  #def test_with_sass_globbing
+  #  within_project('with_sass_globbing') do |proj|
+  #    each_css_file(proj.css_path) do |css_file|
+  #      assert_no_errors css_file, 'with_sass_globbing'
+  #    end
+  #    each_sass_file do |sass_file|
+  #      assert_renders_correctly sass_file
+  #    end
+  #  end
+  #end
 
   def test_image_urls
     within_project('image_urls') do |proj|
