--- libgd-2.1.0/configure.ac.orig	2013-12-15 19:20:24.569994500 +0000
+++ libgd-2.1.0/configure.ac	2013-12-15 19:20:38.010142489 +0000
@@ -45,7 +45,7 @@
 FEATURES="GD_GIF GD_GIFANIM GD_OPENPOLYGON"
 AC_SUBST(FEATURES)
 
-AM_INIT_AUTOMAKE([foreign dist-bzip2 dist-xz -Wall -Werror])
+AM_INIT_AUTOMAKE([foreign dist-bzip2 dist-xz subdir-objects -Wall -Werror])
 AC_CONFIG_HEADERS([src/config.h:src/config.hin])
 
 dnl newer automake wants this, but we still want to work with older
