diff --git a/src/settings/networkwidget.cpp b/src/settings/networkwidget.cpp
index 5495f0c..7a7001d 100644
--- a/src/settings/networkwidget.cpp
+++ b/src/settings/networkwidget.cpp
@@ -52,15 +52,15 @@ NetworkWidget::NetworkWidget(QWidget *parent)

     KCModuleInfo cacheInfo("cache.desktop");
     _cacheModule = new KCModuleProxy(cacheInfo, parent);
-    tabWidget->addTab(_cacheModule, i18n(cacheInfo.moduleName().toLocal8Bit()));
+    tabWidget->addTab(_cacheModule, i18n(cacheInfo.moduleName().toUtf8()));

     KCModuleInfo cookiesInfo("cookies.desktop");
     _cookiesModule = new KCModuleProxy(cookiesInfo, parent);
-    tabWidget->addTab(_cookiesModule, i18n(cookiesInfo.moduleName().toLocal8Bit()));
+    tabWidget->addTab(_cookiesModule, i18n(cookiesInfo.moduleName().toUtf8()));

     KCModuleInfo proxyInfo("proxy.desktop");
     _proxyModule = new KCModuleProxy(proxyInfo, parent);
-    tabWidget->addTab(_proxyModule, i18n(proxyInfo.moduleName().toLocal8Bit()));
+    tabWidget->addTab(_proxyModule, i18n(proxyInfo.moduleName().toUtf8()));

     connect(_cacheModule,   SIGNAL(changed(bool)), this, SLOT(hasChanged()));
     connect(_cookiesModule, SIGNAL(changed(bool)), this, SLOT(hasChanged()));
diff --git a/src/settings/settingsdialog.cpp b/src/settings/settingsdialog.cpp
index eee2a4e..c02512a 100644
--- a/src/settings/settingsdialog.cpp
+++ b/src/settings/settingsdialog.cpp
@@ -129,7 +129,7 @@ Private::Private(SettingsDialog *parent)
     // -- 8
     KCModuleInfo ebrowsingInfo("ebrowsing.desktop");
     ebrowsingModule = new KCModuleProxy(ebrowsingInfo, parent);
-    pageItem = parent->addPage(ebrowsingModule, i18n(ebrowsingInfo.moduleName().toLocal8Bit()));
+    pageItem = parent->addPage(ebrowsingModule, i18n(ebrowsingInfo.moduleName().toUtf8()));
     pageItem->setIcon(KIcon(ebrowsingInfo.icon()));

     // WARNING
