diff -ur xbmc-git20130114.orig/xbmc/filesystem/SMBDirectory.cpp xbmc-git20130114/xbmc/filesystem/SMBDirectory.cpp
--- xbmc-git20130114.orig/xbmc/filesystem/SMBDirectory.cpp	2013-02-20 10:30:02.410102492 +0000
+++ xbmc-git20130114/xbmc/filesystem/SMBDirectory.cpp	2013-02-20 10:30:46.248105548 +0000
@@ -44,7 +44,7 @@
 #include "threads/SingleLock.h"
 #include "PasswordManager.h"
 
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 
 #if defined(TARGET_DARWIN)
 #define XBMC_SMB_MOUNT_PATH "Library/Application Support/XBMC/Mounts/"
diff -ur xbmc-git20130114.orig/xbmc/filesystem/SmbFile.cpp xbmc-git20130114/xbmc/filesystem/SmbFile.cpp
--- xbmc-git20130114.orig/xbmc/filesystem/SmbFile.cpp	2013-02-20 10:30:02.415102431 +0000
+++ xbmc-git20130114/xbmc/filesystem/SmbFile.cpp	2013-02-20 10:30:58.589105858 +0000
@@ -27,7 +27,7 @@
 #include "PasswordManager.h"
 #include "SMBDirectory.h"
 #include "Util.h"
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include "settings/AdvancedSettings.h"
 #include "settings/GUISettings.h"
 #include "threads/SingleLock.h"
