--- Docs/sample/gnokiirc~	2012-04-17 13:09:03.000000000 -0500
+++ Docs/sample/gnokiirc	2012-04-17 13:10:42.793893402 -0500
@@ -238,7 +238,7 @@
 # In particular ensure that mgnokiidev is in this location, with
 # permissions 4750, owned by root, group gnokii.  Ensure you
 # are in the gnokii group and that the group exists...
-bindir = /usr/local/sbin/
+bindir = /usr/sbin/
 
 # Any entries in the following two sections will be set as environment
 # variables when running the scripts.
