--- xgnokii/xgnokii_cfg.c.orig	2008-10-06 08:42:09.000000000 +0100
+++ xgnokii/xgnokii_cfg.c	2008-10-06 08:41:03.000000000 +0100
@@ -75,7 +75,7 @@ static void GetDefaultValues()
 	xgnokiiConfig.user.fax = g_strdup("");
 	xgnokiiConfig.user.email = g_strdup("");
 	xgnokiiConfig.user.address = g_strdup("");
-	xgnokiiConfig.helpviewer = g_strdup("firefox");
+	xgnokiiConfig.helpviewer = g_strdup("htmlview");
 	if ((homedir = g_get_home_dir()) == NULL)
 		homedir = "";
 	xgnokiiConfig.mailbox = g_strdup_printf("%s/Mail/smsbox", homedir);
