diff -up xterm-261/xterm.desktop.desk xterm-261/xterm.desktop
--- xterm-261/xterm.desktop.desk	2010-06-15 12:58:13.000000000 +0200
+++ xterm-261/xterm.desktop	2010-07-02 13:02:56.000000000 +0200
@@ -32,10 +32,11 @@
 # -----------------------------------------------------------------------------
 [Desktop Entry]
 Name=XTerm
+GenericName=Terminal
 Comment=standard terminal emulator for the X window system
 Exec=xterm
 Terminal=false
 Type=Application
 Encoding=UTF-8
-Icon=xterm-color_48x48
+Icon=gnome-xterm
 Categories=System;TerminalEmulator;
