diff -up xterm-245/XTerm.ad.resources xterm-245/XTerm.ad
--- xterm-245/XTerm.ad.resources	2009-02-13 01:14:13.000000000 +0100
+++ xterm-245/XTerm.ad	2009-08-13 12:28:59.000000000 +0200
@@ -210,3 +210,13 @@
 !*allowTcapOps: false
 !*allowTitleOps: false
 !*allowWindowOps: false
+
+! Red Hat Defaults:
+*allowFontOps: false
+*allowTcapOps: false
+*allowWindowOps: false
+*VT100*eightBitInput: false
+*VT100*metaSendsEscape: true
+*VT100*backarrowKey: false
+*VT100*scrollBar: true
+*VT100*utf8Title: true
