diff --git a/Makefile.in b/Makefile.in
index 55b21b7..9340fdc 100755
--- a/Makefile.in
+++ b/Makefile.in
@@ -64,7 +64,7 @@ endif
 
 CFLAGS=@CPPFLAGS@ -g -I$(srcdir)/include -Iinclude -Ilib -Ilib/util -I$(srcdir) \
        $(TALLOC_CFLAGS) $(TEVENT_CFLAGS) $(TDB_CFLAGS) -I@libreplacedir@ \
-	-DVARDIR=\"$(localstatedir)\" -DETCDIR=\"$(etcdir)\" \
+	-DVARDIR=\"$(localstatedir)/lib\" -DETCDIR=\"$(etcdir)\" \
 	-DCTDB_VARDIR=\"$(localstatedir)/lib/ctdb\" \
 	-DLOGDIR=\"$(logdir)\" -DBINDIR=\"$(bindir)\" \
 	-DSOCKPATH=\"$(sockpath)\" \
-- 
1.8.3.1

