--- xfce4-session/engines/simple/simple.c.orig	2006-09-02 19:27:37.000000000 +0200
+++ xfce4-session/engines/simple/simple.c	2006-10-06 15:19:59.000000000 +0200
@@ -202,7 +202,6 @@
 simple_next (XfsmSplashEngine *engine, const gchar *text)
 {
   Simple *simple = (Simple *) engine->user_data;
-  GdkColor shcolor;
   gint tw, th, tx, ty;
 
   pango_layout_set_text (simple->layout, text, -1);
@@ -220,15 +219,6 @@
 
   gdk_gc_set_clip_rectangle (simple->gc, &simple->textbox);
 
-  /* draw shadow */
-  shcolor.red = (simple->fgcolor.red + simple->bgcolor.red) / 2;
-  shcolor.green = (simple->fgcolor.green + simple->bgcolor.green) / 2;
-  shcolor.blue = (simple->fgcolor.blue + simple->bgcolor.blue) / 2;
-  shcolor.red = (shcolor.red + shcolor.green + shcolor.blue) / 3;
-  shcolor.green = shcolor.red;
-  shcolor.blue = shcolor.red;
-
-  gdk_gc_set_rgb_fg_color (simple->gc, &shcolor);
   gdk_draw_layout (simple->pixmap, simple->gc,
                    tx + 2, ty + 2, simple->layout);
 
