diff -ur kdelibs-3.5.13.orig/tdecore/kcharsets.cpp kdelibs-3.5.13/tdecore/kcharsets.cpp
--- kdelibs-3.5.13.orig/tdecore/kcharsets.cpp	2010-11-27 10:10:13.000000000 -0500
+++ kdelibs-3.5.13/tdecore/kcharsets.cpp	2010-11-27 10:14:36.000000000 -0500
@@ -547,9 +547,10 @@
 
     if (name.isEmpty()) {
       ok = false;
-      return TQTextCodec::codecForName("iso8859-1");
-    }
-
+      return TQTextCodec::codecForLocale();
+      }
+    if (name=="gb2312"||name=="gbk")
+    name="gb18030";
     codec = TQTextCodec::codecForName(name);
 
     if(codec) {
