--- kdelibs-3.3.0/mimetypes/application/vnd.rn-realmedia.desktop.org	2004-08-23 20:04:21.485002096 +0800
+++ kdelibs-3.3.0/mimetypes/application/vnd.rn-realmedia.desktop	2004-08-23 20:04:57.605510944 +0800
@@ -3,7 +3,7 @@
 Type=MimeType
 MimeType=application/vnd.rn-realmedia
 Icon=video
-Patterns=*.rm;*.RM;
+Patterns=*.rm;*.RM;*.rmvb;*.RMVB;
 Comment=RealMedia File
 Comment[ar]=وسيط RealMedia
 Comment[az]=RealMedia Faylı
