diff -ur kdelibs-3.5.10.orig/kioslave/ftp/ftp.cc kdelibs-3.5.10/kioslave/ftp/ftp.cc
--- kdelibs-3.5.10.orig/kioslave/ftp/ftp.cc	2009-05-25 11:37:41.000000000 +0800
+++ kdelibs-3.5.10/kioslave/ftp/ftp.cc	2009-05-25 11:38:13.000000000 +0800
@@ -876,9 +876,9 @@
   // The usual answer is '227 Entering Passive Mode. (160,39,200,55,6,245)'
   // but anonftpd gives '227 =160,39,200,55,6,245'
   int i[6];
-  char *start = strchr(ftpResponse(3), '(');
+  char *start = (char *)strchr(ftpResponse(3), '(');
   if ( !start )
-    start = strchr(ftpResponse(3), '=');
+    start = (char *)strchr(ftpResponse(3), '=');
   if ( !start ||
        ( sscanf(start, "(%d,%d,%d,%d,%d,%d)",&i[0], &i[1], &i[2], &i[3], &i[4], &i[5]) != 6 &&
          sscanf(start, "=%d,%d,%d,%d,%d,%d", &i[0], &i[1], &i[2], &i[3], &i[4], &i[5]) != 6 ) )
@@ -931,7 +931,7 @@
     return ERR_INTERNAL;
   }
 
-  char *start = strchr(ftpResponse(3), '|');
+  char *start = (char *)strchr(ftpResponse(3), '|');
   if ( !start || sscanf(start, "|||%d|", &portnum) != 1)
     return ERR_INTERNAL;
 
