--- kdelibs-3.1.95/mimetypes/application/x-shockwave-flash.desktop.caps2	2004-01-17 06:53:25.000000000 -0600
+++ kdelibs-3.1.95/mimetypes/application/x-shockwave-flash.desktop	2004-01-19 12:32:41.000000000 -0600
@@ -3,7 +3,7 @@
 Type=MimeType
 MimeType=application/x-shockwave-flash
 Icon=multimedia
-Patterns=*.swf
+Patterns=*.swf;*.SWF;
 Comment=Shockwave Flash Media
 Comment[ar]=ملف Shockwave Flash
 Comment[az]=Shockwave Flash Mediyası
