diff -Nur kdelibs-3.5.4/tdecore/eventsrc kdelibs-3.5.4n/tdecore/eventsrc
--- kdelibs-3.5.4/tdecore/eventsrc	2006-07-22 04:16:41.000000000 -0400
+++ kdelibs-3.5.4n/tdecore/eventsrc	2006-07-24 21:16:59.428442648 -0400
@@ -1058,7 +1058,7 @@
 Comment[zh_HK]=程式發生了特殊情況
 Comment[zh_TW]=程式中發生了特殊的情況
 default_presentation=1
-default_sound=KDE_Beep.ogg
+default_sound=KDE_Beep.wav
 level=1
 
 [warning]
@@ -1490,7 +1490,7 @@
 Comment[zh_HK]=KDE 正在啟動
 Comment[zh_TW]=KDE 正在啟動
 default_presentation=1
-default_sound=KDE_Startup_1.ogg
+default_sound=KDE_Startup_1.wav
 
 [exitkde]
 Name=Logout
@@ -1631,7 +1631,7 @@
 Comment[zh_HK]=KDE 正在結束
 Comment[zh_TW]=KDE 正在結束
 default_presentation=1
-default_sound=KDE_Logout_3.ogg
+default_sound=KDE_Logout_3.wav
 
 [printerror]
 Name=Print error
@@ -1929,7 +1929,7 @@
 Comment[zh_CN]=正在显示信息性消息
 Comment[zh_HK]=顯示一般訊息
 Comment[zh_TW]=資訊訊息已顯示
-default_sound=KDE_Chimes_2.ogg
+default_sound=KDE_Chimes_2.wav
 default_presentation=65
 nopresentation=18
 
@@ -2075,7 +2075,7 @@
 Comment[zh_CN]=正在显示警告消息
 Comment[zh_HK]=顯示警告訊息
 Comment[zh_TW]=警告訊息已顯示
-default_sound=KDE_Error_1.ogg
+default_sound=KDE_Error_1.wav
 default_presentation=65
 nopresentation=18
 level=2
@@ -2367,7 +2367,7 @@
 Comment[zh_CN]=正在提问
 Comment[zh_HK]=顯示提問訊息
 Comment[zh_TW]=問題詢問中
-default_sound=KDE_Vox_Ahem.ogg
+default_sound=KDE_Vox_Ahem.wav
 default_presentation=65
 nopresentation=18
 
