diff -Nur kdelibs-3.5.1/tdeui/kdatepicker.cpp kdelibs-3.5.1new/tdeui/kdatepicker.cpp
--- kdelibs-3.5.1/tdeui/kdatepicker.cpp	2005-10-10 11:06:38.000000000 -0400
+++ kdelibs-3.5.1new/tdeui/kdatepicker.cpp	2006-03-06 00:51:00.101236560 -0500
@@ -46,6 +46,8 @@
 #include "kdatetbl.h"
 #include "kdatepicker.moc"
 
+#include <stdlib.h>
+#include <tqfile.h>
 // Week numbers are defined by ISO 8601
 // See http://www.merlyn.demon.co.uk/weekinfo.htm for details
 
@@ -249,7 +251,47 @@
 
     const KCalendarSystem * calendar = KGlobal::locale()->calendar();
 
+   // line->setText(KGlobal::locale()->formatDate(date, true));
+	   if(TQString(getenv("LANG")).upper().contains("ZH"))
+   {
+   if(TQFile("/usr/bin/ccal-kdatepicker").exists())
+   {
+   TQString cmd = TQString("ccal-kdatepicker -g ")+ TQString("%1").arg(date.month())+TQString(" ")+TQString("%1").arg(date.year());
+   FILE *pipe;
+//     TQString s;
+   fprintf(stderr,"the cmd is %s\n\n",cmd.latin1());
+   if ((pipe = popen(cmd.latin1(), "r")) == NULL) {
+      pclose(pipe);
+   }
+   TQTextStream *t=new TQTextStream(pipe, IO_ReadOnly);
+   int i = 0;
+   while (!t->atEnd()) {
+       TQString tmp = t->readLine();
+       if(tmp.stripWhiteSpace().length()>0)
+           {
+           i++;
+         fprintf(stderr,"list is %s\n",tmp.stripWhiteSpace().latin1());
+//             s += tmp.stripWhiteSpace();
+           if(i == date.day())
+               {
+                    line->setText(/*TQString::fromLocal8Bit(*/tmp.stripWhiteSpace()/*.latin1())*/);
+                    break;
+               }
+           }
+   }
+   pclose(pipe);
+   }
+
+   else
+   {
+        line->setText(KGlobal::locale()->formatDate(date, true));
+   }
+   }
+   else
+   {
     line->setText(KGlobal::locale()->formatDate(date, true));
+   }
+
     selectMonth->setText(calendar->monthName(date, false));
     fillWeeksCombo(date);
 
