diff -ur kdelibs-3.5.13.orig/kio/kio/kremoteencoding.cpp kdelibs-3.5.13/kio/kio/kremoteencoding.cpp
--- kdelibs-3.5.13.orig/kio/kio/kremoteencoding.cpp	2010-11-27 10:10:09.000000000 -0500
+++ kdelibs-3.5.13/kio/kio/kremoteencoding.cpp	2010-11-27 10:41:02.000000000 -0500
@@ -84,7 +84,10 @@
     codec = TQTextCodec::codecForMib( 106 ); // fallback to UTF-8
 
   if (codec == 0L)
-    codec = TQTextCodec::codecForMib(1);
+    codec = TQTextCodec::codecForLocale();
+
+  if ( TQCString(codec->name()).lower()=="gb18030")      
+    codec = TQTextCodec::codecForMib(113);
 
   kdDebug() << k_funcinfo << "setting encoding " << codec->name()
 	    << " for name=" << name << endl;
