--- xorg-cf-files-0.99.2/site.def.orig	2005-11-28 12:18:47.000000000 +0100
+++ xorg-cf-files-1.0.0/site.def	2005-11-28 12:21:06.000000000 +0100
@@ -72,7 +72,7 @@
 #ifdef AfterVendorCF
 
 #ifndef ProjectRoot
-#define ProjectRoot /usr/X11R6
+#define ProjectRoot /usr
 #endif
 
 /*
--- xorg-cf-files-0.99.2/X11.tmpl.orig	2005-11-28 12:19:03.000000000 +0100
+++ xorg-cf-files-1.0.0/X11.tmpl	2005-11-28 12:31:56.000000000 +0100
@@ -3386,7 +3386,7 @@
 #if HasMotif
 
 #ifndef MotifDir
-#define MotifDir /usr/X11R6
+#define MotifDir /usr
 #endif
 #ifndef MotifDirStandard
 #define MotifDirStandard NO
