diff -up glew-1.9.0/config/Makefile.linux.jx glew-1.9.0/config/Makefile.linux
--- glew-1.9.0/config/Makefile.linux.jx	2012-08-06 11:59:08.000000000 -0400
+++ glew-1.9.0/config/Makefile.linux	2012-12-13 13:03:05.044983152 -0500
@@ -3,14 +3,8 @@ CC = cc
 LD = cc
 PICFLAG = -fPIC
 M_ARCH ?= $(shell uname -m)
-ifeq (x86_64,${M_ARCH})
-LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
-LIBDIR = $(GLEW_DEST)/lib64
-else
-LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
-LIBDIR = $(GLEW_DEST)/lib
-endif
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LIBDIR = $(GLEW_DEST)$(libdir)
+LDFLAGS.GL = -lGLU -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 NAME = GLEW
diff -up glew-1.9.0/Makefile.jx glew-1.9.0/Makefile
--- glew-1.9.0/Makefile.jx	2012-12-13 13:00:59.000000000 -0500
+++ glew-1.9.0/Makefile	2012-12-13 13:04:58.027629027 -0500
@@ -41,9 +41,9 @@ $(error "Platform '$(SYSTEM)' not suppor
 endif
 
 GLEW_DEST ?= /usr
-BINDIR    ?= $(GLEW_DEST)/bin
-LIBDIR    ?= $(GLEW_DEST)/lib
-INCDIR    ?= $(GLEW_DEST)/include/GL
+BINDIR    ?= $(GLEW_DEST)$(bindir)
+LIBDIR    ?= $(GLEW_DEST)$(libdir)
+INCDIR    ?= $(GLEW_DEST)$(includedir)/GL
 
 TARDIR = ../glew-$(GLEW_VERSION)
 TARBALL = ../glew-$(GLEW_VERSION).tar.gz
@@ -224,7 +224,7 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTE
 	$(INSTALL) -d -m 0755 $(BINDIR)
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED) $(BINDIR)/
 else
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED) $(LIBDIR)/
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED) $(LIBDIR)/
 endif
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.SONAME)
@@ -246,7 +246,7 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTE
 	$(INSTALL) -d -m 0755 $(BINDIR)
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED.MX) $(BINDIR)/
 else
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED.MX) $(LIBDIR)/
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED.MX) $(LIBDIR)/
 endif
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED.MX) $(LIBDIR)/$(LIB.SONAME.MX)
