diff -uNr pnm2ppa-0.99pre3/pnm2ppa.conf pnm2ppa-0.99pre3-redhat/pnm2ppa.conf
--- pnm2ppa-0.99pre3/pnm2ppa.conf	Fri May 26 01:35:52 2000
+++ pnm2ppa-0.99pre3-redhat/pnm2ppa.conf	Fri May 26 04:03:56 2000
@@ -12,7 +12,6 @@
 # will be used.   The printer version can also be set with the command line
 # option e.g., "-v 720".
 
-version   ERROR    # PNM2PPA WIIL *NOT* WORK UNTIUL THIS ENTRY IS REMOVED!!!!
 #version  720	# 710, 712, 722 also acceptable
 #version  820
 #version 1000
--- pnm2ppa/pbm2ppa-0.8.6/Makefile.rh	Thu Feb 10 15:28:55 2000
+++ pnm2ppa/pbm2ppa-0.8.6/Makefile	Thu Feb 10 15:30:06 2000
@@ -1,5 +1,5 @@
 CC=gcc
-CFLAGS=-Wall -O2 
+CFLAGS=$(RPM_OPT_FLAGS)
 LDFLAGS=
 PRINTER=HP720
 
--- pnm2ppa/Makefile.rh	Thu Feb 10 15:29:06 2000
+++ pnm2ppa/Makefile	Thu Feb 10 15:29:39 2000
@@ -9,7 +9,7 @@
 INSTALL=install
 
 # production build
-CFLAGS=-Wall -pedantic -O2 -g -DNDEBUG 
+CFLAGS=$(RPM_OPT_FLAGS) -DNDEBUG 
 LDFLAGS=-lm
 OBJS = pnm2ppa.o ppa.o image.o dither.o cutswath.o hash_ink.o
 SRC =  pnm3ppa.c ppa.c image.c dither.c cutswath.c hash_ink.c
--- pnm2ppa.conf	Tue Dec 12 05:13:59 2000
+++ foo	Tue Dec 12 05:15:57 2000

