---
 build/source/Build |   22 ++++++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

Index: texlive-bin-2007/build/source/Build
===================================================================
--- texlive-bin-2007.orig/build/source/Build	2006-12-14 03:12:38.000000000 +0100
+++ texlive-bin-2007/build/source/Build	2007-02-14 15:56:00.000000000 +0100
@@ -47,11 +47,25 @@
 
 # Force building xdvi-xaw; motif is not always free and/or available.
 # --enable-gif is for sam2p; the gif patent has expired worldwide.
-(set -x; \
-  time $TL_CONFIGURE \
+case $1 in
+  configure)
+    shift
+    (set -x; \
+      $TL_CONFIGURE \
          --prefix=$TL_INSTALL_DEST \
          --datadir=$TL_INSTALL_DEST \
          $TL_XDVI_TOOLKIT \
          $TL_CONFIGURE_ARGS \
-         "$@" \
- && time $TL_MAKE $TL_TARGET)
+         "$@" )
+    ;;
+  build)
+    # not make world, this would strip
+    shift
+    make all
+    make install
+    ;;
+  *)
+    echo "This script must be invoked with argument configure or build"
+    ;;
+esac
+
