diff -up texlive-2007/texk/web2c/omegafonts/Makefile.in.makej texlive-2007/texk/web2c/omegafonts/Makefile.in
--- texlive-2007/texk/web2c/omegafonts/Makefile.in.makej	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/web2c/omegafonts/Makefile.in	2008-08-27 13:28:57.000000000 +0200
@@ -34,12 +34,13 @@ omfonts: $(omfonts_objects)
 	$(kpathsea_link) $(omfonts_objects) $(LEXLIB) $(LOADLIBES) 
 
 omfonts.o: omfonts.c y_tab.h
-y_tab.c y_tab.h: pl.y
+y_tab.c: y_tab.h
+y_tab.h: pl.y
 	$(YACC) -d -v $(srcdir)/pl.y
 	-test -f y.tab.c && mv -f y.tab.c y_tab.c
 	-test -f y.tab.h && mv -f y.tab.h y_tab.h
 $(LEX_OUTPUT_ROOT).o: parser.h
-$(LEX_OUTPUT_ROOT).c: pl.l
+$(LEX_OUTPUT_ROOT).c: y_tab.h pl.l
 	$(LEX) $(srcdir)/pl.l
 
 
diff -up texlive-2007/texk/web2c/otps/Makefile.in.makej texlive-2007/texk/web2c/otps/Makefile.in
--- texlive-2007/texk/web2c/otps/Makefile.in.makej	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/web2c/otps/Makefile.in	2008-08-27 13:28:57.000000000 +0200
@@ -21,12 +21,13 @@ otp2ocp: $(otp2ocp_objects)
 	$(kpathsea_link) $(otp2ocp_objects) $(LEXLIB) $(LOADLIBES) 
 
 otp2ocp.o: otp2ocp.c y_tab.h
-y_tab.c y_tab.h: otp.y
+y_tab.c: y_tab.h
+y_tab.h: otp.y
 	$(YACC) -d -v $(srcdir)/otp.y
 	-test -f y.tab.c && mv -f y.tab.c y_tab.c
 	-test -f y.tab.h && mv -f y.tab.h y_tab.h
 $(LEX_OUTPUT_ROOT).o: otp.h
-$(LEX_OUTPUT_ROOT).c: otp.l
+$(LEX_OUTPUT_ROOT).c: y_tab.h otp.l
 	$(LEX) $(srcdir)/otp.l
 
 outocp: outocp.o
diff -up texlive-2007/texk/web2c/web2c/Makefile.in.makej texlive-2007/texk/web2c/web2c/Makefile.in
--- texlive-2007/texk/web2c/web2c/Makefile.in.makej	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/web2c/web2c/Makefile.in	2008-08-27 13:28:57.000000000 +0200
@@ -25,13 +25,14 @@ web2c: $(web2c_objects)
 
 # We use y_tab.* for the sake of MS-DOS.
 main.o: main.c y_tab.h
-y_tab.c y_tab.h: web2c.y
+y_tab.c: y_tab.h
+y_tab.h: web2c.y
 	@echo Expect one shift/reduce conflict.
 	$(YACC) -d -v $(srcdir)/web2c.y
 	-test -f y.tab.c && mv -f y.tab.c y_tab.c
 	-test -f y.tab.h && mv -f y.tab.h y_tab.h
 $(LEX_OUTPUT_ROOT).o: y_tab.h
-$(LEX_OUTPUT_ROOT).c: web2c.l
+$(LEX_OUTPUT_ROOT).c: y_tab.h web2c.l
 	$(LEX) $(srcdir)/web2c.l
 
 fixwrites: fixwrites.o kps.o
