diff -ur tetex-src-3.0/texk/tetex/texlinks.man tetex-src-3.0-copyright-utf8-man/texk/tetex/texlinks.man
--- tetex-src-3.0/texk/tetex/texlinks.man	2005-01-20 08:58:46.000000000 -0800
+++ tetex-src-3.0-copyright-utf8-man/texk/tetex/texlinks.man	2006-05-12 01:35:35.000000000 -0700
@@ -54,7 +54,7 @@
 .SH "AUTHOR" 
 .PP 
 \fBtexlinks\fP was written by Thomas Esser 
-<te@dbs\&.uni-hannover\&.de> and is Copyright © 1999\&.
+<te@dbs\&.uni-hannover\&.de> and is Copyright Â© 1999\&.
 \fBtexlinks\fP has been released into the public domain\&.
 .PP 
 This manual page was written by C\&.M\&. Connelly
diff -ur tetex-src-3.0/texk/tetex/thumbpdf.man tetex-src-3.0-copyright-utf8-man/texk/tetex/thumbpdf.man
--- tetex-src-3.0/texk/tetex/thumbpdf.man	2005-01-20 08:50:12.000000000 -0800
+++ tetex-src-3.0-copyright-utf8-man/texk/tetex/thumbpdf.man	2006-05-12 01:35:58.000000000 -0700
@@ -83,7 +83,7 @@
 .PP 
 .SH "COPYRIGHT" 
 .PP 
-Copyright © 1999\&.  \fBthumbpdf\fP is distributed under
+Copyright Â© 1999\&.  \fBthumbpdf\fP is distributed under
 version 1, or (at your option) a later version of the LaTeX Project
 Public License\&.
 .PP 
