diff -up texlive-2007/configure.t1lib texlive-2007/configure
--- texlive-2007/configure.t1lib	2008-08-27 13:37:53.000000000 +0200
+++ texlive-2007/configure	2008-08-27 13:37:53.000000000 +0200
@@ -4097,6 +4097,7 @@ int main() {
     char *p = testbuf;
     int i;
     int version_num;
+    memset(p, 0, sizeof(testbuf));
     /* T1_GetLibIdent() might return something like '0.8-beta2';
        convert the leading numbers to a single digit: */
     for (i = 0; i < BUFLEN - 1; i++) {
diff -up texlive-2007/libs/configure.t1lib texlive-2007/libs/configure
--- texlive-2007/libs/configure.t1lib	2008-08-27 13:37:53.000000000 +0200
+++ texlive-2007/libs/configure	2008-08-27 13:37:53.000000000 +0200
@@ -1757,6 +1757,7 @@ int main() {
     char *p = testbuf;
     int i;
     int version_num;
+    memset(p, 0, sizeof(testbuf));
     /* T1_GetLibIdent() might return something like '0.8-beta2';
        convert the leading numbers to a single digit: */
     for (i = 0; i < BUFLEN - 1; i++) {
diff -up texlive-2007/libs/t1lib/t1lib.ac.t1lib texlive-2007/libs/t1lib/t1lib.ac
--- texlive-2007/libs/t1lib/t1lib.ac.t1lib	2006-01-16 01:09:26.000000000 +0100
+++ texlive-2007/libs/t1lib/t1lib.ac	2008-08-27 13:37:53.000000000 +0200
@@ -45,6 +45,7 @@ int main() {
     char *p = testbuf;
     int i;
     int version_num;
+    memset(p, 0, sizeof(testbuf));
     /* T1_GetLibIdent() might return something like '0.8-beta2';
        convert the leading numbers to a single digit: */
     for (i = 0; i < BUFLEN - 1; i++) {
diff -up texlive-2007/texk/dvipng/configure.t1lib texlive-2007/texk/dvipng/configure
--- texlive-2007/texk/dvipng/configure.t1lib	2007-01-04 17:14:52.000000000 +0100
+++ texlive-2007/texk/dvipng/configure	2008-08-27 13:37:53.000000000 +0200
@@ -5680,6 +5680,7 @@ int main() {
     char *p = testbuf;
     int i;
     int version_num;
+    memset(p, 0, sizeof(testbuf));
     /* T1_GetLibIdent() might return something like '0.8-beta2';
        convert the leading numbers to a single digit: */
     for (i = 0; i < BUFLEN - 1; i++) {
diff -up texlive-2007/texk/xdvik/configure.t1lib texlive-2007/texk/xdvik/configure
--- texlive-2007/texk/xdvik/configure.t1lib	2006-07-16 19:26:38.000000000 +0200
+++ texlive-2007/texk/xdvik/configure	2008-08-27 13:37:53.000000000 +0200
@@ -4280,6 +4280,7 @@ int main() {
     char *p = testbuf;
     int i;
     int version_num;
+    memset(p, 0, sizeof(testbuf));
     /* T1_GetLibIdent() might return something like '0.8-beta2';
        convert the leading numbers to a single digit: */
     for (i = 0; i < BUFLEN - 1; i++) {
