diff -up texlive/texk/make/paths.mk.fedora_paths texlive/texk/make/paths.mk
--- texlive/texk/make/paths.mk.fedora_paths	2007-08-19 08:46:09.000000000 +0200
+++ texlive/texk/make/paths.mk	2007-08-19 08:55:17.000000000 +0200
@@ -59,7 +59,7 @@ libexecdir = @libexecdir@
 texmf = @texmfmain@
 
 # The directory used by varfonts.
-vartexfonts = /var/tmp/texfonts
+vartexfonts = $(localstatedir)/cache/fonts
 
 # Regular input files.
 texinputdir = $(texmf)/tex
diff -up texlive/texk/kpathsea/texmf.in.fedora_paths texlive/texk/kpathsea/texmf.in
--- texlive/texk/kpathsea/texmf.in.fedora_paths	2007-08-19 08:47:04.000000000 +0200
+++ texlive/texk/kpathsea/texmf.in	2007-08-21 14:16:49.000000000 +0200
@@ -59,12 +59,12 @@
 TEXMFMAIN = @texmf@
 
 % The main distribution tree:
-TEXMFDIST = @texmf@-dist
+TEXMFDIST = @texmf@
 
 % A place for local additions to a "standard" texmf tree.
 % This tree is not used for local configuration maintained by
 % texconfig, it uses TEXMFCONFIG below.
-TEXMFLOCAL = @texmf@-local
+TEXMFLOCAL = /usr/local/share/texmf
 
 % TEXMFSYSVAR, where texconfig-sys stores variable runtime data.
 % With teTeX-3.0 or later, this must be set.
@@ -72,7 +72,7 @@ TEXMFLOCAL = @texmf@-local
 %   TEXMFSYSVAR = $TEXMFMAIN
 % For using a separate tree:
 %   TEXMFSYSVAR = @texmf@-var
-TEXMFSYSVAR = @texmf@-var
+TEXMFSYSVAR = @localstatedir@/lib/texmf
 
 % TEXMFSYSCONFIG, where texconfig-sys stores configuration data.
 % With teTeX-3.0 or later, this must be set.
@@ -80,7 +80,7 @@ TEXMFSYSVAR = @texmf@-var
 %   TEXMFSYSCONFIG = $TEXMFMAIN
 % For using a separate tree:
 %   TEXMFSYSCONFIG = @texmf@-config
-TEXMFSYSCONFIG = @texmf@-config
+TEXMFSYSCONFIG = @sysconfdir@/texmf
 
 % User texmf trees are allowed as follows.
 % This used to be HOMETEXMF.
@@ -111,7 +111,7 @@ TEXMFCONFIG = $HOME/.texmf-config
 %
 % For texconfig to work properly, TEXMFCONFIG and TEXMFVAR should be named
 % explicitly and before all other trees.
-TEXMF = {$TEXMFCONFIG,$TEXMFVAR,$TEXMFHOME,!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFMAIN,!!$TEXMFLOCAL,!!$TEXMFDIST}
+TEXMF = {$TEXMFCONFIG,$TEXMFVAR,$TEXMFHOME,$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFLOCAL,!!$TEXMFMAIN,!!$TEXMFDIST}
 
 % The system trees.  These are the trees that are shared by all the users.
 % If a tree appears in this list, the mktex* scripts will use
@@ -379,7 +379,7 @@ TEXMFSCRIPTS = $TEXMF/scripts//
 %
 % For security reasons, it is better not to have . part of the path.
 %
-TEXMFCNF = {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT}{,{/share,}/texmf{-local,}/web2c};/.$TETEXDIR;$TEXMF/web2c;@web2c@
+TEXMFCNF = {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT}{,{/share,}/texmf-local/web2c};@sysconfdir@/texmf/web2c;/usr/local/share/texmf/web2c;$TEXMF/web2c;@web2c@
 
 # $progname: kpathsea v. 3.5.3 or later overwrites this at runtime. To
 # avoid empty expansions from binaries linked against an earlier
