diff -up texlive-2007/texk/web2c/xetexdir/xetex.mk.system_teckit texlive-2007/texk/web2c/xetexdir/xetex.mk
--- texlive-2007/texk/web2c/xetexdir/xetex.mk.system_teckit	2007-08-18 16:29:37.000000000 +0200
+++ texlive-2007/texk/web2c/xetexdir/xetex.mk	2007-08-18 16:30:06.000000000 +0200
@@ -56,8 +56,8 @@ FREETYPE2DEP = @FREETYPE2DEP@
 FREETYPE2DIR = ../../libs/freetype2
 FREETYPE2SRCDIR = $(srcdir)/$(FREETYPE2DIR)
 
-TECKITFLAGS = @TECKITCPPFLAGS@
-LDTECKIT = @LDTECKIT@
+TECKITFLAGS = -I/usr/include/teckit
+LDTECKIT = -lTECkit
 TECKITDEP = @TECKITDEP@
 
 TECKITDIR=../../libs/teckit
