--- libs/configure
+++ libs/configure	2007-02-07 12:21:24.000000000 +0000
@@ -2976,11 +2976,11 @@
   fi
 
   OLD_LDFLAGS=$LDFLAGS
-  OLD_CPP=$CPP
+  OLD_CPPFLAGS=$CPPFLAGS
   OLD_LIBS=$LIBS
 
   LDFLAGS="$LDFLAGS $EXTRA_ICU_LIBDIR"
-  CPP="$CPP $EXTRA_ICU_INCLUDE"
+  CPPFLAGS="$CPPFLAGS $EXTRA_ICU_INCLUDE -include urename.h"
 
   if test "$with_system_icu" = yes; then
     echo $ac_n "checking for ubidi_open in -licuuc""... $ac_c" 1>&6
@@ -3082,7 +3082,7 @@
   fi
 
   LDFLAGS=$OLD_LDFLAGS
-  CPP=$OLD_CPP
+  CPPFLAGS=$OLD_CPPFLAGS
   LIBS=$OLD_LIBS
 
   if test "$with_system_icu" != yes; then
--- libs/icu-xetex/icu-xetex.ac
+++ libs/icu-xetex/icu-xetex.ac	2007-02-07 12:21:39.000000000 +0000
@@ -16,11 +16,11 @@
   fi
 
   OLD_LDFLAGS=$LDFLAGS
-  OLD_CPP=$CPP
+  OLD_CPPFLAGS=$CPPFLAGS
   OLD_LIBS=$LIBS
 
   LDFLAGS="$LDFLAGS $EXTRA_ICU_LIBDIR"
-  CPP="$CPP $EXTRA_ICU_INCLUDE"
+  CPPFLAGS="$CPPFLAGS $EXTRA_ICU_INCLUDE -include urename.h"
 
   if test "$with_system_icu" = yes; then
     AC_CHECK_LIB(icuuc, ubidi_open,,
@@ -37,7 +37,7 @@
   fi
 
   LDFLAGS=$OLD_LDFLAGS
-  CPP=$OLD_CPP
+  CPPFLAGS=$OLD_CPPFLAGS
   LIBS=$OLD_LIBS
 
   if test "$with_system_icu" != yes; then
--- texk/web2c/configure
+++ texk/web2c/configure	2007-02-07 12:21:55.000000000 +0000
@@ -3018,11 +3018,11 @@
   fi
 
   OLD_LDFLAGS=$LDFLAGS
-  OLD_CPP=$CPP
+  OLD_CPPFLAGS=$CPPFLAGS
   OLD_LIBS=$LIBS
 
   LDFLAGS="$LDFLAGS $EXTRA_ICU_LIBDIR"
-  CPP="$CPP $EXTRA_ICU_INCLUDE"
+  CPPFLAGS="$CPPFLAGS $EXTRA_ICU_INCLUDE -include urename.h"
 
   if test "$with_system_icu" = yes; then
     echo $ac_n "checking for ubidi_open in -licuuc""... $ac_c" 1>&6
@@ -3124,7 +3124,7 @@
   fi
 
   LDFLAGS=$OLD_LDFLAGS
-  CPP=$OLD_CPP
+  CPPFLAGS=$OLD_CPPFLAGS
   LIBS=$OLD_LIBS
 
   if test "$with_system_icu" != yes; then
--- configure
+++ configure	2007-02-07 12:22:06.000000000 +0000
@@ -4662,11 +4662,11 @@
   fi
 
   OLD_LDFLAGS=$LDFLAGS
-  OLD_CPP=$CPP
+  OLD_CPPFLAGS=$CPPFLAGS
   OLD_LIBS=$LIBS
 
   LDFLAGS="$LDFLAGS $EXTRA_ICU_LIBDIR"
-  CPP="$CPP $EXTRA_ICU_INCLUDE"
+  CPPFLAGS="$CPPFLAGS $EXTRA_ICU_INCLUDE -include urename.h"
 
   if test "$with_system_icu" = yes; then
     echo $ac_n "checking for ubidi_open in -licuuc""... $ac_c" 1>&6
@@ -4768,7 +4768,7 @@
   fi
 
   LDFLAGS=$OLD_LDFLAGS
-  CPP=$OLD_CPP
+  CPPFLAGS=$OLD_CPPFLAGS
   LIBS=$OLD_LIBS
 
   if test "$with_system_icu" != yes; then
