--- texk/kpathsea/tex-file.c.kpse-extensions	2006-05-27 21:43:07.000000000 -0700
+++ texk/kpathsea/tex-file.c	2006-05-27 21:40:06.000000000 -0700
@@ -413,14 +413,16 @@
          times with token pasting, but it doesn't seem worth it.  */
     case kpse_gf_format:
       INIT_FORMAT ("gf", DEFAULT_GFFONTS, GF_ENVS);
-      SUFFIXES ("gf");
+#define GF_SUFFIXES "gf", "GF"
+      SUFFIXES (GF_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       FMT_INFO.binmode = true;
       break;
     case kpse_pk_format:
       init_maketex (format, "mktexpk", MKTEXPK_ARGS);
       INIT_FORMAT ("pk", DEFAULT_PKFONTS, PK_ENVS);
-      SUFFIXES ("pk");
+#define PK_SUFFIXES "pk", "PK"
+      SUFFIXES (PK_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       FMT_INFO.binmode = true;
       break;
@@ -434,77 +436,92 @@
       /* Must come before kpse_ofm_format. */
       init_maketex (format, "mktextfm", NULL);
       INIT_FORMAT ("tfm", DEFAULT_TFMFONTS, TFM_ENVS);
-      SUFFIXES (".tfm");
+#define TFM_SUFFIXES ".tfm", ".TFM"
+      SUFFIXES (TFM_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       FMT_INFO.binmode = true;
       break;
     case kpse_afm_format:
       INIT_FORMAT ("afm", DEFAULT_AFMFONTS, AFM_ENVS);
-      SUFFIXES (".afm");
+#define AFM_SUFFIXES ".afm", ".AFM"
+      SUFFIXES (AFM_SUFFIXES);
       break;
     case kpse_base_format:
       init_maketex (format, "mktexfmt", NULL);
       INIT_FORMAT ("base", DEFAULT_MFBASES, BASE_ENVS);
-      SUFFIXES (".base");
+#define BASE_SUFFIXES ".base", ".BASE"
+      SUFFIXES (BASE_SUFFIXES);
       FMT_INFO.binmode = true;
       break;
     case kpse_bib_format:
       INIT_FORMAT ("bib", DEFAULT_BIBINPUTS, BIB_ENVS);
-      SUFFIXES (".bib");
+#define BIB_SUFFIXES ".bib", ".BIB"
+      SUFFIXES (BIB_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       break;
     case kpse_bst_format:
       INIT_FORMAT ("bst", DEFAULT_BSTINPUTS, BST_ENVS);
-      SUFFIXES (".bst");
+#define BST_SUFFIXES ".bst", ".BST"
+      SUFFIXES (BST_SUFFIXES);
       break;
     case kpse_cnf_format:
       INIT_FORMAT ("cnf", DEFAULT_TEXMFCNF, CNF_ENVS);
-      SUFFIXES (".cnf");
+#define CNF_SUFFIXES ".cnf", ".CNF"
+      SUFFIXES (CNF_SUFFIXES);
       break;
     case kpse_db_format:
       INIT_FORMAT ("ls-R", DEFAULT_TEXMFDBS, DB_ENVS);
-#define LSR_SUFFIXES "ls-R", "ls-r"
+#define LSR_SUFFIXES "ls-R", "ls-r", "LS-R"
       SUFFIXES (LSR_SUFFIXES);
       FMT_INFO.path = remove_dbonly (FMT_INFO.path);
       break;
     case kpse_fmt_format:
       init_maketex (format, "mktexfmt", NULL);
       INIT_FORMAT ("fmt", DEFAULT_TEXFORMATS, FMT_ENVS);
-      SUFFIXES (".fmt");
-#define FMT_SUFFIXES ".efmt",".efm",".ofmt",".ofm",".oft",".eofmt",".eoft",".eof",".pfmt",".pfm",".epfmt",".epf",".xpfmt",".xpf",".afmt",".afm"
-      ALT_SUFFIXES (FMT_SUFFIXES);
+#define FMT_SUFFIXES ".fmt", ".FMT"
+      SUFFIXES (FMT_SUFFIXES);
+#define FMT_ALT_SUFFIXES ".efmt",".efm",".ofmt",".ofm",".oft",".eofmt",".eoft",".eof",".pfmt",".pfm",".epfmt",".epf",".xpfmt",".xpf",".afmt",".afm" \
+			 ".EFMT",".EFM",".OFMT",".OFM",".OFT",".EOFMT",".EOFT",".EOF",".PFMT",".PFM",".EPFMT",".EPF",".XPFMT",".XPF",".AFMT",".AFM"
+      ALT_SUFFIXES (FMT_ALT_SUFFIXES);
       FMT_INFO.binmode = true;
       break;
     case kpse_fontmap_format:
       INIT_FORMAT ("map", DEFAULT_TEXFONTMAPS, FONTMAP_ENVS);
-      SUFFIXES (".map");
+#define MAP_SUFFIXES ".map", ".MAP"
+      SUFFIXES (MAP_SUFFIXES);
       break;
     case kpse_mem_format:
       init_maketex (format, "mktexfmt", NULL);
       INIT_FORMAT ("mem", DEFAULT_MPMEMS, MEM_ENVS);
-      SUFFIXES (".mem");
+#define MEM_SUFFIXES ".mem", ".MEM"
+      SUFFIXES (MEM_SUFFIXES);
       FMT_INFO.binmode = true;
       break;
     case kpse_mf_format:
       init_maketex (format, "mktexmf", NULL);
       INIT_FORMAT ("mf", DEFAULT_MFINPUTS, MF_ENVS);
-      SUFFIXES (".mf");
+#define MF_SUFFIXES ".mf", ".MF"
+      SUFFIXES (MF_SUFFIXES);
       break;
     case kpse_mft_format:
       INIT_FORMAT ("mft", DEFAULT_MFTINPUTS, MFT_ENVS);
-      SUFFIXES (".mft");
+#define MFT_SUFFIXES ".mft", ".MFT"
+      SUFFIXES (MFT_SUFFIXES);
       break;
     case kpse_mfpool_format:
       INIT_FORMAT ("mfpool", DEFAULT_MFPOOL, MFPOOL_ENVS);
-      SUFFIXES (".pool");
+#define MFPOOL_SUFFIXES ".pool", ".POOL"
+      SUFFIXES (MFPOOL_SUFFIXES);
       break;
     case kpse_mp_format:
       INIT_FORMAT ("mp", DEFAULT_MPINPUTS, MP_ENVS);
-      SUFFIXES (".mp");
+#define MP_SUFFIXES ".mp", ".MP"
+      SUFFIXES (MP_SUFFIXES);
       break;
     case kpse_mppool_format:
       INIT_FORMAT ("mppool", DEFAULT_MPPOOL, MPPOOL_ENVS);
-      SUFFIXES (".pool");
+#define MPPOOL_SUFFIXES ".pool", ".POOL"
+      SUFFIXES (MPPOOL_SUFFIXES);
       break;
     case kpse_mpsupport_format:
       INIT_FORMAT ("MetaPost support", DEFAULT_MPSUPPORT, MPSUPPORT_ENVS);
@@ -512,49 +529,55 @@
     case kpse_ocp_format:
       init_maketex (format, "mkocp", NULL);
       INIT_FORMAT ("ocp", DEFAULT_OCPINPUTS, OCP_ENVS);
-      SUFFIXES (".ocp");
+#define OCP_SUFFIXES ".ocp", ".OCP"
+      SUFFIXES (OCP_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       FMT_INFO.binmode = true;
       break;
     case kpse_ofm_format:
       init_maketex (format, "mkofm", NULL);
       INIT_FORMAT ("ofm", DEFAULT_OFMFONTS, OFM_ENVS);
-#define OFM_SUFFIXES ".ofm", ".tfm"
+#define OFM_SUFFIXES ".ofm", ".tfm", ".OFM", ".TFM"
       SUFFIXES (OFM_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       FMT_INFO.binmode = true;
       break;
     case kpse_opl_format:
       INIT_FORMAT ("opl", DEFAULT_OPLFONTS, OPL_ENVS);
-      SUFFIXES (".opl");
+#define OPL_SUFFIXES ".opl", ".OPL"
+      SUFFIXES (OPL_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       break;
     case kpse_otp_format:
       INIT_FORMAT ("otp", DEFAULT_OTPINPUTS, OTP_ENVS);
-      SUFFIXES (".otp");
+#define OTP_SUFFIXES ".otp", ".OTP"
+      SUFFIXES (OTP_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       break;
     case kpse_ovf_format:
       INIT_FORMAT ("ovf", DEFAULT_OVFFONTS, OVF_ENVS);
-      SUFFIXES (".ovf");
+#define OVF_SUFFIXES ".ovf", ".OVF"
+      SUFFIXES (OVF_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       FMT_INFO.binmode = true;
       break;
     case kpse_ovp_format:
       INIT_FORMAT ("ovp", DEFAULT_OVPFONTS, OVP_ENVS);
-      SUFFIXES (".ovp");
+#define OVP_SUFFIXES ".ovp", ".OVP"
+      SUFFIXES (OVP_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       break;
     case kpse_pict_format:
       INIT_FORMAT ("graphic/figure", DEFAULT_TEXINPUTS, PICT_ENVS);
-#define PICT_SUFFIXES ".eps", ".epsi"
+#define PICT_SUFFIXES ".eps", ".epsi", ".EPS", ".EPSI"
       ALT_SUFFIXES (PICT_SUFFIXES);
       FMT_INFO.binmode = true;
       break;
     case kpse_tex_format:
       init_maketex (format, "mktextex", NULL);
       INIT_FORMAT ("tex", DEFAULT_TEXINPUTS, TEX_ENVS);
-      SUFFIXES (".tex");
+#define TEX_SUFFIXES ".tex", ".TEX"
+      SUFFIXES (TEX_SUFFIXES);
       /* We don't maintain a list of alternate TeX suffixes.  Such a list
          could never be complete.  */
       break;
@@ -562,7 +585,7 @@
       INIT_FORMAT ("PostScript header", DEFAULT_TEXPSHEADERS,
                    TEX_PS_HEADER_ENVS);
 /* Unfortunately, dvipsk uses this format for type1 fonts.  */
-#define TEXPSHEADER_SUFFIXES ".pro"
+#define TEXPSHEADER_SUFFIXES ".pro", ".PRO"
       ALT_SUFFIXES (TEXPSHEADER_SUFFIXES);
       FMT_INFO.binmode = true;
       break;
@@ -571,7 +594,8 @@
       break;
     case kpse_texpool_format:
       INIT_FORMAT ("texpool", DEFAULT_TEXPOOL, TEXPOOL_ENVS);
-      SUFFIXES (".pool");
+#define TEXPOOL_SUFFIXES ".pool", ".POOL"
+      SUFFIXES (TEXPOOL_SUFFIXES);
       break;
     case kpse_texsource_format:
       INIT_FORMAT ("TeX system sources", DEFAULT_TEXSOURCES, TEXSOURCE_ENVS);
@@ -582,13 +606,14 @@
       break;
     case kpse_type1_format:
       INIT_FORMAT ("type1 fonts", DEFAULT_T1FONTS, TYPE1_ENVS);
-#define TYPE1_SUFFIXES ".pfa", ".pfb"
+#define TYPE1_SUFFIXES ".pfa", ".pfb", ".PFA", ".PFB"
       SUFFIXES (TYPE1_SUFFIXES);
       FMT_INFO.binmode = true;
       break;
     case kpse_vf_format:
       INIT_FORMAT ("vf", DEFAULT_VFFONTS, VF_ENVS);
-      SUFFIXES (".vf");
+#define VF_SUFFIXES ".vf", ".VF"
+      SUFFIXES (VF_SUFFIXES);
       FMT_INFO.suffix_search_only = true;
       FMT_INFO.binmode = true;
       break;
@@ -597,7 +622,8 @@
       break;
     case kpse_ist_format:
       INIT_FORMAT ("ist", DEFAULT_INDEXSTYLE, IST_ENVS);
-      SUFFIXES (".ist");
+#define IST_SUFFIXES ".ist", ".IST"
+      SUFFIXES (IST_SUFFIXES);
       break;
     case kpse_truetype_format:
       INIT_FORMAT ("truetype fonts", DEFAULT_TTFONTS, TRUETYPE_ENVS);
@@ -632,26 +658,32 @@
       break;
     case kpse_web_format:
       INIT_FORMAT ("web", DEFAULT_WEBINPUTS, WEB_ENVS);
-      SUFFIXES (".web");
-      ALT_SUFFIXES (".ch");
+#define WEB_SUFFIXES ".web", ".WEB"
+      SUFFIXES (WEB_SUFFIXES);
+#define WEB_ALT_SUFFIXES ".ch", ".CH"
+      ALT_SUFFIXES (WEB_ALT_SUFFIXES);
       break;
     case kpse_cweb_format:
       INIT_FORMAT ("cweb", DEFAULT_CWEBINPUTS, CWEB_ENVS);
-#define CWEB_SUFFIXES ".w", ".web"
+#define CWEB_SUFFIXES ".w", ".web", ".W", ".WEB"
       SUFFIXES (CWEB_SUFFIXES);
-      ALT_SUFFIXES (".ch");
+#define CWEB_ALT_SUFFIXES ".ch", ".CH"
+      ALT_SUFFIXES (CWEB_ALT_SUFFIXES);
       break;
     case kpse_enc_format:
       INIT_FORMAT ("enc files", DEFAULT_ENCFONTS, ENC_ENVS);
-      SUFFIXES (".enc");
+#define ENC_SUFFIXES ".enc", ".ENC"
+      SUFFIXES (ENC_SUFFIXES);
       break;
     case kpse_cmap_format:
       INIT_FORMAT ("cmap files", DEFAULT_CMAPFONTS, CMAP_ENVS);
-      SUFFIXES (".cmap");      
+#define CMAP_SUFFIXES ".cmap", ".CMAP"
+      SUFFIXES (CMAP_SUFFIXES);
       break;
     case kpse_sfd_format:
       INIT_FORMAT ("subfont definition files", DEFAULT_SFDFONTS, SFD_ENVS);
-      SUFFIXES (".sfd");
+#define SFD_SUFFIXES ".sfd", ".SFD"
+      SUFFIXES (SFD_SUFFIXES);
       break;
     case kpse_opentype_format:
       INIT_FORMAT ("opentype fonts", DEFAULT_OPENTYPEFONTS, OPENTYPE_ENVS);
@@ -663,7 +695,8 @@
       break;
     case kpse_lig_format:
       INIT_FORMAT ("lig files", DEFAULT_LIGFONTS, LIG_ENVS);
-      SUFFIXES (".lig");
+#define LIG_SUFFIXES ".lig", ".LIG"
+      SUFFIXES (LIG_SUFFIXES);
       break;
     case kpse_texmfscripts_format:
       INIT_FORMAT ("texmfscripts", DEFAULT_TEXMFSCRIPTS, TEXMFSCRIPTS_ENVS);
