diff -up texlive-2007/texk/dvipdfm/dvipdft.1.notetex texlive-2007/texk/dvipdfm/dvipdft.1
--- texlive-2007/texk/dvipdfm/dvipdft.1.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/dvipdfm/dvipdft.1	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "dvipdft" "1" "May 2004" "teTeX" "teTeX" 
+.TH "dvipdft" "1" "May 2004" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 dvipdft \- create thumbnail images for use with dvipdfm
diff -up texlive-2007/texk/dvipdfm/ebb.1.notetex texlive-2007/texk/dvipdfm/ebb.1
--- texlive-2007/texk/dvipdfm/ebb.1.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/dvipdfm/ebb.1	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "ebb" "1" "May 2004" "teTeX" "teTeX" 
+.TH "ebb" "1" "May 2004" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 ebb \- extract a bounding box from JPEG, PNG, and PDF files
diff -up texlive-2007/texk/makeindexk/mkindex.1.notetex texlive-2007/texk/makeindexk/mkindex.1
--- texlive-2007/texk/makeindexk/mkindex.1.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/makeindexk/mkindex.1	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "mkindex" "1" "September 2000" "teTeX" "teTeX" 
+.TH "mkindex" "1" "September 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 mkindex \- script to process LaTeX index and glossary files
@@ -27,7 +27,7 @@ MakeIndex glossary style file
 .PP 
 .SH "BUGS" 
 .PP 
-None known, but report any bugs found to <tetex@informatik\&.uni-hannover\&.de> (mailing list)\&.
+None known, but report any bugs found to <tex-live@tug\&.org> (mailing list)\&.
 .PP 
 .SH "AUTHOR" 
 .PP 
diff -up texlive-2007/texk/tetex/allcm.man.notetex texlive-2007/texk/tetex/allcm.man
--- texlive-2007/texk/tetex/allcm.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/allcm.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH allcm 1 "11/94" "teTeX" "teTeX"
+.TH allcm 1 "11/94" "TeXLive" "TeXLive"
 .SH NAME
 allcm \- force the most important Computer\-Modern\-fonts to be calculated
 .SH SYNOPSIS
diff -up texlive-2007/texk/tetex/allec.man.notetex texlive-2007/texk/tetex/allec.man
--- texlive-2007/texk/tetex/allec.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/allec.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH alldc 1 "11/94" "teTeX" "teTeX"
+.TH alldc 1 "11/94" "TeXLive" "TeXLive"
 .SH NAME
 allec \- force the most important ec\-fonts to be calculated
 .SH SYNOPSIS
diff -up texlive-2007/texk/tetex/allneeded.man.notetex texlive-2007/texk/tetex/allneeded.man
--- texlive-2007/texk/tetex/allneeded.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/allneeded.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH allneeded 1 "11/94" "teTeX" "teTeX"
+.TH allneeded 1 "11/94" "TeXLive" "TeXLive"
 .SH NAME
 allneeded \- force the calculation of all fonts now needed
 .SH SYNOPSIS
diff -up texlive-2007/texk/tetex/dvired.man.notetex texlive-2007/texk/tetex/dvired.man
--- texlive-2007/texk/tetex/dvired.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/dvired.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH dvired 1 "09/94" "teTeX" "teTeX"
+.TH dvired 1 "09/94" "TeXLive" "TeXLive"
 .SH NAME
 dvired \- print dvi-files
 .SH SYNOPSIS
diff -up texlive-2007/texk/tetex/e2pall.man.notetex texlive-2007/texk/tetex/e2pall.man
--- texlive-2007/texk/tetex/e2pall.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/e2pall.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "e2pall" "1" "August 2000" "teTeX" "teTeX" 
+.TH "e2pall" "1" "August 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 e2pall \- convert all EPS files in a LaTeX document to PDF
diff -up texlive-2007/texk/tetex/fmtutil.cnf.5.notetex texlive-2007/texk/tetex/fmtutil.cnf.5
--- texlive-2007/texk/tetex/fmtutil.cnf.5.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/fmtutil.cnf.5	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "fmtutil\&.cnf" "5" "October 2000" "teTeX" "teTeX" 
+.TH "fmtutil\&.cnf" "5" "October 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 fmtutil\&.cnf \- configuration file for fmtutil
@@ -55,11 +55,11 @@ hyphenation pattern file
 \fBamstex\fP(1), \fBetex\fP(1), \fBfmtutil\fP(8),
 \fBtex\fP(1), \fBtexconfig\fP(8), \fBtexlinks\fP(8)\&.
 .PP 
-Web page: <http://tug\&.org/teTeX/>
+Web page: <http://tug\&.org/texlive/>
 .PP 
 .SH "BUGS" 
 .PP 
-None known, but report any bugs found to <tetex@dbs\&.uni-hannover\&.de> (mailing list)\&.
+None known, but report any bugs found to <tex-live@tug\&.org> (mailing list)\&.
 .PP 
 .SH "AUTHOR" 
 .PP 
diff -up texlive-2007/texk/tetex/fmtutil.man.notetex texlive-2007/texk/tetex/fmtutil.man
--- texlive-2007/texk/tetex/fmtutil.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/fmtutil.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "fmtutil" "8" "February 2005" "teTeX" "teTeX" 
+.TH "fmtutil" "8" "February 2005" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 fmtutil \- utility for maintaining TeX format files
@@ -63,11 +63,11 @@ default configuration file
 .PP 
 \fBkpsewhich\fP(1)
 .PP 
-Web page: <http://tug\&.org/teTeX/>
+Web page: <http://tug\&.org/texlive/>
 .PP 
 .SH "BUGS" 
 .PP 
-None known, but report any bugs found to <tetex@dbs\&.uni-hannover\&.de> (mailing list)\&.
+None known, but report any bugs found to <tex-live@tug\&.org> (mailing list)\&.
 .PP 
 .SH "AUTHOR" 
 .PP 
diff -up texlive-2007/texk/tetex/fontinst.man.notetex texlive-2007/texk/tetex/fontinst.man
--- texlive-2007/texk/tetex/fontinst.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/fontinst.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "fontinst" "1" "May 2000" "teTeX" "teTeX" 
+.TH "fontinst" "1" "May 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 fontinst \- utility to run TeX as fontinst 
diff -up texlive-2007/texk/tetex/kpsetool.man.notetex texlive-2007/texk/tetex/kpsetool.man
--- texlive-2007/texk/tetex/kpsetool.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/kpsetool.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,7 +1,7 @@
-.TH "kpsetool" "1" "September 2000" "teTeX" "teTeX" 
+.TH "kpsetool" "1" "September 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
-kpsetool \- script to make teTeX-style \fBkpsetool\fP,
+kpsetool \- script to make TeXLive-style \fBkpsetool\fP,
 \fBkpsexpand\fP, and \fBkpsepath\fP available
 .PP 
 .SH "SYNOPSIS" 
@@ -14,11 +14,8 @@ kpsetool \- script to make teTeX-style \
 .PP 
 .SH "DESCRIPTION" 
 .PP 
-\fBkpsetool\fP is a Bourne shell script that makes a teTeX-style
+\fBkpsetool\fP is a Bourne shell script that makes a TeXLive-style
 \fBkpsetool\fP, \fBkpsexpand\fP, and \fBkpsepath\fP available\&.
-\fBkpsetool\fP uses \fBweb2c\fP\'s (new) \fBkpsexpand\fP binary, so the
-syntax for \fBkpsexpand\fP is incompatible with teTeX\'s as of version
-0\&.4\&.
 .PP 
 .SH "OPTIONS" 
 .PP 
@@ -86,11 +83,11 @@ virtual font
 .PP 
 \fIKpathsea: A library for path searching\fP (info or DVI file)
 .PP 
-Web page: <http://tug\&.org/teTeX/>
+Web page: <http://tug\&.org/texlive/>
 .PP 
 .SH "BUGS" 
 .PP 
-None known, but report any bugs found to <tetex@dbs\&.uni-hannover\&.de> (mailing list)\&.
+None known, but report any bugs found to <tex-live@tug\&.org> (mailing list)\&.
 .PP 
 .SH "AUTHOR" 
 .PP 
diff -up texlive-2007/texk/tetex/kpsewhere.man.notetex texlive-2007/texk/tetex/kpsewhere.man
--- texlive-2007/texk/tetex/kpsewhere.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/kpsewhere.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "kpsewhere" "1" "March 2004" "teTeX" "teTeX" 
+.TH "kpsewhere" "1" "March 2004" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 kpsewhere \- Expanding \fBkpsewhich\fP to separately iterate over each
@@ -28,11 +28,11 @@ all other options are directly handed to
 .PP 
 \fIKpathsea: A library for path searching\fP (info or DVI file)
 .PP 
-Web page: <http://tug\&.org/teTeX/>
+Web page: <http://tug\&.org/texlive/>
 .PP 
 .SH "BUGS" 
 .PP 
-None known, but report any bugs found to <tetex@dbs\&.uni-hannover\&.de> (mailing list)\&.
+None known, but report any bugs found to <tex-live@tug\&.org> (mailing list)\&.
 .PP 
 .SH "AUTHOR" 
 .PP 
diff -up texlive-2007/texk/tetex/mkofm.man.notetex texlive-2007/texk/tetex/mkofm.man
--- texlive-2007/texk/tetex/mkofm.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/mkofm.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "mkofm" "1" "September 2000" "teTeX" "teTeX" 
+.TH "mkofm" "1" "September 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 mkofm \- front end to \fBmktextfm\fP(1)
diff -up texlive-2007/texk/tetex/pslatex.man.notetex texlive-2007/texk/tetex/pslatex.man
--- texlive-2007/texk/tetex/pslatex.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/pslatex.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "pslatex" "1" "September 2000" "teTeX" "teTeX" 
+.TH "pslatex" "1" "September 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 pslatex \- utility to typeset LaTeX files using PostScript fonts
diff -up texlive-2007/texk/tetex/rubibtex.man.notetex texlive-2007/texk/tetex/rubibtex.man
--- texlive-2007/texk/tetex/rubibtex.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/rubibtex.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "rubibtex" "1" "October 2000" "teTeX" "teTeX" 
+.TH "rubibtex" "1" "October 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 rubibtex \- make a bibliography for (La)TeX using Russian
diff -up texlive-2007/texk/tetex/rumakeindex.man.notetex texlive-2007/texk/tetex/rumakeindex.man
--- texlive-2007/texk/tetex/rumakeindex.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/rumakeindex.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "rumakeindex" "1" "September 2000" "teTeX" "teTeX" 
+.TH "rumakeindex" "1" "September 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 rumakeindex \- process a LaTeX index using Russian Cyrillic
diff -up texlive-2007/texk/tetex/texconfig.man.notetex texlive-2007/texk/tetex/texconfig.man
--- texlive-2007/texk/tetex/texconfig.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/texconfig.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,8 +1,8 @@
-.TH texconfig 1 "February 2005" "teTeX" "teTeX"
+.TH texconfig 1 "February 2005" "TeXLive" "TeXLive"
 .SH NAME
-texconfig \- configures teTeX
+texconfig \- configures TeXLive
 .LP
-texconfig-sys \- configures teTeX system\-wide
+texconfig-sys \- configures TeXLive system\-wide
 .SH SYNOPSIS
 .B texconfig
 .LP
@@ -10,7 +10,7 @@ texconfig-sys \- configures teTeX system
 .SH DESCRIPTION
 .I texconfig
 allows to configure and maintain
-.I teTeX
+.I TeXLive
 in an easy and convenient manner. It uses the
 .I dialog
 program to draw nice input boxes. After answering the dialoges, the
diff -up texlive-2007/texk/tetex/texdoc.man.notetex texlive-2007/texk/tetex/texdoc.man
--- texlive-2007/texk/tetex/texdoc.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/texdoc.man	2008-08-27 13:33:41.000000000 +0200
@@ -9,7 +9,7 @@
 .ft R
 .fi
 ..
-.TH "texdoc" "1" "September 2000" "teTeX" "teTeX" 
+.TH "texdoc" "1" "September 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 texdoc \- Utility to look up TeX documentation
@@ -23,7 +23,7 @@ texdoc \- Utility to look up TeX documen
 \fBtexdoc\fP seaches for \fINAME\fP in the TeX documentation and
 starts an appropriate viewer\&.
 .PP 
-\fINAME\fP should be the name of a command or package in the teTeX
+\fINAME\fP should be the name of a command or package in the TeX
 distribution\&.
 .PP 
 .SH "OPTIONS" 
@@ -71,11 +71,11 @@ hostname$ export TEXDOCVIEW_html
 \fBacroread\fP(1), \fBgv\fP(1), \fBkpsewhich\fP(1),
 \fBless\fP(1), \fBmore\fP(1), \fBxdvi\fP(1)
 .PP 
-Web page: <http://tug\&.org/teTeX/>
+Web page: <http://tug\&.org/texlive/>
 .PP 
 .SH "BUGS" 
 .PP 
-None known, but report any bugs found to <tetex@dbs\&.uni-hannover\&.de> (mailing list)\&.
+None known, but report any bugs found to <tex-live@tug\&.org> (mailing list)\&.
 .PP 
 .SH "AUTHOR" 
 .PP 
diff -up texlive-2007/texk/tetex/texdoctk.man.notetex texlive-2007/texk/tetex/texdoctk.man
--- texlive-2007/texk/tetex/texdoctk.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/texdoctk.man	2008-08-27 13:33:41.000000000 +0200
@@ -80,7 +80,7 @@ If you want to print the documentations,
 non-PS files into PostScript. Here are some suggestions:
 
 \& dvi->ps:
-dvips (is part of teTeX) (http://www.radicaleye.com/dvips.html)
+dvips (is part of TeXLive) (http://www.radicaleye.com/dvips.html)
 
 \& pdf->ps:
 pdf2ps (http://www.cs.wisc.edu/~ghost) or
diff -up texlive-2007/texk/tetex/texlinks.man.notetex texlive-2007/texk/tetex/texlinks.man
--- texlive-2007/texk/tetex/texlinks.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/texlinks.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "texlinks" "8" "September 2000" "teTeX" "teTeX" 
+.TH "texlinks" "8" "September 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 texlinks \- maintain symbolic links from format to engine
@@ -45,11 +45,11 @@ default configuration file
 .PP 
 \fBfmtutil\fP(1), \fBfmtutil\&.cnf\fP(5)
 .PP 
-Web page: <http://tug\&.org/teTeX/>
+Web page: <http://tug\&.org/texlive/>
 .PP 
 .SH "BUGS" 
 .PP 
-None known, but report any bugs found to <tetex@dbs\&.uni-hannover\&.de> (mailing list)\&.
+None known, but report any bugs found to <tex-live@tug\&.org> (mailing list)\&.
 .PP 
 .SH "AUTHOR" 
 .PP 
diff -up texlive-2007/texk/tetex/thumbpdf.man.notetex texlive-2007/texk/tetex/thumbpdf.man
--- texlive-2007/texk/tetex/thumbpdf.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/thumbpdf.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "thumbpdf" "1" "September 2000" "teTeX" "teTeX" 
+.TH "thumbpdf" "1" "September 2000" "TeXLive" "TeXLive" 
 .PP 
 .SH "NAME" 
 thumbpdf \- generate thumbnail images for a PDF file
@@ -75,7 +75,7 @@ for which a user wants to generate thumb
 .PP 
 \fBgs\fP(1), \fBpdftex\fP(1)\&.
 .PP 
-Web page: <http://tug\&.org/teTeX/>
+Web page: <http://tug\&.org/texlive/>
 .PP 
 .SH "BUGS" 
 .PP 
diff -up texlive-2007/texk/tetex/updmap.cfg.5.notetex texlive-2007/texk/tetex/updmap.cfg.5
--- texlive-2007/texk/tetex/updmap.cfg.5.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/updmap.cfg.5	2008-08-27 13:33:41.000000000 +0200
@@ -23,7 +23,7 @@
 friends.
 .SH "DESCRIPTION"
 The file \fBupdmap.cfg\fR is the central font configuration file of a
-teTeX system and is read by \fBupdmap\fP(1). It describes if and how
+TeX system and is read by \fBupdmap\fP(1). It describes if and how
 fonts should be included (or not included) into PS-/PDF-files. This
 works for Outline (Postscript Type1) and Bitmap Fonts (Postscript
 Type3) mostly coming from MetaFont.
diff -up texlive-2007/texk/tetex/updmap.man.notetex texlive-2007/texk/tetex/updmap.man
--- texlive-2007/texk/tetex/updmap.man.notetex	2006-01-17 22:41:51.000000000 +0100
+++ texlive-2007/texk/tetex/updmap.man	2008-08-27 13:33:41.000000000 +0200
@@ -1,4 +1,4 @@
-.TH "updmap" "1" "February 2005" "teTeX" "teTeX"
+.TH "updmap" "1" "February 2005" "TeXLive" "TeXLive"
 .PP
 .SH "NAME"
 updmap \- Update font map files for TeX output drivers\&.
@@ -206,7 +206,7 @@ forces all fonts to be downloaded, so th
 \fBxdvi\fP(1) and \fBps2pk\fP(1)\&.
 .IP
 .PP
-teTeX configuration files for \fBdvips\fP(1) tuned for use with \fBupdmap\fP:
+TeXLive configuration files for \fBdvips\fP(1) tuned for use with \fBupdmap\fP:
 .IP
 .IP "\fIconfig\&.builtin35\fP"
 Loads \fIbuildin35\&.map\fP instead
@@ -231,7 +231,7 @@ of \fIpsfonts\&.map\fP\&.
 .IP "\fIconfig\&.www\fP"
 Loads \fIpsfonts_t1\&.map\fP instead
 of \fIpsfonts\&.map\fP\&.
-(For compatibility with previous versions of teTeX\&.)
+(For compatibility with previous versions of teTeX and TeXLive\&.)
 .IP
 .IP "\fIconfig\&.gstopk\fP"
 Loads \fIpsfonts_t1\&.map\fP instead
@@ -244,11 +244,11 @@ of \fIpsfonts\&.map\fP\&.
 \fBgsftopk\fP(1), \fBpdftex\fP(1), \fBps2pk\fP(1),
 \fBtexhash\fP(1), \fBxdvi\fP(1)\&.
 .PP
-Web page: \fI<http://tug\&.org/teTeX/>\fP
+Web page: \fI<http://tug\&.org/texlive/>\fP
 .PP
 .SH "BUGS"
 .PP
-None known, but report any bugs found to <tetex@dbs\&.uni-hannover\&.de> (mailing list)\&.
+None known, but report any bugs found to <tex-live@tug\&.org> (mailing list)\&.
 .PP
 .SH "AUTHOR"
 .PP
