--- tetex-src-3.0/texk/xdvik/browser.c.browser	2004-11-12 10:25:26.000000000 +0900
+++ tetex-src-3.0/texk/xdvik/browser.c	2005-02-07 15:22:31.849488410 +0900
@@ -39,7 +39,10 @@
 #include "statusline.h"
 
 static const char *const default_browser_cmd =
-	"netscape -raise -remote \"openURL(%s,new-window)\""
+	"xdg-open %s"
+	":firefox -remote \"openURL(%s,new-window)\""
+	":mozilla -remote \"openURL(%s,new-window)\""
+	":xterm -e w3m %s"
 	":xterm -e lynx %s"
 	":xterm -e wget %s";
 
--- tetex-src-3.0/texk/xdvik/gui/xm_prefsP.h.browser	2004-04-22 06:12:09.000000000 +0900
+++ tetex-src-3.0/texk/xdvik/gui/xm_prefsP.h	2005-02-07 15:22:31.850488229 +0900
@@ -96,8 +96,10 @@
 #define Xdvi_APPLY_STR				"Apply"
 
 #define Xdvi_PREFS_BROWSER_DEFAULTS \
+	"xdg-open %s\n" \
+	"firefox -remote 'openURL(%s,new-window)'\n" \
 	"mozilla -remote 'openURL(%s,new-window)'\n" \
-	"netscape -raise -remote 'openURL(%s,new-window)'\n" \
+	"xterm -e w3m %s\n" \
 	"xterm -e lynx %s\n" \
 	Xdvi_ADD_COMMAND_STR
 #define Xdvi_PREFS_EDITOR_DEFAULTS \
--- tetex-src-3.0/texk/xdvik/xdvi.1.in.browser	2005-01-21 01:51:45.000000000 +0900
+++ tetex-src-3.0/texk/xdvik/xdvi.1.in	2005-02-07 15:22:31.854487501 +0900
@@ -365,7 +365,7 @@
 .B WWWBROWSER
 (in that order) are used to determine the browser command. If these are
 not set either, the following default value is used:
-.SB netscape -raise -remote "openURL(%s,new-window)":xterm -e lynx %s:xterm -e wget %s
+.SB xdg-open %s:firefox -remote -remote "openURL(%s,new-window)":mozilla -remote "openURL(%s,new-window)":xterm -e w3m %s:xterm -e lynx %s:xterm -e wget %s
 .\"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 .TP
 .BI \-bw " width"
