diff -up mikmod-3.2.2-beta1/configure~ mikmod-3.2.2-beta1/configure
--- mikmod-3.2.2-beta1/configure~	2008-04-03 15:41:20.000000000 +0200
+++ mikmod-3.2.2-beta1/configure	2008-04-03 15:41:20.000000000 +0200
@@ -3338,6 +3338,7 @@ cat >>conftest.$ac_ext <<_ACEOF
 #include <mikmod.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 char* my_strdup (char *str)
 {
@@ -4116,8 +4116,8 @@ main ()
   for (i = 0; i < 256; i++)
     if (XOR (islower (i), ISLOWER (i))
 	|| toupper (i) != TOUPPER (i))
-      exit(2);
-  exit (0);
+      return (2);
+  return (0);
 }
 _ACEOF
 rm -f conftest$ac_exeext
diff -up mikmod-3.2.2-beta1/src/mutilities.c~ mikmod-3.2.2-beta1/src/mutilities.c
--- mikmod-3.2.2-beta1/src/mutilities.c~	2008-04-03 15:47:52.000000000 +0200
+++ mikmod-3.2.2-beta1/src/mutilities.c	2008-04-03 15:47:52.000000000 +0200
@@ -35,6 +35,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <errno.h>
+#include <ctype.h>
 
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
