diff --git a/pdf-core.gemspec b/pdf-core.gemspec
index 49ab202..43e4c58 100644
--- a/pdf-core.gemspec
+++ b/pdf-core.gemspec
@@ -22,20 +22,20 @@ Gem::Specification.new do |s|
     s.specification_version = 4
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
-      s.add_development_dependency(%q<pdf-reader>, ["~> 1.2"])
+      s.add_development_dependency(%q<pdf-reader>, [">= 1.2", "< 1.3.4"])
       s.add_development_dependency(%q<simplecov>, [">= 0"])
       s.add_development_dependency(%q<pdf-inspector>, ["~> 1.1.0"])
       s.add_development_dependency(%q<rspec>, [">= 0"])
       s.add_development_dependency(%q<rake>, [">= 0"])
     else
-      s.add_dependency(%q<pdf-reader>, ["~> 1.2"])
+      s.add_dependency(%q<pdf-reader>, [">= 1.2", "< 1.3.4"])
       s.add_dependency(%q<simplecov>, [">= 0"])
       s.add_dependency(%q<pdf-inspector>, ["~> 1.1.0"])
       s.add_dependency(%q<rspec>, [">= 0"])
       s.add_dependency(%q<rake>, [">= 0"])
     end
   else
-    s.add_dependency(%q<pdf-reader>, ["~> 1.2"])
+    s.add_dependency(%q<pdf-reader>, [">= 1.2", "< 1.3.4"])
     s.add_dependency(%q<simplecov>, [">= 0"])
     s.add_dependency(%q<pdf-inspector>, ["~> 1.1.0"])
     s.add_dependency(%q<rspec>, [">= 0"])
