diff -up wimax-20110419/InfraStack/OSDependent/Linux/Config/config.xml.in.foo wimax-20110419/InfraStack/OSDependent/Linux/Config/config.xml.in
--- wimax-20110419/InfraStack/OSDependent/Linux/Config/config.xml.in.foo	2011-04-19 16:12:21.824000013 -0400
+++ wimax-20110419/InfraStack/OSDependent/Linux/Config/config.xml.in	2011-04-19 16:12:28.038000012 -0400
@@ -15,7 +15,7 @@
 	<ModeOfOperationProduction>0</ModeOfOperationProduction>
 	<OutputTarget>console</OutputTarget>
 	<SdkIpAddress>0.0.0.0</SdkIpAddress>
-	<IPRenew>1</IPRenew>
+	<IPRenew>0</IPRenew>
 	<DisableObfuscate>0</DisableObfuscate>
 	<SdkPortNum>49001</SdkPortNum>
 	<StartMocMsgProxy>False</StartMocMsgProxy>
