From 2f3cb337e1cda9e897480f256453f369914380b3 Mon Sep 17 00:00:00 2001
From: Vitezslav Cizek <vcizek@suse.cz>
Date: Mon, 1 Oct 2012 14:48:16 +0200
Subject: [PATCH 207/208] su: fixed a typo in pam error message

Signed-off-by: Vitezslav Cizek <vcizek@suse.cz>
---
 login-utils/su-common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/login-utils/su-common.c b/login-utils/su-common.c
index 118e080..ea6864e 100644
--- a/login-utils/su-common.c
+++ b/login-utils/su-common.c
@@ -223,7 +223,7 @@ create_watching_parent (void)
   if (is_pam_failure(retval))
     {
       cleanup_pam (retval);
-      errx (EXIT_FAILURE, _("cannot not open session: %s"),
+      errx (EXIT_FAILURE, _("cannot open session: %s"),
 	     pam_strerror (pamh, retval));
     }
   else
-- 
1.7.11.7

