diff -up util-linux-2.19/floppy-0.18/superfloppy.c.kzak util-linux-2.19/floppy-0.18/superfloppy.c
--- util-linux-2.19/floppy-0.18/superfloppy.c.kzak	2009-09-23 02:47:26.000000000 +0200
+++ util-linux-2.19/floppy-0.18/superfloppy.c	2011-03-07 10:37:20.000000000 +0100
@@ -13,6 +13,7 @@
 #include <popt.h>
 #include <locale.h>
 #include <libintl.h>
+#include <locale.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
