diff --git a/src/add-caldav-object.c b/src/add-caldav-object.c
index 68a4377..35aef48 100644
--- a/src/add-caldav-object.c
+++ b/src/add-caldav-object.c
@@ -103,7 +103,6 @@ gboolean caldav_add(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "PUT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	if (res != 0) {
 		error->code = -1;
diff --git a/src/delete-caldav-object.c b/src/delete-caldav-object.c
index 9a85168..ea5733f 100644
--- a/src/delete-caldav-object.c
+++ b/src/delete-caldav-object.c
@@ -150,7 +150,6 @@ gboolean caldav_delete(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	g_free(search);
 	curl_slist_free_all(http_header);
@@ -249,7 +248,6 @@ gboolean caldav_delete(caldav_settings* settings, caldav_error* error) {
 					curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "DELETE");
 					curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 					curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-					curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 					res = curl_easy_perform(curl);
 					if (LOCKSUPPORT && lock_token) {
 						caldav_unlock_object(
@@ -380,7 +378,6 @@ gboolean caldav_tasks_delete(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	g_free(search);
 	curl_slist_free_all(http_header);
@@ -479,7 +476,6 @@ gboolean caldav_tasks_delete(caldav_settings* settings, caldav_error* error) {
 					curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "DELETE");
 					curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 					curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-					curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 					res = curl_easy_perform(curl);
 					if (LOCKSUPPORT && lock_token) {
 						caldav_unlock_object(
diff --git a/src/get-caldav-report.c b/src/get-caldav-report.c
index d916259..0021cb8 100644
--- a/src/get-caldav-report.c
+++ b/src/get-caldav-report.c
@@ -169,7 +169,6 @@ gboolean caldav_getall(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	if (res != 0) {
 		error->code = -1;
@@ -266,7 +265,6 @@ gboolean caldav_getrange(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	if (res != 0) {
 		error->code = -1;
@@ -349,7 +347,6 @@ gboolean caldav_tasks_getall(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	if (res != 0) {
 		error->code = -1;
@@ -446,7 +443,6 @@ gboolean caldav_tasks_getrange(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	if (res != 0) {
 		error->code = -1;
diff --git a/src/get-display-name.c b/src/get-display-name.c
index 8fc0ff8..c205efd 100644
--- a/src/get-display-name.c
+++ b/src/get-display-name.c
@@ -99,7 +99,6 @@ gboolean caldav_getname(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "PROPFIND");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	if (res != 0) {
 		error->code = -1;
diff --git a/src/get-freebusy-report.c b/src/get-freebusy-report.c
index f75ab92..40a5a3c 100644
--- a/src/get-freebusy-report.c
+++ b/src/get-freebusy-report.c
@@ -104,7 +104,6 @@ gboolean caldav_freebusy(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	if (res != 0) {
 		error->code = -1;
diff --git a/src/lock-caldav-object.c b/src/lock-caldav-object.c
index 30f2d20..6845551 100644
--- a/src/lock-caldav-object.c
+++ b/src/lock-caldav-object.c
@@ -107,7 +107,6 @@ gchar* caldav_lock_object(
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "LOCK");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	curl_slist_free_all(http_header);
 	if (res != 0) {
@@ -211,7 +210,6 @@ gboolean caldav_unlock_object(gchar* lock_token, gchar* URI,
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "UNLOCK");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	curl_slist_free_all(http_header);
 	if (res != 0) {
diff --git a/src/modify-caldav-object.c b/src/modify-caldav-object.c
index fb3a4d0..f3563d8 100644
--- a/src/modify-caldav-object.c
+++ b/src/modify-caldav-object.c
@@ -151,7 +151,6 @@ gboolean caldav_modify(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	curl_slist_free_all(http_header);
 	http_header = NULL;
@@ -249,7 +248,6 @@ gboolean caldav_modify(caldav_settings* settings, caldav_error* error) {
 									strlen(settings->file));
 						curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 						curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-						curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 						curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "PUT");
 						res = curl_easy_perform(curl);
 						if (LOCKSUPPORT && lock_token) {
@@ -391,7 +389,6 @@ gboolean caldav_tasks_modify(caldav_settings* settings, caldav_error* error) {
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "REPORT");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	curl_slist_free_all(http_header);
 	http_header = NULL;
@@ -489,7 +486,6 @@ gboolean caldav_tasks_modify(caldav_settings* settings, caldav_error* error) {
 									strlen(settings->file));
 						curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 						curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-						curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 						curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "PUT");
 						res = curl_easy_perform(curl);
 						if (LOCKSUPPORT && lock_token) {
diff --git a/src/options-caldav-server.c b/src/options-caldav-server.c
index 1d97d04..0045afe 100644
--- a/src/options-caldav-server.c
+++ b/src/options-caldav-server.c
@@ -74,7 +74,6 @@ gboolean caldav_getoptions(CURL* curl, caldav_settings* settings, response* resu
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "OPTIONS");
 	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1);
 	curl_easy_setopt(curl, CURLOPT_UNRESTRICTED_AUTH, 1);
-	curl_easy_setopt(curl, CURLOPT_POSTREDIR, CURL_REDIR_POST_ALL);
 	res = curl_easy_perform(curl);
 	if (res == 0) {
 		gchar* head;
@@ -102,15 +101,12 @@ gboolean caldav_getoptions(CURL* curl, caldav_settings* settings, response* resu
 	}
 	else if (
 		(res == CURLE_SSL_CONNECT_ERROR ||
-		 CURLE_PEER_FAILED_VERIFICATION ||
 		 CURLE_SSL_ENGINE_NOTFOUND ||
 		 CURLE_SSL_ENGINE_SETFAILED ||
 		 CURLE_SSL_CERTPROBLEM ||
 		 CURLE_SSL_CIPHER ||
 		 CURLE_SSL_CACERT ||
-		 CURLE_SSL_CACERT_BADFILE ||
-		 CURLE_SSL_CRL_BADFILE ||
-		 CURLE_SSL_ISSUER_ERROR) && settings->usehttps) {
+		 CURLE_SSL_CRL_BADFILE ) && settings->usehttps) {
 		error->code = -2;
 		error->str = g_strdup(error_buf);
 	}
