diff -ur k3b-i18n-1.0.5.orig/configure k3b-i18n-1.0.5/configure
--- k3b-i18n-1.0.5.orig/configure	2011-12-15 12:17:00.000000000 -0500
+++ k3b-i18n-1.0.5/configure	2011-12-15 12:25:35.000000000 -0500
@@ -31498,7 +31498,7 @@
 kde_widgetdir=NO
 for i in $kde_libdirs;
 do
-  for j in "kde3/plugins/designer/kdewidgets.la";
+  for j in "trinity/plugins/designer/kdewidgets.la";
   do
     echo "configure: 31503: $i/$j" >&5
     if test -r "$i/$j"; then
@@ -31520,7 +31520,7 @@
    { (exit 1); exit 1; }; }
 fi
 
-if test -n "$kde_widgetdir" && test ! -r "$kde_widgetdir/kde3/plugins/designer/kdewidgets.la"; then
+if test -n "$kde_widgetdir" && test ! -r "$kde_widgetdir/trinity/plugins/designer/kdewidgets.la"; then
 { { echo "$as_me:$LINENO: error:
 I can't find the designer plugins. These are required and should have been installed
 by kdelibs" >&5
@@ -31531,7 +31531,7 @@
 fi
 
 if test -n "$kde_widgetdir"; then
-    kde_widgetdir="$kde_widgetdir/kde3/plugins/designer"
+    kde_widgetdir="$kde_widgetdir/trinity/plugins/designer"
 fi
 
 
@@ -31778,10 +31778,10 @@
     fi
   fi
   if test -z "$kde_styledir"; then
-    kde_styledir='\${libdir}/kde3/plugins/styles'
+    kde_styledir='\${libdir}/trinity/plugins/styles'
   fi
   if test -z "$kde_widgetdir"; then
-    kde_widgetdir='\${libdir}/kde3/plugins/designer'
+    kde_widgetdir='\${libdir}/trinity/plugins/designer'
   fi
   if test -z "$xdg_appsdir"; then
     xdg_appsdir='\${datadir}/applications/kde'
@@ -31924,7 +31924,7 @@
 
 { echo "$as_me:$LINENO: checking for KDE libraries installed" >&5
 echo $ECHO_N "checking for KDE libraries installed... $ECHO_C" >&6; }
-ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -lkdecore $LIBQT $KDE_TEST_RPATH 1>&5'
+ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -ltdecore $LIBQT $KDE_TEST_RPATH 1>&5'
 
 if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
   (eval $ac_link) 2>&5
@@ -32080,10 +32080,10 @@
     fi
   fi
   if test -z "$kde_styledir"; then
-    kde_styledir='\${libdir}/kde3/plugins/styles'
+    kde_styledir='\${libdir}/trinity/plugins/styles'
   fi
   if test -z "$kde_widgetdir"; then
-    kde_widgetdir='\${libdir}/kde3/plugins/designer'
+    kde_widgetdir='\${libdir}/trinity/plugins/designer'
   fi
   if test -z "$xdg_appsdir"; then
     xdg_appsdir='\${datadir}/applications/kde'
@@ -32226,7 +32226,7 @@
 
 { echo "$as_me:$LINENO: checking for KDE libraries installed" >&5
 echo $ECHO_N "checking for KDE libraries installed... $ECHO_C" >&6; }
-ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -lkdecore $LIBQT $KDE_TEST_RPATH 1>&5'
+ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -ltdecore $LIBQT $KDE_TEST_RPATH 1>&5'
 
 if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
   (eval $ac_link) 2>&5
@@ -33098,9 +33098,9 @@
        esac
    LIB_KDED=$lib_kded
 
-   LIB_KDECORE="-lkdecore"
+   LIB_KDECORE="-ltdecore"
 
-   LIB_KDEUI="-lkdeui"
+   LIB_KDEUI="-ltdeui"
 
    LIB_KIO="-lkio"
 
@@ -33138,9 +33138,9 @@
    LIB_KFILE="-lkio"
 
 elif test $kde_qtver = 2; then
-   LIB_KDECORE="-lkdecore"
+   LIB_KDECORE="-ltdecore"
 
-   LIB_KDEUI="-lkdeui"
+   LIB_KDEUI="-ltdeui"
 
    LIB_KIO="-lkio"
 
@@ -33161,9 +33161,9 @@
    LIB_KDEPRINT="-lkdeprint"
 
 else
-   LIB_KDECORE="-lkdecore -lXext $(LIB_QT)"
+   LIB_KDECORE="-ltdecore -lXext $(LIB_QT)"
 
-   LIB_KDEUI="-lkdeui $(LIB_KDECORE)"
+   LIB_KDEUI="-ltdeui $(LIB_KDECORE)"
 
    LIB_KFM="-lkfm $(LIB_KDECORE)"
 
