diff -ur lilypond-2.16.2-orig/lily/freetype-error.cc lilypond-2.16.2/lily/freetype-error.cc
--- lilypond-2.16.2-orig/lily/freetype-error.cc	2014-05-16 09:32:27.348273363 +0000
+++ lilypond-2.16.2/lily/freetype-error.cc	2014-05-16 09:56:29.380261683 +0000
@@ -31,7 +31,7 @@
   const char *err_msg;
 } ft_errors[] =
 
-#include <freetype/fterrors.h>
+#include <fterrors.h>
 
   ;
 
diff -ur lilypond-2.16.2-orig/lily/open-type-font.cc lilypond-2.16.2/lily/open-type-font.cc
--- lilypond-2.16.2-orig/lily/open-type-font.cc	2014-05-16 09:32:28.930305423 +0000
+++ lilypond-2.16.2/lily/open-type-font.cc	2014-05-16 09:56:40.933001973 +0000
@@ -23,7 +23,7 @@
 
 using namespace std;
 
-#include <freetype/tttables.h>
+#include <tttables.h>
 
 #include "dimensions.hh"
 #include "international.hh"
diff -ur lilypond-2.16.2-orig/lily/pango-font.cc lilypond-2.16.2/lily/pango-font.cc
--- lilypond-2.16.2-orig/lily/pango-font.cc	2014-05-16 09:32:29.452766628 +0000
+++ lilypond-2.16.2/lily/pango-font.cc	2014-05-16 09:56:55.588282914 +0000
@@ -22,7 +22,7 @@
 #define PANGO_ENABLE_BACKEND
 
 #include <pango/pangoft2.h>
-#include <freetype/ftxf86.h>
+#include <ftxf86.h>
 
 #include <map>
 #include <cstdio>
diff -ur lilypond-2.16.2-orig/lily/ttf.cc lilypond-2.16.2/lily/ttf.cc
--- lilypond-2.16.2-orig/lily/ttf.cc	2014-05-16 09:32:28.137336267 +0000
+++ lilypond-2.16.2/lily/ttf.cc	2014-05-16 09:58:15.174261162 +0000
@@ -20,7 +20,7 @@
 #include <cstdio>
 #include "freetype.hh"
 
-#include <freetype/tttables.h>
+#include <tttables.h>
 
 #include "international.hh"
 #include "memory-stream.hh"
