diff -up cups-1.5.2/cups/language.c.translation cups-1.5.2/cups/language.c
--- cups-1.5.2/cups/language.c.translation	2011-11-01 07:06:15.000000000 +0100
+++ cups-1.5.2/cups/language.c	2012-02-28 10:54:58.448744034 +0100
@@ -1134,7 +1134,7 @@ _cupsMessageLookup(cups_array_t *a,	/* I
   }
 #endif /* __APPLE__ && CUPS_BUNDLEDIR */
 
-  if (match && match->str)
+  if (match && match->str && (strlen(match->str) > 0))
     return (match->str);
   else
     return (m);
