diff --git a/coregrind/vg_preloaded.c b/coregrind/vg_preloaded.c
index fe8b50a..e36d90d 100644
--- a/coregrind/vg_preloaded.c
+++ b/coregrind/vg_preloaded.c
@@ -83,7 +83,7 @@ void * VG_NOTIFY_ON_LOAD(ifunc_wrapper) (void)
     VALGRIND_GET_ORIG_FN(fn);
     CALL_FN_W_v(result, fn);
 
-#if defined(VGP_ppc64_linux)
+#if defined(VGP_ppc64be_linux)
    /* ppc64 uses function descriptors, so get the actual function entry
       address for the client request, but return the function descriptor
       from this function. 
