diff -Nuri valgrind-3.11.0-orig/include/valgrind.h valgrind-3.11.0/include/valgrind.h
--- valgrind-3.11.0-orig/include/valgrind.h     2015-11-03 10:56:07.117788401 +0800
+++ valgrind-3.11.0/include/valgrind.h  2015-11-03 10:57:57.195932971 +0800
@@ -156,7 +156,7 @@
#  define PLAT_arm64_linux 1
#elif defined(__linux__) && defined(__s390__) && defined(__s390x__)
#  define PLAT_s390x_linux 1
-#elif defined(__linux__) && defined(__mips__) && (__mips==64)
+#elif defined(__linux__) && defined(__mips__) && ((__mips==64) || defined(__mips64))
#  define PLAT_mips64_linux 1
#elif defined(__linux__) && defined(__mips__) && (__mips!=64)
#  define PLAT_mips32_linux 1
