diff -up pango-1.29.5/pango/Makefile.am.lib64 pango-1.29.5/pango/Makefile.am
--- pango-1.29.5/pango/Makefile.am.lib64	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/Makefile.am	2011-11-22 16:39:49.407443964 -0500
@@ -25,6 +25,7 @@ INCLUDES =						\
 	-DPANGO_ENABLE_BACKEND				\
 	-DPANGO_ENABLE_ENGINE				\
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
+	-DHOST=\"$(host_triplet)\"			\
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
 	-I$(top_builddir)				\
diff -up pango-1.29.5/pango/Makefile.in.lib64 pango-1.29.5/pango/Makefile.in
--- pango-1.29.5/pango/Makefile.in.lib64	2011-11-22 14:07:59.000000000 -0500
+++ pango-1.29.5/pango/Makefile.in	2011-11-22 16:40:20.090442868 -0500
@@ -557,6 +557,7 @@ INCLUDES = \
 	-DPANGO_ENABLE_BACKEND				\
 	-DPANGO_ENABLE_ENGINE				\
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
+	-DHOST=\"$(host_triplet)\"			\
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
 	-I$(top_builddir)				\
diff -up pango-1.29.5/pango/modules.c.lib64 pango-1.29.5/pango/modules.c
--- pango-1.29.5/pango/modules.c.lib64	2011-08-15 21:11:08.000000000 -0400
+++ pango-1.29.5/pango/modules.c	2011-11-22 16:38:52.317446005 -0500
@@ -529,6 +529,7 @@ read_modules (void)
 
   if (!file_str)
     file_str = g_build_filename (pango_get_sysconf_subdirectory (),
+				 HOST,
 				 "pango.modules",
 				 NULL);
 
