From 4c3be42cdf4c49785298ddb6b34beed1c8cc9d40 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Sat, 21 Jul 2012 12:52:01 +0200
Subject: [PATCH] dracut.8: add more description about calling dracut

---
 dracut.8.asc | 56 +++++++++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 55 insertions(+), 1 deletion(-)

diff --git a/dracut.8.asc b/dracut.8.asc
index 0cd1663..24af936 100644
--- a/dracut.8.asc
+++ b/dracut.8.asc
@@ -14,6 +14,12 @@ SYNOPSIS
 
 DESCRIPTION
 -----------
+
+Create an initramfs <image> for the kernel with the version <kernel version>.
+If <kernel version> is omitted, then the version of the actual running
+kernel is used. If <image> is omitted or empty, then the default location
+/boot/initramfs-<kernel version>.img is used.
+
 dracut creates an initial image used by the kernel for preloading the block
 device modules (such as IDE, SCSI or RAID) which are needed to access the root
 filesystem, mounting the root filesystem and booting into the real system.
@@ -22,7 +28,55 @@ At boot time, the kernel unpacks that archive into RAM disk, mounts and uses it
 as initial root file system. All finding of the root device happens in this
 early userspace.
 
-For a complete list of kernel command line options see *dracut.cmdline*(7)
+For a complete list of kernel command line options see *dracut.cmdline*(7).
+
+EXAMPLE
+-------
+
+To create a initramfs image, the most simple command is:
+----
+# dracut
+----
+
+This will generate a general purpose initramfs image, with all possible
+functionality resulting of the combination of the installed dracut modules and
+system tools. The image is /boot/initramfs-_++<kernel version>++_.img and
+contains the kernel modules of the currently active kernel with version
+_++<kernel version>++_.
+
+If the initramfs image already exists, dracut will display an error message, and
+to overwrite the existing image, you have to use the --force option.
+----
+# dracut --force
+----
+
+If you want to specify another filename for the resulting image you would issue
+a command like:
+----
+# dracut foobar.img
+----
+
+To generate an image for a specific kernel version, the command would be:
+----
+# dracut foobar.img 2.6.40-1.rc5.f20
+----
+
+A shortcut to generate the image at the default location for a specific kernel
+version is:
+----
+# dracut --kver 2.6.40-1.rc5.f20
+----
+
+If you want to create lighter, smaller initramfs images, you may want to specify
+the --host-only or -H option. Using this option, the resulting image will
+contain only those dracut modules, kernel modules and filesystems, which are
+needed to boot this specific machine. This has the drawback, that you can't put
+the disk on another controller or machine, and that you can't switch to another
+root filesystem, without recreating the initramfs image. The usage of the
+--host-only option is only for experts and you will have to keep the broken
+pieces. At least keep a copy of a general purpose image (and corresponding
+kernel) as a fallback to rescue your system.
+
 
 OPTIONS
 -------
