From 472189da152be23b8a7d0dabc0a3e80ea61e866b Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Sat, 21 Jul 2012 13:03:56 +0200
Subject: [PATCH] dracut.sh: add -N option for --no-hostonly

---
 dracut.8.asc | 2 +-
 dracut.sh    | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/dracut.8.asc b/dracut.8.asc
index 24af936..0902b3b 100644
--- a/dracut.8.asc
+++ b/dracut.8.asc
@@ -308,7 +308,7 @@ If chrooted to another root other than the real root device, use "--fstab" and
 provide a valid _/etc/fstab_.
 ====
 
-**--no-hostonly**::
+**-N, --no-hostonly**::
     Disable Host-Only mode
 
 **--fstab**::
diff --git a/dracut.sh b/dracut.sh
index ffdcbcc..d8b3354 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -126,7 +126,7 @@ Creates initial ramdisk images for preloading modules
                          Useful when running dracut from a git checkout.
   -H, --hostonly        Host-Only mode: Install only what is needed for
                          booting the local host instead of a generic host.
-  --no-hostonly         Disables Host-Only mode
+  -N, --no-hostonly     Disables Host-Only mode
   --fstab               Use /etc/fstab to determine the root device.
   --add-fstab [FILE]    Add file to the initramfs fstab
   --mount "[DEV] [MP] [FSTYPE] [FSOPTS]"
@@ -244,7 +244,7 @@ set -- "${@/--include/++include}"
 set -- "${@/%-i/++include}"
 
 TEMP=$(unset POSIXLY_CORRECT; getopt \
-    -o "a:m:o:d:I:k:c:L:fvqlHhM" \
+    -o "a:m:o:d:I:k:c:L:fvqlHhMN" \
     --long kver: \
     --long add: \
     --long force-add: \
@@ -356,7 +356,7 @@ while :; do
                            && dracutbasedir="$(readlink -f ${0%/*})"
                        ;;
         -H|--hostonly) hostonly_l="yes" ;;
-        --no-hostonly) hostonly_l="no" ;;
+        -N|--no-hostonly) hostonly_l="no" ;;
         --fstab)       use_fstab_l="yes" ;;
         -h|--help)     long_usage; exit 1 ;;
         -i|--include)  push include_src "$2"
