From 7b2d3d995c97652a13880b279bfc602e47bcda9c Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 9 Feb 2012 13:58:13 +0100
Subject: [PATCH] dracut: honor binaries in sbin first

---
 dracut |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/dracut b/dracut
index 545f9db..fb46d70 100755
--- a/dracut
+++ b/dracut
@@ -279,14 +279,14 @@ if ! [[ $kernel ]]; then
 fi
 [[ $outfile ]] || outfile="/boot/initramfs-$kernel.img"
 
-for i in /usr/bin /bin /usr/sbin /sbin; do
+for i in /usr/sbin /sbin /usr/bin /bin; do
     rl=$i
     if [ -L "$i" ]; then
         rl=$(readlink -f $i)
     fi
-    NPATH="$NPATH:$rl"
+    NPATH+=":$rl"
 done
-export PATH="$NPATH"
+export PATH="${NPATH#:}"
 unset NPATH
 unset LD_LIBRARY_PATH
 unset GREP_OPTIONS
