From edea870c3cade3a9f8836e75afa98587945908d2 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Sat, 25 Feb 2012 16:09:38 +0100
Subject: [PATCH] dracut-functions.sh: instmods() removed special case for
 "=ata"

---
 dracut-functions.sh |    8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index 81801e1..bd3203d 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -1103,13 +1103,7 @@ instmods() {
         local _mod="$1"
         case $_mod in
             =*)
-                # This introduces 2 incompatible meanings for =* arguments
-                # to instmods.  We need to decide which one to keep.
-                if [[ $_mod = =ata && -f $srcmods/modules.block ]]; then
-                    ( [[ "$_mpargs" ]] && echo $_mpargs
-                      egrep 'ata|ahci' "${srcmods}/modules.block" ) \
-                    | instmods
-                elif [ -f $srcmods/modules.${_mod#=} ]; then
+                if [ -f $srcmods/modules.${_mod#=} ]; then
                     ( [[ "$_mpargs" ]] && echo $_mpargs
                       cat "${srcmods}/modules.${_mod#=}" ) \
                     | instmods
