From 81dcf7c90a853d91009888dbff6670bb483c0716 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 9 Feb 2012 14:16:31 +0100
Subject: [PATCH] 99shutdown/shutdown: don't do console_init on shutdown
 emergency

---
 modules.d/99shutdown/shutdown |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/modules.d/99shutdown/shutdown b/modules.d/99shutdown/shutdown
index 64937fa..2576687 100755
--- a/modules.d/99shutdown/shutdown
+++ b/modules.d/99shutdown/shutdown
@@ -26,7 +26,6 @@ emergency_shell()
     source_hook shutdown-emergency
     echo
     if getargbool 1 rd.shell -y rdshell || getarg rd.break rdbreak; then
-        [ -x /lib/udev/console_init ] && /lib/udev/console_init /dev/console
         echo "Dropping to debug shell."
         echo
         export PS1="$_rdshell_name:\${PWD}# "
