From 0e95d84892d06ff14e4d6b321dad05344724934e Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Sat, 30 Jun 2012 11:27:36 +0200
Subject: [PATCH] dracut-functions.sh: set LC_ALL=C to get correct parsing
 information

---
 dracut-functions.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index 5acf6f6..cd9426e 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -19,7 +19,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-
+export LC_ALL=C
 
 if [[ $DRACUT_KERNEL_LAZY ]] && ! [[ $DRACUT_KERNEL_LAZY_HASHDIR ]]; then
     if ! [[ -d "$initdir/.kernelmodseen" ]]; then
