From 7a5f1ee46a07e00d1687a0c2cd6c13319626f17d Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Sat, 25 Feb 2012 16:40:38 +0100
Subject: [PATCH] dracut.spec: require "hardlink"

---
 dracut.spec |    1 +
 1 file changed, 1 insertion(+)

diff --git a/dracut.spec b/dracut.spec
index 1ec8b68..4102bb0 100644
--- a/dracut.spec
+++ b/dracut.spec
@@ -71,6 +71,7 @@ Requires: cpio
 Requires: filesystem >= 2.1.0
 Requires: findutils
 Requires: grep
+Requires: hardlink
 Requires: gzip
 Requires: module-init-tools >= 3.7-9
 Requires: sed
