From 34e43ceb0df05081b737f932b0bc2f0426fb5276 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Sat, 30 Jun 2012 11:31:38 +0200
Subject: [PATCH] dracut-install.c: give info that SOURCE argument is missing

---
 install/dracut-install.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/install/dracut-install.c b/install/dracut-install.c
index 86c32db..778881a 100644
--- a/install/dracut-install.c
+++ b/install/dracut-install.c
@@ -601,6 +601,7 @@ static int parse_argv(int argc, char *argv[])
         }
 
         if (!optind || optind == argc) {
+                log_error("No SOURCE argument given");
                 usage(EXIT_FAILURE);
         }
 
