From db9fbaf9b8753bad46fb7c43d860e234919d2bc9 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 5 Apr 2013 11:52:33 +0200
Subject: [PATCH] TEST-99-RPM: ignore rescue images

---
 test/TEST-99-RPM/test.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/TEST-99-RPM/test.sh b/test/TEST-99-RPM/test.sh
index 66f0beb..f030cb0 100755
--- a/test/TEST-99-RPM/test.sh
+++ b/test/TEST-99-RPM/test.sh
@@ -50,6 +50,7 @@ find / -xdev -type f -not -path '/var/*' \
   -not -path '/test.output' \
   -not -path '/etc/nsswitch.conf.bak' \
   -not -path '/etc/iscsi/initiatorname.iscsi' \
+  -not -path '/boot/*0-rescue*' \
   -not -path '/dev/null' \
   -exec rpm -qf '{}' ';' | \
   fgrep 'not owned' &> /test.output
