From ddd01008ff0b7b3daf5708ac32a6ccd0e843b3fc Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Sat, 25 Feb 2012 16:51:35 +0100
Subject: [PATCH] 95terminfo/module-setup.sh: no need to call "find"

---
 modules.d/95terminfo/module-setup.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/95terminfo/module-setup.sh b/modules.d/95terminfo/module-setup.sh
index aec3582..32a3fe9 100755
--- a/modules.d/95terminfo/module-setup.sh
+++ b/modules.d/95terminfo/module-setup.sh
@@ -10,7 +10,7 @@ install() {
     done
 
     if [ -d ${_terminfodir} ]; then
-        for f in $(find ${_terminfodir} -type f); do
+        for f in ${_terminfodir}/*/*; do
             inst_simple $f
         done
     fi
