From cf38fc7389d4136d0f77d9709c87acc4283f6bd6 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Sat, 25 Feb 2012 17:18:51 +0100
Subject: [PATCH] 10i18n/module-setup.sh: optimize install of all kbd files

---
 modules.d/10i18n/module-setup.sh |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/10i18n/module-setup.sh b/modules.d/10i18n/module-setup.sh
index a5a3388..b4c19e1 100755
--- a/modules.d/10i18n/module-setup.sh
+++ b/modules.d/10i18n/module-setup.sh
@@ -92,8 +92,8 @@ install() {
     install_all_kbd() {
         local rel f
 
-        for f in $(eval find ${kbddir}/{${KBDSUBDIRS}} -type f -print)
-        do
+        find $(eval echo ${kbddir}/{${KBDSUBDIRS}}) -type f -print | \
+            while read f; do
             inst_simple $f
         done
 
