From a1837eaf84714cefb082ae8701210ab2c876a62f Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 29 Nov 2012 16:07:35 +0100
Subject: [PATCH] nfs/nfsroot-cleanup.sh: mount --bind instead of --move

---
 modules.d/95nfs/nfsroot-cleanup.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules.d/95nfs/nfsroot-cleanup.sh b/modules.d/95nfs/nfsroot-cleanup.sh
index 5610016..586749c 100755
--- a/modules.d/95nfs/nfsroot-cleanup.sh
+++ b/modules.d/95nfs/nfsroot-cleanup.sh
@@ -22,7 +22,9 @@ if incol2 /proc/mounts /var/lib/nfs/rpc_pipefs; then
         mkdir -m 0755 -p $NEWROOT/$rpcpipefspath 2>/dev/null
 
     if [ -d $NEWROOT/$rpcpipefspath ]; then
-        mount --move /var/lib/nfs/rpc_pipefs $NEWROOT/$rpcpipefspath
+        # mount --move does not seem to work???
+        mount --bind /var/lib/nfs/rpc_pipefs $NEWROOT/$rpcpipefspath
+        umount /var/lib/nfs/rpc_pipefs
     else
         umount /var/lib/nfs/rpc_pipefs
     fi
