From ec61f0a3af894d2498d696464f12da2d500b9f39 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 2 Jul 2012 22:14:49 +0200
Subject: [PATCH] dracut-functions.sh: forgot --set-version $kernel for
 modprobe

---
 dracut-functions.sh |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index eabbb0d..2898c30 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -1201,7 +1201,8 @@ dracut_kernel_post() {
     local _moddirname=${srcmods%%/lib/modules/*}
 
     if [[ $DRACUT_KERNEL_LAZY_HASHDIR ]] && [[ -f "$DRACUT_KERNEL_LAZY_HASHDIR/lazylist" ]]; then
-        xargs -r modprobe -a ${_moddirname+-d ${_moddirname}/} --ignore-install --show-depends \
+        xargs -r modprobe -a ${_moddirname+-d ${_moddirname}/} \
+            --ignore-install --show-depends --set-version $kernel \
             < "$DRACUT_KERNEL_LAZY_HASHDIR/lazylist" 2>/dev/null \
             | sort -u \
             | while read _cmd _modpath _options; do
