From 14201d23e7d1dd3b1e6cc7444768c297df03db54 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 11 Apr 2013 10:10:49 +0200
Subject: [PATCH] AUTHORS: update

---
 .mailmap | 27 +++++++++++++++------------
 AUTHORS  | 27 ++++++++++++++++++---------
 2 files changed, 33 insertions(+), 21 deletions(-)

diff --git a/.mailmap b/.mailmap
index bce9b7c..852f269 100644
--- a/.mailmap
+++ b/.mailmap
@@ -1,12 +1,15 @@
-Philippe Seewer <philippe.seewer@bfh.ch> <philippe.seewer-omB+W0Dpw2o@public.gmane.org>
-Seewer Philippe <philippe.seewer@bfh.ch> <philippe.seewer@bfh.ch>
-Philippe Seewer <philippe.seewer@bfh.ch> <philippe.seewer@bfh.ch>
-Victor Lowther <victor.lowther@gmail.com> <victor.lowther-Re5JQEeQqe8AvxtiuMwx3w@public.gmane.org>
-Harald Hoyer <harald@redhat.com> <harald-H+wXaHxf7aLQT0dZR+AlfA@public.gmane.org>
-Harald Hoyer <harald@redhat.com> <harald@eeepc.(none)>
-Mike Snitzer <snitzer@redhat.com> <msnitzer@redhat.com> 
-Amerigo Wang <amwang@redhat.com> <xiyou.wangcong@gmail.com>
-Andrey Borzenkov <arvidjaar@gmail.com> <arvidjaar@mail.ru>
-Dan Horák <dhorak@redhat.com> <dan@danny.cz>
-John Reiser <jreiser@bitwagon.com> <jreiser@BitWagon.com>
-Luca Berra <bluca@vodka.it> <bluca@comedia.it>
+Philippe Seewer <philippe.seewer@bfh.ch> 	<philippe.seewer-omB+W0Dpw2o@public.gmane.org>
+Seewer Philippe <philippe.seewer@bfh.ch> 	<philippe.seewer@bfh.ch>
+Philippe Seewer <philippe.seewer@bfh.ch> 	<philippe.seewer@bfh.ch>
+Victor Lowther <victor.lowther@gmail.com> 	<victor.lowther-Re5JQEeQqe8AvxtiuMwx3w@public.gmane.org>
+Harald Hoyer <harald@redhat.com> 		<harald-H+wXaHxf7aLQT0dZR+AlfA@public.gmane.org>
+Harald Hoyer <harald@redhat.com> 		<harald@eeepc.(none)>
+Mike Snitzer <snitzer@redhat.com> 		<msnitzer@redhat.com>
+Amerigo Wang <amwang@redhat.com> 		<xiyou.wangcong@gmail.com>
+Andrey Borzenkov <arvidjaar@gmail.com> 		<arvidjaar@mail.ru>
+Dan Horák <dhorak@redhat.com> 			<dan@danny.cz>
+John Reiser <jreiser@bitwagon.com> 		<jreiser@BitWagon.com>
+Luca Berra <bluca@vodka.it> 			<bluca@comedia.it>
+Dave Young <dyoung@redhat.com>	 		dyoung@redhat.com
+Frederick Grose <fgrose@sugarlabs.org> 		<fgrose@gmail.com>
+Frederic Crozat <fcrozat@suse.com> 		<fcrozat@mandriva.com>
diff --git a/AUTHORS b/AUTHORS
index fa69b91..3512cad 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -6,9 +6,9 @@ Warren Togami <wtogami@redhat.com>
 Will Woods <wwoods@redhat.com>
 Jeremy Katz <katzj@redhat.com>
 David Dillow <dave@thedillows.org>
+Dave Young <dyoung@redhat.com>
 Michal Soltys <soltys@ziu.info>
 Amerigo Wang <amwang@redhat.com>
-Dave Young <dyoung@redhat.com>
 Colin Guthrie <colin@mageia.org>
 Andrey Borzenkov <arvidjaar@gmail.com>
 Andreas Thienemann <andreas@bawue.net>
@@ -17,53 +17,61 @@ Peter Jones <pjones@redhat.com>
 John Reiser <jreiser@bitwagon.com>
 Luca Berra <bluca@vodka.it>
 Daniel Drake <dsd@laptop.org>
+Dan Horák <dhorak@redhat.com>
 Leho Kraav <leho@kraav.com>
 Brian C. Lane <bcl@redhat.com>
 Marc Grimme <grimme@atix.de>
 Chao Wang <chaowang@redhat.com>
-Dan Horák <dhorak@redhat.com>
+Frederic Crozat <fcrozat@suse.com>
 Jesse Keating <jkeating@redhat.com>
 Milan Broz <mbroz@redhat.com>
 Roberto Sassu <roberto.sassu@polito.it>
-dyoung@redhat.com <dyoung@redhat.com>
 Anton Blanchard <anton@samba.org>
 Bill Nottingham <notting@redhat.com>
 David Cantrell <dcantrell@redhat.com>
-Frederic Crozat <fcrozat@suse.com>
 Jon Ander Hernandez <jonan.h@gmail.com>
 Juan RP <xtraeme@gmail.com>
 Lance Albertson <lance@osuosl.org>
 Marian Ganisin <mganisin@redhat.com>
 Michael Ploujnikov <plouj@somanetworks.com>
 Peter Rajnoha <prajnoha@redhat.com>
+Radek Vykydal <rvykydal@redhat.com>
+WANG Chao <chaowang@redhat.com>
+Wim Muskee <wimmuskee@gmail.com>
 Alan Jenkins <alan-jenkins@tuffmail.co.uk>
 Alan Pevec <apevec@redhat.com>
+Baoquan He <bhe@redhat.com>
 Colin Walters <walters@verbum.org>
 Dennis Gilmore <dennis@ausil.us>
+Frederick Grose <fgrose@sugarlabs.org>
 Ian Dall <ian@beware.dropbear.id.au>
 James Buren <ryuo@frugalware.org>
 Joey Boggs <jboggs@redhat.com>
 Mike Snitzer <snitzer@redhat.com>
 Przemysław Rudy <prudy1@o2.pl>
+Stefan Reimer <it@startux.de>
 Thomas Lange <lange@informatik.uni-koeln.de>
 Vivek Goyal <vgoyal@redhat.com>
 Vladislav Bogdanov <bubble@hoster-ok.com>
-Wim Muskee <wimmuskee@gmail.com>
 Alexander Todorov <atodorov@redhat.com>
 Andy Lutomirski <luto@mit.edu>
 Anssi Hannula <anssi@mageia.org>
 Canek Peláez Valdés <caneko@gmail.com>
 Chris Leech <cleech@redhat.com>
 Christian Heinz <christian.ch.heinz@gmail.com>
+Cong Wang <amwang@redhat.com>
 Dave Jones <davej@redhat.com>
+Dave Young <dave@redhat.com>
 Duane Griffin <duaneg@dghda.com>
-Frederic Crozat <fcrozat@mandriva.com>
-Frederick Grose <fgrose@gmail.com>
 Glen Gray <slaine@slaine.org>
 Hermann Gausterer <git-dracut-2012@mrq1.org>
 James Laska <jlaska@redhat.com>
 Jan Stodola <jstodola@redhat.com>
+Jiri Pirko <jiri@resnulli.us>
+Joe Lawrence <Joe.Lawrence@stratus.com>
+Kevin Yung <Kevin.Yung@myob.com>
 Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
+Lars R. Damerow <lars@pixar.com>
 Lennert Buytenhek <buytenh@wantstofly.org>
 Lubomir Rintel <lkundrak@v3.sk>
 Matt <smoothsailing72@hotmail.com>
@@ -71,17 +79,18 @@ Matt Smith <shadowfax@gmx.com>
 Michal Schmidt <mschmidt@redhat.com>
 Mike Gorse <mgorse@suse.com>
 Munehiro Matsuda <haro@kgt.co.jp>
+Nicolas Chauvet <kwizart@gmail.com>
 Nikoli <nikoli@lavabit.com>
 Olivier Blin <dev@blino.org>
 Paolo Bonzini <pbonzini@redhat.com>
 Peter Robinson <pbrobinson@fedoraproject.org>
 Pádraig Brady <P@draigBrady.com>
 Quentin Armitage <quentin@armitage.org.uk>
-Radek Vykydal <rvykydal@redhat.com>
+Robert Buchholz <rbu@goodpoint.de>
 Sergey Fionov <fionov@gmail.com>
 Srinivasa T N <seenutn@linux.vnet.ibm.com>
-Stefan Reimer <it@startux.de>
 Thilo Bangert <thilo.bangert@gmx.net>
+Thomas Backlund <tmb@mageia.org>
 Tomasz Torcz <tomek@pipebreaker.pl>
 Vadim Kuznetsov <vadimk@gentoo.org>
 Ville Skyttä <ville.skytta@iki.fi>
