From a3f00efc99ac4736be3a1e24259d691bb9a9e19d Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 8 Mar 2012 10:58:16 +0100
Subject: [PATCH] 40network/kill-dhclient.sh: kill dhclient silently

---
 modules.d/40network/kill-dhclient.sh |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/40network/kill-dhclient.sh b/modules.d/40network/kill-dhclient.sh
index 2f834c1..78060f5 100755
--- a/modules.d/40network/kill-dhclient.sh
+++ b/modules.d/40network/kill-dhclient.sh
@@ -5,7 +5,7 @@
 for f in /tmp/dhclient.*.pid; do
     [ -e $f ] || continue
     read PID < $f;
-    kill $PID;
+    kill $PID >/dev/null 2>&1
 done
 
 sleep 0.1
@@ -13,5 +13,5 @@ sleep 0.1
 for f in /tmp/dhclient.*.pid; do
     [ -e $f ] || continue
     read PID < $f;
-    kill -9 $PID;
+    kill -9 $PID >/dev/null 2>&1
 done
