From 9f355169f454987405fcb79dc3eba6a280981a8c Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 3 Jul 2012 14:20:14 +0200
Subject: [PATCH] manpages: simplify AUTHORS

---
 dracut.8.asc         |   24 ++++++++----------------
 dracut.cmdline.7.asc |    3 +--
 2 files changed, 9 insertions(+), 18 deletions(-)

diff --git a/dracut.8.asc b/dracut.8.asc
index 196e2e8..3c0efb9 100644
--- a/dracut.8.asc
+++ b/dracut.8.asc
@@ -364,29 +364,21 @@ link:$$https://dracut.wiki.kernel.org$$[https://dracut.wiki.kernel.org]
 
 AUTHORS
 -------
-*Harald Hoyer*::
-    Project Leader and Developer
+Harald Hoyer
 
-*Victor Lowther*::
-    Developer
+Victor Lowther
 
-*Philippe Seewer*::
-    Developer
+Philippe Seewer
 
-*Warren Togami*::
-    Developer
+Warren Togami
 
-*Amadeusz Żołnowski*::
-    Developer
+Amadeusz Żołnowski
 
-*Jeremy Katz*::
-    Developer
+Jeremy Katz
 
-*David Dillow*::
-    Developer
+David Dillow
 
-*Will Woods*::
-    Developer
+Will Woods
 
 SEE ALSO
 --------
diff --git a/dracut.cmdline.7.asc b/dracut.cmdline.7.asc
index 3bfb53a..997c513 100644
--- a/dracut.cmdline.7.asc
+++ b/dracut.cmdline.7.asc
@@ -650,8 +650,7 @@ _/etc/cmdline.d/*.conf_::
 
 AUTHOR
 ------
-*Harald Hoyer*::
-    Project Leader and Developer
+Harald Hoyer
 
 SEE ALSO
 --------
