From f0e10d9351be491f5ff703b6ba4ed230a5673609 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 3 Jul 2012 18:20:46 +0200
Subject: [PATCH] usrmount/mount-usr.sh: check the right path with ismounted

---
 modules.d/98usrmount/mount-usr.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/98usrmount/mount-usr.sh b/modules.d/98usrmount/mount-usr.sh
index 3173241..c277d74 100755
--- a/modules.d/98usrmount/mount-usr.sh
+++ b/modules.d/98usrmount/mount-usr.sh
@@ -84,7 +84,7 @@ mount_usr()
             info "Mounting /usr"
             mount "$NEWROOT/usr" 2>&1 | vinfo
         fi
-        if ! ismounted /usr; then
+        if ! ismounted "$NEWROOT/usr"; then
             warn "Mounting /usr to $NEWROOT/usr failed"
             warn "*** Dropping you to a shell; the system will continue"
             warn "*** when you leave the shell."
