From 5861184e87e9d9164c093d3600d85c670c828886 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 2 Mar 2012 12:03:38 +0100
Subject: [PATCH] ifup.sh: check for "-m" to set manualup

---
 modules.d/40network/ifup.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/40network/ifup.sh b/modules.d/40network/ifup.sh
index b138e8e..b55362c 100755
--- a/modules.d/40network/ifup.sh
+++ b/modules.d/40network/ifup.sh
@@ -50,7 +50,7 @@ fi
 
 # disable manual ifup while netroot is set for simplifying our logic
 # in netroot case we prefer netroot to bringup $netif automaticlly
-[ -n "$2" ] && [ -z "$netroot" ] && manualup="$2"
+[ -n "$2" -a "$2" = "-m" ] && [ -z "$netroot" ] && manualup="$2"
 [ -z "$netroot" ] && [ -z "$manualup" ] && exit 0
 [ -n "$manualup" ] && >/tmp/net.$netif.manualup
 
