From d8caa679e06e779a04e2353cfeab80de47477d2a Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 3 Jul 2012 18:23:59 +0200
Subject: [PATCH] lsinitrd.sh: get rid of awk call

---
 lsinitrd.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lsinitrd.sh b/lsinitrd.sh
index f27f755..fc4b8b0 100755
--- a/lsinitrd.sh
+++ b/lsinitrd.sh
@@ -63,7 +63,7 @@ if [[ $# -eq 2 ]]; then
     exit $?
 fi
 
-echo "$image: $(du -h $image | awk '{print $1}')"
+echo "$image: $(du -h $image | while read a b; do echo $a;done)"
 echo "========================================================================"
 $CAT "$image" | cpio --extract --verbose --quiet --to-stdout 'lib/dracut/dracut-*' 2>/dev/null
 echo "========================================================================"
