From d0eedc4a5fa0bb23a5cf0aa6cea2f45095c74e1d Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 16 Apr 2013 13:38:16 +0200
Subject: [PATCH] .gitignore: ignore more files

---
 .gitignore | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/.gitignore b/.gitignore
index bec31ab..25506f1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -13,3 +13,7 @@ test*.img
 /.project
 /dracut-version.sh
 /install/dracut-install
+/*.rpm
+/*.[0-9]
+/modules.d/98systemd/*.service.8
+/*.sign
