From eaf4cb6bbb6367c967d1af3a00af5be3eba98a56 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 5 Jul 2012 06:57:11 +0200
Subject: [PATCH] Makefile: mkinitrd man page install typo

---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 78cf85d..07741d0 100644
--- a/Makefile
+++ b/Makefile
@@ -74,7 +74,7 @@ install: doc dracut-version.sh
 	install -m 0755 dracut-initramfs-restore.sh $(DESTDIR)$(pkglibdir)/dracut-initramfs-restore
 	cp -arx modules.d $(DESTDIR)$(pkglibdir)
 	install -m 0644 lsinitrd.1 $(DESTDIR)$(mandir)/man1/lsinitrd.1
-	install -m 0644 mkdinitrd.8 $(DESTDIR)$(mandir)/man8/mkinitrd.8
+	install -m 0644 mkinitrd.8 $(DESTDIR)$(mandir)/man8/mkinitrd.8
 	install -m 0644 dracut.8 $(DESTDIR)$(mandir)/man8/dracut.8
 	install -m 0644 dracut-catimages.8 \
                         $(DESTDIR)$(mandir)/man8/dracut-catimages.8
