From 0635530dda6e12e2475b43bd65ca4943ce36a7e7 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 8 Mar 2012 12:25:30 +0100
Subject: [PATCH] TEST-20-NFS/test.sh: fsck with "-a"

---
 test/TEST-20-NFS/test.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/TEST-20-NFS/test.sh b/test/TEST-20-NFS/test.sh
index fcd424f..a6bde0e 100755
--- a/test/TEST-20-NFS/test.sh
+++ b/test/TEST-20-NFS/test.sh
@@ -12,7 +12,7 @@ run_server() {
     # Start server first
     echo "NFS TEST SETUP: Starting DHCP/NFS server"
 
-    fsck $TESTDIR/server.ext3 || return 1
+    fsck -a $TESTDIR/server.ext3 || return 1
     $testdir/run-qemu \
 	-hda $TESTDIR/server.ext3 -m 256M -nographic \
 	-net nic,macaddr=52:54:00:12:34:56,model=e1000 \
