From d33d60774f38f43053c2cd1811b78c4f1af0fdc9 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 8 Mar 2012 13:40:21 +0100
Subject: [PATCH] dracut.spec: add 98pollcdrom

---
 dracut.spec |    1 +
 1 file changed, 1 insertion(+)

diff --git a/dracut.spec b/dracut.spec
index 91b3451..0275af3 100644
--- a/dracut.spec
+++ b/dracut.spec
@@ -272,6 +272,7 @@ rm -rf $RPM_BUILD_ROOT
 %{dracutlibdir}/modules.d/97masterkey
 %{dracutlibdir}/modules.d/98ecryptfs
 %{dracutlibdir}/modules.d/98integrity
+%{dracutlibdir}/modules.d/98pollcdrom
 %{dracutlibdir}/modules.d/98selinux
 %{dracutlibdir}/modules.d/98syslog
 %{dracutlibdir}/modules.d/98usrmount
