From e5efb6a798788195e2c1154deb582abf2dc0c6f5 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 6 Jul 2012 12:29:44 +0200
Subject: [PATCH] dracut.asc: add lsinitrd and mkinitrd

---
 dracut.asc |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dracut.asc b/dracut.asc
index 6d93379..e1f756c 100644
--- a/dracut.asc
+++ b/dracut.asc
@@ -1022,6 +1022,12 @@ include::dracut.conf.5.asc[]
 [[dracutcmdline7]]
 include::dracut.cmdline.7.asc[]
 
+[[lsinitrd1]]
+include::lsinitrd.1.asc[]
+
+[[mkinitrd8]]
+include::mkinitrd.8.asc[]
+
 :leveloffset: 0
 [appendix]
 License
