From 18595cede8d84212bc8f9055f5710f97998ed165 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 6 Jul 2012 12:30:03 +0200
Subject: [PATCH] dracut.8.asc: fixup NOTE sections

---
 dracut.8.asc |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/dracut.8.asc b/dracut.8.asc
index 3c0efb9..619fcaf 100644
--- a/dracut.8.asc
+++ b/dracut.8.asc
@@ -46,7 +46,7 @@ example:
 **-o, --omit** _<list of dracut modules>_::
     omit a space-separated list of dracut modules. This parameter can be
     specified multiple times.
-
++
 [NOTE]
 ===============================
 If [LIST] has multiple arguments, then you have to put these in quotes. For
@@ -130,7 +130,7 @@ example:
     specify a space-separated list of kernel filesystem modules to exclusively
     include in the generic initramfs. This parameter can be specified multiple
     times.
-
++
 [NOTE]
 ===============================
 If [LIST] has multiple arguments, then you have to put these in quotes. For
@@ -273,7 +273,6 @@ provide a valid _/etc/fstab_.
 ===============================
 If [LIST] has multiple arguments, then you have to put these in quotes. For
 example:
-+
 ----
 # dracut --install "/bin/foo /sbin/bar"  ...
 ----
