From 5d0404e3b373f29b80f4383fdd87a8525537d6e1 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 6 Jul 2012 12:30:50 +0200
Subject: [PATCH] dracut.cmdline.7.asc: fixup

---
 dracut.cmdline.7.asc |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dracut.cmdline.7.asc b/dracut.cmdline.7.asc
index 997c513..5f0396c 100644
--- a/dracut.cmdline.7.asc
+++ b/dracut.cmdline.7.asc
@@ -65,7 +65,8 @@ rootfstype=ext3
     do not honor special mount options for the root filesystem found in
     _/etc/fstab_ of the real root.
 
-**resume=**_<path to resume partition>_
+**resume=**_<path to resume partition>_::
+    resume from a swap partition
 +
 E.g.:
 +
