From 3a00cf9eccaadb44fd22d8f49c99b4d6dfe16009 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 14 Mar 2012 11:50:07 +0100
Subject: [PATCH] AUTHORS: update

---
 AUTHORS |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/AUTHORS b/AUTHORS
index 12dbe15..f7a1007 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -5,24 +5,24 @@ Warren Togami <wtogami@redhat.com>
 Amadeusz Żołnowski <aidecoe@aidecoe.name>
 Jeremy Katz <katzj@redhat.com>
 David Dillow <dave@thedillows.org>
-Michal Soltys <soltys@ziu.info>
 Will Woods <wwoods@redhat.com>
+Michal Soltys <soltys@ziu.info>
+Dave Young <dyoung@redhat.com>
+Amerigo Wang <amwang@redhat.com>
 Andrey Borzenkov <arvidjaar@gmail.com>
+Colin Guthrie <colin@mageia.org>
 Andreas Thienemann <andreas@bawue.net>
 Hans de Goede <hdegoede@redhat.com>
 Peter Jones <pjones@redhat.com>
-Amerigo Wang <amwang@redhat.com>
-Colin Guthrie <colin@mageia.org>
 John Reiser <jreiser@bitwagon.com>
 Luca Berra <bluca@vodka.it>
-Dave Young <dyoung@redhat.com>
 Daniel Drake <dsd@laptop.org>
 Marc Grimme <grimme@atix.de>
+Dan Horák <dhorak@redhat.com>
 Roberto Sassu <roberto.sassu@polito.it>
 Anton Blanchard <anton@samba.org>
 Bill Nottingham <notting@redhat.com>
 Brian C. Lane <bcl@redhat.com>
-Dan Horák <dhorak@redhat.com>
 David Cantrell <dcantrell@redhat.com>
 Lance Albertson <lance@osuosl.org>
 Marian Ganisin <mganisin@redhat.com>
@@ -47,6 +47,7 @@ Frederic Crozat <fcrozat@mandriva.com>
 Glen Gray <slaine@slaine.org>
 Hermann Gausterer <git-dracut-2012@mrq1.org>
 James Laska <jlaska@redhat.com>
+Jan Stodola <jstodola@redhat.com>
 Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
 Leho Kraav <leho@kraav.com>
 Lubomir Rintel <lkundrak@v3.sk>
