From 6d385c7111c81fe730c1823b232d19e8d42f50d4 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 20 Sep 2011 11:16:53 +0200
Subject: [PATCH] mount securityfs in a seperate dracut module

---
 dracut.spec |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/dracut.spec b/dracut.spec
index 76f4fe1..f9848ed 100644
--- a/dracut.spec
+++ b/dracut.spec
@@ -247,6 +247,7 @@ rm -rf $RPM_BUILD_ROOT
 %{_datadir}/dracut/modules.d/95zfcp
 %{_datadir}/dracut/modules.d/95terminfo
 %{_datadir}/dracut/modules.d/95udev-rules
+%{_datadir}/dracut/modules.d/96securityfs
 %{_datadir}/dracut/modules.d/97biosdevname
 %{_datadir}/dracut/modules.d/97masterkey
 %{_datadir}/dracut/modules.d/98ecryptfs
