From a421016671d59d99d182194223e83a6a616dedf2 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 6 Jul 2012 13:48:05 +0200
Subject: [PATCH] dracut.sh: keep vim syntax highlighting happy

---
 dracut.sh |    1 +
 1 file changed, 1 insertion(+)

diff --git a/dracut.sh b/dracut.sh
index c1be619..db2e33b 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -641,6 +641,7 @@ unset omit_drivers_corrected
 for ((i=0; i < ${#dracut_args[@]}; i++)); do
     strstr "${dracut_args[$i]}" " " && \
         dracut_args[$i]="\"${dracut_args[$i]}\""
+        #" keep vim happy
 done
 ddebug "Executing: $0 ${dracut_args[@]}"
 
