From 0028ffac74574e39e54507f4903ca93b042a04a1 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 9 Jul 2012 20:49:24 +0200
Subject: [PATCH] plymouth: add plymouth-wait-quit.service to initrd

---
 modules.d/50plymouth/module-setup.sh |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules.d/50plymouth/module-setup.sh b/modules.d/50plymouth/module-setup.sh
index 10aa0da..14d9e59 100755
--- a/modules.d/50plymouth/module-setup.sh
+++ b/modules.d/50plymouth/module-setup.sh
@@ -76,5 +76,9 @@ install() {
     inst_hook pre-trigger 10 "$moddir"/plymouth-pretrigger.sh
     inst_hook emergency 50 "$moddir"/plymouth-emergency.sh
     dracut_install readlink
+
+    if [[ -x $systemdutildir/systemd ]]; then
+        dracut_install -o $systemdsystemunitdir/plymouth-quit-wait.service
+    fi
 }
 
