From fc5b6b03281b10fc7c4de460574f1d5efb5f309a Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 11 Jul 2012 08:41:31 +0200
Subject: [PATCH] systemd/initrd-switch-root.target: add ConditionPathExist

---
 modules.d/98systemd/initrd-switch-root.target |    1 +
 1 file changed, 1 insertion(+)

diff --git a/modules.d/98systemd/initrd-switch-root.target b/modules.d/98systemd/initrd-switch-root.target
index feb7162..74647a5 100644
--- a/modules.d/98systemd/initrd-switch-root.target
+++ b/modules.d/98systemd/initrd-switch-root.target
@@ -14,3 +14,4 @@ Requires=initrd-switch-root.service
 Before=initrd-switch-root.service
 AllowIsolate=yes
 Wants=systemd-journald.service
+ConditionPathExists=/etc/initrd-release
