From 98eb6d57dffda383620f237eaee08c97bf35d863 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 11 Jul 2012 15:25:49 +0200
Subject: [PATCH] dracut.sh: for --include copy also the symbolic links

---
 dracut.sh |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dracut.sh b/dracut.sh
index 9be7cac..afd076a 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -934,9 +934,9 @@ while pop include_src src && pop include_target tgt; do
                         mkdir -m 0755 -p "$s"
                         chmod --reference="$i" "$s"
                     fi
-                    cp --reflink=auto --sparse=auto -pfLr -t "$s" "$i"/*
+                    cp --reflink=auto --sparse=auto -fa -t "$s" "$i"/*
                 else
-                    cp --reflink=auto --sparse=auto -pfLr -t "$s" "$i"
+                    cp --reflink=auto --sparse=auto -fa -t "$s" "$i"
                 fi
             done
         fi
