From d3be5a89e5714f43a6c288878e4b2746aa781159 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 19 Oct 2011 14:24:07 +0200
Subject: [PATCH] netroot: do not die, if arping failed

---
 modules.d/40network/netroot |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules.d/40network/netroot b/modules.d/40network/netroot
index 462da51..9d996cc 100755
--- a/modules.d/40network/netroot
+++ b/modules.d/40network/netroot
@@ -119,7 +119,7 @@ if [ -n "$netroot_ip" ]; then
     fi
 fi
 if [ -n "$dest" ] && ! arping -q -f -w 60 -I $netif $dest ; then
-    die "Resolving $dest via ARP on $netif failed"
+    dinfo "Resolving $dest via ARP on $netif failed"
 fi
 
 # Source netroot hooks before we start the handler
