From 624c70d763e9a201b5450fea10ac5e32939cf2a2 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 21 Oct 2011 10:15:09 +0200
Subject: [PATCH] 90livenet: do not install by default

---
 modules.d/90livenet/module-setup.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules.d/90livenet/module-setup.sh b/modules.d/90livenet/module-setup.sh
index a6ee3e0..5cfbb75 100755
--- a/modules.d/90livenet/module-setup.sh
+++ b/modules.d/90livenet/module-setup.sh
@@ -5,7 +5,7 @@ check() {
     # a live, host-only image doesn't really make a lot of sense
     [[ $hostonly ]] && return 1
     command -v wget >/dev/null || return 1
-    return 0
+    return 255
 }
 
 depends() {
