--- ./src/main/org/h2/util/DbDriverActivator.java.orig	2014-06-12 10:13:52.276707162 +0100
+++ ./src/main/org/h2/util/DbDriverActivator.java	2014-06-12 10:14:21.632227371 +0100
@@ -6,11 +6,8 @@
  */
 package org.h2.util;
 
-import java.util.Properties;
-import org.h2.engine.Constants;
 import org.osgi.framework.BundleActivator;
 import org.osgi.framework.BundleContext;
-import org.osgi.service.jdbc.DataSourceFactory;
 
 /**
  * The driver activator loads the H2 driver when starting the bundle. The driver
@@ -27,19 +24,6 @@
     @Override
     public void start(BundleContext bundleContext) {
         org.h2.Driver driver = org.h2.Driver.load();
-        Properties properties = new Properties();
-        properties.put(
-                DataSourceFactory.OSGI_JDBC_DRIVER_CLASS,
-                org.h2.Driver.class.getName());
-        properties.put(
-                DataSourceFactory.OSGI_JDBC_DRIVER_NAME,
-                "H2 JDBC Driver");
-        properties.put(
-                DataSourceFactory.OSGI_JDBC_DRIVER_VERSION,
-                Constants.getFullVersion());
-        bundleContext.registerService(
-                DataSourceFactory.class.getName(),
-                new OsgiDataSourceFactory(driver), properties);
     }
 
     /**
