--- iputils-s20121221-orig/ping_common.c	2013-02-01 08:28:29.836191171 +0100
+++ iputils-s20121221-new/ping_common.c	2013-02-01 08:28:11.013152725 +0100
@@ -269,7 +269,9 @@ void common_options(int ch)
 		char *ep;
 
 		errno = 0;
+		setlocale(LC_ALL, "C");
 		dbl = strtod(optarg, &ep);
+		setlocale(LC_ALL, "");
 
 		if (errno || *ep != '\0' ||
 		    !finite(dbl) || dbl < 0.0 || dbl >= (double)INT_MAX / 1000 - 1.0) {
