diff -up rng-tools-3/rngd.8.in.man rng-tools-3/rngd.8.in
--- rng-tools-3/rngd.8.in.man	2009-12-24 07:47:24.000000000 +0100
+++ rng-tools-3/rngd.8.in	2012-01-12 10:10:25.558096794 +0100
@@ -25,7 +25,7 @@ it is properly random.
 .PP
 The \fB\-f\fR or \fB\-\-foreground\fR options can be used to tell
 \fBrngd\fR to avoid forking on startup.  This is typically used for
-debugging.  The \fB\-f\fR or \fB\-\-foreground\fR options, which fork and put
+debugging.  The \fB\-b\fR or \fB\-\-background\fR options, which fork and put
 \fBrngd\fR into the background automatically, are the default.
 .PP
 The \fB\-r\fR or \fB\-\-rng-device\fR options can be used to select an
diff -up rng-tools-3/rngtest.1.in.man rng-tools-3/rngtest.1.in
--- rng-tools-3/rngtest.1.in.man	2009-12-24 07:47:24.000000000 +0100
+++ rng-tools-3/rngtest.1.in	2012-01-12 10:12:27.909567190 +0100
@@ -43,7 +43,7 @@ Exit after processing n input blocks, if
 Dump statistics every n blocks, if n is not zero.
 .TP
 \fB\-t\fR \fIn\fR, \fB\-\-timedstats=\fIn\fR (default: 0)
-Dump statistics every n secods, if n is not zero.
+Dump statistics every n seconds, if n is not zero.
 .TP
 \fB\-?\fR, \fB\-\-help\fR
 Give a short summary of all program options.
@@ -62,7 +62,7 @@ other statistics show a breakdown of the
 tests are defined on FIPS 140-1 and FIPS 140-2 errata of 2001-10-10. They
 were removed in FIPS 140-2 errata of 2002-12-03).
 .PP
-The speed statistics are taken for every 20000-bit block trasferred or
+The speed statistics are taken for every 20000-bit block transferred or
 processed.
 
 .SH EXIT STATUS
