diff -ur clang-3.4/lib/Driver/ToolChains.cpp clang-3.4.new/lib/Driver/ToolChains.cpp
--- clang-3.4/lib/Driver/ToolChains.cpp	2013-12-02 10:00:07.000000000 +0000
+++ clang-3.4.new/lib/Driver/ToolChains.cpp	2014-05-21 22:46:30.139499936 +0000
@@ -1118,18 +1118,21 @@
   static const char *const ARMTriples[] = { "arm-linux-gnueabi",
                                             "arm-linux-androideabi" };
   static const char *const ARMHFTriples[] = { "arm-linux-gnueabihf",
-                                              "armv7hl-redhat-linux-gnueabi" };
+                                              "armv7hl-redhat-linux-gnueabi",
+   					       "armv7hl-magic-linux-gnueabi" };
 
   static const char *const X86_64LibDirs[] = { "/lib64", "/lib" };
   static const char *const X86_64Triples[] = {
     "x86_64-linux-gnu", "x86_64-unknown-linux-gnu", "x86_64-pc-linux-gnu",
     "x86_64-redhat-linux6E", "x86_64-redhat-linux", "x86_64-suse-linux",
+    "x86_64-magic-linux",
     "x86_64-manbo-linux-gnu", "x86_64-linux-gnu", "x86_64-slackware-linux"
   };
   static const char *const X86LibDirs[] = { "/lib32", "/lib" };
   static const char *const X86Triples[] = {
     "i686-linux-gnu", "i686-pc-linux-gnu", "i486-linux-gnu", "i386-linux-gnu",
     "i386-redhat-linux6E", "i686-redhat-linux", "i586-redhat-linux",
+    "i686-magic-linux",
     "i386-redhat-linux", "i586-suse-linux", "i486-slackware-linux",
     "i686-montavista-linux"
   };
@@ -1139,6 +1142,8 @@
                                              "mips-mti-linux-gnu" };
   static const char *const MIPSELLibDirs[] = { "/lib" };
   static const char *const MIPSELTriples[] = { "mipsel-linux-gnu",
+					        "mipsel-magic-linux",
+					        "mipsel-magic-linux-gnu",
                                                "mipsel-linux-android" };
 
   static const char *const MIPS64LibDirs[] = { "/lib64", "/lib" };
@@ -1146,6 +1151,8 @@
                                                "mips-mti-linux-gnu" };
   static const char *const MIPS64ELLibDirs[] = { "/lib64", "/lib" };
   static const char *const MIPS64ELTriples[] = { "mips64el-linux-gnu",
+						  "mips64el-magic-linux",
+						  "mips64el-magic-linux-gnu",
                                                  "mips-mti-linux-gnu" };
 
   static const char *const PPCLibDirs[] = { "/lib32", "/lib" };
@@ -1157,17 +1164,19 @@
   static const char *const PPC64Triples[] = { "powerpc64-linux-gnu",
                                               "powerpc64-unknown-linux-gnu",
                                               "powerpc64-suse-linux",
-                                              "ppc64-redhat-linux" };
+                                              "ppc64-redhat-linux",
+					       "ppc64-magic-linux" };
   static const char *const PPC64LELibDirs[] = { "/lib64", "/lib" };
   static const char *const PPC64LETriples[] = { "powerpc64le-linux-gnu",
                                                 "powerpc64le-unknown-linux-gnu",
                                                 "powerpc64le-suse-linux",
-                                                "ppc64le-redhat-linux" };
+                                                "ppc64le-redhat-linux",
+						 "ppc64le-magic-linux" };
 
   static const char *const SystemZLibDirs[] = { "/lib64", "/lib" };
   static const char *const SystemZTriples[] = {
     "s390x-linux-gnu", "s390x-unknown-linux-gnu", "s390x-ibm-linux-gnu",
-    "s390x-suse-linux", "s390x-redhat-linux"
+    "s390x-suse-linux", "s390x-redhat-linux", "s390x-magic-linux"
   };
 
   switch (TargetTriple.getArch()) {
@@ -2161,6 +2170,7 @@
   RHEL5,
   RHEL6,
   Fedora,
+  Magic,
   OpenSUSE,
   UbuntuHardy,
   UbuntuIntrepid,
@@ -2186,6 +2196,10 @@
   return Distro == OpenSUSE;
 }
 
+static bool IsMagic(enum Distro Distro) {
+  return Distro == Magic;
+}
+
 static bool IsDebian(enum Distro Distro) {
   return Distro >= DebianLenny && Distro <= DebianJessie;
 }
@@ -2255,6 +2269,9 @@
   if (llvm::sys::fs::exists("/etc/SuSE-release"))
     return OpenSUSE;
 
+  if (llvm::sys::fs::exists("/etc/magic-release"))
+    return Magic;
+
   if (llvm::sys::fs::exists("/etc/exherbo-release"))
     return Exherbo;
 
@@ -2400,7 +2417,7 @@
   // ABI requires a mapping between the GOT and the symbol table.
   // Android loader does not support .gnu.hash.
   if (!IsMips && !IsAndroid) {
-    if (IsRedhat(Distro) || IsOpenSUSE(Distro) ||
+    if (IsRedhat(Distro) || IsOpenSUSE(Distro) || IsMagic(Distro) ||
         (IsUbuntu(Distro) && Distro >= UbuntuMaverick))
       ExtraOpts.push_back("--hash-style=gnu");
 
@@ -2409,12 +2426,12 @@
       ExtraOpts.push_back("--hash-style=both");
   }
 
-  if (IsRedhat(Distro))
+  if (IsRedhat(Distro) || IsMagic(Distro))
     ExtraOpts.push_back("--no-add-needed");
 
   if (Distro == DebianSqueeze || Distro == DebianWheezy ||
       Distro == DebianJessie || IsOpenSUSE(Distro) ||
-      (IsRedhat(Distro) && Distro != RHEL4 && Distro != RHEL5) ||
+      (IsRedhat(Distro) && Distro != RHEL4 && Distro != RHEL5) || IsMagic(Distro) || 
       (IsUbuntu(Distro) && Distro >= UbuntuKarmic))
     ExtraOpts.push_back("--build-id");
 
