diff -ur cfe-3.6.2.src.orig/lib/Driver/ToolChains.cpp cfe-3.6.2.src/lib/Driver/ToolChains.cpp
--- cfe-3.6.2.src.orig/lib/Driver/ToolChains.cpp	2015-08-23 17:32:22.118569402 +0000
+++ cfe-3.6.2.src/lib/Driver/ToolChains.cpp	2015-08-23 17:42:28.142553870 +0000
@@ -1296,7 +1296,8 @@
   static const char *const ARMTriples[] = { "arm-linux-gnueabi",
                                             "arm-linux-androideabi" };
   static const char *const ARMHFTriples[] = { "arm-linux-gnueabihf",
-                                              "armv7hl-redhat-linux-gnueabi" };
+                                              "armv7hl-redhat-linux-gnueabi",
+					      "armv7hl-magic-linux-gnueabi" };
   static const char *const ARMebLibDirs[] = { "/lib" };
   static const char *const ARMebTriples[] = { "armeb-linux-gnueabi",
                                               "armeb-linux-androideabi" };
@@ -1307,6 +1308,7 @@
   static const char *const X86_64Triples[] = {
     "x86_64-linux-gnu", "x86_64-unknown-linux-gnu", "x86_64-pc-linux-gnu",
     "x86_64-redhat-linux6E", "x86_64-redhat-linux", "x86_64-suse-linux",
+    "x86_64-magic-linux",
     "x86_64-manbo-linux-gnu", "x86_64-linux-gnu", "x86_64-slackware-linux",
     "x86_64-linux-android", "x86_64-unknown-linux"
   };
@@ -1315,6 +1317,7 @@
   static const char *const X86Triples[] = {
     "i686-linux-gnu", "i686-pc-linux-gnu", "i486-linux-gnu", "i386-linux-gnu",
     "i386-redhat-linux6E", "i686-redhat-linux", "i586-redhat-linux",
+    "i386-magic-linux", "i686-magic-linux",
     "i386-redhat-linux", "i586-suse-linux", "i486-slackware-linux",
     "i686-montavista-linux", "i686-linux-android", "i586-linux-gnu"
   };
@@ -1326,6 +1329,8 @@
   static const char *const MIPSELLibDirs[] = { "/lib" };
   static const char *const MIPSELTriples[] = { "mipsel-linux-gnu",
                                                "mipsel-linux-android",
+					       "mipsel-magic-linux",
+					       "mipsel-magic-linux-gnu",
                                                "mips-img-linux-gnu" };
 
   static const char *const MIPS64LibDirs[] = { "/lib64", "/lib" };
@@ -1338,6 +1343,8 @@
                                                  "mips-mti-linux-gnu",
                                                  "mips-img-linux-gnu",
                                                  "mips64el-linux-android",
+						 "mips64el-magic-linux",
+						 "mips64el-magic-linux-gnu",
                                                  "mips64el-linux-gnuabi64" };
 
   static const char *const PPCLibDirs[] = { "/lib32", "/lib" };
@@ -1349,12 +1356,14 @@
   static const char *const PPC64Triples[] = { "powerpc64-linux-gnu",
                                               "powerpc64-unknown-linux-gnu",
                                               "powerpc64-suse-linux",
-                                              "ppc64-redhat-linux" };
+                                              "ppc64-redhat-linux",
+					      "ppc64-magic-linux" };
   static const char *const PPC64LELibDirs[] = { "/lib64", "/lib" };
   static const char *const PPC64LETriples[] = { "powerpc64le-linux-gnu",
                                                 "powerpc64le-unknown-linux-gnu",
                                                 "powerpc64le-suse-linux",
-                                                "ppc64le-redhat-linux" };
+                                                "ppc64le-redhat-linux",
+						"ppc64le-magic-linux" };
 
   static const char *const SPARCv8LibDirs[] = { "/lib32", "/lib" };
   static const char *const SPARCv8Triples[] = { "sparc-linux-gnu",
@@ -1366,7 +1375,7 @@
   static const char *const SystemZLibDirs[] = { "/lib64", "/lib" };
   static const char *const SystemZTriples[] = {
     "s390x-linux-gnu", "s390x-unknown-linux-gnu", "s390x-ibm-linux-gnu",
-    "s390x-suse-linux", "s390x-redhat-linux"
+    "s390x-suse-linux", "s390x-redhat-linux", "s390x-magic-linux"
   };
 
   using std::begin;
@@ -2682,6 +2691,7 @@
   RHEL5,
   RHEL6,
   Fedora,
+  Magic,
   OpenSUSE,
   UbuntuHardy,
   UbuntuIntrepid,
@@ -2707,6 +2717,10 @@
   return Distro == OpenSUSE;
 }
 
+static bool IsMagic(enum Distro Distro) {
+  return Distro == Magic;
+}
+
 static bool IsDebian(enum Distro Distro) {
   return Distro >= DebianLenny && Distro <= DebianJessie;
 }
@@ -2760,6 +2774,18 @@
     return UnknownDistro;
   }
 
+  /*File = llvm::MemoryBuffer::getFile("/etc/magic-release");
+  if (File) {
+    StringRef Data = File.get()->getBuffer();
+    if ( Data.startswith("Magic") {
+      if (Data.find("release 4") != StringRef::npos)
+        return Magic;
+      else if (Data.find("release 3") != StringRef::npos)
+        return Magic;
+    }
+    return UnknownDistro;
+  }*/
+
   File = llvm::MemoryBuffer::getFile("/etc/debian_version");
   if (File) {
     StringRef Data = File.get()->getBuffer();
@@ -2777,6 +2803,9 @@
   if (llvm::sys::fs::exists("/etc/SuSE-release"))
     return OpenSUSE;
 
+  if (llvm::sys::fs::exists("/etc/magic-release"))
+    return Magic;
+
   if (llvm::sys::fs::exists("/etc/exherbo-release"))
     return Exherbo;
 
@@ -2955,7 +2984,7 @@
   // ABI requires a mapping between the GOT and the symbol table.
   // Android loader does not support .gnu.hash.
   if (!IsMips && !IsAndroid) {
-    if (IsRedhat(Distro) || IsOpenSUSE(Distro) ||
+    if (IsRedhat(Distro) || IsOpenSUSE(Distro) || IsMagic(Distro) ||
         (IsUbuntu(Distro) && Distro >= UbuntuMaverick))
       ExtraOpts.push_back("--hash-style=gnu");
 
@@ -2964,12 +2993,12 @@
       ExtraOpts.push_back("--hash-style=both");
   }
 
-  if (IsRedhat(Distro))
+  if (IsRedhat(Distro) || IsMagic(Distro))
     ExtraOpts.push_back("--no-add-needed");
 
   if (Distro == DebianSqueeze || Distro == DebianWheezy ||
       Distro == DebianJessie || IsOpenSUSE(Distro) ||
-      (IsRedhat(Distro) && Distro != RHEL4 && Distro != RHEL5) ||
+      (IsRedhat(Distro) && Distro != RHEL4 && Distro != RHEL5) || IsMagic(Distro) || 
       (IsUbuntu(Distro) && Distro >= UbuntuKarmic))
     ExtraOpts.push_back("--build-id");
 
