diff -ur clang-3.1.src.orig/lib/Driver/ToolChains.cpp clang-3.1.src/lib/Driver/ToolChains.cpp
--- clang-3.1.src.orig/lib/Driver/ToolChains.cpp	2012-12-20 17:36:19.000000000 +0000
+++ clang-3.1.src/lib/Driver/ToolChains.cpp	2012-12-20 17:37:31.000000000 +0000
@@ -1195,6 +1195,7 @@
     "x86_64-pc-linux-gnu",
     "x86_64-redhat-linux6E",
     "x86_64-redhat-linux",
+    "x86_64-magic-linux",
     "x86_64-suse-linux",
     "x86_64-manbo-linux-gnu",
     "x86_64-linux-gnu",
@@ -1209,6 +1210,7 @@
     "i686-redhat-linux",
     "i586-redhat-linux",
     "i386-redhat-linux",
+    "i686-magic-linux",
     "i586-suse-linux",
     "i486-slackware-linux"
   };
