diff -ur llvm-3.3.src-orig/tools/lldb/source/Plugins/Process/Linux/LinuxSignals.cpp llvm-3.3.src/tools/lldb/source/Plugins/Process/Linux/LinuxSignals.cpp
--- llvm-3.3.src-orig/tools/lldb/source/Plugins/Process/Linux/LinuxSignals.cpp	2014-05-17 10:43:35.258249880 +0000
+++ llvm-3.3.src/tools/lldb/source/Plugins/Process/Linux/LinuxSignals.cpp	2014-05-17 10:45:46.034684025 +0000
@@ -45,7 +45,9 @@
     ADDSIGNAL(PIPE,   false,  true,  true, "write to pipe with reading end closed");
     ADDSIGNAL(ALRM,   false,  false, true, "alarm");
     ADDSIGNAL(TERM,   false,  true,  true, "termination requested");
+#ifdef SIGSTKFLT
     ADDSIGNAL(STKFLT, false,  true,  true, "stack fault");
+#endif
     ADDSIGNAL(CHLD,   false,  false, true, "child process exit");
     ADDSIGNAL(CONT,   false,  true,  true, "process continue");
     ADDSIGNAL(STOP,   false,  true,  true, "process stop");
diff -ur llvm-3.3.src-orig/tools/lldb/source/Plugins/Process/Utility/RegisterContextMacOSXFrameBackchain.cpp llvm-3.3.src/tools/lldb/source/Plugins/Process/Utility/RegisterContextMacOSXFrameBackchain.cpp
--- llvm-3.3.src-orig/tools/lldb/source/Plugins/Process/Utility/RegisterContextMacOSXFrameBackchain.cpp	2014-05-17 10:43:35.276804577 +0000
+++ llvm-3.3.src/tools/lldb/source/Plugins/Process/Utility/RegisterContextMacOSXFrameBackchain.cpp	2014-05-17 10:47:57.618735771 +0000
@@ -149,7 +149,7 @@
 
             // TOOD: need a better way to detect when "long double" types are 
             // the same bytes size as "double"
-#if !defined(__arm__)
+#if !defined(__arm__) && !defined(_MSC_VER) && !defined(__mips__)
         case sizeof (long double):
             if (sizeof (long double) == sizeof(uint32_t))
             {
