diff -ur dpkg-1.17.6.orig/debian/dpkg.logrotate dpkg-1.17.6/debian/dpkg.logrotate
--- dpkg-1.17.6.orig/debian/dpkg.logrotate	2014-03-24 15:34:13.011287452 +0000
+++ dpkg-1.17.6/debian/dpkg.logrotate	2014-03-24 15:36:00.589284695 +0000
@@ -7,12 +7,3 @@
 	notifempty
 	create 644 root root
 }
-/var/log/alternatives.log {
-	monthly
-	rotate 12
-	compress
-	delaycompress
-	missingok
-	notifempty
-	create 644 root root
-}
 
