diff -ur dpkg-1.17.6.orig/scripts/Makefile.am dpkg-1.17.6/scripts/Makefile.am
--- dpkg-1.17.6.orig/scripts/Makefile.am	2014-03-24 15:34:13.085287450 +0000
+++ dpkg-1.17.6/scripts/Makefile.am	2014-03-24 15:35:11.851285944 +0000
@@ -21,7 +21,7 @@
 	dpkg-source \
 	dpkg-vendor
 
-changelogdir = $(pkglibdir)/parsechangelog
+changelogdir = $(pkglibexecdir)/parsechangelog
 changelog_SCRIPTS = \
 	changelog/debian
 
@@ -121,7 +121,7 @@
 		sed -e "s:^\#![[:space:]]*/usr/bin/perl:\#!$(PERL):" \
 		    -e "s:\$$CONFDIR[[:space:]]*=[[:space:]]*['\"][^'\"]*['\"]:\$$CONFDIR='$(pkgconfdir)':" \
 		    -e "s:\$$ADMINDIR[[:space:]]*=[[:space:]]*['\"][^'\"]*['\"]:\$$ADMINDIR='$(admindir)':" \
-		    -e "s:\$$LIBDIR[[:space:]]*=[[:space:]]*['\"][^'\"]*['\"]:\$$LIBDIR='$(pkglibdir)':" \
+		    -e "s:\$$LIBDIR[[:space:]]*=[[:space:]]*['\"][^'\"]*['\"]:\$$LIBDIR='$(pkglibexecdir)':" \
 		    -e "s:\$$DATADIR[[:space:]]*=[[:space:]]*['\"][^'\"]*['\"]:\$$DATADIR='$(pkgdatadir)':" \
 		    -e "s:\$$PROGVERSION[[:space:]]*=[[:space:]]*['\"][^'\"]*[\"']:\$$PROGVERSION='$(PACKAGE_VERSION)':"
 
