diff -up PackageKit-0.7.4/configure.ac.xulrunner2 PackageKit-0.7.4/configure.ac
--- PackageKit-0.7.4/configure.ac.xulrunner2	2012-04-24 03:27:24.000000000 -0500
+++ PackageKit-0.7.4/configure.ac	2012-05-25 09:15:04.170891611 -0500
@@ -349,7 +349,7 @@ AC_ARG_ENABLE(browser_plugin, AS_HELP_ST
 	      enable_browser_plugin=$enableval,enable_browser_plugin=yes)
 if test x$enable_browser_plugin = xyes; then
 	PKG_CHECK_MODULES(PK_BROWSER_PLUGIN,	\
-			  mozilla-plugin >= 8.0	\
+			  mozilla-plugin >= 2	\
 			  gio-unix-2.0		\
 			  nspr >= 4.8		\
 			  cairo			\
