diff -ur PackageKit-0.6.20.orig/contrib/browser-plugin/pk-main.c PackageKit-0.6.20/contrib/browser-plugin/pk-main.c
--- PackageKit-0.6.20.orig/contrib/browser-plugin/pk-main.c	2012-01-05 03:39:42.000000000 -0500
+++ PackageKit-0.6.20/contrib/browser-plugin/pk-main.c	2012-01-05 03:40:02.000000000 -0500
@@ -543,7 +543,7 @@
 /**
  * NP_GetMIMEDescription:
  **/
-char *
+const char *
 NP_GetMIMEDescription (void)
 {
 	pk_debug ("NP_GetMIMEDescription");
