--- lisp/frame.el.orig	2011-04-29 07:31:09.000000000 -0600
+++ lisp/frame.el	2011-08-05 12:13:03.208221581 -0600
@@ -602,7 +602,7 @@
 	 (next (next-frame this 'visible-nomini)))
     ;; then search iconified frames
     (if (eq this next)
-	(setq next (next-frame 'visible-iconic-nomini)))
+	(setq next (next-frame this 'visible-iconic-nomini)))
     (if (eq this next)
 	;; otherwise, make a new frame
 	(make-frame)
