diff --git a/configure.ac b/configure.ac
index 059ca8b..2ff18a8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -772,7 +772,7 @@ LUA_EOF
           done
         fi
 
-        LUA_RRD_LIBDIR="$langpref/lib/lua/$lua_vdot"
+        LUA_RRD_LIBDIR="$libdir/lua/$lua_vdot"
         # if lua 5.0 can't find compat-5.1, force installation of
         # compat-5.1.lua together with RRDtool.
         if test "$lua_vdot" = "5.0" -a "$LUA_HAVE_COMPAT51" != "HAVE_COMPAT51"; then
diff --git a/bindings/tcl/Makefile.am b/bindings/tcl/Makefile.am
index c0e8b0f..265191d 100644
--- a/bindings/tcl/Makefile.am
+++ b/bindings/tcl/Makefile.am
@@ -27,8 +27,9 @@ tclpkgdir = @TCL_PACKAGE_DIR@
 tclpkg_DATA = pkgIndex.tcl
 tclpkg_SCRIPTS = ifOctets.tcl
 else
-pkgdata_DATA = pkgIndex.tcl
-pkgdata_SCRIPTS = ifOctets.tcl
+pkgindexdir = $(pkglibdir)
+pkgindex_DATA = pkgIndex.tcl
+pkgindex_SCRIPTS = ifOctets.tcl
 endif
 
 # Automake doen't like `tclrrd$(VERSION)$(TCL_SHLIB_SUFFIX)' as
