diff -ur ClusterLabs-pacemaker-394e906.orig/lib/services/systemd.c ClusterLabs-pacemaker-394e906/lib/services/systemd.c
--- ClusterLabs-pacemaker-394e906.orig/lib/services/systemd.c	2013-01-06 05:18:57.000000000 +0000
+++ ClusterLabs-pacemaker-394e906/lib/services/systemd.c	2013-01-06 05:19:22.000000000 +0000
@@ -55,7 +55,7 @@
     GError *error = NULL;
     GDBusProxy *proxy = NULL;
     
-    g_type_init();
+    //g_type_init();
 
     if(path == NULL) {
         path = BUS_PATH;
diff -ur ClusterLabs-pacemaker-394e906.orig/lib/services/upstart.c ClusterLabs-pacemaker-394e906/lib/services/upstart.c
--- ClusterLabs-pacemaker-394e906.orig/lib/services/upstart.c	2013-01-06 05:18:57.000000000 +0000
+++ ClusterLabs-pacemaker-394e906/lib/services/upstart.c	2013-01-06 05:19:13.000000000 +0000
@@ -55,7 +55,7 @@
     GError *error = NULL;
     GDBusProxy *proxy = NULL;
     
-    g_type_init();
+    //g_type_init();
 
     if(path == NULL) {
         path = BUS_PATH;
