From 25828196a0862118a013c00efe1e63e5831da6d4 Mon Sep 17 00:00:00 2001
From: Angus Salkeld <asalkeld@redhat.com>
Date: Mon, 29 Oct 2012 11:54:23 +1100
Subject: [PATCH] TEST: fix typo s/,/; in check_ipc.c

Signed-off-by: Angus Salkeld <asalkeld@redhat.com>
---
 tests/check_ipc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/check_ipc.c b/tests/check_ipc.c
index c396d61..09ad619 100644
--- a/tests/check_ipc.c
+++ b/tests/check_ipc.c
@@ -620,7 +620,7 @@ test_ipc_bulk_events(void)
 
 	events_received = 0;
 	cl = qb_loop_create();
-	res = qb_ipcc_fd_get(conn, &fd),
+	res = qb_ipcc_fd_get(conn, &fd);
 	ck_assert_int_eq(res, 0);
 	res = qb_loop_poll_add(cl, QB_LOOP_MED,
 			 fd, POLLIN,
@@ -692,7 +692,7 @@ test_ipc_event_on_created(void)
 
 	events_received = 0;
 	cl = qb_loop_create();
-	res = qb_ipcc_fd_get(conn, &fd),
+	res = qb_ipcc_fd_get(conn, &fd);
 	ck_assert_int_eq(res, 0);
 	res = qb_loop_poll_add(cl, QB_LOOP_MED,
 			 fd, POLLIN,
-- 
1.7.11.7

