--- tidyp-1.00/configure.ac	2010-04-25 06:25:59.000000000 +0100
+++ tidyp-1.00/configure.ac	2010-04-26 11:19:26.379099622 +0100
@@ -62,7 +62,7 @@
 if test $debug_build = yes; then
 	CFLAGS="$save_cflags -g"
 else
-	CFLAGS="-O2 $save_cflags"
+	CFLAGS="$save_cflags"
 fi
 #
 # =============================================
--- tidyp-0.99/configure	2010-02-22 06:25:50.000000000 +0000
+++ tidyp-0.99/configure	2010-03-01 14:39:35.461711475 +0000
@@ -3110,7 +3110,7 @@
 if test $debug_build = yes; then
 	CFLAGS="$save_cflags -g"
 else
-	CFLAGS="-O2 $save_cflags"
+	CFLAGS="$save_cflags"
 fi
 #
 # =============================================
