diff -up telepathy-qt-0.9.3/cmake/modules/FindTelepathyFarstream.cmake.compat telepathy-qt-0.9.3/cmake/modules/FindTelepathyFarstream.cmake
--- telepathy-qt-0.9.3/cmake/modules/FindTelepathyFarstream.cmake.compat	2012-07-13 12:40:23.000000000 -0500
+++ telepathy-qt-0.9.3/cmake/modules/FindTelepathyFarstream.cmake	2012-10-31 14:46:50.455757306 -0500
@@ -23,9 +23,9 @@ if (NOT WIN32)
     # in the find_path() and find_library() calls
     find_package(PkgConfig)
     if (TELEPATHY_FARSTREAM_MIN_VERSION)
-        PKG_CHECK_MODULES(PC_TELEPATHY_FARSTREAM telepathy-farstream>=${TELEPATHY_FARSTREAM_MIN_VERSION})
+        PKG_CHECK_MODULES(PC_TELEPATHY_FARSTREAM telepathy-farstream-0.4>=${TELEPATHY_FARSTREAM_MIN_VERSION} telepathy-farstream>=${TELEPATHY_FARSTREAM_MIN_VERSION})
     else (TELEPATHY_FARSTREAM_MIN_VERSION)
-        PKG_CHECK_MODULES(PC_TELEPATHY_FARSTREAM telepathy-farstream)
+        PKG_CHECK_MODULES(PC_TELEPATHY_FARSTREAM telepathy-farstream-0.4 telepathy-farstream)
     endif (TELEPATHY_FARSTREAM_MIN_VERSION)
     set(TELEPATHY_FARSTREAM_DEFINITIONS ${PC_TELEPATHY_FARSTREAM_CFLAGS_OTHER})
 endif (NOT WIN32)
@@ -34,10 +34,10 @@ find_path(TELEPATHY_FARSTREAM_INCLUDE_DI
    PATHS
    ${PC_TELEPATHY_FARSTREAM_INCLUDEDIR}
    ${PC_TELEPATHY_FARSTREAM_INCLUDE_DIRS}
-   PATH_SUFFIXES telepathy-1.0
+   PATH_SUFFIXES telepathy-1.0-compat telepathy-1.0
    )
 
-find_library(TELEPATHY_FARSTREAM_LIBRARIES NAMES telepathy-farstream
+find_library(TELEPATHY_FARSTREAM_LIBRARIES NAMES telepathy-farstream-0.4 telepathy-farstream
    PATHS
    ${PC_TELEPATHY_FARSTREAM_LIBDIR}
    ${PC_TELEPATHY_FARSTREAM_LIBRARY_DIRS}
