https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=728675
Message #17 ï¼š
tags 728675 +patch
thanks


Hi,

the number of arguments in QT4_CREATE_MOC_COMMAND seems to have
changed [1].   By applying the following simple patch the package
builds again:

--- telepathy-qt-0.9.3.orig/cmake/modules/FindQt.cmake
+++ telepathy-qt-0.9.3/cmake/modules/FindQt.cmake
@@ -74,7 +74,7 @@ ENDMACRO(QT_GET_MOC_FLAGS)

 MACRO(QT_CREATE_MOC_COMMAND infile outfile moc_flags moc_options)
   IF(QT_VERSION_MAJOR MATCHES 4)
-    QT4_CREATE_MOC_COMMAND(${infile} ${outfile} "${moc_flags}" "${moc_options}")
+    QT4_CREATE_MOC_COMMAND(${infile} ${outfile} "${moc_flags}" "${moc_options}" "${moc_target}")
   ELSE(QT_VERSION_MAJOR MATCHES 4)
     IF(QT_VERSION_MAJOR MATCHES 5)
       QT5_CREATE_MOC_COMMAND(${infile} ${outfile} "${moc_flags}" "${moc_options}")
