commit b4881a61a24b3df9af47ee6d7e7de4ca8d369f81
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1324747996 -0600

    Rename a few stragglers

diff --git a/src/gvcore/imageviewconfig.kcfg b/src/gvcore/imageviewconfig.kcfg
index e5c1e04..d93051a 100644
--- a/src/gvcore/imageviewconfig.kcfg
+++ b/src/gvcore/imageviewconfig.kcfg
@@ -2,7 +2,7 @@
 <!DOCTYPE kcfg SYSTEM "http://www.kde.org/standards/kcfg/1.0/kcfg.dtd">
 <kcfg> 
   <include>tqapplication.h</include>
-  <include>tqpalette.h</include>
+  <include>palette.h</include>
   <kcfgfile name="gwenviewrc"/> 
   <group name="pixmap widget">
     <entry name="smoothAlgorithm" key="smooth scale" type="Enum"> 
