commit 59c7639f5e9f3ef40b06cb0e011216cc748e3f85
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1337933593 -0500

    Define QT_CLEAN_NAMESPACE during libmng checks

diff --git a/configure.in.in b/configure.in.in
index f5e959c..1cdda0b 100644
--- a/configure.in.in
+++ b/configure.in.in
@@ -98,6 +98,8 @@ AC_SUBST(GV_ASM_DEFS)
 # libmng check (for gvmngformattype.*)
 #
 LIBMNG=
+CXXFLAGS_SAVE="$CXXFLAGS"
+CXXFLAGS="$CXXFLAGS -DQT_CLEAN_NAMESPACE"
 KDE_CHECK_HEADER(libmng.h,
     [
     KDE_CHECK_LIB(mng, mng_initialize,
@@ -106,6 +108,7 @@ KDE_CHECK_HEADER(libmng.h,
         LIBMNG="-lmng $LIBZ -lm"
         ], [], $LIBZ -lm)
     ])
+CXXFLAGS="$CXXFLAGS_SAVE"
 AC_SUBST(LIBMNG)
 if test -z "$LIBMNG"; then
     AC_WARN([Can't find libmng.h, Gwenview won't be compiled with MNG support])
