From e921bfe82068a1767311181ad7f4c78ed86e2577 Mon Sep 17 00:00:00 2001
From: Tomas Heinrich <theinric@redhat.com>
Date: Thu, 16 Jan 2014 11:43:55 +0100
Subject: [PATCH] bugfix: invert the logic for issuing a warning

---
 plugins/omjournal/omjournal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/omjournal/omjournal.c b/plugins/omjournal/omjournal.c
index 160c369..937d277 100644
--- a/plugins/omjournal/omjournal.c
+++ b/plugins/omjournal/omjournal.c
@@ -153,7 +153,7 @@ ENDdoAction
 BEGINparseSelectorAct
 CODESTARTparseSelectorAct
 CODE_STD_STRING_REQUESTparseSelectorAct(1)
-	if(strncmp((char*) p, ":omjournal:", sizeof(":omjournal:") - 1)) {
+	if(!strncmp((char*) p, ":omjournal:", sizeof(":omjournal:") - 1)) {
 		errmsg.LogError(0, RS_RET_LEGA_ACT_NOT_SUPPORTED,
 			"omjournal supports only v6+ config format, use: "
 			"action(type=\"omjournal\" ...)");
-- 
1.8.4.3

