From 612f881d541aaf3a8725276d41ff4ed3948e2d01 Mon Sep 17 00:00:00 2001
From: Rainer Gerhards <rgerhards@adiscon.com>
Date: Fri, 18 Oct 2013 07:11:44 +0200
Subject: [PATCH] bugfix: omelasticsearch did not compile on platforms without
 atomic instructions

---
 plugins/omelasticsearch/omelasticsearch.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/plugins/omelasticsearch/omelasticsearch.c b/plugins/omelasticsearch/omelasticsearch.c
index b82968d..8f51e1a 100644
--- a/plugins/omelasticsearch/omelasticsearch.c
+++ b/plugins/omelasticsearch/omelasticsearch.c
@@ -608,7 +608,7 @@ curlPost(instanceData *pData, uchar *message, int msglen, uchar **tpls, int nmsg
 		case CURLE_COULDNT_RESOLVE_PROXY:
 		case CURLE_COULDNT_CONNECT:
 		case CURLE_WRITE_ERROR:
-			STATSCOUNTER_INC(indexHTTPReqFail, mutHTTPReqFail);
+			STATSCOUNTER_INC(indexHTTPReqFail, mutIndexHTTPReqFail);
 			indexHTTPFail += nmsgs;
 			DBGPRINTF("omelasticsearch: we are suspending ourselfs due "
 				  "to failure %lld of curl_easy_perform()\n",
@@ -1001,18 +1001,18 @@ CODEmodInit_QueryRegCFSLineHdlr
 	/* support statistics gathering */
 	CHKiRet(statsobj.Construct(&indexStats));
 	CHKiRet(statsobj.SetName(indexStats, (uchar *)"omelasticsearch"));
-	STATSCOUNTER_INIT(indexSubmit, mutCtrIndexSubmit);
+	STATSCOUNTER_INIT(indexSubmit, mutIndexSubmit);
 	CHKiRet(statsobj.AddCounter(indexStats, (uchar *)"submitted",
-		ctrType_IntCtr, &indexSubmit));
-	STATSCOUNTER_INIT(indexHTTPFail, mutCtrIndexHTTPFail);
+		ctrType_Int, &indexSubmit));
+	STATSCOUNTER_INIT(indexHTTPFail, mutIndexHTTPFail);
 	CHKiRet(statsobj.AddCounter(indexStats, (uchar *)"failed.http",
-		ctrType_IntCtr, &indexHTTPFail));
-	STATSCOUNTER_INIT(indexHTTPReqFail, mutCtrIndexHTTPReqFail);
+		ctrType_Int, &indexHTTPFail));
+	STATSCOUNTER_INIT(indexHTTPReqFail, mutIndexHTTPReqFail);
 	CHKiRet(statsobj.AddCounter(indexStats, (uchar *)"failed.httprequests",
-		ctrType_IntCtr, &indexHTTPReqFail));
-	STATSCOUNTER_INIT(indexESFail, mutCtrIndexESFail);
+		ctrType_Int, &indexHTTPReqFail));
+	STATSCOUNTER_INIT(indexESFail, mutIndexESFail);
 	CHKiRet(statsobj.AddCounter(indexStats, (uchar *)"failed.es",
-		ctrType_IntCtr, &indexESFail));
+		ctrType_Int, &indexESFail));
 	CHKiRet(statsobj.ConstructFinalize(indexStats));
 ENDmodInit
 
-- 
1.8.5.3

