diff -up jsoncpp/SConstruct.optflags jsoncpp/SConstruct
--- jsoncpp/SConstruct.optflags	2010-03-13 05:59:50.000000000 -0500
+++ jsoncpp/SConstruct	2012-06-26 13:27:15.551157402 -0400
@@ -119,7 +119,7 @@ elif platform == 'mingw':
     env.Append( CPPDEFINES=[ "WIN32", "NDEBUG", "_MT" ] )
 elif platform.startswith('linux-gcc'):
     env.Tool( 'default' )
-    env.Append( LIBS = ['pthread'], CCFLAGS = "-Wall" )
+    env.Append( LIBS = ['pthread'], CCFLAGS = "-Wall @@OPTFLAGS@@" )
     env['SHARED_LIB_ENABLED'] = True
 else:
     print "UNSUPPORTED PLATFORM."
