--- branches/kdesupport/3.5/kdesupport/akode/plugins/xiph_decoder/flac113_decoder.cpp	2007/04/14 22:11:43	654042
+++ branches/kdesupport/3.5/kdesupport/akode/plugins/xiph_decoder/flac113_decoder.cpp	2007/04/14 22:14:06	654043
@@ -108,7 +108,7 @@
 static FLAC__StreamDecoderReadStatus flac_read_callback(
         const FLAC__StreamDecoder *,
         FLAC__byte buffer[],
-        unsigned *bytes,
+        size_t *bytes,
         void *client_data)
 {
     FLACDecoder::private_data *data = (FLACDecoder::private_data*)client_data;
