--- kdebase-runtime-4.5.0/kioslave/thumbnail/textcreator.cpp	2010-08-28 09:26:09.000000000 +0800
+++ kdebase-runtime-4.5.0/kioslave/thumbnail/textcreator.cpp	2010-08-28 09:32:18.638158660 +0800
@@ -53,15 +53,14 @@ TextCreator::~TextCreator()

 static QTextCodec *codecFromContent(const char *data, int dataSize)
 {
-#if 0 // ### Use this when KEncodingProber does not return junk encoding for UTF-8 data)
     KEncodingProber prober;
     prober.feed(data, dataSize);
-    return QTextCodec::codecForName(prober.encoding());
-#else
+    if (prober.confidence() > 0.6)
+        return QTextCodec::codecForName(prober.encoding());
+
     QByteArray ba = QByteArray::fromRawData(data, dataSize);
     // try to detect UTF text, fall back to locale default (which is usually UTF-8)
     return QTextCodec::codecForUtfText(ba, KGlobal::locale()->codecForEncoding());
-#endif
 }

 bool TextCreator::create(const QString &path, int width, int height, QImage &img)
