diff -ur colord-0.1.18.orig/configure colord-0.1.18/configure
--- colord-0.1.18.orig/configure	2012-04-13 01:36:59.000000000 -0400
+++ colord-0.1.18/configure	2012-04-13 01:38:05.000000000 -0400
@@ -16797,7 +16797,7 @@
 
 # udev rules go in /lib, not /usr/lib
 if test "$prefix" = "/usr" -o "$prefix" = "/usr/local" ; then
-  slashlibdir=/lib
+  slashlibdir=$prefix/lib
 else
   slashlibdir=$prefix/lib
 fi
diff -ur colord-0.1.18.orig/configure.ac colord-0.1.18/configure.ac
--- colord-0.1.18.orig/configure.ac	2012-04-13 01:36:59.000000000 -0400
+++ colord-0.1.18/configure.ac	2012-04-13 01:37:50.000000000 -0400
@@ -266,7 +266,7 @@
 
 # udev rules go in /lib, not /usr/lib
 if test "$prefix" = "/usr" -o "$prefix" = "/usr/local" ; then
-  slashlibdir=/lib
+  slashlibdir=$prefix/lib
 else
   slashlibdir=$prefix/lib
 fi
