From 102fee5e1e7661356502ca89d6a73889b826af6b Mon Sep 17 00:00:00 2001
From: Stanislav Ochotnicky <sochotnicky@redhat.com>
Date: Wed, 29 Jan 2014 17:23:28 +0100
Subject: [PATCH 01/15] [depgenerators] Switch to require java-headless instead
 java

---
 depgenerators/maven.req | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/depgenerators/maven.req b/depgenerators/maven.req
index 5bd6c34..c161e3d 100755
--- a/depgenerators/maven.req
+++ b/depgenerators/maven.req
@@ -158,7 +158,7 @@ class TagBuilder:
                             "%mvn_package {art} <package_name>\n".format(art=skipped)
             raise Exception(skipped_msg)
 
-        reqs = {'java': depmap.get_java_requires(),
+        reqs = {'java-headless': depmap.get_java_requires(),
                 'java-devel': depmap.get_java_devel_requires()}
         for name, version in reqs.iteritems():
             self.__process_jvm_req(name, version)
-- 
1.8.5.3

