From 7cf7e60b9ce308cf7d0cbbe3da142781ce736696 Mon Sep 17 00:00:00 2001
From: Stanislav Ochotnicky <sochotnicky@redhat.com>
Date: Wed, 29 Jan 2014 17:40:39 +0100
Subject: [PATCH 02/15] [test] Fix testsuite for java-headless change

---
 test/maven_req_test.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/maven_req_test.py b/test/maven_req_test.py
index 4551bb5..b99672f 100644
--- a/test/maven_req_test.py
+++ b/test/maven_req_test.py
@@ -26,7 +26,7 @@ class TestMavenReq(unittest.TestCase):
     @mavenreq(["require-java/require.xml"])
     def test_require_java(self, stdout, stderr, return_value):
         self.assertEquals(return_value, 0, stderr)
-        self.assertEquals(stdout, "jpackage-utils\nmvn(org.apache.maven:maven-project)\njava >= 1:1.6\n")
+        self.assertEquals(stdout, "jpackage-utils\nmvn(org.apache.maven:maven-project)\njava-headless >= 1:1.6\n")
 
     @mavenreq(["require-java-devel/require.xml"])
     def test_require_java_devel(self, stdout, stderr, return_value):
@@ -36,7 +36,7 @@ class TestMavenReq(unittest.TestCase):
     @mavenreq(["require-java-both/require.xml"])
     def test_require_java_both(self, stdout, stderr, return_value):
         self.assertEquals(return_value, 0, stderr)
-        self.assertEquals(stdout, "jpackage-utils\nmvn(org.apache.maven:maven-project)\njava >= 1:1.6\njava-devel >= 1:1.6\n")
+        self.assertEquals(stdout, "jpackage-utils\nmvn(org.apache.maven:maven-project)\njava-headless >= 1:1.6\njava-devel >= 1:1.6\n")
 
     @mavenreq(["require_parent/require.xml"])
     def test_require_parent(self, stdout, stderr, return_value):
-- 
1.8.5.3

