From 424bba5b517fa1c458d7f11b10e577ba59585bd2 Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Thu, 29 May 2014 13:08:11 +0200
Subject: [PATCH 06/10] [maven.req] Fix self-dependency detection code

---
 depgenerators/maven.req | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/depgenerators/maven.req b/depgenerators/maven.req
index 56fd27f..a42162e 100755
--- a/depgenerators/maven.req
+++ b/depgenerators/maven.req
@@ -242,13 +242,10 @@ class TagBuilder(object):
                     for compatVer in provided.compatVersions:
                         if dependency.resolvedVersion == compatVer:
                             return True
-                        else:
-                            return False
                 elif (not dependency.resolvedVersion and
                     not provided.is_compat()):
                     return True
-                else:
-                    return False
+        return False
 
 
     def filter_requires(self, requires):
-- 
1.9.3

