diff -up netkit-rsh-0.17/rshd/rshd.c.nohostcheck-compat netkit-rsh-0.17/rshd/rshd.c
--- netkit-rsh-0.17/rshd/rshd.c.nohostcheck-compat	2007-10-03 13:06:08.000000000 +0200
+++ netkit-rsh-0.17/rshd/rshd.c	2007-10-03 13:07:15.000000000 +0200
@@ -90,7 +90,7 @@ char rcsid[] = 
 static pam_handle_t *pamh;
 #endif /* USE_PAM */
 
-#define	OPTIONS	"ahlLn"
+#define	OPTIONS	"aDhlLn"
 
 static int keepalive = 1;
 static int check_all = 0;
@@ -658,6 +658,9 @@ main(int argc, char *argv[])
 			check_all = 1;
 			break;
 
+		case 'D':
+			break;
+
 		case 'h':
 			allow_root_rhosts = 1;
 			break;
diff -up netkit-rsh-0.17/rlogind/rlogind.c.nohostcheck-compat netkit-rsh-0.17/rlogind/rlogind.c
--- netkit-rsh-0.17/rlogind/rlogind.c.nohostcheck-compat	2007-10-03 13:06:08.000000000 +0200
+++ netkit-rsh-0.17/rlogind/rlogind.c	2007-10-03 13:06:45.000000000 +0200
@@ -450,9 +450,10 @@ int main(int argc, char **argv) {
     openlog("rlogind", LOG_PID | LOG_CONS, LOG_AUTH);
 
     opterr = 0;
-    while ((ch = getopt(argc, argv, "ahLln")) != EOF) {
+    while ((ch = getopt(argc, argv, "aDhLln")) != EOF) {
 	switch (ch) {
 	    case 'a': check_all = 1; break;
+	    case 'D': break;
 	    case 'h': allow_root_rhosts = 1; break;
 	    case 'L': deny_all_rhosts_hequiv = 1; break;
 	    case 'l': use_rhosts = 0; break;
