diff -up IPC-Shareable-0.60/lib/IPC/Shareable.pm.BAD IPC-Shareable-0.60/lib/IPC/Shareable.pm
--- IPC-Shareable-0.60/lib/IPC/Shareable.pm.BAD	2008-02-03 11:13:12.000000000 -0500
+++ IPC-Shareable-0.60/lib/IPC/Shareable.pm	2008-02-03 11:13:18.000000000 -0500
@@ -2,6 +2,7 @@ package IPC::Shareable;
 
 require 5.00503;
 use strict;
+no strict 'refs';
 use IPC::Semaphore;
 use IPC::Shareable::SharedMem;
 use IPC::SysV qw(
diff -up IPC-Shareable-0.60/lib/IPC/Shareable.pm.BAD IPC-Shareable-0.60/lib/IPC/Shareable.pm
--- IPC-Shareable-0.60/lib/IPC/Shareable.pm.BAD	2008-02-04 13:11:24.000000000 -0500
+++ IPC-Shareable-0.60/lib/IPC/Shareable.pm	2008-02-04 13:11:35.000000000 -0500
@@ -453,7 +453,7 @@ sub STORESIZE {
     my $n    = shift;
 
     $self->{_data} = _thaw($self->{_shm}) unless $self->{_lock};
-    $#{@{$self->{_data}}} = $n - 1;
+    $#{$self->{_data}} = $n - 1;
     if ($self->{_lock} & LOCK_EX) {
         $self->{_was_changed} = 1;
     } else {
