diff -upr mate-control-center-1.8.1-orig/capplets/keyboard/mate-keyboard-properties-a11y.c mate-control-center-1.8.1/capplets/keyboard/mate-keyboard-properties-a11y.c
--- mate-control-center-1.8.1-orig/capplets/keyboard/mate-keyboard-properties-a11y.c	2014-03-05 21:55:21.000000000 +0100
+++ mate-control-center-1.8.1/capplets/keyboard/mate-keyboard-properties-a11y.c	2014-04-19 13:40:12.534518738 +0200
@@ -145,7 +145,7 @@ notifications_button_clicked_cb (GtkWidg
 	g_settings_bind (a11y_settings, "feature-state-change-beep", w, "active", G_SETTINGS_BIND_DEFAULT);
 
 	w = NWID ("togglekeys_enable");
-	g_settings_bind (a11y_settings, "togglekeys_-enable", w, "active", G_SETTINGS_BIND_DEFAULT);
+	g_settings_bind (a11y_settings, "togglekeys-enable", w, "active", G_SETTINGS_BIND_DEFAULT);
 
 	w = NWID ("stickykeys_modifier_beep");
 	g_settings_bind (a11y_settings, "stickykeys-modifier-beep", w, "active", G_SETTINGS_BIND_DEFAULT);
