--- trunk/qterm-qt4/CMakeLists.txt	2011/10/15 21:11:02	1295
+++ trunk/qterm-qt4/CMakeLists.txt	2011/12/17 07:10:08	1311
@@ -1,13 +1,12 @@
 project(qterm)
 include(CTest)
 cmake_minimum_required(VERSION 2.6.2 FATAL_ERROR)
-set(QT_MIN_VERSION "4.5.0")
 set(QTERM_VERSION_MAJOR 0)
 set(QTERM_VERSION_MINOR 5)
 set(QTERM_VERSION_RELEASE 12)
 set(QTERM_VERSION "${QTERM_VERSION_MAJOR}.${QTERM_VERSION_MINOR}.${QTERM_VERSION_RELEASE}" )
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-find_package(Qt4 REQUIRED)
+find_package(Qt4 4.5.0 REQUIRED)
 find_package(KDE4)
 
 IF(WIN32 AND MSVC)
