From 0e50780282bbe7cb56ca432657c965f064ee9cf2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 1 Feb 2012 10:40:07 +0100
Subject: [PATCH] Fix 64-bit warnings

---
 FindRef.xs |   15 ++++++++++-----
 1 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/FindRef.xs b/FindRef.xs
index 679958b..94920f6 100644
--- a/FindRef.xs
+++ b/FindRef.xs
@@ -151,7 +151,8 @@ find_ (SV *target_ref)
 				    {
 				      STRLEN length;
 				      char *name = HePV(he, length);
-				      res_pair (form ("the member '%.*s' of", length, name));
+				      int ilength = length;
+				      res_pair (form ("the member '%.*s' of", ilength, name));
 				    }
                               }
 
@@ -239,13 +240,17 @@ find_ (SV *target_ref)
                                     MAGIC *mg = mg_find (sv, PERL_MAGIC_defelem);
 
                                     if (mg && mg->mg_obj)
-                                      res_pair (form ("the target for the lvalue hash element '%.*s',",
-                                                      SvCUR (mg->mg_obj), SvPV_nolen (mg->mg_obj)));
+                                      {
+                                        STRLEN length = SvCUR (mg->mg_obj);
+                                        int ilength = length;
+                                        res_pair (form ("the target for the lvalue hash element '%.*s',",
+                                                        ilength, SvPV_nolen (mg->mg_obj)));
+                                      }
                                     else
-                                      res_pair (form ("the target for the lvalue array element #%d,", LvTARGOFF (sv)));
+                                      res_pair (form ("the target for the lvalue array element #%zu,", LvTARGOFF (sv)));
                                   }
                                 else
-                                  res_pair (form ("an lvalue reference target (type '%c', ofs %d, len %d),",
+                                  res_pair (form ("an lvalue reference target (type '%c', ofs %zu, len %zu),",
                                                   LvTYPE (sv), LvTARGOFF (sv), LvTARGLEN (sv)));
                               }
 
-- 
1.7.7.6

