From 02c5c3f2181381a44e03cdfb8962bcbf67a8139a Mon Sep 17 00:00:00 2001
From: Jarod Wilson <jarod@redhat.com>
Date: Tue, 7 Jun 2011 00:14:20 -0400
Subject: [PATCH 11/17] libusb has no libusb-config any longer, use pkg-config
 by default

Signed-off-by: Jarod Wilson <jarod@redhat.com>
---
 configure.ac | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6c07e0d..1a3347e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,11 +20,11 @@ AC_PROG_MAKE_SET
 AC_PATH_PROG(mknod, mknod, /bin/mknod)
 AC_PATH_PROG(mkfifo, mkfifo, /usr/bin/mkfifo)
 AC_PATH_PROG(depmod, depmod, /sbin/depmod, $PATH:/sbin)
-AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
 AM_PATH_PYTHON(,, [:])
 AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
+LIBUSB_CONFIG="pkg-config libusb"
 
 dnl Checks for header files.
 AC_HEADER_STDC
@@ -310,15 +310,14 @@ AH_TEMPLATE([SYSCONFDIR],
 AH_TEMPLATE([USE_SYSLOG],
 	[define if you want to log to syslog instead of logfile])
 
-if test -n "${LIBUSB_CONFIG}"; then
-  AC_DEFINE(HAVE_LIBUSB)
-  possible_drivers="${possible_drivers} (atilibusb) (awlibusb) (dfclibusb) (commandir) (srm7500libusb)" 
-fi
-
 AC_CHECK_LIB(caraca_client, caraca_init,
   AC_DEFINE(HAVE_LIBCARACA)
   possible_drivers="${possible_drivers} (caraca)"
 )
+AC_CHECK_HEADER(usb.h,
+  AC_DEFINE(HAVE_LIBUSB)
+  possible_drivers="${possible_drivers} (atilibusb) (awlibusb) (dfclibusb) (commandir) (srm7500libusb)" 
+)
 AC_CHECK_HEADER(ftdi.h,
   AC_DEFINE(HAVE_FTDI)
   possible_drivers="${possible_drivers} (ftdi)"
@@ -1449,7 +1448,7 @@ Get it at:
 Or install these packages from your distro:
 
    libusb
-   libusb-dev
+   libusb-devel
 
 ])
     ;;
-- 
1.8.3.1

