--- xfce-utils-4.7.1.orig/scripts/xflock4	2010-11-07 10:52:14.000000000 +0100
+++ xfce-utils-4.7.1/scripts/xflock4	2010-11-13 11:19:27.326293458 +0100
@@ -19,9 +19,9 @@
 #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 #
 
-if test x"`which xscreensaver-command 2>/dev/null`" != x""; then
+if ps ux | grep x[s]creensaver > /dev/null 2>&1; then
     xscreensaver-command -lock 
-elif test x"`which gnome-screensaver-command 2>/dev/null`" != x""; then
+elif ps ux | grep gnome-[s]creensaver > /dev/null 2>&1; then
     gnome-screensaver-command --lock
 elif test x"`which slock 2>/dev/null`" != x""; then
    slock
