diff -Naur procps-ng-3.3.9.orig/vmstat.c procps-ng-3.3.9/vmstat.c
--- procps-ng-3.3.9.orig/vmstat.c	2014-02-05 12:55:02.000000000 +0100
+++ procps-ng-3.3.9/vmstat.c	2014-02-05 16:58:35.786000000 +0100
@@ -204,10 +204,10 @@
 	    "%2s %2s %12s %12s %12s %12s %4s %4s %5s %5s %4s %4s %3s %3s %3s %3s %3s";
 
 
-	printf(w_option ? wide_header : header);
+	printf("%s", w_option ? wide_header : header);
 
 	if (t_option) {
-		printf(timestamp_header);
+		printf("%s", timestamp_header);
 	}
 
 	printf("\n");
@@ -528,10 +528,10 @@
 	const char wide_format[] =
 	    "%5s %9s %9s %11s %11s %9s %9s %11s %11s %7s %7s";
 
-	printf(w_option ? wide_header : header);
+	printf("%s", w_option ? wide_header : header);
 
 	if (t_option) {
-		printf(timestamp_header);
+		printf("%s", timestamp_header);
 	}
 
 	printf("\n");
