diff -u -r BioPerl-1.6.1/Bio/Variation/IO.pm BioPerl-1.6.1.mod/Bio/Variation/IO.pm
--- BioPerl-1.6.1/Bio/Variation/IO.pm	2007-02-14 04:37:48.000000000 -0700
+++ BioPerl-1.6.1.mod/Bio/Variation/IO.pm	2008-09-25 02:03:26.000000000 -0700
@@ -86,7 +86,7 @@
 
 This makes the simplest ever reformatter
 
-    #!/usr/local/bin/perl
+    #!/usr/bin/perl
 
     $format1 = shift;
     $format2 = shift;
Only in BioPerl-1.6.1.mod/Bio/Variation: IO.pm.orig
diff -u -r BioPerl-1.6.1/Build.PL BioPerl-1.6.1.mod/Build.PL
--- BioPerl-1.6.1/Build.PL	2009-09-29 12:33:45.000000000 -0400
+++ BioPerl-1.6.1.mod/Build.PL	2009-11-11 22:23:17.000000000 -0500
@@ -31,13 +31,13 @@
                             'DB_File'                   => 0,
                             'Data::Stag'                => 0.11, # Bio::SeqIO::swiss, we can change to 'recommend' if needed
                             'Scalar::Util'              => 0,    # not in Perl 5.6.1, arrived in core in 5.7.3
-                            'ExtUtils::Manifest'        => '1.52', # allows spaces in file names
+#                            'ExtUtils::Manifest'        => '1.52', # allows spaces in file names
                            },
     build_requires      => {
                             'Test::More'                => 0,
                             'Module::Build'             => 0.2805,
-                            'Test::Harness'             => 2.62,
-                            'CPAN'                      => 1.81
+#                            'Test::Harness'             => 2.62,
+#                            'CPAN'                      => 1.81
                            },
     recommends          => { # does what you would expect of recommends, except more informative output and generates optional_features in META.yml
                             'Ace'                       => '0/access of ACeDB database/Bio::DB::Ace,Bio::DB::GFF::Adaptor::ace',
diff -u -r BioPerl-1.6.1/examples/align/align_on_codons.pl BioPerl-1.6.1.mod/examples/align/align_on_codons.pl
--- BioPerl-1.6.1/examples/align/align_on_codons.pl	2007-02-14 04:37:47.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/align/align_on_codons.pl	2008-09-25 02:03:26.000000000 -0700
@@ -13,10 +13,10 @@
 BEGIN {
     $CODONSIZE = 3; # parametrize everything like a good little programmer
     if( ! defined $ENV{'CLUSTALDIR'} ) { 
-	$ENV{'CLUSTALDIR'} = '/usr/local/bin';
+	$ENV{'CLUSTALDIR'} = '/usr/bin';
     } 
     if( ! defined $ENV{'TCOFFEEDIR'} ) { 
-	$ENV{'TCOFFEEDIR'} = '/usr/local/bin';
+	$ENV{'TCOFFEEDIR'} = '/usr/bin';
     }
     $USAGE = 
 qq{align_on_codons.pl < file.fa
diff -u -r BioPerl-1.6.1/examples/db/dbfetch BioPerl-1.6.1.mod/examples/db/dbfetch
--- BioPerl-1.6.1/examples/db/dbfetch	2007-02-14 04:37:48.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/db/dbfetch	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -- # -*-Perl-*-
+#!/usr/bin/perl -- # -*-Perl-*-
 
 =head1 NAME
 
diff -u -r BioPerl-1.6.1/examples/db/getGenBank.pl BioPerl-1.6.1.mod/examples/db/getGenBank.pl
--- BioPerl-1.6.1/examples/db/getGenBank.pl	2007-02-14 04:37:47.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/db/getGenBank.pl	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 #
 # How to retrieve GenBank entries over the Web
 #
diff -u -r BioPerl-1.6.1/examples/db/get_seqs.pl BioPerl-1.6.1.mod/examples/db/get_seqs.pl
--- BioPerl-1.6.1/examples/db/get_seqs.pl	2007-02-14 04:37:48.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/db/get_seqs.pl	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 use strict;
 use vars qw($USAGE);
 use Carp;
diff -u -r BioPerl-1.6.1/examples/generate_random_seq.pl BioPerl-1.6.1.mod/examples/generate_random_seq.pl
--- BioPerl-1.6.1/examples/generate_random_seq.pl	2007-02-14 04:37:47.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/generate_random_seq.pl	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/bin/perl -w
+#!/usr/bin/perl -w
 use strict;
 use vars qw($USAGE);
 
diff -u -r BioPerl-1.6.1/examples/searchio/psiblast_features.pl BioPerl-1.6.1.mod/examples/searchio/psiblast_features.pl
--- BioPerl-1.6.1/examples/searchio/psiblast_features.pl	2007-02-14 04:37:48.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/searchio/psiblast_features.pl	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl 
+#!/usr/bin/perl 
 
 # Example usage of a SearchIO::psiblast parser of traditional format Blast 
 # and PSI-Blast reports.
diff -u -r BioPerl-1.6.1/examples/structure/structure-io.pl BioPerl-1.6.1.mod/examples/structure/structure-io.pl
--- BioPerl-1.6.1/examples/structure/structure-io.pl	2007-02-14 04:37:48.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/structure/structure-io.pl	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/bin/perl -w
+#!/usr/bin/perl -w
 # Getting  Entry, Chain, Residue, and Atom objects given a PDB file
 
 use Bio::Structure::IO;
diff -u -r BioPerl-1.6.1/examples/tk/hitdisplay.pl BioPerl-1.6.1.mod/examples/tk/hitdisplay.pl
--- BioPerl-1.6.1/examples/tk/hitdisplay.pl	2007-02-14 04:37:47.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/tk/hitdisplay.pl	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 #
 # PROGRAM  : hitdisplay.pl
 # PURPOSE  : Demonstrate Bio::Tk::HitDisplay
diff -u -r BioPerl-1.6.1/examples/tools/gb_to_gff.pl BioPerl-1.6.1.mod/examples/tools/gb_to_gff.pl
--- BioPerl-1.6.1/examples/tools/gb_to_gff.pl	2007-02-14 04:37:47.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/tools/gb_to_gff.pl	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 use strict;
 
 use Bio::Tools::GFF;
diff -u -r BioPerl-1.6.1/examples/tools/gff2ps.pl BioPerl-1.6.1.mod/examples/tools/gff2ps.pl
--- BioPerl-1.6.1/examples/tools/gff2ps.pl	2007-02-14 04:37:47.000000000 -0700
+++ BioPerl-1.6.1.mod/examples/tools/gff2ps.pl	2008-09-25 02:03:26.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 
 =head1 NAME
--- BioPerl-1.6.1/MANIFEST.orig	2008-12-27 20:29:43.000000000 -0700
+++ BioPerl-1.6.1/MANIFEST	2008-12-27 20:31:23.000000000 -0700
@@ -424,15 +424,6 @@
 Bio/Phenotype/OMIM/OMIMparser.pm
 Bio/Phenotype/Phenotype.pm
 Bio/Phenotype/PhenotypeI.pm
-Bio/PhyloNetwork.pm
-Bio/PhyloNetwork/Factory.pm
-Bio/PhyloNetwork/FactoryX.pm
-Bio/PhyloNetwork/GraphViz.pm
-Bio/PhyloNetwork/muVector.pm
-Bio/PhyloNetwork/RandomFactory.pm
-Bio/PhyloNetwork/TreeFactory.pm
-Bio/PhyloNetwork/TreeFactoryMulti.pm
-Bio/PhyloNetwork/TreeFactoryX.pm
 Bio/PopGen/Genotype.pm
 Bio/PopGen/GenotypeI.pm
 Bio/PopGen/HtSNP.pm
@@ -1994,12 +1985,6 @@
 t/Tools/tRNAscanSE.t
 t/Tree/Compatible.t
 t/Tree/Node.t
-t/Tree/PhyloNetwork/Factory.t
-t/Tree/PhyloNetwork/GraphViz.t
-t/Tree/PhyloNetwork/MuVector.t
-t/Tree/PhyloNetwork/PhyloNetwork.t
-t/Tree/PhyloNetwork/RandomFactory.t
-t/Tree/PhyloNetwork/TreeFactory.t
 t/Tree/RandomTreeFactory.t
 t/Tree/Tree.t
 t/Tree/TreeIO.t
