diff -up gnupg-1.4.13/cipher/idea.c.endian gnupg-1.4.13/cipher/idea.c
--- gnupg-1.4.13/cipher/idea.c.endian	2013-01-07 08:37:48.899033247 -0500
+++ gnupg-1.4.13/cipher/idea.c	2013-01-07 08:39:03.239030102 -0500
@@ -201,7 +201,7 @@ cipher( byte *outbuf, const byte *inbuf,
     x2 = *in++;
     x3 = *in++;
     x4 = *in;
-  #ifndef WORDS_BIGENDIAN
+  #ifndef BIG_ENDIAN_HOST
     x1 = (x1>>8) | (x1<<8);
     x2 = (x2>>8) | (x2<<8);
     x3 = (x3>>8) | (x3<<8);
@@ -234,7 +234,7 @@ cipher( byte *outbuf, const byte *inbuf,
     MUL(x4, *key);
 
     out = (u16*)outbuf;
-  #ifndef WORDS_BIGENDIAN
+  #ifndef BIG_ENDIAN_HOST
     *out++ = (x1>>8) | (x1<<8);
     *out++ = (x3>>8) | (x3<<8);
     *out++ = (x2>>8) | (x2<<8);
