diff -Nur gnupg-1.4.13-orig/mpi/longlong.h gnupg-1.4.13/mpi/longlong.h
--- gnupg-1.4.13-orig/mpi/longlong.h 2013-05-17 22:51:40.000000000 -0400
+++ gnupg-1.4.13/mpi/longlong.h 2013-05-17 22:52:07.000000000 -0400
@@ -752,7 +752,7 @@
     (w1) = _r >> 64;                                                    \
     (w0) = (UDItype) _r;                                                \
   } while (0)
-# elif if __GNUC__ > 2 || __GNUC_MINOR__ >= 7
+# elif __GNUC__ > 2 || __GNUC_MINOR__ >= 7
 #  define umul_ppmm(w1, w0, u, v)                                       \
      __asm__ ("dmultu %2,%3"                                            \
               : "=l" ((UDItype)(w0)),                                   \
