--- kvirc/src/kvirc/build/Makefile.am.ORI	2011-11-26 22:37:04.065565368 +0100
+++ kvirc/src/kvirc/build/Makefile.am	2011-11-26 22:37:17.139565032 +0100
@@ -8,7 +8,7 @@
 AM_CPPFLAGS = -I$(SS_TOPSRCDIR)/src/kvilib/include/ -I$(SS_TOPSRCDIR)/src/kvirc/include/ \
 $(SS_INCDIRS) $(SS_CPPFLAGS) -DGLOBAL_KVIRC_DIR=\"$(globalkvircdir)\"
 
-kvirc_LDFLAGS = -export-dynamic $(SS_LDFLAGS) $(SS_LIBDIRS) $(SS_RPATH)
+kvirc_LDFLAGS = -export-dynamic $(SS_LDFLAGS) $(SS_LIBDIRS) $(SS_RPATH) -lDCOP -lkdefx -lkio -lkdecore -lkdeui
 kvirc_LDADD = $(SS_LIBLINK) ../../kvilib/build/libkvilib.la
 
 bin_PROGRAMS = kvirc
