diff -up ./src/cpp/CMakeLists.txt.fixsoname ./src/cpp/CMakeLists.txt
--- ./src/cpp/CMakeLists.txt.fixsoname	2011-05-13 20:03:10.766371331 -0400
+++ ./src/cpp/CMakeLists.txt	2011-05-13 20:03:54.134635748 -0400
@@ -36,6 +36,8 @@ set_target_properties(flann_cpp PROPERTI
 add_library(flann_cpp-gd SHARED ${CPP_SOURCES})
 set_target_properties(flann_cpp-gd PROPERTIES 
     COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG}
+    VERSION ${FLANN_VERSION}
+    SOVERSION ${FLANN_VERSION}
     DEFINE_SYMBOL FLANN_EXPORTS
 )
 
