diff -up openmpi-1.8.3/oshmem/tools/wrappers/shmemcc-wrapper-data.txt.in.oshmem openmpi-1.8.3/oshmem/tools/wrappers/shmemcc-wrapper-data.txt.in
--- openmpi-1.8.3/oshmem/tools/wrappers/shmemcc-wrapper-data.txt.in.oshmem	2014-07-23 15:43:33.000000000 -0600
+++ openmpi-1.8.3/oshmem/tools/wrappers/shmemcc-wrapper-data.txt.in	2014-10-03 15:26:43.728048287 -0600
@@ -31,7 +31,7 @@ linker_flags=@OMPI_WRAPPER_EXTRA_LDFLAGS
 # applications to only use the SHMEM and MPI APIs.
 libs=-loshmem -lmpi
 libs_static=-loshmem -lmpi -lopen-rte -lopen-pal @OMPI_WRAPPER_EXTRA_LIBS@
-dyn_lib_file=libsohmem.@OPAL_DYN_LIB_SUFFIX@
+dyn_lib_file=liboshmem.@OPAL_DYN_LIB_SUFFIX@
 static_lib_file=liboshmem.a
 required_file=
 includedir=${includedir}
