diff -ur mpeg4ip-1.6.1.orig/configure.in mpeg4ip-1.6.1/configure.in
--- mpeg4ip-1.6.1.orig/configure.in	2012-01-17 17:42:39.000000000 -0500
+++ mpeg4ip-1.6.1/configure.in	2012-01-17 17:42:55.000000000 -0500
@@ -487,7 +487,7 @@
   if test x$have_ffmpeg_arg = xtrue; then
      AC_MSG_NOTICE([have enable ffmpeg $enable_ffmpeg])
      FFMPEG_INC=-I${enable_ffmpeg}/libavcodec
-     FFMPEG_LIB="${enable_ffmpeg}/libavcodec/libavcodec.a -lz"
+     FFMPEG_LIB="-lavcodec -lz"
      saveCFLAGS="$CFLAGS"
      CFLAGS="$CFLAGS $FFMPEG_INC"
      AC_CHECK_TYPES([AVRational], , , [#include <avcodec.h>])
@@ -512,7 +512,7 @@
      if test x$have_ffmpeg = xtrue; then
         AC_DEFINE(HAVE_FFMPEG_INSTALLED, 1, [defined if ffmpeg is installed])
         FFMPEG_INC=
-        FFMPEG_LIB="-lavcodec -lz $ffmpeg_add_libs"
+        FFMPEG_LIB="-lavcodec -lz $ffmpeg_add_libs $(pkg-config libswscale --libs --silence-errors)"
         AC_CHECK_TYPES([AVRational], , , [#include <ffmpeg/avcodec.h>])
 	AC_CHECK_MEMBERS(AVCodecContext.time_base, , , [#include <ffmpeg/avcodec.h>])
 	AC_CHECK_DECLS([avcodec_decode_audio2], , ,[#include <ffmpeg/avcodec.h>])
diff -ur mpeg4ip-1.6.1.orig/server/mp4live/audio_ffmpeg.h mpeg4ip-1.6.1/server/mp4live/audio_ffmpeg.h
--- mpeg4ip-1.6.1.orig/server/mp4live/audio_ffmpeg.h	2012-01-17 17:42:40.000000000 -0500
+++ mpeg4ip-1.6.1/server/mp4live/audio_ffmpeg.h	2012-01-17 17:42:55.000000000 -0500
@@ -25,7 +25,7 @@
 #include "audio_encoder.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -ur mpeg4ip-1.6.1.orig/server/mp4live/video_ffmpeg.h mpeg4ip-1.6.1/server/mp4live/video_ffmpeg.h
--- mpeg4ip-1.6.1.orig/server/mp4live/video_ffmpeg.h	2012-01-17 17:42:40.000000000 -0500
+++ mpeg4ip-1.6.1/server/mp4live/video_ffmpeg.h	2012-01-17 17:42:55.000000000 -0500
@@ -25,7 +25,7 @@
 #include "video_encoder.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
