--- qt-x11-free-3.3.5/src/kernel/qapplication_x11.cpp.orig	2005-11-29 03:19:49.000000000 +0800
+++ qt-x11-free-3.3.5/src/kernel/qapplication_x11.cpp	2005-11-29 03:24:40.000000000 +0800
@@ -2452,8 +2452,8 @@
     XSetErrorHandler( original_x_errhandler );
     XSetIOErrorHandler( original_xio_errhandler );
 
-    if ( qt_is_gui_used && !appForeignDpy )
-	XCloseDisplay( appDpy );		// close X display
+//    if ( qt_is_gui_used && !appForeignDpy )
+//	XCloseDisplay( appDpy );		// close X display
     appDpy = 0;
 
     qt_x11_wm_client_leader = 0;
