--- qt3/src/kernel/qscriptengine_x11.cpp.fontrendering-gu-228452	2012-09-29 11:30:23.821539127 +0200
+++ qt3/src/kernel/qscriptengine_x11.cpp	2012-09-29 11:31:59.679413873 +0200
@@ -261,7 +261,7 @@
     Other, Other, Other, Other,
     Other, Other, Other, Other,
 
-    Other, Other, Other, Other,
+    Other, Other, VowelMark, VowelMark,
     Other, Other, Other, Other,
     Other, Other, Other, Other,
     Other, Other, Other, Other,
