--- qt-x11-free-3.3.6/src/kernel/qscriptengine_x11.cpp.orig	2006-10-20 19:26:38.000000000 +0200
+++ qt-x11-free-3.3.6/src/kernel/qscriptengine_x11.cpp	2006-10-20 19:28:14.000000000 +0200
@@ -159,7 +159,7 @@
     Matra, Matra, Matra, Matra,
     Matra, Invalid, Invalid, Matra,
     Matra, Invalid, Invalid, Matra,
-    Matra, Halant, Unknown, Unknown,
+    Matra, Halant, Consonant, Unknown,				//here Consonant was Unknown before modification
 
     Invalid, Invalid, Invalid, Invalid,
     Invalid, Invalid, Invalid, VowelMark,
