diff -u -r qt-x11-free-3.3.8.orig/src/kernel/qapplication_x11.cpp qt-x11-free-3.3.8/src/kernel/qapplication_x11.cpp
--- qt-x11-free-3.3.8.orig/src/kernel/qapplication_x11.cpp	2007-06-03 21:52:19.000000000 +0800
+++ qt-x11-free-3.3.8/src/kernel/qapplication_x11.cpp	2007-06-03 21:53:13.000000000 +0800
@@ -5046,7 +5046,7 @@
 
     if ( type == QEvent::KeyPress ) {
 	bool mb=FALSE;
-	if ( qt_xim ) {
+	if ( qt_xim && !tlw->isDesktop() && !tlw->isPopup() ) {
 	    QTLWExtra*  xd = tlw->topData();
 	    QInputContext *qic = (QInputContext *) xd->xic;
 	    if ( qic ) {
diff -u -r qt-x11-free-3.3.8.orig/src/kernel/qinputcontext_x11.cpp qt-x11-free-3.3.8/src/kernel/qinputcontext_x11.cpp
--- qt-x11-free-3.3.8.orig/src/kernel/qinputcontext_x11.cpp	2007-06-03 21:52:19.000000000 +0800
+++ qt-x11-free-3.3.8/src/kernel/qinputcontext_x11.cpp	2007-06-03 21:53:13.000000000 +0800
@@ -145,7 +145,7 @@
 	}
 
 	bool send_imstart = FALSE;
-	if (qApp->focusWidget() != qic->focusWidget && qic->text.isEmpty()) {
+	if (qApp->focusWidget() != qic->focusWidget) {
 	    if (qic->focusWidget) {
 #ifdef QT_XIM_DEBUG
 		qDebug( "sending IMEnd (empty) to %p", qic->focusWidget );
@@ -324,7 +324,7 @@
 	return;
     }
 
-    if (! widget->isTopLevel()) {
+    if (! widget->isTopLevel() || widget->isDesktop() || widget->isPopup()) {
 	qWarning("QInputContext: cannot create input context for non-toplevel widgets");
 	return;
     }
diff -u -r qt-x11-free-3.3.8.orig/src/kernel/qwidget_x11.cpp qt-x11-free-3.3.8/src/kernel/qwidget_x11.cpp
--- qt-x11-free-3.3.8.orig/src/kernel/qwidget_x11.cpp	2007-06-03 21:52:19.000000000 +0800
+++ qt-x11-free-3.3.8/src/kernel/qwidget_x11.cpp	2007-06-03 21:54:45.000000000 +0800
@@ -962,6 +962,10 @@
 	QWidget* tlw = topLevelWidget();
 	QTLWExtra *topdata = tlw->topData();
 
+        //check tlw is desktop or popup or not
+        if (tlw->isDesktop() || tlw->isPopup())
+                return
+
 	// trigger input context creation if it hasn't happened already
 	createInputContext();
 	QInputContext *qic = (QInputContext *) topdata->xic;
@@ -1440,7 +1444,7 @@
 void QWidget::setActiveWindow()
 {
     QWidget *tlw = topLevelWidget();
-    if ( tlw->isVisible() && !tlw->topData()->embedded && !qt_deferred_map_contains(tlw) ) {
+    if ( tlw->isVisible() && !tlw->topData()->embedded && !qt_deferred_map_contains(tlw) && !tlw->isDesktop() && !tlw->isPopup()) {
 	XSetInputFocus( x11Display(), tlw->winId(), RevertToNone, qt_x_time);
         focusInputContext();
     }
@@ -2666,7 +2670,7 @@
     QTLWExtra *topdata = tlw->topData();
 
 #ifndef QT_NO_XIM
-    if (qt_xim) {
+    if (qt_xim && !tlw->isDesktop() && !tlw->isPopup()) {
 	if (! topdata->xic) {
 	    QInputContext *qic = new QInputContext(tlw);
 	    topdata->xic = (void *) qic;
@@ -2701,6 +2705,9 @@
 	QWidget *tlw = topLevelWidget();
 	QTLWExtra *topdata = tlw->topData();
 
+	//check tlw is desktop or popup or not
+	if (tlw->isDesktop() || tlw->isPopup())
+		return
 	// trigger input context creation if it hasn't happened already
 	createInputContext();
 
Only in qt-x11-free-3.3.8/src/kernel: qwidget_x11.cpp.orig
