diff -ur qt-x11-free-3.3.8b.orig/include/qvaluelist.h qt-x11-free-3.3.8b/include/qvaluelist.h
--- qt-x11-free-3.3.8b.orig/include/qvaluelist.h	2011-04-26 09:51:17.000000000 -0400
+++ qt-x11-free-3.3.8b/include/qvaluelist.h	2011-04-26 09:52:27.000000000 -0400
@@ -50,6 +50,7 @@
 #ifndef QT_NO_STL
 #include <iterator>
 #include <list>
+#include <cstddef>
 #endif
 
 //#define QT_CHECK_VALUELIST_RANGE
diff -ur qt-x11-free-3.3.8b.orig/include/qvaluevector.h qt-x11-free-3.3.8b/include/qvaluevector.h
--- qt-x11-free-3.3.8b.orig/include/qvaluevector.h	2011-04-26 09:51:17.000000000 -0400
+++ qt-x11-free-3.3.8b/include/qvaluevector.h	2011-05-16 10:04:31.000000000 -0400
@@ -49,6 +49,8 @@
 #include <vector>
 #endif
 
+#include <cstddef>
+
 template <class T>
 class QValueVectorPrivate : public QShared
 {
diff -ur qt-x11-free-3.3.8b.orig/src/tools/qvaluelist.h qt-x11-free-3.3.8b/src/tools/qvaluelist.h
--- qt-x11-free-3.3.8b.orig/src/tools/qvaluelist.h	2011-04-26 09:51:17.000000000 -0400
+++ qt-x11-free-3.3.8b/src/tools/qvaluelist.h	2011-04-26 09:52:27.000000000 -0400
@@ -50,6 +50,7 @@
 #ifndef QT_NO_STL
 #include <iterator>
 #include <list>
+#include <cstddef>
 #endif
 
 //#define QT_CHECK_VALUELIST_RANGE
diff -ur qt-x11-free-3.3.8b.orig/src/tools/qvaluevector.h qt-x11-free-3.3.8b/src/tools/qvaluevector.h
--- qt-x11-free-3.3.8b.orig/src/tools/qvaluevector.h	2011-04-26 09:51:17.000000000 -0400
+++ qt-x11-free-3.3.8b/src/tools/qvaluevector.h	2011-05-16 10:04:31.000000000 -0400
@@ -49,6 +49,8 @@
 #include <vector>
 #endif
 
+#include <cstddef>
+
 template <class T>
 class QValueVectorPrivate : public QShared
 {
