--- usbmuxd-1.0.9/udev/39-usbmuxd.rules.in.orig	2014-04-21 18:57:45.137269481 +0100
+++ usbmuxd-1.0.9/udev/39-usbmuxd.rules.in	2014-04-21 18:58:47.615294201 +0100
@@ -1,7 +1,7 @@
 # usbmuxd ("Apple Mobile Device" muxer listening on /var/run/usbmuxd)
 
 # Initialize iOS devices into "deactivated" USB configuration state and run usbmuxd
-ACTION=="add", SUBSYSTEM=="usb", ATTR{idVendor}=="05ac", ATTR{idProduct}=="12[9a][0-9a-f]", ENV{USBMUX_SUPPORTED}="1", ATTR{bConfigurationValue}="0", OWNER="usbmux", RUN+="@prefix@/sbin/usbmuxd -u -U usbmux"
+ACTION=="add", SUBSYSTEM=="usb", ATTR{idVendor}=="05ac", ATTR{idProduct}=="12[9a][0-9a-f]", ENV{USBMUX_SUPPORTED}="1", ATTR{bConfigurationValue}="0", OWNER="usbmux", TAG+="systemd", ENV{SYSTEMD_WANTS}="usbmuxd.service"
 
 # Exit usbmuxd when the last device is removed
 ACTION=="remove", SUBSYSTEM=="usb", ENV{PRODUCT}=="5ac/12[9a][0-9a-f]/*", ENV{INTERFACE}=="255/*", RUN+="@prefix@/sbin/usbmuxd -x"
--- /dev/null
+++ b/udev/usbmuxd.service.in
@@ -0,0 +1,7 @@
+[Unit]
+Description=iOS USB Muxer
+
+[Service]
+Type=simple
+ExecStart=/usr/sbin/usbmuxd -u -U usbmux
+PIDFile=/var/run/usbmuxd.pid
