--- usbmuxd-1.0.9/src/main.c.orig	2014-05-05 17:33:42.447319977 +0100
+++ usbmuxd-1.0.9/src/main.c	2014-05-05 17:34:55.110508314 +0100
@@ -20,7 +20,12 @@
 
 */
 
-#define _BSD_SOURCE
+#if defined __linux__ || defined __GNU__ || defined __GLIBC__
+#define _DEFAULT_SOURCE 1
+#else
+#define _BSD_SOURCE 1
+#endif
+
 #define _GNU_SOURCE
 
 #ifdef HAVE_CONFIG_H
--- usbmuxd-1.0.9/src/device.c.orig	2014-05-05 17:33:53.094494058 +0100
+++ usbmuxd-1.0.9/src/device.c	2014-05-05 17:34:37.537220856 +0100
@@ -19,7 +19,11 @@
 
 */
 
-#define _BSD_SOURCE
+#if defined __linux__ || defined __GNU__ || defined __GLIBC__
+#define _DEFAULT_SOURCE 1
+#else
+#define _BSD_SOURCE 1
+#endif
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
