diff -ur kdevelop-3.5.4.orig/lib/antlr/antlr/CharScanner.hpp kdevelop-3.5.4/lib/antlr/antlr/CharScanner.hpp
--- kdevelop-3.5.4.orig/lib/antlr/antlr/CharScanner.hpp	2009-05-19 08:02:32.000000000 +0800
+++ kdevelop-3.5.4/lib/antlr/antlr/CharScanner.hpp	2009-05-19 08:04:25.000000000 +0800
@@ -18,6 +18,7 @@
 #include <cctype>
 #endif
 #include <cstring>
+#include <cstdio>
 
 #if ( _MSC_VER == 1200 )
 // VC6 seems to need this
