--- kdevelop-3.2.1/lib/Makefile.am.orig	2005-05-23 20:09:56.000000000 +0800
+++ kdevelop-3.2.1/lib/Makefile.am	2005-06-03 09:51:59.000000000 +0800
@@ -16,9 +16,11 @@
 lib_LTLIBRARIES = libkdevelop.la
 libkdevelop_la_LDFLAGS = -no-undefined $(all_libraries) -version-info 1:0
 libkdevelop_la_LIBADD = $(top_builddir)/lib/interfaces/libkdevinterfaces.la \
-	$(top_builddir)/lib/util/libkdevutil.la $(top_builddir)/lib/widgets/propeditor/libkdevpropertyeditor.la \
-	$(top_builddir)/lib/interfaces/external/libkinterfacedesigner.la $(top_builddir)/lib/interfaces/extensions/libkdevextensions.la \
-	$(LIB_KDEVCOMPAT) -lktexteditor $(LIB_KPARTS) $(LIB_KIO)  -lkscript
+        $(top_builddir)/lib/widgets/propeditor/libkdevpropertyeditor.la $(LIB_KDEVCOMPAT) $(LIB_KIO) 
+ 
+#       $(top_builddir)/lib/util/libkdevutil.la $(top_builddir)/lib/widgets/propeditor/libkdevpropertyeditor.la \ 
+#       $(top_builddir)/lib/interfaces/external/libkinterfacedesigner.la $(top_builddir)/lib/interfaces/extensions/libkdevextensions.la \ 
+#       $(LIB_KDEVCOMPAT) -lktexteditor $(LIB_KPARTS) $(LIB_KIO)  -lkscript 
 libkdevelop_la_SOURCES = dummy.cpp
 
 
--- kdevelop-3.2.1/lib/interfaces/Makefile.am.orig	2005-05-23 20:09:53.000000000 +0800
+++ kdevelop-3.2.1/lib/interfaces/Makefile.am	2005-06-03 09:49:03.000000000 +0800
@@ -12,6 +12,10 @@
 	kdevprojectiface.cpp kdevprojectiface.skel kdevdesignerintegrationiface.cpp \
 	kdevdesignerintegrationiface.skel
 
+ libkdevinterfaces_la_LDFLAGS = -no-undefined $(all_libraries) $(LIB_KDEUI) $(LIB_KHTML) $(LIB_KPARTS) -lktexteditor -lkscript \ 
+         $(top_builddir)/lib/interfaces/external/libkinterfacedesigner.la \ 
+         $(top_builddir)/lib/util/libkdevutil.la 
+
 kdevelopincludedir = $(includedir)/kdevelop/interfaces
 kdevelopinclude_HEADERS = kdevlanguagesupport.h kdevmainwindow.h \
 	kdevpartcontroller.h kdevplugin.h kdevproject.h kdevcore.h kdevcoderepository.h codemodel.h \
--- kdevelop-3.2.1/lib/widgets/Makefile.am.orig	2005-05-23 20:09:54.000000000 +0800
+++ kdevelop-3.2.1/lib/widgets/Makefile.am	2005-06-03 09:50:02.000000000 +0800
@@ -14,6 +14,7 @@
 	processwidget.h
 SUBDIRS = propeditor
 libkdevwidgets_la_LIBADD = $(top_builddir)/lib/interfaces/libkdevinterfaces.la
+libkdevwidgets_la_LDFLAGS = -no-undefined $(all_libraries)
 rcdir = $(kde_datadir)/kdevelop
 
 rc_DATA =  kdevhtml_partui.rc
