--- mkrf-0.2.3/Rakefile.newrake	2013-03-07 14:18:40.000000000 +0900
+++ mkrf-0.2.3/Rakefile	2013-03-07 14:27:03.000000000 +0900
@@ -1,8 +1,8 @@
 require 'rake'
 require 'rake/testtask'
 require 'rake/packagetask'
-require 'rake/gempackagetask'
-require 'rake/rdoctask'
+require 'rubygems/package_task'
+require 'rdoc/task'
 require 'rubygems'
 
 $:.unshift(File.dirname(__FILE__) + "/lib")
@@ -121,7 +121,7 @@
   s.files = s.files + Dir.glob( "test/**/*" ).delete_if { |item| item.include?( "\.svn" ) }
 end
   
-Rake::GemPackageTask.new(spec) do |p|
+Gem::PackageTask.new(spec) do |p|
   p.gem_spec = spec
   p.need_tar = true
   p.need_zip = true
