diff -up dyninst/testsuite/make.module.tmpl.testsuite-opt dyninst/testsuite/make.module.tmpl
--- dyninst/testsuite/make.module.tmpl.testsuite-opt	2013-12-11 17:01:03.278606211 -0800
+++ dyninst/testsuite/make.module.tmpl	2013-12-11 17:02:24.593743942 -0800
@@ -23,7 +23,7 @@ LFLAGS = -L../../common/$(PLATFORM) -L..
 # Turn off NO_INITIALIZER_LIST_SUPPORT if we compile test suite with C++11
 CFLAGS = $(UNIFIED_TEST_DEF:%=%_test) $(ARCH_FLAGS) $(AC_DEF) $(FIRST_INCLUDE) $(IFLAGS) -I../src -g $(BOOST_INCDIR) $(CPPFLAGS)
 CXXFLAGS = $(CFLAGS)
-CXXFLAGS += $(PLATFORM_CXXFLAGS) $(CXX_REQ_FLAGS)
+CXXFLAGS += $(PLATFORM_CXXFLAGS) -std=c++11
 
 CFLAGS_NATIVE = $(CFLAGS)
 CXXFLAGS_NATIVE = $(CXXFLAGS)
