commit 6401cf3b040bee82a3d1146519aaaf3123b169bb
Author: Andrew Bernat <bernat@cs.wisc.edu>
Date:   Thu Mar 14 19:30:52 2013 -0500

    Symlink path fix as suggested by Josh Stone

diff --git a/make.library.tmpl b/make.library.tmpl
index fb2dce4..61f7651 100644
--- a/make.library.tmpl
+++ b/make.library.tmpl
@@ -153,10 +153,10 @@ $(DEST)/$(TARGET).$(RELEASE_NUM): $(TARGET).$(RELEASE_NUM) $(TARGET)
 ifndef STATIC_LIBRARY
 $(DEST)/$(TARGET).$(SONAME_NUM): $(DEST)/$(TARGET).$(RELEASE_NUM)
 	@echo Making symlink $(TARGET).$(RELEASE_NUM) to $(DEST)/$(TARGET).$(SONAME_NUM)
-	$(HIDE_COMP)-$(LN) -f $(DEST)/$(TARGET).$(RELEASE_NUM) $(DEST)/$(TARGET).$(SONAME_NUM)
+	$(HIDE_COMP)-$(LN) -f $(TARGET).$(RELEASE_NUM) $(DEST)/$(TARGET).$(SONAME_NUM)
 $(DEST)/$(TARGET): $(DEST)/$(TARGET).$(SONAME_NUM)
 	@echo Making symlink $(TARGET).$(SONAME_NUM) to $(DEST)/$(TARGET)
-	$(HIDE_COMP)-$(LN) -f $(DEST)/$(TARGET).$(SONAME_NUM) $(DEST)/$(TARGET)
+	$(HIDE_COMP)-$(LN) -f $(TARGET).$(SONAME_NUM) $(DEST)/$(TARGET)
 else
 $(DEST)/$(TARGET): $(TARGET)
 	@echo Installing $(TARGET) to $(DEST)/$(TARGET)
