diff -ur llk_linux-2.3.orig/src/Makefile.am llk_linux-2.3/src/Makefile.am
--- llk_linux-2.3.orig/src/Makefile.am	2012-12-31 06:35:26.000000000 +0000
+++ llk_linux-2.3/src/Makefile.am	2012-12-31 06:35:40.000000000 +0000
@@ -30,5 +30,5 @@
 
 llk_linux_LDADD =  \
 	$(PACKAGE_LIBS)\
-	 -lc -lesd
+	 -lc -lesd -lX11
 
diff -ur llk_linux-2.3.orig/src/Makefile.in llk_linux-2.3/src/Makefile.in
--- llk_linux-2.3.orig/src/Makefile.in	2012-12-31 06:35:26.000000000 +0000
+++ llk_linux-2.3/src/Makefile.in	2012-12-31 06:35:50.000000000 +0000
@@ -226,7 +226,7 @@
 
 llk_linux_LDADD = \
 	$(PACKAGE_LIBS)\
-	 -lc -lesd
+	 -lc -lesd -lX11
 
 all: all-am
 
