diff -ur linphone-3.5.2.orig/coreapi/sal_eXosip2_sdp.c linphone-3.5.2/coreapi/sal_eXosip2_sdp.c
--- linphone-3.5.2.orig/coreapi/sal_eXosip2_sdp.c	2013-02-18 18:40:56.383463031 +0000
+++ linphone-3.5.2/coreapi/sal_eXosip2_sdp.c	2013-02-18 18:41:29.122462650 +0000
@@ -23,7 +23,7 @@
 #include "sal.h"
 #include <eXosip2/eXosip.h>
 
-#define keywordcmp(key,b) strncmp(key,b,sizeof(key))
+#define keywordcmp(key,b) strcmp(key,b)
 
 #ifdef FOR_LATER
 
diff -ur linphone-3.5.2.orig/gtk/Makefile.am linphone-3.5.2/gtk/Makefile.am
--- linphone-3.5.2.orig/gtk/Makefile.am	2013-02-18 18:40:56.390463038 +0000
+++ linphone-3.5.2/gtk/Makefile.am	2013-02-18 18:43:23.920466042 +0000
@@ -75,7 +75,7 @@
 endif
 
 
-AM_CFLAGS= -DIN_LINPHONE -I$(top_srcdir)/coreapi/ \
+AM_CFLAGS= -DIN_LINPHONE -I$(top_srcdir)/coreapi/ -Wno-error=deprecated-declarations\
 		$(MEDIASTREAMER_CFLAGS) \
 		$(ORTP_CFLAGS) \
 		$(STRICT_OPTIONS) $(LIBGTK_CFLAGS) $(LIBGTKMAC_CFLAGS) $(IPV6_CFLAGS) \
diff -ur linphone-3.5.2.orig/gtk/Makefile.in linphone-3.5.2/gtk/Makefile.in
--- linphone-3.5.2.orig/gtk/Makefile.in	2013-02-18 18:40:56.391463039 +0000
+++ linphone-3.5.2/gtk/Makefile.in	2013-02-18 18:44:14.851467107 +0000
@@ -380,7 +380,7 @@
 @BUILD_GTK_UI_TRUE@@BUILD_WIN32_TRUE@linphone_LDFLAGS = -Wl,--export-all-symbols -mwindows
 @BUILD_GTK_UI_TRUE@uidir = $(datadir)/linphone
 @BUILD_GTK_UI_TRUE@ui_DATA = $(UI_FILES) $(PIXMAPS) $(top_srcdir)/COPYING
-AM_CFLAGS = -DIN_LINPHONE -I$(top_srcdir)/coreapi/ \
+AM_CFLAGS = -DIN_LINPHONE -I$(top_srcdir)/coreapi/ -Wno-error=deprecated-declarations \
 		$(MEDIASTREAMER_CFLAGS) \
 		$(ORTP_CFLAGS) \
 		$(STRICT_OPTIONS) $(LIBGTK_CFLAGS) $(LIBGTKMAC_CFLAGS) $(IPV6_CFLAGS) \
