diff -uNr kdebase-3.4.0.orig/configure.in.in kdebase-3.4.0/configure.in.in
--- kdebase-3.4.0.orig/configure.in.in	2005-02-23 12:24:24.000000000 +0100
+++ kdebase-3.4.0/configure.in.in	2005-03-03 12:29:05.000000000 +0100
@@ -138,9 +138,6 @@
    fi
 fi
 
-dnl check for java
-KDE_CHECK_JAVA_DIR
-
 dnl check for an md5sum implementation. the GNU versions
 dnl produce different output from the FreeBSD / Solaris 
 dnl versions, so MD5SUM_CUT shows how to retrieve the
diff -uNr kdebase-3.4.0.orig/kcontrol/konqhtml/configure.in.in kdebase-3.4.0/kcontrol/konqhtml/configure.in.in
--- kdebase-3.4.0.orig/kcontrol/konqhtml/configure.in.in	1970-01-01 01:00:00.000000000 +0100
+++ kdebase-3.4.0/kcontrol/konqhtml/configure.in.in	2005-03-03 12:30:42.000000000 +0100
@@ -0,0 +1,9 @@
+AC_ARG_WITH(java-dir, AC_HELP_STRING([--with-java-dir=javadir],[set the path where Java is installed]),
+	ac_java_dir=$withval,
+	ac_java_dir=no)
+
+if test "x$ac_java_dir" != xno; then
+	if test -x "$ac_java_dir/bin/java"; then
+		AC_DEFINE_UNQUOTED(PATH_JAVA, "$ac_java_dir/bin/java", [Define where your java executable is])
+	fi
+fi
