--- kdebase-3.5.2/kcontrol/kfontinst/kio/KioFonts.cpp.orig	2006-03-17 18:17:16.000000000 +0800
+++ kdebase-3.5.2/kcontrol/kfontinst/kio/KioFonts.cpp	2006-04-15 22:06:26.000000000 +0800
@@ -766,8 +766,8 @@
         itsFolders[FOLDER_USER].location=dir;
     }
 
-    TQString sysDefault("/usr/local/share/fonts/"),
-            sysDir(getFontFolder(sysDefault, "/usr/local/share/", dirs));
+    TQString sysDefault("/usr/share/fonts/"),
+            sysDir(getFontFolder(sysDefault, "/usr/share/", dirs));
 
     if(sysDir.isEmpty())
     {
