diff -ur tdebase-git20111213.orig/kcontrol/taskbar/kcmtaskbarui.ui tdebase-git20111213/kcontrol/taskbar/kcmtaskbarui.ui
--- tdebase-git20111213.orig/kcontrol/taskbar/kcmtaskbarui.ui	2011-12-14 07:09:53.000000000 -0500
+++ tdebase-git20111213/kcontrol/taskbar/kcmtaskbarui.ui	2011-12-14 12:27:35.000000000 -0500
@@ -8,8 +8,8 @@
         <rect>
             <x>0</x>
             <y>0</y>
-            <width>358</width>
-            <height>351</height>
+            <width>390</width>
+            <height>553</height>
         </rect>
     </property>
     <property name="caption">
@@ -39,53 +39,78 @@
                 <property name="name">
                     <cstring>unnamed</cstring>
                 </property>
-                <widget class="TQCheckBox" row="1" column="1" rowspan="1" colspan="2">
+                <widget class="TQCheckBox" row="0" column="0" rowspan="1" colspan="3">
                     <property name="name">
-                        <cstring>kcfg_SortByDesktop</cstring>
+                        <cstring>kcfg_ShowAllWindows</cstring>
                     </property>
                     <property name="text">
-                        <string>Sort windows by desk&amp;top</string>
+                        <string>&amp;Show windows from all desktops</string>
+                    </property>
+                    <property name="accel">
+                        <string>Alt+S</string>
                     </property>
                     <property name="checked">
                         <bool>true</bool>
                     </property>
                     <property name="whatsThis" stdset="0">
-                        <string>Selecting this option causes the taskbar to show windows in order of the desktop they appear on.
+                        <string>Turning this option off will cause the taskbar to display &lt;b&gt;only&lt;/b&gt; the windows on the current desktop. 
 
-By default this option is selected.</string>
+By default, this option is selected and all windows are shown.</string>
                     </property>
                 </widget>
-                <widget class="TQCheckBox" row="0" column="0" rowspan="1" colspan="3">
+                <widget class="TQCheckBox" row="3" column="0" rowspan="1" colspan="3">
                     <property name="name">
-                        <cstring>kcfg_ShowAllWindows</cstring>
+                        <cstring>showAllScreens</cstring>
                     </property>
                     <property name="text">
-                        <string>&amp;Show windows from all desktops</string>
+                        <string>Show windows from all sc&amp;reens</string>
+                    </property>
+                    <property name="accel">
+                        <string>Alt+R</string>
                     </property>
                     <property name="checked">
                         <bool>true</bool>
                     </property>
                     <property name="whatsThis" stdset="0">
-                        <string>Turning this option off will cause the taskbar to display &lt;b&gt;only&lt;/b&gt; the windows on the current desktop. 
+                        <string>Turning this option off will cause the taskbar to display &lt;b&gt;only&lt;/b&gt; windows which are on the same Xinerama screen as the taskbar.
 
 By default, this option is selected and all windows are shown.</string>
                     </property>
                 </widget>
-                <widget class="TQCheckBox" row="6" column="0" rowspan="1" colspan="3">
+		<widget class="TQCheckBox" row="10" column="0" rowspan="1" colspan="2">
                     <property name="name">
-                        <cstring>kcfg_ShowWindowListBtn</cstring>
+                    	<cstring>kcfg_UseCustomColors</cstring>
+		    </property>
+                    <property name="enabled">
+                        <bool>true</bool>
                     </property>
                     <property name="text">
-                        <string>Show window list &amp;button</string>
+                        <string>Use &amp;custom colors</string>
+                    </property>
+                    <property name="text">
+                        <string>Use &amp;custom colors</string>
                     </property>
                     <property name="checked">
                         <bool>true</bool>
                     </property>
-                    <property name="whatsThis" stdset="0">
-                        <string>Selecting this option causes the taskbar to display a button that, when clicked, shows a list of all windows in a popup menu.</string>
+                </widget>
+                <widget class="TQComboBox" row="9" column="2" rowspan="2" colspan="1">
+                    <property name="name">
+                        <cstring>appearance</cstring>
                     </property>
                 </widget>
-                <widget class="TQComboBox" row="7" column="2">
+                <widget class="TQLabel" row="8" column="0" rowspan="1" colspan="2">
+                    <property name="name">
+                        <cstring>groupTasksLabel</cstring>
+                    </property>
+                    <property name="text">
+                        <string>&amp;Group similar tasks:</string>
+                    </property>
+                    <property name="buddy" stdset="0">
+                        <cstring>kcfg_GroupTasks</cstring>
+                    </property>
+                </widget>
+                <widget class="TQComboBox" row="8" column="2">
                     <property name="name">
                         <cstring>kcfg_GroupTasks</cstring>
                     </property>
@@ -98,44 +123,36 @@
                         </sizepolicy>
                     </property>
                     <property name="whatsThis" stdset="0">
-                        <string>The taskbar can group similar windows into single buttons. When one of these window group buttons are clicked on a menu appears showing all the windows in that group. This can be especially useful with the &lt;em&gt;Show all windows&lt;/em&gt; option.
+                        <string>The taskbar can group similar windows into single buttons. When one of these window group buttons are clicked on a menu appears showing all the wi
+ndows in that group. This can be especially useful with the &lt;em&gt;Show all windows&lt;/em&gt; option.
 
-You can set the taskbar to &lt;strong&gt;Never&lt;/strong&gt; group windows, to &lt;strong&gt;Always&lt;/strong&gt; group windows or to group windows only &lt;strong&gt;When the Taskbar is Full&lt;/strong&gt;.
+You can set the taskbar to &lt;strong&gt;Never&lt;/strong&gt; group windows, to &lt;strong&gt;Always&lt;/strong&gt; group windows or to group windows only &lt;strong&gt;When the 
+Taskbar is Full&lt;/strong&gt;.
 
 By default the taskbar groups windows when it is full.</string>
                     </property>
                 </widget>
-                <widget class="TQLabel" row="7" column="0" rowspan="1" colspan="2">
+                <widget class="TQLabel" row="9" column="0" rowspan="1" colspan="2">
                     <property name="name">
-                        <cstring>groupTasksLabel</cstring>
+                        <cstring>textLabel1</cstring>
                     </property>
                     <property name="text">
-                        <string>&amp;Group similar tasks:</string>
+                        <string>A&amp;ppearance:</string>
                     </property>
                     <property name="buddy" stdset="0">
-                        <cstring>kcfg_GroupTasks</cstring>
-                    </property>
-                </widget>
-                <widget class="TQCheckBox" row="4" column="0" rowspan="1" colspan="3">
-                    <property name="name">
-                        <cstring>kcfg_ShowOnlyIconified</cstring>
-                    </property>
-                    <property name="text">
-                        <string>Show o&amp;nly minimized windows</string>
-                    </property>
-                    <property name="whatsThis" stdset="0">
-                        <string>Select this option if you want the taskbar to display &lt;b&gt;only&lt;/b&gt; minimized windows. 
-
-By default, this option is not selected and the taskbar will show all windows.</string>
+                        <cstring>appearance</cstring>
                     </property>
                 </widget>
-                <widget class="TQCheckBox" row="5" column="0" rowspan="1" colspan="3">
+                <widget class="TQCheckBox" row="6" column="0" rowspan="1" colspan="3">
                     <property name="name">
                         <cstring>kcfg_ShowIcon</cstring>
                     </property>
                     <property name="text">
                         <string>Sho&amp;w application icons</string>
                     </property>
+                    <property name="accel">
+                        <string>Alt+W</string>
+                    </property>
                     <property name="checked">
                         <bool>true</bool>
                     </property>
@@ -145,53 +162,73 @@
 By default this option is selected.</string>
                     </property>
                 </widget>
-                <widget class="TQCheckBox" row="3" column="0" rowspan="1" colspan="3">
+                <widget class="TQCheckBox" row="7" column="0" rowspan="1" colspan="3">
                     <property name="name">
-                        <cstring>showAllScreens</cstring>
+                        <cstring>kcfg_ShowWindowListBtn</cstring>
                     </property>
                     <property name="text">
-                        <string>Show windows from all sc&amp;reens</string>
+                        <string>Show window list &amp;button</string>
+                    </property>
+                    <property name="accel">
+                        <string>Alt+B</string>
                     </property>
                     <property name="checked">
                         <bool>true</bool>
                     </property>
                     <property name="whatsThis" stdset="0">
-                        <string>Turning this option off will cause the taskbar to display &lt;b&gt;only&lt;/b&gt; windows which are on the same Xinerama screen as the taskbar.
-
-By default, this option is selected and all windows are shown.</string>
+                        <string>Selecting this option causes the taskbar to display a button that, when clicked, shows a list of all windows in a popup menu.</string>
                     </property>
                 </widget>
-                <widget class="TQComboBox" row="8" column="2">
+                <widget class="TQCheckBox" row="5" column="0" rowspan="1" colspan="3">
                     <property name="name">
-                        <cstring>appearance</cstring>
+                        <cstring>kcfg_ShowTaskBarTips</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Show Task Bar Tips</string>
+                    </property>
+                    <property name="accel">
+                        <string>Alt+Y</string>
+                    </property>
+                    <property name="checked">
+                        <bool>true</bool>
                     </property>
                 </widget>
-                <widget class="TQLabel" row="8" column="0" rowspan="1" colspan="2">
+                <widget class="TQCheckBox" row="1" column="1" rowspan="1" colspan="2">
                     <property name="name">
-                        <cstring>textLabel1</cstring>
+                        <cstring>kcfg_SortByDesktop</cstring>
                     </property>
                     <property name="text">
-                        <string>A&amp;ppearance:</string>
+                        <string>Sort windows by desk&amp;top</string>
                     </property>
-                    <property name="buddy" stdset="0">
-                        <cstring>appearance</cstring>
+                    <property name="accel">
+                        <string>Alt+T</string>
+                    </property>
+                    <property name="checked">
+                        <bool>true</bool>
+                    </property>
+                    <property name="whatsThis" stdset="0">
+                        <string>Selecting this option causes the taskbar to show windows in order of the desktop they appear on.
+
+By default this option is selected.</string>
                     </property>
                 </widget>
-                <widget class="TQCheckBox" row="9" column="0" rowspan="1" colspan="2">
+                <widget class="TQCheckBox" row="4" column="0" rowspan="1" colspan="3">
                     <property name="name">
-                        <cstring>kcfg_UseCustomColors</cstring>
-                    </property>
-                    <property name="enabled">
-                        <bool>true</bool>
+                        <cstring>kcfg_ShowOnlyIconified</cstring>
                     </property>
                     <property name="text">
-                        <string>Use &amp;custom colors</string>
+                        <string>Show o&amp;nly minimized windows</string>
                     </property>
-                    <property name="checked">
-                        <bool>true</bool>
+                    <property name="accel">
+                        <string>Alt+N</string>
+                    </property>
+                    <property name="whatsThis" stdset="0">
+                        <string>Select this option if you want the taskbar to display &lt;b&gt;only&lt;/b&gt; minimized windows. 
+
+By default, this option is not selected and the taskbar will show all windows.</string>
                     </property>
                 </widget>
-                <widget class="TQLayoutWidget" row="10" column="0" rowspan="1" colspan="3">
+                <widget class="TQLayoutWidget" row="11" column="0" rowspan="1" colspan="3">
                     <property name="name">
                         <cstring>layout9</cstring>
                     </property>
