diff -Nur kdebase-3.5.1/kcontrol/clock/dtime.cpp kdebase-3.5.1new/kcontrol/clock/dtime.cpp
--- kdebase-3.5.1/kcontrol/clock/dtime.cpp	2005-10-10 11:03:48.000000000 -0400
+++ kdebase-3.5.1new/kcontrol/clock/dtime.cpp	2006-03-12 21:53:31.044200768 -0500
@@ -243,11 +243,11 @@
   KConfig config("kcmclockrc", true, false);
   config.setGroup("NTP");
   timeServerList->insertStringList(TQStringList::split(',', config.readEntry("servers",
-    i18n("Public Time Server (pool.ntp.org),\
+    "pool.ntp.org,\
 asia.pool.ntp.org,\
 europe.pool.ntp.org,\
 north-america.pool.ntp.org,\
-oceania.pool.ntp.org"))));
+oceania.pool.ntp.org")));
   setDateTimeAuto->setChecked(config.readBoolEntry("enabled", false));
 
   // Reset to the current date and time
diff -Nur kdebase-3.5.1/kcontrol/clock/tzone.cpp kdebase-3.5.1new/kcontrol/clock/tzone.cpp
--- kdebase-3.5.1/kcontrol/clock/tzone.cpp	2006-01-19 12:00:53.000000000 -0500
+++ kdebase-3.5.1new/kcontrol/clock/tzone.cpp	2006-03-12 21:53:39.039985224 -0500
@@ -45,13 +45,19 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #endif
-
+#include <klistviewsearchline.h>
+#include <qlayout.h>
 Tzone::Tzone(TQWidget * parent, const char *name)
   : TQVGroupBox(parent, name)
 {
     setTitle(i18n("To change the timezone, select your area from the list below"));
 
+
+    KListViewSearchLine * tzsearch = new KListViewSearchLine(this,"");	
     tzonelist = new KTimezoneWidget(this, "ComboBox_1", &m_zoneDb);
+
+//    KListViewSearchLine * tzsearch = new KListViewSearchLine(this,tzonelist,"");	
+	tzsearch->setListView(tzonelist);
     connect( tzonelist, TQT_SIGNAL(selectionChanged()), TQT_SLOT(handleZoneChange()) );
 
     m_local = new TQLabel(this);
