diff -Nur kdebase-3.5.1/kicker/kicker/core/kicker.cpp kdebase-3.5.1n/kicker/kicker/core/kicker.cpp
--- kdebase-3.5.1/kicker/kicker/core/kicker.cpp	2005-10-10 23:03:59.000000000 +0800
+++ kdebase-3.5.1n/kicker/kicker/core/kicker.cpp	2006-03-04 14:03:25.000000000 +0800
@@ -74,7 +74,7 @@
     // initialize the configuration object
     KickerSettings::instance(instanceName() + "rc");
 
-    if (KCrash::crashHandler() == 0 )
+   /* if (KCrash::crashHandler() == 0 )
     {
         // this means we've most likely crashed once. so let's see if we
         // stay up for more than 2 minutes time, and if so reset the
@@ -82,12 +82,12 @@
         QTimer::singleShot(120000, this, SLOT(setCrashHandler()));
     }
     else
-    {
+    {*/
         // See if a crash handler was installed. It was if the -nocrashhandler
         // argument was given, but the app eats the kde options so we can't
         // check that directly. If it wasn't, don't install our handler either.
         setCrashHandler();
-    }
+    //}
 
     // Make kicker immutable if configuration modules have been marked immutable
     if (isImmutable() && kapp->authorizeControlModules(Kicker::configModules(true)).isEmpty())
@@ -158,16 +158,18 @@
 
 void Kicker::setCrashHandler()
 {
-    KCrash::setEmergencySaveFunction(Kicker::crashHandler);
+//    KCrash::setEmergencySaveFunction(Kicker::crashHandler);
+	KCrash::setCrashHandler(Kicker::crashHandler);
 }
 
 void Kicker::crashHandler(int /* signal */)
 {
-    fprintf(stderr, "kicker: crashHandler called\n");
+ /*   fprintf(stderr, "kicker: crashHandler called\n");
 
     DCOPClient::emergencyClose();
     sleep(1);
-    system("kicker --nocrashhandler &"); // try to restart
+    system("kicker --nocrashhandler &"); // try to restart*/
+	exit(0);
 }
 
 void Kicker::slotToggleShowDesktop()
