Index: ksysguard/ksysguardd/Linux/lmsensors.c
===================================================================
--- ksysguard/ksysguardd/Linux/lmsensors.c	(Revision 779545)
+++ ksysguard/ksysguardd/Linux/lmsensors.c	(Revision 779546)
@@ -29,7 +29,7 @@
 
 #include "lmsensors.h"
 
-#ifdef HAVE_LMSENSORS
+#ifdef HAVE_SENSORS_SENSORS_H
 #include <sensors/sensors.h>
 
 #ifndef SENSORS_API_VERSION
@@ -293,7 +293,7 @@
 #endif
 }
 
-#else /* HAVE_LMSENSORS */
+#else /* HAVE_SENSORS_SENSORS_H */
 
 /* dummy version for systems that have no lmsensors support */
 
