diff -up kdebase-3.5.8/konsole/konsole/main.cpp.konsole-bz#244906 kdebase-3.5.8/konsole/konsole/main.cpp
--- kdebase-3.5.8/konsole/konsole/main.cpp.konsole-bz#244906	2007-10-08 04:51:33.000000000 -0500
+++ kdebase-3.5.8/konsole/konsole/main.cpp	2007-10-13 06:59:48.000000000 -0500
@@ -46,9 +46,11 @@
 
 #include "konsole.h"
 
+#if 0
 #if defined(Q_WS_X11) && defined(HAVE_XRENDER) && QT_VERSION >= 0x030300
 #define COMPOSITE
 #endif
+#endif
 
 #ifdef COMPOSITE
 # include <X11/Xlib.h>
