--- kdebase-3.5.1/kdm/kfrontend/genkdmconf.c.tn	2006-02-15 22:57:04.000000000 +0100
+++ kdebase-3.5.1/kdm/kfrontend/genkdmconf.c	2006-02-15 23:05:31.000000000 +0100
@@ -2845,7 +2845,7 @@
 	f = Create( newkdmrc, kdmrcmode );
 	wrconf( f );
 	fclose( f );
-
+#if 0
 	ASPrintf( &nname, "%s/README", newdir );
 	f = Create( nname, 0644 );
 	fprintf( f,
@@ -2894,6 +2894,6 @@
 "\nTry 'genkdmconf --help' if you want to generate another configuration.\n"
 "\nYou may delete this README.\n" );
 	fclose( f );
-
+#endif
 	return 0;
 }
--- kdebase-3.5.1/kdm/Makefile.am.tn	2006-02-15 22:56:02.000000000 +0100
+++ kdebase-3.5.1/kdm/Makefile.am	2006-02-15 22:55:56.000000000 +0100
@@ -4,9 +4,6 @@
 
 noinst_DATA=config.ci
 
-kdmdocdir = $(datadir)/doc/kdm
-kdmdoc_DATA = README
-
 install-data-local:
 	-@test -n "$(DESTDIR)" || test -z "$(PAM)" || { $(top_srcdir)/mkpamserv $(PAM) && $(top_srcdir)/mkpamserv -P $(PAM)-np; }
 
