diff -ur kdebase-3.5.10.orig/kcontrol/kfontinst/kfontinst/Fontmap.cpp kdebase-3.5.10/kcontrol/kfontinst/kfontinst/Fontmap.cpp
--- kdebase-3.5.10.orig/kcontrol/kfontinst/kfontinst/Fontmap.cpp	2009-05-18 08:54:22.000000000 +0800
+++ kdebase-3.5.10/kcontrol/kfontinst/kfontinst/Fontmap.cpp	2009-05-18 12:03:58.000000000 +0800
@@ -44,7 +44,7 @@
 
 using namespace std;
 
-static char * findSpace(char *str)
+static char * findSpace(char *str)
 {
     while(str && *str!=' ' && *str!='\t')
         str++;
@@ -65,7 +65,7 @@
     char a[constMaxLen+1],
          b[constFileMaxLen+1];
 
-    char *slash1=strchr(line, '/'),
+    char *slash1=(char *)strchr(line, '/'),
          *space1=slash1 ? findSpace(slash1) : NULL, //strchr(slash1, ' ') : NULL,
          *ob=slash1 ? strchr(slash1, '(') : NULL,
          *cb=ob ? strchr(ob, ')') : NULL,
