--- kdebase-3.4.3.bak/nsplugins/pluginscan.cpp.orig	2005-10-05 21:38:27.000000000 +0800
+++ kdebase-3.4.3.bak/nsplugins/pluginscan.cpp	2005-10-20 18:26:44.000000000 +0800
@@ -184,7 +184,7 @@
         if (!description.isEmpty())
             ts << "Name=" << description << endl;
         else
-            ts << "Name=" << i18n("Netscape plugin mimeinfo") << endl;
+            ts << "Name=" << "Netscape plugin mimeinfo" << endl;
 
         f.close();
     }
@@ -489,10 +489,10 @@
         QTextStream ts(&f);
 
         ts << "[Desktop Entry]" << endl;
-        ts << "Name=" << i18n("Netscape plugin viewer") << endl;
+        ts << "Name=" << "Netscape plugin viewer" << endl;
         ts << "Type=Service" << endl;
         ts << "Icon=netscape" << endl;
-        ts << "Comment=" << i18n("Netscape plugin viewer") << endl;
+        ts << "Comment=" << "Netscape plugin viewer" << endl;
         ts << "X-KDE-Library=libnsplugin" << endl;
         ts << "InitialPreference=0" << endl;
         ts << "ServiceTypes=KParts/ReadOnlyPart,Browser/View" << endl;
