diff -Nur kdebase-3.2.0/kcontrol/launch/kcmlaunch.h kdebase-3.2.0new/kcontrol/launch/kcmlaunch.h
--- kdebase-3.2.0/kcontrol/launch/kcmlaunch.h	2003-08-03 06:35:04.000000000 +0800
+++ kdebase-3.2.0new/kcontrol/launch/kcmlaunch.h	2003-08-05 23:48:23.000000000 +0800
@@ -54,7 +54,7 @@
       BusyCursor            = 1 << 0,
       TaskbarButton         = 1 << 1,
 
-      Default = BusyCursor | TaskbarButton
+      Default = TaskbarButton
     };
 
 
