--- kdebase-3.5.4/khelpcenter/searchhandlers/khc_htsearch.pl.orig	2006-08-18 16:31:15.000000000 +0200
+++ kdebase-3.5.4/khelpcenter/searchhandlers/khc_htsearch.pl	2006-08-18 16:39:30.000000000 +0200
@@ -8,7 +8,7 @@
 use open IO => ':utf8';
 use open ':std';
 
-my $htsearchpath="/srv/www/cgi-bin/htsearch";
+my $htsearchpath="/var/www/cgi-bin/htsearch";
 
 my $config;
 my $format;
--- kdebase-3.5.4/khelpcenter/searchhandlers/khc_htdig.pl.in.orig	2006-08-18 16:30:16.000000000 +0200
+++ kdebase-3.5.4/khelpcenter/searchhandlers/khc_htdig.pl.in	2006-08-18 16:30:37.000000000 +0200
@@ -26,7 +26,7 @@
 
 use Getopt::Long;
 
-my $htdigdata = "/srv/www/htdig/common/";
+my $htdigdata = "/usr/share/htdig/";
 my $htdigbin = "/usr/bin";
 my $kdeprefix = "@prefix@";
 chomp $kdeprefix;
--- kdebase-3.5.4/khelpcenter/searchhandlers/khc_docbookdig.pl.in.orig	2006-08-18 16:28:11.000000000 +0200
+++ kdebase-3.5.4/khelpcenter/searchhandlers/khc_docbookdig.pl.in	2006-08-18 16:28:55.000000000 +0200
@@ -26,7 +26,7 @@
 
 use Getopt::Long;
 
-my $htdigdata = "/srv/www/htdig/common/";
+my $htdigdata = "/usr/share/htdig/";
 my $htdigbin = "/usr/bin";
 my $kdeprefix = "@prefix@";
 chomp $kdeprefix;
