diff -u -r kdebase-635804.orig/konsole/eventsrc kdebase-635804/konsole/eventsrc
--- kdebase-635804.orig/konsole/eventsrc	2007-02-27 15:26:53.000000000 +0800
+++ kdebase-635804/konsole/eventsrc	2007-02-27 15:27:57.000000000 +0800
@@ -161,7 +161,7 @@
 Comment[vi]=Chuông phát ra trong một phiên chạy nhìn thấy
 Comment[zh_CN]=可见会话中发生的响铃
 Comment[zh_TW]=在可見的工作階段使用響鈴
-default_sound=KDE_Beep_Connect.ogg
+default_sound=KDE_Beep_Connect.wav
 default_presentation=1
 
 [BellInvisible]
@@ -294,7 +294,7 @@
 Comment[vi]=Chuông phát ra trong một phiên chạy không nhìn thấy
 Comment[zh_CN]=不可见会话中发生的响铃
 Comment[zh_TW]=在不可見的工作階段使用響鈴
-default_sound=KDE_Beep_Connect.ogg
+default_sound=KDE_Beep_Connect.wav
 default_presentation=16
 
 [Activity]
