--- kdebase-3.4.3.bak/nsplugins/viewer/nsplugin.cpp	2005-10-05 21:38:27.000000000 +0800
+++ kdebase-3.4.3/nsplugins/viewer/nsplugin.cpp	2005-10-21 19:40:23.000000000 +0800
@@ -912,6 +912,10 @@
 
    _win.x = 0;
    _win.y = 0;
+   //_height and _width can't equal to zero. patched by KanKer.
+   if (_height==0 || _width==0)
+	return NPERR_NO_ERROR;
+
    _win.height = _height;
    _win.width = _width;
    _win.type = NPWindowTypeWindow;
