diff -ur kdebase-3.5.13.orig/konqueror/konq_tabs.cc kdebase-3.5.13/konqueror/konq_tabs.cc
--- kdebase-3.5.13.orig/konqueror/konq_tabs.cc	2010-11-29 01:53:25.000000000 -0500
+++ kdebase-3.5.13/konqueror/konq_tabs.cc	2010-11-29 02:41:34.000000000 -0500
@@ -173,6 +173,9 @@
            TQT_SLOT( slotReceivedDropEvent( TQWidget *, TQDropEvent * ) ) );
   connect( this, TQT_SIGNAL( initiateDrag( TQWidget * )),
            TQT_SLOT( slotInitiateDrag( TQWidget * ) ) );
+  //doubleclick close tab.
+  connect( this, TQT_SIGNAL( mouseDoubleClick( TQWidget * ) ),
+           m_pViewManager->mainWindow(), TQT_SLOT( slotRemoveTab()  ) );
 }
 
 KonqFrameTabs::~KonqFrameTabs()

