diff -ur tdebase-git20111213.orig/CMakeLists.txt tdebase-git20111213/CMakeLists.txt
--- tdebase-git20111213.orig/CMakeLists.txt	2011-12-14 07:09:51.000000000 -0500
+++ tdebase-git20111213/CMakeLists.txt	2011-12-14 13:05:02.000000000 -0500
@@ -59,6 +59,7 @@
 option( WITH_SAMBA "Enable SAMBA support" OFF )
 option( WITH_OPENEXR "Enable openexr support" OFF )
 option( WITH_XCOMPOSITE "Enable xcomposite support" OFF )
+option( WITH_XTST "Enable xtst support" OFF)
 option( WITH_XCURSOR "Enable xcursor support" OFF )
 option( WITH_XFIXES "Enable xfixes support" OFF )
 option( WITH_XRANDR "Enable xrandr support" OFF )
diff -ur tdebase-git20111213.orig/ConfigureChecks.cmake tdebase-git20111213/ConfigureChecks.cmake
--- tdebase-git20111213.orig/ConfigureChecks.cmake	2011-12-14 07:09:52.000000000 -0500
+++ tdebase-git20111213/ConfigureChecks.cmake	2011-12-14 13:06:23.000000000 -0500
@@ -76,6 +76,15 @@
   endif( XCOMPOSITE_FOUND )
 endif( )
 
+# xtst (ktrash)
+if( WITH_XTST)
+  pkg_search_module( XTST xtst )
+  if( XTST_FOUND )
+    set( HAVE_XTST 1 )
+  else( XTST_FOUND )
+    tde_message_fatal( "xtst is requested, but was not found on your system" )
+  endif( XTST_FOUND )
+endif( )
 
 # xfixes (klipper, kicker)
 if( WITH_XFIXES )
