--- src/cmd/zdb/zdb.c~	2012-01-09 02:31:29.000000000 -0600
+++ src/cmd/zdb/zdb.c	2014-02-10 10:42:59.775869701 -0600
@@ -1483,7 +1483,7 @@
 {
 	time_t timestamp = ub->ub_timestamp;
 
-	(void) printf(header ? header : "");
+	(void) printf(header ? "%s", header : "%s", "");
 	(void) printf("\tmagic = %016llx\n", (u_longlong_t)ub->ub_magic);
 	(void) printf("\tversion = %llu\n", (u_longlong_t)ub->ub_version);
 	(void) printf("\ttxg = %llu\n", (u_longlong_t)ub->ub_txg);
@@ -1495,7 +1495,7 @@
 		sprintf_blkptr(blkbuf, &ub->ub_rootbp);
 		(void) printf("\trootbp = %s\n", blkbuf);
 	}
-	(void) printf(footer ? footer : "");
+	(void) printf(footer ? "%s", footer : "%s", "");
 }
 
 static void
