diff -up gpsd-3.5/SConstruct.libcap gpsd-3.5/SConstruct
--- gpsd-3.5/SConstruct.libcap	2012-04-14 15:36:18.000000000 +0200
+++ gpsd-3.5/SConstruct	2012-04-30 10:39:45.370120340 +0200
@@ -458,7 +458,7 @@ else:
     confdefs.append("/* #undef HAVE_LIBRT */\n")
     rtlibs = []
 
-if config.CheckLib('libcap'):
+if False and config.CheckLib('libcap'):
     confdefs.append("#define HAVE_LIBCAP 1\n")
     # System library - no special flags
     rtlibs = ["-lcap"]
