From 2a809cfbe5335bfa13a9f6496c4e04fffd93d930 Mon Sep 17 00:00:00 2001
From: Markus Steinborn <gnugv_maintainer@yahoo.de>
Date: Sat, 24 Apr 2010 16:55:21 +0200
Subject: [PATCH] Xaw3d-1.5E-cast.patch

---
 xc/lib/Xaw3d/Scrollbar.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git libXaw3d-1.6.1/src/Scrollbar.c libXaw3d-1.6.1/src/Scrollbar.c
index 0566733..6ed1480 100644
--- libXaw3d-1.6.1/src/Scrollbar.c
+++ libXaw3d-1.6.1/src/Scrollbar.c
@@ -1375,7 +1375,7 @@ static void NotifyThumb (w, event, params, num_params )
        the arrows. */
     top += 0.0001;
 /* #endif */
-    XtCallCallbacks (w, XtNthumbProc, *(XtPointer*)&top);
+    XtCallCallbacks (w, XtNthumbProc, (XtPointer)(long)top);
     XtCallCallbacks (w, XtNjumpProc, (XtPointer)&top);
 }
 
-- 
1.6.1

