diff -up libXaw3d-1.6.1/include/X11/Xaw3d/Text.h.compat libXaw3d-1.6.1/include/X11/Xaw3d/Text.h
--- libXaw3d-1.6.1/include/X11/Xaw3d/Text.h.compat	2011-09-19 15:42:36.000000000 -0600
+++ libXaw3d-1.6.1/include/X11/Xaw3d/Text.h	2012-02-26 17:35:36.777982148 -0700
@@ -242,6 +242,12 @@ extern Widget XawTextGetSource(
     Widget		/* w */
 );
 
+extern Widget XawTextGetSink(
+#if NeedFunctionPrototypes
+    Widget		/* w */
+#endif
+);
+
 extern XawTextPosition XawTextSearch(
     Widget			/* w */,
 #if NeedWidePrototypes
diff -up libXaw3d-1.6.1/src/Command.c.compat libXaw3d-1.6.1/src/Command.c
--- libXaw3d-1.6.1/src/Command.c.compat	2011-09-19 15:42:36.000000000 -0600
+++ libXaw3d-1.6.1/src/Command.c	2012-02-26 17:33:12.007416808 -0700
@@ -92,6 +92,8 @@ static XtResource resources[] = {
    {XtNcornerRoundPercent, XtCCornerRoundPercent, XtRDimension,
         sizeof(Dimension), offset(command.corner_round), XtRImmediate,
 	(XtPointer) 25},
+   {XtNshadowWidth, XtCShadowWidth, XtRDimension, sizeof(Dimension),
+	offset(threeD.shadow_width), XtRImmediate, (XtPointer) 2},
    {XtNborderWidth, XtCBorderWidth, XtRDimension, sizeof(Dimension),
       XtOffsetOf(RectObjRec,rectangle.border_width), XtRImmediate,
       (XtPointer) 0}
diff -up libXaw3d-1.6.1/src/Label.c.compat libXaw3d-1.6.1/src/Label.c
--- libXaw3d-1.6.1/src/Label.c.compat	2012-02-26 17:33:11.990417563 -0700
+++ libXaw3d-1.6.1/src/Label.c	2012-02-26 17:33:12.008416763 -0700
@@ -115,6 +115,8 @@ static XtResource resources[] = {
 	offset(label.pixmap), XtRImmediate, (XtPointer)None},
     {XtNresize, XtCResize, XtRBoolean, sizeof(Boolean),
 	offset(label.resize), XtRImmediate, (XtPointer)True},
+    {XtNshadowWidth, XtCShadowWidth, XtRDimension, sizeof(Dimension),
+	offset(threeD.shadow_width), XtRImmediate, (XtPointer) 0},
     {XtNborderWidth, XtCBorderWidth, XtRDimension, sizeof(Dimension),
          XtOffsetOf(RectObjRec,rectangle.border_width), XtRImmediate,
          (XtPointer)1}
diff -up libXaw3d-1.6.1/src/Text.c.compat libXaw3d-1.6.1/src/Text.c
--- libXaw3d-1.6.1/src/Text.c.compat	2011-10-06 13:17:19.000000000 -0600
+++ libXaw3d-1.6.1/src/Text.c	2012-02-26 17:34:39.884510665 -0700
@@ -3312,6 +3312,17 @@ XawTextGetSource(Widget w)
   return ((TextWidget)w)->text.source;
 }
 
+Widget
+#if NeedFunctionPrototypes
+XawTextGetSink(Widget w)
+#else
+XawTextGetSink(w)
+Widget w;
+#endif
+{
+  return (((TextWidget)w)->text.sink);
+}
+
 void
 XawTextDisplayCaret (Widget w,
 #if NeedWidePrototypes
