From 23b3ca50395f180cba4a923eb8827dcc961629da Mon Sep 17 00:00:00 2001
From: Markus Steinborn <gnugv_maintainer@yahoo.de>
Date: Sat, 24 Apr 2010 16:48:38 +0200
Subject: [PATCH] Xaw3d-1.5E-close-destroy-crash.patch

---
 xc/lib/Xaw3d/XawIm.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git libXaw3d-1.6.1/src/XawIm.c libXaw3d-1.6.1/src/XawIm.c
index ead7166..730c393 100644
--- libXaw3d-1.6.1/src/XawIm.c
+++ libXaw3d-1.6.1/src/XawIm.c
@@ -331,12 +331,12 @@ static void DestroyAllIM( ve )
     /*
      * Close Input Method
      */
-    CloseIM(ve);
     if (!XFindContext(XDisplayOfIM(ve->im.xim), (Window)ve->im.xim, errContext,
 		      (XPointer*)&contextErrData)) {
 	if (contextErrData) XtFree((char *)contextErrData);
     }
     XDeleteContext(XDisplayOfIM(ve->im.xim), (Window)ve->im.xim, errContext);
+    CloseIM(ve);
     ve->im.xim = NULL;
 
     /*
-- 
1.6.1

