diff --git a/Functions/Prompts/prompt_bigfade_setup b/Functions/Prompts/prompt_bigfade_setup
index 2d95f31..f2f27f4 100644
--- a/Functions/Prompts/prompt_bigfade_setup
+++ b/Functions/Prompts/prompt_bigfade_setup
@@ -7,7 +7,7 @@ prompt_bigfade_help () {
   cat <<EOH
 This prompt is color-scheme-able.  You can invoke it thus:
 
-  prompt bigfade [<fade-bar> [<userhost> [<date> [<cwd>]]]]
+  prompt bigfade [8bit] [<fade-bar> [<userhost> [<date> [<cwd>]]]]
 
 where the parameters are the colors for the fade-bar, user@host text,
 date text, and current working directory respectively.  The default
@@ -27,9 +27,12 @@ prompt_bigfade_setup () {
   local date=${3:-'white'}
   local cwd=${4:-'yellow'}
 
+  if [[ $1 == '8bit' ]]; then
+  shift
   local -A schars
   autoload -Uz prompt_special_chars
   prompt_special_chars
+  fi
 
   PS1="%B%F{$fadebar}$schars[333]$schars[262]$schars[261]$schars[260]%B%F{$userhost}%K{$fadebar}%n@%m%b%k%f%F{$fadebar}%K{black}$schars[260]$schars[261]$schars[262]$schars[333]%b%f%k%F{$fadebar}%K{black}$schars[333]$schars[262]$schars[261]$schars[260]%B%F{$date}%K{black} %D{%a %b %d} %D{%I:%M:%S%P}$prompt_newline%B%F{$cwd}%K{black}%d>%b%f%k "
   PS2="%B%F{$fadebar}$schars[333]$schars[262]$schars[261]$schars[260]%b%F{$fadebar}%K{black}$schars[260]$schars[261]$schars[262]$schars[333]%F{$fadebar}%K{black}$schars[333]$schars[262]$schars[261]$schars[260]%B%F{$fadebar}>%b%f%k "
diff --git a/Functions/Prompts/prompt_elite2_setup b/Functions/Prompts/prompt_elite2_setup
index 754bf60..0006665 100644
--- a/Functions/Prompts/prompt_elite2_setup
+++ b/Functions/Prompts/prompt_elite2_setup
@@ -6,7 +6,7 @@ prompt_elite2_help () {
   cat <<EOH
 This prompt is color-scheme-able.  You can invoke it thus:
 
-  prompt elite2 [<text-color> [<parentheses-color>]]
+  prompt elite2 [8bit] [<text-color> [<parentheses-color>]]
 
 The default colors are both cyan.  This theme works best with a dark
 background.
@@ -21,9 +21,12 @@ prompt_elite2_setup () {
   local text_col=${1:-'cyan'}
   local parens_col=${2:-$text_col}
 
+  if [[ $1 == '8bit' ]]; then
+  shift
   local -A schars
   autoload -Uz prompt_special_chars
   prompt_special_chars
+  fi
 
   local text="%b%F{$text_col}"
   local parens="%B%F{$parens_col}"
diff --git a/Functions/Prompts/prompt_elite_setup b/Functions/Prompts/prompt_elite_setup
index 0d57f50..e4b3fb1 100644
--- a/Functions/Prompts/prompt_elite_setup
+++ b/Functions/Prompts/prompt_elite_setup
@@ -6,7 +6,7 @@ prompt_elite_help () {
   cat <<EOH
 This prompt is color-scheme-able.  You can invoke it thus:
 
-  prompt elite [<text-color> [<punctuation-color>]]
+  prompt elite [8bit] [<text-color> [<punctuation-color>]]
 
 The default colors are red and blue respectively.  This theme is
 intended for use with a black background.
@@ -21,9 +21,12 @@ prompt_elite_setup () {
   local text=${1:-'red'}
   local punctuation=${2:-'blue'}
 
+  if [[ $1 == '8bit' ]]; then
+  shift
   local -A schars
   autoload -Uz prompt_special_chars
   prompt_special_chars
+  fi
 
   PS1="%F{$text}$schars[332]$schars[304]%F{$punctuation}(%F{$text}%n%F{$punctuation}@%F{$text}%m%F{$punctuation})%F{$text}-%F{$punctuation}(%F{$text}%D{%I:%M%P}%F{$punctuation}-:-%F{$text}%D{%m}%F{$punctuation}%F{$text}/%D{%d}%F{$punctuation})%F{$text}$schars[304]-%F{$punctuation}$schars[371]%F{$text}-$schars[371]$schars[371]%F{$punctuation}$schars[372]$prompt_newline%F{$text}$schars[300]$schars[304]%F{$punctuation}(%F{$text}%1~%F{$punctuation})%F{$text}$schars[304]$schars[371]%F{$punctuation}$schars[372]%f"
   PS2="> "
diff --git a/Functions/Prompts/prompt_fade_setup b/Functions/Prompts/prompt_fade_setup
index d915cc0..ddd0162 100644
--- a/Functions/Prompts/prompt_fade_setup
+++ b/Functions/Prompts/prompt_fade_setup
@@ -7,7 +7,7 @@ prompt_fade_help () {
   cat <<EOH
 This prompt is color-scheme-able.  You can invoke it thus:
 
-  prompt fade [<fade-bar-and-cwd> [<userhost> [<date>]]] 
+  prompt fade [8bit] [<fade-bar-and-cwd> [<userhost> [<date>]]] 
 
 where the parameters are the colors for the fade-bar and current
 working directory, user@host text, and date text respectively.  The
@@ -27,9 +27,12 @@ prompt_fade_setup () {
   local userhost=${2:-'white'}
   local date=${3:-'white'}
 
+  if [[ $1 == '8bit' ]]; then
+  shift
   local -A schars
   autoload -Uz prompt_special_chars
   prompt_special_chars
+  fi
 
   PS1="%F{$fadebar_cwd}%B%K{$fadebar_cwd}$schars[333]$schars[262]$schars[261]$schars[260]%F{$userhost}%K{$fadebar_cwd}%B%n@%m%b%F{$fadebar_cwd}%K{black}$schars[333]$schars[262]$schars[261]$schars[260]%F{$date}%K{black}%B %D{%a %b %d} %D{%I:%M:%S%P} $prompt_newline%F{$fadebar_cwd}%K{black}%B%~/%b%k%f "
   PS2="%F{$fadebar_cwd}%K{black}$schars[333]$schars[262]$schars[261]$schars[260]%f%k>"
diff --git a/Functions/Prompts/prompt_fire_setup b/Functions/Prompts/prompt_fire_setup
index 7bb8367..3217980 100644
--- a/Functions/Prompts/prompt_fire_setup
+++ b/Functions/Prompts/prompt_fire_setup
@@ -8,7 +8,7 @@ prompt_fire_help () {
   cat <<EOH
 This prompt is color-scheme-able.  You can invoke it thus:
 
-  prompt fire [<fire1> [<fire2> [<fire3> [<userhost> [<date> [<cwd>]]]]]]
+  prompt fire [8bit] [<fire1> [<fire2> [<fire3> [<userhost> [<date> [<cwd>]]]]]]
 
 where the parameters are the three fire colors, and the colors for the
 user@host text, date text, and current working directory respectively.
@@ -29,9 +29,12 @@ prompt_fire_setup () {
   local date=${5:-'white'}
   local cwd=${6:-'yellow'}
 
+  if [[ $1 == '8bit' ]]; then
+  shift
   local -a schars
   autoload -Uz prompt_special_chars
   prompt_special_chars
+  fi
 
   local GRAD1="%{$schars[333]$schars[262]$schars[261]$schars[260]%}"
   local GRAD2="%{$schars[260]$schars[261]$schars[262]$schars[333]%}"
