diff -ur libvisual-plugins-0.2.0/plugins/actor/G-Force/Makefile.am libvisual-plugins-0.2.0.mkinstalldirs/plugins/actor/G-Force/Makefile.am
--- libvisual-plugins-0.2.0/plugins/actor/G-Force/Makefile.am	2005-01-21 07:25:08.000000000 -0800
+++ libvisual-plugins-0.2.0.mkinstalldirs/plugins/actor/G-Force/Makefile.am	2005-05-07 13:05:50.000000000 -0700
@@ -26,7 +26,7 @@
 	unix/libvisual/libactor_gforce.la
 
 install-data-local:
-	test -d "$(DESTDIR)$(pkgdatadir)" || mkdirhier "$(DESTDIR)$(pkgdatadir)"
+	test -d "$(DESTDIR)$(pkgdatadir)" || $(mkinstalldirs) "$(DESTDIR)$(pkgdatadir)"
 	cp -r deffont "GForceColorMaps" "GForceDeltaFields" "GForceParticles" "GForceWaveShapes" "$(DESTDIR)$(pkgdatadir)"
 
 uninstall-local:
diff -ur libvisual-plugins-0.2.0/plugins/actor/G-Force/Makefile.in libvisual-plugins-0.2.0.mkinstalldirs/plugins/actor/G-Force/Makefile.in
--- libvisual-plugins-0.2.0/plugins/actor/G-Force/Makefile.in	2005-01-24 06:44:21.000000000 -0800
+++ libvisual-plugins-0.2.0.mkinstalldirs/plugins/actor/G-Force/Makefile.in	2005-05-07 13:06:39.000000000 -0700
@@ -613,7 +613,7 @@
 
 
 install-data-local:
-	test -d "$(DESTDIR)$(pkgdatadir)" || mkdirhier "$(DESTDIR)$(pkgdatadir)"
+	test -d "$(DESTDIR)$(pkgdatadir)" || $(mkinstalldirs) "$(DESTDIR)$(pkgdatadir)"
 	cp -r deffont "GForceColorMaps" "GForceDeltaFields" "GForceParticles" "GForceWaveShapes" "$(DESTDIR)$(pkgdatadir)"
 
 uninstall-local:
