*** ../binutils-2.23.2.orig/ld/ld.texinfo	2013-04-25 16:44:07.538768821 +0100
--- ld/ld.texinfo	2013-04-25 16:47:27.040774352 +0100
*************** in the section entitled ``GNU Free Docum
*** 142,153 ****
  @ifset ARM
  * ARM::				ld and the ARM family
  @end ifset
- @ifset HPPA
- * HPPA ELF32::                  ld and HPPA 32-bit ELF
- @end ifset
  @ifset M68HC11
  * M68HC11/68HC12::              ld and the Motorola 68HC11 and 68HC12 families
  @end ifset
  @ifset M68K
  * M68K::                        ld and Motorola 68K family
  @end ifset
--- 142,153 ----
  @ifset ARM
  * ARM::				ld and the ARM family
  @end ifset
  @ifset M68HC11
  * M68HC11/68HC12::              ld and the Motorola 68HC11 and 68HC12 families
  @end ifset
+ @ifset HPPA
+ * HPPA ELF32::                  ld and HPPA 32-bit ELF
+ @end ifset
  @ifset M68K
  * M68K::                        ld and Motorola 68K family
  @end ifset
*************** functionality are not listed.
*** 5998,6003 ****
--- 5998,6006 ----
  @ifset I960
  * i960::                        @command{ld} and the Intel 960 family
  @end ifset
+ @ifset M68HC11
+ * M68HC11/68HC12::		@code{ld} and the Motorola 68HC11 and 68HC12 families
+ @end ifset
  @ifset ARM
  * ARM::				@command{ld} and the ARM family
  @end ifset
*************** functionality are not listed.
*** 6013,6021 ****
  @ifset MSP430
  * MSP430::			@command{ld} and MSP430
  @end ifset
- @ifset M68HC11
- * M68HC11/68HC12::		@code{ld} and the Motorola 68HC11 and 68HC12 families
- @end ifset
  @ifset POWERPC
  * PowerPC ELF32::		@command{ld} and PowerPC 32-bit ELF Support
  @end ifset
--- 6016,6021 ----
*************** If you have more than one @code{SECT} st
*** 7863,7869 ****
  @printindex cp
  
  @tex
! % I think something like @colophon should be in texinfo.  In the
  % meantime:
  \long\def\colophon{\hbox to0pt{}\vfill
  \centerline{The body of this manual is set in}
--- 7863,7869 ----
  @printindex cp
  
  @tex
! % I think something like @@colophon should be in texinfo.  In the
  % meantime:
  \long\def\colophon{\hbox to0pt{}\vfill
  \centerline{The body of this manual is set in}
*************** If you have more than one @code{SECT} st
*** 7874,7880 ****
  \centerline{{\sl\fontname\tensl\/}}
  \centerline{are used for emphasis.}\vfill}
  \page\colophon
! % Blame: doc@cygnus.com, 28mar91.
  @end tex
  
  @bye
--- 7874,7880 ----
  \centerline{{\sl\fontname\tensl\/}}
  \centerline{are used for emphasis.}\vfill}
  \page\colophon
! % Blame: doc@@cygnus.com, 28mar91.
  @end tex
  
  @bye
