--- star-1.5/DEFAULTS/Defaults.linux.fix	2007-08-23 13:35:54.000000000 +0200
+++ star-1.5/DEFAULTS/Defaults.linux	2007-08-23 13:51:33.000000000 +0200
@@ -8,8 +8,8 @@
 # Compiler stuff
 #
 ###########################################################################
-DEFCCOM=	cc
-#DEFCCOM=	gcc
+#DEFCCOM=	cc
+DEFCCOM=	gcc
 
 ###########################################################################
 #
@@ -40,22 +40,21 @@ __LINUX_SRC_INCLUDE=	$(_LINUX_SRC_INCLUD
 LINUX_SRC_INCLUDE=      $(__LINUX_SRC_INCLUDE:$(_UNIQ)no%=%)
 
 DEFINCDIRS=	$(SRCROOT)/include
-LDPATH=		-L/opt/schily/lib
-#RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+LDPATH= 
+RUNPATH= 
 
 ###########################################################################
 #
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/opt/schily
-INS_KBASE=	/
-INS_RBASE=	/
+INS_BASE=	$(RPM_INSTALLDIR)/usr/
+INS_KBASE=	$(RPM_INSTALLDIR)/
+INS_RBASE=	$(RPM_INSTALLDIR)/
 #
 DEFUMASK=	002
 #
 DEFINSMODEF=	444
 DEFINSMODEX=	755
 DEFINSUSR=	root
-DEFINSGRP=	bin
+DEFINSGRP=	root
