diff -ur tdegraphics-git20111223/ksnapshot/CMakeLists.txt tdegraphics-git20111223.new/ksnapshot/CMakeLists.txt
--- tdegraphics-git20111223/ksnapshot/CMakeLists.txt	2011-12-23 07:16:58.000000000 -0500
+++ tdegraphics-git20111223.new/ksnapshot/CMakeLists.txt	2011-12-23 08:03:31.000000000 -0500
@@ -40,6 +40,6 @@
   SOURCES
     ksnapshotiface.skel main.cpp ksnapshot.cpp regiongrabber.cpp
     windowgrabber.cpp ksnapshotwidget.ui
-  LINK tdeprint-shared
+  LINK tdeprint-shared ${IMLIB_LIBRARIES}
   DESTINATION ${BIN_INSTALL_DIR}
 )
