diff -r 87dea3f5ebe7 src/pkg/runtime/pprof/pprof_test.go
--- a/src/pkg/runtime/pprof/pprof_test.go	Fri Nov 29 08:32:31 2013 +1100
+++ b/src/pkg/runtime/pprof/pprof_test.go	Fri Jan 24 13:47:42 2014 -0500
@@ -32,7 +32,7 @@
 	})
 }
 
-func TestCPUProfileMultithreaded(t *testing.T) {
+func testCPUProfileMultithreaded(t *testing.T) {
 	buf := make([]byte, 100000)
 	defer runtime.GOMAXPROCS(runtime.GOMAXPROCS(2))
 	testCPUProfile(t, []string{"crc32.ChecksumIEEE", "crc32.Update"}, func() {
