diff -ur rpmdevtools-8.4/rpmdev-bumpspec rpmdevtools-8.4.new/rpmdev-bumpspec
--- rpmdevtools-8.4/rpmdev-bumpspec	2013-07-31 20:50:51.000000000 +0000
+++ rpmdevtools-8.4.new/rpmdev-bumpspec	2014-02-26 10:16:09.785969458 +0000
@@ -215,6 +215,8 @@
     parser.add_option("-s", "--string", default=None,
                       help="bump trailing .STRING<DIGIT> component if found, "
                       "append .STRING1 if not; trumps -r")
+    parser.add_option("-n", "--nobump", default=False, action='store_true',
+		      help="no bump trailing, if specified no-op -s or -r") 
     parser.add_option("-V", "--verbose", default=False, action='store_true',
                       help="more output")
     parser.add_option("-v", "--version", default=False, action='store_true',
@@ -259,8 +261,9 @@
         except:
             # Not actually a parser error, but... meh.
             parser.error(sys.exc_info()[1])
-        s.bumpRelease()
-        s.writeFile(aspec)
+        if not opts.nobump:
+            s.bumpRelease()
+            s.writeFile(aspec)
 
         # Get EVR for changelog entry.
         cmd = ("rpm", "-q", "--specfile", "--define", "dist %{nil}",
