diff -up Crypt-OpenSSL-PKCS10-0.06/PKCS10.xs.openssl Crypt-OpenSSL-PKCS10-0.06/PKCS10.xs
--- Crypt-OpenSSL-PKCS10-0.06/PKCS10.xs.openssl	2006-02-24 22:15:56.000000000 +0100
+++ Crypt-OpenSSL-PKCS10-0.06/PKCS10.xs	2009-08-25 21:09:05.000000000 +0200
@@ -32,7 +32,7 @@ typedef struct
   if (New(0, p_var, p_size, p_type) == NULL) \
     { PACKAGE_CROAK("unable to alloc buffer"); }
 
-int add_ext(STACK_OF(X509_REQUEST) *sk, int nid, char *value);
+int add_ext(STACK_OF(X509_EXTENSION) *sk, int nid, char *value);
 X509_NAME *parse_name(char *str, long chtype, int multirdn); 
 
 /*
@@ -181,7 +181,7 @@ static void callback(int p, int n, void 
  * because we wont reference any other sections.
  */
 
-int add_ext(STACK_OF(X509_REQUEST) *sk, int nid, char *value)
+int add_ext(STACK_OF(X509_EXTENSION) *sk, int nid, char *value)
 	{
 	X509_EXTENSION *ex;
 	ex = X509V3_EXT_conf_nid(NULL, NULL, nid, value);
