diff -Nur kaffeine/src/input/audiobrowser/playlist.cpp kaffeine.fix_alternatecodec_gb18030/src/input/audiobrowser/playlist.cpp
--- kaffeine/src/input/audiobrowser/playlist.cpp	2008-01-21 01:50:48.000000000 +0800
+++ kaffeine.fix_alternatecodec_gb18030/src/input/audiobrowser/playlist.cpp	2008-07-04 08:45:55.000000000 +0800
@@ -278,7 +278,7 @@
 PlayList::PlayList( QWidget* parent, QObject *objParent, const char *name ) : KaffeineInput(objParent , name),
 		m_playTime(0), m_playTimeVisible(0), m_countVisible(0), m_searchSelection(false),
 		m_metaOnLoading(true), m_sortAscending(true), m_currentEntry(NULL), m_currentRandomListEntry(-1),
-		m_endless(false), m_random(false), m_useAlternateEncoding(false), m_alternateEncoding("ISO 8859-1")
+		m_endless(false), m_random(false), m_useAlternateEncoding(false), m_alternateEncoding("GB18030")
 {
 	google = NULL;
 
diff -Nur kaffeine/src/kaffeine.cpp kaffeine.fix_alternatecodec_gb18030/src/kaffeine.cpp
--- kaffeine/src/kaffeine.cpp	2008-07-03 22:53:25.000000000 +0800
+++ kaffeine.fix_alternatecodec_gb18030/src/kaffeine.cpp	2008-07-04 08:45:38.000000000 +0800
@@ -782,7 +782,7 @@
 	m_useAlternateEncoding = config->readBoolEntry("Use Alternate Encoding", false);
 	slotUseAlternateEncoding(m_useAlternateEncoding);
 
-	m_alternateEncoding = config->readEntry("Alternate Encoding Name", "ISO 8859-1");
+	m_alternateEncoding = config->readEntry("Alternate Encoding Name", "GB18030");
 	slotAlternateEncoding(m_alternateEncoding);
 
 	m_pauseVideo = config->readBoolEntry("PauseHiddenVideo", true);
