diff -up xf86-video-nouveau-1.0.1/src/nv_driver.c.ppc xf86-video-nouveau-1.0.1/src/nv_driver.c
--- xf86-video-nouveau-1.0.1/src/nv_driver.c.ppc	2012-07-02 12:08:19.172163488 +0200
+++ xf86-video-nouveau-1.0.1/src/nv_driver.c	2012-07-02 12:08:57.547809381 +0200
@@ -25,7 +25,9 @@
 #include "nv_include.h"
 
 #include "xorg-server.h"
+#if !defined(__powerpc__)
 #include "xf86int10.h"
+#endif
 #include "xf86drm.h"
 #include "xf86drmMode.h"
 #include "nouveau_drm.h"
diff -up xf86-video-nouveau-1.0.1/src/nv_include.h.ppc xf86-video-nouveau-1.0.1/src/nv_include.h
--- xf86-video-nouveau-1.0.1/src/nv_include.h.ppc	2012-07-02 11:40:16.658531972 +0200
+++ xf86-video-nouveau-1.0.1/src/nv_include.h	2012-07-02 11:46:14.203915773 +0200
@@ -30,7 +30,9 @@
 
 #include "xf86DDC.h"
 
+#if !defined(__powerpc__)
 #include "vbe.h"
+#endif
 
 #include "xf86RandR12.h"
 
diff -up xf86-video-nouveau-1.0.1/src/nv_type.h.ppc xf86-video-nouveau-1.0.1/src/nv_type.h
--- xf86-video-nouveau-1.0.1/src/nv_type.h.ppc	2012-07-02 12:07:22.543161675 +0200
+++ xf86-video-nouveau-1.0.1/src/nv_type.h	2012-07-02 12:09:03.701592241 +0200
@@ -3,7 +3,9 @@
 
 #include "colormapst.h"
 #include "xf86Cursor.h"
+#if !defined(__powerpc__)
 #include "xf86int10.h"
+#endif
 #include "exa.h"
 #ifdef XF86DRI
 #define _XF86DRI_SERVER_
