From bccfb8ac43da56cb1f53c4b421e09dcac1fd1af0 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Wed, 13 Feb 2013 15:17:08 +0100
Subject: [PATCH] lynx-2.8.6-redhat.patch rebased for 2.8.9

---
 lynx.cfg   | 14 +++++++++++---
 userdefs.h | 18 +++++++++---------
 2 files changed, 20 insertions(+), 12 deletions(-)

diff --git a/lynx.cfg b/lynx.cfg
index f01782f..ccc0846 100644
--- a/lynx.cfg
+++ b/lynx.cfg
@@ -1,8 +1,11 @@
 # $LynxId: lynx.cfg,v 1.247 2014/03/10 22:07:34 tom Exp $
 # lynx.cfg file.
-# The default placement for this file is /usr/local/lib/lynx.cfg (Unix)
+# The default placement for this file is /etc/lynx.cfg (Magic Linux)
 #                                     or Lynx_Dir:lynx.cfg (VMS)
 #
+# Please don't edit this file directly (it is updated with every Red Hat
+# Linux update, overwriting your changes). Instead, edit /etc/lynx-site.cfg.
+#
 # $Format: "#PRCS LYNX_VERSION \"$ProjectVersion$\""$
 #PRCS LYNX_VERSION "2.8.9dev.1"
 #
@@ -90,7 +93,7 @@
 #
 # Normally we expect you will connect to a remote site, e.g., the Lynx starting
 # site:
-STARTFILE:http://lynx.isc.org/
+STARTFILE:http://www.magiclinux.org
 #
 # As an alternative, you may want to use a local URL.  A good choice for this is
 # the user's home directory:
@@ -436,7 +439,7 @@ DEFAULT_INDEX_FILE:http://scout.wisc.edu/
 # Find RFC 1345 at
 .url http://tools.ietf.org/html/rfc1345
 #
-#CHARACTER_SET:iso-8859-1
+CHARACTER_SET:utf-8
 
 .h2 LOCALE_CHARSET
 # LOCALE_CHARSET overrides CHARACTER_SET if true, using the current locale to
@@ -1823,6 +1826,10 @@ DEFAULT_INDEX_FILE:http://scout.wisc.edu/
 .ex
 #DOWNLOADER:Save OS/390 binary file: iconv -f IBM-1047 -t ISO8859-1 %s >%s:FALSE
 
+# Added by Red Hat:
+DOWNLOADER:View with less:less %s:TRUE
+
+ 
 .h1 Interaction
 
 .h2 NO_DOT_FILES
@@ -3422,6 +3429,7 @@ COLOR:6:brightred:black
 #ENABLE_LYNXRC:VI_KEYS:ON
 #ENABLE_LYNXRC:VISITED_LINKS:ON
 .fi
+INCLUDE:/etc/lynx-site.cfg
 
 .h1 External Programs
 # Any of the compiled-in pathnames of external programs can be overridden
diff --git a/userdefs.h b/userdefs.h
index 40a954b..4afe835 100644
--- a/userdefs.h
+++ b/userdefs.h
@@ -105,7 +105,7 @@
  * mailcap files (see the examples in the samples directory).
  */
 #ifndef LYNX_CFG_FILE
-#define LYNX_CFG_FILE "Lynx_Dir:lynx.cfg"
+#define LYNX_CFG_FILE "/etc/lynx.cfg"
 #endif /* LYNX_CFG_FILE */
 
 #ifndef LYNX_CFG_PATH
@@ -119,8 +119,8 @@
  * Mappings in these global and personal files override any SUFFIX
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_EXTENSION_MAP "Lynx_Dir:mime.types"
-#define PERSONAL_EXTENSION_MAP "mime.types"
+#define GLOBAL_EXTENSION_MAP "/etc/mime.types"
+#define PERSONAL_EXTENSION_MAP ".mime.types"
 
 /**************************
  * The MAILCAP file allows you to map file MIME types to
@@ -129,7 +129,7 @@
  * Mappings in these global and personal files override any VIEWER
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_MAILCAP "Lynx_Dir:mailcap"
+#define GLOBAL_MAILCAP "/etc/mailcap"
 #define PERSONAL_MAILCAP ".mailcap"
 
 /**************************
@@ -300,8 +300,8 @@
 #define LYNX_CFG_PATH "."
 #define LYNX_CFG_FILE "./lynx.cfg"
 #else
-#define LYNX_CFG_PATH "/usr/local/lib"
-#define LYNX_CFG_FILE "/usr/local/lib/lynx.cfg"
+#define LYNX_CFG_PATH "/etc"
+#define LYNX_CFG_FILE "/etc/lynx.cfg"
 #endif /* DOSPATH */
 #endif /* LYNX_CFG_FILE */
 #endif /* HAVE_CONFIG_H */
@@ -345,7 +345,7 @@
  * use any default viewers for image types.  Note that open is used as
  * the default for NeXT, instead of the XLOADIMAGE_COMMAND definition.
  */
-#define XLOADIMAGE_COMMAND "xli %s &"
+#define XLOADIMAGE_COMMAND "display %s &"
 
 /**************************
  * For UNIX systems, SYSTEM_MAIL and SYSTEM_MAIL_FLAGS are set by the
@@ -366,7 +366,7 @@
  * the "TMPDIR" (unix), or "TEMP" or "TMP" (Windows,DOS,OS/2)
  * variable.
  */
-#define TEMP_SPACE "/tmp/"
+#define TEMP_SPACE "~"
 
 /********************************
  * Comment this line out to disable code that implements command logging
@@ -1172,7 +1172,7 @@
  * OSU server distribution.
  */
 #ifndef HAVE_CONFIG_H
-/* #define LYNXCGI_LINKS */
+#define LYNXCGI_LINKS
 #endif
 
 /*********************************
-- 
1.8.3.1

