From: Andrew Wilcox <AWilcox@Wilcox-Tech.com>
Bugzilla: https://bugzilla.redhat.com/782995

--- xf86-video-vmware-12.0.1/src/vmware.old	2012-03-08 11:13:16.000000000 -0500
+++ xf86-video-vmware-12.0.1/src/vmware.c	2012-03-16 18:20:25.406907242 -0400
@@ -354,6 +354,7 @@ 
      * Save the current video state.  Do it here before VMXGetVMwareSvgaId
      * writes to any registers.
      */
+    vgaHWSetStdFuncs(VGAHWPTR(pScrn));
     VMWARESave(pScrn);
 
     id = VMXGetVMwareSvgaId(pVMWARE);
