diff -up xf86-video-vmware-11.0.3/src/vmwarexinerama.c.jx xf86-video-vmware-11.0.3/src/vmwarexinerama.c
--- xf86-video-vmware-11.0.3/src/vmwarexinerama.c.jx	2011-11-16 19:12:55.000000000 -0500
+++ xf86-video-vmware-11.0.3/src/vmwarexinerama.c	2011-11-16 19:15:30.667991582 -0500
@@ -134,7 +134,6 @@ VMwareXineramaGetState(ClientPtr client)
     if(client->swapped) {
        vmswaps (&rep.sequenceNumber, n);
        vmswapl (&rep.length, n);
-       vmswaps (&rep.state, n);
     }
     WriteToClient(client, sizeof(xPanoramiXGetStateReply), (char *)&rep);
     return client->noClientException;
@@ -186,7 +185,6 @@ VMwareXineramaGetScreenCount(ClientPtr c
     if(client->swapped) {
        vmswaps(&rep.sequenceNumber, n);
        vmswapl(&rep.length, n);
-       vmswaps(&rep.ScreenCount, n);
     }
     WriteToClient(client, sizeof(xPanoramiXGetScreenCountReply), (char *)&rep);
     return client->noClientException;
@@ -239,8 +237,8 @@ VMwareXineramaGetScreenSize(ClientPtr cl
     if(client->swapped) {
        vmswaps(&rep.sequenceNumber, n);
        vmswapl(&rep.length, n);
-       vmswaps(&rep.width, n);
-       vmswaps(&rep.height, n);
+       vmswapl(&rep.width, n);
+       vmswapl(&rep.height, n);
     }
     WriteToClient(client, sizeof(xPanoramiXGetScreenSizeReply), (char *)&rep);
     return client->noClientException;
