diff -ur libindi-0.9.8/libs/indibase/alignment/CMakeLists.txt libindi-0.9.8.math/libs/indibase/alignment/CMakeLists.txt
--- libindi-0.9.8/libs/indibase/alignment/CMakeLists.txt	2014-05-09 16:53:05.404183770 +0200
+++ libindi-0.9.8.math/libs/indibase/alignment/CMakeLists.txt	2014-05-09 16:54:31.110896117 +0200
@@ -8,7 +8,7 @@
 	set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} ${CMAKE_GSL_CXX_FLAGS})
 endif (GSL_FOUND)
 
-set(INDI_MATH_PLUGINS_DIRECTORY "${DATA_INSTALL_DIR}MathPlugins/" CACHE PATH "Path to Math Plugins directory")
+set(INDI_MATH_PLUGINS_DIRECTORY "${LIB_DESTINATION}/indi/MathPlugins/" CACHE PATH "Path to Math Plugins directory")
 
 set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS INDI_MATH_PLUGINS_DIRECTORY="${INDI_MATH_PLUGINS_DIRECTORY}")
 
