diff -up PyQt-x11-gpl-4.9/sip/QtGui/qabstractitemview.sip.doItemsLayout PyQt-x11-gpl-4.9/sip/QtGui/qabstractitemview.sip
--- PyQt-x11-gpl-4.9/sip/QtGui/qabstractitemview.sip.doItemsLayout	2011-12-22 03:22:56.000000000 -0600
+++ PyQt-x11-gpl-4.9/sip/QtGui/qabstractitemview.sip	2012-01-09 08:14:27.766433278 -0600
@@ -160,7 +160,6 @@ public:
 public slots:
     virtual void reset();
     virtual void setRootIndex(const QModelIndex &index);
-    virtual void doItemsLayout();
     virtual void selectAll();
     void edit(const QModelIndex &index);
     void clearSelection();
diff -up PyQt-x11-gpl-4.9/sip/QtGui/qheaderview.sip.doItemsLayout PyQt-x11-gpl-4.9/sip/QtGui/qheaderview.sip
--- PyQt-x11-gpl-4.9/sip/QtGui/qheaderview.sip.doItemsLayout	2011-12-22 03:22:56.000000000 -0600
+++ PyQt-x11-gpl-4.9/sip/QtGui/qheaderview.sip	2012-01-09 08:14:36.467324502 -0600
@@ -82,7 +82,6 @@ public:
     Qt::SortOrder sortIndicatorOrder() const;
     bool stretchLastSection() const;
     void setStretchLastSection(bool stretch);
-    virtual void doItemsLayout();
     bool sectionsMoved() const;
 
 public slots:
diff -up PyQt-x11-gpl-4.9/sip/QtGui/qlistview.sip.doItemsLayout PyQt-x11-gpl-4.9/sip/QtGui/qlistview.sip
--- PyQt-x11-gpl-4.9/sip/QtGui/qlistview.sip.doItemsLayout	2011-12-22 03:22:56.000000000 -0600
+++ PyQt-x11-gpl-4.9/sip/QtGui/qlistview.sip	2012-01-09 08:12:34.066854718 -0600
@@ -91,7 +91,6 @@ public:
     virtual QRect visualRect(const QModelIndex &index) const;
     virtual void scrollTo(const QModelIndex &index, QAbstractItemView::ScrollHint hint = QAbstractItemView::EnsureVisible);
     virtual QModelIndex indexAt(const QPoint &p) const;
-    virtual void doItemsLayout();
     virtual void reset();
     virtual void setRootIndex(const QModelIndex &index);
 
diff -up PyQt-x11-gpl-4.9/sip/QtGui/qtreeview.sip.doItemsLayout PyQt-x11-gpl-4.9/sip/QtGui/qtreeview.sip
--- PyQt-x11-gpl-4.9/sip/QtGui/qtreeview.sip.doItemsLayout	2011-12-22 03:22:57.000000000 -0600
+++ PyQt-x11-gpl-4.9/sip/QtGui/qtreeview.sip	2012-01-09 08:14:43.661234565 -0600
@@ -62,7 +62,6 @@ public:
     virtual QModelIndex indexAt(const QPoint &p) const;
     QModelIndex indexAbove(const QModelIndex &index) const;
     QModelIndex indexBelow(const QModelIndex &index) const;
-    virtual void doItemsLayout();
     virtual void reset();
 
 signals:
