diff -up PyQt-x11-gpl-4.9.4/configure.py.webkit PyQt-x11-gpl-4.9.4/configure.py
--- PyQt-x11-gpl-4.9.4/configure.py.webkit	2012-07-30 17:35:01.000000000 +0200
+++ PyQt-x11-gpl-4.9.4/configure.py	2012-07-30 17:41:03.768738467 +0200
@@ -377,7 +377,6 @@ class ConfigurePyQt4:
                 extra_libs=sql_libs)
         if qt_version < 0x050000: check_module("QtSvg", "qsvgwidget.h", "new QSvgWidget()")
         if qt_version < 0x050000: check_module("QtTest", "QtTest", "QTest::qSleep(0)")
-        if qt_version < 0x050000: check_module("QtWebKit", "qwebpage.h", "new QWebPage()")
         if qt_version < 0x050000: check_module("QtXml", "qdom.h", "new QDomDocument()")
         if qt_version < 0x050000: check_module("QtXmlPatterns", "qxmlname.h", "new QXmlName()")
         if qt_version < 0x050000: check_module("phonon", "phonon/videowidget.h",
