From 1f11faf7b505ac94bee1242e90dbd3ae37a086b4 Mon Sep 17 00:00:00 2001
From: Connor Lane Smith <cls@lubutu.com>
Date: Fri, 10 Feb 2012 00:37:42 +0000
Subject: [PATCH 18/28] listen for C-S-[jm]

---
 dmenu.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dmenu.c b/dmenu.c
index fad4443..f7eba87 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -254,8 +254,10 @@ keypress(XKeyEvent *ev) {
 		case XK_g: ksym = XK_Escape;    break;
 		case XK_h: ksym = XK_BackSpace; break;
 		case XK_i: ksym = XK_Tab;       break;
-		case XK_j: ksym = XK_Return;    break;
-		case XK_m: ksym = XK_Return;    break;
+		case XK_j: /* fallthrough */
+		case XK_J: ksym = XK_Return;    break;
+		case XK_m: /* fallthrough */
+		case XK_M: ksym = XK_Return;    break;
 		case XK_n: ksym = XK_Down;      break;
 		case XK_p: ksym = XK_Up;        break;
 
-- 
1.9.0

