diff -up lksctp-tools-1.0.11/src/withsctp/withsctp.in.orig lksctp-tools-1.0.11/src/withsctp/withsctp.in
--- lksctp-tools-1.0.11/src/withsctp/withsctp.in.orig	2009-10-24 02:49:15.000000000 +0200
+++ lksctp-tools-1.0.11/src/withsctp/withsctp.in	2009-12-01 13:46:30.000000000 +0100
@@ -1,6 +1,6 @@
 #!/bin/sh
 # -*- sh -*-
-LIBDIR=@libdir@/@PACKAGE@
+LIBDIR=`rpm --eval "%{_libdir}"`/@PACKAGE@
 BINDIR=@bindir@
 export LD_PRELOAD=${LIBDIR}/libwithsctp.so.1.0.11
 if ! ${BINDIR}/checksctp 2> /dev/null
