diff -up wicd-1.7.2.4/curses/netentry_curses.py.unicode wicd-1.7.2.4/curses/netentry_curses.py
--- wicd-1.7.2.4/curses/netentry_curses.py.unicode	2012-02-11 15:35:45.000000000 -0500
+++ wicd-1.7.2.4/curses/netentry_curses.py	2012-10-23 14:06:36.537948070 -0400
@@ -264,7 +264,7 @@ class WiredSettingsDialog(AdvancedSettin
         self.set_values()
         
         self.prof_name = name
-        title = _('Configuring preferences for wired profile "$A"').replace('$A',self.prof_name)
+        title = ''.join([_('Configuring preferences for wired profile'), self.prog_name])
         self._w.header = urwid.Text( ('header',title),align='right' )
 
         self.set_values()
@@ -386,7 +386,7 @@ class WirelessSettingsDialog(AdvancedSet
         self.encrypt_types = misc.LoadEncryptionMethods()
         self.set_values()
 
-        title = _('Configuring preferences for wireless network "$A" ($B)').replace('$A',wireless.GetWirelessProperty(networkID,'essid')).replace('$B',wireless.GetWirelessProperty(networkID,'bssid'))
+        title = ''.join([_('Configuring preferences for wireless network'), wireless.GetWirelessProperty(networkID,'essid'), "(%s)" % wireless.GetWirelessProperty(networkID,'bssid')])
         self._w.header = urwid.Text(('header',title),align='right' )
 
     def set_values(self):
