diff -up wicd-1.7.1b2/cli/wicd-cli.py.orig wicd-1.7.1b2/cli/wicd-cli.py
--- wicd-1.7.1b2/cli/wicd-cli.py.orig	2010-10-29 11:36:55.000000000 -0400
+++ wicd-1.7.1b2/cli/wicd-cli.py	2011-08-19 17:10:38.582160338 -0400
@@ -191,6 +191,9 @@ if options.connect:
 
 		check = lambda: wired.CheckIfWiredConnecting()
 		message = lambda: wired.CheckWiredConnectingMessage()
+	else:
+		check = lambda: False
+		message = lambda: False
 
 	# update user on what the daemon is doing
 	last = None
