diff -up ./dummyserver/handlers.py.orig ./dummyserver/handlers.py
--- ./dummyserver/handlers.py.orig	2014-04-21 01:19:35.209031172 -0700
+++ ./dummyserver/handlers.py	2014-04-21 01:19:54.911126953 -0700
@@ -190,7 +190,7 @@ def _parse_header(line):
     """
     import tornado.httputil
     import email.utils
-    from urllib3.packages import six
+    import six
     if not six.PY3:
         line = line.encode('utf-8')
     parts = tornado.httputil._parseparam(';' + line)
diff -up ./setup.py.orig ./setup.py
--- ./setup.py.orig	2014-04-21 00:58:19.713830394 -0700
+++ ./setup.py	2014-04-21 01:30:40.966267728 -0700
@@ -45,7 +45,6 @@ setup(name='urllib3',
       url='http://urllib3.readthedocs.org/',
       license='MIT',
       packages=['urllib3',
-                'urllib3.packages', 'urllib3.packages.ssl_match_hostname',
                 'urllib3.contrib', 'urllib3.util',
                 ],
       requires=requirements,
diff -up ./test-requirements.txt.orig ./test-requirements.txt
--- ./test-requirements.txt.orig	2014-04-21 01:21:44.452659485 -0700
+++ ./test-requirements.txt	2014-04-21 01:21:59.317731751 -0700
@@ -2,3 +2,5 @@ nose==1.3
 mock==1.0.1
 tornado==3.1.1
 coverage==3.6
+six
+backports.ssl_match_hostname
diff -up ./test/test_collections.py.orig ./test/test_collections.py
--- ./test/test_collections.py.orig	2014-04-21 00:58:40.569931786 -0700
+++ ./test/test_collections.py	2014-04-21 01:18:59.041855346 -0700
@@ -4,7 +4,7 @@ from urllib3._collections import (
     HTTPHeaderDict,
     RecentlyUsedContainer as Container
 )
-from urllib3.packages import six
+import six
 xrange = six.moves.xrange
 
 
diff -up ./test/test_connectionpool.py.orig ./test/test_connectionpool.py
--- ./test/test_connectionpool.py.orig	2014-04-21 00:58:48.097968383 -0700
+++ ./test/test_connectionpool.py	2014-04-21 01:18:48.335803299 -0700
@@ -6,7 +6,14 @@ from urllib3.connectionpool import (
     HTTPConnectionPool,
 )
 from urllib3.util import Timeout
-from urllib3.packages.ssl_match_hostname import CertificateError
+try:
+    # python3.2+
+    from ssl import CertificateError
+except ImportError:
+    # Older python where the backport from pypi is installed
+    from backports.ssl_match_hostname import CertificateError
+
+
 from urllib3.exceptions import (
     ClosedPoolError,
     EmptyPoolError,
diff -up ./test/test_fields.py.orig ./test/test_fields.py
--- ./test/test_fields.py.orig	2014-04-21 00:58:57.501014095 -0700
+++ ./test/test_fields.py	2014-04-21 01:18:55.321837262 -0700
@@ -1,7 +1,7 @@
 import unittest
 
 from urllib3.fields import guess_content_type, RequestField
-from urllib3.packages.six import u
+from six import u
 
 
 class TestRequestField(unittest.TestCase):
diff -up ./test/test_filepost.py.orig ./test/test_filepost.py
--- ./test/test_filepost.py.orig	2014-04-21 00:59:05.128051174 -0700
+++ ./test/test_filepost.py	2014-04-21 01:18:52.414823129 -0700
@@ -2,7 +2,7 @@ import unittest
 
 from urllib3.filepost import encode_multipart_formdata, iter_fields
 from urllib3.fields import RequestField
-from urllib3.packages.six import b, u
+from six import b, u
 
 
 BOUNDARY = '!! test boundary !!'
diff -up ./urllib3/_collections.py.orig ./urllib3/_collections.py
--- ./urllib3/_collections.py.orig	2014-04-21 00:59:11.682083036 -0700
+++ ./urllib3/_collections.py	2014-04-21 01:18:57.203846411 -0700
@@ -19,8 +19,8 @@ except ImportError: # Platform-specific:
 try: # Python 2.7+
     from collections import OrderedDict
 except ImportError:
-    from .packages.ordered_dict import OrderedDict
-from .packages.six import itervalues
+    from ordered_dict import OrderedDict
+from six import itervalues
 
 
 __all__ = ['RecentlyUsedContainer', 'HTTPHeaderDict']
diff -up ./urllib3/connectionpool.py.orig ./urllib3/connectionpool.py
--- ./urllib3/connectionpool.py.orig	2014-04-21 00:59:20.406125448 -0700
+++ ./urllib3/connectionpool.py	2014-04-21 01:06:56.579343119 -0700
@@ -31,8 +31,15 @@ from .exceptions import (
     ReadTimeoutError,
     ProxyError,
 )
-from .packages.ssl_match_hostname import CertificateError
-from .packages import six
+try:
+    # python3.2+
+    from ssl import match_hostname, CertificateError
+except ImportError:
+    # Older python where the backport from pypi is installed
+    from backports.ssl_match_hostname import match_hostname, CertificateError
+import six
+
+
 from .connection import (
     port_by_scheme,
     DummyConnection,
diff -up ./urllib3/connection.py.orig ./urllib3/connection.py
--- ./urllib3/connection.py.orig	2014-04-21 01:14:17.899488582 -0700
+++ ./urllib3/connection.py	2014-04-21 01:16:00.490987327 -0700
@@ -38,8 +38,15 @@ except (ImportError, AttributeError): #
 from .exceptions import (
     ConnectTimeoutError,
 )
-from .packages.ssl_match_hostname import match_hostname
-from .packages import six
+try:
+    # python3.2+
+    from ssl import match_hostname, CertificateError
+except ImportError:
+    # Older python where the backport from pypi is installed
+    from backports.ssl_match_hostname import match_hostname, CertificateError
+
+import six
+
 from .util import (
     assert_fingerprint,
     resolve_cert_reqs,
diff -up ./urllib3/fields.py.orig ./urllib3/fields.py
--- ./urllib3/fields.py.orig	2014-04-21 00:59:31.661180164 -0700
+++ ./urllib3/fields.py	2014-04-21 01:09:04.310964082 -0700
@@ -7,7 +7,7 @@
 import email.utils
 import mimetypes
 
-from .packages import six
+import six
 
 
 def guess_content_type(filename, default='application/octet-stream'):
diff -up ./urllib3/filepost.py.orig ./urllib3/filepost.py
--- ./urllib3/filepost.py.orig	2014-04-21 00:59:39.538218457 -0700
+++ ./urllib3/filepost.py	2014-04-21 01:08:14.138720171 -0700
@@ -10,8 +10,8 @@ import mimetypes
 from uuid import uuid4
 from io import BytesIO
 
-from .packages import six
-from .packages.six import b
+import six
+from six import b
 from .fields import RequestField
 
 writer = codecs.lookup('utf-8')[3]
diff -up ./urllib3/response.py.orig ./urllib3/response.py
--- ./urllib3/response.py.orig	2014-04-21 00:59:47.622257758 -0700
+++ ./urllib3/response.py	2014-04-21 01:09:15.423018103 -0700
@@ -11,7 +11,7 @@ import io
 
 from ._collections import HTTPHeaderDict
 from .exceptions import DecodeError
-from .packages.six import string_types as basestring, binary_type
+from  six import string_types as basestring, binary_type
 from .util import is_fp_closed
 
 
diff -up ./urllib3/util/request.py.orig ./urllib3/util/request.py
--- ./urllib3/util/request.py.orig	2014-04-21 01:10:59.339523289 -0700
+++ ./urllib3/util/request.py	2014-04-21 01:12:02.663831138 -0700
@@ -1,6 +1,6 @@
 from base64 import b64encode
 
-from ..packages import six
+import six
 
 
 ACCEPT_ENCODING = 'gzip,deflate'
