diff --git a/src/formula.cpp b/src/formula.cpp
index ad37782..182ddaa 100644
--- a/src/formula.cpp
+++ b/src/formula.cpp
@@ -132,6 +132,7 @@ void FormulaList::generateBitmaps(const char *path)
       {
         err("Problems running dvips. Check your installation!\n");
         portable_sysTimerStop();
+        QDir::setCurrent(oldDir);
         return;
       }
       portable_sysTimerStop();
@@ -192,6 +193,7 @@ void FormulaList::generateBitmaps(const char *path)
       {
         err("Problem running ghostscript %s %s. Check your installation!\n",portable_ghostScriptCommand(),gsArgs);
         portable_sysTimerStop();
+        QDir::setCurrent(oldDir);
         return;
       }
       portable_sysTimerStop();
diff --git a/src/rtfgen.cpp b/src/rtfgen.cpp
index 7baaa3c..e10b638 100644
--- a/src/rtfgen.cpp
+++ b/src/rtfgen.cpp
@@ -2590,6 +2590,7 @@ bool RTFGenerator::preProcessFileInplace(const char *path,const char *name)
   if (!outf.open(IO_WriteOnly))
   {
     err("Failed to open %s for writing!\n",combinedName.data());
+    QDir::setCurrent(oldDir);
     return FALSE;
   }
   FTextStream outt(&outf);
