diff --git a/src/config.xml b/src/config.xml
index acbee8e..faad651 100644
--- a/src/config.xml
+++ b/src/config.xml
@@ -2647,6 +2647,16 @@ or
 ]]>
       </docs>
     </option>
+    <option type='bool' id='LATEX_TIMESTAMP' defval='0' depends='GENERATE_LATEX'>
+      <docs>
+<![CDATA[
+ If the \c LATEX_TIMESTAMP tag is set to \c YES then the footer of
+ each generated page will contain the date and time when the page
+ was generated. Setting this to \c NO can help when comparing the output of
+ multiple runs.
+]]>
+      </docs>
+    </option>
   </group>
   <group name='RTF' docs='Configuration options related to the RTF output'>
     <option type='bool' id='GENERATE_RTF' defval='0'>
diff --git a/src/latexgen.cpp b/src/latexgen.cpp
index 681b575..d8a3220 100644
--- a/src/latexgen.cpp
+++ b/src/latexgen.cpp
@@ -571,11 +571,18 @@ static void writeDefaultHeaderPart1(FTextStream &t)
 
   // Headers & footers
   QGString genString;
+  QCString generatedBy;
+  static bool timeStamp = Config_getBool("LATEX_TIMESTAMP");
   FTextStream tg(&genString);
-  filterLatexString(tg,
-                    theTranslator->trGeneratedAt(dateToString(TRUE),
-                       Config_getString("PROJECT_NAME")),
-                    FALSE,FALSE,FALSE);
+  if (timeStamp)
+  {
+    generatedBy = theTranslator->trGeneratedAt(dateToString(TRUE), Config_getString("PROJECT_NAME"));
+  }
+  else
+  {
+    generatedBy = theTranslator->trGeneratedBy();
+  }
+  filterLatexString(tg, generatedBy, FALSE,FALSE,FALSE);
   t << "% Headers & footers\n"
        "\\usepackage{fancyhdr}\n"
        "\\pagestyle{fancyplain}\n"
@@ -700,10 +707,11 @@ static void writeDefaultHeaderPart3(FTextStream &t)
 {
   // part 3
   // Finalize project number
-  t << " Doxygen " << versionString << "}\\\\\n"
-       "\\vspace*{0.5cm}\n"
-       "{\\small " << dateToString(TRUE) << "}\\\\\n"
-       "\\end{center}\n"
+  t << " Doxygen " << versionString << "}\\\\\n";
+  if (Config_getBool("LATEX_TIMESTAMP"))
+    t << "\\vspace*{0.5cm}\n"
+         "{\\small " << dateToString(TRUE) << "}\\\\\n";
+  t << "\\end{center}\n"
        "\\end{titlepage}\n";
   bool compactLatex = Config_getBool("COMPACT_LATEX");
   if (!compactLatex)
