Index: x11-xserver-utils-7.7+4/sessreg-1.1.0/man/Makefile.am
===================================================================
--- x11-xserver-utils-7.7+4.orig/sessreg-1.1.0/man/Makefile.am
+++ x11-xserver-utils-7.7+4/sessreg-1.1.0/man/Makefile.am
@@ -10,7 +10,7 @@ SUFFIXES = .$(APP_MAN_SUFFIX) .man
 AM_CPPFLAGS = -I$(top_buil%description -l zh_CN.UTF-8ir) -I$(top_srcdir)
 filenames.sed: filenames.sed.c
 	$(AM_V_GEN)$(CPP) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	    $(AM_CPPFLAGS) $(CPPFLAGS) $(srcdir)/filenames.sed.c | \
+	    $(AM_CPPFLAGS) $(CPPFLAGS) -P $(srcdir)/filenames.sed.c | \
 	    $(SED) -n -e '/s|__/ p' -e '/^\/__/ p' > $@
 
 # String replacements in MAN_SUBSTS now come from xorg-macros.m4 via configure
