diff -up ./src/CMakeLists.txt.configure ./src/CMakeLists.txt
--- ./src/CMakeLists.txt.configure	2010-01-12 18:22:13.135730380 +0000
+++ ./src/CMakeLists.txt	2010-01-12 18:22:59.590732057 +0000
@@ -34,6 +34,6 @@ install(TARGETS qedje LIBRARY DESTINATIO
 install(FILES ${qedje_HEADERS} DESTINATION include/)
 
 if (NOT WIN32)
-    configure_file(qedje.pc.cmake qedje.pc @ONLY)
-    install(FILES ${CMAKE_BINARY_DIR}/qedje.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig/)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/qedje.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/qedje.pc @ONLY)
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/qedje.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig/)
 endif (NOT WIN32)
