diff -up Tk-Text-SuperText-0.9.4/test.pl.test Tk-Text-SuperText-0.9.4/test.pl
--- Tk-Text-SuperText-0.9.4/test.pl.test	2009-06-10 10:23:11.000000000 +0200
+++ Tk-Text-SuperText-0.9.4/test.pl	2009-06-10 10:23:36.000000000 +0200
@@ -58,7 +58,7 @@ $text->focus;
 $text->bind('Tk::Text::SuperText','<<pippo>>',\&pippo);
 $text->eventAdd('<<pippo>>','<Control-p>','<Control-Key-1>');
 
-MainLoop;
+#MainLoop;
 
 print "ok 2\n";
 
