diff -ur libQtGTL-0.9.1.orig/QtShiva/KernelParametersWidget.cpp libQtGTL-0.9.1/QtShiva/KernelParametersWidget.cpp
--- libQtGTL-0.9.1.orig/QtShiva/KernelParametersWidget.cpp	2012-01-14 18:04:51.000000000 -0500
+++ libQtGTL-0.9.1/QtShiva/KernelParametersWidget.cpp	2012-01-14 18:05:31.000000000 -0500
@@ -23,7 +23,7 @@
 
 #include <GTLCore/Value.h>
 #include <OpenShiva/Kernel.h>
-#include <OpenShiva/Metadata.h>
+#include <GTLFragment/Metadata.h>
 
 using namespace QtShiva;
 
diff -ur libQtGTL-0.9.1.orig/QtShiva/SourceParametersWidget.cpp libQtGTL-0.9.1/QtShiva/SourceParametersWidget.cpp
--- libQtGTL-0.9.1.orig/QtShiva/SourceParametersWidget.cpp	2012-01-14 18:04:51.000000000 -0500
+++ libQtGTL-0.9.1/QtShiva/SourceParametersWidget.cpp	2012-01-14 18:05:56.000000000 -0500
@@ -25,7 +25,7 @@
 
 #include <GTLCore/Value.h>
 #include <OpenShiva/Source.h>
-#include <OpenShiva/Metadata.h>
+#include <GTLFragment/Metadata.h>
 
 class SourceParametersWidget::SourceParametrisation : public ParametersWidgetBase::Parametrisation {
   public:
