diff -ur libtpcmisc.orig/Makefile libtpcmisc/Makefile
--- libtpcmisc.orig/Makefile	2014-08-01 12:28:05.007079725 +0000
+++ libtpcmisc/Makefile	2014-08-01 12:31:31.761074426 +0000
@@ -62,9 +62,12 @@
                   libtpcmiscv.c
 OBJECTS 	= ${SOURCES:.c=.o}
 LIBARCHIVE	= ${LIBRARY}.a
+SHLIBSHORT     = ${LIBRARY}.so
+SHLIBMAJOR     = ${SHLIBSHORT}.0
+SHLIBARCHIVE   = ${SHLIBMAJOR}.0.0
 DOXYGEN		= doxygen
 CC		= gcc
-CFLAGS 		= -std=gnu99 -s -Wall -O3 ${INCLUDE} ${ANSI}
+CFLAGS 		= -std=gnu99 -s -Wall -O3 ${INCLUDE} ${ANSI} -fPIC -DPIC
 LIBS 		= ${LIBPATH} -ltpcmisc -L.
 ZIP		= zip
 PRVERS		= progvers -opt=-b
@@ -79,14 +82,20 @@
 LIBEXEC 	= ${LIBRARY}${EXT}
 
 # ------------- Compilation: --------------------------------------------------#
-compile: ${LIBARCHIVE} ${LIBEXEC}
+compile: ${LIBARCHIVE} ${SHLIBARCHIVE} ${LIBEXEC}
 
-# Compile library
+# Compile static library
 ${LIBARCHIVE}: ${OBJECTS}
 	${AR} -rvs ${LIBARCHIVE} ${OBJECTS}
 
+# Compile shared library
+${SHLIBARCHIVE}: ${OBJECTS}
+	${CC} -shared -o $@ ${OBJECTS} -Wl,-soname=${SHLIBMAJOR} -lm
+	ln -s ${SHLIBARCHIVE} ${SHLIBMAJOR}
+	ln -s ${SHLIBARCHIVE} ${SHLIBSHORT}
+
 # Compile executable
-${LIBEXEC}: ${LIBARCHIVE} ${LIBRARY}.c
+${LIBEXEC}: ${SHLIBARCHIVE} ${LIBRARY}.c
 	${CC} ${LIBRARY}.c -o ${LIBEXEC} ${LIBPATH} ${CFLAGS} ${LIBS}
 
 ifneq ($(OS),Windows_NT)
@@ -102,8 +111,9 @@
 # ------------- Installation: -------------------------------------------------#
 install: libversion libinstall execinstall
 
-libinstall: ${LIBARCHIVE}
+libinstall: ${LIBARCHIVE} ${SHAREDLIBARCHIVE}
 	${ADDDATE} -i=${LIBRARY}.version -o=${PET_LIBPATH} -nodate -copy ${LIBARCHIVE}
+	${ADDDATE} -i=${LIBRARY}.version -o=${PET_LIB} -nodate -copy ${SHLIBARCHIVE}
 
 execinstall: ${LIBEXEC}
 	${ADDDATE} -i=${LIBRARY}.version -o=${PET_BIN} -nodate -copy ${LIBEXEC}
