--- qterm-0.4.0pre3/qterm/qterm.cfg.orig	2005-09-01 14:58:19.000000000 +0000
+++ qterm-0.4.0pre3/qterm/qterm.cfg	2006-01-09 14:40:53.000000000 +0000
@@ -11,7 +11,7 @@
 keybar=1 2 1 0 0
 bbsbar=0 2 2 1 21
 mainbar=1 2 0 1 0
-language=eng
+language=chs
 clipcodec=0
 vscrollpos=0
 statusbar=1
@@ -23,7 +23,7 @@
 name0=post
 
 [quick list]
-num=1
+num=2
 
 [preference]
 xim=0
@@ -35,11 +35,15 @@
 blinktab=1
 warn=1
 beep=1
-http=mozilla
+http=konqueror
+image=gwenview
 antialias=0
 tray=0
 
 [quick 0]
-addr=debian
+addr=newsmth.net
 port=23
 
+[quick 1]
+addr=debian
+port=23
