diff -up /dev/null akode-2.0.2/akode/akode.pc.in
--- /dev/null	2007-12-13 13:41:14.965729569 -0600
+++ akode-2.0.2/akode/akode.pc.in	2007-12-23 21:32:59.000000000 -0600
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: akode 
+Description: aKode is a simple audio-decoding frame-work 
+Version: 2.0.2 
+Libs: -L${libdir} -lakode
+Cflags: -I${includedir}
+
diff -up akode-2.0.2/akode/configure.in.in.multilib akode-2.0.2/akode/configure.in.in
--- akode-2.0.2/akode/configure.in.in.multilib	2007-12-23 21:32:59.000000000 -0600
+++ akode-2.0.2/akode/configure.in.in	2007-12-23 21:32:59.000000000 -0600
@@ -1,5 +1,6 @@
 dnl don't remove the below
 dnl AC_OUTPUT(akode/akode-config)
+dnl AC_OUTPUT(akode/akode.pc)
 
 AM_CONFIG_HEADER(akode/lib/akode_export.h)
 
diff -up akode-2.0.2/akode/akode-config.in.multilib akode-2.0.2/akode/akode-config.in
--- akode-2.0.2/akode/akode-config.in.multilib	2007-04-06 08:16:47.000000000 -0500
+++ akode-2.0.2/akode/akode-config.in	2007-12-23 21:37:40.000000000 -0600
@@ -15,9 +15,8 @@ EOH
 }
 
 prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
+akode_libs="`pkg-config --libs akode`"
+akode_cflags="`pkg-config --cflags akode`"
 
 flags=""
 
@@ -29,10 +28,10 @@ while test $# -gt 0
 do
   case $1 in
     --libs)
-	  flags="$flags -L$libdir -lakode"
+	  flags="$flags $akode_libs"
 	  ;;
     --cflags)
-	  flags="$flags -I$includedir"
+	  flags="$flags $akode_cflags"
 	  ;;
     --version)
 	  echo 2.0.2
diff -up akode-2.0.2/akode/Makefile.am.multilib akode-2.0.2/akode/Makefile.am
--- akode-2.0.2/akode/Makefile.am.multilib	2005-07-14 09:27:18.000000000 -0500
+++ akode-2.0.2/akode/Makefile.am	2007-12-23 21:32:59.000000000 -0600
@@ -1,3 +1,8 @@
 bin_SCRIPTS = akode-config
 
 SUBDIRS= lib plugins akodeplay
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = akode.pc
+
+EXTRA_DIST = akode.pc.in
