diff -urNp tla-1.3.5.orig/src/tla/libarch/invent.c tla-1.3.5/src/tla/libarch/invent.c
--- tla-1.3.5.orig/src/tla/libarch/invent.c	2006-07-20 12:04:40.000000000 +0530
+++ tla-1.3.5/src/tla/libarch/invent.c	2007-12-13 23:18:49.000000000 +0530
@@ -16,6 +16,7 @@
 #include "hackerlab/arrays/ar.h"
 #include "hackerlab/fs/file-names.h"
 #include "hackerlab/vu/safe.h"
+#include "tla/libarch/cmdutils.h"
 #include "tla/libarch/inode-sig.h"
 #include "tla/libarch/inv-ids.h"
 #include "tla/libarch/invent.h"
