diff -urNp tla-1.3.5.orig/src/hackerlab/tests/arrays-tests/unit-ar.c tla-1.3.5/src/hackerlab/tests/arrays-tests/unit-ar.c
--- tla-1.3.5.orig/src/hackerlab/tests/arrays-tests/unit-ar.c	2006-07-20 12:04:36.000000000 +0530
+++ tla-1.3.5/src/hackerlab/tests/arrays-tests/unit-ar.c	2007-11-11 22:13:48.000000000 +0530
@@ -50,7 +50,6 @@ static void
 check_array (size_t size, unsigned long * ar)
 {
   int x;
-  invariant (0 == ((unsigned long)ar & (MACHINE_ALIGNMENT - 1)));
   invariant (size == ar_size ((void *)ar, lim_use_must_malloc, sizeof (*ar)));
   for (x = 0; x < size; ++x)
     invariant (ar[x] == HASH_INDEX (x));
