diff -Nuri libxfcegui4-4.10.0-orig/libxfcegui4/xfce-exec.c libxfcegui4-4.10.0/libxfcegui4/xfce-exec.c
--- libxfcegui4-4.10.0-orig/libxfcegui4/xfce-exec.c	2014-11-25 21:16:11.933756840 +0800
+++ libxfcegui4-4.10.0/libxfcegui4/xfce-exec.c	2014-11-25 21:16:49.645690211 +0800
@@ -79,7 +79,7 @@
 #ifndef G_OS_WIN32
     gchar *dpyname = user_data;
     
-    xfce_setenv ("DISPLAY", dpyname, TRUE);
+    g_setenv ("DISPLAY", dpyname, TRUE);
 #endif
 }
 
