From 0d63e935edf151dd936b5cf43948a1e0a996d42b Mon Sep 17 00:00:00 2001
From: dgod <dgod.osa@gmail.com>
Date: Fri, 24 Feb 2012 22:47:36 +0800
Subject: [PATCH] lxtask can be closed by pressing the Escape Or Ctrl+W now

---
 src/interface.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/src/interface.c b/src/interface.c
index 62e89ff..45cf256 100644
--- a/src/interface.c
+++ b/src/interface.c
@@ -66,11 +66,25 @@ GtkWidget* create_main_window (void)
 
 	/* build menu */
 	menu = gtk_menu_new();
+	
 	item = gtk_menu_item_new_with_mnemonic( _("_File") );
 	gtk_menu_item_set_submenu(GTK_MENU_ITEM (item), menu );
 	gtk_menu_shell_append( (GtkMenuShell*)menubar, item );
 
 	item = gtk_image_menu_item_new_from_stock( GTK_STOCK_QUIT, NULL );
+	GtkAccelGroup* accel_group = gtk_accel_group_new();
+	gtk_window_add_accel_group(GTK_WINDOW(window), accel_group);
+#if GTK_CHECK_VERSION(3,0,0)
+	gtk_widget_add_accelerator(item, "activate", accel_group, 
+		GDK_KEY_Escape, (GdkModifierType)0, GTK_ACCEL_VISIBLE);
+	gtk_widget_add_accelerator(item, "activate", accel_group, 
+		GDK_KEY_W, GDK_CONTROL_MASK, GTK_ACCEL_VISIBLE);
+#else
+	gtk_widget_add_accelerator(item, "activate", accel_group, 
+		GDK_Escape, (GdkModifierType)NULL, GTK_ACCEL_VISIBLE);
+	gtk_widget_add_accelerator(item, "activate", accel_group, 
+		GDK_W, GDK_CONTROL_MASK, GTK_ACCEL_VISIBLE);
+#endif
 	gtk_menu_shell_append( (GtkMenuShell*)menu, item );
 	g_signal_connect( item, "activate", G_CALLBACK(gtk_main_quit), NULL );
 
-- 
1.7.4.1

