diff -ur physfs-2.0.2.orig/test/wxtest_physfs.cpp physfs-2.0.2/test/wxtest_physfs.cpp
--- physfs-2.0.2.orig/test/wxtest_physfs.cpp	2012-01-26 14:01:45.000000000 -0500
+++ physfs-2.0.2/test/wxtest_physfs.cpp	2012-01-26 14:02:07.000000000 -0500
@@ -415,7 +415,7 @@
 
 void WxTestPhysfsFrame::onMenuInit(wxCommandEvent &evt)
 {
-    wxString argv0(wxGetApp().argv[0] == NULL ? wxT("") : wxGetApp().argv[0]);
+    wxString argv0((const char*)wxGetApp().argv[0] == NULL ? wxT("") : wxGetApp().argv[0]);
     wxString str(wxGetTextFromUser(wxT("PHYSFS_init"),
                  wxT("argv[0]? (cancel for NULL)"), argv0));
     char *cstr = str.IsEmpty() ? NULL : newutf8(str);
