diff -ur ../xmedcon-0.10.7.orig/configure ./configure
--- ../xmedcon-0.10.7.orig/configure	2010-09-15 03:06:18.000000000 +0530
+++ ./configure	2011-06-18 10:02:10.656245908 +0530
@@ -8437,7 +8437,7 @@
 
 
   lt_prog_compiler_wl=
-lt_prog_compiler_pic=
+lt_prog_compiler_pic='-fPIC'
 lt_prog_compiler_static=
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $compiler option to produce PIC" >&5
@@ -12177,7 +12177,7 @@
 
 if test x"$tpc_prefix" != x; then
     TPC_LDFLAGS="-L$tpc_prefix/lib -ltpcimgio -L$tpc_prefix/lib -ltpcmisc"
-    TPC_CFLAGS="-I$tpc_prefix/include"
+    TPC_CFLAGS="-I$tpc_prefix/include/libtpcimgio  -I$tpc_prefix/include/libtpcmisc"
     LDFLAGS="$LDFLAGS $TPC_LDFLAGS"
     CPPFLAGS="$CPPFLAGS $TPC_CFLAGS"
 else
@@ -13296,7 +13296,7 @@
 fi
 
 
-ac_config_files="$ac_config_files Makefile xmedcon-config libs/Makefile libs/ljpg/Makefile libs/dicom/Makefile libs/nifti/Makefile libs/tpc/Makefile macros/Makefile source/Makefile source/m-config.h etc/Makefile etc/xmedcon.spec etc/xmedcon-$VERSION-1.iss:etc/xmedcon.iss.in etc/xmedcon-$VERSION-1.info:etc/xmedcon.info.in etc/xmedcon-$VERSION.ebuild:etc/xmedcon.ebuild.in man/Makefile"
+ac_config_files="$ac_config_files Makefile xmedcon-config libs/Makefile libs/ljpg/Makefile libs/dicom/Makefile macros/Makefile source/Makefile source/m-config.h etc/Makefile etc/xmedcon.spec etc/xmedcon-$VERSION-1.iss:etc/xmedcon.iss.in etc/xmedcon-$VERSION-1.info:etc/xmedcon.info.in etc/xmedcon-$VERSION.ebuild:etc/xmedcon.ebuild.in man/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -14337,8 +14337,6 @@
     "libs/Makefile") CONFIG_FILES="$CONFIG_FILES libs/Makefile" ;;
     "libs/ljpg/Makefile") CONFIG_FILES="$CONFIG_FILES libs/ljpg/Makefile" ;;
     "libs/dicom/Makefile") CONFIG_FILES="$CONFIG_FILES libs/dicom/Makefile" ;;
-    "libs/nifti/Makefile") CONFIG_FILES="$CONFIG_FILES libs/nifti/Makefile" ;;
-    "libs/tpc/Makefile") CONFIG_FILES="$CONFIG_FILES libs/tpc/Makefile" ;;
     "macros/Makefile") CONFIG_FILES="$CONFIG_FILES macros/Makefile" ;;
     "source/Makefile") CONFIG_FILES="$CONFIG_FILES source/Makefile" ;;
     "source/m-config.h") CONFIG_FILES="$CONFIG_FILES source/m-config.h" ;;
