 CHANGELOG   |    5 +++++
 dmidecode.c |    2 +-
 2 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/CHANGELOG b/CHANGELOG
index 0545f8a..de3d7e8 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,8 @@
+2011-01-25  Jean Delvare  <khali@linux-fr.org>
+
+	* dmidecode.c: Fix boundary checks of memory array location codes
+	  (DMI type 16). Reported by Andrey Matveyev.
+
 2010-11-24  Jean Delvare  <khali@linux-fr.org>
 
 	* dmidecode.c: Assume that the max power capacity is expressed in
diff --git a/dmidecode.c b/dmidecode.c
index 5c7ad0e..7b081f9 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -2100,7 +2100,7 @@ static const char *dmi_memory_array_location(u8 code)
 
 	if (code >= 0x01 && code <= 0x0A)
 		return location[code - 0x01];
-	if (code >= 0xA0 && code <= 0xA4)
+	if (code >= 0xA0 && code <= 0xA3)
 		return location_0xA0[code - 0xA0];
 	return out_of_spec;
 }
