diff -up dmidecode-2.11/AUTHORS.dmioem_hp_fix dmidecode-2.11/AUTHORS
--- dmidecode-2.11/AUTHORS.dmioem_hp_fix	2012-03-12 10:26:58.166021817 +0100
+++ dmidecode-2.11/AUTHORS	2012-03-12 10:27:18.818022115 +0100
@@ -83,3 +83,4 @@ Lars Mueller
 Thomas Mingarelli
 Andrey Matveyev
 Stefan Tauner
+Naga Chumbalkar
diff -up dmidecode-2.11/CHANGELOG.dmioem_hp_fix dmidecode-2.11/CHANGELOG
--- dmidecode-2.11/CHANGELOG.dmioem_hp_fix	2012-03-12 10:27:05.076021906 +0100
+++ dmidecode-2.11/CHANGELOG	2012-03-12 10:27:18.818022115 +0100
@@ -1,3 +1,8 @@
+2012-03-12  Anton Arapov  <anton@redhat.com>
+
+	* dmioem.c: Add "PXE" to the HP OEM Type 209 output so it is similar to
+	  the "iSCSI" description in HP OEM Type 221. Patch from Naga Chumbalkar.
+
 2011-11-18  Jean Delvare  <khali@linux-fr.org>
 
 	* dmidecode.c: Properly print the hexadecimal value of invalid
diff -up dmidecode-2.11/dmioem.c.dmioem_hp_fix dmidecode-2.11/dmioem.c
--- dmidecode-2.11/dmioem.c.dmioem_hp_fix	2012-03-12 10:27:13.108022027 +0100
+++ dmidecode-2.11/dmioem.c	2012-03-12 10:27:18.819022112 +0100
@@ -83,7 +83,7 @@ static int dmi_decode_hp(const struct dm
 			 */
 			printf(h->type == 221 ?
 				"HP BIOS iSCSI NIC PCI and MAC Information\n" :
-				"HP BIOS NIC PCI and MAC Information\n");
+				"HP BIOS PXE NIC PCI and MAC Information\n");
 			nic = 1;
 			ptr = 4;
 			while (h->length >= ptr + 8)
