*** ../binutils-2.24.orig/opcodes/s390-mkopc.c	2013-12-05 13:06:00.082812690 +0000
--- opcodes/s390-mkopc.c	2013-12-05 13:13:40.206798138 +0000
*************** dumpTable (void)
*** 298,304 ****
    int  ix;
  
    /*  Write hash table entries (slots).  */
!   printf (file_header);
  
    for (ix = 0; ix < no_ops; ix++)
      {
--- 298,304 ----
    int  ix;
  
    /*  Write hash table entries (slots).  */
!   printf ("%s", file_header);
  
    for (ix = 0; ix < no_ops; ix++)
      {
