diff -Nur garcon-0.1.9.orig/data/xfce/Makefile.am garcon-0.1.9/data/xfce/Makefile.am
--- garcon-0.1.9.orig/data/xfce/Makefile.am	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/Makefile.am	2011-10-22 20:59:51.692343459 +0200
@@ -23,18 +23,8 @@
 
 directorydir = $(datadir)/desktop-directories
 directory_in_files =							\
-	xfce-accessories.directory.in					\
-	xfce-development.directory.in					\
-	xfce-education.directory.in					\
-	xfce-games.directory.in						\
-	xfce-graphics.directory.in					\
-	xfce-multimedia.directory.in					\
-	xfce-network.directory.in					\
-	xfce-office.directory.in					\
 	xfce-other.directory.in						\
-	xfce-screensavers.directory.in					\
-	xfce-settings.directory.in					\
-	xfce-system.directory.in
+	xfce-settings.directory.in
 
 directory_DATA = $(directory_in_files:.directory.in=.directory)
 @INTLTOOL_DIRECTORY_RULE@
diff -Nur garcon-0.1.9.orig/data/xfce/Makefile.in garcon-0.1.9/data/xfce/Makefile.in
--- garcon-0.1.9.orig/data/xfce/Makefile.in	2011-09-12 20:24:26.000000000 +0200
+++ garcon-0.1.9/data/xfce/Makefile.in	2011-10-22 20:59:51.693343452 +0200
@@ -281,18 +281,8 @@
 
 directorydir = $(datadir)/desktop-directories
 directory_in_files = \
-	xfce-accessories.directory.in					\
-	xfce-development.directory.in					\
-	xfce-education.directory.in					\
-	xfce-games.directory.in						\
-	xfce-graphics.directory.in					\
-	xfce-multimedia.directory.in					\
-	xfce-network.directory.in					\
-	xfce-office.directory.in					\
 	xfce-other.directory.in						\
-	xfce-screensavers.directory.in					\
-	xfce-settings.directory.in					\
-	xfce-system.directory.in
+	xfce-settings.directory.in
 
 directory_DATA = $(directory_in_files:.directory.in=.directory)
 EXTRA_DIST = \
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-accessories.directory.in garcon-0.1.9/data/xfce/xfce-accessories.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-accessories.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-accessories.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-accessories
-_Name=Accessories
-_Comment=Common desktop tools and applications
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-applications.menu garcon-0.1.9/data/xfce/xfce-applications.menu
--- garcon-0.1.9.orig/data/xfce/xfce-applications.menu	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-applications.menu	2011-10-22 20:59:51.700343406 +0200
@@ -21,20 +21,25 @@
         <Filename>exo-mail-reader.desktop</Filename>
         <Filename>exo-web-browser.desktop</Filename>
         <Separator/>
-        <Menuname>Settings</Menuname>
+        <Menuname>Preferences</Menuname>
+        <Menuname>Administration</Menuname>        
         <Separator/>
         <Merge type="all"/>
         <Separator/>
-        <Filename>xfhelp4.desktop</Filename>
-        <Filename>xfce4-about.desktop</Filename>
         <Filename>xfce4-session-logout.desktop</Filename>
     </Layout>
 
     <Menu>
-        <Name>Settings</Name>
+        <Name>Preferences</Name>
         <Directory>xfce-settings.directory</Directory>
         <Include>
-            <Category>Settings</Category>
+            <And>
+                <Category>Settings</Category>
+                <Not><Category>System</Category></Not>
+                <Not><Category>Screensaver</Category></Not>
+                <Not><Filename>fedora-im-chooser.desktop</Filename></Not>
+                <Not><Filename>gnome-default-applications.desktop</Filename></Not>
+            </And>
         </Include>
 
         <Layout>
@@ -43,25 +48,31 @@
             <Merge type="all"/>
         </Layout>
 
-        <Menu>
-            <Name>Screensavers</Name>
-            <Directory>xfce-screensavers.directory</Directory>
-            <Include>
-                <Category>Screensaver</Category>
-            </Include>
-        </Menu>
+    </Menu>
+
+    <Menu>
+        <Name>Administration</Name>
+        <Directory>SystemConfig.directory</Directory>
+        <Include>
+            <And>
+                <Category>Settings</Category>
+                <Category>System</Category>
+                <Not><Category>Screensaver</Category></Not>
+            </And>
+            <Filename>gpk-application.desktop</Filename>
+            <Filename>gpk-update-viewer.desktop</Filename>
+        </Include>
+        
     </Menu>
 
     <Menu>
         <Name>Accessories</Name>
-        <Directory>xfce-accessories.directory</Directory>
+        <Directory>Utility.directory</Directory>
         <Include>
-            <Or>
-                <Category>Accessibility</Category>
-                <Category>Core</Category>
-                <Category>Legacy</Category>
+            <And>
                 <Category>Utility</Category>
-            </Or>
+                <Not><Category>System</Category></Not>
+            </And>
         </Include>
         <Exclude>
             <Or>
@@ -75,23 +86,34 @@
 
     <Menu>
         <Name>Development</Name>
-        <Directory>xfce-development.directory</Directory>
+        <Directory>Development.directory</Directory>
         <Include>
             <Category>Development</Category>
         </Include>
     </Menu>
 
     <Menu>
+        <Name>Documentation</Name>
+        <Directory>Documentation.directory</Directory>
+        <Include>
+            <Category>Documentation</Category>
+        </Include>
+    </Menu>
+        
+    <Menu>
         <Name>Education</Name>
-        <Directory>xfce-education.directory</Directory>
+        <Directory>Education.directory</Directory>
         <Include>
+            <Or>
             <Category>Education</Category>
+            <Category>Science</Category>
+        </Or>
         </Include>
     </Menu>
 
     <Menu>
         <Name>Games</Name>
-        <Directory>xfce-games.directory</Directory>
+        <Directory>Game.directory</Directory>
         <Include>
             <Category>Game</Category>
         </Include>
@@ -99,7 +121,7 @@
 
     <Menu>
         <Name>Graphics</Name>
-        <Directory>xfce-graphics.directory</Directory>
+        <Directory>Graphics.directory</Directory>
         <Include>
             <Category>Graphics</Category>
         </Include>
@@ -107,7 +129,7 @@
 
     <Menu>
         <Name>Multimedia</Name>
-        <Directory>xfce-multimedia.directory</Directory>
+        <Directory>AudioVideo.directory</Directory>
         <Include>
             <Category>Audio</Category>
             <Category>Video</Category>
@@ -117,7 +139,7 @@
 
     <Menu>
         <Name>Network</Name>
-        <Directory>xfce-network.directory</Directory>
+        <Directory>Network.directory</Directory>
         <Include>
             <Category>Network</Category>
         </Include>
@@ -131,7 +153,7 @@
 
     <Menu>
         <Name>Office</Name>
-        <Directory>xfce-office.directory</Directory>
+        <Directory>Office.directory</Directory>
         <Include>
             <Category>Office</Category>
         </Include>
@@ -139,12 +161,18 @@
 
     <Menu>
         <Name>System</Name>
-        <Directory>xfce-system.directory</Directory>
+        <Directory>System.directory</Directory>
         <Include>
-            <Or>
-                <Category>Emulator</Category>
-                <Category>System</Category>
-            </Or>
+            <And>
+                <Or>
+                    <Category>Emulator</Category>
+                    <Category>System</Category>
+                </Or>
+                <Not><Category>Settings</Category></Not>
+                <Not><Category>Screensaver</Category></Not>
+                <Not><Filename>gpk-application.desktop</Filename></Not>
+                <Not><Filename>gpk-update-viewer.desktop</Filename></Not>
+            </And>
         </Include>
         <Exclude>
             <Or>
@@ -158,7 +186,13 @@
         <Directory>xfce-other.directory</Directory>
         <OnlyUnallocated/>
         <Include>
-            <All/>
+            <And>
+                <Not><Category>Core</Category></Not>
+                <Not><Category>Settings</Category></Not>
+                <Not><Category>SystemSetup</Category></Not>
+                <Not><Category>X-Red-Hat-ServerConfig</Category></Not>
+                <Not><Category>Screensaver</Category></Not>
+            </And>
         </Include>
     </Menu>
 
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-development.directory.in garcon-0.1.9/data/xfce/xfce-development.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-development.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-development.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-development
-_Name=Development
-_Comment=Software development tools
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-education.directory.in garcon-0.1.9/data/xfce/xfce-education.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-education.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-education.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-education
-_Name=Education
-_Comment=Educational software
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-games.directory.in garcon-0.1.9/data/xfce/xfce-games.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-games.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-games.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-games
-_Name=Games
-_Comment=Games, puzzles, and other fun software
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-graphics.directory.in garcon-0.1.9/data/xfce/xfce-graphics.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-graphics.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-graphics.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-graphics
-_Name=Graphics
-_Comment=Graphics creation and manipulation applications
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-multimedia.directory.in garcon-0.1.9/data/xfce/xfce-multimedia.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-multimedia.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-multimedia.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-multimedia
-_Name=Multimedia
-_Comment=Audio and video players and editors
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-network.directory.in garcon-0.1.9/data/xfce/xfce-network.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-network.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-network.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-internet
-_Name=Internet
-_Comment=Applications for Internet access
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-office.directory.in garcon-0.1.9/data/xfce/xfce-office.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-office.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-office.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-office
-_Name=Office
-_Comment=Office and productivity applications
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-screensavers.directory.in garcon-0.1.9/data/xfce/xfce-screensavers.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-screensavers.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-screensavers.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=preferences-desktop-screensaver
-_Name=Screensavers
-_Comment=Screensaver applets
diff -Nur garcon-0.1.9.orig/data/xfce/xfce-system.directory.in garcon-0.1.9/data/xfce/xfce-system.directory.in
--- garcon-0.1.9.orig/data/xfce/xfce-system.directory.in	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9/data/xfce/xfce-system.directory.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[Desktop Entry]
-Version=1.0
-Type=Directory
-Icon=applications-system
-_Name=System
-_Comment=System tools and utilities
