--- liblo-0.28/doc/reference.doxygen.in	2014-01-25 13:44:13.000000000 +0100
+++ liblo-0.28/doc/reference.doxygen.in.mod	2015-07-09 23:07:52.261825418 +0200
@@ -1110,7 +1110,7 @@ HTML_COLORSTYLE_GAMMA  = 80
 # The default value is: YES.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
-HTML_TIMESTAMP         = YES
+HTML_TIMESTAMP         = NO
 
 # If the HTML_DYNAMIC_SECTIONS tag is set to YES then the generated HTML
 # documentation will contain sections that can be hidden and shown after the
