diff -ur qterm-0.4.1.orig/qterm/Makefile.am qterm-0.4.1/qterm/Makefile.am
--- qterm-0.4.1.orig/qterm/Makefile.am	2012-01-30 09:18:50.000000000 -0500
+++ qterm-0.4.1/qterm/Makefile.am	2012-01-30 09:43:58.000000000 -0500
@@ -11,11 +11,11 @@
 TRANSLATIONS = 
 KDE_OPTIONS = qtonly
 if WANT_SSL
-qterm_LDADD = $(top_builddir)/qterm/ssh/libssh.a $(LIB_QPE) $(LIBPYTHON) $(ESD_LIB) $(LIB_SOUNDSERVER) $(LIB_QT) $(LIBSOCKET) $(LIB_OPENSSL)
+qterm_LDADD = $(top_builddir)/qterm/ssh/libssh.a $(LIB_QPE) $(LIBPYTHON) $(ESD_LIB) $(LIB_SOUNDSERVER) $(LIB_QT) $(LIBSOCKET) $(LIB_OPENSSL) -lcrypto
 
 SUBDIRS = cursor pic script docs schema ui po ssh
 else
-qterm_LDADD = $(LIB_QPE) $(LIBPYTHON) $(ESD_LIB) $(LIB_SOUNDSERVER) $(LIB_QT) $(LIBSOCKET)
+qterm_LDADD = $(LIB_QPE) $(LIBPYTHON) $(ESD_LIB) $(LIB_SOUNDSERVER) $(LIB_QT) $(LIBSOCKET) -lcrypto
 
 SUBDIRS = cursor pic script docs schema ui po
 endif
diff -ur qterm-0.4.1.orig/qterm/Makefile.in qterm-0.4.1/qterm/Makefile.in
--- qterm-0.4.1.orig/qterm/Makefile.in	2012-01-30 09:18:50.000000000 -0500
+++ qterm-0.4.1/qterm/Makefile.in	2012-01-30 09:43:10.000000000 -0500
@@ -112,7 +112,7 @@
 LIB_OPENSSL = @LIB_OPENSSL@
 LIB_POLL = @LIB_POLL@
 LIB_QPE = @LIB_QPE@
-LIB_QT = @LIB_QT@
+LIB_QT = @LIB_QT@ -lcrypto
 LIB_SOUNDSERVER = @LIB_SOUNDSERVER@
 LIB_X11 = @LIB_X11@
 LIB_XEXT = @LIB_XEXT@
