diff -ur phonon-backend-xine-4.4.4.orig/xine/CMakeLists.txt phonon-backend-xine-4.4.4/xine/CMakeLists.txt
--- phonon-backend-xine-4.4.4.orig/xine/CMakeLists.txt	2012-03-30 07:13:18.000000000 -0400
+++ phonon-backend-xine-4.4.4/xine/CMakeLists.txt	2012-03-30 07:13:45.000000000 -0400
@@ -29,7 +29,7 @@
 
 string(REGEX REPLACE "^[0-9]*\\.([0-9]*)\\.[0-9]*$" "\\1" XINE_MINOR_VERSION "${XINE_VERSION}")
 # if an older xinelib is installed give a warning about problems but continue
-if(XINE_MINOR_VERSION LESS 2 AND XINE_BUGFIX_VERSION LESS 90)
+if(XINE_MINOR_VERSION LESS 3 AND XINE_BUGFIX_VERSION LESS 90)
     if(XINE_BUGFIX_VERSION LESS 7)
         if(XINE_BUGFIX_VERSION LESS 5)
             message(WARNING "xine-lib v${XINE_VERSION} was found on your system. This version does not provide video support and has known security issues. Recommend upgrading to version 1.1.9 or above." "http://sourceforge.net/project/showfiles.php?group_id=9655&package_id=9732")
@@ -41,9 +41,9 @@
             message(WARNING "xine-lib v${XINE_VERSION} was found on your system. This version is known to have problems when playing short sounds. Consider upgrading to version 1.1.9 or above." "http://sourceforge.net/project/showfiles.php?group_id=9655&package_id=9732")
         endif(XINE_BUGFIX_VERSION LESS 9)
     endif(XINE_BUGFIX_VERSION LESS 7)
-else(XINE_MINOR_VERSION LESS 2 AND XINE_BUGFIX_VERSION LESS 90)
+else(XINE_MINOR_VERSION LESS 3 AND XINE_BUGFIX_VERSION LESS 90)
     message(FATAL_ERROR "xine-lib v${XINE_VERSION} was found on your system. This version contains major changes compared to 1.1.x and has not been tested properly for use with this backend. Consider downgrading to version 1.1.9." "http://sourceforge.net/project/showfiles.php?group_id=9655&package_id=9732")
-endif(XINE_MINOR_VERSION LESS 2 AND XINE_BUGFIX_VERSION LESS 90)
+endif(XINE_MINOR_VERSION LESS 3 AND XINE_BUGFIX_VERSION LESS 90)
 
 set(phonon_xine_SRCS
    sinknode.cpp
