Index: SuperLU_4.3/TESTING/Makefile
===================================================================
--- SuperLU_4.3.orig/TESTING/Makefile
+++ SuperLU_4.3/TESTING/Makefile
@@ -52,9 +52,9 @@ testmat:
 
 single: ./stest stest.out
 
-./stest: $(SLINTST) $(ALINTST) $(SUPERLULIB) $(TMGLIB)
+./stest: $(SLINTST) $(ALINTST) $(TMGLIB)
 	$(LOADER) $(LOADOPTS) $(SLINTST) $(ALINTST) \
-        $(TMGLIB) $(SUPERLULIB) $(BLASLIB) -lm -o $@
+        $(TMGLIB) -Wl,-rpath,../SRC $(SUPERLULIB) $(BLASLIB) -lm -o $@
 
 stest.out: stest stest.csh
 	@echo Testing SINGLE PRECISION linear equation routines 
@@ -62,9 +62,9 @@ stest.out: stest stest.csh
 
 double: ./dtest dtest.out
 
-./dtest: $(DLINTST) $(ALINTST) $(SUPERLULIB) $(TMGLIB)
+./dtest: $(DLINTST) $(ALINTST) $(TMGLIB)
 	$(LOADER) $(LOADOPTS) $(DLINTST) $(ALINTST) \
-        $(TMGLIB) $(SUPERLULIB) $(BLASLIB) -lm -o $@
+        $(TMGLIB) -Wl,-rpath,../SRC $(SUPERLULIB) $(BLASLIB) -lm -o $@
 
 dtest.out: dtest dtest.csh
 	@echo Testing DOUBLE PRECISION linear equation routines 
@@ -72,9 +72,9 @@ dtest.out: dtest dtest.csh
 
 complex: ./ctest ctest.out
 
-./ctest: $(CLINTST) $(ALINTST) $(SUPERLULIB) $(TMGLIB)
+./ctest: $(CLINTST) $(ALINTST) $(TMGLIB)
 	$(LOADER) $(LOADOPTS) $(CLINTST) $(ALINTST) \
-	$(TMGLIB) $(SUPERLULIB) $(BLASLIB) -lm -o $@
+	$(TMGLIB) -Wl,-rpath,../SRC $(SUPERLULIB) $(BLASLIB) -lm -o $@
 
 ctest.out: ctest ctest.csh
 	@echo Testing SINGLE COMPLEX linear equation routines 
@@ -82,9 +82,9 @@ ctest.out: ctest ctest.csh
 
 complex16: ./ztest ztest.out
 
-./ztest: $(ZLINTST) $(ALINTST) $(SUPERLULIB) $(TMGLIB)
+./ztest: $(ZLINTST) $(ALINTST) $(TMGLIB)
 	$(LOADER) $(LOADOPTS) $(ZLINTST) $(ALINTST) \
-	$(TMGLIB) $(SUPERLULIB) $(BLASLIB) -lm -o $@
+	$(TMGLIB) -Wl,-rpath,../SRC $(SUPERLULIB) $(BLASLIB) -lm -o $@
 
 ztest.out: ztest ztest.csh
 	@echo Testing DOUBLE COMPLEX linear equation routines 
