diff --git a/data/ibus.schemas.in b/data/ibus.schemas.in
index 007fc66..6c75562 100644
--- a/data/ibus.schemas.in
+++ b/data/ibus.schemas.in
@@ -143,7 +143,7 @@ se,si,sk,sy,sy(ku),th,tj,tr,ua,uz,vn
       <owner>ibus</owner>
       <type>list</type>
       <list_type>string</list_type>
-      <default>[&lt;Super&gt;space]</default>
+      <default>[&lt;Control&gt;space]</default>
       <locale name="C">
         <short>Trigger shortcut keys for gtk_accelerator_parse</short>
           <long>The shortcut keys for turning input method on or off</long>
diff --git a/ui/gtk3/panel.vala b/ui/gtk3/panel.vala
index 54251e7..867dbd2 100644
--- a/ui/gtk3/panel.vala
+++ b/ui/gtk3/panel.vala
@@ -354,6 +354,7 @@ class Panel : IBus.PanelService {
     }
 
     private void update_version_1_5_3() {
+#if 0
 #if ENABLE_LIBNOTIFY
         if (!Notify.is_initted()) {
             Notify.init ("ibus");
@@ -374,6 +375,7 @@ class Panel : IBus.PanelService {
 #else
         warning(_("Super+space is now the default hotkey."));
 #endif
+#endif
     }
 
     private void set_version() {
