--- knetworkconf-0.6.1/backends/network-conf.in.orig	2005-01-26 10:21:01.000000000 +0800
+++ knetworkconf-0.6.1/backends/network-conf.in	2005-06-23 11:25:15.000000000 +0800
@@ -61,20 +61,7 @@
 
 $name = "network";
 $version = "@VERSION@";
-@platforms = ("redhat-5.2", "redhat-6.0", "redhat-6.1", "redhat-6.2", "redhat-7.0", "redhat-7.1",
-              "redhat-7.2", "redhat-8.0", "redhat-9",
-              "openna-1.0",
-              "mandrake-7.1", "mandrake-7.2", "mandrake-9.0", "mandrake-9.1", "mandrake-9.2",
-              "mandrake-10.0", "mandrake-10.1",
-              "blackpanther-4.0",
-              "debian-2.2", "debian-3.0", "debian-sarge",
-              "suse-7.0", "suse-9.0", "suse-9.1",
-              "turbolinux-7.0", "fedora-1", "fedora-2", "specifix",
-              "pld-1.0", "pld-1.1", "pld-1.99",
-              "conectiva-9", "conectiva-10",
-              "vine-3.0", "vine-3.1",
-              "slackware-9.1.0", "slackware-10.0.0",
-              "gentoo", "freebsd-5", "freebsd-6");
+@platforms = ( "magic-1.2");
 
 $description =<<"end_of_description;";
        Configures all network parameters and interfaces.
--- knetworkconf-0.6.1/backends/network.pl.in.orig	2005-02-03 20:50:31.000000000 +0000
+++ knetworkconf-0.6.1/backends/network.pl.in	2005-12-08 09:41:46.000000000 +0000
@@ -458,44 +458,7 @@
        );
   my %dist_map =
       (
-       "redhat-6.0"   => "redhat-6.2",
-       "redhat-6.1"   => "redhat-6.2",
-       "redhat-6.2"   => "redhat-6.2",
-       "redhat-7.0"   => "redhat-6.2",
-       "redhat-7.1"   => "redhat-6.2",
-       "redhat-7.2"   => "redhat-6.2",
-       "redhat-8.0"   => "redhat-6.2",
-       "redhat-9"     => "redhat-6.2",
-       "openna-1.0"   => "redhat-6.2",
-       "debian-2.2"   => "debian-2.2",
-       "debian-3.0"   => "debian-2.2",
-       "debian-sarge" => "debian-2.2",
-       "mandrake-7.1" => "debian-2.2",
-       "mandrake-7.2" => "debian-2.2",
-       "mandrake-9.0" => "debian-2.2",
-       "mandrake-9.1" => "debian-2.2",
-       "mandrake-9.2" => "debian-2.2",
-       "mandrake-10.0" => "debian-2.2",
-       "mandrake-10.1" => "redhat-6.2",
-       "blackpanther-4.0" => "debian-2.2",
-       "suse-7.0"     => "debian-2.2",
-       "suse-9.0"     => "redhat-6.2",
-       "suse-9.1"     => "redhat-6.2",
-       "pld-1.0"      => "redhat-6.2",
-       "pld-1.1"      => "redhat-6.2",
-       "pld-1.99"     => "redhat-6.2",
-       "conectiva-9"  => "debian-2.2",
-       "conectiva-10" => "debian-2.2",
-       "fedora-1"     => "redhat-6.2",
-       "fedora-2"     => "redhat-6.2",
-       "specifix"     => "redhat-6.2",
-       "vine-3.0"     => "redhat-6.2",
-       "vine-3.1"     => "redhat-6.2",
-       "slackware-9.1.0" => "redhat-6.2",
-       "slackware-10.0.0" => "redhat-6.2",
-       "gentoo"       => "debian-2.2",
-       "freebsd-5"    => "debian-2.2",
-       "freebsd-6"    => "debian-2.2",
+       "magic-1.2"   => "redhat-6.2",
        );
 
   return $cmd_map{$dist_map{$gst_dist}};
@@ -1734,45 +1697,7 @@
 
   my %dist_map =
 	 (
-	  "redhat-6.0"   => \&gst_network_rh62_get_file,
-	  "redhat-6.1"   => \&gst_network_rh62_get_file,
-	  "redhat-6.2"   => \&gst_network_rh62_get_file,
-	  "redhat-7.0"   => \&gst_network_rh62_get_file,
-	  "redhat-7.1"   => \&gst_network_rh62_get_file,
-	  "redhat-7.2"   => \&gst_network_rh72_get_file,
-    "redhat-8.0"   => \&gst_network_rh72_get_file,
-    "redhat-9"     => \&gst_network_rh72_get_file,
-	  "openna-1.0"   => \&gst_network_rh62_get_file,
-	  "mandrake-7.2" => \&gst_network_rh62_get_file,
-		"mandrake-7.1" => \&gst_network_rh62_get_file,
-		"mandrake-9.0" => \&gst_network_rh62_get_file,
-		"mandrake-9.1" => \&gst_network_rh62_get_file,
-		"mandrake-9.2" => \&gst_network_rh62_get_file,
-    "mandrake-10.0" => \&gst_network_rh62_get_file,
-    "mandrake-10.1" => \&gst_network_rh62_get_file,
-    "blackpanther-4.0" => \&gst_network_rh62_get_file,
- 		"conectiva-9"  => \&gst_network_rh62_get_file,
- 		"conectiva-10" => \&gst_network_rh62_get_file,
-    "debian-2.2"   => \&gst_network_deb22_get_file,
-    "debian-3.0"   => \&gst_network_deb22_get_file,
-    "debian-sarge" => \&gst_network_deb22_get_file,
-    "suse-7.0"     => \&gst_network_suse70_get_file,
-    "suse-9.0"     => \&gst_network_deb22_get_file,
-    "suse-9.1"     => \&gst_network_deb22_get_file,
-	  "turbolinux-7.0"   => \&gst_network_rh62_get_file,
-    "pld-1.0"      => \&gst_network_pld10_get_file,
-    "pld-1.1"      => \&gst_network_pld10_get_file,
-    "pld-1.99"     => \&gst_network_pld10_get_file,
-    "fedora-1"     => \&gst_network_rh72_get_file,
-    "fedora-2"     => \&gst_network_rh72_get_file,
-    "specifix"     => \&gst_network_rh72_get_file,
-    "vine-3.0"     => \&gst_network_rh62_get_file,
-    "vine-3.1"     => \&gst_network_rh62_get_file,
-    "slackware-9.1.0" => \&gst_network_deb22_get_file,
-    "slackware-10.0.0" => \&gst_network_deb22_get_file,
-    "gentoo"       => \&gst_network_deb22_get_file,
-    "freebsd-5"    => \&gst_network_deb22_get_file,
-    "freebsd-6"    => \&gst_network_deb22_get_file,
+	  "magic-1.2"   => \&gst_network_rh62_get_file,
 	  );
 
   $proc = $dist_map{$gst_dist};
@@ -2706,7 +2631,7 @@
 	 (
 	  "bootp" => "bootp",
 	  "dhcp"  => "dhcp",
-	  "none"  => "none"
+	  "static"  => "none"
 	  );
   my $ret;
 
@@ -2727,7 +2652,7 @@
 	 (
 	  "bootp"    => "bootp",
 	  "dhcp"     => "dhcp",
-	  "none"     => "none"
+	  "none"     => "static"
 	  );
 
   return &gst_replace_sh ($file, $key, $proto_name_to_rh62{$value});
@@ -3794,46 +3719,7 @@
   my $dev;
   my %dist_map =
 	 (
-    "redhat-5.2"   => "lo",
-	  "redhat-6.0"   => "lo",
-	  "redhat-6.1"   => "lo",
-	  "redhat-6.2"   => "lo",
-	  "redhat-7.0"   => "lo",
-	  "redhat-7.1"   => "lo",
-	  "redhat-7.2"   => "lo",
-    "redhat-8.0"   => "lo",
-    "redhat-9"     => "",
-	  "openna-1.0"   => "lo",
-	  "mandrake-7.1" => "lo",
-	  "mandrake-7.2" => "lo",
-	  "mandrake-9.0" => "lo",
-	  "mandrake-9.1" => "lo",
-	  "mandrake-9.2" => "lo",
-    "mandrake-10.0" => "lo",
-    "mandrake-10.1" => "lo",
-    "blackpanther-4.0" => "lo",
- 	  "conectiva-9"  => "lo",
-    "conectiva-10" => "lo",
-	  "debian-2.2"   => "lo",
-    "debian-3.0"   => "lo",
-    "debian-sarge" => "lo",
-    "suse-7.0"     => "",
-    "suse-9.0"     => "",
-    "suse-9.1"     => "",
-	  "turbolinux-7.0"   => "lo",
-    "pld-1.0"      => "lo",
-    "pld-1.1"      => "lo",
-    "pld-1.99"     => "lo",
-    "fedora-1"     => "",
-    "fedora-2"     => "",
-    "specifix"     => "",
-    "vine-3.0"     => "lo",
-    "vine-3.1"     => "lo",
-    "slackware-9.1.0" => "",
-    "slackware-10.0.0" => "",
-    "gentoo"       => "",
-    "freebsd-5"    => "",
-    "freebsd-6"    => "",
+    "magic-1.2"   => "lo",
 	  );
 
   $dev = $dist_map {$gst_dist};
@@ -3918,47 +3804,8 @@
 {
   my %dist_map =
 	 (
-    "redhat-5.2"   => "redhat-6.2",
-	  "redhat-6.0"   => "redhat-6.2",
-	  "redhat-6.1"   => "redhat-6.2",
-	  "redhat-6.2"   => "redhat-6.2",
-	  "redhat-7.0"   => "redhat-7.0",
-	  "redhat-7.1"   => "redhat-7.0",
-	  "redhat-7.2"   => "redhat-7.2",
-    "redhat-8.0"   => "redhat-7.2",
-    "redhat-9"     => "redhat-7.2",
-	  "openna-1.0"   => "redhat-6.2",
-	  "mandrake-7.1" => "redhat-6.2",
-	  "mandrake-7.2" => "redhat-6.2",
-    "mandrake-9.0" => "redhat-7.0",
-    "mandrake-9.1" => "redhat-7.0",
-    "mandrake-9.2" => "redhat-7.0",
-    "mandrake-10.0" => "redhat-7.0",
-    "mandrake-10.1" => "redhat-7.0",
-    "blackpanther-4.0" => "redhat-7.0",
-    "conectiva-9"  => "redhat-7.0", 
-    "conectiva-10" => "redhat-7.0", 
-    "debian-2.2"   => "debian-2.2",
-    "debian-3.0"   => "debian-2.2",
-    "debian-sarge" => "debian-2.2",
-    "suse-7.0"     => "suse-7.0",
-    "suse-9.0"     => "suse-9.0",
-    "suse-9.1"     => "suse-9.0",
-	  "turbolinux-7.0"  => "redhat-7.0",
-    "pld-1.0"      => "pld-1.0",
-    "pld-1.1"      => "pld-1.0",
-    "pld-1.99"     => "pld-1.0",
-    "fedora-1"     => "redhat-7.2",
-    "fedora-2"     => "redhat-7.2",
-    "specifix"     => "redhat-7.2",
-    "vine-3.0"     => "redhat-7.0",
-    "vine-3.1"     => "redhat-7.0",
-    "slackware-9.1.0" => "slackware-9.1.0",
-    "slackware-10.0.0" => "slackware-9.1.0",
-    "gentoo"       => "gentoo",
-    "freebsd-5"    => "freebsd-5",
-    "freebsd-6"    => "freebsd-5",
-	  );
+    "magic-1.2" => "redhat-7.0",
+ 	  );
 
   my %dist_tables =
 	 (
@@ -4325,46 +4172,7 @@
 {
   my %dist_map =
 	 (
-    "redhat-5.2"   => "redhat-6.2",
-	  "redhat-6.0"   => "redhat-6.2",
-	  "redhat-6.1"   => "redhat-6.2",
-	  "redhat-6.2"   => "redhat-6.2",
-	  "redhat-7.0"   => "redhat-6.2",
-	  "redhat-7.1"   => "redhat-6.2",
-	  "redhat-7.2"   => "redhat-7.2",
-    "redhat-8.0"   => "redhat-8.0",
-    "redhat-9"     => "redhat-8.0",
-	  "openna-1.0"   => "redhat-6.2",
-	  "mandrake-7.1" => "redhat-6.2",
-    "mandrake-7.2" => "redhat-6.2",
-    "mandrake-9.0" => "mandrake-9.0",
-    "mandrake-9.1" => "mandrake-9.0",
-    "mandrake-9.2" => "mandrake-9.0",
-    "mandrake-10.0" => "mandrake-9.0",
-    "mandrake-10.1" => "mandrake-9.0",
-    "blackpanther-4.0" => "mandrake-9.0",
-    "conectiva-9"  => "conectiva-9",
-    "conectiva-10" => "conectiva-9",
-    "debian-2.2"   => "debian-2.2",
-    "debian-3.0"   => "debian-3.0",
-    "debian-sarge" => "debian-3.0",
-    "suse-7.0"     => "suse-7.0",
-    "suse-9.0"     => "suse-9.0",
-    "suse-9.1"     => "suse-9.0",
-	  "turbolinux-7.0"   => "redhat-6.2",
-    "pld-1.0"      => "pld-1.0",
-    "pld-1.1"      => "pld-1.0",
-    "pld-1.99"     => "pld-1.0",
-    "fedora-1"     => "redhat-7.2",
-    "fedora-2"     => "redhat-7.2",
-    "specifix"     => "redhat-7.2",
-    "vine-3.0"     => "vine-3.0",
-    "vine-3.1"     => "vine-3.0",
-    "slackware-9.1.0" => "slackware-9.1.0",
-    "slackware-10.0.0" => "slackware-9.1.0",
-    "gentoo"       => "gentoo",
-    "freebsd-5"    => "freebsd-5",
-    "freebsd-6"    => "freebsd-5",
+    "magic-1.2"   => "redhat-6.2",
 	  );
   
   my %dist_tables =
@@ -5140,45 +4948,7 @@
 {
   my %dist_map =
 	 (
-    "redhat-5.2"   => "redhat-5.2",
-	  "redhat-6.0"   => "redhat-6.2",
-	  "redhat-6.1"   => "redhat-6.2",
-	  "redhat-6.2"   => "redhat-6.2",
-	  "redhat-7.0"   => "redhat-7.0",
-	  "redhat-7.1"   => "redhat-7.0",
-	  "redhat-7.2"   => "redhat-7.2",
-    "redhat-8.0"   => "redhat-7.2",
-    "redhat-9"     => "redhat-7.2",
-	  "mandrake-7.1" => "redhat-6.2",
-	  "mandrake-7.2" => "redhat-6.2",
-	  "mandrake-9.0" => "redhat-7.0",
-	  "mandrake-9.1" => "redhat-7.0",
-	  "mandrake-9.2" => "redhat-7.0",
-    "mandrake-10.0" => "redhat-7.0",
-    "mandrake-10.1" => "redhat-7.0",
-    "blackpanther-4.0" => "redhat-7.0",
- 	  "conectiva-9"  => "redhat-7.0",
- 	  "conectiva-10" => "redhat-7.0",
-    "debian-2.2"   => "debian-2.2",
-    "debian-3.0" => "debian-2.2",
-    "debian-sarge" => "debian-2.2",
-    "suse-7.0"     => "suse-7.0",
-    "suse-9.0"     => "suse-9.0",
-    "suse-9.1"     => "suse-9.0",
-	  "turbolinux-7.0"   => "redhat-7.0",
-    "pld-1.0"      => "pld-1.0",
-    "pld-1.1"      => "pld-1.0",
-    "pld-1.99"     => "pld-1.0",
-    "fedora-1"     => "redhat-7.2",
-    "fedora-2"     => "redhat-7.2",
-    "specifix"     => "redhat-7.2",
-    "vine-3.0"     => "redhat-7.0",
-    "vine-3.1"     => "redhat-7.0",
-    "slackware-9.1.0" => "slackware-9.1.0",
-    "slackware-10.0.0" => "slackware-9.1.0",
-    "gentoo"       => "gentoo",
-    "freebsd-5"    => "freebsd-5",
-    "freebsd-6"    => "freebsd-5",
+    "magic-1.2"   => "redhat-7.0",
 	  );
 
   my %dist_tables =
@@ -5533,45 +5303,7 @@
 {
   my %dist_map =
 	 (
-    "redhat-5.2"   => "redhat-5.2",
-	  "redhat-6.0"   => "redhat-6.2",
-	  "redhat-6.1"   => "redhat-6.2",
-	  "redhat-6.2"   => "redhat-6.2",
-	  "redhat-7.0"   => "redhat-6.2",
-	  "redhat-7.1"   => "redhat-6.2",
-	  "redhat-7.2"   => "redhat-7.2",
-    "redhat-8.0"   => "redhat-8.0",
-    "redhat-9"     => "redhat-8.0",
-	  "mandrake-7.1" => "redhat-6.2",
-    "mandrake-7.2" => "redhat-6.2",
-    "mandrake-9.0" => "mandrake-9.0",
-    "mandrake-9.1" => "mandrake-9.0",
-    "mandrake-9.2" => "mandrake-9.0",
-    "mandrake-10.0" => "mandrake-9.0",
-    "mandrake-10.1" => "mandrake-9.0",
-    "blackpanther-4.0" => "mandrake-9.0",
-    "conectiva-9"  => "conectiva-9",
-    "conectiva-10" => "conectiva-9",
-    "debian-2.2"   => "debian-2.2",
-    "debian-3.0"   => "debian-3.0",
-    "debian-sarge" => "debian-3.0",
-    "suse-7.0"     => "suse-7.0",
-    "suse-9.0"     => "suse-9.0",
-    "suse-9.1"     => "suse-9.0",
-	  "turbolinux-7.0"   => "redhat-6.2",
-    "pld-1.0"      => "pld-1.0",
-    "pld-1.1"      => "pld-1.0",
-    "pld-1.99"     => "pld-1.0",
-    "fedora-1"     => "redhat-7.2",
-    "fedora-2"     => "redhat-7.2",
-    "specifix"     => "redhat-7.2",
-    "vine-3.0"     => "vine-3.0",
-    "vine-3.1"     => "vine-3.0",
-    "slackware-9.1.0" => "slackware-9.1.0",
-    "slackware-10.0.0" => "slackware-9.1.0",
-    "gentoo"       => "gentoo",
-    "freebsd-5"    => "freebsd-5",
-    "freebsd-6"    => "freebsd-5",
+    "magic-1.2"   => "redhat-6.2",
 	  );
 
   my %dist_tables =
--- knetworkconf-0.6.1/backends/platform.pl.in.orig	2005-01-26 10:21:01.000000000 +0800
+++ knetworkconf-0.6.1/backends/platform.pl.in	2005-06-23 11:44:24.000000000 +0800
@@ -38,53 +38,7 @@
 
 
 $PLATFORM_INFO = {
-  "debian-2.2"      => "Debian GNU/Linux 2.2 Potato",
-  "debian-3.0"      => "Debian GNU/Linux 3.0 Woody",
-  "debian-sarge"    => "Debian GNU/Linux Sarge",
-  "redhat-5.2"      => "Red Hat Linux 5.2 Apollo",
-  "redhat-6.0"      => "Red Hat Linux 6.0 Hedwig",
-  "redhat-6.1"      => "Red Hat Linux 6.1 Cartman",
-  "redhat-6.2"      => "Red Hat Linux 6.2 Zoot",
-  "redhat-7.0"      => "Red Hat Linux 7.0 Guinness",
-  "redhat-7.1"      => "Red Hat Linux 7.1 Seawolf",
-  "redhat-7.2"      => "Red Hat Linux 7.2 Enigma",
-  "redhat-7.3"      => "Red Hat Linux 7.3 Valhalla",
-  "redhat-8.0"      => "Red Hat Linux 8.0 Psyche",
-  "redhat-9"        => "Red Hat Linux 9.0 Shrike",
-  "openna-1.0"      => "OpenNA Linux 1.0 VSLC",
-  "mandrake-7.1"    => "Linux Mandrake 7.1",
-  "mandrake-7.2"    => "Linux Mandrake 7.2 Odyssey",
-  "mandrake-8.0"    => "Linux Mandrake 8.0 Traktopel",
-  "mandrake-9.0"    => "Linux Mandrake 9.0 Dolphin",
-  "mandrake-9.1"    => "Linux Mandrake 9.1 Bamboo",
-  "mandrake-9.2"    => "Linux Mandrake 9.2 FiveStar",
-  "mandrake-10.0"   => "Linux Mandrake 10.0",
-  "mandrake-10.1"   => "Linux Mandrake 10.1",
-  "blackpanther-4.0" => "Black Panther OS 4.0",
-  "conectiva-9"     => "Conectiva Linux 9",
-  "conectiva-10"    => "Conectiva Linux 10",
-  "suse-7.0"        => "SuSE Linux 7.0",
-  "suse-9.0"        => "SuSE Linux 9.0",
-  "suse-9.1"        => "SuSE Linux 9.1",
-  "turbolinux-7.0"  => "Turbolinux 7.0",
-  "slackware-8.0.0" => "Slackware 8.0.0",
-  "slackware-8.1"   => "Slackware 8.1",
-  "slackware-9.0.0" => "Slackware 9.0.0",
-  "slackware-9.1.0" => "Slackware 9.1.0",
-  "slackware-10.0.0" => "Slackware 10.0.0",
-  "freebsd-4"       => "FreeBSD 4",
-  "freebsd-5"       => "FreeBSD 5",
-  "freebsd-6"       => "FreeBSD 6",
-  "gentoo"          => "Gentoo Linux",
-  "archlinux-0.7"   => "Arch Linux 0.7",
-  "pld-1.0"         => "PLD 1.0 Ra",
-  "pld-1.1"         => "PLD 1.1 Ra",
-  "pld-1.99"        => "PLD 1.99 Ac-pre",
-  "vine-3.0"        => "Vine Linux 3.0",
-  "vine-3.1"        => "Vine Linux 3.1",
-  "fedora-1"        => "Fedora Core 1 (Yarrow)",
-  "fedora-2"        => "Fedora Core 2 (Tettnang)",
-  "specifix"        => "Specifix Linux",
+  "magic-1.2"       => "Magic Linux 1.2",
 };
   
 sub check_lsb
@@ -441,6 +395,13 @@
   return -1;
 }
 
+sub check_magic
+{
+    open MAGIC, "$gst_prefix/etc/magic-release" or return -1;
+    close MAGIC;
+        return "magic-1.2";
+}
+
 sub gst_platform_get_system
 {
   my ($tool) = @_;
@@ -456,9 +417,7 @@
 
   my %check = (
     # Red Hat check must run after Vine, Mandrake and Fedora, and Mandrake after BlackPanther
-    "Linux" => [ \&check_lsb,      \&check_debian,   \&check_caldera, \&check_suse, \&check_blackpanther, \&check_vine,
-                 \&check_fedora, \&check_mandrake, \&check_conectiva, \&check_linuxppc, \&check_redhat,  \&check_openna,
-                 \&check_turbolinux, \&check_slackware, \&check_gentoo,  \&check_pld, \&check_specifix, \&check_archlinux ],
+    "Linux" => [  \&check_magic ],
     "FreeBSD" => [ \&check_freebsd ],
     "SunOS"    => [ \&check_solaris ]
                );
--- knetworkconf-0.6.1/backends/service.pl.in.orig	2005-01-26 10:21:01.000000000 +0800
+++ knetworkconf-0.6.1/backends/service.pl.in	2005-06-23 11:49:41.000000000 +0800
@@ -48,52 +48,7 @@
   my %dist_map =
       (
        # gst_dist => [rc.X dirs location, init.d scripts location, relative path location]
-       "redhat-5.2"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-6.0"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-6.1"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-6.2"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-7.0"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-7.1"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-7.2"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-7.3"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-8.0"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "redhat-9"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "openna-1.0"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-
-       "mandrake-7.1" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "mandrake-7.2" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "mandrake-9.0" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "mandrake-9.1" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "mandrake-9.2" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "mandrake-10.0" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "mandrake-10.1" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-
-       "blackpanther-4.0" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-
-       "conectiva-9"  => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "conectiva-10" => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-
-       "debian-2.2"   => ["$gst_prefix/etc", "$gst_prefix/etc/init.d", "../init.d"],
-       "debian-3.0"   => ["$gst_prefix/etc", "$gst_prefix/etc/init.d", "../init.d"],
-       "debian-sarge" => ["$gst_prefix/etc", "$gst_prefix/etc/init.d", "../init.d"],
-
-       "suse-7.0"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d", "../"],
-       "suse-9.0"     => ["$gst_prefix/etc/init.d", "$gst_prefix/etc/init.d", "../"],
-       "suse-9.1"     => ["$gst_prefix/etc/init.d", "$gst_prefix/etc/init.d", "../"],
-
-       "turbolinux-7.0"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-
-       "pld-1.0"      => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "pld-1.1"      => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "pld-1.99"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-
-       "fedora-1"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "fedora-2"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-
-       "specifix"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-
-       "vine-3.0"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
-       "vine-3.1"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
+       "magic-1.2"   => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
        );
   my $res;
 
@@ -108,53 +63,8 @@
 {
   my %dist_map =
       (
-       "redhat-5.2"     => [3, 5],
-       "redhat-6.0"     => [3, 5],
-       "redhat-6.1"     => [3, 5],
-       "redhat-6.2"     => [3, 5],
-       "redhat-7.0"     => [3, 5],
-       "redhat-7.1"     => [3, 5],
-       "redhat-7.2"     => [3, 5],
-       "redhat-7.3"     => [3, 5],
-       "redhat-8.0"     => [3, 5],
-       "redhat-9"       => [3, 5],
-       "openna-1.0"     => [3, 5],
-
-       "mandrake-7.1"   => [3, 5],
-       "mandrake-7.2"   => [3, 5],
-       "mandrake-9.0"   => [3, 5],
-       "mandrake-9.1"   => [3, 5],
-       "mandrake-9.2"   => [3, 5],
-       "mandrake-10.0"  => [3, 5],
-       "mandrake-10.1"  => [3, 5],
-
-       "blackpanther-4.0" => [3, 5],
-
-       "conectiva-9"    => [3, 5],
-       "conectiva-10"   => [3, 5],
-
-       "debian-2.2"     => [2, 3],
-       "debian-3.0"     => [2, 3],
-       "debian-sarge"   => [2, 3],
-
-       "suse-7.0"       => [3, 5],
-       "suse-9.0"       => [3, 5],
-       "suse-9.1"       => [3, 5],
-
-       "turbolinux-7.0" => [3, 5],
-
-       "pld-1.0"        => [3, 5],
-       "pld-1.1"        => [3, 5],
-       "pld-1.99"       => [3, 5],
-
-       "fedora-1"       => [3, 5],
-       "fedora-2"       => [3, 5],
-
-       "specifix"       => [3, 5],
-       
-       "vine-3.0"       => [3, 5],
-       "vine-3.1"       => [3, 5],
-       );
+       "magic-1.2"     => [3, 5],
+        );
   my $res;
 
   $res = $dist_map{$gst_dist};
@@ -167,58 +77,7 @@
   my (%dist_map, %runlevels, $desc, $distro);
   %dist_map =
     (
-     "redhat-5.2"     => "redhat-5.2",
-     "redhat-6.0"     => "redhat-5.2",
-     "redhat-6.1"     => "redhat-5.2",
-     "redhat-6.2"     => "redhat-5.2",
-     "redhat-7.0"     => "redhat-5.2",
-     "redhat-7.1"     => "redhat-5.2",
-     "redhat-7.2"     => "redhat-5.2",
-     "redhat-7.3"     => "redhat-5.2",
-     "redhat-8.0"     => "redhat-5.2",
-     "redhat-9"       => "redhat-5.2",
-     "openna-1.0"     => "redhat-5.2",
-     
-     "mandrake-7.1"   => "redhat-5.2",
-     "mandrake-7.2"   => "redhat-5.2",
-     "mandrake-9.0"   => "redhat-5.2",
-     "mandrake-9.1"   => "redhat-5.2",
-     "mandrake-9.2"   => "redhat-5.2",
-     "mandrake-10.0"  => "redhat-5.2",
-     "mandrake-10.1"  => "redhat-5.2",
-     
-     "blackpanther-4.0" => "redhat-5.2",
-
-     "conectiva-9"    => "redhat-5.2",
-     "conectiva-10"   => "redhat-5.2",
-     
-     "debian-2.2"     => "debian-2.2",
-     "debian-3.0"     => "debian-2.2",
-     "debian-sarge"   => "debian-2.2",
-     
-     "suse-7.0"       => "redhat-5.2",
-     "suse-9.0"       => "redhat-5.2",
-     "suse-9.1"       => "redhat-5.2",
-     
-     "turbolinux-7.0" => "redhat-5.2",
-     "pld-1.0"        => "redhat-5.2",
-     "pld-1.1"        => "redhat-5.2",
-     "pld-1.99"       => "redhat-5.2",
-     "fedora-1"       => "redhat-5.2",
-     "fedora-2"       => "redhat-5.2",
-
-     "specifix"       => "redhat-5.2",
-
-     "vine-3.0"       => "redhat-5.2",
-     "vine-3.1"       => "redhat-5.2",
-
-     "slackware-9.1.0" => "slackware-9.1.0",
-     "slackware-10.0.0" => "slackware-9.1.0",
-
-     "gentoo"         => "gentoo",
-
-     "freebsd-5"      => "freebsd-5",
-     "freebsd-6"      => "freebsd-5",
+     "magic-1.2"     => "redhat-5.2",
     );
 
   %runlevels=
@@ -1570,32 +1429,7 @@
   my ($service) = @_;
   my %dist_map =
       (
-       "debian-2.2"   => \&gst_service_debian_get_status,
-       "debian-3.0"   => \&gst_service_debian_get_status,
-       "debian-sarge" => \&gst_service_debian_get_status,
-       
-       "redhat-5.2"   => \&gst_service_redhat_get_status,
-       "redhat-6.0"   => \&gst_service_redhat_get_status,
-       "redhat-6.1"   => \&gst_service_redhat_get_status,
-       "redhat-6.2"   => \&gst_service_redhat_get_status,
-       "redhat-7.0"   => \&gst_service_redhat_get_status,
-       "redhat-7.1"   => \&gst_service_redhat_get_status,
-       "redhat-7.2"   => \&gst_service_redhat_get_status,
-       "redhat-7.3"   => \&gst_service_redhat_get_status,
-       "redhat-8.0"   => \&gst_service_redhat_get_status,
-       "redhat-9"     => \&gst_service_redhat_get_status,
-       "mandrake-7.2" => \&gst_service_redhat_get_status,
-       "fedora-1"     => \&gst_service_redhat_get_status,
-       "fedora-2"     => \&gst_service_redhat_get_status,
-       "specifix"     => \&gst_service_redhat_get_status,
-
-       "suse-9.0"     => \&gst_service_suse_get_status,
-       "suse-9.1"     => \&gst_service_suse_get_status,
-
-       "gentoo"       => \&gst_service_gentoo_get_status,
-
-       "freebsd-5"    => \&gst_service_rcng_get_status,
-       "freebsd-6"    => \&gst_service_rcng_get_status,
+       "magic-1.2"   => \&gst_service_redhat_get_status,
       );
   my $proc;
 
--- knetworkconf-0.6.1/knetworkconf/knetworkconfigparser.cpp.orig	2005-02-14 16:18:09.000000000 +0800
+++ knetworkconf-0.6.1/knetworkconf/knetworkconfigparser.cpp	2005-06-23 11:23:54.000000000 +0800
@@ -32,7 +32,7 @@
 /** Runs the gst backend to get network values. You can pass to the script a specific platform to load using the platform parameter.  */
 void KNetworkConfigParser::runDetectionScript(QString platform){
   KDetectDistroDlg* dialog = new KDetectDistroDlg(0, 0);
-  dialog->show();
+//  dialog->show();
   procDetect = new QProcess(this);
   QString pathToProgram =  locate("data",BACKEND_PATH);
   if (pathToProgram == "")
