diff -ur webrtc-20130531svn3704-orig/webrtc/modules/audio_processing/aec/Makefile.am webrtc-20130531svn3704/webrtc/modules/audio_processing/aec/Makefile.am
--- webrtc-20130531svn3704-orig/webrtc/modules/audio_processing/aec/Makefile.am	2014-07-03 02:11:52.030138260 +0000
+++ webrtc-20130531svn3704/webrtc/modules/audio_processing/aec/Makefile.am	2014-07-03 02:13:50.188401882 +0000
@@ -8,8 +8,8 @@
 if ARM
 # Do nothing for now.
 else
-libaec_la_SOURCES += aec_core_sse2.c \
-                     aec_rdft_sse2.c
+# libaec_la_SOURCES += aec_core_sse2.c \
+#                      aec_rdft_sse2.c
 endif
 
 libaec_la_LIBADD = ../utility/libapm_util.la \
@@ -30,5 +30,5 @@
 if ARM
 # Do nothing for now.
 else
-AM_CFLAGS += -msse2
+# AM_CFLAGS += -msse2
 endif
diff -ur webrtc-20130531svn3704-orig/webrtc/modules/video_processing/main/source/Makefile.am webrtc-20130531svn3704/webrtc/modules/video_processing/main/source/Makefile.am
--- webrtc-20130531svn3704-orig/webrtc/modules/video_processing/main/source/Makefile.am	2014-07-03 02:11:52.154161761 +0000
+++ webrtc-20130531svn3704/webrtc/modules/video_processing/main/source/Makefile.am	2014-07-03 02:20:09.481564830 +0000
@@ -14,7 +14,7 @@
 if ARM
 # Do nothing for now.
 else
-libvideo_processing_la_SOURCES += content_analysis_sse2.cc
+# libvideo_processing_la_SOURCES += content_analysis_sse2.cc
 endif
 
 libvideo_processing_la_LIBADD = ../../../../common_video/libcommon_video.la \
@@ -44,5 +44,5 @@
 if ARM
 # Do nothing for now.
 else
-AM_CXXFLAGS += -msse2
+# AM_CXXFLAGS += -msse2
 endif
