diff -up webrtc-20130326svn3237/webrtc/modules/video_render/incoming_video_stream.cc.doubleslash webrtc-20130326svn3237/webrtc/modules/video_render/incoming_video_stream.cc
--- webrtc-20130326svn3237/webrtc/modules/video_render/incoming_video_stream.cc.doubleslash	2012-10-31 01:22:11.000000000 -0400
+++ webrtc-20130326svn3237/webrtc/modules/video_render/incoming_video_stream.cc	2013-03-26 17:00:54.729908965 -0400
@@ -8,7 +8,7 @@
  *  be found in the AUTHORS file in the root of the source tree.
  */
 
-#include "webrtc/modules/video_render//incoming_video_stream.h"
+#include "webrtc/modules/video_render/incoming_video_stream.h"
 
 #include <cassert>
 
@@ -22,7 +22,7 @@
 #endif
 
 #include "common_video/libyuv/include/webrtc_libyuv.h"
-#include "webrtc/modules/video_render//video_render_frames.h"
+#include "webrtc/modules/video_render/video_render_frames.h"
 #include "system_wrappers/interface/critical_section_wrapper.h"
 #include "system_wrappers/interface/event_wrapper.h"
 #include "system_wrappers/interface/map_wrapper.h"
diff -up webrtc-20130326svn3237/webrtc/modules/video_render/video_render_frames.cc.doubleslash webrtc-20130326svn3237/webrtc/modules/video_render/video_render_frames.cc
--- webrtc-20130326svn3237/webrtc/modules/video_render/video_render_frames.cc.doubleslash	2012-10-31 01:22:11.000000000 -0400
+++ webrtc-20130326svn3237/webrtc/modules/video_render/video_render_frames.cc	2013-03-26 16:50:51.665908965 -0400
@@ -8,7 +8,7 @@
  *  be found in the AUTHORS file in the root of the source tree.
  */
 
-#include "webrtc/modules/video_render//video_render_frames.h"
+#include "webrtc/modules/video_render/video_render_frames.h"
 
 #include <cassert>
 
