diff -up tree-1.6.0/Makefile.no-strip tree-1.6.0/Makefile
--- tree-1.6.0/Makefile.no-strip	2011-06-24 15:25:27.000000000 +0100
+++ tree-1.6.0/Makefile	2011-06-27 16:10:48.812529754 +0100
@@ -97,7 +97,7 @@ install: tree
 	install -d $(BINDIR)
 	install -d $(MANDIR)
 	if [ -e $(TREE_DEST) ]; then \
-		install -s $(TREE_DEST) $(BINDIR)/$(TREE_DEST); \
+		install $(TREE_DEST) $(BINDIR)/$(TREE_DEST); \
 	fi
 	install doc/$(MAN) $(MANDIR)/$(MAN)
 
