diff -up tree-1.6.0/tree.c.no-color-by-default tree-1.6.0/tree.c
--- tree-1.6.0/tree.c.no-color-by-default	2011-06-27 16:14:14.190548385 +0100
+++ tree-1.6.0/tree.c	2011-06-27 16:14:27.257295243 +0100
@@ -79,7 +79,8 @@ int main(int argc, char **argv)
   q = p = dtotal = ftotal = 0;
   aflag = dflag = fflag = lflag = pflag = sflag = Fflag = uflag = gflag = FALSE;
   Dflag = qflag = Nflag = Qflag = Rflag = hflag = Hflag = siflag = cflag = FALSE;
-  noindent = force_color = nocolor = xdev = noreport = nolinks = FALSE;
+  noindent = force_color = xdev = noreport = nolinks = FALSE;
+  nocolor = TRUE;
   dirsfirst = nosort = inodeflag = devflag = Xflag = FALSE;
   duflag = pruneflag = FALSE;
   flimit = 0;
