diff -up tree-1.6.0/Makefile.preserve-timestamps tree-1.6.0/Makefile
--- tree-1.6.0/Makefile.preserve-timestamps	2011-06-27 16:11:26.000000000 +0100
+++ tree-1.6.0/Makefile	2011-06-27 16:12:36.394443602 +0100
@@ -97,9 +97,9 @@ install: tree
 	install -d $(BINDIR)
 	install -d $(MANDIR)
 	if [ -e $(TREE_DEST) ]; then \
-		install $(TREE_DEST) $(BINDIR)/$(TREE_DEST); \
+		install -p $(TREE_DEST) $(BINDIR)/$(TREE_DEST); \
 	fi
-	install doc/$(MAN) $(MANDIR)/$(MAN)
+	install -p doc/$(MAN) $(MANDIR)/$(MAN)
 
 distclean:
 	if [ -f tree.o ]; then rm *.o; fi
