diff -Nuar kdesdk-3.5.13.1/CMakeLists.txt kdesdk-3.5.13.1.patched/CMakeLists.txt
--- kdesdk-3.5.13.1/CMakeLists.txt	2012-09-08 12:06:50.000000000 +0200
+++ kdesdk-3.5.13.1.patched/CMakeLists.txt	2012-10-01 19:24:39.244352538 +0200
@@ -109,7 +109,7 @@
 tde_conditional_add_subdirectory( BUILD_POXML poxml )
 tde_conditional_add_subdirectory( BUILD_SCHECK scheck )
 tde_conditional_add_subdirectory( BUILD_SCRIPTS scripts )
-tde_conditional_add_subdirectory( BUILD_KDECACHEGRIND kdecachegrind )
+tde_conditional_add_subdirectory( BUILD_KDECACHEGRIND kcachegrind )
 tde_conditional_add_subdirectory( BUILD_UMBRELLO umbrello )
 
 
diff -Nuar kdesdk-3.5.13.1/kcachegrind/CMakeLists.txt kdesdk-3.5.13.1.patched/kcachegrind/CMakeLists.txt
--- kdesdk-3.5.13.1/kcachegrind/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ kdesdk-3.5.13.1.patched/kcachegrind/CMakeLists.txt	2012-10-01 19:35:12.689266467 +0200
@@ -0,0 +1,21 @@
+#################################################
+#
+#  (C) 2012 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+include( ConfigureChecks.cmake )
+
+set( KCACHEGRIND_VERSION "0.4.6kde" )
+configure_file( version.h.in version.h @ONLY )
+configure_file( kcachegrind.spec.in kcachegrind.spec @ONLY )
+configure_file( kcachegrind.lsm.in kcachegrind.lsm @ONLY )
+
+add_subdirectory( kcachegrind )
+add_subdirectory( pics )
+add_subdirectory( converters )
diff -Nuar kdesdk-3.5.13.1/kcachegrind/ConfigureChecks.cmake kdesdk-3.5.13.1.patched/kcachegrind/ConfigureChecks.cmake
--- kdesdk-3.5.13.1/kcachegrind/ConfigureChecks.cmake	1970-01-01 01:00:00.000000000 +0100
+++ kdesdk-3.5.13.1.patched/kcachegrind/ConfigureChecks.cmake	2012-10-01 19:35:15.546204776 +0200
@@ -0,0 +1,12 @@
+#################################################
+#
+#  (C) 2012 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+check_function_exists( mmap HAVE_MMAP )
diff -Nuar kdesdk-3.5.13.1/kcachegrind/converters/CMakeLists.txt kdesdk-3.5.13.1.patched/kcachegrind/converters/CMakeLists.txt
--- kdesdk-3.5.13.1/kcachegrind/converters/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ kdesdk-3.5.13.1.patched/kcachegrind/converters/CMakeLists.txt	2012-10-01 19:33:48.381089049 +0200
@@ -0,0 +1,15 @@
+#################################################
+#
+#  (C) 2012 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+install( PROGRAMS
+    hotshot2calltree op2calltree pprof2calltree
+    dprof2calltree memprof2calltree
+  DESTINATION ${BIN_INSTALL_DIR} )
diff -Nuar kdesdk-3.5.13.1/kcachegrind/kcachegrind/CMakeLists.txt kdesdk-3.5.13.1.patched/kcachegrind/kcachegrind/CMakeLists.txt
--- kdesdk-3.5.13.1/kcachegrind/kcachegrind/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ kdesdk-3.5.13.1.patched/kcachegrind/kcachegrind/CMakeLists.txt	2012-10-01 19:35:46.872528600 +0200
@@ -0,0 +1,50 @@
+#################################################
+#
+#  (C) 2012 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+include_directories(
+  ${CMAKE_CURRENT_BINARY_DIR}
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${CMAKE_BINARY_DIR}
+  ${TDE_INCLUDE_DIR}
+  ${TQT_INCLUDE_DIRS}
+)
+
+link_directories(
+  ${TQT_LIBRARY_DIRS}
+)
+
+
+##### other data ################################
+
+tde_install_icons( )
+install( FILES kcachegrind.desktop DESTINATION ${XDG_APPS_INSTALL_DIR} )
+install( FILES kcachegrindui.rc tips DESTINATION ${DATA_INSTALL_DIR}/kcachegrind )
+install( FILES x-kcachegrind.desktop DESTINATION ${MIME_INSTALL_DIR}/application )
+
+
+##### kcachegrind (executable) ##################
+
+tde_add_executable( kcachegrind AUTOMOC
+  SOURCES
+    functionselectionbase.ui stackselectionbase.ui partselectionbase.ui
+    configdlgbase.ui loader.cpp cachegrindloader.cpp treemap.cpp pool.cpp
+    main.cpp configuration.cpp functionselection.cpp coverage.cpp
+    partgraph.cpp toplevel.cpp stackselection.cpp stackbrowser.cpp
+    subcost.cpp tracedata.cpp partselection.cpp configdlg.cpp
+    utils.cpp fixcost.cpp traceitemview.cpp instrview.cpp tabview.cpp
+    sourceview.cpp callmapview.cpp callview.cpp coverageview.cpp
+    costtypeview.cpp partview.cpp listutils.cpp costtypeitem.cpp
+    multiview.cpp callitem.cpp coverageitem.cpp sourceitem.cpp
+    costlistitem.cpp partlistitem.cpp functionitem.cpp instritem.cpp
+    stackitem.cpp callgraphview.cpp
+  LINK kio-shared
+  DESTINATION ${BIN_INSTALL_DIR}
+)
\ Pas de fin de ligne à la fin du fichier
diff -Nuar kdesdk-3.5.13.1/kcachegrind/pics/CMakeLists.txt kdesdk-3.5.13.1.patched/kcachegrind/pics/CMakeLists.txt
--- kdesdk-3.5.13.1/kcachegrind/pics/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ kdesdk-3.5.13.1.patched/kcachegrind/pics/CMakeLists.txt	2012-10-01 19:35:59.697251930 +0200
@@ -0,0 +1,14 @@
+#################################################
+#
+#  (C) 2012 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+add_subdirectory( hicolor )
+
+tde_install_icons( DESTINATION ${DATA_INSTALL_DIR}/kcachegrind/icons )
diff -Nuar kdesdk-3.5.13.1/kcachegrind/pics/hicolor/CMakeLists.txt kdesdk-3.5.13.1.patched/kcachegrind/pics/hicolor/CMakeLists.txt
--- kdesdk-3.5.13.1/kcachegrind/pics/hicolor/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ kdesdk-3.5.13.1.patched/kcachegrind/pics/hicolor/CMakeLists.txt	2012-10-01 19:36:11.193004007 +0200
@@ -0,0 +1,12 @@
+#################################################
+#
+#  (C) 2012 Serghei Amelian
+#  serghei (DOT) amelian (AT) gmail.com
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+tde_install_icons( DESTINATION ${DATA_INSTALL_DIR}/kcachegrind/icons )
diff -Nuar kdesdk-3.5.13.1/kdecachegrind/CMakeLists.txt kdesdk-3.5.13.1.patched/kdecachegrind/CMakeLists.txt
--- kdesdk-3.5.13.1/kdecachegrind/CMakeLists.txt	2012-09-08 12:06:51.000000000 +0200
+++ kdesdk-3.5.13.1.patched/kdecachegrind/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
@@ -1,21 +0,0 @@
-#################################################
-#
-#  (C) 2012 Serghei Amelian
-#  serghei (DOT) amelian (AT) gmail.com
-#
-#  Improvements and feedback are welcome
-#
-#  This file is released under GPL >= 2
-#
-#################################################
-
-include( ConfigureChecks.cmake )
-
-set( KCACHEGRIND_VERSION "0.4.6kde" )
-configure_file( version.h.in version.h @ONLY )
-configure_file( kdecachegrind.spec.in kcachegrind.spec @ONLY )
-configure_file( kdecachegrind.lsm.in kcachegrind.lsm @ONLY )
-
-add_subdirectory( kdecachegrind )
-add_subdirectory( pics )
-add_subdirectory( converters )
diff -Nuar kdesdk-3.5.13.1/kdecachegrind/ConfigureChecks.cmake kdesdk-3.5.13.1.patched/kdecachegrind/ConfigureChecks.cmake
--- kdesdk-3.5.13.1/kdecachegrind/ConfigureChecks.cmake	2012-09-08 12:06:51.000000000 +0200
+++ kdesdk-3.5.13.1.patched/kdecachegrind/ConfigureChecks.cmake	1970-01-01 01:00:00.000000000 +0100
@@ -1,12 +0,0 @@
-#################################################
-#
-#  (C) 2012 Serghei Amelian
-#  serghei (DOT) amelian (AT) gmail.com
-#
-#  Improvements and feedback are welcome
-#
-#  This file is released under GPL >= 2
-#
-#################################################
-
-check_function_exists( mmap HAVE_MMAP )
diff -Nuar kdesdk-3.5.13.1/kdecachegrind/converters/CMakeLists.txt kdesdk-3.5.13.1.patched/kdecachegrind/converters/CMakeLists.txt
--- kdesdk-3.5.13.1/kdecachegrind/converters/CMakeLists.txt	2012-09-08 12:06:51.000000000 +0200
+++ kdesdk-3.5.13.1.patched/kdecachegrind/converters/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
@@ -1,15 +0,0 @@
-#################################################
-#
-#  (C) 2012 Serghei Amelian
-#  serghei (DOT) amelian (AT) gmail.com
-#
-#  Improvements and feedback are welcome
-#
-#  This file is released under GPL >= 2
-#
-#################################################
-
-install( PROGRAMS
-    hotshot2calltree op2calltree pprof2calltree
-    dprof2calltree memprof2calltree
-  DESTINATION ${BIN_INSTALL_DIR} )
diff -Nuar kdesdk-3.5.13.1/kdecachegrind/kdecachegrind/CMakeLists.txt kdesdk-3.5.13.1.patched/kdecachegrind/kdecachegrind/CMakeLists.txt
--- kdesdk-3.5.13.1/kdecachegrind/kdecachegrind/CMakeLists.txt	2012-09-08 12:06:51.000000000 +0200
+++ kdesdk-3.5.13.1.patched/kdecachegrind/kdecachegrind/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
@@ -1,50 +0,0 @@
-#################################################
-#
-#  (C) 2012 Serghei Amelian
-#  serghei (DOT) amelian (AT) gmail.com
-#
-#  Improvements and feedback are welcome
-#
-#  This file is released under GPL >= 2
-#
-#################################################
-
-include_directories(
-  ${CMAKE_CURRENT_BINARY_DIR}
-  ${CMAKE_CURRENT_SOURCE_DIR}
-  ${CMAKE_BINARY_DIR}
-  ${TDE_INCLUDE_DIR}
-  ${TQT_INCLUDE_DIRS}
-)
-
-link_directories(
-  ${TQT_LIBRARY_DIRS}
-)
-
-
-##### other data ################################
-
-tde_install_icons( )
-install( FILES kdecachegrind.desktop DESTINATION ${XDG_APPS_INSTALL_DIR} )
-install( FILES kdecachegrindui.rc tips DESTINATION ${DATA_INSTALL_DIR}/kcachegrind )
-install( FILES x-kdecachegrind.desktop DESTINATION ${MIME_INSTALL_DIR}/application )
-
-
-##### kdecachegrind (executable) ##################
-
-tde_add_executable( kdecachegrind AUTOMOC
-  SOURCES
-    functionselectionbase.ui stackselectionbase.ui partselectionbase.ui
-    configdlgbase.ui loader.cpp cachegrindloader.cpp treemap.cpp pool.cpp
-    main.cpp configuration.cpp functionselection.cpp coverage.cpp
-    partgraph.cpp toplevel.cpp stackselection.cpp stackbrowser.cpp
-    subcost.cpp tracedata.cpp partselection.cpp configdlg.cpp
-    utils.cpp fixcost.cpp traceitemview.cpp instrview.cpp tabview.cpp
-    sourceview.cpp callmapview.cpp callview.cpp coverageview.cpp
-    costtypeview.cpp partview.cpp listutils.cpp costtypeitem.cpp
-    multiview.cpp callitem.cpp coverageitem.cpp sourceitem.cpp
-    costlistitem.cpp partlistitem.cpp functionitem.cpp instritem.cpp
-    stackitem.cpp callgraphview.cpp
-  LINK kio-shared
-  DESTINATION ${BIN_INSTALL_DIR}
-)
diff -Nuar kdesdk-3.5.13.1/kdecachegrind/pics/CMakeLists.txt kdesdk-3.5.13.1.patched/kdecachegrind/pics/CMakeLists.txt
--- kdesdk-3.5.13.1/kdecachegrind/pics/CMakeLists.txt	2012-09-08 12:06:51.000000000 +0200
+++ kdesdk-3.5.13.1.patched/kdecachegrind/pics/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
@@ -1,14 +0,0 @@
-#################################################
-#
-#  (C) 2012 Serghei Amelian
-#  serghei (DOT) amelian (AT) gmail.com
-#
-#  Improvements and feedback are welcome
-#
-#  This file is released under GPL >= 2
-#
-#################################################
-
-add_subdirectory( hicolor )
-
-tde_install_icons( DESTINATION ${DATA_INSTALL_DIR}/kdecachegrind/icons )
diff -Nuar kdesdk-3.5.13.1/kdecachegrind/pics/hicolor/CMakeLists.txt kdesdk-3.5.13.1.patched/kdecachegrind/pics/hicolor/CMakeLists.txt
--- kdesdk-3.5.13.1/kdecachegrind/pics/hicolor/CMakeLists.txt	2012-09-08 12:06:51.000000000 +0200
+++ kdesdk-3.5.13.1.patched/kdecachegrind/pics/hicolor/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
@@ -1,12 +0,0 @@
-#################################################
-#
-#  (C) 2012 Serghei Amelian
-#  serghei (DOT) amelian (AT) gmail.com
-#
-#  Improvements and feedback are welcome
-#
-#  This file is released under GPL >= 2
-#
-#################################################
-
-tde_install_icons( DESTINATION ${DATA_INSTALL_DIR}/kdecachegrind/icons )
