diff -up boost_1_54_0/boost/interprocess/detail/atomic.hpp\~ boost_1_54_0/boost/interprocess/detail/atomic.hpp
--- boost_1_54_0/boost/interprocess/detail/atomic.hpp~	2012-09-24 14:17:34.000000000 +0200
+++ boost_1_54_0/boost/interprocess/detail/atomic.hpp	2013-08-29 16:36:51.682606594 +0200
@@ -213,7 +213,7 @@ inline boost::uint32_t atomic_cas32
                  "bne-   1b\n\t"
                  "2:"
                  : "=&r"(prev)
-                 : "b" (mem), "r"(cmp), "r" (with)
+                 : "b" (mem), "r" (with), "r" (cmp)
                  : "cc", "memory");
    return prev;
 }

Diff finished.  Thu Aug 29 16:36:55 2013
