diff -Naur libXaw-1.0.12.orig/src/DisplayList.c libXaw-1.0.12/src/DisplayList.c
--- libXaw-1.0.12.orig/src/DisplayList.c	2013-09-08 05:57:54.000000000 +0200
+++ libXaw-1.0.12/src/DisplayList.c	2014-04-16 18:08:03.440000000 +0200
@@ -287,7 +287,7 @@
 	}
       if (fp)
 	{
-	  snprintf(cname, fp - fname + 1, fname);
+	  snprintf(cname, fp - fname + 1, "%s", fname);
 	  memmove(fname, fp + 1, strlen(fp));
 	  lc = cname[0] ? XawGetDisplayListClass(cname) : xlibc;
 	  if (!lc)
