diff -Nuarp kdepim-runtime-4.4.2.orig/agents/CMakeLists.txt kdepim-runtime-4.4.2/agents/CMakeLists.txt
--- kdepim-runtime-4.4.2.orig/agents/CMakeLists.txt	2010-04-10 17:50:31.000000000 +0800
+++ kdepim-runtime-4.4.2/agents/CMakeLists.txt	2010-04-10 18:01:28.000000000 +0800
@@ -1,4 +1,3 @@
-add_subdirectory( maildispatcher )
 
 if( Nepomuk_FOUND AND STRIGI_FOUND )
     include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/nepomukfeeder ${CMAKE_CURRENT_BINARY_DIR}/ontologies ${STRIGI_INCLUDE_DIR})
diff -Nuarp kdepim-runtime-4.4.2.orig/resources/CMakeLists.txt kdepim-runtime-4.4.2/resources/CMakeLists.txt
--- kdepim-runtime-4.4.2.orig/resources/CMakeLists.txt	2010-04-10 17:50:31.000000000 +0800
+++ kdepim-runtime-4.4.2/resources/CMakeLists.txt	2010-04-10 18:02:28.000000000 +0800
@@ -37,7 +37,6 @@ add_subdirectory( maildir )
 add_subdirectory( mailtransport_dummy )
 add_subdirectory( mbox )
 add_subdirectory( microblog )
-add_subdirectory( pop3 )
 
 if ( NEPOMUK_FOUND )
   add_subdirectory( nepomuktag )
