diff -Nuarp kdepim-runtime-4.3.4/libkdepim-copy/CMakeLists.txt kdepim-runtime-4.3.4.fix_enablefinal/libkdepim-copy/CMakeLists.txt
--- kdepim-runtime-4.3.4/libkdepim-copy/CMakeLists.txt	2009-06-25 00:19:59.000000000 +0800
+++ kdepim-runtime-4.3.4.fix_enablefinal/libkdepim-copy/CMakeLists.txt	2009-12-28 21:22:42.000000000 +0800
@@ -1,3 +1,4 @@
+kde4_no_enable_final(libkdepim-copy)
 
 add_definitions(-DQT3_SUPPORT -DQT3_SUPPORT_WARNINGS)
 
diff -Nuarp kdepim-runtime-4.3.4/resources/kolabproxy/CMakeLists.txt kdepim-runtime-4.3.4.fix_enablefinal/resources/kolabproxy/CMakeLists.txt
--- kdepim-runtime-4.3.4/resources/kolabproxy/CMakeLists.txt	2009-08-04 06:09:41.000000000 +0800
+++ kdepim-runtime-4.3.4.fix_enablefinal/resources/kolabproxy/CMakeLists.txt	2009-12-28 21:25:22.000000000 +0800
@@ -1,4 +1,5 @@
 project(kolabproxy)
+kde4_no_enable_final(kolabproxy)
 
 include_directories(
     ${kdepim-runtime_SOURCE_DIR}
diff -Nuarp kdepim-runtime-4.3.4/tray/CMakeLists.txt kdepim-runtime-4.3.4.fix_enablefinal/tray/CMakeLists.txt
--- kdepim-runtime-4.3.4/tray/CMakeLists.txt	2009-10-02 16:20:43.000000000 +0800
+++ kdepim-runtime-4.3.4.fix_enablefinal/tray/CMakeLists.txt	2009-12-28 21:23:09.000000000 +0800
@@ -1,3 +1,4 @@
+kde4_no_enable_final(tray)
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}
                      ${QT_INCLUDE_DIR} ${QT_QTDBUS_INCLUDE_DIR} )
 include_directories( BEFORE ${LIBKNOTIFICATIONITEM-1_INCLUDE_DIR} )
