--- kaffeine-3.5.13.1/kaffeine/src/player-parts/xine-part/kxinewidget.cpp.ORI	2012-10-03 21:39:27.711879801 +0200
+++ kaffeine-3.5.13.1/kaffeine/src/player-parts/xine-part/kxinewidget.cpp	2012-10-03 21:40:35.174370892 +0200
@@ -2642,7 +2642,11 @@
 
 bool KXineWidget::getAutoplayPluginURLS(const TQString& plugin, TQStringList& list)
 {
+#if XINE_MAJOR_VERSION > 1 || ( XINE_MAJOR_VERSION == 1 && XINE_MINOR_VERSION >= 2 )
+	const char* const* urls = NULL;
+#else
 	char** urls = NULL;
+#endif
 	int num;
 	int i = 0;
 
