diff -urN cfitsio/fitsio2.h cfitsio_new/fitsio2.h
--- cfitsio/fitsio2.h	2010-01-26 18:40:25.000000000 +0100
+++ cfitsio_new/fitsio2.h	2010-06-30 17:24:59.000000000 +0200
@@ -85,6 +85,14 @@
 #define MACHINE NATIVE
 #define LONGSIZE 64   
 
+#elif defined(__s390x__)
+#define BYTESWAPPED FALSE
+#define LONGSIZE 64
+
+#elif defined(__s390__)
+#define BYTESWAPPED FALSE
+#define LONGSIZE 32
+
 #elif defined(__ia64__)  || defined(__x86_64__)
                   /*  Intel itanium 64-bit PC, or AMD opteron 64-bit PC */
 #define BYTESWAPPED TRUE
