diff -ur cfitsio/cfitsio.pc.in cfitsio.zlib/cfitsio.pc.in
--- cfitsio/cfitsio.pc.in	2011-09-22 22:04:57.000000000 +0200
+++ cfitsio.zlib/cfitsio.pc.in	2011-12-06 18:01:48.491309113 +0100
@@ -1,11 +1,11 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/cfitsio
 
 Name: cfitsio
 Description: FITS File Subroutine Library
-Version: 3.29
-Libs: -L${libdir} -lcfitsio @LIBS@
-Libs.private: -lm
-Cflags: -I${includedir}
+Version: 3.290
+Libs: -L${libdir} -lcfitsio
+Libs.private: @LIBS@ -lm
+Cflags: -D_REENTRANT -I${includedir}
Sólo en cfitsio: cfitsio.pc.in.orig
Sólo en cfitsio: cfitsio.pc.in.rej
