--- qt-x11-free-3.3.6/tools/designer/uic/main.cpp.orig	2006-05-15 19:43:34.000000000 +0200
+++ qt-x11-free-3.3.6/tools/designer/uic/main.cpp	2006-05-15 19:45:24.000000000 +0200
@@ -316,8 +316,7 @@
 	out << "/****************************************************************************" << endl;
 	out << "** Form "<< (impl? "implementation" : "interface") << " generated from reading ui file '" << fileName << "'" << endl;
 	out << "**" << endl;
-	out << "** Created: " << QDateTime::currentDateTime().toString() << endl;
-	out << "**      by: The User Interface Compiler ($Id: qt/main.cpp   3.3.7   edited Aug 31 2005 $)" << endl;
+	out << "** Created by: The User Interface Compiler ($Id: qt/main.cpp   3.3.7   edited Aug 31 2005 $)" << endl;
 	out << "**" << endl;
 	out << "** WARNING! All changes made in this file will be lost!" << endl;
 	out << "****************************************************************************/" << endl << endl;
--- qt-x11-free-3.3.6/tools/designer/uic/embed.cpp.orig	2006-05-15 19:45:32.000000000 +0200
+++ qt-x11-free-3.3.6/tools/designer/uic/embed.cpp	2006-05-15 19:46:07.000000000 +0200
@@ -130,8 +130,7 @@
     for ( it = images.begin(); it != images.end(); ++it )
 	out << "**      " << *it << "\n";
     out << "**\n";
-    out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
-    out << "**      by: The User Interface Compiler ($Id: qt/embed.cpp   3.3.7   edited Aug 31 2005 $)\n";
+    out << "** Created by: The User Interface Compiler ($Id: qt/embed.cpp   3.3.7   edited Aug 31 2005 $)\n";
     out << "**\n";
     out << "** WARNING! All changes made in this file will be lost!\n";
     out << "****************************************************************************/\n";
