--- qt-x11-free-3.3.6-snapshot-20051225/src/tools/qunicodetables.cpp.orig	2005-12-25 00:51:28.000000000 +0000
+++ qt-x11-free-3.3.6-snapshot-20051225/src/tools/qunicodetables.cpp	2005-12-28 10:14:51.000000000 +0000
@@ -13136,7 +13136,7 @@
 #define SCRIPTS_19 52
     0x4f, Limbu, 0x7f, TaiLe, 0xdf, Unicode, 0xff, Khmer,
 #define SCRIPTS_20 60
-    0x0a, Unicode, 0x0d, UnknownScript, 0x6f, Unicode, 0x9f, NumberForms,
+    0x0a, Unicode, 0x0d, Han, 0x6f, Han, 0x9f, NumberForms,
     0xab, CurrencySymbols, 0xac, Latin,
     0xcf, CurrencySymbols, 0xff, CombiningMarks,		// row 0x20, index 50
 #define SCRIPTS_21 76
--- qt-x11-free-3.3.6-snapshot-20060220/src/kernel/qfontdatabase.cpp.orig	2006-02-20 09:28:46.000000000 +0800
+++ qt-x11-free-3.3.6-snapshot-20060220/src/kernel/qfontdatabase.cpp	2006-02-21 09:50:58.000000000 +0800
@@ -470,7 +470,7 @@
     // Ogham,
     { 0x1680, 0x0 },
     // SpacingModifiers,
-    { 0x02c6, 0x0 },
+    { 0x02c7, 0x0 },
     // CombiningMarks,
     { 0x0300, 0x0 },
 
@@ -542,13 +542,13 @@
 
     // Symbols
     // CurrencySymbols,
-    { 0x20aa, 0x0 },
+    { 0x20ac, 0x0 },
     // LetterlikeSymbols,
     { 0x2103, 0x0 },
     // NumberForms,
     { 0x2160, 0x0 },
     // MathematicalOperators,
-    { 0x222b, 0x0 },
+    { 0x2199, 0x0 },
     // TechnicalSymbols,
     { 0x2312, 0x0 },
     // GeometricSymbols,
@@ -574,7 +574,7 @@
     { 0x1770, 0x0 },
 
     // KatakanaHalfWidth
-    { 0xff65, 0x0 },
+    { 0xffe4, 0x0 },
 
     // Limbu
     { 0x1901, 0x0 },
