--- qt-copy/src/kernel/qapplication_x11.cpp.org	2004-08-23 13:30:58.843935240 +0800
+++ qt-copy/src/kernel/qapplication_x11.cpp	2004-08-23 13:33:02.059203672 +0800
@@ -1029,7 +1029,7 @@
     if (xim_preferred_style == 0) {
         QString ximInputStyle =
             settings.readEntry( "/qt/XIMInputStyle",
-                                QObject::trUtf8( "On The Spot" ) ).lower();
+                                QObject::trUtf8( "Over The Spot" ) ).lower();
         if ( ximInputStyle == "on the spot" )
             xim_preferred_style = XIMPreeditCallbacks | XIMStatusNothing;
         else if ( ximInputStyle == "over the spot" )
