diff -ur qt-x11-free-3.3.8d.orig/tools/qtconfig/mainwindow.cpp qt-x11-free-3.3.8d/tools/qtconfig/mainwindow.cpp
--- qt-x11-free-3.3.8d.orig/tools/qtconfig/mainwindow.cpp	2011-11-25 13:20:05.000000000 -0500
+++ qt-x11-free-3.3.8d/tools/qtconfig/mainwindow.cpp	2011-11-25 14:58:40.000000000 -0500
@@ -376,7 +376,7 @@
     rtlExtensions->setChecked( settings.readBoolEntry( "/qt/useRtlExtensions", FALSE ) );
 
 #if defined(Q_WS_X11)
-    inputStyle->setCurrentText( settings.readEntry( "/qt/XIMInputStyle", trUtf8( "On The Spot" ) ) );
+    inputStyle->setCurrentText( settings.readEntry( "/qt/XIMInputStyle", trUtf8( "Over The Spot" ) ) );
 #else
     inputStyle->hide();
     inputStyleLabel->hide();
@@ -510,9 +510,12 @@
 
 #ifdef Q_WS_X11
 	QString style = inputStyle->currentText();
-	QString str = "On The Spot";
-	if ( style == trUtf8( "Over The Spot" ) )
-	    str = "Over The Spot";
+	// QString str = "On The Spot";
+	QString str = "Over The Spot";
+	// if ( style == trUtf8( "Over The Spot" ) )
+	//     str = "Over The Spot";
+	if ( style == trUtf8( "On The Spot" ) )
+	    str = "On The Spot";
 	else if ( style == trUtf8( "Off The Spot" ) )
 	    str = "Off The Spot";
 	else if ( style == trUtf8( "Root" ) )
