diff -up gambas2-2.8.1/main/configure.ac.orig gambas2-2.8.1/main/configure.ac
--- gambas2-2.8.1/main/configure.ac.orig	2008-08-21 13:24:51.000000000 -0400
+++ gambas2-2.8.1/main/configure.ac	2008-08-21 13:31:29.000000000 -0400
@@ -27,8 +27,8 @@ GB_COMPONENT(
   [optional external charset conversion library],
   [],
   [GB_FIND(iconv.h, /usr/local /usr, include)],
-  [GB_FIND(libiconv.$SHLIBEXT, /usr/local /usr, lib)],
-  [-liconv],
+  [],
+  [],
   [],
   [This library may be located inside the system C library, so let's go on...])
 
