diff -ur gambas2-2.23.1.orig/gb.qt.kde/configure.ac gambas2-2.23.1/gb.qt.kde/configure.ac
--- gambas2-2.23.1.orig/gb.qt.kde/configure.ac	2011-11-28 07:43:16.000000000 -0500
+++ gambas2-2.23.1/gb.qt.kde/configure.ac	2011-11-28 09:13:08.000000000 -0500
@@ -28,7 +28,7 @@
   [src],
   [GB_FIND(kapplication.h, `kde-config --prefix 2> /dev/null`, include)],
   [GB_FIND(libkdecore.$SHLIBEXT, `kde-config --prefix 2> /dev/null`, lib)],
-  [-lkdecore -lkdeui -lDCOP -lkio])
+  [-ltdecore -ltdeui -lDCOP -lkio])
 
 fi
   
