--- gambas2-2.23.1/gb.pdf/src/CPdfDocument.cpp	2011-05-27 02:01:38.000000000 +0200
+++ gambas2-2.23.1/gb.pdf/src/CPdfDocument.cpp	2011-08-08 14:03:56.000000000 +0200
@@ -51,6 +51,7 @@
 #include <Outline.h>
 #include <Link.h>
 #include <Gfx.h>
+#include <glib/poppler-features.h>
 
 /*****************************************************************************
 
@@ -959,10 +960,14 @@ END_METHOD
 ******************************************************************************/
 void aux_fill_links(void *_object)
 {
+#if POPPLER_CHECK_VERSION(0, 17, 0)
+	THIS->links = new Links (THIS->page->getAnnots (THIS->doc->getCatalog()));
+#else
 	Object obj;
 	
 	THIS->links = new Links (THIS->page->getAnnots (&obj),THIS->doc->getCatalog()->getBaseURI ());
 	obj.free();
+#endif
 }
 
 BEGIN_PROPERTY (PDFPAGELINKS_count)
