diff -ur kdeaddons-3.5.13.orig/kate/tabbarextension/plugin_katetabbarextension.h kdeaddons-3.5.13/kate/tabbarextension/plugin_katetabbarextension.h
--- kdeaddons-3.5.13.orig/kate/tabbarextension/plugin_katetabbarextension.h	2010-12-07 11:26:57.000000000 -0500
+++ kdeaddons-3.5.13/kate/tabbarextension/plugin_katetabbarextension.h	2010-12-07 12:04:20.000000000 -0500
@@ -53,7 +53,7 @@
 class MyPtrList : public TQPtrList <KateTabBarButton>
 {
   public:
-  MyPtrList() { TQPtrList<KateTabBarButton>::TQPtrList(); }
+  MyPtrList() { TQPtrList(); }
 
   virtual int compareItems ( TQPtrCollection::Item item1_, TQPtrCollection::Item item2_ );
 };

