diff -u -r kdeaddons-3.5.6.orig/kfile-plugins/html/kfile_html.cpp kdeaddons-3.5.6/kfile-plugins/html/kfile_html.cpp
--- kdeaddons-3.5.6.orig/kfile-plugins/html/kfile_html.cpp	2007-01-26 17:03:47.000000000 +0800
+++ kdeaddons-3.5.6/kfile-plugins/html/kfile_html.cpp	2007-01-26 17:07:48.000000000 +0800
@@ -87,7 +87,7 @@
     if (exp.search(s, last) != -1)
     {
         kdDebug(7034) << "DocType: " << TQString(exp.capturedTexts().join("-")) << endl;
-        appendItem(group, "Doctype", exp.cap(1));
+        appendItem(group, "Doctype", TQString::fromLocal8Bit(exp.cap(1).latin1()));
         last += exp.matchedLength();
     }
 
@@ -95,7 +95,7 @@
     exp.setPattern("<\\s*title\\s*>\\s*(.*)\\s*<\\s*/\\s*title\\s*>");
     if (exp.search(s, last) != -1)
     {
-        title = exp.cap(1);
+        title = TQString::fromLocal8Bit(exp.cap(1).latin1());
         last += exp.matchedLength();
     }
 
@@ -114,8 +114,8 @@
     {
         if ((start=exp.search(s, last)) == -1)
             break;
-        meta = exp.cap(1);
-        last = start+exp.matchedLength();
+        meta = TQString::fromLocal8Bit(exp.cap(1).latin1());
+	last = start+exp.matchedLength();
 
         kdDebug(7034) << "Found Meta: " << meta << endl;
 
