diff -ur tdeaddons-git20111222.orig/admin/ltmain.sh tdeaddons-git20111222/admin/ltmain.sh
--- tdeaddons-git20111222.orig/admin/ltmain.sh	2011-12-22 08:29:08.000000000 -0500
+++ tdeaddons-git20111222/admin/ltmain.sh	2011-12-22 08:29:40.000000000 -0500
@@ -73,9 +73,9 @@
 
 PROGRAM=ltmain.sh
 PACKAGE=libtool
-VERSION="2.2.6b Debian-2.2.6b-2ubuntu1"
+VERSION="2.4"
 TIMESTAMP=""
-package_revision=1.3017
+package_revision=1.3293
 
 # Be Bourne compatible
 if test -n "${ZSH_VERSION+set}" && (emulate sh) >/dev/null 2>&1; then

