--- jemalloc-2.2.5/include/jemalloc/internal/atomic.h.orig	2012-04-21 10:50:06.000000000 -0400
+++ jemalloc-2.2.5/include/jemalloc/internal/atomic.h	2012-04-21 10:50:21.000000000 -0400
@@ -105,7 +105,9 @@
 
 /******************************************************************************/
 /* 32-bit operations. */
-#ifdef __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4
+#if ( defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4) \
+   || defined(__ARM_ARCH_5T__) \
+   || defined(__ARM_ARCH_5TE__) )
 JEMALLOC_INLINE uint32_t
 atomic_add_uint32(uint32_t *p, uint32_t x)
 {
