--- kdegraphics.ORI/kpdf/xpdf/aconf.h	2011-04-27 19:32:57.000000000 +0200
+++ kdegraphics/kpdf/xpdf/aconf.h	2011-10-30 14:54:24.000000000 +0100
@@ -7,7 +7,7 @@
 #define HAVE_FREETYPE_FREETYPE_H HAVE_FREETYPE
 #define OPI_SUPPORT 0
 #define TEXTOUT_WORD_LIST 0
-#define HAVE_MKSTEMPS 1 //libkdefakes provides it
+#define HAVE_MKSTEMPS 0
 #define HAVE_SPLASH 1
 #define SPLASH_CMYK 1
 #define HAVE_XPDFCORE 0
