--- kdegraphics/kfile-plugins/dependencies/poppler-tqt/CMakeLists.txt.ORI	2012-08-01 19:42:18.057554419 +0200
+++ kdegraphics/kfile-plugins/dependencies/poppler-tqt/CMakeLists.txt	2012-08-01 19:42:28.320371892 +0200
@@ -38,7 +38,7 @@
 
 install( FILES
     poppler-qt.h poppler-page-transition.h poppler-link-qt3.h
-  DESTINATION ${INCLUDE_INSTALL_DIR}/kde )
+  DESTINATION ${INCLUDE_INSTALL_DIR} )
 
 ##### poppler-tqt (shared) ############################
 
