--- ./configure.gstreamer	2006-04-09 23:08:39.000000000 +0200
+++ ./configure	2006-04-14 10:23:09.000000000 +0200
@@ -37922,7 +37922,7 @@
 # BEGIN DO_NOT_COMPILE CHECK
 ###############################################################################
 
-if test x$build_xine = xno -a x$build_helix = xno; then
+if test x$build_xine = xno -a x$build_helix = xno -a x$build_gstreamer10 = xno; then
 
     DO_NOT_COMPILE="$DO_NOT_COMPILE amarok"
     no_engine=yes
@@ -40316,7 +40316,7 @@
     echo " =================================="
     echo " ="
     echo " = No suitable multimedia framework was detected. You need to install at least"
-    echo " = the Xine or Helix framework as detailed in the amaroK README."
+    echo " = the Xine, Helix framework or GStreamer as detailed in the amaroK README."
     echo " ="
 fi
 
