--- amarok-1.4.8/amarok/src/konquisidebar/universalamarok.cpp	2007-12-17 23:53:19.000000000 +0800
+++ amarok-1.4.8/amarok/src/konquisidebar/universalamarok.cpp	2008-04-12 20:07:00.000000000 +0800
@@ -171,7 +171,7 @@
         map->insert ("Name", i18n ("Amarok"));
         map->insert ("Open", "true");
         map->insert ("X-KDE-KonqSidebarModule","konqsidebar_universalamarok");
-        fn->setLatin1 ("amarok.desktop");
+        fn->setLatin1 ("amarok%1.desktop");
         return true;
     }
 }
