diff -ur tdeedu-git20111222.orig/admin/ltmain.sh tdeedu-git20111222/admin/ltmain.sh
--- tdeedu-git20111222.orig/admin/ltmain.sh	2011-12-22 13:01:14.000000000 -0500
+++ tdeedu-git20111222/admin/ltmain.sh	2011-12-22 13:01:42.000000000 -0500
@@ -73,9 +73,9 @@
 
 PROGRAM=ltmain.sh
 PACKAGE=libtool
-VERSION="2.2.6b Debian-2.2.6b-2ubuntu1"
+VERSION="2.4"
 TIMESTAMP=""
-package_revision=1.3017
+package_revision=1.3293
 
 # Be Bourne compatible
 if test -n "${ZSH_VERSION+set}" && (emulate sh) >/dev/null 2>&1; then
