diff --git a/cmake/ConfigOptions.cmake b/cmake/ConfigOptions.cmake
index 87ffefe..7eacca5 100644
--- a/cmake/ConfigOptions.cmake
+++ b/cmake/ConfigOptions.cmake
@@ -6,6 +6,8 @@ elseif((CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|AMD64") AND (CMAKE_SIZEOF_VOID_P
 elseif((CMAKE_SYSTEM_PROCESSOR MATCHES "i386") AND (CMAKE_SIZEOF_VOID_P EQUAL 8) AND (APPLE))
 	# Mac is weird like that.
 	set(TARGET_ARCH "x64")
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+	set(TARGET_ARCH "AARCH64")
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "arm*")
 	set(TARGET_ARCH "ARM")
 endif()
@@ -20,6 +22,9 @@ else()
 	option(WITH_SSE2 "Enable SSE2 optimization." OFF)
 endif()
 
+if(TARGET_ARCH MATCHES "AARCH64")
+endif()
+
 if(TARGET_ARCH MATCHES "ARM")
 	if (NOT DEFINED WITH_NEON)
 		option(WITH_NEON "Enable NEON optimization." ON)
