diff -rcb k3b.ori/src/k3b.desktop k3b/src/k3b.desktop
*** k3b.ori/src/k3b.desktop	Wed Dec 15 14:23:05 2010
--- k3b/src/k3b.desktop	Wed Dec 15 14:28:55 2010
***************
*** 116,123 ****
  Name[ar]= K3b
  Name[bn]=কে-থ্রি-বি
  Name[hi]=के3बी
! MimeType=application/x-k3b;application/x-iso;application/x-cue
  X-KDE-StartupNotify=true
! X-KDE-NativeMimeType=application/x-k3b;application/x-iso;application/x-cue
! Categories=KDE;Application;AudioVideo;DiscBurning;
  X-DCOP-ServiceType=Unique
--- 116,124 ----
  Name[ar]= K3b
  Name[bn]=কে-থ্রি-বি
  Name[hi]=के3बी
! Encoding=UTF-8
! MimeType=application/x-k3b;application/x-iso;application/x-cue;
  X-KDE-StartupNotify=true
! X-KDE-NativeMimeType=application/x-k3b;application/x-iso;application/x-cue;
! Categories=KDE;AudioVideo;DiscBurning;
  X-DCOP-ServiceType=Unique
