From: Gerald Combs <gerald@wireshark.org>
Date: Fri, 15 Nov 2013 23:16:14 +0000
Subject: [PATCH] From Dirk Jagdmann: Make sure err_str is initialized.

svn path=/trunk-1.10/; revision=53348

diff --git a/ui/gtk/main_welcome.c b/ui/gtk/main_welcome.c
index 6de0fea..017f2e4 100644
--- a/ui/gtk/main_welcome.c
+++ b/ui/gtk/main_welcome.c
@@ -981,7 +981,7 @@ static void fill_capture_box(void)
     GtkCellRenderer   *renderer;
     GtkTreeViewColumn *column;
     int               error = 0;
-    gchar             *label_text, *err_str;
+    gchar             *label_text = NULL, *err_str = NULL;
 #ifdef _WIN32
     DWORD reg_ret;
     DWORD chimney_enabled = 0;
