diff -Nur lrcShow-II.orig/gui.py lrcShow-II/gui.py
--- gui.py	2007-12-23 12:46:57.000000000 +0800
+++ gui.py	2007-12-23 19:27:53.000000000 +0800
@@ -251,12 +251,12 @@
 		else:
 			self.proxy={'http':self.proxy}
 		if(self.ifDownload=='yes'):
-			self.childMenu.childMenu2.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/apply.png')),_(u'是'),1,1)
+			self.childMenu.childMenu2.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/apply.png')),_(u'是'),1,1)
 			self.childMenu.childMenu2.insertItem(_(u'否'),2,2)
 			
 		else:
 			self.childMenu.childMenu2.insertItem(_(u'是'),1,1)
-			self.childMenu.childMenu2.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/apply.png')),_(u'否'),2,2)
+			self.childMenu.childMenu2.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/apply.png')),_(u'否'),2,2)
 		self.childMenu.childMenu2.insertItem(_(u'仅当前'),3,3)
 		self.childMenu.childMenu2.setItemEnabled(3,False)
 		self.path=unicode(self.path,'utf8').encode(getdefaultlocale()[1])
@@ -276,7 +276,7 @@
 		self.setMinimumHeight(140)
 		self.setMinimumWidth(420)	
 		self.setFont(QFont(self.fontType,self.fontSize))
-		self.setIcon(QPixmap('../scripts/lrcShow-II/icon/logo.png'))
+		self.setIcon(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/logo.png'))
 		self.mytimer=QTimer()
 		self.connect(self.mytimer,SIGNAL("timeout()"),self.scroll)
 		self.connect(self.childBox.seekSlowButton,SIGNAL("clicked()"),self.seekSlow)
@@ -770,7 +770,7 @@
 				pass
 			else:
 				self.childMenu.childMenu2.clear()
-				self.childMenu.childMenu2.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/apply.png')),_(u'是'),1,1)
+				self.childMenu.childMenu2.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/apply.png')),_(u'是'),1,1)
 				self.childMenu.childMenu2.insertItem(_(u'否'),2,2)
 				self.childMenu.childMenu2.insertItem(_(u'仅当前'),3,3)
 				self.childMenu.childMenu2.setItemEnabled(3,False)
@@ -810,7 +810,7 @@
 			else:
 				self.childMenu.childMenu2.clear()
 				self.childMenu.childMenu2.insertItem(u'是',1,1)
-				self.childMenu.childMenu2.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/apply.png')),u'否',2,2)
+				self.childMenu.childMenu2.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/apply.png')),u'否',2,2)
 				self.childMenu.childMenu2.insertItem(u'仅当前',3,3)
 				if(self.originalLrc):
 					self.childMenu.childMenu2.setItemEnabled(3,True)
@@ -852,13 +852,13 @@
 				pass
 			else:
 				self.childMenu.childMenu3.clear()
-				self.childMenu.childMenu3.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/apply.png')),_(u'普通模式'),1,1)
+				self.childMenu.childMenu3.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/apply.png')),_(u'普通模式'),1,1)
 				self.childMenu.childMenu3.insertItem(_(u'全屏模式'),2,2)
 				self.childMenu.childMenu3.insertItem(_(u'OSD模式'),3,3)
 				self.childMenu.setItemEnabled(6,True)
 				if(self.showMode=='osd'):
 					self.reparent(None,self.WFlags,self.pos(),False)
-					self.setIcon(QPixmap('../scripts/lrcShow-II/icon/logo.png'))
+					self.setIcon(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/logo.png'))
 					self.show()
 				elif(self.showMode=='fullscreen'):
 					#self.setWindowState(Qt.WindowNoState)
@@ -876,7 +876,7 @@
 				self.childMenu.childMenu3.clear()
 				self.childMenu.childMenu3.insertItem(_(u'普通模式'),1,1)
 				self.childMenu.childMenu3.setAccel('Esc',1)
-				self.childMenu.childMenu3.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/apply.png')),_(u'全屏模式'),2,2)
+				self.childMenu.childMenu3.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/apply.png')),_(u'全屏模式'),2,2)
 				self.childMenu.childMenu3.insertItem(_(u'OSD模式'),3,3)
 				self.childMenu.childMenu3.setItemEnabled(3,False)
 				self.childMenu.setItemEnabled(6,False)
@@ -894,10 +894,10 @@
 				self.childMenu.childMenu3.insertItem(_(u'普通模式'),1,1)
 				self.childMenu.childMenu3.setAccel('Esc',1)
 				self.childMenu.childMenu3.insertItem(_(u'全屏模式'),2,2)
-				self.childMenu.childMenu3.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/apply.png')),_(u'OSD模式'),3,3)
+				self.childMenu.childMenu3.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/apply.png')),_(u'OSD模式'),3,3)
 				self.childMenu.childMenu3.setItemEnabled(2,False)
 				self.reparent(None,Qt.WStyle_Customize+Qt.WStyle_StaysOnTop,self.pos(),False)
-				self.setIcon(QPixmap('../scripts/lrcShow-II/icon/logo.png'))
+				self.setIcon(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/logo.png'))
 				self.show()
 				self.showMode='osd'
 	
@@ -943,20 +943,20 @@
 		
 		self.seekSlowButton=QPushButton(self,'seekSlowButton')
 		self.seekSlowButton.setGeometry(QRect(0,0,30,20))
-		self.seekSlowButton.setPixmap(QPixmap('../scripts/lrcShow-II/icon/backward.png'))
+		self.seekSlowButton.setPixmap(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/backward.png'))
 		self.seekSlowButton.setPaletteBackgroundColor(color)
 		QToolTip.add(self.seekSlowButton,_(u'歌词整体滞后0.2秒'))
 		
 		self.seekFastButton=QPushButton(self,'seekFastButton')
 		self.seekFastButton.setGeometry(QRect(30,0,30,20))
-		self.seekFastButton.setPixmap(QPixmap('../scripts/lrcShow-II/icon/forward.png'))
+		self.seekFastButton.setPixmap(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/forward.png'))
 		self.seekFastButton.setPaletteBackgroundColor(color)
 		QToolTip.add(self.seekFastButton,_(u'歌词整体提前0.2秒'))
 		
 		self.saveButton=QPushButton(self,'saveButton')
 		self.saveButton.setGeometry(QRect(60,0,30,20))
 		#self.saveButton.setText('保存'.decode('utf8'))
-		self.saveButton.setPixmap(QPixmap('../scripts/lrcShow-II/icon/save.png'))
+		self.saveButton.setPixmap(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/save.png'))
 		self.saveButton.setPaletteBackgroundColor(color)
 		self.saveButton.setEnabled(False)
 		QToolTip.add(self.saveButton,_(u'保存新偏移量至文件'))
@@ -964,14 +964,14 @@
 		self.transferButton=QPushButton(self,'transferButton')
 		self.transferButton.setGeometry(QRect(100,0,30,20))
 		#self.transferButton.setText(u'简繁')
-		self.transferButton.setPixmap(QPixmap('../scripts/lrcShow-II/icon/st.png'))
+		self.transferButton.setPixmap(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/st.png'))
 		self.transferButton.setPaletteBackgroundColor(color)
 		QToolTip.add(self.transferButton,_(u'简繁体快速转换'))
 		
 		self.editButton=QPushButton(self,'editButton')
 		self.editButton.setGeometry(QRect(140,0,30,20))
 		#self.editButton.setText(u'编辑')
-		self.editButton.setPixmap(QPixmap('../scripts/lrcShow-II/icon/edit.png'))
+		self.editButton.setPixmap(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/edit.png'))
 		self.editButton.setPaletteBackgroundColor(color)
 		QToolTip.add(self.editButton,_(u'开启lrcShow-II歌词编辑器'))
 
@@ -979,7 +979,7 @@
 	
 	def __init__(self,f,e,*args):
 		apply(QWidget.__init__,(self,)+args)
-		self.setIcon(QPixmap('../scripts/lrcShow-II/icon/logo.png'))
+		self.setIcon(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/logo.png'))
 		self.setCaption(_(u'lrcShow-II歌词编辑器'))
 		self.setMinimumSize(QSize(480,280))
 		layout = QVBoxLayout(self,2,6,"layout")
@@ -1099,37 +1099,37 @@
 		self.childMenu1=colorMenu(self)
 		self.childMenu2=QPopupMenu()
 		self.childMenu3=QPopupMenu()
-		self.childMenu3.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/apply.png')),_(u'普通模式'),1,1)
+		self.childMenu3.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/apply.png')),_(u'普通模式'),1,1)
 		self.childMenu3.insertItem(_(u'全屏模式'),2,2)
 		self.childMenu3.insertItem(_(u'OSD模式'),3,3)###
 		
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/backward.png')),_(u'歌词滞后0.2秒'),1,1)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/backward.png')),_(u'歌词滞后0.2秒'),1,1)
 		self.setAccel('-',1)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/forward.png')),_(u'歌词提前0.2秒'),2,2)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/forward.png')),_(u'歌词提前0.2秒'),2,2)
 		self.setAccel('+',2)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/save.png')),_(u'保存偏移量'),3,3)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/colors.png')),_(u'设置配色'),self.childMenu1,4,4)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/font.png')),_(u'设置字体...'),5,5)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/line.png')),_(u'设置显示行数...'),6,6)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/st.png')),_(u'简繁体互转'),7,7)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/folder.png')),_(u'设置本地歌词路径...'),8,8)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/download.png')),_(u'是否从网络下载'),self.childMenu2,9,9)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/screen.png')),_(u'设置显示模式'),self.childMenu3,10,10)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/proxy.png')),_(u'设置网络代理...'),11,11)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/edit.png')),_(u'歌词编辑器...'),12,12)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/engine.png')),_(u'设置搜索引擎...'),13,13)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/search.png')),_(u'从网络搜索'),14,14)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/locale.png')),_(u'设置本地歌词编码...'),15,15)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/help.png')),_(u'查看帮助...'),16,16)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/logo.png')),_(u'关于lrcShow-II...'),17,17)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/exit.png')),_(u'退出lrcShow-II'),18,18)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/save.png')),_(u'保存偏移量'),3,3)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/colors.png')),_(u'设置配色'),self.childMenu1,4,4)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/font.png')),_(u'设置字体...'),5,5)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/line.png')),_(u'设置显示行数...'),6,6)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/st.png')),_(u'简繁体互转'),7,7)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/folder.png')),_(u'设置本地歌词路径...'),8,8)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/download.png')),_(u'是否从网络下载'),self.childMenu2,9,9)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/screen.png')),_(u'设置显示模式'),self.childMenu3,10,10)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/proxy.png')),_(u'设置网络代理...'),11,11)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/edit.png')),_(u'歌词编辑器...'),12,12)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/engine.png')),_(u'设置搜索引擎...'),13,13)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/search.png')),_(u'从网络搜索'),14,14)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/locale.png')),_(u'设置本地歌词编码...'),15,15)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/help.png')),_(u'查看帮助...'),16,16)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/logo.png')),_(u'关于lrcShow-II...'),17,17)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/exit.png')),_(u'退出lrcShow-II'),18,18)
 		self.setItemEnabled(3,False)
 		
 class colorMenu(QPopupMenu):
 	
 	def __init__(self,*args):
 		apply(QPopupMenu.__init__,(self,)+args)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/background.png')),_(u'修改背景色...'),1,1)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/fontcolor.png')),_(u'修改字体色...'),2,2)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/highlight.png')),_(u'修改标亮色...'),3,3)
-		self.insertItem(QIconSet(QPixmap('../scripts/lrcShow-II/icon/undo.png')),_(u'返回默认配色'),4,4)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/background.png')),_(u'修改背景色...'),1,1)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/fontcolor.png')),_(u'修改字体色...'),2,2)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/highlight.png')),_(u'修改标亮色...'),3,3)
+		self.insertItem(QIconSet(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/undo.png')),_(u'返回默认配色'),4,4)
diff -Nur lrcShow-II.orig/helpBrowser.py lrcShow-II/helpBrowser.py
--- helpBrowser.py	2007-12-20 20:17:27.000000000 +0800
+++ helpBrowser.py	2007-12-23 19:28:41.000000000 +0800
@@ -12,7 +12,7 @@
 	def __init__(self,*args):
         	apply(QDialog.__init__,(self,) + args)
 		self.setCaption(_(u'lrcShow-II文档查看器'))
-		self.setIcon(QPixmap('../scripts/lrcShow-II/icon/help.png'))
+		self.setIcon(QPixmap('/usr/share/apps/amarok/scripts/lrcShow-II/icon/help.png'))
 		self.label=QLabel(self)
 		self.label.setText(_(u'请选择相关条目：'))
 		self.contentBro=browserView(self)
@@ -31,11 +31,11 @@
 	def moveTo(self,f):
 		self.contentBro.clear()
 		if(f==0):
-			fileName='../scripts/lrcShow-II/document/document-zh.html'
+			fileName='/usr/share/apps/amarok/scripts/lrcShow-II/document/document-zh.html'
 		elif(f==1):
-			fileName='../scripts/lrcShow-II/document/document-en.html'
+			fileName='/usr/share/apps/amarok/scripts/lrcShow-II/document/document-en.html'
 		elif(f==2):
-			fileName='../scripts/lrcShow-II/document/COPYING'
+			fileName='/usr/share/apps/amarok/scripts/lrcShow-II/document/COPYING'
 		file=open(fileName)
 		a=file.read()
 		file.close()
@@ -56,7 +56,7 @@
         	apply(QTextBrowser.__init__,(self,) + args)
 		self.setWrapPolicy(QTextEdit.AtWordBoundary)
 		self.setHScrollBarMode(QTextEdit.AlwaysOff)
-		file=open('../scripts/lrcShow-II/document/document-zh.html')
+		file=open('/usr/share/apps/amarok/scripts/lrcShow-II/document/document-zh.html')
 		a=file.read()
 		file.close()
 		self.setText(a.decode('utf8'))
diff -Nur lrcShow-II.orig/lrcShow-II.py lrcShow-II/lrcShow-II.py
--- lrcShow-II.py	2007-12-22 23:29:01.000000000 +0800
+++ lrcShow-II.py	2007-12-23 19:29:08.000000000 +0800
@@ -37,9 +37,9 @@
 
 if __name__=="__main__":
 	try:
-		gettext.translation('lrcShow-II','../scripts/lrcShow-II/locale').install(unicode=True)
+		gettext.translation('lrcShow-II','/usr/share/apps/amarok/scripts/lrcShow-II/locale').install(unicode=True)
 	except IOError:
-		gettext.translation('lrcShow-II','../scripts/lrcShow-II/locale',languages=['en_US']).install(unicode=True)
+		gettext.translation('lrcShow-II','/usr/share/apps/amarok/scripts/lrcShow-II/locale',languages=['en_US']).install(unicode=True)
 	
 	if(os.path.exists('lrcShow-II.conf')):
 		pass
diff -Nur lrcShow-II.orig/s2t.py lrcShow-II/s2t.py
--- s2t.py	2007-12-20 20:17:27.000000000 +0800
+++ s2t.py	2007-12-23 19:29:35.000000000 +0800
@@ -11,10 +11,10 @@
 		self.string = string
 	
 	def transfer(self):
-		fileS = open('../scripts/lrcShow-II/dict/simplified')
+		fileS = open('/usr/share/apps/amarok/scripts/lrcShow-II/dict/simplified')
 		dictS = fileS.read()
 		fileS.close()
-		fileT = open('../scripts/lrcShow-II/dict/traditional')
+		fileT = open('/usr/share/apps/amarok/scripts/lrcShow-II/dict/traditional')
 		dictT = fileT.read()
 		fileT.close()
 		
