--- qtbase-opensource-src-5.3.0/mkspecs/features/qt_module_pris.prf.orig	2014-08-01 10:31:38.994403099 +0000
+++ qtbase-opensource-src-5.3.0/mkspecs/features/qt_module_pris.prf	2014-08-01 10:33:11.034489286 +0000
@@ -177,9 +177,8 @@
         include($$mod_work_pfx/qt_lib_$${pri}.pri)
     for(mod, mods_to_load) {
         for(var, $$list(VERSION MAJOR_VERSION MINOR_VERSION PATCH_VERSION \
-                        name depends module_config CONFIG DEFINES sources \
+                        name depends run_depends plugin_types module_config CONFIG DEFINES \
                         includes bins libs libexecs plugins imports qml \
-                        rpath_link \
                 )):defined(QT.$${mod}.$$var, var):cache(QT.$${mod}.$$var, transient)
     }
     cache(QT_MODULES, transient)
