diff -Nuarp konversation-1.2.1/src/dcc/chat.cpp konversation-1.2.1.utf8_default/src/dcc/chat.cpp
--- konversation-1.2.1/src/dcc/chat.cpp	2009-11-12 13:24:42.000000000 +0800
+++ konversation-1.2.1.utf8_default/src/dcc/chat.cpp	2009-12-29 20:16:04.000000000 +0800
@@ -48,7 +48,7 @@ namespace Konversation
             // set default values
             m_reverse = Preferences::self()->dccPassiveSend();

-            setEncoding(Konversation::IRCCharsets::self()->encodingForLocale());
+            setEncoding(QString("UTF-8"));
         }

         Chat::~Chat()
diff -Nuarp konversation-1.2.1/src/identity.cpp konversation-1.2.1.utf8_default/src/identity.cpp
--- konversation-1.2.1/src/identity.cpp	2009-11-12 13:24:49.000000000 +0800
+++ konversation-1.2.1.utf8_default/src/identity.cpp	2009-12-29 20:17:44.000000000 +0800
@@ -55,7 +55,7 @@ Identity::~Identity()
 
 void Identity::init()
 {
-    setCodecName(Konversation::IRCCharsets::self()->encodingForLocale());
+    setCodecName(QString("UTF-8")/*Konversation::IRCCharsets::self()->encodingForLocale()*/);
 
     setInsertRememberLineOnAway(false);
 
@@ -161,7 +161,7 @@ void Identity::setCodecName(const QStrin
     // never set an empty or borked codec!
     QString codecName=newCodecName;
     if(!Konversation::IRCCharsets::self()->isValidEncoding(codecName))
-        codecName=Konversation::IRCCharsets::self()->encodingForLocale();
+        codecName=QString("UTF-8")/*Konversation::IRCCharsets::self()->encodingForLocale()*/;
 
     m_codecName=codecName;
     m_codec=Konversation::IRCCharsets::self()->codecForName(codecName);
 
