--- metacity-2.30.0/src/Makefile.am	2010-04-20 00:56:08.110849725 -0400
+++ hacked2/src/Makefile.am	2010-04-20 01:03:18.002854963 -0400
@@ -192,10 +192,12 @@
 install-data-local:
 endif
 
-IMAGES=stock_maximize.png stock_minimize.png stock_delete.png
+IMAGES=stock_maximize.png stock_minimize.png stock_delete.png window.png mini-window.png
 VARIABLES=stock_maximize_data $(srcdir)/stock_maximize.png \
           stock_minimize_data $(srcdir)/stock_minimize.png \
-          stock_delete_data $(srcdir)/stock_delete.png
+          stock_delete_data $(srcdir)/stock_delete.png \
+          mini_window_data $(srcdir)/mini-window.png \
+          window_data $(srcdir)/window.png
 
 BUILT_SOURCES = inlinepixbufs.h
 CLEANFILES = inlinepixbufs.h metacity.desktop metacity-wm.desktop metacity.schemas metacity.schemas.in 50-metacity-desktop-key.xml 50-metacity-key.xml
--- metacity-2.30.0/src/ui/ui.c	2010-04-20 00:56:08.190599271 -0400
+++ hacked2/src/ui/ui.c	2010-04-20 01:04:24.220853635 -0400
@@ -588,10 +588,8 @@
                                                    0,
                                                    NULL);
       else
-          default_icon = gtk_icon_theme_load_icon (theme,
-                                                   "gtk-missing-image",
-                                                   META_ICON_WIDTH,
-                                                   0,
+        default_icon = gdk_pixbuf_new_from_inline (-1, window_data,
+                                                   FALSE,
                                                    NULL);
 
       g_assert (default_icon);
@@ -623,10 +621,8 @@
                                                    0,
                                                    NULL);
       else
-          default_icon = gtk_icon_theme_load_icon (theme,
-                                                   "gtk-missing-image",
-                                                   META_MINI_ICON_WIDTH,
-                                                   0,
+        default_icon = gdk_pixbuf_new_from_inline (-1, mini_window_data,
+                                                   FALSE,
                                                    NULL);
 
       g_assert (default_icon);
