---
 multipathd/main.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

Index: multipath-tools-130222/multipathd/main.c
===================================================================
--- multipath-tools-130222.orig/multipathd/main.c
+++ multipath-tools-130222/multipathd/main.c
@@ -537,7 +537,8 @@ rescan:
 			goto fail_map;
 
 	if (retries >= 0) {
-		condlog(2, "%s path added to devmap %s", pp->dev, mpp->alias);
+		condlog(2, "%s [%s]: path added to devmap %s",
+			pp->dev, pp->dev_t, mpp->alias);
 		return 0;
 	}
 	else
@@ -642,8 +643,8 @@ ev_remove_path (struct path *pp, struct
 			}
 			sync_map_state(mpp);
 
-			condlog(2, "%s: path removed from map %s",
-				pp->dev, mpp->alias);
+			condlog(2, "%s [%s]: path removed from map %s",
+				pp->dev, pp->dev_t, mpp->alias);
 		}
 	}
 
