---
 multipathd/main.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

Index: multipath-tools-130222/multipathd/main.c
===================================================================
--- multipath-tools-130222.orig/multipathd/main.c
+++ multipath-tools-130222/multipathd/main.c
@@ -1226,11 +1226,10 @@ check_path (struct vectors * vecs, struc
 					pp->checkint = 2 * pp->checkint;
 				else
 					pp->checkint = conf->max_checkint;
-
-				pp->tick = pp->checkint;
-				condlog(4, "%s: delay next check %is",
-					pp->dev_t, pp->tick);
 			}
+			pp->tick = pp->checkint;
+			condlog(4, "%s: delay next check %is",
+				pp->dev_t, pp->tick);
 		}
 	}
 	else if (newstate == PATH_DOWN) {
