---
 libmultipath/dict.c |    4 ----
 1 file changed, 4 deletions(-)

Index: multipath-tools-130222/libmultipath/dict.c
===================================================================
--- multipath-tools-130222.orig/libmultipath/dict.c
+++ multipath-tools-130222/libmultipath/dict.c
@@ -2468,16 +2468,12 @@ snprint_def_verbosity (char * buff, int
 static int
 snprint_def_max_polling_interval (char * buff, int len, void * data)
 {
-	if (conf->max_checkint == MAX_CHECKINT(conf->checkint))
-		return 0;
 	return snprintf(buff, len, "%i", conf->max_checkint);
 }
 
 static int
 snprint_reassign_maps (char * buff, int len, void * data)
 {
-	if (conf->reassign_maps == DEFAULT_REASSIGN_MAPS)
-		return 0;
 	return snprintf(buff, len, "\"%s\"",
 			conf->reassign_maps?"yes":"no");
 }
