---
 libmultipath/dict.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Index: multipath-tools-130222/libmultipath/dict.c
===================================================================
--- multipath-tools-130222.orig/libmultipath/dict.c
+++ multipath-tools-130222/libmultipath/dict.c
@@ -1126,11 +1126,11 @@ hw_failback_handler(vector strvec)
 
 	buff = set_value(strvec);
 
-	if (strlen(buff) == 6 && !strcmp(buff, "\"manual\""))
+       if (strlen(buff) == 6 && !strcmp(buff, "manual"))
 		hwe->pgfailback = -FAILBACK_MANUAL;
-	else if (strlen(buff) == 9 && !strcmp(buff, "\"immediate\""))
+       else if (strlen(buff) == 9 && !strcmp(buff, "immediate"))
 		hwe->pgfailback = -FAILBACK_IMMEDIATE;
-	else if (strlen(buff) == 10 && !strcmp(buff, "\"followover\""))
+       else if (strlen(buff) == 10 && !strcmp(buff, "followover"))
 		hwe->pgfailback = -FAILBACK_FOLLOWOVER;
 	else
 		hwe->pgfailback = atoi(buff);
