---
 kpartx/dasd.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: multipath-tools-130222/kpartx/dasd.c
===================================================================
--- multipath-tools-130222.orig/kpartx/dasd.c
+++ multipath-tools-130222/kpartx/dasd.c
@@ -46,6 +46,8 @@ unsigned long long sectors512(unsigned l
 	return sectors * (blocksize >> 9);
 }
 
+typedef unsigned int __attribute__((__may_alias__)) label_ints_t;
+
 /*
  */
 int 
@@ -169,7 +171,7 @@ read_dasd_pt(int fd, struct slice all, s
 		/*
 		 * VM style CMS1 labeled disk
 		 */
-		unsigned int *label = (unsigned int *) &vlabel;
+		label_ints_t *label = (label_ints_t *) &vlabel;
 
 		blocksize = label[4];
 		if (label[14] != 0) {
