diff -ur libmpeg3-1.8.orig/mpeg3tocutil.c libmpeg3-1.8/mpeg3tocutil.c
--- libmpeg3-1.8.orig/mpeg3tocutil.c	2014-07-21 17:56:08.715933427 +0000
+++ libmpeg3-1.8/mpeg3tocutil.c	2014-07-21 17:57:16.438931691 +0000
@@ -1278,7 +1278,7 @@
 
 // Store file information
 	PUT_INT32(FILE_INFO);
-	fprintf(file->toc_fd, file->fs->path);
+	fprintf(file->toc_fd, "%s", file->fs->path);
 	for(j = strlen(file->fs->path); j < MPEG3_STRLEN; j++)
 			fputc(0, file->toc_fd);
 	PUT_INT64(file->source_date);
@@ -1309,7 +1309,7 @@
 // Path
 		PUT_INT32(TITLE_PATH);
 
-		fprintf(file->toc_fd, title->fs->path);
+		fprintf(file->toc_fd, "%s", title->fs->path);
 
 // Pad path with 0
 		for(j = strlen(title->fs->path); j < MPEG3_STRLEN; j++)
