diff -up kdepim/akregator/src/akregator.kcfg~ kdepim/akregator/src/akregator.kcfg
--- kdepim/akregator/src/akregator.kcfg~	2006-11-30 11:13:45.000000000 +0200
+++ kdepim/akregator/src/akregator.kcfg	2007-10-02 21:54:41.000000000 +0300
@@ -154,7 +154,7 @@
   </entry>
   <entry key="External Browser Custom Command" type="String" >
    <whatsthis>Command to launch external browser.  URL will substitute for %u.</whatsthis>
-   <default>firefox %u</default>
+   <default>xdg-open %u</default>
   </entry>
   <entry key="LMB Behaviour" type="Enum" >
    <whatsthis>What the click with left mouse button should do.</whatsthis>
diff -up kdepim/akregator/src/settings_browser.ui~ kdepim/akregator/src/settings_browser.ui
--- kdepim/akregator/src/settings_browser.ui~	2005-11-19 12:38:13.000000000 +0200
+++ kdepim/akregator/src/settings_browser.ui	2007-10-02 21:54:32.000000000 +0300
@@ -64,7 +64,7 @@
                         <bool>false</bool>
                     </property>
                     <property name="text">
-                        <string>firefox %u</string>
+                        <string>xdg-open %u</string>
                     </property>
                 </widget>
             </grid>
