diff -u -r kdepim-3.5.10.orig/configure.in.in kdepim-3.5.10/configure.in.in
--- kdepim-3.5.10.orig/configure.in.in	2008-09-28 23:10:26.000000000 +0800
+++ kdepim-3.5.10/configure.in.in	2008-09-28 23:16:15.000000000 +0800
@@ -45,8 +45,8 @@
        [ LIB_GNOKII="-lgnokii $LIB_BLUETOOTH" ],
        [
     	AC_CHECK_LIB(gnokii, gn_cfg_get,
-       	[ LIB_GNOKII="-lgnokii -lintl $LIB_BLUETOOTH" ],
-       	[], $all_libraries -lintl -lXpm -lX11 $LIB_BLUETOOTH )
+       	[ LIB_GNOKII="-lgnokii $LIB_BLUETOOTH" ],
+       	[], $all_libraries -lXpm -lX11 $LIB_BLUETOOTH )
        ],
        $all_libraries -lXpm -lX11 $LIB_BLUETOOTH )
   AC_SUBST(LIB_GNOKII)
