diff -up jing-trang-20091111/mod/schematron/src/main/com/thaiopensource/validate/schematron/NewSaxonSchemaReaderFactory.java~ jing-trang-20091111/mod/schematron/src/main/com/thaiopensource/validate/schematron/NewSaxonSchemaReaderFactory.java
--- jing-trang-20091111/mod/schematron/src/main/com/thaiopensource/validate/schematron/NewSaxonSchemaReaderFactory.java~	2008-11-07 04:49:44.000000000 +0200
+++ jing-trang-20091111/mod/schematron/src/main/com/thaiopensource/validate/schematron/NewSaxonSchemaReaderFactory.java	2011-06-24 11:58:33.065604560 +0300
@@ -1,6 +1,6 @@
 package com.thaiopensource.validate.schematron;
 
-import net.sf.saxon.FeatureKeys;
+import net.sf.saxon.lib.FeatureKeys;
 import net.sf.saxon.TransformerFactoryImpl;
 
 import javax.xml.transform.TransformerFactory;
@@ -12,6 +12,7 @@ public class NewSaxonSchemaReaderFactory
   }
 
   public void initTransformerFactory(TransformerFactory factory) {
+    factory.setAttribute(FeatureKeys.XSLT_VERSION, "2.0");
     factory.setAttribute(FeatureKeys.LINE_NUMBERING, Boolean.TRUE);
     factory.setAttribute(FeatureKeys.VERSION_WARNING, Boolean.FALSE);
   }
