diff -up ./openjdk-boot/jdk/make/sun/security/pkcs11/Makefile.sav ./openjdk-boot/jdk/make/sun/security/pkcs11/Makefile
--- ./openjdk-boot/jdk/make/sun/security/pkcs11/Makefile.sav	2011-07-06 15:46:38.437692753 -0400
+++ ./openjdk-boot/jdk/make/sun/security/pkcs11/Makefile	2011-07-06 15:46:57.772418577 -0400
@@ -147,7 +147,7 @@ OTHER_INCLUDES += \
 # Rules
 #
 CLASSDESTDIR = $(TEMPDIR)/classes
-JAVAHFLAGS += -Xbootclasspath/p:$(CLASSDESTDIR)
+JAVAHFLAGS += -classpath $(CLASSDESTDIR)
 
 include $(BUILDDIR)/common/Mapfile-vers.gmk
 
