diff --git a/configure.ac b/configure.ac
index db18c8d..6a79194 100644
--- a/configure.ac
+++ b/configure.ac
@@ -219,6 +219,8 @@ if test "${srtp_prefix}" != "none" ; then
 
 	if test "$have_srtp_headers$have_srtp_lib" = "yesyes" ; then
 		have_srtp=yes
+		AC_CHECK_LIB(srtp,sha1_update,[
+			AC_MSG_ERROR([This libsrtp version exports symbols conflicting with other libraries. Please use the one from git://git.linphone.org/srtp.git])], have_srtp=yes)
 		AC_DEFINE(HAVE_SRTP, 1, [Defined when srtp support is compiled])
 	else
 		AC_MSG_NOTICE([Could not find libsrtp headers or lib, cryto transport disabled.])
