diff -ur k3bmonkeyaudioplugin-3.1.orig/src/libmonkeyaudio/APELink.cpp k3bmonkeyaudioplugin-3.1/src/libmonkeyaudio/APELink.cpp
--- k3bmonkeyaudioplugin-3.1.orig/src/libmonkeyaudio/APELink.cpp	2009-05-18 18:17:27.000000000 +0800
+++ k3bmonkeyaudioplugin-3.1/src/libmonkeyaudio/APELink.cpp	2009-05-18 18:17:56.000000000 +0800
@@ -63,10 +63,10 @@
     if (pData != NULL)
     {
         // parse out the information
-        char * pHeader = strstr(pData, APE_LINK_HEADER);
-        char * pImageFile = strstr(pData, APE_LINK_IMAGE_FILE_TAG);
-        char * pStartBlock = strstr(pData, APE_LINK_START_BLOCK_TAG);
-        char * pFinishBlock = strstr(pData, APE_LINK_FINISH_BLOCK_TAG);
+        char * pHeader = (char *)strstr(pData, APE_LINK_HEADER);
+        char * pImageFile = (char *)strstr(pData, APE_LINK_IMAGE_FILE_TAG);
+        char * pStartBlock = (char *)strstr(pData, APE_LINK_START_BLOCK_TAG);
+        char * pFinishBlock = (char *)strstr(pData, APE_LINK_FINISH_BLOCK_TAG);
 
         if (pHeader && pImageFile && pStartBlock && pFinishBlock)
         {
