diff -ur faac-1.28.orig/common/mp4v2/mpeg4ip.h faac-1.28/common/mp4v2/mpeg4ip.h
--- faac-1.28.orig/common/mp4v2/mpeg4ip.h	2012-12-19 13:53:13.000000000 +0000
+++ faac-1.28/common/mp4v2/mpeg4ip.h	2012-12-19 13:54:04.000000000 +0000
@@ -121,9 +121,9 @@
 #include <sys/param.h>
 
 #ifdef __cplusplus
-extern "C" {
+extern "C++" {
 #endif
-char *strcasestr(const char *haystack, const char *needle);
+const char *strcasestr(const char *haystack, const char *needle);
 #ifdef __cplusplus
 }
 #endif
