From b35fcb64c3ba3df95f62d0d129adb791ce2bad15 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Vr=C3=A1til?= <dvratil@redhat.com>
Date: Fri, 19 Sep 2014 11:10:13 +0200
Subject: [PATCH 08/30] Suppress unused variable warnings in release mode

---
 server/tests/unittest/collectionreferencetest.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/server/tests/unittest/collectionreferencetest.cpp b/server/tests/unittest/collectionreferencetest.cpp
index 1b10c55..9c98f28 100644
--- a/server/tests/unittest/collectionreferencetest.cpp
+++ b/server/tests/unittest/collectionreferencetest.cpp
@@ -47,6 +47,7 @@ public:
         res.setName(QLatin1String(name));
         const bool success = res.insert();
         Q_ASSERT(success);
+        Q_UNUSED(success);
         mResource = res;
         return res;
     }
@@ -60,6 +61,7 @@ public:
         col.setResource(mResource);
         const bool success = col.insert();
         Q_ASSERT(success);
+        Q_UNUSED(success);
         return col;
     }
 
-- 
2.1.0

