From c23607679fa1451f0c6890bd4a5656c07d519853 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Vr=C3=A1til?= <dvratil@redhat.com>
Date: Fri, 6 Feb 2015 13:33:50 +0100
Subject: [PATCH 31/34] Less C++11, fixes build with clang

---
 server/tests/unittest/searchtest.cpp | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/server/tests/unittest/searchtest.cpp b/server/tests/unittest/searchtest.cpp
index f523b09..969d296 100644
--- a/server/tests/unittest/searchtest.cpp
+++ b/server/tests/unittest/searchtest.cpp
@@ -119,21 +119,21 @@ private Q_SLOTS:
         QTest::addColumn<QStringList>("mimetypes");
         QTest::addColumn<QVector<qint64>>("expectedResults");
 
-        QTest::newRow("") << QVector<qint64>({ 0 })
-                          << QStringList({ QLatin1String("text/plain") })
-                          << QVector<qint64>({ col4.id(), col5.id(), col7.id() });
-        QTest::newRow("") << QVector<qint64>({ 0 })
-                          << QStringList({ QLatin1String("application/octet-stream") })
-                          << QVector<qint64>({ col2.id(), col3.id(), col6.id(), col8.id() });
-        QTest::newRow("") << QVector<qint64>({ col1.id() })
-                          << QStringList({ QLatin1String("text/plain") })
-                          << QVector<qint64>({ col4.id() });
-        QTest::newRow("") << QVector<qint64>({ col1.id() })
-                          << QStringList({ QLatin1String("unique/mime-type") })
+        QTest::newRow("") << (QVector<qint64>() << 0)
+                          << (QStringList() << QLatin1String("text/plain"))
+                          << (QVector<qint64>() << col4.id() << col5.id() << col7.id());
+        QTest::newRow("") << (QVector<qint64>() << 0)
+                          << (QStringList() << QLatin1String("application/octet-stream"))
+                          << (QVector<qint64>() << col2.id() << col3.id() << col6.id() << col8.id());
+        QTest::newRow("") << (QVector<qint64>() << col1.id())
+                          << (QStringList() << QLatin1String("text/plain"))
+                          << (QVector<qint64>() << col4.id());
+        QTest::newRow("") << (QVector<qint64>() << col1.id())
+                          << (QStringList() << QLatin1String("unique/mime-type"))
                           << QVector<qint64>();
-        QTest::newRow("") << QVector<qint64>({ col2.id(), col7.id() })
-                          << QStringList({ QLatin1String("application/octet-stream") })
-                          << QVector<qint64>({ col3.id(), col8.id() });
+        QTest::newRow("") << (QVector<qint64>() << col2.id() << col7.id())
+                          << (QStringList() << QLatin1String("application/octet-stream"))
+                          << (QVector<qint64>() << col3.id() << col8.id());
     }
 
     void testSearchHelperCollectionListing()
-- 
2.4.3

