From 18ed37d89b8185ac15a8bfe245de8a88d17f2c64 Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Tue, 28 Jul 2015 12:47:44 +0200
Subject: [PATCH 34/34] set cmake_min_req to match kdelibs4 and enable newer
 policies

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2d790c9..a64e724 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
+cmake_minimum_required(VERSION 2.8.9)
 project(Akonadi)
 
-cmake_minimum_required(VERSION 2.8.8 FATAL_ERROR)
 
 # where to look first for cmake modules, before ${CMAKE_ROOT}/Modules/ is checked
 set(CMAKE_MODULE_PATH "${Akonadi_SOURCE_DIR}/cmake/modules")
-- 
2.4.3

