--- UFconfig/UFconfig.mk	2011-01-26 14:49:41.000000000 -0500
+++ UFconfig/UFconfig.mk.new	2011-02-04 10:12:02.201378469 -0500
@@ -49,7 +49,7 @@
 MV = mv -f
 
 # Fortran compiler (not normally required)
-F77 = f77
+F77 = gfortran
 F77FLAGS = -O
 F77LIB =
 
@@ -91,8 +91,8 @@
 # naming the BLAS and LAPACK library (*.a or *.so) files.
 
 # This is probably slow ... it might connect to the Standard Reference BLAS:
-BLAS = -lblas -lgfortran
-LAPACK = -llapack
+#BLAS = -lblas -lgfortran
+#LAPACK = -llapack
 
 # NOTE: this next option for the "Goto BLAS" has nothing to do with a "goto"
 # statement.  Rather, the Goto BLAS is written by Dr. Kazushige Goto.
@@ -106,6 +106,10 @@
 # BLAS = -lblas_plain -lgfortran -lgfortranbegin
 # LAPACK = -llapack
 
+# Using ATLAS optimized BLAS and Lapack:  Fedora specific addition
+BLAS = -lcblas 
+LAPACK = -lclapack
+
 # The BLAS might not contain xerbla, an error-handling routine for LAPACK and
 # the BLAS.  Also, the standard xerbla requires the Fortran I/O library, and
 # stops the application program if an error occurs.  A C version of xerbla
@@ -156,7 +160,7 @@
 # -DNRECIPROCAL	do not multiply by the reciprocal
 # -DNO_DIVIDE_BY_ZERO	do not divide by zero
 
-UMFPACK_CONFIG = 
+UMFPACK_CONFIG = -DNPARTITION
 
 #------------------------------------------------------------------------------
 # CHOLMOD configuration
@@ -194,7 +198,7 @@
 # -DNSUNPERF	    for Solaris only.  If defined, do not use the Sun
 #			Performance Library
 
-CHOLMOD_CONFIG =
+CHOLMOD_CONFIG = -DNPARTITION
 
 #------------------------------------------------------------------------------
 # SuiteSparseQR configuration:
@@ -208,7 +212,7 @@
 # -DHAVE_TBB        enable the use of Intel's Threading Building Blocks (TBB)
 
 # default, without timing, without TBB:
-SPQR_CONFIG =
+SPQR_CONFIG = -DNPARTITION
 # with timing and TBB:
 # SPQR_CONFIG = -DTIMING -DHAVE_TBB
 # with timing
@@ -280,6 +284,10 @@
 # F77 = gfortran
 # BLAS = -lgoto -lgfortran
 
+# Fedora, with ATLAS Blas
+F77 = gfortran
+BLAS = -lf77blas -lgfortran
+
 #------------------------------------------------------------------------------
 # Mac
 #------------------------------------------------------------------------------
