diff --git a/sshd.c b/sshd.c
index 8dcfdf2..95b63ad 100644
--- a/sshd.c
+++ b/sshd.c
@@ -1592,6 +1592,10 @@ main(int ac, char **av)
 	parse_server_config(&options, rexeced_flag ? "rexec" : config_file_name,
 	    &cfg, NULL, NULL, NULL);
 
+	/* 'UsePAM no' is not supported in Fedora */
+	if (! options.use_pam)
+		logit("WARNING: 'UsePAM no' is not supported in Fedora and may cause several problems.");
+
 	seed_rng();
 
 	/* Fill in default values for those options not explicitly set. */
diff --git a/sshd_config b/sshd_config
index 8c16754..9f28b04 100644
--- a/sshd_config
+++ b/sshd_config
@@ -92,6 +92,8 @@ GSSAPICleanupCredentials yes
 # If you just want the PAM account and session checks to run without
 # PAM authentication, then enable this but set PasswordAuthentication
 # and ChallengeResponseAuthentication to 'no'.
+# WARNING: 'UsePAM no' is not supported in Fedora and may cause several
+# problems.
 #UsePAM no
 UsePAM yes
 
