--- kdebindings-3.5.13.1/qtruby/rubylib/qtruby/Qt.cpp.OLD	2012-09-29 20:08:00.156164195 +0200
+++ kdebindings-3.5.13.1/qtruby/rubylib/qtruby/Qt.cpp	2012-09-29 20:08:43.936538689 +0200
@@ -1782,7 +1782,7 @@
 VALUE
 getmetainfo(VALUE self, int &offset, int &index)
 {
-    const char * signalname = rb_id2name(rb_frame_this_func());
+    const char * signalname = rb_id2name(rb_frame_last_func());
     VALUE metaObject_value = rb_funcall(qt_internal_module, rb_intern("getMetaObject"), 1, self);
 
     smokeruby_object *ometa = value_obj_info(metaObject_value);
@@ -1872,7 +1872,7 @@
     // Now, I need to find out if this means me
     int index;
     char *slotname;
-    bool isSignal = qstrcmp(rb_id2name(rb_frame_this_func()), "qt_emit") == 0;
+    bool isSignal = qstrcmp(rb_id2name(rb_frame_last_func()), "qt_emit") == 0;
     VALUE mocArgs = getslotinfo(self, id, slotname, index, isSignal);
     if(mocArgs == Qnil) {
 		// No ruby slot/signal found, assume the target is a C++ one
--- kdebindings-3.5.13.1/korundum/rubylib/korundum/Korundum.cpp.OLD	2012-09-29 20:11:43.597168589 +0200
+++ kdebindings-3.5.13.1/korundum/rubylib/korundum/Korundum.cpp	2012-09-29 20:11:56.908163792 +0200
@@ -936,7 +936,7 @@
 {
 	VALUE dcopObject = rb_funcall(kde_module, rb_intern("createDCOPObject"), 1, self);
 	
-    TQString signalname(rb_id2name(rb_frame_this_func()));
+    TQString signalname(rb_id2name(rb_frame_last_func()));
     VALUE args = getdcopinfo(self, signalname);
 
     if(args == Qnil) return Qfalse;
