--- ProcessList.c.orig	2008-09-23 08:23:14.000000000 +0200
+++ ProcessList.c	2009-02-05 19:11:29.000000000 +0100
@@ -563,9 +563,9 @@
                process->pid = pid;
             }
          }
-         if (parent) {
-            process->tgid = parent->pid;
-         }
+
+         process->tgid = parent ? parent->pid : pid;
+
 
          if (showUserlandThreads && (!parent || pid != parent->pid)) {
             char subdirname[MAX_NAME+1];
