diff -ur kdeutils-3.5.13.orig/klaptopdaemon/acpi.cpp kdeutils-3.5.13/klaptopdaemon/acpi.cpp
--- kdeutils-3.5.13.orig/klaptopdaemon/acpi.cpp	2010-12-02 11:48:18.000000000 -0500
+++ kdeutils-3.5.13/klaptopdaemon/acpi.cpp	2010-12-02 11:48:32.000000000 -0500
@@ -126,6 +126,7 @@
     enableThrottle->setEnabled(can_enable);
     connect( enableThrottle, TQT_SIGNAL(clicked()), this, TQT_SLOT(configChanged()) );
 
+#if 0
     tmp_label = new TQLabel(i18n("If the above boxes are disabled then there is no 'helper' "
 				"application set up to help change ACPI states, there are two "
 				"ways you can enable this application, either make the file "
@@ -142,7 +143,7 @@
     ll->addWidget(setupButton);
     ll->addStretch(8);
     top_layout->addLayout(ll);
-
+#endif
 
     top_layout->addStretch(1);
     top_layout->addWidget( new TQLabel( i18n("Version: %1").arg(LAPTOP_VERSION), this), 0, Qt::AlignRight );
@@ -162,6 +163,7 @@
 
 void AcpiConfig::setupHelper()
 {
+#if 0
 	unsigned long len, crc;
 	TQString helper = KStandardDirs::findExe("klaptop_acpi_helper");
 	checkcrc(TQFile::encodeName(helper), len, crc);
@@ -185,6 +187,7 @@
 		KMessageBox::sorry(0, i18n("The ACPI helper cannot be enabled because kdesu cannot be found.  Please make sure that it is installed correctly."),
 				i18n("KLaptopDaemon"));
 	}
+#endif
 	laptop_portable::acpi_set_mask(enablestandby, enablesuspend, enablehibernate, enableperformance, enablethrottle);
     	bool can_enable = laptop_portable::has_acpi(1);	// is helper ready
     	enableStandby->setEnabled(can_enable);
diff -ur kdeutils-3.5.13.orig/klaptopdaemon/acpi_helper.cpp kdeutils-3.5.13/klaptopdaemon/acpi_helper.cpp
--- kdeutils-3.5.13.orig/klaptopdaemon/acpi_helper.cpp	2010-12-02 11:48:18.000000000 -0500
+++ kdeutils-3.5.13/klaptopdaemon/acpi_helper.cpp	2010-12-02 12:15:36.000000000 -0500
@@ -100,7 +100,7 @@
    Returns only if the program does not exist; if the program exists
    and is unsafe, exit; if the program exists and is safe, run it
    and never return. */
-void run_program(const char *path, const int action)
+void run_program(const char *path)
 {
 	struct stat sb;
 	int err;
@@ -118,13 +118,7 @@
 		exit(1);
 	}
 	::setuid(::geteuid());					// otherwise bash will throw it away
-	if (action == 1) {
-		system("/usr/sbin/pmi action hibernate");
-	} else if (action == 2) {
-		system("/usr/sbin/pmi action sleep");
-	} else {
-		::execl(path, NULL);	// this is not KDE environment code 
-	}
+	system(path);
 	exit(0);
 }
 
@@ -145,7 +139,7 @@
 	for (i = 1; i < argc; i++)
 	if (strcmp(argv[i], "--suspend") == 0 || strcmp(argv[i], "-suspend") == 0) {
 		/* Returns only if suspend does not exist. */
-		run_program("/usr/sbin/pmi", 2);
+		run_program("/usr/sbin/pm-suspend");
 		/*
 		if (useSysPower)
 			write_to_power("mem");
@@ -166,7 +160,7 @@
 		exit(0);
 	} else
 	if (strcmp(argv[i], "--hibernate") == 0 || strcmp(argv[i], "-hibernate") == 0) {
-		run_program("/usr/sbin/pmi", 1);
+		run_program("/usr/sbin/pm-hibernate");
 		/*
 		if (useSysPower)
 			write_to_power("disk");
@@ -176,7 +170,7 @@
 		exit(0);
 	} else
 	if (strcmp(argv[i], "--software-suspend") == 0 || strcmp(argv[i], "-software-suspend") == 0) {
-		run_program("/usr/sbin/hibernate", 0);
+		run_program("/usr/sbin/hibernate");
 		exit(0);
 	} else
 	if (strcmp(argv[i], "--throttling") == 0 || strcmp(argv[i], "-throttling") == 0) {
diff -ur kdeutils-3.5.13.orig/klaptopdaemon/portable.cpp kdeutils-3.5.13/klaptopdaemon/portable.cpp
--- kdeutils-3.5.13.orig/klaptopdaemon/portable.cpp	2010-12-02 11:48:18.000000000 -0500
+++ kdeutils-3.5.13/klaptopdaemon/portable.cpp	2010-12-02 11:48:32.000000000 -0500
@@ -1356,6 +1356,7 @@
 	if (known[type] == last_seed)
 		return(known_res[type]);
 	known[type] = last_seed;
+#if 0
 	known_res[type] = 0;
 	struct stat sb;
 	TQString str = KStandardDirs::findExe("klaptop_acpi_helper");
@@ -1370,6 +1371,7 @@
 		return(0);	
 	if (!(sb.st_mode&0x111))
 		return(0);
+#endif
 	known_res[type] = 1;
 	return(1);
 }
