--- kdeutils-3.5.4/kedit/kedit.cpp.orig	2006-09-12 12:54:45.000000000 +0200
+++ kdeutils-3.5.4/kedit/kedit.cpp	2006-09-12 15:21:50.000000000 +0200
@@ -902,7 +902,6 @@
 
     TQFont printFont = eframe->font();
     TQFont headerFont(printFont);
-    headerFont.setBold(true);
 
     TQFontMetrics printFontMetrics(printFont);
     TQFontMetrics headerFontMetrics(headerFont);
@@ -922,7 +921,6 @@
 
         int dy = 0;
 
-	p->setFont(headerFont);
         int w = printFontMetrics.width("M");
         p->setTabStops(8*w);
 
@@ -1325,6 +1323,7 @@
   TQWidget *font = new TQWidget(0, "FontSetting");
   TQVBoxLayout *topLayout = new TQVBoxLayout(font, 0, KDialog::spacingHint());
   KFontChooser *mFontChooser = new KFontChooser(font, "kcfg_Font", false, TQStringList(), false, 6);
+  mFontChooser->enableColumn(KFontChooser::StyleList, false);
   topLayout->addWidget(mFontChooser);
   addPage(font, i18n("Font"), "fonts", i18n("Editor Font"));
 
