--- kdeutils-3.5.4/klaptopdaemon/warning.cpp.orig	2006-09-21 14:39:36.000000000 +0200
+++ kdeutils-3.5.4/klaptopdaemon/warning.cpp	2006-09-21 14:59:48.000000000 +0200
@@ -595,10 +595,14 @@
 		}
 		checkNone->setChecked(!do_suspend&!do_standby&!do_hibernate&!logout&!shutdown);
 		editRunCommand->setURL(runcommand_val);
-		editLowTime->setValue(low_val_time);
-                editLowPercent->setValue(low_val_percent);
-                editCriticalTime->setValue(critical_val_time);
-                editCriticalPercent->setValue(critical_val_percent);
+		if (editLowTime)
+			editLowTime->setValue(low_val_time);
+		if (editLowPercent)
+             	   editLowPercent->setValue(low_val_percent);
+		if (editCriticalTime)
+                	editCriticalTime->setValue(critical_val_time);
+		if (editCriticalPercent)
+                	editCriticalPercent->setValue(critical_val_percent);
 		editPlaySound->setURL(sound_val);
 	}
 	changed(true);
