diff -ur tdeutils-git20111228/superkaramba/src/taskmanager.cpp tdeutils-git20111228.new/superkaramba/src/taskmanager.cpp
--- tdeutils-git20111228/superkaramba/src/taskmanager.cpp	2011-12-28 16:45:23.000000000 -0500
+++ tdeutils-git20111228.new/superkaramba/src/taskmanager.cpp	2011-12-29 04:34:59.000000000 -0500
@@ -774,11 +774,11 @@
    // on slower machines.
    //
   TQWidget *rootWin = TQT_TQWIDGET(tqApp->desktop());
-  TQRect geom = _info.geometry();
+  /*TQRect geom = _info.geometry();
    _grab = TQPixmap::grabWindow( rootWin->winId(),
         geom.x(), geom.y(),
         geom.width(), geom.height() );
-
+	*/
    if ( !_grab.isNull() )
      TQTimer::singleShot( 200, this, TQT_SLOT( generateThumbnail() ) );
 }
