--- kdeutils-3.0.3/kdf/disklist.cpp.label	2002-08-19 14:36:13.000000000 +0200
+++ kdeutils-3.0.3/kdf/disklist.cpp	2002-08-19 14:37:34.000000000 +0200
@@ -163,8 +163,9 @@
     while (! t.eof()) {
       s=t.readLine();
       s=s.simplifyWhiteSpace();
-      if ( (!s.isEmpty() ) && (s.find(DELIMITER)!=0) ) {
+      if ( (!s.isEmpty() ) && (s.find(DELIMITER)!=0) && (s.find("LABEL")!=0) ) {
                // not empty or commented out by '#'
+	       // skip LABEL entries as long as kdf can't handle them properly
 	//	kdDebug() << "GOT: [" << s << "]" << endl;
 	disk = new DiskEntry();// Q_CHECK_PTR(disk);
         disk->setMounted(FALSE);
