--- sendmail-8.14.1/sendmail/helpfile.noversion	2007-02-01 19:29:44.000000000 +0100
+++ sendmail-8.14.1/sendmail/helpfile	2007-04-10 16:58:16.000000000 +0200
@@ -11,9 +11,7 @@
 cpyr	forth in the LICENSE file which can be found at the top level of
 cpyr	the sendmail distribution.
 cpyr
-cpyr	$$Id: helpfile,v 8.48 2007/02/01 18:29:44 ca Exp $$
-cpyr
-smtp	This is sendmail version $v
+smtp	This is sendmail
 smtp	Topics:
 smtp		HELO	EHLO	MAIL	RCPT	DATA
 smtp		RSET	NOOP	QUIT	HELP	VRFY
