Index: emerald-0.8.8/misc/emerald-theme-manager.desktop
===================================================================
--- emerald-0.8.8.orig/misc/emerald-theme-manager.desktop	2011-04-21 19:43:42.000000000 +0700
+++ emerald-0.8.8/misc/emerald-theme-manager.desktop	2011-06-04 14:08:33.786049494 +0700
@@ -1,13 +1,18 @@
 [Desktop Entry]
-Encoding=UTF-8
+Type=Application
 Name=Emerald Theme Manager
 Name[fr]=Gestionnaire de thèmes Emerald
+Name[ca]=Gestor de temes de l'Emerald
+Name[de]=Emerald Motiv Manager
+Name[es]=Gestor de temas del Emerald
 Comment=Configure Emerald themes
 Comment[fr]=Configurerer les options des thèmes Emerald
-Icon=/usr/share/pixmaps/emerald-theme-manager-icon.png
+Comment[ca]=Configureu els temes de l'Emerald
+Comment[de]=Konfiguriert Emerald Motive
+Comment[es]=Configura los temas del Emerald
+Icon=emerald-theme-manager-icon
 Terminal=false
-Type=Application
 Categories=Settings;DesktopSettings;
-Exec=emerald-theme-manager -i %f
+Exec=emerald-theme-manager %f
 MimeType=application/x-emerald-theme;
 
