diff -up Tk-ToolBar-0.10/Makefile.PL.orig Tk-ToolBar-0.10/Makefile.PL
--- Tk-ToolBar-0.10/Makefile.PL.orig	2010-02-27 16:11:44.000000000 -0700
+++ Tk-ToolBar-0.10/Makefile.PL	2012-02-20 17:16:18.012721646 -0700
@@ -23,10 +23,6 @@ WriteMakefile1(
 	      },
 	      PM               => {
 				   'ToolBar.pm'       => '$(INST_LIB)/Tk/ToolBar.pm',
-				   'toolbar.pl'       => ($] >= 5.005 ?
-							  '$(INST_ARCHLIB)' :
-							  '$(INST_LIB)') .
-				   '/Tk/demos/widtrib/toolbar.pl',
 				   'ToolBar/tkIcons'  => '$(INST_LIB)/Tk/ToolBar/tkIcons',
 				  },
 	     );
