diff -ur hal-0.5.13/fdi/policy/10osvendor/10-keymap.fdi hal-0.5.13.new/fdi/policy/10osvendor/10-keymap.fdi
--- hal-0.5.13/fdi/policy/10osvendor/10-keymap.fdi	2008-08-10 21:50:10.000000000 +0800
+++ hal-0.5.13.new/fdi/policy/10osvendor/10-keymap.fdi	2009-11-02 10:54:49.000000000 +0800
@@ -18,6 +18,7 @@
 
       <merge key="input.xkb.layout" type="string">us</merge>
       <merge key="input.xkb.variant" type="string" />
+      <merge key="input.xkb.options" type="string">terminate:ctrl_alt_bksp</merge>
     </match>
   </device>
 </deviceinfo>
