From 27bef9545edaee7eac91865ef648752ac94f2dcf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=B1=9F=E6=B6=9B?= <jiangtao9999@163.com>
Date: Wed, 23 Oct 2013 09:58:43 +0800
Subject: [PATCH 04/50] =?UTF-8?q?=E5=A2=9E=E5=8A=A0mips64el=E7=BC=96=E8=AF?=
 =?UTF-8?q?=91?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 gcc/gcc.spec | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/gcc/gcc.spec b/gcc/gcc.spec
index d920576..c153883 100644
--- a/gcc/gcc.spec
+++ b/gcc/gcc.spec
@@ -36,7 +36,7 @@
 %else
 %global build_libtsan 0
 %endif
-%ifarch %{ix86} x86_64 ppc ppc64 s390 s390x %{arm}
+%ifarch %{ix86} x86_64 ppc ppc64 s390 s390x %{arm} mips64el
 %global build_libatomic 1
 %else
 %global build_libatomic 0
@@ -1005,7 +1005,12 @@ CC="$CC" CFLAGS="$OPT_FLAGS" \
 	--with-bugurl=http://www.magiclinux.org/bugs --enable-bootstrap \
 	--enable-shared --enable-threads=posix --enable-checking=release \
 	--with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions \
-	--enable-gnu-unique-object --enable-linker-build-id --with-linker-hash-style=gnu \
+        --enable-gnu-unique-object --enable-linker-build-id \
+%ifarch mips64el
+        --with-linker-hash-style=sysv \
+%else
+        --with-linker-hash-style=gnu \
+%endif
 	--enable-languages=c,c++,objc,obj-c++,java,fortran${enablelada}${enablelgo},lto \
 	--enable-plugin --enable-initfini-array \
 %if !%{build_java}
@@ -1057,6 +1062,9 @@ CC="$CC" CFLAGS="$OPT_FLAGS" \
 %ifarch %{ix86}
 	--with-arch=i686 \
 %endif
+%ifarch mips64el
+        --with-arch=mips3 --with-abi=64 --disable-multilib \
+%endif
 %ifarch x86_64
 	--with-arch_32=i686 \
 %endif
@@ -1287,14 +1295,14 @@ find %{buildroot} -name libgcj.a -o -name libgtkpeer.a \
 		     -o -name libgij.a -o -name libgcj_bc.a -o -name libjavamath.a \
   | xargs rm -f
 
-mv %{buildroot}%{_prefix}/lib/libgcj.spec $FULLPATH/
+mv %{buildroot}%{_prefix}/lib*/libgcj.spec $FULLPATH/
 sed -i -e 's/lib: /&%%{static:%%eJava programs cannot be linked statically}/' \
   $FULLPATH/libgcj.spec
 %endif
 
-mv %{buildroot}%{_prefix}/lib/libgfortran.spec $FULLPATH/
+mv %{buildroot}%{_prefix}/lib*/libgfortran.spec $FULLPATH/
 %if %{build_libitm}
-mv %{buildroot}%{_prefix}/lib/libitm.spec $FULLPATH/
+mv %{buildroot}%{_prefix}/lib*/libitm.spec $FULLPATH/
 %endif
 
 mkdir -p %{buildroot}/%{_lib}
-- 
1.8.1

