diff -up Python-2.7.3/Lib/site.py.00121 Python-2.7.3/Lib/site.py
--- Python-2.7.3/Lib/site.py.00121	2013-02-19 14:07:19.156978291 -0500
+++ Python-2.7.3/Lib/site.py	2013-02-19 14:07:19.233978288 -0500
@@ -123,7 +123,7 @@ def addbuilddir():
     s = "build/lib.%s-%.3s" % (get_platform(), sys.version)
     if hasattr(sys, 'gettotalrefcount'):
         s += '-pydebug'
-    s = os.path.join(os.path.dirname(sys.path.pop()), s)
+    s = os.path.join(os.path.dirname(sys.path[-1]), s)
     sys.path.append(s)
 
 
diff -up Python-2.7.3/Misc/NEWS.00121 Python-2.7.3/Misc/NEWS
