diff --git a/magic/Magdir/linux b/magic/Magdir/linux
index 8d4c60a..aa8d1aa 100644
--- a/magic/Magdir/linux
+++ b/magic/Magdir/linux
@@ -88,6 +88,7 @@
 # From Daniel Novotny <dnovotny@redhat.com>
 # swap file for PowerPC
 65526	string		SWAPSPACE2	Linux/ppc swap file
+16374   string          SWAPSPACE2      Linux/ia64 swap file
 #
 # Linux kernel boot images, from Albert Cahalan <acahalan@cs.uml.edu>
 # and others such as Axel Kohlmeyer <akohlmey@rincewind.chemie.uni-ulm.de>
