--- bibletime/sword.m4.ftbfs	2011-05-17 20:42:02.000000000 +0200
+++ bibletime/sword.m4	2012-08-03 17:40:00.461746639 +0200
@@ -119,7 +119,7 @@
 EOF
 
 
-ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+ac_link='$LIBTOOL_SHELL --silent --mode=link --tag=LD ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
 if AC_TRY_EVAL(ac_link) && test -s conftest; then
 	if test -x conftest; then
 		eval ac_cv_installed_sword_version=`./conftest 2>&5`
@@ -192,7 +192,7 @@
 }
 EOF
 
-ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+ac_link='$LIBTOOL_SHELL --silent --mode=link --tag=LD ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
 if AC_TRY_EVAL(ac_link) && test -s conftest; then
 	if test -x conftest; then
 		eval ac_cv_sword_recent_version=`./conftest $1 2>&5`
--- bibletime/clucene.m4.ORI	2012-08-03 17:41:19.989067063 +0200
+++ bibletime/clucene.m4	2012-08-03 17:41:35.675734952 +0200
@@ -119,7 +119,7 @@
 EOF
 
 
-ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+ac_link='$LIBTOOL_SHELL --silent --mode=link --tag=LD ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
 if AC_TRY_EVAL(ac_link) && test -s conftest; then
 	if test -x conftest; then
 		eval ac_cv_installed_clucene_version=`./conftest 2>&5`
