diff -ur mono-addins-1.1.orig/configure.ac mono-addins-1.1/configure.ac
--- mono-addins-1.1.orig/configure.ac	2014-10-21 11:35:17.148690425 +0000
+++ mono-addins-1.1/configure.ac	2014-10-21 11:36:08.603689106 +0000
@@ -33,8 +33,8 @@
 	AC_MSG_ERROR([No al tool found. You need to install either the mono or .Net SDK.])
 fi
 
-GACUTIL_FLAGS='/package mono-addins /root $(DESTDIR)$(libdir)'
-GACUTIL_POLICY_FLAGS='/root $(DESTDIR)$(libdir)'
+GACUTIL_FLAGS='/package mono-addins /root $(DESTDIR)$(prefix)/lib'
+GACUTIL_POLICY_FLAGS='/root $(DESTDIR)$(prefix)/lib'
 
 AC_SUBST(GACUTIL)
 AC_SUBST(GACUTIL_FLAGS)
diff -ur mono-addins-1.1.orig/mautil/Makefile.am mono-addins-1.1/mautil/Makefile.am
--- mono-addins-1.1.orig/mautil/Makefile.am	2014-10-21 11:35:17.164690425 +0000
+++ mono-addins-1.1/mautil/Makefile.am	2014-10-21 11:36:23.499688725 +0000
@@ -8,7 +8,7 @@
 include $(top_srcdir)/Makefile.include
 include $(top_srcdir)/xbuild.include
 
-programfilesdir = $(libdir)/mono/mono-addins
+programfilesdir = $(prefix)/lib/mono/mono-addins
 programfiles_DATA = $(ASSEMBLY) $(build_datafiles)
 
 man_MANS = mautil.1
