diff -ur smplayer-0.5.21.orig/src/inforeader.cpp smplayer-0.5.21/src/inforeader.cpp
--- smplayer-0.5.21.orig/src/inforeader.cpp	2009-06-09 17:30:16.000000000 +0800
+++ smplayer-0.5.21/src/inforeader.cpp	2009-06-09 17:30:27.000000000 +0800
@@ -51,7 +51,7 @@
 	ao_list.clear();
 	demuxer_list.clear();
 
-	run("-vo help -ao help -demuxer help -vc help -ac help");
+	run("-identify -vo help -ao help -demuxer help -vc help -ac help");
 	//run("-demuxer help");
 
 	list();
@@ -89,11 +89,11 @@
 
 }
 
-static QRegExp rx_vo_key("^Available video output drivers");
-static QRegExp rx_ao_key("^Available audio output drivers");
-static QRegExp rx_demuxer_key("^Available demuxers");
-static QRegExp rx_ac_key("^Available audio codecs");
-static QRegExp rx_vc_key("^Available video codecs");
+static QRegExp rx_vo_key("^ID_VIDEO_OUTPUTS");
+static QRegExp rx_ao_key("^ID_AUDIO_OUTPUTS");
+static QRegExp rx_demuxer_key("^ID_DEMUXERS");
+static QRegExp rx_ac_key("^ID_AUDIO_CODECS");
+static QRegExp rx_vc_key("^ID_VIDEO_CODECS");
 
 static QRegExp rx_driver("\\t(.*)\\t(.*)");
 static QRegExp rx_demuxer("^\\s+([A-Z,a-z,0-9]+)\\s+(\\d+)\\s+(\\S.*)");
diff -ur smplayer-0.5.21.orig/src/mplayerprocess.cpp smplayer-0.5.21/src/mplayerprocess.cpp
--- smplayer-0.5.21.orig/src/mplayerprocess.cpp	2009-06-09 17:30:16.000000000 +0800
+++ smplayer-0.5.21/src/mplayerprocess.cpp	2009-06-09 17:30:44.000000000 +0800
@@ -158,7 +158,7 @@
 static QRegExp rx_connecting("^Connecting to .*");
 static QRegExp rx_resolving("^Resolving .*");
 static QRegExp rx_screenshot("^\\*\\*\\* screenshot '(.*)'");
-static QRegExp rx_endoffile("^Exiting... \\(End of file\\)");
+static QRegExp rx_endoffile("^Exiting... \\(End of file\\)|^ID_EXIT=EOF");
 static QRegExp rx_mkvchapters("\\[mkv\\] Chapter (\\d+) from");
 static QRegExp rx_aspect2("^Movie-Aspect is ([0-9,.]+):1");
  
