--- kchmviewer/lib/kio-msits/Makefile.am.ORI	2011-11-21 00:43:10.415241950 +0100
+++ kchmviewer/lib/kio-msits/Makefile.am	2011-11-21 00:43:22.929241583 +0100
@@ -5,7 +5,7 @@
 
 kio_msits_la_SOURCES = msits.cpp
 kio_msits_la_LIBADD = -lkio $(CHM_LIBS)
-kio_msits_la_LDFLAGS = -module -avoid-version $(KDE_LDFLAGS) $(LIB_QT) $(LIB_KIO) $(LIB_KDECORE) $(KDE_PLUGIN)
+kio_msits_la_LDFLAGS = -module -avoid-version $(KDE_LDFLAGS) $(LIB_QT) $(LIB_KIO) $(LIB_KDECORE) $(KDE_PLUGIN) $(QT_LDFLAGS)
 
 protocol_DATA = msits.protocol
 protocoldir = $(kde_servicesdir)
