--- a/Makefile	2009-10-31 18:59:06.000000000 +0100
+++ b/Makefile	2014-12-17 09:42:37.586079225 +0100
@@ -13,22 +13,11 @@
 #
 CONFIG=/etc/openldap/check_password.conf
 
-OPT=-g -O2 -Wall -fpic 						\
-	-DHAVE_CRACKLIB -DCRACKLIB_DICTPATH="\"$(CRACKLIB)\""	\
-	-DCONFIG_FILE="\"$(CONFIG)\""					\
+CFLAGS+=-fpic                                                  \
+	-DHAVE_CRACKLIB -DCRACKLIB_DICTPATH="\"$(CRACKLIB)\""  \
+	-DCONFIG_FILE="\"$(CONFIG)\""                          \
 	-DDEBUG
 
-# Where to find the OpenLDAP headers.
-#
-LDAP_INC=-I/home/pyb/tmp/openldap-2.3.39/include \
-	 -I/home/pyb/tmp/openldap-2.3.39/servers/slapd
-
-# Where to find the CrackLib headers.
-#
-CRACK_INC=
-
-INCS=$(LDAP_INC) $(CRACK_INC)
-
 LDAP_LIB=-lldap_r -llber
 
 # Comment out this line if you do NOT want to use the cracklib.
@@ -45,10 +34,10 @@
 all: 	check_password
 
 check_password.o:
-	$(CC) $(OPT) -c $(INCS) check_password.c
+	$(CC) $(CFLAGS) -c $(LDAP_INC) check_password.c
 
 check_password: clean check_password.o
-	$(CC) -shared -o check_password.so check_password.o $(CRACKLIB_LIB)
+	$(CC) $(LDFLAGS) -shared -o check_password.so check_password.o $(CRACKLIB_LIB)
 
 install: check_password
 	cp -f check_password.so ../../../usr/lib/openldap/modules/
