diff -Nur kaffeine-0.8.5/kaffeine/servicemenus/kaffeine_append_file.desktop kaffeine-0.8.5.new/kaffeine/servicemenus/kaffeine_append_file.desktop
--- kaffeine-0.8.5/kaffeine/servicemenus/kaffeine_append_file.desktop	2007-07-29 00:45:37.000000000 +0800
+++ kaffeine-0.8.5.new/kaffeine/servicemenus/kaffeine_append_file.desktop	2007-11-03 04:19:06.000000000 +0800
@@ -1,5 +1,5 @@
 [Desktop Entry]
-ServiceTypes=application/x-ogg,audio/basic,audio/vnd.rn-realaudio,audio/x-aiff,audio/x-mp3,audio/x-mpeg,audio/x-mpegurl,audio/x-ms-wma,audio/x-ogg,audio/x-pn-realaudio,audio/x-pn-realaudio-plugin,audio/x-scpls,audio/x-wav,audio/x-flac,video/x-matroska,audio/x-matroska,video/mpeg,video/msvideo,video/quicktime,video/vnd.rn-realvideo,video/x-avi,video/x-fli,video/x-flic,video/x-ms-asf,video/x-ms-wmv,video/x-msvideo,application/x-mplayer2,application/smil,application/x-kaffeine,audio/x-musepack
+ServiceTypes=application/x-ogg,audio/basic,audio/vnd.rn-realaudio,audio/x-aiff,audio/x-mp3,audio/x-mpeg,audio/x-mpegurl,audio/x-ms-wma,audio/x-ogg,audio/x-pn-realaudio,audio/x-pn-realaudio-plugin,audio/x-scpls,audio/x-wav,audio/x-flac,video/x-matroska,audio/x-matroska,video/mpeg,video/msvideo,video/quicktime,video/vnd.rn-realvideo,video/x-avi,video/x-fli,video/x-flic,video/x-ms-asf,video/x-ms-wmv,video/x-msvideo,application/x-mplayer2,application/smil,application/x-kaffeine,audio/x-musepack,application/vnd.rn-realmedia,video/x-flv
 Actions=kaffeine_append_file;
 Encoding=UTF-8
 
