diff -u -r kaffeine-mozilla-0.2.orig/src/plugin.c kaffeine-mozilla-0.2/src/plugin.c
--- kaffeine-mozilla-0.2.orig/src/plugin.c	2004-05-12 16:12:52.000000000 +0800
+++ kaffeine-mozilla-0.2/src/plugin.c	2006-04-03 18:26:38.000000000 +0800
@@ -165,6 +165,9 @@
          "audio/x-ms-wma: wma: Microsoft Media Audio;"
          "video/x-ms-wmv: wmv: Microsoft Media Video;"
          "audio/x-mpegurl: m3u: Streaming-MPEG-Audio;"
+	 "audio/x-mp3: mp3: mp3 audio;"
+	 "application/vnd.rn-realmedia: rm, rmvb: RealMedia Video;"
+	 "audio/vnd.rn-realaudio: ra, ram: Real audio;"
 	 ;
 }
 
