diff -u -r kaffeine-mozilla-0.2.orig/configure kaffeine-mozilla-0.2/configure
--- kaffeine-mozilla-0.2.orig/configure	2004-05-14 00:46:54.000000000 +0800
+++ kaffeine-mozilla-0.2/configure	2006-09-12 14:54:52.000000000 +0800
@@ -20552,7 +20552,7 @@
 echo "${ECHO_T}$ac_cv_header_X11_Xaw_Form_h" >&6
 
 fi
-if test $ac_cv_header_X11_Xaw_Form_h = yes; then
+if test $ac_cv_header_X11_Xaw_Form_h = no; then
   :
 else
   { { echo "$as_me:$LINENO: error: you need to install libxaw header files (-dev package)" >&5
