diff -up vlc-2.0.5/modules/codec/flac.c.FLAC13 vlc-2.0.5/modules/codec/flac.c
--- vlc-2.0.5/modules/codec/flac.c.FLAC13	2011-12-08 19:00:26.000000000 +0100
+++ vlc-2.0.5/modules/codec/flac.c	2013-01-02 20:48:33.569532663 +0100
@@ -38,8 +38,8 @@
 #include <vlc_codec.h>
 #include <vlc_aout.h>
 
-#include <stream_decoder.h>
-#include <stream_encoder.h>
+#include <FLAC/stream_decoder.h>
+#include <FLAC/stream_encoder.h>
 
 #include <vlc_block_helper.h>
 #include <vlc_bits.h>
