diff -up vlc-2.0.2/modules/access/screen/xcb.c.discard vlc-2.0.2/modules/access/screen/xcb.c
--- vlc-2.0.2/modules/access/screen/xcb.c.discard	2011-12-08 19:00:26.000000000 +0100
+++ vlc-2.0.2/modules/access/screen/xcb.c	2012-07-11 15:07:43.092366202 +0200
@@ -319,7 +319,11 @@ static void Demux (void *data)
         msg_Err (demux, "bad X11 drawable 0x%08"PRIx32, sys->window);
 discard:
         if (sys->follow_mouse)
+#if 0
             xcb_discard_reply (conn, gc.sequence);
+#else
+            msg_Warn (demux, "xcb_discard_reply not available in libxcb < 1.6") ;
+#endif
         return;
     }
 
