diff -up vlc-1.1.0/modules/Makefile.am.mock vlc-1.1.0/modules/Makefile.am
--- vlc-1.1.0/modules/Makefile.am.mock	2010-04-05 19:46:04.000000000 -0500
+++ vlc-1.1.0/modules/Makefile.am	2010-04-16 12:24:52.745993752 -0500
@@ -63,7 +63,7 @@ MOSTLYCLEANFILES = stamp-cache plugins-*
 
 stamp-cache:
 	$(AM_V_GEN)if test "$(build)" = "$(host)"; then \
-		../bin/vlc-cache-gen$(EXEEXT) . ; \
+		../bin/vlc-cache-gen$(EXEEXT) . ||: ; \
 	else \
 		echo "Cross-compilation: cache generation skipped!" ; \
 	fi
@@ -71,7 +71,7 @@ stamp-cache:
 
 install-exec-hook:
 	if test -z "$(DESTDIR)"; then \
-		../bin/vlc-cache-gen$(EXEEXT) "$(vlclibdir)/plugins" ; \
+		../bin/vlc-cache-gen$(EXEEXT) "$(vlclibdir)/plugins" ||: ; \
 	else \
 		echo "Staged installation: cache generation skipped!" ; \
 	fi

