--- configure.ac.orig	2010-08-18 19:52:28.000000000 +0800
+++ configure.ac	2010-08-19 11:10:49.057014652 +0800
@@ -937,7 +937,7 @@ AC_ARG_ENABLE(non-utf8,
   [  --enable-non-utf8       Legacy non-UTF-8 systems support (default enabled)], [
   enable_non_utf8="yes"
 ])
-AS_IF([test "${enable_non_utf8}" != "no"], [
+AS_IF([test "${enable_non_utf8}" == "no"], [
   AC_DEFINE([ASSUME_UTF8], [1],
             [Define to 1 if the operating system uses UTF-8 internally])
 ])
