diff -ur vlc-1.1.13.orig/projects/mozilla/support/npunix.cpp vlc-1.1.13/projects/mozilla/support/npunix.cpp
--- vlc-1.1.13.orig/projects/mozilla/support/npunix.cpp	2012-01-26 15:09:25.000000000 -0500
+++ vlc-1.1.13/projects/mozilla/support/npunix.cpp	2012-01-26 15:12:04.000000000 -0500
@@ -767,11 +767,11 @@
  *  - Netscape uses the return value to identify when an object instance
  *    of this plugin should be created.
  */
-char *
+/*char *
 NP_GetMIMEDescription(void)
 {
     return NPP_GetMIMEDescription();
-}
+}*/
 
 /*
  * NP_GetValue [optional]
diff -ur vlc-1.1.13.orig/projects/mozilla/vlcshell.cpp vlc-1.1.13/projects/mozilla/vlcshell.cpp
--- vlc-1.1.13.orig/projects/mozilla/vlcshell.cpp	2012-01-26 15:09:25.000000000 -0500
+++ vlc-1.1.13/projects/mozilla/vlcshell.cpp	2012-01-26 15:11:26.000000000 -0500
@@ -71,11 +71,11 @@
 /******************************************************************************
  * UNIX-only API calls
  *****************************************************************************/
-char * NPP_GetMIMEDescription( void )
+/*char * NPP_GetMIMEDescription( void )
 {
     static char mimetype[] = PLUGIN_MIMETYPES;
     return mimetype;
-}
+}*/
 
 NPError NPP_GetValue( NPP instance, NPPVariable variable, void *value )
 {
diff -ur vlc-1.1.13.orig/projects/mozilla/vlcshell.h vlc-1.1.13/projects/mozilla/vlcshell.h
--- vlc-1.1.13.orig/projects/mozilla/vlcshell.h	2012-01-26 15:09:25.000000000 -0500
+++ vlc-1.1.13/projects/mozilla/vlcshell.h	2012-01-26 15:11:08.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef __VLCSHELL_H__
 #define __VLCSHELL_H__
 
-char * NPP_GetMIMEDescription( void );
+//char * NPP_GetMIMEDescription( void );
 
 NPError NPP_Initialize( void );
 
