--- lxde-common-0.3.2.1.orig/lxde-logout.in	2008-04-20 10:26:19.000000000 +0200
+++ lxde-common-0.3.2.1/lxde-logout.in	2008-11-30 17:08:25.000000000 +0100
@@ -1,2 +1,2 @@
 #!/bin/sh
-lxsession-logout --banner "@prefix@/share/lxde/images/logout-banner.png" --side=top
+lxsession-logout --banner "@prefix@/share/pixmaps/fedora-logo-small.png" --side=top
--- lxde-common-0.3.2.1.orig/lxde-settings/config	2008-04-20 10:26:08.000000000 +0200
+++ lxde-common-0.3.2.1/lxde-settings/config	2008-10-10 00:25:20.000000000 +0200
@@ -4,8 +4,8 @@
 #iNet/DndDragThreshold
 #iNet/CursorBlink
 #iNet/CursorBlinkTime
-sNet/ThemeName=Clearlooks
-sNet/IconThemeName=nuoveXT2
+sNet/ThemeName=Nodoka
+sNet/IconThemeName=Fedora
 #iGtk/CanChangeAccels=0
 #cGtk/ColorPalette
 sGtk/FontName=Sans 10
--- lxde-common-0.3.2.1.orig/lxpanel/panel.in	2008-04-20 12:09:34.000000000 +0200
+++ lxde-common-0.3.2.1/lxpanel/panel.in	2008-05-06 02:52:48.000000000 +0200
@@ -29,7 +29,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxpanel/images/my-computer.png
+        image=@prefix@/share/icons/Fedora/24x24/places/start-here.png
         system {
         }
         separator {
--- lxde-common-0.3.2.1.orig/pcmanfm/main.lxde.in	2008-03-19 07:36:29.000000000 +0100
+++ lxde-common-0.3.2.1/pcmanfm/main.lxde.in	2008-04-13 05:25:51.000000000 +0200
@@ -1,6 +1,6 @@
 [Desktop]
 show_wallpaper=1
-wallpaper=@prefix@/share/lxde/wallpapers/lxde_blue.jpg
+wallpaper=@prefix@/share/backgrounds/images/default.png
 
 [General]
 terminal=lxterminal
