--- lxde-common-0.3.2.1.orig/lxpanel/config	2008-04-20 11:25:36.000000000 +0200
+++ lxde-common-0.3.2.1/lxpanel/config	2008-05-06 02:03:45.000000000 +0200
@@ -1,4 +1,4 @@
 [Command]
 FileManager=pcmanfm %s
-Terminal=xterm -e
+Terminal=lxterminal -e
 
--- lxde-common-0.3.2.1.orig/lxpanel/panel.in	2008-04-20 12:09:34.000000000 +0200
+++ lxde-common-0.3.2.1/lxpanel/panel.in	2008-10-10 01:11:31.000000000 +0200
@@ -13,9 +13,9 @@
     alpha=0
     setdocktype=1
     setpartialstrut=1
-    usefontcolor=1
+    usefontcolor=0
     fontcolor=#ffffff
-    background=1
+    background=0
     backgroundfile=@prefix@/share/lxpanel/images/background.png
 }
 
@@ -35,6 +35,7 @@
         separator {
         }
         item {
+            image=gnome-run
             command=run
         }
         separator {
@@ -50,13 +51,13 @@
     type = launchbar
     Config {
         Button {
-            id=pcmanfm.desktop
+            id=magic-pcmanfm.desktop
         }
         Button {
-            id=gnome-terminal.desktop
+            id=magic-lxterminal.desktop
         }
         Button {
-            id=firefox.desktop
+            id=mozilla-firefox.desktop
         }
     }
 }
@@ -133,3 +134,7 @@
     }
 }
 
+Plugin {
+    type = volumealsa
+}
+
