--- lxde-common-0.3.2.1.orig/startlxde.in	2008-04-20 10:26:19.000000000 +0200
+++ lxde-common-0.3.2.1/startlxde.in	2008-11-01 01:52:54.000000000 +0100
@@ -18,6 +18,14 @@
     cp @prefix@/share/lxde/openbox/rc.xml "$OPENBOX_CONF_DIR/lxde-rc.xml"
 fi
 
+# Workaround for gdm, openbox and the number of desktops
+NUMBER_DESKTOPS=`grep \<number\> "$OPENBOX_CONF_DIR/lxde-rc.xml" | awk -F \> {'print $2}' | awk -F \< {'print $1}'`
+if test x"`which wmctrl 2>/dev/null`" != x""; then
+    (until [ `wmctrl -d | wc -l ` = "$NUMBER_DESKTOPS" ];do
+	wmctrl -n "$NUMBER_DESKTOPS"
+    done) &
+fi
+
 # Enable GTK+2 integration for OpenOffice.org, if available.
 export SAL_USE_VCLPLUGIN=gtk
 
