diff -up gpgme-1.3.0/src/gpgme-config.in.config_extras gpgme-1.3.0/src/gpgme-config.in
--- gpgme-1.3.0/src/gpgme-config.in.config_extras	2009-10-23 08:44:56.000000000 -0500
+++ gpgme-1.3.0/src/gpgme-config.in	2011-03-17 13:15:12.511565537 -0500
@@ -12,7 +12,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 includedir=@includedir@
-libdir=@libdir@
+libdir=@exec_prefix@/lib
 
 # Make sure that no weird locale setting messes up our sed regexps etc.
 LC_COLLATE=C
@@ -20,26 +20,26 @@ LC_ALL=C
 LANG=C
 
 # Network libraries.
-assuan_cflags="@LIBASSUAN_CFLAGS@"
-assuan_libs="@LIBASSUAN_LIBS@"
+#assuan_cflags="@LIBASSUAN_CFLAGS@"
+#assuan_libs="@LIBASSUAN_LIBS@"
 
 # Configure libgpg-error.
 gpg_error_cflags="@GPG_ERROR_CFLAGS@"
-gpg_error_libs="@GPG_ERROR_LIBS@"
+#gpg_error_libs="@GPG_ERROR_LIBS@"
 
 # Configure thread packages.
 thread_modules=""
 
 @HAVE_PTH_TRUE@thread_modules="$thread_modules pth"
-libs_pth="@PTH_LDFLAGS@ @PTH_LIBS@"
+#libs_pth="@PTH_LDFLAGS@ @PTH_LIBS@"
 cflags_pth="@PTH_CFLAGS@"
 
 @HAVE_PTHREAD_TRUE@thread_modules="$thread_modules pthread"
-libs_pthread="-lpthread"
+#libs_pthread="-lpthread"
 cflags_pthread=""
 
 # Configure glib.
-libs_glib="@GLIB_LIBS@"
+#libs_glib="@GLIB_LIBS@"
 cflags_glib="@GLIB_CFLAGS@"
 with_glib=
 
