diff -ur reciteword-0.8.4.orig/src/bookfile.cpp reciteword-0.8.4/src/bookfile.cpp
--- reciteword-0.8.4.orig/src/bookfile.cpp	2009-05-18 09:02:32.000000000 +0800
+++ reciteword-0.8.4/src/bookfile.cpp	2009-05-18 09:05:54.000000000 +0800
@@ -88,7 +88,7 @@
 	
 
 static gchar *
-rw_book_get_value (const gchar * str1, gchar * str2,gint utf8)
+rw_book_get_value (gchar * str1, gchar * str2,gint utf8)
 {
 	gchar tmp, *tmp1, *tmp2;
 
