Index: redhat-rpm-config-9.1.0/macros
===================================================================
--- redhat-rpm-config-9.1.0.orig/macros
+++ redhat-rpm-config-9.1.0/macros
@@ -239,7 +239,7 @@ kernel_module_package_release	1
 		done \
 	fi \
 	echo "%%global flavors_to_build ${flavors_to_build:-%%nil}" \
-	echo "%%global kernel_source() /usr/src/kernels/%kverrel-\\\$([ %%%%{1} = default ] || echo "%%%%{1}-")%_target_cpu" \
+	echo "%%global kernel_source() /usr/src/kernels/%kverrel-\\\$([ %%%%{1} = default ] || echo "%%%%{1}.")%_target_cpu" \
 	if [ ! -z "%{-f*}" ] \
 	then \
 		filelist="%{-f*}" \
