--- kdemultimedia-3.2.2/mpeglib_artsplug/splayPlayObject_impl.cpp.buffer	2003-05-05 07:19:44.000000000 -0500
+++ kdemultimedia-3.2.2/mpeglib_artsplug/splayPlayObject_impl.cpp	2004-05-21 09:37:26.312000308 -0500
@@ -17,7 +17,7 @@
 
 #include "debug.h"
 
-#define INPUT_SIZE  8192
+#define INPUT_SIZE  32768 
 
 /**
    Problems with streaming, which must be solved:
--- kdemultimedia-3.2.2/mpeglib/lib/decoder/splayPlugin.cpp.buffer	2003-06-27 08:44:43.000000000 -0500
+++ kdemultimedia-3.2.2/mpeglib/lib/decoder/splayPlugin.cpp	2004-05-21 09:38:12.550471578 -0500
@@ -25,7 +25,7 @@
 
 using namespace std;
 
-#define INPUTBUFFER_SIZE 8192
+#define INPUTBUFFER_SIZE 32768 
 
 SplayPlugin::SplayPlugin() {
   pow(6.0,3.0);            // fixes bug in __math.h
