diff -ur tdemultimedia-git20111223/kscd/libwm/audio/audio_arts.c tdemultimedia-git20111223.orig/kscd/libwm/audio/audio_arts.c
--- tdemultimedia-git20111223/kscd/libwm/audio/audio_arts.c	2011-12-23 15:15:42.000000000 -0500
+++ tdemultimedia-git20111223.orig/kscd/libwm/audio/audio_arts.c	2011-12-27 09:50:24.000000000 -0500
@@ -26,7 +26,7 @@
 
 #ifdef USE_ARTS
 
-#include <artsc.h>
+#include <artsc/artsc.h>
 
 #include "audio.h"
 
diff -ur tdemultimedia-git20111223/noatun/library/vequalizer.cpp tdemultimedia-git20111223.orig/noatun/library/vequalizer.cpp
--- tdemultimedia-git20111223/noatun/library/vequalizer.cpp	2011-12-23 15:15:42.000000000 -0500
+++ tdemultimedia-git20111223.orig/noatun/library/vequalizer.cpp	2011-12-27 09:50:04.000000000 -0500
@@ -315,7 +315,7 @@
 void VEqualizer::init()
 {
 	KURL url;
-	url.setPath(kapp->dirs()->localtdedir()+"/share/apps/noatun/equalizer");
+	url.setPath(kapp->dirs()->localkdedir()+"/share/apps/noatun/equalizer");
 	if(!load(url))
 	{
 		setPreamp(0);
@@ -333,7 +333,7 @@
 VEqualizer::~VEqualizer()
 {
 	KURL url;
-	url.setPath(kapp->dirs()->localtdedir()+"/share/apps/noatun/equalizer");
+	url.setPath(kapp->dirs()->localkdedir()+"/share/apps/noatun/equalizer");
 	save(url, "auto");
 
 	delete d;
@@ -664,7 +664,7 @@
 
 static TQString makePresetFile()
 {
-	TQString basedir=kapp->dirs()->localtdedir()+"/share/apps/noatun/eq.preset/";
+	TQString basedir=kapp->dirs()->localkdedir()+"/share/apps/noatun/eq.preset/";
 	// now append a filename that doesn't exist
 	KStandardDirs::makeDir(basedir);
 	TQString fullpath;
@@ -920,7 +920,7 @@
 
 	emit EQ->removed(*this);
 
-	if (file().find(kapp->dirs()->localtdedir())==0)
+	if (file().find(kapp->dirs()->localkdedir())==0)
 	{
 		TQFile f(file());
 		f.remove();
