Index: Makefile.in
===================================================================
--- Makefile.in	(revision 200)
+++ Makefile.in	(working copy)
@@ -42,8 +42,8 @@
 OCAMLCFLAGS	= -g
 OCAMLCPACKAGES	=
 OCAMLCLIBS	= -linkpkg
-OCAMLOPTFLAGS	=
-OCAMLOPTPACKAGES =
+OCAMLOPTFLAGS	= $(OCAMLCFLAGS)
+OCAMLOPTPACKAGES = $(OCAMLCPACKAGES)
 OCAMLOPTLIBS	= -linkpkg
 
 ifneq ($(enable_coverage),no)
@@ -88,7 +88,7 @@
 	$(OCAMLMKLIB) -o bitstring $^
 
 bitstring_persistent.cma: bitstring_persistent.cmo
-	$(OCAMLFIND) ocamlc -a -o $@ $^
+	$(OCAMLFIND) ocamlc -a $(OCAMLCFLAGS) -o $@ $^
 
 bitstring_persistent.cmo: bitstring_persistent.ml
 	$(OCAMLFIND) ocamlc $(OCAMLCFLAGS) $(OCAMLCPACKAGES) \
@@ -99,7 +99,7 @@
 	$(OCAMLMKLIB) -o bitstring $^
 
 bitstring_persistent.cmxa: bitstring_persistent.cmx
-	$(OCAMLFIND) ocamlopt -a -o $@ $^
+	$(OCAMLFIND) ocamlopt -a $(OCAMLOPTFLAGS) -o $@ $^
 
 bitstring_persistent.cmx: bitstring_persistent.ml
 	$(OCAMLFIND) ocamlopt $(OCAMLOPTFLAGS) $(OCAMLOPTPACKAGES) \
@@ -115,7 +115,7 @@
 
 bitstring-objinfo: bitstring_objinfo.cmo bitstring.cma bitstring_persistent.cma
 	$(OCAMLFIND) ocamlc -I +camlp4 unix.cma dynlink.cma camlp4lib.cma \
-	  $(OCAMLCLIBS) \
+	  $(OCAMLCFLAGS) $(OCAMLCLIBS) \
 	  bitstring.cma bitstring_persistent.cma \
 	  $< -o $@
 
