*** lib/echoe.rb	2010-11-09 11:19:37.011348562 +0100
--- lib/echoe.rb.new	2010-11-09 11:19:22.788214143 +0100
***************
*** 3,10 ****
  
  require 'rake'
  require 'rake/clean'
! require "#{$HERE}/../vendor/rake/lib/rake/contrib/compositepublisher"
! require "#{$HERE}/../vendor/rake/lib/rake/contrib/sshpublisher"
  require 'rake/gempackagetask'
  require 'rake/rdoctask'
  require 'rake/testtask'
--- 3,10 ----
  
  require 'rake'
  require 'rake/clean'
! require "rake/contrib/compositepublisher"
! require "rake/contrib/sshpublisher"
  require 'rake/gempackagetask'
  require 'rake/rdoctask'
  require 'rake/testtask'
