--- elinks-0.12pre4/configure.in.pkgconfig
+++ elinks-0.12pre4/configure.in
@@ -1034,6 +1034,11 @@
 
 else
 	cf_result=no
+	if pkg-config openssl ; then
+		CFLAGS="$CFLAGS_X `pkg-config --cflags openssl`"
+		LIBS="$LIBS_X `pkg-config --libs openssl`"
+		cf_result=yes
+	else
 	for ssldir in "$with_openssl" "" /usr /usr/local/openssl \
 			 /usr/lib/openssl /usr/local/ssl \
 			 /usr/local/www /usr/lib/ssl /usr/local \
@@ -1058,6 +1063,7 @@
 			fi
 		fi
 	done
+	fi
 
 	if test "$cf_result" != yes; then
 		if test "${with_openssl-no}" != "no"; then
