diff -up openbabel-75414ad4e043f16ba72ae51c7ca60f448576688d/scripts/CMakeLists.txt.rpm openbabel-75414ad4e043f16ba72ae51c7ca60f448576688d/scripts/CMakeLists.txt
--- openbabel-75414ad4e043f16ba72ae51c7ca60f448576688d/scripts/CMakeLists.txt.rpm	2015-02-04 20:04:57.000000000 +0100
+++ openbabel-75414ad4e043f16ba72ae51c7ca60f448576688d/scripts/CMakeLists.txt	2015-02-07 23:38:54.649356914 +0100
@@ -336,12 +336,23 @@ if (DO_PERL_BINDINGS)
     if(NOT BINDINGS_ONLY)
       add_dependencies(bindings_perl openbabel)
     endif()
+    execute_process(
+      COMMAND
+        ${PERL_EXECUTABLE} -V:installvendorarch
+      OUTPUT_VARIABLE
+        PERL_INSTALL_DIR_OUTPUT
+      RESULT_VARIABLE
+        PERL_INSTALL_DIR_RESULT
+      )
+    if (NOT PERL_INSTALL_DIR_RESULT)
+      string(REGEX REPLACE "installvendorarch='([^']+)'.*" "\\1" PERL_INSTALL_DIR ${PERL_INSTALL_DIR_OUTPUT})
+    endif (NOT PERL_INSTALL_DIR_RESULT)
 
     install(TARGETS bindings_perl
-            LIBRARY DESTINATION ${LIB_INSTALL_DIR}/auto/Chemistry/OpenBabel/
+            LIBRARY DESTINATION ${PERL_INSTALL_DIR}/auto/Chemistry/OpenBabel/
             COMPONENT bindings_perl)
     install(FILES ${openbabel_SOURCE_DIR}/scripts/perl/OpenBabel.pm
-            DESTINATION ${LIB_INSTALL_DIR}/Chemistry/
+            DESTINATION ${PERL_INSTALL_DIR}/Chemistry/
             COMPONENT bindings_perl)
 
 endif (DO_PERL_BINDINGS)
