diff -ur alsa-plugins-1.0.27.orig/a52/pcm_a52.c alsa-plugins-1.0.27/a52/pcm_a52.c
--- alsa-plugins-1.0.27.orig/a52/pcm_a52.c	2013-07-06 11:27:16.815116421 +0000
+++ alsa-plugins-1.0.27/a52/pcm_a52.c	2013-07-06 11:27:36.019116137 +0000
@@ -27,7 +27,7 @@
 #include <alsa/pcm_plugin.h>
 #include AVCODEC_HEADER
 
-#if LIBAVCODEC_VERSION_MAJOR >= 53 && LIBAVCODEC_VERSION_MINOR >= 34
+#if LIBAVCODEC_VERSION_MAJOR >= 55 && LIBAVCODEC_VERSION_MINOR >= 18
 #include <libavutil/audioconvert.h>
 #include <libavutil/mem.h>
 #define USE_AVCODEC_FRAME
