diff -up xf86-video-fbdev-0.4.3/src/fbdev.c.dma xf86-video-fbdev-0.4.3/src/fbdev.c
--- xf86-video-fbdev-0.4.3/src/fbdev.c.dma	2012-07-02 12:46:46.000000000 +0100
+++ xf86-video-fbdev-0.4.3/src/fbdev.c	2012-07-02 13:02:40.545956779 +0100
@@ -673,6 +673,8 @@ FBDevScreenInit(SCREEN_INIT_ARGS_DECL)
 	       pScrn->offset.red,pScrn->offset.green,pScrn->offset.blue);
 #endif
 
+	pScreen->canDoBGNoneRoot = 1;
+
 	if (NULL == (fPtr->fbmem = fbdevHWMapVidmem(pScrn))) {
 	        xf86DrvMsg(pScrn->scrnIndex,X_ERROR,"mapping of video memory"
 			   " failed\n");
