--- tellico/src/translators/btparse/bibtex_ast.c.ORI	2012-08-06 15:44:03.978791275 +0200
+++ tellico/src/translators/btparse/bibtex_ast.c	2012-08-06 15:44:23.449376138 +0200
@@ -57,7 +57,7 @@
 void dump_ast (char *msg, AST *root)
 {
    if (msg != NULL)
-      printf (msg);
+      printf ("%s", msg);
    dump (root, 0);
    printf ("\n");
 }
--- tellico/src/translators/btparse/error.c.ORI	2012-08-06 15:44:38.954046066 +0200
+++ tellico/src/translators/btparse/error.c	2012-08-06 15:45:08.152425553 +0200
@@ -88,7 +88,7 @@
 
    if (err->filename)
    {
-      fprintf (stderr, err->filename);
+      fprintf (stderr, "%s", err->filename);
       something_printed = TRUE;
    }
    if (err->line > 0)                   /* going to print a line number? */
@@ -111,7 +111,7 @@
    {
       if (something_printed)
          fprintf (stderr, ", ");
-      fprintf (stderr, name);
+      fprintf (stderr, "%s", name);
       something_printed = TRUE;
    }
 
