--- kvirc/admin/acinclude.m4.in.ORI	2011-11-25 19:57:17.296988361 +0100
+++ kvirc/admin/acinclude.m4.in	2011-11-25 19:57:47.365665354 +0100
@@ -2603,10 +2603,10 @@
 	helpdir="\${globalkvircdir}/help/en"
 	AC_SUBST(helpdir)
 
-	pluglibdir="\${globalkvircdir}/modules"
+	pluglibdir="\${libdir}/kvirc/$SS_DIRTAG/modules"
 	AC_SUBST(pluglibdir)
 
-	modulelibdir="\${globalkvircdir}/modules"
+	modulelibdir="\${libdir}/kvirc/$SS_DIRTAG/modules"
 	AC_SUBST(modulelibdir)
 
 	defscriptdir="\${globalkvircdir}/defscript"
--- kvirc/configure.in.ORI	2012-08-06 13:48:54.586710376 +0200
+++ kvirc/configure.in	2012-08-06 13:50:31.344841522 +0200
@@ -100,7 +100,7 @@
 
 AC_SS_HEADER([Checking KDE])
 
-SS_KDE_GENERAL_FILE_TO_SEARCH="include/kwin.h"
+SS_KDE_GENERAL_FILE_TO_SEARCH="include/kwin.h include/tde/kwin.h include/kde/kwin.h"
 SS_KDE_INCLUDE_FILE_TO_SEARCH="kwin.h"
 
 AC_SS_FIND_KDE
