--- kvirc-3.5.13.1/admin/acinclude.m4.in.ORI	2012-10-06 19:19:30.073415124 +0200
+++ kvirc-3.5.13.1/admin/acinclude.m4.in	2012-10-06 19:19:42.785136998 +0200
@@ -1105,7 +1105,7 @@
 			ss_save_LIBS="$LIBS"
 			CPPFLAGS="-I$SS_X_INCDIR -I$SS_QT3_INCDIR -I$SS_KDE_INCDIR $CPPFLAGS $X_CFLAGS"
 			CXXFLAGS="-O2 -Wall $CXXFLAGS"
-			LIBS="-L$SS_KDE_LIBDIR -L$SS_X_LIBDIR $SS_X_LDFLAGS $SS_X_LIBLINK -L$SS_QT3_LIBDIR $LIBS -lsoundserver_idl -lkdecore -l$SS_QTLIB_NAME -Wl,-rpath $SS_KDE_LIBDIR"
+			LIBS="-L$SS_KDE_LIBDIR -L$SS_X_LIBDIR $SS_X_LDFLAGS $SS_X_LIBLINK -L$SS_QT3_LIBDIR $LIBS -lsoundserver_idl -lkdecore -lmcop -l$SS_QTLIB_NAME -Wl,-rpath $SS_KDE_LIBDIR"
 			if test "$SS_LINK_TO_LIBDL" = "yes"; then
 				LIBS="$LIBS -ldl";
 			fi
