diff -ur tde-i18n-zh_CN-git20111225.orig/admin/acinclude.m4.in tde-i18n-zh_CN-git20111225/admin/acinclude.m4.in
--- tde-i18n-zh_CN-git20111225.orig/admin/acinclude.m4.in	2011-12-28 10:28:32.000000000 -0500
+++ tde-i18n-zh_CN-git20111225/admin/acinclude.m4.in	2011-12-28 10:29:49.000000000 -0500
@@ -471,7 +471,7 @@
 if test "$1" = "default"; then
 
   if test -z "$kde_htmldir"; then
-    kde_htmldir='\${datadir}/doc/HTML'
+    kde_htmldir='\${datadir}/doc/kde/HTML'
   fi
   if test -z "$kde_appsdir"; then
     kde_appsdir='\${datadir}/applnk'
