diff -ur arts-1.5.10/soundserver/cpuusage.cc arts-1.5.10-cpu-overload-quiet/soundserver/cpuusage.cc
--- arts-1.5.10/soundserver/cpuusage.cc	2005-09-10 10:13:31.000000000 +0200
+++ arts-1.5.10-cpu-overload-quiet/soundserver/cpuusage.cc	2009-12-16 09:28:14.000000000 +0100
@@ -24,6 +24,7 @@
 #include "dispatcher.h"
 #include "debug.h"
 
+#include <stdlib.h>
 #include <time.h>
 #include <unistd.h>
 #include <sys/time.h>
@@ -121,7 +122,7 @@
 
 		// ok, cancel synthesis due to cpu overload! brutal method
 		if(d->stalled > 15)
-			arts_fatal("cpu overload, aborting");
+			exit(1);
 	}
 
 	// prepare for next checkpoint
