From: Jiang Tao <jiangtao9999@163.com>

Fix mips64 compile error.

diff -ur xorg-server-1.12.4-orig/hw/xfree86/common/compiler.h xorg-server-1.12.4/hw/xfree86/common/compiler.h
--- xorg-server-1.12.4-orig/hw/xfree86/common/compiler.h  2013-07-05 05:50:05.775811380 +0000
+++ xorg-server-1.12.4/hw/xfree86/common/compiler.h	2013-07-05 05:53:46.132277897 +0000
@@ -704,7 +704,7 @@
 }
 
 #elif defined(__mips__) || (defined(__arm32__) && !defined(__linux__))
-#ifdef __arm32__
+#if defined(__arm32__) || defined(__mips64)
 #define PORT_SIZE long
 #else
 #define PORT_SIZE short
