From d39ea5f4a0d3048c5109550833ce8af743540a0e Mon Sep 17 00:00:00 2001
From: Avesh Agarwal <avagarwa@redhat.com>
Date: Tue, 10 Dec 2013 13:41:36 -0500
Subject: [PATCH 03/20] Fixed Networkmanager and related lib dependencies.

---
 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 10af655..27e2242 100644
--- a/configure.ac
+++ b/configure.ac
@@ -66,10 +66,10 @@ AC_SUBST(DBUS_CFLAGS)
 AC_SUBST(DBUS_LIBS)
 
 PKG_CHECK_MODULES(NM,
-	NetworkManager >= 0.9.6
-	libnm-util >= 0.9.6
-	libnm-glib >= 0.9.6
-	libnm-glib-vpn >= 0.9.6)
+	NetworkManager >= 0.9.8
+	libnm-util >= 0.9.8
+	libnm-glib >= 0.9.8
+	libnm-glib-vpn >= 0.9.8)
 AC_SUBST(NM_CFLAGS)
 AC_SUBST(NM_LIBS)
 
-- 
1.8.3.1

