From 41fd9ad25736659b3a465630bca7da294b1204e5 Mon Sep 17 00:00:00 2001
From: Avesh Agarwal <avagarwa@redhat.com>
Date: Tue, 10 Dec 2013 13:42:50 -0500
Subject: [PATCH 04/20] Fixed gnomekeyring lib dependencies.

---
 properties/Makefile.am | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/properties/Makefile.am b/properties/Makefile.am
index f599e21..abbf208 100644
--- a/properties/Makefile.am
+++ b/properties/Makefile.am
@@ -12,6 +12,7 @@ ui_DATA = nm-openswan-dialog.ui
 
 libnm_openswan_properties_la_CPPFLAGS = \
 	$(GTK_CFLAGS) \
+	$(GNOMEKEYRING_CFLAGS) \
 	$(NM_CFLAGS) \
 	-DICONDIR=\""$(datadir)/pixmaps"\" \
 	-DUIDIR=\""$(uidir)"\" \
@@ -22,7 +23,8 @@ libnm_openswan_properties_la_CPPFLAGS = \
 
 libnm_openswan_properties_la_LIBADD = \
 	$(GTK_LIBS) \
-	$(NM_LIBS)
+	$(NM_LIBS) \
+	$(GNOMEKEYRING_LIBS)
 
 libnm_openswan_properties_la_LDFLAGS = \
         -avoid-version
-- 
1.8.3.1

