diff -up Babel-1.3/setup.py.bak Babel-1.3/setup.py
--- Babel-1.3/setup.py.bak	2014-12-17 08:15:47.607637242 -0800
+++ Babel-1.3/setup.py	2014-12-17 08:17:04.981559771 -0800
@@ -57,7 +57,10 @@ setup(
         # This version identifier is currently necessary as
         # pytz otherwise does not install on pip 1.4 or
         # higher.
-        'pytz>=0a',
+        ### But the version confuses setuptools 8 and higher so remove it in the
+        ### system package
+        #'pytz>=0a',
+        'pytz',
     ],
 
     cmdclass={'build_doc': build_doc, 'test_doc': test_doc,
