diff -ur rpm-4.5.90.git8426.orig/rpmrc.in rpm-4.5.90.git8426/rpmrc.in
--- rpm-4.5.90.git8426.orig/rpmrc.in	2008-07-25 08:55:50.000000000 -0400
+++ rpm-4.5.90.git8426/rpmrc.in	2008-07-25 08:57:55.000000000 -0400
@@ -198,16 +198,16 @@
 #############################################################
 # For a given uname().machine, the default build arch
 
-buildarchtranslate: osfmach3_i686: i386
+buildarchtranslate: osfmach3_i686: i686
 buildarchtranslate: osfmach3_i586: i386
 buildarchtranslate: osfmach3_i486: i386
 buildarchtranslate: osfmach3_i386: i386
 
-buildarchtranslate: athlon: i386
-buildarchtranslate: geode: i386
-buildarchtranslate: pentium4: i386
-buildarchtranslate: pentium3: i386
-buildarchtranslate: i686: i386
+buildarchtranslate: athlon: i686
+buildarchtranslate: geode: i686
+buildarchtranslate: pentium4: i686
+buildarchtranslate: pentium3: i686
+buildarchtranslate: i686: i686
 buildarchtranslate: i586: i386
 buildarchtranslate: i486: i386
 buildarchtranslate: i386: i386
