--- kiso-0.8.3/src/dirops.cpp.orig	2005-08-27 17:45:36.000000000 +0000
+++ kiso-0.8.3/src/dirops.cpp	2005-12-30 16:39:52.000000000 +0000
@@ -31,6 +31,7 @@
 	mkdir(dest,7*64+5*8);
 	dp = opendir(source);
 
+	if (dp != NULL){
 	for (entry = readdir(dp);entry != NULL;entry = readdir(dp))
         if (QString(entry->d_name) != "." && QString(entry->d_name) != "..")
         {
@@ -50,6 +51,7 @@
 	};
 
 	closedir(dp);
+	}
 }
 
 void GetDirView (QListView* Object, QDir directory, const char* filter,
