--- kdesdk/kapptemplate/CMakeLists.txt.ORI	2012-07-30 17:58:45.681535290 +0200
+++ kdesdk/kapptemplate/CMakeLists.txt	2012-07-30 17:58:55.232338294 +0200
@@ -23,7 +23,7 @@
 
 install( PROGRAMS
     ${CMAKE_CURRENT_BINARY_DIR}/kapptemplate
-  DESTINATION ${DATA_INSTALL_DIR}/kapptemplate/bin )
+  DESTINATION ${BIN_INSTALL_DIR}/kapptemplate/bin )
 
 install( FILES
     kapptemplate.common kapptemplate.module
--- kdesdk/kbugbuster/kresources/CMakeLists.txt.ORI	2012-07-30 18:00:21.039575551 +0200
+++ kdesdk/kbugbuster/kresources/CMakeLists.txt	2012-07-30 18:00:25.414486000 +0200
@@ -17,6 +17,7 @@
   ${CMAKE_CURRENT_BINARY_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}/../backend
   ${TDE_INCLUDE_DIR}
+  ${TDE_INCLUDE_DIR}/kde
   ${TQT_INCLUDE_DIRS}
 )
 
