commit 572169a2e18705e526990c52924a268b218ff52b
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1340737965 -0500

    Fix FTBFS on newer subversion libraries
    This closes Bug 872

diff --git a/kioslave/svn/ConfigureChecks.cmake b/kioslave/svn/ConfigureChecks.cmake
index 81358c6..3d46c74 100644
--- a/kioslave/svn/ConfigureChecks.cmake
+++ b/kioslave/svn/ConfigureChecks.cmake
@@ -19,12 +19,16 @@ endif( )
 
 # check for subversion library
 
-check_library_exists( "svn_client-1" "svn_pool_create_ex" "${SVN_LIBRARY_DIR}" HAVE_SVN_POOL_CREATE_EX )
-
-if( HAVE_SVN_POOL_CREATE_EX )
+check_library_exists( "svn_client-1" "svn_pool_create_ex" "${SVN_LIBRARY_DIR}" HAVE_SVN_POOL_CREATE_EX_SVN_CLIENT )
+if( HAVE_SVN_POOL_CREATE_EX_SVN_CLIENT )
   set( SVN_LIBRARIES "svn_client-1;svn_subr-1;svn_ra-1" )
 else( )
-  tde_message_fatal( "svn_client-1 library was not found on your system.\n Subversion is installed?\n Try to set SVN_LIBRARY_DIR to subversion library directory." )
+  check_library_exists( "svn_subr-1" "svn_pool_create_ex" "${SVN_LIBRARY_DIR}" HAVE_SVN_POOL_CREATE_EX_SUBR )
+  if( HAVE_SVN_POOL_CREATE_EX_SUBR )
+    set( SVN_LIBRARIES "svn_client-1;svn_subr-1;svn_ra-1" )
+  else( )
+    tde_message_fatal( "svn_client-1 or svn_subr-1 library was not found on your system.\n Subversion is installed?\n Try to set SVN_LIBRARY_DIR to subversion library directory." )
+  endif( )
 endif( )
 
 
