--- kdesdk-3.5.13.1/kioslave/svn/ConfigureChecks.cmake	2012-09-30 20:05:52.627163660 +0200
+++ kdesdk/kioslave/svn/ConfigureChecks.cmake	2012-09-30 20:02:29.267164082 +0200
@@ -34,11 +34,11 @@
 
 # check for subversion headers
 
-tde_save_and_set( CMAKE_REQUIRED_FLAGS "-I${APR_INCLUDE_DIRS}" )
-tde_save_and_set( CMAKE_REQUIRED_DEFINITIONS "${APR_CFLAGS}" )
+set( CMAKE_REQUIRED_FLAGS "-I${APR_INCLUDE_DIRS}" )
+set( CMAKE_REQUIRED_DEFINITIONS "${APR_CFLAGS}" )
 
 if( SVN_INCLUDE_DIR )
-  set_and_save( CMAKE_REQUIRED_INCLUDES "${SVN_INCLUDE_DIR}" )
+  set( CMAKE_REQUIRED_INCLUDES "${SVN_INCLUDE_DIR}" )
   check_include_file( "svn_types.h" HAVE_SVN_TYPES_H )
   tde_restore( CMAKE_REQUIRED_INCLUDES )
 else( )
