--- compiz-with-glib-mainloop/plugins/cube/cube.xml.in	2011-01-15 13:20:16.396170197 +0000
+++ compiz-with-glib-mainloop/plugins/cube/cube.xml.in.new	2011-01-15 14:14:24.005435105 +0000
@@ -86,18 +86,18 @@
 				<_short>Top</_short>
 				<_long>Color of top face of the cube</_long>
 				<default>
-					<red>0xffff</red>
-					<green>0xffff</green>
-					<blue>0xffff</blue>
+					<red>0x8888</red>
+					<green>0x0000</green>
+					<blue>0x0000</blue>
 				</default>
 			</option>
 			<option name="bottom_color" type="color">
 				<_short>Bottom</_short>
 				<_long>Color of bottom face of the cube</_long>
 				<default>
-					<red>0xffff</red>
-					<green>0xffff</green>
-					<blue>0xffff</blue>
+					<red>0x8888</red>
+					<green>0x0000</green>
+					<blue>0x0000</blue>
 				</default>
 			</option>
 			</subgroup>
@@ -112,6 +112,7 @@
 			<_short>Skydome Image</_short>
 			<_long>Image to use as texture for the skydome</_long>
 			<hints>file;image;</hints>
+			<value>/usr/share/pixmaps/redhat/shadowman-transparent</value>
 			</option>
 			<option name="skydome_animated" type="bool">
 			<_short>Animate Skydome</_short>
@@ -122,18 +123,18 @@
 			<_short>Skydome Gradient Start Color</_short>
 			<_long>Color to use for the top color-stop of the skydome-fallback gradient</_long>
 			<default>
-				<red>0x0d0d</red>
-				<green>0xb1b1</green>
-				<blue>0xfdfd</blue>
+				<red>0x8888</red>
+				<green>0x0000</green>
+				<blue>0x0000</blue>
 			</default>
 			</option>
 			<option name="skydome_gradient_end_color" type="color">
 			<_short>Skydome Gradient End Color</_short>
 			<_long>Color to use for the bottom color-stop of the skydome-fallback gradient</_long>
 			<default>
-				<red>0xfefe</red>
-				<green>0xffff</green>
-				<blue>0xc7c7</blue>
+				<red>0x8888</red>
+				<green>0x0000</green>
+				<blue>0x0000</blue>
 			</default>
 			</option>
 			</subgroup>
