diff -uprN compiz-0.8.8-orig/configure.ac compiz-0.8.8/configure.ac
--- compiz-0.8.8-orig/configure.ac	2013-04-19 14:12:09.341857000 +0200
+++ compiz-0.8.8/configure.ac	2013-04-19 14:19:14.618100294 +0200
@@ -260,12 +260,6 @@ if test "x$use_gtk" = "xyes"; then
     if test "x$use_marco" = "xyes"; then
       PKG_CHECK_MODULES(MARCO, libmarco-private,
 			[use_marco=yes], [use_marco=no])
-      PKG_CHECK_EXISTS(libmarco-private >= 2.15.21,
-		       [have_marco_2_15_21=yes], [have_marco_2_15_21=no])
-      PKG_CHECK_EXISTS(libmarco-private >= 2.17.0,
-		       [have_marco_2_17_0=yes], [have_marco_2_17_0=no])
-      PKG_CHECK_EXISTS(libmarco-private >= 2.23.2,
-		       [have_marco_2_23_2=yes], [have_marco_2_23_2=no])
     fi
   else
     use_marco="no"
@@ -292,18 +286,6 @@ fi
 AM_CONDITIONAL(USE_MARCO, test "x$use_marco" = "xyes")
 if test "$use_marco" = yes; then
   AC_DEFINE(USE_MARCO, 1, [Build marco theme support])
-  if test "$have_marco_2_15_21" = yes; then
-    AC_DEFINE(HAVE_MARCO_2_15_21, 1,
-	      [Define to 1 if marco version >= 2.15.21])
-  fi
-  if test "$have_marco_2_17_0" = yes; then
-    AC_DEFINE(HAVE_MARCO_2_17_0, 1,
-	      [Define to 1 if marco version >= 2.17.0])
-  fi
-  if test "$have_marco_2_23_2" = yes; then
-    AC_DEFINE(HAVE_MARCO_2_23_2, 1,
-	      [Define to 1 if marco version >= 2.23.2])
-  fi
 fi
 
 AM_CONDITIONAL(USE_MATE, test "x$use_mate" = "xyes")
diff -uprN compiz-0.8.8-orig/gtk/window-decorator/gtk-window-decorator.c compiz-0.8.8/gtk/window-decorator/gtk-window-decorator.c
--- compiz-0.8.8-orig/gtk/window-decorator/gtk-window-decorator.c	2013-04-19 14:12:09.342857000 +0200
+++ compiz-0.8.8/gtk/window-decorator/gtk-window-decorator.c	2013-04-19 14:28:30.521589122 +0200
@@ -1327,19 +1327,10 @@ meta_get_corner_radius (const MetaFrameG
 			int		        *bottom_left_radius,
 			int			*bottom_right_radius)
 {
-
-#ifdef HAVE_MARCO_2_17_0
     *top_left_radius     = fgeom->top_left_corner_rounded_radius;
     *top_right_radius    = fgeom->top_right_corner_rounded_radius;
     *bottom_left_radius  = fgeom->bottom_left_corner_rounded_radius;
     *bottom_right_radius = fgeom->bottom_right_corner_rounded_radius;
-#else
-    *top_left_radius     = fgeom->top_left_corner_rounded ? 5 : 0;
-    *top_right_radius    = fgeom->top_right_corner_rounded ? 5 : 0;
-    *bottom_left_radius  = fgeom->bottom_left_corner_rounded ? 5 : 0;
-    *bottom_right_radius = fgeom->bottom_right_corner_rounded ? 5 : 0;
-#endif
-
 }
 
 static int
@@ -1548,8 +1539,6 @@ meta_function_to_type (MetaButtonFunctio
 	return META_BUTTON_TYPE_MAXIMIZE;
     case META_BUTTON_FUNCTION_CLOSE:
 	return META_BUTTON_TYPE_CLOSE;
-
-#ifdef HAVE_MARCO_2_17_0
     case META_BUTTON_FUNCTION_SHADE:
 	return META_BUTTON_TYPE_SHADE;
     case META_BUTTON_FUNCTION_ABOVE:
@@ -1562,7 +1551,6 @@ meta_function_to_type (MetaButtonFunctio
 	return META_BUTTON_TYPE_UNABOVE;
     case META_BUTTON_FUNCTION_UNSTICK:
 	return META_BUTTON_TYPE_UNSTICK;
-#endif
 
     default:
 	break;
@@ -1606,8 +1594,6 @@ meta_button_state_for_button_type (decor
 	return meta_button_state (d->button_states[BUTTON_MIN]);
     case META_BUTTON_TYPE_MENU:
 	return meta_button_state (d->button_states[BUTTON_MENU]);
-
-#ifdef HAVE_MARCO_2_17_0
     case META_BUTTON_TYPE_SHADE:
 	return meta_button_state (d->button_states[BUTTON_SHADE]);
     case META_BUTTON_TYPE_ABOVE:
@@ -1620,7 +1606,6 @@ meta_button_state_for_button_type (decor
 	return meta_button_state (d->button_states[BUTTON_UNABOVE]);
     case META_BUTTON_TYPE_UNSTICK:
 	return meta_button_state (d->button_states[BUTTON_UNSTICK]);
-#endif
 
     default:
 	break;
@@ -1708,10 +1693,8 @@ meta_get_decoration_geometry (decor_t		*
     if (d->state & WNCK_WINDOW_STATE_SHADED)
 	*flags |= META_FRAME_SHADED;
 
-#ifdef HAVE_MARCO_2_17_0
     if (d->state & WNCK_WINDOW_STATE_ABOVE)
 	*flags |= META_FRAME_ABOVE;
-#endif
 
     meta_theme_get_frame_borders (theme,
 				  META_FRAME_TYPE_NORMAL,
@@ -1808,7 +1791,6 @@ meta_draw_window_decoration (decor_t *d)
     bg_color = style->bg[GTK_STATE_NORMAL];
     bg_alpha = 1.0;
 
-#ifdef HAVE_MARCO_2_17_0
     if (frame_style->window_background_color)
     {
 	meta_color_spec_render (frame_style->window_background_color,
@@ -1817,7 +1799,6 @@ meta_draw_window_decoration (decor_t *d)
 
 	bg_alpha = frame_style->window_background_alpha / 255.0;
     }
-#endif
 
     cairo_destroy (cr);
 
@@ -2844,12 +2825,7 @@ meta_get_button_position (decor_t *d,
     MetaFrameFlags    flags;
     MetaTheme	      *theme;
     GdkRectangle      clip;
-
-#ifdef HAVE_MARCO_2_15_21
     MetaButtonSpace   *space;
-#else
-    GdkRectangle      *space;
-#endif
 
     if (!d->context)
     {
@@ -2890,7 +2866,6 @@ meta_get_button_position (decor_t *d,
 	space = &fgeom.close_rect;
 	break;
 
-#if defined (HAVE_MARCO_2_17_0) && defined (HAVE_LIBWNCK_2_18_1)
     case BUTTON_SHADE:
 	if (!meta_button_present (&button_layout, META_BUTTON_FUNCTION_SHADE))
 	    return FALSE;
@@ -2927,13 +2902,11 @@ meta_get_button_position (decor_t *d,
 
 	space = &fgeom.unstick_rect;
 	break;
-#endif
 
     default:
 	return FALSE;
     }
 
-#ifdef HAVE_MARCO_2_15_21
     if (!space->clickable.width && !space->clickable.height)
 	return FALSE;
 
@@ -2941,15 +2914,6 @@ meta_get_button_position (decor_t *d,
     *y = space->clickable.y;
     *w = space->clickable.width;
     *h = space->clickable.height;
-#else
-    if (!space->width && !space->height)
-	return FALSE;
-
-    *x = space->x;
-    *y = space->y;
-    *w = space->width;
-    *h = space->height;
-#endif
 
     return TRUE;
 }
@@ -5763,8 +5727,6 @@ meta_button_function_from_string (const
 	return META_BUTTON_FUNCTION_MAXIMIZE;
     else if (strcmp (str, "close") == 0)
 	return META_BUTTON_FUNCTION_CLOSE;
-
-#ifdef HAVE_MARCO_2_17_0
     else if (strcmp (str, "shade") == 0)
 	return META_BUTTON_FUNCTION_SHADE;
     else if (strcmp (str, "above") == 0)
@@ -5777,8 +5739,6 @@ meta_button_function_from_string (const
 	return META_BUTTON_FUNCTION_UNABOVE;
     else if (strcmp (str, "unstick") == 0)
 	return META_BUTTON_FUNCTION_UNSTICK;
-#endif
-
     else
 	return META_BUTTON_FUNCTION_LAST;
 }
@@ -5788,7 +5748,6 @@ meta_button_opposite_function (MetaButto
 {
     switch (ofwhat)
     {
-#ifdef HAVE_MARCO_2_17_0
     case META_BUTTON_FUNCTION_SHADE:
 	return META_BUTTON_FUNCTION_UNSHADE;
     case META_BUTTON_FUNCTION_UNSHADE:
@@ -5803,7 +5762,6 @@ meta_button_opposite_function (MetaButto
 	return META_BUTTON_FUNCTION_UNSTICK;
     case META_BUTTON_FUNCTION_UNSTICK:
 	return META_BUTTON_FUNCTION_STICK;
-#endif
 
     default:
 	return META_BUTTON_FUNCTION_LAST;
@@ -5819,10 +5777,8 @@ meta_initialize_button_layout (MetaButto
     {
 	layout->left_buttons[i] = META_BUTTON_FUNCTION_LAST;
 	layout->right_buttons[i] = META_BUTTON_FUNCTION_LAST;
-#ifdef HAVE_MARCO_2_23_2
 	layout->left_buttons_has_spacer[i] = FALSE;
 	layout->right_buttons_has_spacer[i] = FALSE;
-#endif
     }
 }
 
@@ -5853,7 +5809,6 @@ meta_update_button_layout (const char *v
 	while (buttons[b] != NULL)
 	{
 	    f = meta_button_function_from_string (buttons[b]);
-#ifdef HAVE_MARCO_2_23_2
 	    if (i > 0 && strcmp("spacer", buttons[b]) == 0)
             {
 	       new_layout.left_buttons_has_spacer[i - 1] = TRUE;
@@ -5863,7 +5818,6 @@ meta_update_button_layout (const char *v
                   new_layout.left_buttons_has_spacer[i - 2] = TRUE;
             }
 	    else
-#endif
 	    {
 	       if (f != META_BUTTON_FUNCTION_LAST && !used[f])
 	       {
@@ -5900,7 +5854,6 @@ meta_update_button_layout (const char *v
 	    while (buttons[b] != NULL)
 	    {
 	       f = meta_button_function_from_string (buttons[b]);
-#ifdef HAVE_MARCO_2_23_2
 	       if (i > 0 && strcmp("spacer", buttons[b]) == 0)
 	       {
 		  new_layout.right_buttons_has_spacer[i - 1] = TRUE;
@@ -5909,7 +5862,6 @@ meta_update_button_layout (const char *v
 		     new_layout.right_buttons_has_spacer[i - 2] = TRUE;
 	       }
 	       else
-#endif
 	       {
 		   if (f != META_BUTTON_FUNCTION_LAST && !used[f])
 		   {
@@ -5953,14 +5905,12 @@ meta_update_button_layout (const char *v
 	for (j = 0; j < i; j++)
 	{
 	    rtl_layout.right_buttons[j] = new_layout.left_buttons[i - j - 1];
-#ifdef HAVE_MARCO_2_23_2
 	    if (j == 0)
 		rtl_layout.right_buttons_has_spacer[i - 1] =
 		    new_layout.left_buttons_has_spacer[i - j - 1];
 	    else
 		rtl_layout.right_buttons_has_spacer[j - 1] =
 		    new_layout.left_buttons_has_spacer[i - j - 1];
-#endif
 	}
 
 	i = 0;
@@ -5970,14 +5920,12 @@ meta_update_button_layout (const char *v
 	for (j = 0; j < i; j++)
 	{
 	    rtl_layout.left_buttons[j] = new_layout.right_buttons[i - j - 1];
-#ifdef HAVE_MARCO_2_23_2
 	    if (j == 0)
 		rtl_layout.left_buttons_has_spacer[i - 1] =
 		    new_layout.right_buttons_has_spacer[i - j - 1];
 	    else
 		rtl_layout.left_buttons_has_spacer[j - 1] =
 		    new_layout.right_buttons_has_spacer[i - j - 1];
-#endif
 	}
 
 	new_layout = rtl_layout;
