diff -up HTML-Template-2.10/lib/HTML/Template.pm.old HTML-Template-2.10/lib/HTML/Template.pm
--- HTML-Template-2.10/lib/HTML/Template.pm.old	2011-07-14 15:09:24.064824098 -0400
+++ HTML-Template-2.10/lib/HTML/Template.pm	2011-07-14 15:11:42.792823364 -0400
@@ -298,7 +298,7 @@ if available.  As a final attempt, the f
 directly.  See below for more information on C<HTML_TEMPLATE_ROOT>
 and the C<path> option to C<new()>.
 
-As a protection against infinitly recursive includes, an arbitrary
+As a protection against infinitely recursive includes, an arbitrary
 limit of 10 levels deep is imposed.  You can alter this limit with the
 C<max_includes> option.  See the entry for the C<max_includes> option
 below for more details.
@@ -980,7 +980,7 @@ C<!!!ZAP_VAR FOO!!!> and transforms them
     );
 
 More complicated usages are possible.  You can request that your
-filter receieve the template text as an array of lines rather than
+filter receive the template text as an array of lines rather than
 as a single scalar.  To do that you need to specify your filter using
 a hash-ref.  In this form you specify the filter using the C<sub> key
 and the desired argument format using the C<format> key.  The available
