diff -up HTML-Template-2.10/lib/HTML/Template.pm~ HTML-Template-2.10/lib/HTML/Template.pm
--- HTML-Template-2.10/lib/HTML/Template.pm~	2011-08-30 20:44:01.075540073 +0300
+++ HTML-Template-2.10/lib/HTML/Template.pm	2011-08-30 20:45:10.256004737 +0300
@@ -1,6 +1,7 @@
 package HTML::Template;
 
-$HTML::Template::VERSION = '2.10';
+# https://bugzilla.redhat.com/show_bug.cgi?id=734253
+$HTML::Template::VERSION = '2.91';
 
 =head1 NAME
 
