From: inkscape upstream
Fix compilation against libpng-1.5

=== modified file 'src/sp-image.cpp'
--- src/sp-image.cpp	2011-02-21 07:59:34 +0000
+++ src/sp-image.cpp	2011-02-21 08:57:28 +0000
@@ -387,9 +387,13 @@
 
 #if defined(PNG_iCCP_SUPPORTED)
                 {
-                    char* name = 0;
+                    png_charp name = 0;
                     int compression_type = 0;
-                    char* profile = 0;
+#if (PNG_LIBPNG_VER < 10500)
+                    png_charp profile = 0;
+#else
+                    png_bytep profile = 0;
+#endif
                     png_uint_32 proflen = 0;
                     if ( png_get_iCCP(pngPtr, infoPtr, &name, &compression_type, &profile, &proflen) ) {
 //                                         g_message("Found an iCCP chunk named [%s] with %d bytes and comp %d", name, proflen, compression_type);

