Adjust for new glib.h include rules.

diff --git a/src/application/application.h b/src/application/application.h
index fce6bd4..749790a 100644
--- a/src/application/application.h
+++ b/src/application/application.h
@@ -13,7 +13,7 @@
 #ifndef INKSCAPE_APPLICATION_APPLICATION_H
 #define INKSCAPE_APPLICATION_APPLICATION_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Gtk {
 class Main;
diff --git a/src/application/editor.h b/src/application/editor.h
index 4545022..6a11ec9 100644
--- a/src/application/editor.h
+++ b/src/application/editor.h
@@ -16,7 +16,7 @@
 #define INKSCAPE_APPLICATION_EDITOR_H
 
 #include <sigc++/sigc++.h>
-#include <glib/gslist.h>
+#include <glib.h>
 #include <glibmm/ustring.h>
 #include <set>
 #include "app-prototype.h"
diff --git a/src/attributes-test.h b/src/attributes-test.h
index 6677294..2621f93 100644
--- a/src/attributes-test.h
+++ b/src/attributes-test.h
@@ -6,7 +6,6 @@
 
 #include <vector>
 #include <glib.h>
-#include <glib/gprintf.h>
 #include "attributes.h"
 #include "streq.h"
 
diff --git a/src/attributes.cpp b/src/attributes.cpp
index 5d3a008..97b23c6 100644
--- a/src/attributes.cpp
+++ b/src/attributes.cpp
@@ -17,8 +17,7 @@
 # include "config.h"
 #endif
 
-#include <glib.h> // g_assert()
-#include <glib/ghash.h>
+#include <glib.h>
 #include "attributes.h"
 
 typedef struct {
diff --git a/src/attributes.h b/src/attributes.h
index 82ac962..4cd3459 100644
--- a/src/attributes.h
+++ b/src/attributes.h
@@ -13,8 +13,7 @@
  *
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
-#include <glib/gtypes.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 
 unsigned int sp_attribute_lookup(gchar const *key);
 unsigned char const *sp_attribute_name(unsigned int id);
diff --git a/src/bind/javabind.cpp b/src/bind/javabind.cpp
index 6dc8c9a..36d55f6 100644
--- a/src/bind/javabind.cpp
+++ b/src/bind/javabind.cpp
@@ -52,7 +52,7 @@
 #include "javabind-private.h"
 #include <path-prefix.h>
 #include <prefix.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 
 //For repr and document
 #include <document.h>
diff --git a/src/color-profile-fns.h b/src/color-profile-fns.h
index c8c51b5..41d8fc6 100644
--- a/src/color-profile-fns.h
+++ b/src/color-profile-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #if ENABLE_LCMS
 #include <vector>
 #include <glibmm/ustring.h>
diff --git a/src/color-profile.h b/src/color-profile.h
index fa8f353..029b232 100644
--- a/src/color-profile.h
+++ b/src/color-profile.h
@@ -5,7 +5,7 @@
  * SPColorProfile: SVG <color-profile> implementation
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <sp-object.h>
 #include <glibmm/ustring.h>
 #if ENABLE_LCMS
diff --git a/src/color-rgba.h b/src/color-rgba.h
index fc52b19..39b2975 100644
--- a/src/color-rgba.h
+++ b/src/color-rgba.h
@@ -12,8 +12,7 @@
 #ifndef SEEN_COLOR_RGBA_H
 #define SEEN_COLOR_RGBA_H
 
-#include <glib.h> // g_assert()
-#include <glib/gmessages.h>
+#include <glib.h>
 #include "libnr/nr-pixops.h"
 #include "decimal-round.h"
 
diff --git a/src/conn-avoid-ref.h b/src/conn-avoid-ref.h
index 653b1d7..f276634 100644
--- a/src/conn-avoid-ref.h
+++ b/src/conn-avoid-ref.h
@@ -13,7 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gslist.h>
+#include <glib.h>
 #include <stddef.h>
 #include <sigc++/connection.h>
 
diff --git a/src/debug/logger.cpp b/src/debug/logger.cpp
index f28d275..27c6d19 100644
--- a/src/debug/logger.cpp
+++ b/src/debug/logger.cpp
@@ -11,7 +11,7 @@
 
 #include <fstream>
 #include <vector>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include "inkscape-version.h"
 #include "debug/logger.h"
 #include "debug/simple-event.h"
diff --git a/src/debug/simple-event.h b/src/debug/simple-event.h
index 74301a6..ea28cf5 100644
--- a/src/debug/simple-event.h
+++ b/src/debug/simple-event.h
@@ -14,9 +14,7 @@
 
 #include <stdarg.h>
 #include <vector>
-#include <glib.h> // g_assert()
-#include <glib/gstrfuncs.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 
 #include "gc-alloc.h"
 #include "debug/event.h"
diff --git a/src/debug/timestamp.cpp b/src/debug/timestamp.cpp
index 8814f37..26a9ce0 100644
--- a/src/debug/timestamp.cpp
+++ b/src/debug/timestamp.cpp
@@ -10,8 +10,7 @@
  */
 
 
-#include <glib/gtypes.h>
-#include <glib/gmain.h>
+#include <glib.h>
 #include <glibmm/ustring.h>
 #include "debug/simple-event.h"
 
diff --git a/src/desktop-style.h b/src/desktop-style.h
index e1ca5b3..95b3bf9 100644
--- a/src/desktop-style.h
+++ b/src/desktop-style.h
@@ -13,7 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 class ColorRGBA;
 struct SPCSSAttr;
diff --git a/src/dialogs/clonetiler.cpp b/src/dialogs/clonetiler.cpp
index 55884fe..b846f48 100644
--- a/src/dialogs/clonetiler.cpp
+++ b/src/dialogs/clonetiler.cpp
@@ -12,7 +12,7 @@
 #ifdef HAVE_CONFIG_H
 # include "config.h"
 #endif
-#include <glib/gmem.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 #include <glibmm/i18n.h>
 
diff --git a/src/dir-util.cpp b/src/dir-util.cpp
index 67db036..8fe1315 100644
--- a/src/dir-util.cpp
+++ b/src/dir-util.cpp
@@ -7,11 +7,7 @@
 #include <errno.h>
 #include <string>
 #include <cstring>
-#include <glib/gutils.h>
-#include <glib/gmem.h>
-#include <glib/gerror.h>
-#include <glib/gconvert.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 /** Returns a form of \a path relative to \a base if that is easy to construct (e.g. if \a path
     appears to be in the directory specified by \a base), otherwise returns \a path.
diff --git a/src/dir-util.h b/src/dir-util.h
index 9bdfafa..6b16fcc 100644
--- a/src/dir-util.h
+++ b/src/dir-util.h
@@ -10,7 +10,7 @@
  */
 
 #include <stdlib.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 char const *sp_relative_path_from_path(char const *path, char const *base);
 char const *sp_extension_from_path(char const *path);
diff --git a/src/display/canvas-bpath.h b/src/display/canvas-bpath.h
index b97bbcc..6db6fa3 100644
--- a/src/display/canvas-bpath.h
+++ b/src/display/canvas-bpath.h
@@ -13,7 +13,7 @@
  *
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include <display/sp-canvas.h>
 
diff --git a/src/display/curve.cpp b/src/display/curve.cpp
index 73b8dc3..b17ef51 100644
--- a/src/display/curve.cpp
+++ b/src/display/curve.cpp
@@ -19,7 +19,7 @@
 
 #include "display/curve.h"
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <2geom/pathvector.h>
 #include <2geom/sbasis-geometric.h>
 #include <2geom/sbasis-to-bezier.h>
diff --git a/src/display/curve.h b/src/display/curve.h
index fe07201..64731c4 100644
--- a/src/display/curve.h
+++ b/src/display/curve.h
@@ -15,8 +15,7 @@
  * Released under GNU GPL
  */
 
-#include <glib/gtypes.h>
-#include <glib/gslist.h>
+#include <glib.h>
 
 #include <2geom/forward.h>
 
diff --git a/src/display/gnome-canvas-acetate.h b/src/display/gnome-canvas-acetate.h
index 40574e1..2f84931 100644
--- a/src/display/gnome-canvas-acetate.h
+++ b/src/display/gnome-canvas-acetate.h
@@ -15,7 +15,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "display/sp-canvas.h"
 
 
diff --git a/src/display/nr-3dutils.cpp b/src/display/nr-3dutils.cpp
index 89c2194..db8800f 100644
--- a/src/display/nr-3dutils.cpp
+++ b/src/display/nr-3dutils.cpp
@@ -9,7 +9,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gmessages.h>
+#include <glib.h>
 
 #include "libnr/nr-pixblock.h"
 #include "display/nr-3dutils.h"
diff --git a/src/display/nr-arena.h b/src/display/nr-arena.h
index 76b071e..1d23511 100644
--- a/src/display/nr-arena.h
+++ b/src/display/nr-arena.h
@@ -13,7 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gmacros.h>
+#include <glib.h>
 
 #include "display/rendermode.h"
 
diff --git a/src/display/nr-filter-diffuselighting.cpp b/src/display/nr-filter-diffuselighting.cpp
index 0fe4c59..1e08e11 100644
--- a/src/display/nr-filter-diffuselighting.cpp
+++ b/src/display/nr-filter-diffuselighting.cpp
@@ -10,7 +10,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gmessages.h>
+#include <glib.h>
 
 #include "display/nr-3dutils.h"
 #include "display/nr-arena-item.h"
diff --git a/src/display/nr-filter-specularlighting.cpp b/src/display/nr-filter-specularlighting.cpp
index 6a6ce38..dbe6764 100644
--- a/src/display/nr-filter-specularlighting.cpp
+++ b/src/display/nr-filter-specularlighting.cpp
@@ -10,7 +10,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <cmath>
 
 #include "display/nr-3dutils.h"
diff --git a/src/display/nr-plain-stuff.cpp b/src/display/nr-plain-stuff.cpp
index 62a6110..e37ad96 100644
--- a/src/display/nr-plain-stuff.cpp
+++ b/src/display/nr-plain-stuff.cpp
@@ -11,7 +11,7 @@
  * Released under GNU GPL
  */
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libnr/nr-pixops.h>
 #include "nr-plain-stuff.h"
 
diff --git a/src/display/nr-plain-stuff.h b/src/display/nr-plain-stuff.h
index c568f38..99176a3 100644
--- a/src/display/nr-plain-stuff.h
+++ b/src/display/nr-plain-stuff.h
@@ -12,7 +12,7 @@
  * Released under GNU GPL
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 void nr_render_checkerboard_rgb (guchar *px, gint w, gint h, gint rs, gint xoff, gint yoff);
 void nr_render_checkerboard_rgb_custom (guchar *px, gint w, gint h, gint rs, gint xoff, gint yoff, guint32 c0, guint32 c1, gint sizep2);
diff --git a/src/display/sodipodi-ctrlrect.h b/src/display/sodipodi-ctrlrect.h
index 7e5f515..7e00565 100644
--- a/src/display/sodipodi-ctrlrect.h
+++ b/src/display/sodipodi-ctrlrect.h
@@ -16,7 +16,7 @@
  *
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "sp-canvas.h"
 
 #define SP_TYPE_CTRLRECT (sp_ctrlrect_get_type ())
diff --git a/src/display/sp-canvas.h b/src/display/sp-canvas.h
index 3d23bf0..14c1410 100644
--- a/src/display/sp-canvas.h
+++ b/src/display/sp-canvas.h
@@ -27,7 +27,7 @@
 # endif
 #endif
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gdk/gdk.h>
 #include <gtk/gtk.h>
 
diff --git a/src/document-subset.cpp b/src/document-subset.cpp
index 1988865..020e6af 100644
--- a/src/document-subset.cpp
+++ b/src/document-subset.cpp
@@ -12,7 +12,7 @@
 #include "document.h"
 #include "sp-object.h"
 
-#include <glib/gmessages.h>
+#include <glib.h>
 
 #include <sigc++/signal.h>
 #include <sigc++/functors/mem_fun.h>
diff --git a/src/draw-anchor.h b/src/draw-anchor.h
index 0277616..af281d3 100644
--- a/src/draw-anchor.h
+++ b/src/draw-anchor.h
@@ -5,7 +5,7 @@
  * Drawing anchors. 
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <2geom/point.h>
 
 struct SPDrawContext;
diff --git a/src/dyna-draw-context.cpp b/src/dyna-draw-context.cpp
index bb8e690..7a61a88 100644
--- a/src/dyna-draw-context.cpp
+++ b/src/dyna-draw-context.cpp
@@ -38,7 +38,7 @@
 #include <2geom/pathvector.h>
 #include <2geom/bezier-utils.h>
 #include "display/curve.h"
-#include <glib/gmem.h>
+#include <glib.h>
 #include "macros.h"
 #include "document.h"
 #include "selection.h"
diff --git a/src/eraser-context.cpp b/src/eraser-context.cpp
index c198d93..c0bdec6 100644
--- a/src/eraser-context.cpp
+++ b/src/eraser-context.cpp
@@ -36,7 +36,7 @@
 #include "display/canvas-bpath.h"
 #include <2geom/bezier-utils.h>
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include "macros.h"
 #include "document.h"
 #include "selection.h"
diff --git a/src/extension/internal/bitmap/imagemagick.cpp b/src/extension/internal/bitmap/imagemagick.cpp
index e907612..9a5a506 100644
--- a/src/extension/internal/bitmap/imagemagick.cpp
+++ b/src/extension/internal/bitmap/imagemagick.cpp
@@ -15,6 +15,7 @@
 #include <gtkmm/spinbutton.h>
 #include <gtkmm.h>
 
+#include <glib.h>
 #include <glib/gstdio.h>
 
 #include "desktop.h"
diff --git a/src/extension/internal/cairo-render-context.cpp b/src/extension/internal/cairo-render-context.cpp
index 596f1fe..d65722b 100644
--- a/src/extension/internal/cairo-render-context.cpp
+++ b/src/extension/internal/cairo-render-context.cpp
@@ -29,7 +29,7 @@
 #include <errno.h>
 #include <2geom/pathvector.h>
 
-#include <glib/gmem.h>
+#include <glib.h>
 
 #include <glibmm/i18n.h>
 #include "display/nr-arena.h"
diff --git a/src/extension/internal/cairo-renderer.cpp b/src/extension/internal/cairo-renderer.cpp
index 7060251..489af18 100644
--- a/src/extension/internal/cairo-renderer.cpp
+++ b/src/extension/internal/cairo-renderer.cpp
@@ -33,7 +33,7 @@
 #include <2geom/transforms.h>
 #include <2geom/pathvector.h>
 
-#include <glib/gmem.h>
+#include <glib.h>
 
 #include <glibmm/i18n.h>
 #include "display/nr-arena.h"
diff --git a/src/extension/internal/gdkpixbuf-input.cpp b/src/extension/internal/gdkpixbuf-input.cpp
index df7f348..a999c43 100644
--- a/src/extension/internal/gdkpixbuf-input.cpp
+++ b/src/extension/internal/gdkpixbuf-input.cpp
@@ -1,7 +1,7 @@
 #ifdef HAVE_CONFIG_H
 # include <config.h>
 #endif
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <glibmm/i18n.h>
 #include "document-private.h"
 #include <dir-util.h>
diff --git a/src/extension/internal/pdfinput/svg-builder.h b/src/extension/internal/pdfinput/svg-builder.h
index f0062bb..7f19151 100644
--- a/src/extension/internal/pdfinput/svg-builder.h
+++ b/src/extension/internal/pdfinput/svg-builder.h
@@ -49,7 +49,7 @@ class XRef;
 class SPCSSAttr;
 
 #include <vector>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace Extension {
diff --git a/src/extension/internal/win32.cpp b/src/extension/internal/win32.cpp
index 21f2788..18855b2 100644
--- a/src/extension/internal/win32.cpp
+++ b/src/extension/internal/win32.cpp
@@ -13,7 +13,7 @@
 # include "config.h"
 #endif
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include <libnr/nr-macros.h>
 #include <2geom/transforms.h>
 
diff --git a/src/extract-uri.h b/src/extract-uri.h
index 1975d9b..884b43a 100644
--- a/src/extract-uri.h
+++ b/src/extract-uri.h
@@ -1,7 +1,7 @@
 #ifndef SEEN_EXTRACT_URI_H
 #define SEEN_EXTRACT_URI_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 gchar *extract_uri(gchar const *s, gchar const** endptr = 0);
 
diff --git a/src/file.cpp b/src/file.cpp
index 1128a5a..3714560 100644
--- a/src/file.cpp
+++ b/src/file.cpp
@@ -27,7 +27,7 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <glib/gmem.h>
+#include <glib.h>
 #include <glibmm/i18n.h>
 #include <libnr/nr-pixops.h>
 
diff --git a/src/file.h b/src/file.h
index dec7e3f..88d0fa0 100644
--- a/src/file.h
+++ b/src/file.h
@@ -16,7 +16,7 @@
  */
 
 #include <gtkmm.h>
-#include <glib/gslist.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 
 #include "extension/extension-forward.h"
diff --git a/src/filters/blend-fns.h b/src/filters/blend-fns.h
index f08ed9d..39bf2b4 100644
--- a/src/filters/blend-fns.h
+++ b/src/filters/blend-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/colormatrix-fns.h b/src/filters/colormatrix-fns.h
index 3a4a8d3..e023d4d 100644
--- a/src/filters/colormatrix-fns.h
+++ b/src/filters/colormatrix-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/componenttransfer-fns.h b/src/filters/componenttransfer-fns.h
index 4998377..06248d6 100644
--- a/src/filters/componenttransfer-fns.h
+++ b/src/filters/componenttransfer-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/composite-fns.h b/src/filters/composite-fns.h
index c79cb17..51610fd 100644
--- a/src/filters/composite-fns.h
+++ b/src/filters/composite-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/convolvematrix-fns.h b/src/filters/convolvematrix-fns.h
index 76baf7f..931ad94 100644
--- a/src/filters/convolvematrix-fns.h
+++ b/src/filters/convolvematrix-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/diffuselighting-fns.h b/src/filters/diffuselighting-fns.h
index b91ed80..18a9375 100644
--- a/src/filters/diffuselighting-fns.h
+++ b/src/filters/diffuselighting-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/displacementmap-fns.h b/src/filters/displacementmap-fns.h
index 6d92c6b..4b8ed74 100644
--- a/src/filters/displacementmap-fns.h
+++ b/src/filters/displacementmap-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/flood-fns.h b/src/filters/flood-fns.h
index 8cc5072..0cd4bd3 100644
--- a/src/filters/flood-fns.h
+++ b/src/filters/flood-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/image-fns.h b/src/filters/image-fns.h
index 0a8b453..5748486 100644
--- a/src/filters/image-fns.h
+++ b/src/filters/image-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/merge-fns.h b/src/filters/merge-fns.h
index 24bda1a..9787d59 100644
--- a/src/filters/merge-fns.h
+++ b/src/filters/merge-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/morphology-fns.h b/src/filters/morphology-fns.h
index a055040..a6f1b58 100644
--- a/src/filters/morphology-fns.h
+++ b/src/filters/morphology-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/offset-fns.h b/src/filters/offset-fns.h
index 38561c1..32fbda7 100644
--- a/src/filters/offset-fns.h
+++ b/src/filters/offset-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/specularlighting-fns.h b/src/filters/specularlighting-fns.h
index bd48ba6..70cd5d4 100644
--- a/src/filters/specularlighting-fns.h
+++ b/src/filters/specularlighting-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/tile-fns.h b/src/filters/tile-fns.h
index b7c4c5f..4a99a51 100644
--- a/src/filters/tile-fns.h
+++ b/src/filters/tile-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/filters/turbulence-fns.h b/src/filters/turbulence-fns.h
index 43b4450..de9f228 100644
--- a/src/filters/turbulence-fns.h
+++ b/src/filters/turbulence-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/gc-anchored.h b/src/gc-anchored.h
index 917930a..119b86a 100644
--- a/src/gc-anchored.h
+++ b/src/gc-anchored.h
@@ -11,7 +11,7 @@
 #ifndef SEEN_INKSCAPE_GC_ANCHORED_H
 #define SEEN_INKSCAPE_GC_ANCHORED_H
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include "gc-managed.h"
 
 namespace Inkscape {
diff --git a/src/gc-core.h b/src/gc-core.h
index af327dc..b99e1c5 100644
--- a/src/gc-core.h
+++ b/src/gc-core.h
@@ -24,7 +24,7 @@
 #else
 # include <gc.h>
 #endif
-#include <glib/gmain.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace GC {
diff --git a/src/gc.cpp b/src/gc.cpp
index 8622059..5a636e7 100644
--- a/src/gc.cpp
+++ b/src/gc.cpp
@@ -13,7 +13,7 @@
 #include <stdexcept>
 #include <cstring>
 #include <string>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <sigc++/functors/ptr_fun.h>
 #include <glibmm/main.h>
 #include <cstddef>
diff --git a/src/gradient-drag.h b/src/gradient-drag.h
index 40ab065..4fa471b 100644
--- a/src/gradient-drag.h
+++ b/src/gradient-drag.h
@@ -14,7 +14,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gslist.h>
+#include <glib.h>
 #include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <vector>
diff --git a/src/help.h b/src/help.h
index ef5d703..7241c82 100644
--- a/src/help.h
+++ b/src/help.h
@@ -13,7 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 
 void sp_help_about(void);
diff --git a/src/helper/gnome-utils.h b/src/helper/gnome-utils.h
index 0a28c95..62676c7 100644
--- a/src/helper/gnome-utils.h
+++ b/src/helper/gnome-utils.h
@@ -15,8 +15,7 @@
 #ifndef __GNOME_UTILS_H__
 #define __GNOME_UTILS_H__
 
-#include <glib/gtypes.h>
-#include <glib/glist.h>
+#include <glib.h>
 
 GList *gnome_uri_list_extract_uris(gchar const *uri_list);
 
diff --git a/src/helper/pixbuf-ops.cpp b/src/helper/pixbuf-ops.cpp
index 3be63aa..d7d2e57 100644
--- a/src/helper/pixbuf-ops.cpp
+++ b/src/helper/pixbuf-ops.cpp
@@ -18,7 +18,6 @@
 #include <interface.h>
 #include <libnr/nr-pixops.h>
 #include <glib.h>
-#include <glib/gmessages.h>
 #include <png.h>
 #include "png-write.h"
 #include <display/nr-arena-item.h>
diff --git a/src/helper/pixbuf-ops.h b/src/helper/pixbuf-ops.h
index a985be2..af57327 100644
--- a/src/helper/pixbuf-ops.h
+++ b/src/helper/pixbuf-ops.h
@@ -12,7 +12,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 struct SPDocument;
 
diff --git a/src/helper/png-write.cpp b/src/helper/png-write.cpp
index b1c135d..5cb37d9 100644
--- a/src/helper/png-write.cpp
+++ b/src/helper/png-write.cpp
@@ -20,7 +20,7 @@
 #include <libnr/nr-pixops.h>
 #include <libnr/nr-translate-scale-ops.h>
 #include <2geom/rect.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <png.h>
 #include "png-write.h"
 #include "io/sys.h"
diff --git a/src/helper/png-write.h b/src/helper/png-write.h
index 83321aa..f8ba4bb 100644
--- a/src/helper/png-write.h
+++ b/src/helper/png-write.h
@@ -12,7 +12,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <2geom/forward.h>
 struct SPDocument;
 
diff --git a/src/helper/stlport.h b/src/helper/stlport.h
index c9389e8..c7b00eb 100644
--- a/src/helper/stlport.h
+++ b/src/helper/stlport.h
@@ -3,8 +3,7 @@
 
 
 #include <list>
-#include <glib/glist.h>
-#include <glib/gslist.h>
+#include <glib.h>
 
 template <typename T>
 class StlConv {
diff --git a/src/helper/stock-items.h b/src/helper/stock-items.h
index ddad554..4576d51 100644
--- a/src/helper/stock-items.h
+++ b/src/helper/stock-items.h
@@ -12,7 +12,7 @@
  *
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include <forward.h>
 
diff --git a/src/helper/unit-menu.h b/src/helper/unit-menu.h
index cdaa826..fc8a600 100644
--- a/src/helper/unit-menu.h
+++ b/src/helper/unit-menu.h
@@ -10,7 +10,7 @@
  *
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 
 #include <helper/helper-forward.h>
diff --git a/src/helper/units.h b/src/helper/units.h
index 8dc62fe..93bd706 100644
--- a/src/helper/units.h
+++ b/src/helper/units.h
@@ -15,9 +15,7 @@
  *
  */
 
-#include <glib/gmessages.h>
-#include <glib/gslist.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "sp-metric.h"
 
 
diff --git a/src/inkscape.h b/src/inkscape.h
index ca28942..5538d4f 100644
--- a/src/inkscape.h
+++ b/src/inkscape.h
@@ -13,7 +13,7 @@
  */
 
 #include <list>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 struct SPDesktop;
 struct SPDocument;
diff --git a/src/inkview.cpp b/src/inkview.cpp
index 6fe989e..9d76378 100644
--- a/src/inkview.cpp
+++ b/src/inkview.cpp
@@ -39,7 +39,7 @@
 #include <sys/stat.h>
 #include <locale.h>
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include <libnr/nr-macros.h>
 
 // #include <stropts.h>
diff --git a/src/io/inkjar.h b/src/io/inkjar.h
index 0fe088b..ea4b0ee 100644
--- a/src/io/inkjar.h
+++ b/src/io/inkjar.h
@@ -26,8 +26,7 @@
 # endif
 #endif
 
-#include <glib/garray.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkjar {
 
diff --git a/src/io/resource.cpp b/src/io/resource.cpp
index 1f6f845..14b8beb 100644
--- a/src/io/resource.cpp
+++ b/src/io/resource.cpp
@@ -16,10 +16,7 @@
 #include "config.h"
 #endif
 
-#include <glib.h> // g_assert()
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gfileutils.h>
+#include <glib.h>
 #include "path-prefix.h"
 #include "inkscape.h"
 #include "io/resource.h"
diff --git a/src/io/sys.cpp b/src/io/sys.cpp
index 143108a..529f7ce 100644
--- a/src/io/sys.cpp
+++ b/src/io/sys.cpp
@@ -16,8 +16,6 @@
 #endif
 
 #include <glib.h>
-#include <glib/gstdio.h>
-#include <glib/gutils.h>
 #include <glibmm/fileutils.h>
 #if GLIB_CHECK_VERSION(2,6,0)
     #include <glib/gstdio.h>
diff --git a/src/io/sys.h b/src/io/sys.h
index 8623f6b..83ffdb4 100644
--- a/src/io/sys.h
+++ b/src/io/sys.h
@@ -15,9 +15,7 @@
 #include <stdio.h>
 #include <sys/stat.h>
 #include <sys/types.h>
-#include <glib/gtypes.h>
-#include <glib/gdir.h>
-#include <glib/gfileutils.h>
+#include <glib.h>
 #include <glibmm/spawn.h>
 #include <string>
 
diff --git a/src/knot-holder-entity.h b/src/knot-holder-entity.h
index c8fd29d..84afaa4 100644
--- a/src/knot-holder-entity.h
+++ b/src/knot-holder-entity.h
@@ -17,7 +17,7 @@
  * Released under GNU GPL
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "knot.h"
 #include <2geom/forward.h>
 #include "snapper.h"
diff --git a/src/knotholder.h b/src/knotholder.h
index 0b37d21..7feb1c7 100644
--- a/src/knotholder.h
+++ b/src/knotholder.h
@@ -17,7 +17,7 @@
  *
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "knot-enums.h"
 #include "forward.h"
 #include "libnr/nr-forward.h"
diff --git a/src/libcroco/cr-libxml-node-iface.h b/src/libcroco/cr-libxml-node-iface.h
index 5da0d9a..b4a6212 100644
--- a/src/libcroco/cr-libxml-node-iface.h
+++ b/src/libcroco/cr-libxml-node-iface.h
@@ -1,7 +1,7 @@
 #ifndef __CR_LIBXML_NODE_IFACE_H__
 #define __CR_LIBXML_NODE_IFACE_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include "cr-node-iface.h"
 
 G_BEGIN_DECLS
diff --git a/src/libcroco/cr-node-iface.h b/src/libcroco/cr-node-iface.h
index 9c2d30e..01898d6 100644
--- a/src/libcroco/cr-node-iface.h
+++ b/src/libcroco/cr-node-iface.h
@@ -1,8 +1,7 @@
 #ifndef __CR_NODE_IFACE_H__
 #define __CR_NODE_IFACE_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
diff --git a/src/libgdl/gdl-stock.h b/src/libgdl/gdl-stock.h
index cb6f7ab..459d00c 100644
--- a/src/libgdl/gdl-stock.h
+++ b/src/libgdl/gdl-stock.h
@@ -22,7 +22,7 @@
 #ifndef __GDL_STOCK_H__
 #define __GDL_STOCK_H__
 
-#include <glib/gmacros.h>   // G_BEGIN_DECLS
+#include <glib.h>   // G_BEGIN_DECLS
 
 G_BEGIN_DECLS
 
diff --git a/src/libnr/in-svg-plane-test.h b/src/libnr/in-svg-plane-test.h
index 304182f..1d51125 100644
--- a/src/libnr/in-svg-plane-test.h
+++ b/src/libnr/in-svg-plane-test.h
@@ -1,6 +1,6 @@
 #include <cxxtest/TestSuite.h>
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <cmath>
 
 #include "libnr/in-svg-plane.h"
diff --git a/src/libnr/nr-gradient.cpp b/src/libnr/nr-gradient.cpp
index e6eb9b7..32dac7d 100644
--- a/src/libnr/nr-gradient.cpp
+++ b/src/libnr/nr-gradient.cpp
@@ -26,7 +26,7 @@
 #include <libnr/nr-blit.h>
 #include <libnr/nr-gradient.h>
 #include <libnr/nr-matrix-ops.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <stdio.h>
 
 /* Common */
diff --git a/src/libnr/nr-i-coord.h b/src/libnr/nr-i-coord.h
index f87dea3..c9e6b3f 100644
--- a/src/libnr/nr-i-coord.h
+++ b/src/libnr/nr-i-coord.h
@@ -1,7 +1,7 @@
 #ifndef SEEN_NR_I_COORD_H
 #define SEEN_NR_I_COORD_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace NR {
 
diff --git a/src/libnr/nr-matrix.h b/src/libnr/nr-matrix.h
index b1f9d58..471381c 100644
--- a/src/libnr/nr-matrix.h
+++ b/src/libnr/nr-matrix.h
@@ -17,8 +17,7 @@
  * This code is in public domain.
  */
 
-#include <glib.h> // g_assert()
-#include <glib/gmessages.h>
+#include <glib.h>
 
 #include "libnr/nr-coord.h"
 #include "libnr/nr-values.h"
diff --git a/src/libnr/nr-object.cpp b/src/libnr/nr-object.cpp
index d92052d..e4268f1 100644
--- a/src/libnr/nr-object.cpp
+++ b/src/libnr/nr-object.cpp
@@ -15,7 +15,7 @@
 
 #include <typeinfo>
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include <libnr/nr-macros.h>
 
 #include "nr-object.h"
diff --git a/src/libnr/nr-object.h b/src/libnr/nr-object.h
index 2691302..da89385 100644
--- a/src/libnr/nr-object.h
+++ b/src/libnr/nr-object.h
@@ -15,7 +15,7 @@
 #include "config.h"
 #endif
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "gc-managed.h"
 #include "gc-finalized.h"
 #include "gc-anchored.h"
diff --git a/src/libnr/nr-pixblock-pattern.cpp b/src/libnr/nr-pixblock-pattern.cpp
index aa32462..a09c957 100644
--- a/src/libnr/nr-pixblock-pattern.cpp
+++ b/src/libnr/nr-pixblock-pattern.cpp
@@ -10,7 +10,7 @@
  */
 
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include "nr-pixops.h"
 #include "nr-pixblock-pattern.h"
 
diff --git a/src/libnr/nr-pixblock.cpp b/src/libnr/nr-pixblock.cpp
index 6b2b12b..a881a37 100644
--- a/src/libnr/nr-pixblock.cpp
+++ b/src/libnr/nr-pixblock.cpp
@@ -13,7 +13,7 @@
 #include <cstring>
 #include <string>
 #include <string.h>
-#include <glib/gmem.h>
+#include <glib.h>
 #include "nr-pixblock.h"
 
 /// Size of buffer that needs no allocation (default 4).
diff --git a/src/libnr/nr-point-fns-test.h b/src/libnr/nr-point-fns-test.h
index 7d28c9c..eeb28f4 100644
--- a/src/libnr/nr-point-fns-test.h
+++ b/src/libnr/nr-point-fns-test.h
@@ -3,7 +3,7 @@
 
 #include <cassert>
 #include <cmath>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <stdlib.h>
 
 #include "libnr/nr-point-fns.h"
diff --git a/src/libnr/nr-rotate-fns-test.h b/src/libnr/nr-rotate-fns-test.h
index e3bfe30..d1c01cd 100644
--- a/src/libnr/nr-rotate-fns-test.h
+++ b/src/libnr/nr-rotate-fns-test.h
@@ -1,7 +1,7 @@
 #include <cxxtest/TestSuite.h>
 
 #include <cmath>
-#include <glib/gmacros.h>
+#include <glib.h>
 
 #include <libnr/nr-rotate-fns.h>
 
diff --git a/src/libnrtype/FontFactory.cpp b/src/libnrtype/FontFactory.cpp
index 067254b..4130bfc 100644
--- a/src/libnrtype/FontFactory.cpp
+++ b/src/libnrtype/FontFactory.cpp
@@ -15,7 +15,7 @@
 #endif
 
 #include <glibmm.h>
-#include <glib/gmem.h>
+#include <glib.h>
 #include <glibmm/i18n.h> // _()
 #include <pango/pangoft2.h>
 #include "libnrtype/FontFactory.h"
diff --git a/src/libnrtype/Layout-TNG-Output.cpp b/src/libnrtype/Layout-TNG-Output.cpp
index f34b93d..ab03b16 100644
--- a/src/libnrtype/Layout-TNG-Output.cpp
+++ b/src/libnrtype/Layout-TNG-Output.cpp
@@ -8,7 +8,7 @@
  *
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
-#include <glib/gmem.h>
+#include <glib.h>
 #include "Layout-TNG.h"
 #include "display/nr-arena-glyphs.h"
 #include "style.h"
diff --git a/src/libnrtype/nr-type-primitives.cpp b/src/libnrtype/nr-type-primitives.cpp
index 34b1e43..698f3f8 100644
--- a/src/libnrtype/nr-type-primitives.cpp
+++ b/src/libnrtype/nr-type-primitives.cpp
@@ -14,7 +14,7 @@
 
 #include <stdlib.h>
 #include <string.h>
-#include <glib/gmem.h>
+#include <glib.h>
 #include <libnr/nr-macros.h>
 #include "nr-type-primitives.h"
 
diff --git a/src/libnrtype/nr-type-primitives.h b/src/libnrtype/nr-type-primitives.h
index 92b94e9..9bb181c 100644
--- a/src/libnrtype/nr-type-primitives.h
+++ b/src/libnrtype/nr-type-primitives.h
@@ -11,7 +11,7 @@
  * This code is in public domain
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 struct NRNameList;
 struct NRStyleList;
diff --git a/src/livarot/AlphaLigne.cpp b/src/livarot/AlphaLigne.cpp
index f878c1b..5b8321b 100644
--- a/src/livarot/AlphaLigne.cpp
+++ b/src/livarot/AlphaLigne.cpp
@@ -12,7 +12,7 @@
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <glib/gmem.h>
+#include <glib.h>
 
 AlphaLigne::AlphaLigne(int iMin,int iMax)
 {
diff --git a/src/livarot/BitLigne.cpp b/src/livarot/BitLigne.cpp
index c4c1346..d7cce26 100644
--- a/src/livarot/BitLigne.cpp
+++ b/src/livarot/BitLigne.cpp
@@ -15,7 +15,7 @@
 #include <string>
 #include <cmath>
 #include <cstdio>
-#include <glib/gmem.h>
+#include <glib.h>
 
 BitLigne::BitLigne(int ist,int ien,float iScale)
 {
diff --git a/src/livarot/PathSimplify.cpp b/src/livarot/PathSimplify.cpp
index 0f440de..efcdfd4 100644
--- a/src/livarot/PathSimplify.cpp
+++ b/src/livarot/PathSimplify.cpp
@@ -6,7 +6,7 @@
  *
  */
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include <libnr/nr-point-matrix-ops.h>
 #include "livarot/Path.h"
 #include "livarot/path-description.h"
diff --git a/src/livarot/Shape.cpp b/src/livarot/Shape.cpp
index 9107844..e260e97 100644
--- a/src/livarot/Shape.cpp
+++ b/src/livarot/Shape.cpp
@@ -8,7 +8,7 @@
 
 #include <cstdio>
 #include <cstdlib>
-#include <glib/gmem.h>
+#include <glib.h>
 #include "Shape.h"
 #include "livarot/sweep-event-queue.h"
 #include "livarot/sweep-tree-list.h"
diff --git a/src/livarot/ShapeSweep.cpp b/src/livarot/ShapeSweep.cpp
index 00a0dd9..d3d5ef1 100644
--- a/src/livarot/ShapeSweep.cpp
+++ b/src/livarot/ShapeSweep.cpp
@@ -9,7 +9,7 @@
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
-#include <glib/gmem.h>
+#include <glib.h>
 #include "Shape.h"
 #include "livarot/sweep-event-queue.h"
 #include "livarot/sweep-tree-list.h"
diff --git a/src/livarot/int-line.cpp b/src/livarot/int-line.cpp
index c1e388f..d03d62c 100644
--- a/src/livarot/int-line.cpp
+++ b/src/livarot/int-line.cpp
@@ -9,7 +9,7 @@
  *
  */
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include <cmath>
 #include <cstring>
 #include <string>
diff --git a/src/livarot/sweep-event.cpp b/src/livarot/sweep-event.cpp
index 88d895e..28fead2 100644
--- a/src/livarot/sweep-event.cpp
+++ b/src/livarot/sweep-event.cpp
@@ -1,4 +1,4 @@
-#include <glib/gmem.h>
+#include <glib.h>
 #include "livarot/sweep-event-queue.h"
 #include "livarot/sweep-tree.h"
 #include "livarot/sweep-event.h"
diff --git a/src/livarot/sweep-tree-list.cpp b/src/livarot/sweep-tree-list.cpp
index 1d2dcec..5d3f69e 100644
--- a/src/livarot/sweep-tree-list.cpp
+++ b/src/livarot/sweep-tree-list.cpp
@@ -1,4 +1,4 @@
-#include <glib/gmem.h>
+#include <glib.h>
 #include "livarot/sweep-tree.h"
 #include "livarot/sweep-tree-list.h"
 
diff --git a/src/live_effects/parameter/array.h b/src/live_effects/parameter/array.h
index e5f2301..ca22a07 100644
--- a/src/live_effects/parameter/array.h
+++ b/src/live_effects/parameter/array.h
@@ -11,7 +11,7 @@
 
 #include <vector>
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include <gtkmm/tooltips.h>
 
diff --git a/src/live_effects/parameter/bool.h b/src/live_effects/parameter/bool.h
index 8f5196d..851476f 100644
--- a/src/live_effects/parameter/bool.h
+++ b/src/live_effects/parameter/bool.h
@@ -9,7 +9,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include "live_effects/parameter/parameter.h"
 
diff --git a/src/live_effects/parameter/enum.h b/src/live_effects/parameter/enum.h
index 05f3bdd..2d2268e 100644
--- a/src/live_effects/parameter/enum.h
+++ b/src/live_effects/parameter/enum.h
@@ -9,7 +9,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include "ui/widget/registered-enums.h"
 #include <gtkmm/tooltips.h>
diff --git a/src/live_effects/parameter/path.h b/src/live_effects/parameter/path.h
index 8fe9382..792c98a 100644
--- a/src/live_effects/parameter/path.h
+++ b/src/live_effects/parameter/path.h
@@ -9,7 +9,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <2geom/path.h>
 
 #include <gtkmm/tooltips.h>
diff --git a/src/live_effects/parameter/point.h b/src/live_effects/parameter/point.h
index ec61fcd..38fab1c 100644
--- a/src/live_effects/parameter/point.h
+++ b/src/live_effects/parameter/point.h
@@ -9,7 +9,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <2geom/point.h>
 
 #include <gtkmm/tooltips.h>
diff --git a/src/live_effects/parameter/text.h b/src/live_effects/parameter/text.h
index 8539a80..8265430 100644
--- a/src/live_effects/parameter/text.h
+++ b/src/live_effects/parameter/text.h
@@ -13,7 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include "display/canvas-bpath.h"
 #include "live_effects/parameter/parameter.h"
diff --git a/src/live_effects/parameter/vector.h b/src/live_effects/parameter/vector.h
index a4c29d3..09581d2 100644
--- a/src/live_effects/parameter/vector.h
+++ b/src/live_effects/parameter/vector.h
@@ -9,7 +9,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <2geom/point.h>
 
 #include <gtkmm/tooltips.h>
diff --git a/src/main-cmdlineact.h b/src/main-cmdlineact.h
index aca039f..03f0eb0 100644
--- a/src/main-cmdlineact.h
+++ b/src/main-cmdlineact.h
@@ -15,7 +15,7 @@
  * Released under GNU GPL v2.x, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 
diff --git a/src/main.cpp b/src/main.cpp
index d4b8271..5852d36 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -48,7 +48,6 @@
 
 #include <libxml/tree.h>
 #include <glib.h>
-#include <glib/gprintf.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
 
diff --git a/src/message-context.cpp b/src/message-context.cpp
index 5055f41..6182c2c 100644
--- a/src/message-context.cpp
+++ b/src/message-context.cpp
@@ -9,7 +9,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include "message-context.h"
 #include "message-stack.h"
 
diff --git a/src/message-stack.cpp b/src/message-stack.cpp
index 05aec87..1c407e3 100644
--- a/src/message-stack.cpp
+++ b/src/message-stack.cpp
@@ -10,7 +10,7 @@
  */
 
 #include <string.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <cstring>
 #include <string>
 #include "message-stack.h"
diff --git a/src/modifier-fns.h b/src/modifier-fns.h
index 02d3e8f..368791b 100644
--- a/src/modifier-fns.h
+++ b/src/modifier-fns.h
@@ -12,7 +12,7 @@
  */
 
 #include <gdk/gdk.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 inline bool
 mod_shift(guint const state)
diff --git a/src/number-opt-number.h b/src/number-opt-number.h
index 0025f2d..6a0e72f 100644
--- a/src/number-opt-number.h
+++ b/src/number-opt-number.h
@@ -18,7 +18,6 @@
 #endif
 
 #include <glib.h>
-#include <glib/gprintf.h>
 //todo: use glib instead of stdlib
 #include <stdlib.h>
 #include "svg/stringstream.h"
diff --git a/src/object-hierarchy.h b/src/object-hierarchy.h
index c127a48..fa23219 100644
--- a/src/object-hierarchy.h
+++ b/src/object-hierarchy.h
@@ -17,7 +17,7 @@
 #include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/signal.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 
 class SPObject;
 
diff --git a/src/path-chemistry.cpp b/src/path-chemistry.cpp
index c44ab5b..c97b7c7 100644
--- a/src/path-chemistry.cpp
+++ b/src/path-chemistry.cpp
@@ -22,7 +22,7 @@
 #include "xml/repr.h"
 #include "svg/svg.h"
 #include "display/curve.h"
-#include <glib/gmem.h>
+#include <glib.h>
 #include <glibmm/i18n.h>
 #include "sp-path.h"
 #include "sp-text.h"
diff --git a/src/removeoverlap.h b/src/removeoverlap.h
index 5b16e70..1ba4157 100644
--- a/src/removeoverlap.h
+++ b/src/removeoverlap.h
@@ -13,7 +13,7 @@
 #ifndef SEEN_REMOVEOVERLAP_H
 #define SEEN_REMOVEOVERLAP_H
 
-#include <glib/gslist.h>
+#include <glib.h>
 
 void removeoverlap(GSList const *items, double xGap, double yGap);
 
diff --git a/src/sp-conn-end-pair.h b/src/sp-conn-end-pair.h
index bc5fa12..7ba42e9 100644
--- a/src/sp-conn-end-pair.h
+++ b/src/sp-conn-end-pair.h
@@ -11,7 +11,7 @@
  *
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include "forward.h"
 #include "libnr/nr-point.h"
diff --git a/src/sp-conn-end.h b/src/sp-conn-end.h
index 052e8dd..032f12e 100644
--- a/src/sp-conn-end.h
+++ b/src/sp-conn-end.h
@@ -1,7 +1,7 @@
 #ifndef SEEN_SP_CONN_END
 #define SEEN_SP_CONN_END
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <stddef.h>
 #include <sigc++/connection.h>
 
diff --git a/src/sp-filter-fns.h b/src/sp-filter-fns.h
index 4e1b012..a1e2075 100644
--- a/src/sp-filter-fns.h
+++ b/src/sp-filter-fns.h
@@ -5,7 +5,7 @@
  * Macros and fn declarations related to filters.
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glib-object.h>
 #include "libnr/nr-forward.h"
 #include "sp-filter-units.h"
diff --git a/src/sp-gaussian-blur-fns.h b/src/sp-gaussian-blur-fns.h
index 0307392..fe696dc 100644
--- a/src/sp-gaussian-blur-fns.h
+++ b/src/sp-gaussian-blur-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/sp-gradient-fns.h b/src/sp-gradient-fns.h
index aabc3ed..e09717b 100644
--- a/src/sp-gradient-fns.h
+++ b/src/sp-gradient-fns.h
@@ -5,7 +5,7 @@
  * Macros and fn declarations related to gradients.
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <2geom/forward.h>
 #include "sp-gradient-spread.h"
diff --git a/src/sp-gradient-vector.h b/src/sp-gradient-vector.h
index 7bdfb51..d341daa 100644
--- a/src/sp-gradient-vector.h
+++ b/src/sp-gradient-vector.h
@@ -1,7 +1,7 @@
 #ifndef SEEN_SP_GRADIENT_VECTOR_H
 #define SEEN_SP_GRADIENT_VECTOR_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <vector>
 #include "color.h"
 
diff --git a/src/sp-image.cpp b/src/sp-image.cpp
index 322e2d1..0c7f9ba 100644
--- a/src/sp-image.cpp
+++ b/src/sp-image.cpp
@@ -30,6 +30,7 @@
 //#include <gdk-pixbuf/gdk-pixbuf-io.h>
 #include "display/nr-arena-image.h"
 #include <display/curve.h>
+#include <glib.h>
 #include <glib/gstdio.h>
 
 //Added for preserveAspectRatio support -- EAF
diff --git a/src/sp-linear-gradient-fns.h b/src/sp-linear-gradient-fns.h
index 0962bae..636cc1c 100644
--- a/src/sp-linear-gradient-fns.h
+++ b/src/sp-linear-gradient-fns.h
@@ -6,7 +6,7 @@
  */
 
 #include <glib-object.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace XML {
diff --git a/src/sp-metrics.h b/src/sp-metrics.h
index 23c1b6c..c2f9687 100644
--- a/src/sp-metrics.h
+++ b/src/sp-metrics.h
@@ -1,8 +1,7 @@
 #ifndef SP_METRICS_H
 #define SP_METRICS_H
 
-#include <glib/gstring.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "sp-metric.h"
 
 gdouble sp_absolute_metric_to_metric (gdouble length_src, const SPMetric metric_src, const SPMetric metric_dst);
diff --git a/src/sp-radial-gradient.h b/src/sp-radial-gradient.h
index bec6cbe..528b527 100644
--- a/src/sp-radial-gradient.h
+++ b/src/sp-radial-gradient.h
@@ -5,7 +5,7 @@
  * SPRadialGradient: SVG <radialgradient> implementtion.
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "sp-gradient.h"
 #include "svg/svg-length.h"
 #include "sp-radial-gradient-fns.h"
diff --git a/src/sp-stop.h b/src/sp-stop.h
index 2cf8ad6..96922eb 100644
--- a/src/sp-stop.h
+++ b/src/sp-stop.h
@@ -8,7 +8,7 @@
  * Authors?
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glibmm/ustring.h>
 #include "sp-object.h"
 #include "color.h"
diff --git a/src/sp-text.h b/src/sp-text.h
index b90fe37..5db61ff 100644
--- a/src/sp-text.h
+++ b/src/sp-text.h
@@ -13,7 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "sp-item.h"
diff --git a/src/sp-textpath.h b/src/sp-textpath.h
index 1931966..18b42ec 100644
--- a/src/sp-textpath.h
+++ b/src/sp-textpath.h
@@ -1,7 +1,7 @@
 #ifndef INKSCAPE_SP_TEXTPATH_H
 #define INKSCAPE_SP_TEXTPATH_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "svg/svg-length.h"
 #include "sp-item.h"
 #include "sp-text.h"
diff --git a/src/sp-tspan.h b/src/sp-tspan.h
index def8051..6c40d86 100644
--- a/src/sp-tspan.h
+++ b/src/sp-tspan.h
@@ -5,7 +5,7 @@
  * tspan and textpath, based on the flowtext routines
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "sp-item.h"
 #include "text-tag-attributes.h"
 
diff --git a/src/splivarot.cpp b/src/splivarot.cpp
index 128fa23..89360d8 100644
--- a/src/splivarot.cpp
+++ b/src/splivarot.cpp
@@ -20,7 +20,7 @@
 #include <cstring>
 #include <string>
 #include <vector>
-#include <glib/gmem.h>
+#include <glib.h>
 #include "xml/repr.h"
 #include "svg/svg.h"
 #include "sp-path.h"
diff --git a/src/spray-context.cpp b/src/spray-context.cpp
index 2bdac19..7d57168 100644
--- a/src/spray-context.cpp
+++ b/src/spray-context.cpp
@@ -29,7 +29,7 @@
 #include "svg/svg.h"
 #include "display/canvas-bpath.h"
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include "macros.h"
 #include "document.h"
 #include "selection.h"
diff --git a/src/svg/css-ostringstream.cpp b/src/svg/css-ostringstream.cpp
index a6eb278..865f6ae 100644
--- a/src/svg/css-ostringstream.cpp
+++ b/src/svg/css-ostringstream.cpp
@@ -1,8 +1,7 @@
 #include "svg/css-ostringstream.h"
 #include "svg/strip-trailing-zeros.h"
 #include "preferences.h"
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 Inkscape::CSSOStringStream::CSSOStringStream()
 {
diff --git a/src/svg/css-ostringstream.h b/src/svg/css-ostringstream.h
index 48ecf29..93bf32d 100644
--- a/src/svg/css-ostringstream.h
+++ b/src/svg/css-ostringstream.h
@@ -1,7 +1,7 @@
 #ifndef SVG_CSS_OSTRINGSTREAM_H_INKSCAPE
 #define SVG_CSS_OSTRINGSTREAM_H_INKSCAPE
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <sstream>
 
 namespace Inkscape {
diff --git a/src/svg/stringstream.h b/src/svg/stringstream.h
index 60ed74e..4623cbc 100644
--- a/src/svg/stringstream.h
+++ b/src/svg/stringstream.h
@@ -1,7 +1,7 @@
 #ifndef INKSCAPE_STRINGSTREAM_H
 #define INKSCAPE_STRINGSTREAM_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <sstream>
 #include <string>
 
diff --git a/src/svg/strip-trailing-zeros.cpp b/src/svg/strip-trailing-zeros.cpp
index 769834d..daad8b9 100644
--- a/src/svg/strip-trailing-zeros.cpp
+++ b/src/svg/strip-trailing-zeros.cpp
@@ -1,7 +1,7 @@
 
 #include <cstring>
 #include <string>
-#include <glib/gmessages.h>
+#include <glib.h>
 
 #include "svg/strip-trailing-zeros.h"
 
diff --git a/src/svg/svg-affine.cpp b/src/svg/svg-affine.cpp
index 91a9fa7..658991a 100644
--- a/src/svg/svg-affine.cpp
+++ b/src/svg/svg-affine.cpp
@@ -19,7 +19,7 @@
 #include <string>
 #include <cstdlib>
 #include <cstdio>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libnr/nr-matrix-fns.h>
 #include <libnr/nr-matrix-ops.h>
 #include <2geom/transforms.h>
diff --git a/src/svg/svg-color.cpp b/src/svg/svg-color.cpp
index 04f3877..4d727c3 100644
--- a/src/svg/svg-color.cpp
+++ b/src/svg/svg-color.cpp
@@ -23,12 +23,7 @@
 #include <string>
 #include <cassert>
 #include <math.h>
-#include <glib/gmem.h>
-#include <glib.h> // g_assert
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/ghash.h>
-#include <glib/gutils.h>
+#include <glib.h>
 #include <errno.h>
 
 #include "strneq.h"
diff --git a/src/svg/svg-color.h b/src/svg/svg-color.h
index a3868c1..d1c7bee 100644
--- a/src/svg/svg-color.h
+++ b/src/svg/svg-color.h
@@ -1,7 +1,7 @@
 #ifndef SVG_SVG_COLOR_H_SEEN
 #define SVG_SVG_COLOR_H_SEEN
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 class SVGICCColor;
 
diff --git a/src/svg/svg-length.cpp b/src/svg/svg-length.cpp
index 94f1cf3..ae5f758 100644
--- a/src/svg/svg-length.cpp
+++ b/src/svg/svg-length.cpp
@@ -19,7 +19,7 @@
 #include <cstring>
 #include <string>
 #include <math.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 #include "svg.h"
 #include "stringstream.h"
diff --git a/src/svg/svg-length.h b/src/svg/svg-length.h
index 1f4be81..f5631b1 100644
--- a/src/svg/svg-length.h
+++ b/src/svg/svg-length.h
@@ -16,7 +16,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 class SVGLength
 {
diff --git a/src/svg/svg-path-geom-test.h b/src/svg/svg-path-geom-test.h
index 1a084d5..a1bd4d7 100644
--- a/src/svg/svg-path-geom-test.h
+++ b/src/svg/svg-path-geom-test.h
@@ -8,7 +8,7 @@
 #include <stdio.h>
 #include <string>
 #include <vector>
-#include <glib/gmem.h>
+#include <glib.h>
 
 class SvgPathGeomTest : public CxxTest::TestSuite
 {
diff --git a/src/svg/svg-path.cpp b/src/svg/svg-path.cpp
index bbf1589..b58ec40 100644
--- a/src/svg/svg-path.cpp
+++ b/src/svg/svg-path.cpp
@@ -31,10 +31,7 @@
 #include <cstring>
 #include <string>
 #include <cassert>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib.h> // g_assert()
+#include <glib.h>
 
 #include "svg/svg.h"
 #include "svg/path-string.h"
diff --git a/src/svg/svg.h b/src/svg/svg.h
index 0b2c3ae..7ff8ecc 100644
--- a/src/svg/svg.h
+++ b/src/svg/svg.h
@@ -11,7 +11,7 @@
  *
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <vector>
 #include <cstring>
 #include <string>
diff --git a/src/text-editing.h b/src/text-editing.h
index 038458f..9777c67 100644
--- a/src/text-editing.h
+++ b/src/text-editing.h
@@ -13,7 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <utility>   // std::pair
 #include "libnrtype/Layout-TNG.h"
 #include <libnr/nr-forward.h>
diff --git a/src/text-tag-attributes.h b/src/text-tag-attributes.h
index 197bfb7..1c73869 100644
--- a/src/text-tag-attributes.h
+++ b/src/text-tag-attributes.h
@@ -2,7 +2,7 @@
 #define INKSCAPE_TEXT_TAG_ATTRIBUTES_H
 
 #include <vector>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "libnrtype/Layout-TNG.h"
 #include "svg/svg-length.h"
 
diff --git a/src/trace/potrace/potracelib.cpp b/src/trace/potrace/potracelib.cpp
index 136f7a9..9505aa0 100644
--- a/src/trace/potrace/potracelib.cpp
+++ b/src/trace/potrace/potracelib.cpp
@@ -4,7 +4,7 @@
 
 #include <stdlib.h>
 #include <string.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 #include "potracelib.h"
 #include "inkscape-version.h"
diff --git a/src/tweak-context.cpp b/src/tweak-context.cpp
index 904d0cb..ce6a515 100644
--- a/src/tweak-context.cpp
+++ b/src/tweak-context.cpp
@@ -22,7 +22,7 @@
 #include "svg/svg.h"
 #include "display/canvas-bpath.h"
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include "macros.h"
 #include "document.h"
 #include "selection.h"
diff --git a/src/ui/cache/svg_preview_cache.cpp b/src/ui/cache/svg_preview_cache.cpp
index 6e05f6c..c13d41f 100644
--- a/src/ui/cache/svg_preview_cache.cpp
+++ b/src/ui/cache/svg_preview_cache.cpp
@@ -19,7 +19,7 @@
 # include "config.h"
 #endif
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 #include "sp-namedview.h"
 #include "selection.h"
diff --git a/src/ui/dialog/desktop-tracker.h b/src/ui/dialog/desktop-tracker.h
index b5b83b5..dc492d2 100644
--- a/src/ui/dialog/desktop-tracker.h
+++ b/src/ui/dialog/desktop-tracker.h
@@ -13,7 +13,7 @@
 
 #include <stddef.h>
 #include <sigc++/connection.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 typedef struct _GtkWidget GtkWidget;
 class SPDesktop;
diff --git a/src/ui/dialog/dialog-manager.h b/src/ui/dialog/dialog-manager.h
index a97b58c..f088b16 100644
--- a/src/ui/dialog/dialog-manager.h
+++ b/src/ui/dialog/dialog-manager.h
@@ -14,7 +14,7 @@
 #ifndef INKSCAPE_UI_DIALOG_MANAGER_H
 #define INKSCAPE_UI_DIALOG_MANAGER_H
 
-#include <glib/gquark.h>
+#include <glib.h>
 #include "dialog.h"
 #include <map>
 
diff --git a/src/ui/dialog/icon-preview.cpp b/src/ui/dialog/icon-preview.cpp
index cbd2769..306e990 100644
--- a/src/ui/dialog/icon-preview.cpp
+++ b/src/ui/dialog/icon-preview.cpp
@@ -16,7 +16,7 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <glib/gmem.h>
+#include <glib.h>
 #include <glibmm/i18n.h>
 #include <gtkmm/alignment.h>
 #include <gtkmm/buttonbox.h>
diff --git a/src/ui/dialog/input.cpp b/src/ui/dialog/input.cpp
index 8c98515..1cd8eac 100644
--- a/src/ui/dialog/input.cpp
+++ b/src/ui/dialog/input.cpp
@@ -11,7 +11,7 @@
 #include <map>
 #include <set>
 #include <list>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <glibmm/i18n.h>
 #include <gtkmm/alignment.h>
 #include <gtkmm/cellrenderercombo.h>
diff --git a/src/ui/widget/icon-widget.cpp b/src/ui/widget/icon-widget.cpp
index 64415f4..210b3dc 100644
--- a/src/ui/widget/icon-widget.cpp
+++ b/src/ui/widget/icon-widget.cpp
@@ -14,7 +14,7 @@
 # include <config.h>
 #endif
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include "icon-widget.h"
 
 namespace Inkscape {
diff --git a/src/ui/widget/spin-slider.cpp b/src/ui/widget/spin-slider.cpp
index e3e73a5..69bfb6d 100644
--- a/src/ui/widget/spin-slider.cpp
+++ b/src/ui/widget/spin-slider.cpp
@@ -10,7 +10,7 @@
  * Released under GNU GPL.  Read the file 'COPYING' for more information.
  */
 
-#include "glib/gstrfuncs.h"
+#include <glib.h>
 #include "glibmm/i18n.h"
 
 #include "spin-slider.h"
diff --git a/src/unclump.h b/src/unclump.h
index c5a8bf7..82dd272 100644
--- a/src/unclump.h
+++ b/src/unclump.h
@@ -11,7 +11,7 @@
 #ifndef SEEN_DIALOGS_UNCLUMP_H
 #define SEEN_DIALOGS_UNCLUMP_H
 
-#include <glib/gslist.h>
+#include <glib.h>
 
 void unclump(GSList *items);
 
diff --git a/src/uri.h b/src/uri.h
index 1c890a6..38a4707 100644
--- a/src/uri.h
+++ b/src/uri.h
@@ -14,7 +14,7 @@
 #ifndef INKSCAPE_URI_H
 #define INKSCAPE_URI_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <exception>
 #include <libxml/uri.h>
 #include "bad-uri-exception.h"
diff --git a/src/util/glib-list-iterators.h b/src/util/glib-list-iterators.h
index 586bc31..9e73fa6 100644
--- a/src/util/glib-list-iterators.h
+++ b/src/util/glib-list-iterators.h
@@ -17,8 +17,7 @@
 
 #include <cstddef>
 #include <iterator>
-#include "glib/gslist.h"
-#include "glib/glist.h"
+#include <glib.h>
 
 namespace Inkscape {
 
diff --git a/src/util/share.cpp b/src/util/share.cpp
index 2f693fa..f74de23 100644
--- a/src/util/share.cpp
+++ b/src/util/share.cpp
@@ -10,7 +10,7 @@
  */
 
 #include "util/share.h"
-#include <glib/gmessages.h>
+#include <glib.h>
 
 namespace Inkscape {
 namespace Util {
diff --git a/src/version.cpp b/src/version.cpp
index edaa600..ef7fbe2 100644
--- a/src/version.cpp
+++ b/src/version.cpp
@@ -12,7 +12,7 @@
  */
 
 #include <stdio.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include "version.h"
 
 gboolean sp_version_from_string(const gchar *string, Inkscape::Version *version)
diff --git a/src/version.h b/src/version.h
index c620631..0a0381c 100644
--- a/src/version.h
+++ b/src/version.h
@@ -10,7 +10,7 @@
 #ifndef SEEN_INKSCAPE_VERSION_H
 #define SEEN_INKSCAPE_VERSION_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #define SVG_VERSION "1.1"
 
diff --git a/src/widgets/icon.cpp b/src/widgets/icon.cpp
index 851abe9..2a814fe 100644
--- a/src/widgets/icon.cpp
+++ b/src/widgets/icon.cpp
@@ -16,7 +16,7 @@
 #endif
 
 #include <cstring>
-#include <glib/gmem.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 #include <gtkmm.h>
 
diff --git a/src/widgets/sp-color-icc-selector.h b/src/widgets/sp-color-icc-selector.h
index f76b258..f6e7cca 100644
--- a/src/widgets/sp-color-icc-selector.h
+++ b/src/widgets/sp-color-icc-selector.h
@@ -1,7 +1,7 @@
 #ifndef SEEN_SP_COLOR_ICC_SELECTOR_H
 #define SEEN_SP_COLOR_ICC_SELECTOR_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 
 #include "../color.h"
diff --git a/src/widgets/sp-color-scales.h b/src/widgets/sp-color-scales.h
index b50c386..38219b5 100644
--- a/src/widgets/sp-color-scales.h
+++ b/src/widgets/sp-color-scales.h
@@ -1,7 +1,7 @@
 #ifndef SEEN_SP_COLOR_SCALES_H
 #define SEEN_SP_COLOR_SCALES_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 
 #include <color.h>
diff --git a/src/widgets/sp-color-wheel-selector.h b/src/widgets/sp-color-wheel-selector.h
index 56bc8af..4124b1f 100644
--- a/src/widgets/sp-color-wheel-selector.h
+++ b/src/widgets/sp-color-wheel-selector.h
@@ -1,7 +1,7 @@
 #ifndef SEEN_SP_COLOR_WHEEL_SELECTOR_H
 #define SEEN_SP_COLOR_WHEEL_SELECTOR_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 
 #include "../color.h"
diff --git a/src/widgets/spinbutton-events.h b/src/widgets/spinbutton-events.h
index a9ff741..5092861 100644
--- a/src/widgets/spinbutton-events.h
+++ b/src/widgets/spinbutton-events.h
@@ -9,7 +9,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gtk/gtk.h>      /* GtkWidget */
 
 gboolean spinbutton_focus_in (GtkWidget *w, GdkEventKey *event, gpointer data);
diff --git a/src/widgets/spw-utilities.h b/src/widgets/spw-utilities.h
index 78f7937..f0ec2f4 100644
--- a/src/widgets/spw-utilities.h
+++ b/src/widgets/spw-utilities.h
@@ -18,7 +18,7 @@
    SPObject, that reacts to modification.
 */
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 typedef struct _GtkWidget GtkWidget;
 
diff --git a/src/widgets/stroke-style.cpp b/src/widgets/stroke-style.cpp
index f020b0c..907952f 100644
--- a/src/widgets/stroke-style.cpp
+++ b/src/widgets/stroke-style.cpp
@@ -18,7 +18,7 @@
 
 #define noSP_SS_VERBOSE
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 #include <glibmm/i18n.h>
 
diff --git a/src/xml/attribute-record.h b/src/xml/attribute-record.h
index bab0b5a..a61329b 100644
--- a/src/xml/attribute-record.h
+++ b/src/xml/attribute-record.h
@@ -5,8 +5,7 @@
 #ifndef SEEN_XML_SP_REPR_ATTR_H
 #define SEEN_XML_SP_REPR_ATTR_H
 
-#include <glib/gquark.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "gc-managed.h"
 #include "util/share.h"
 
diff --git a/src/xml/comment-node.h b/src/xml/comment-node.h
index 698a30a..ec94691 100644
--- a/src/xml/comment-node.h
+++ b/src/xml/comment-node.h
@@ -15,7 +15,7 @@
 #ifndef SEEN_INKSCAPE_XML_COMMENT_NODE_H
 #define SEEN_INKSCAPE_XML_COMMENT_NODE_H
 
-#include <glib/gquark.h>
+#include <glib.h>
 #include "xml/simple-node.h"
 
 namespace Inkscape {
diff --git a/src/xml/croco-node-iface.cpp b/src/xml/croco-node-iface.cpp
index db98e3d..e67eeac 100644
--- a/src/xml/croco-node-iface.cpp
+++ b/src/xml/croco-node-iface.cpp
@@ -1,7 +1,7 @@
 
 #include <cstring>
 #include <string>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 #include "xml/croco-node-iface.h"
 #include "xml/node.h"
diff --git a/src/xml/event.h b/src/xml/event.h
index c622577..acd2bfc 100644
--- a/src/xml/event.h
+++ b/src/xml/event.h
@@ -18,8 +18,7 @@
 #ifndef SEEN_INKSCAPE_XML_SP_REPR_ACTION_H
 #define SEEN_INKSCAPE_XML_SP_REPR_ACTION_H
 
-#include <glib/gtypes.h>
-#include <glib/gquark.h>
+#include <glib.h>
 #include <glibmm/ustring.h>
 
 #include <iterator>
diff --git a/src/xml/node-event-vector.h b/src/xml/node-event-vector.h
index 0c291c2..e639687 100644
--- a/src/xml/node-event-vector.h
+++ b/src/xml/node-event-vector.h
@@ -14,7 +14,7 @@
 #ifndef SEEN_INKSCAPE_XML_SP_REPR_EVENT_VECTOR
 #define SEEN_INKSCAPE_XML_SP_REPR_EVENT_VECTOR
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include "xml/node.h"
 
diff --git a/src/xml/node-observer.h b/src/xml/node-observer.h
index ca787c2..44f11e5 100644
--- a/src/xml/node-observer.h
+++ b/src/xml/node-observer.h
@@ -18,7 +18,7 @@
 #ifndef SEEN_INKSCAPE_XML_NODE_OBSERVER_H
 #define SEEN_INKSCAPE_XML_NODE_OBSERVER_H
 
-#include <glib/gquark.h>
+#include <glib.h>
 #include "util/share.h"
 #include "xml/xml-forward.h"
 
diff --git a/src/xml/node.h b/src/xml/node.h
index abcccdb..ad3f3de 100644
--- a/src/xml/node.h
+++ b/src/xml/node.h
@@ -18,7 +18,7 @@
 #ifndef SEEN_INKSCAPE_XML_NODE_H
 #define SEEN_INKSCAPE_XML_NODE_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "gc-anchored.h"
 #include "util/list.h"
 #include "xml/xml-forward.h"
diff --git a/src/xml/pi-node.h b/src/xml/pi-node.h
index 4aa1f84..548216a 100644
--- a/src/xml/pi-node.h
+++ b/src/xml/pi-node.h
@@ -14,7 +14,7 @@
 #ifndef SEEN_INKSCAPE_XML_PI_NODE_H
 #define SEEN_INKSCAPE_XML_PI_NODE_H
 
-#include <glib/gquark.h>
+#include <glib.h>
 #include "xml/simple-node.h"
 
 namespace Inkscape {
diff --git a/src/xml/quote.cpp b/src/xml/quote.cpp
index 37f00ba..166cf26 100644
--- a/src/xml/quote.cpp
+++ b/src/xml/quote.cpp
@@ -12,7 +12,7 @@
  */
 
 #include <cstring>
-#include <glib/gmem.h>
+#include <glib.h>
 
 
 /** \return strlen(xml_quote_strdup(\a val)) (without doing the malloc).
diff --git a/src/xml/rebase-hrefs.cpp b/src/xml/rebase-hrefs.cpp
index c387672..c27e9dd 100644
--- a/src/xml/rebase-hrefs.cpp
+++ b/src/xml/rebase-hrefs.cpp
@@ -7,9 +7,7 @@
 #include "util/share.h"
 #include "xml/attribute-record.h"
 #include "xml/node.h"
-#include <glib/gmem.h>
-#include <glib/gurifuncs.h>
-#include <glib/gutils.h>
+#include <glib.h>
 using Inkscape::XML::AttributeRecord;
 
 
diff --git a/src/xml/rebase-hrefs.h b/src/xml/rebase-hrefs.h
index b4f288c..ae29a75 100644
--- a/src/xml/rebase-hrefs.h
+++ b/src/xml/rebase-hrefs.h
@@ -1,7 +1,7 @@
 #ifndef REBASE_HREFS_H_SEEN
 #define REBASE_HREFS_H_SEEN
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "util/list.h"
 #include "xml/attribute-record.h"
 struct SPDocument;
diff --git a/src/xml/repr.h b/src/xml/repr.h
index 549822e..b6482e6 100644
--- a/src/xml/repr.h
+++ b/src/xml/repr.h
@@ -14,7 +14,7 @@
 #define __SP_REPR_H__
 
 #include <stdio.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "gc-anchored.h"
 
 #include "xml/node.h"
diff --git a/src/xml/simple-node.cpp b/src/xml/simple-node.cpp
index 7ecdc1b..0c781e2 100644
--- a/src/xml/simple-node.cpp
+++ b/src/xml/simple-node.cpp
@@ -16,7 +16,7 @@
 
 #include <cstring>
 #include <string>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 #include "xml/node.h"
 #include "xml/simple-node.h"
diff --git a/src/xml/text-node.h b/src/xml/text-node.h
index ef6e477..5948bc1 100644
--- a/src/xml/text-node.h
+++ b/src/xml/text-node.h
@@ -15,7 +15,7 @@
 #ifndef SEEN_INKSCAPE_XML_TEXT_NODE_H
 #define SEEN_INKSCAPE_XML_TEXT_NODE_H
 
-#include <glib/gquark.h>
+#include <glib.h>
 #include "xml/simple-node.h"
 
 namespace Inkscape {
