--- emacs-24.3-orig/configure.ac	2014-09-09 14:14:26.305364162 +0000
+++ emacs-24.3/configure.ac	2014-09-09 14:50:51.748865921 +0000
@@ -1088,7 +1088,7 @@
     CRT_DIR=/usr/lib            # default
 
     case "$canonical" in
-      x86_64-*-linux-gnu* | s390x-*-linux-gnu*)
+      x86_64-*-linux-gnu* | s390x-*-linux-gnu* | mips64*-*-linux-* )
       ## On x86-64 and s390x GNU/Linux distributions, the standard library
       ## can be in a variety of places.  We only try /usr/lib64 and /usr/lib.
       ## For anything else (eg /usr/lib32), it is up the user to specify
