--- ./findlib-1.4/src/findlib/Makefile.old	2013-09-13 22:13:25.322978268 +0100
+++ ./findlib-1.4/src/findlib/Makefile	2013-09-13 22:13:38.289999908 +0100
@@ -14,7 +14,7 @@
 
 # Need compiler-libs since ocaml-4.00
 OCAMLC = ocamlc -I +compiler-libs
-OCAMLOPT = ocamlopt -I +compiler-libs
+OCAMLOPT = ocamlopt -I +compiler-libs -g
 OCAMLDEP = ocamldep
 OCAMLLEX = ocamllex
 #CAMLP4O =  camlp4 pa_o.cmo pa_op.cmo pr_o.cmo --
