--- kdenetwork-3.4.3/kfile-plugins/torrent/kfile_torrent.cpp.orig	2005-10-05 21:23:58.000000000 +0800
+++ kdenetwork-3.4.3/kfile-plugins/torrent/kfile_torrent.cpp	2005-10-18 23:13:21.000000000 +0800
@@ -31,6 +31,8 @@
 #include "bint.h"
 #include "bstring.h"
 
+#include <qtextcodec.h>
+
 typedef KGenericFactory<KTorrentPlugin> TorrentFactory;
 K_EXPORT_COMPONENT_FACTORY(kfile_torrent, TorrentFactory("kfile_torrent"))
 
@@ -165,7 +167,15 @@
     // The remainder of this function will consist of a lot of redundancy checks.
     // If a torrent has a key, but it is of the wrong type, then it isn't a valid
     // torrent, and so we should just die.
-    
+    //Added by KanKer <kanker@163.com>
+    BString *enc = m_dict->findStr ("encoding");
+    QString encode;
+    if (!enc)
+	encode = QTextCodec::codecForLocale()->name();
+    else
+        encode = enc->get_string();
+    QTextCodec *codec=QTextCodec::codecForName(encode.latin1());
+
     if (m_dict->contains("announce"))
     {
         BString *str = m_dict->findStr ("announce");
@@ -234,7 +244,7 @@
         if (num_files > 1 && !real_str.endsWith("/"))
             real_str.append('/');
 
-        appendItem (group, "name", real_str);
+        appendItem (group, "name", codec->toUnicode(real_str.latin1()));
     }
 
     // piece length is required as well
@@ -250,7 +260,7 @@
         if (!comment)
             return false;
             
-        appendItem (group, "comment", comment->get_string());
+        appendItem (group, "comment", codec->toUnicode(comment->get_string().latin1()));
     }
     else
         appendItem (group, "comment", QString());
