diff -Nur kopete-0.12b1/kopete/protocols/irc/ircprotocol.cpp kopete-0.12b1f/kopete/protocols/irc/ircprotocol.cpp
--- kopete-0.12b1/kopete/protocols/irc/ircprotocol.cpp	2006-03-08 09:25:24.000000000 +0800
+++ kopete-0.12b1f/kopete/protocols/irc/ircprotocol.cpp	2006-03-14 20:03:46.000000000 +0800
@@ -136,15 +136,15 @@
 			QStringList::split(' ',"irc_voice contact_away_overlay"),  i18n("Away")),
 
 	m_UserStatusOnline(Kopete::OnlineStatus::Online,
-			25, this, Online, TQString(), i18n("Online"), i18n("Online"), Kopete::OnlineStatusManager::Online),
+			25, this, Online, TQString(), ("Online"), i18n("Online"), Kopete::OnlineStatusManager::Online),
 
 	m_UserStatusAway(Kopete::OnlineStatus::Away,
 			2, this, Online | Away, "contact_away_overlay",
-			i18n("Away"), i18n("Away"), Kopete::OnlineStatusManager::Away),
+			("Away"), i18n("Away"), Kopete::OnlineStatusManager::Away),
 	m_UserStatusConnecting(Kopete::OnlineStatus::Connecting,
 			1, this, Connecting, "irc_connecting", i18n("Connecting")),
 	m_UserStatusOffline(Kopete::OnlineStatus::Offline,
-			0, this, Offline, TQString(), i18n("Offline"), i18n("Offline"), Kopete::OnlineStatusManager::Offline),
+			0, this, Offline, TQString(), ("Offline"), i18n("Offline"), Kopete::OnlineStatusManager::Offline),
 
 	m_StatusUnknown(Kopete::OnlineStatus::Unknown,
 			999, this, 999, "status_unknown", i18n("Status not available")),
diff -Nur kopete-0.12b1/kopete/protocols/jabber/jabberprotocol.cpp kopete-0.12b1f/kopete/protocols/jabber/jabberprotocol.cpp
--- kopete-0.12b1/kopete/protocols/jabber/jabberprotocol.cpp	2006-03-08 09:25:21.000000000 +0800
+++ kopete-0.12b1f/kopete/protocols/jabber/jabberprotocol.cpp	2006-03-14 20:03:51.000000000 +0800
@@ -71,14 +71,15 @@
 
 JabberProtocol::JabberProtocol (QObject * parent, const char *name, const TQStringList &)
 : Kopete::Protocol( JabberProtocolFactory::instance(), parent, name ),
-	JabberKOSChatty(Kopete::OnlineStatus::Online,        100, this, JabberFreeForChat, "jabber_chatty", i18n ("Free for Chat"), i18n ("Free for Chat"), Kopete::OnlineStatusManager::FreeForChat, Kopete::OnlineStatusManager::HasAwayMessage ),
-	JabberKOSOnline(Kopete::OnlineStatus::Online,         90, this, JabberOnline, TQString(), i18n ("Online"), i18n ("Online"), Kopete::OnlineStatusManager::Online, Kopete::OnlineStatusManager::HasAwayMessage ),
-	JabberKOSAway(Kopete::OnlineStatus::Away,             80, this, JabberAway, "contact_away_overlay", i18n ("Away"), i18n ("Away"), Kopete::OnlineStatusManager::Away, Kopete::OnlineStatusManager::HasAwayMessage),
-	JabberKOSXA(Kopete::OnlineStatus::Away,               70, this, JabberXA, "contact_xa_overlay", i18n ("Extended Away"), i18n ("Extended Away"), 0, Kopete::OnlineStatusManager::HasAwayMessage),
-	JabberKOSDND(Kopete::OnlineStatus::Away,              60, this, JabberDND, "contact_busy_overlay", i18n ("Do not Disturb"), i18n ("Do not Disturb"), Kopete::OnlineStatusManager::Busy, Kopete::OnlineStatusManager::HasAwayMessage),
-	JabberKOSOffline(Kopete::OnlineStatus::Offline,       50, this, JabberOffline, TQString(), i18n ("Offline") ,i18n ("Offline"), Kopete::OnlineStatusManager::Offline, Kopete::OnlineStatusManager::HasAwayMessage ),
-	JabberKOSInvisible(Kopete::OnlineStatus::Invisible,   40, this, JabberInvisible, "contact_invisible_overlay",   i18n ("Invisible") ,i18n ("Invisible"), Kopete::OnlineStatusManager::Invisible),
-	JabberKOSConnecting(Kopete::OnlineStatus::Connecting, 30, this, JabberConnecting, "jabber_connecting",  i18n("Connecting")),
+	 JabberKOSChatty(Kopete::OnlineStatus::Online,        100, this, 1, "jabber_chatty",      ("Free to Chat"), i18n ("Free to Chat"), Kopete::OnlineStatusManager::Online),
+    JabberKOSOnline(Kopete::OnlineStatus::Online,         90, this, 0, TQString(),        ("Online"), i18n ("Online"), Kopete::OnlineStatusManager::Online),
+    JabberKOSAway(Kopete::OnlineStatus::Away,             80, this, 2, "jabber_away",        ("Away"), i18n ("Away"), Kopete::OnlineStatusManager::Busy,Kopete::OnlineStatusManager::HasAwayMessage),
+    JabberKOSXA(Kopete::OnlineStatus::Away,               70, this, 3, "jabber_xa",          ("Extended Away"), i18n ("Extended Away"), Kopete::OnlineStatusManager::Away),
+    JabberKOSDND(Kopete::OnlineStatus::Away,              60, this, 4, "jabber_na",          ("Do not Disturb"), i18n ("Do not Disturb"), Kopete::OnlineStatusManager::Away),
+    JabberKOSOffline(Kopete::OnlineStatus::Offline,       50, this, 5, TQString(),        ("Offline") ,i18n ("Offline"), Kopete::OnlineStatusManager::Offline),
+    JabberKOSInvisible(Kopete::OnlineStatus::Invisible,   40, this, 6, "jabber_invisible",   ("Invisible") ,i18n ("Invisible"), Kopete::OnlineStatusManager::Invisible),
+    JabberKOSConnecting(Kopete::OnlineStatus::Connecting, 30, this, 7, "jabber_connecting",  i18n("Connecting")),
+
 	propLastSeen(Kopete::Global::Properties::self()->lastSeen()),
 	propAwayMessage(Kopete::Global::Properties::self()->awayMessage()),
 	propFirstName(Kopete::Global::Properties::self()->firstName()),
diff -Nur kopete-0.12b1/kopete/protocols/Makefile.am kopete-0.12b1f/kopete/protocols/Makefile.am
--- kopete-0.12b1/kopete/protocols/Makefile.am	2006-03-08 09:25:24.000000000 +0800
+++ kopete-0.12b1f/kopete/protocols/Makefile.am	2006-03-14 20:03:46.000000000 +0800
@@ -1,21 +1 @@
-if include_meanwhile
-MEANWHILE=meanwhile
-endif
-
-if include_gadu
-GADU=gadu
-endif
-
-if include_jabber
-JABBER=jabber
-endif
-
-if include_testbed
-TESTBED=testbed
-endif
-
-if include_smsgsm
-SMS=sms
-endif
-
-SUBDIRS = $(TESTBED) groupwise msn irc oscar yahoo winpopup $(SMS) $(JABBER) $(GADU) $(MEANWHILE)
+SUBDIRS = jabber msn irc oscar yahoo
diff -Nur kopete-0.12b1/kopete/protocols/msn/msnprotocol.cpp kopete-0.12b1f/kopete/protocols/msn/msnprotocol.cpp
--- kopete-0.12b1/kopete/protocols/msn/msnprotocol.cpp	2006-03-08 09:25:24.000000000 +0800
+++ kopete-0.12b1f/kopete/protocols/msn/msnprotocol.cpp	2006-03-14 20:03:51.000000000 +0800
@@ -49,17 +49,17 @@
 
 MSNProtocol::MSNProtocol( TQObject *parent, const char *name, const TQStringList & /* args */ )
 : Kopete::Protocol( MSNProtocolFactory::instance(), parent, name ),
-	NLN( Kopete::OnlineStatus::Online,    25, this, 1, TQString(),               i18n( "Online" ) ,           i18n( "O&nline" ),        Kopete::OnlineStatusManager::Online,Kopete::OnlineStatusManager::HasAwayMessage ),
-	BSY( Kopete::OnlineStatus::Away,      20, this, 2, "msn_busy",                  i18n( "Busy" ) ,             i18n( "&Busy" ),          Kopete::OnlineStatusManager::Busy,  Kopete::OnlineStatusManager::HasAwayMessage ),
-	BRB( Kopete::OnlineStatus::Away,      22, this, 3, "msn_brb",                   i18n( "Be Right Back" ),     i18n( "Be &Right Back" ) ,  0  ,                              Kopete::OnlineStatusManager::HasAwayMessage ),
-	AWY( Kopete::OnlineStatus::Away,      18, this, 4, "contact_away_overlay",      i18n( "Away From Computer" ),i18n( "&Away" ),          Kopete::OnlineStatusManager::Away,  Kopete::OnlineStatusManager::HasAwayMessage ),
-	PHN( Kopete::OnlineStatus::Away,      12, this, 5, "contact_phone_overlay",     i18n( "On the Phone" ) ,     i18n( "On The &Phone" )  ,  0  ,                              Kopete::OnlineStatusManager::HasAwayMessage ),
-	LUN( Kopete::OnlineStatus::Away,      15, this, 6, "contact_food_overlay",      i18n( "Out to Lunch" ) ,     i18n( "Out To &Lunch" )  ,  0  ,                              Kopete::OnlineStatusManager::HasAwayMessage ),
-	FLN( Kopete::OnlineStatus::Offline,    0, this, 7, TQString(),               i18n( "Offline" ) ,          i18n( "&Offline" ),       Kopete::OnlineStatusManager::Offline,Kopete::OnlineStatusManager::DisabledIfOffline ),
-	HDN( Kopete::OnlineStatus::Invisible,  3, this, 8, "contact_invisible_overlay", i18n( "Invisible" ) ,        i18n( "&Invisible" ),     Kopete::OnlineStatusManager::Invisible ), 
-	IDL( Kopete::OnlineStatus::Away,      10, this, 9, "contact_away_overlay",      i18n( "Idle" ) ,             i18n( "&Idle" ),          Kopete::OnlineStatusManager::Idle , Kopete::OnlineStatusManager::HideFromMenu ),
-	UNK( Kopete::OnlineStatus::Unknown,   25, this, 0, "status_unknown",            i18n( "Status not available" ) ),
-	CNT( Kopete::OnlineStatus::Connecting, 2, this, 10,"msn_connecting",            i18n( "Connecting" ) ),
+	 NLN( Kopete::OnlineStatus::Online,    25, this, 1, TQString(),   ( "Online" ) , i18n( "O&nline" ), Kopete::OnlineStatusManager::Online ),
+    BSY( Kopete::OnlineStatus::Away,      20, this, 2, "msn_busy",      ( "Busy" ) , i18n( "&Busy" ), Kopete::OnlineStatusManager::Busy),
+    BRB( Kopete::OnlineStatus::Away,      22, this, 3, "msn_brb",       ( "Be Right Back" ), i18n( "Be &Right Back" )),
+    AWY( Kopete::OnlineStatus::Away,      18, this, 4, "msn_away",      ( "Away From Computer" ) , i18n( "&Away" ), Kopete::OnlineStatusManager::Away, Kopete::OnlineStatusManager::HasAwayMessage ),
+    PHN( Kopete::OnlineStatus::Away,      12, this, 5, "msn_phone",     ( "On the Phone" ) , i18n( "On The &Phone" )),
+    LUN( Kopete::OnlineStatus::Away,      15, this, 6, "msn_lunch",     ( "Out to Lunch" ) , i18n( "Out To &Lunch" ) ),
+    FLN( Kopete::OnlineStatus::Offline,    0, this, 7, TQString(),   ( "Offline" ) , i18n( "&Offline" ), Kopete::OnlineStatusManager::Offline , Kopete::OnlineStatusManager::DisabledIfOffline),
+    HDN( Kopete::OnlineStatus::Invisible,  3, this, 8, "msn_invisible", ( "Invisible" ) , i18n( "&Invisible" ), Kopete::OnlineStatusManager::Invisible ),
+    IDL( Kopete::OnlineStatus::Away,      10, this, 9, "msn_away",      ( "Idle" ) ,  i18n( "&Idle" ), Kopete::OnlineStatusManager::Invisible ),
+    UNK( Kopete::OnlineStatus::Unknown,   25, this, 0, "status_unknown",( "Status not available" ) ),
+    CNT( Kopete::OnlineStatus::Connecting, 2, this, 10,"msn_connecting",( "Connecting" ) ),
 	propEmail(Kopete::Global::Properties::self()->emailAddress()),
 	propPhoneHome(Kopete::Global::Properties::self()->privatePhone()),
 	propPhoneWork(Kopete::Global::Properties::self()->workPhone()),
diff -Nur kopete-0.12b1/kopete/protocols/oscar/aim/aimprotocol.cpp kopete-0.12b1f/kopete/protocols/oscar/aim/aimprotocol.cpp
--- kopete-0.12b1/kopete/protocols/oscar/aim/aimprotocol.cpp	2006-03-08 09:25:23.000000000 +0800
+++ kopete-0.12b1f/kopete/protocols/oscar/aim/aimprotocol.cpp	2006-03-14 20:03:46.000000000 +0800
@@ -220,14 +220,14 @@
 
 AIMProtocol::AIMProtocol(QObject *parent, const char *name, const TQStringList &)
   : Kopete::Protocol( AIMProtocolFactory::instance(), parent, name ),
-	statusOnline( Kopete::OnlineStatus::Online, 2, this, 0, TQString(), i18n("Online"), i18n("Online"), Kopete::OnlineStatusManager::Online ),
-	statusOffline( Kopete::OnlineStatus::Offline, 2, this, 10, TQString(), i18n("Offline"), i18n("Offline"), Kopete::OnlineStatusManager::Offline ),
-	statusAway( Kopete::OnlineStatus::Away, 2, this, 20, "contact_away_overlay", i18n("Away"), i18n("Away"), Kopete::OnlineStatusManager::Away,
+	statusOnline( Kopete::OnlineStatus::Online, 2, this, 0, TQString(), ("Online"), i18n("Online"), Kopete::OnlineStatusManager::Online ),
+	statusOffline( Kopete::OnlineStatus::Offline, 2, this, 10, TQString(), ("Offline"), i18n("Offline"), Kopete::OnlineStatusManager::Offline ),
+	statusAway( Kopete::OnlineStatus::Away, 2, this, 20, "contact_away_overlay", ("Away"), i18n("Away"), Kopete::OnlineStatusManager::Away,
 							Kopete::OnlineStatusManager::HasAwayMessage ),
-	statusWirelessOnline( Kopete::OnlineStatus::Online, 1, this, 30, "contact_phone_overlay", i18n("Mobile"), i18n("Mobile"),
+	statusWirelessOnline( Kopete::OnlineStatus::Online, 1, this, 30, "contact_phone_overlay", ("Mobile"), i18n("Mobile"),
 	Kopete::OnlineStatusManager::Online, Kopete::OnlineStatusManager::HideFromMenu ),
 	statusWirelessAway( Kopete::OnlineStatus::Away, 1, this, 31, TQStringList::split( " ", "contact_phone_overlay contact_away_overlay"),
-	i18n("Mobile Away"), i18n("Mobile Away"), Kopete::OnlineStatusManager::Away, Kopete::OnlineStatusManager::HideFromMenu ),
+	("Mobile Away"), i18n("Mobile Away"), Kopete::OnlineStatusManager::Away, Kopete::OnlineStatusManager::HideFromMenu ),
 	statusConnecting(Kopete::OnlineStatus::Connecting, 99, this, 99, "aim_connecting", i18n("Connecting...")),
 	awayMessage(Kopete::Global::Properties::self()->awayMessage()),
 	clientFeatures("clientFeatures", i18n("Client Features"), 0, false),
diff -Nur kopete-0.12b1/kopete/protocols/oscar/icq/icqpresence.cpp kopete-0.12b1f/kopete/protocols/oscar/icq/icqpresence.cpp
--- kopete-0.12b1/kopete/protocols/oscar/icq/icqpresence.cpp	2006-03-08 09:25:23.000000000 +0800
+++ kopete-0.12b1f/kopete/protocols/oscar/icq/icqpresence.cpp	2006-03-14 20:03:46.000000000 +0800
@@ -68,13 +68,14 @@
 	 */
 	static const PresenceTypeData data[] =
 	{
-		{ Presence::Offline,      OnlineStatus::Offline, OFFLINE,  OFFLINE, i18n( "O&ffline" ),        i18n("Offline"),        i18n("Offline"),                    0,                      "contact_invisible_overlay", Kopete::OnlineStatusManager::Offline,      0 },
-		{ Presence::DoNotDisturb, OnlineStatus::Away,    SET_DND,  IS_DND,  i18n( "&Do Not Disturb" ), i18n("Do Not Disturb"), i18n("Do Not Disturb (Invisible)"), "contact_busy_overlay", "contact_invisible_overlay", Kopete::OnlineStatusManager::Busy,         Kopete::OnlineStatusManager::HasAwayMessage },
-		{ Presence::Occupied,     OnlineStatus::Away,    SET_OCC,  IS_OCC,  i18n( "O&ccupied" ),       i18n("Occupied"),       i18n("Occupied (Invisible)"),       "contact_busy_overlay", "contact_invisible_overlay", 0,                                         Kopete::OnlineStatusManager::HasAwayMessage },
-		{ Presence::NotAvailable, OnlineStatus::Away,    SET_NA,   IS_NA,   i18n( "Not A&vailable" ),  i18n("Not Available"),  i18n("Not Available (Invisible)"),  "contact_xa_overlay",   "contact_invisible_overlay", Kopete::OnlineStatusManager::ExtendedAway, Kopete::OnlineStatusManager::HasAwayMessage },
-		{ Presence::Away,         OnlineStatus::Away,    SET_AWAY, IS_AWAY, i18n( "&Away" ),           i18n("Away"),           i18n("Away (Invisible)"),           "contact_away_overlay", "contact_invisible_overlay", Kopete::OnlineStatusManager::Away,         Kopete::OnlineStatusManager::HasAwayMessage },
-		{ Presence::FreeForChat,  OnlineStatus::Online,  SET_FFC,  IS_FFC,  i18n( "&Free for Chat" ),  i18n("Free For Chat"),  i18n("Free For Chat (Invisible)"),  "icq_ffc",              "contact_invisible_overlay", Kopete::OnlineStatusManager::FreeForChat,  0 },
-		{ Presence::Online,       OnlineStatus::Online,  ONLINE,   ONLINE,  i18n( "O&nline" ),         i18n("Online"),         i18n("Online (Invisible)"),         0,                      "contact_invisible_overlay", Kopete::OnlineStatusManager::Online,       0 }
+       { Presence::Offline,      OnlineStatus::Offline, OFFLINE,  OFFLINE, i18n( "O&ffline" ),        ("Offline"),        ("Offline"),                    0,                      "contact_invisible_overlay", Kopete::OnlineStatusManager::Offline,      0 },
+       { Presence::DoNotDisturb, OnlineStatus::Away,    SET_DND,  IS_DND,  i18n( "&Do Not Disturb" ), ("Do Not Disturb"), i18n("Do Not Disturb (Invisible)"), "contact_busy_overlay", "contact_invisible_overlay", 0,                                         Kopete::OnlineStatusManager::HasAwayMessage },
+       { Presence::Occupied,     OnlineStatus::Away,    SET_OCC,  IS_OCC,  i18n( "O&ccupied" ),       ("Occupied"),       i18n("Occupied (Invisible)"),       "contact_busy_overlay", "contact_invisible_overlay", Kopete::OnlineStatusManager::Busy,         Kopete::OnlineStatusManager::HasAwayMessage },
+       { Presence::NotAvailable, OnlineStatus::Away,    SET_NA,   IS_NA,   i18n( "Not A&vailable" ),  ("Not Available"),  i18n("Not Available (Invisible)"),  "contact_xa_overlay",   "contact_invisible_overlay", Kopete::OnlineStatusManager::ExtendedAway, Kopete::OnlineStatusManager::HasAwayMessage },
+       { Presence::Away,         OnlineStatus::Away,    SET_AWAY, IS_AWAY, i18n( "&Away" ),           ("Away"),           i18n("Away (Invisible)"),           "contact_away_overlay", "contact_invisible_overlay", Kopete::OnlineStatusManager::Away,         Kopete::OnlineStatusManager::HasAwayMessage },
+       { Presence::FreeForChat,  OnlineStatus::Online,  SET_FFC,  IS_FFC,  i18n( "&Free for Chat" ),  ("Free For Chat"),  i18n("Free For Chat (Invisible)"),  "icq_ffc",              "contact_invisible_overlay", Kopete::OnlineStatusManager::FreeForChat,  0 },
+       { Presence::Online,       OnlineStatus::Online,  ONLINE,   ONLINE,  i18n( "O&nline" ),         ("Online"),         i18n("Online (Invisible)"),         0,                      "contact_invisible_overlay", Kopete::OnlineStatusManager::Online,       0 }
+
 	};
 	return data;
 }
diff -Nur kopete-0.12b1/kopete/protocols/yahoo/yahooprotocol.cpp kopete-0.12b1f/kopete/protocols/yahoo/yahooprotocol.cpp
--- kopete-0.12b1/kopete/protocols/yahoo/yahooprotocol.cpp	2006-03-08 09:25:14.000000000 +0800
+++ kopete-0.12b1f/kopete/protocols/yahoo/yahooprotocol.cpp	2006-03-14 20:03:51.000000000 +0800
@@ -39,20 +39,20 @@
 
 YahooProtocol::YahooProtocol( TQObject *parent, const char *name, const TQStringList & )
 	: Kopete::Protocol( YahooProtocolFactory::instance(), parent, name ),
-	Offline( Kopete::OnlineStatus::Offline,      0, this, 0x5a55aa56, TQString(),  i18n( "Offline" ),       i18n( "Offline" ),     Kopete::OnlineStatusManager::Offline ),
-	Online( Kopete::OnlineStatus::Online,       25, this, 0, TQString(),           i18n( "Online" ),        i18n( "Online" ),      Kopete::OnlineStatusManager::Online, Kopete::OnlineStatusManager::HasAwayMessage  ),
-	BeRightBack( Kopete::OnlineStatus::Away,    22, this, 1, "contact_away_overlay",  i18n( "Be right back" ), i18n( "Be right back" ) ),
-	Busy( Kopete::OnlineStatus::Away,           20, this, 2, "contact_busy_overlay",  i18n( "Busy" ),          i18n( "Busy" ),        Kopete::OnlineStatusManager::Busy, Kopete::OnlineStatusManager::HasAwayMessage ),
-	NotAtHome( Kopete::OnlineStatus::Away,      17, this, 3, "contact_xa_overlay",    i18n( "Not at home" ),   i18n( "Not at home" ), Kopete::OnlineStatusManager::ExtendedAway ),
-	NotAtMyDesk( Kopete::OnlineStatus::Away,    18, this, 4, "contact_xa_overlay",    i18n( "Not at my desk"), i18n( "Not at my desk"), Kopete::OnlineStatusManager::Away ),
-	NotInTheOffice( Kopete::OnlineStatus::Away, 16, this, 5, "contact_xa_overlay",    i18n( "Not in the office" ), i18n( "Not in the office" ) ),
-	OnThePhone( Kopete::OnlineStatus::Away,     12, this, 6, "contact_phone_overlay", i18n( "On the phone" ), i18n( "On the phone" ) ),
-	OnVacation( Kopete::OnlineStatus::Away,      3, this, 7, "contact_xa_overlay",    i18n( "On vacation" ),  i18n( "On vacation" ) ),
-	OutToLunch( Kopete::OnlineStatus::Away,     10, this, 8, "contact_food_overlay",  i18n( "Out to lunch" ), i18n( "Out to lunch" ) ),
-	SteppedOut( Kopete::OnlineStatus::Away,     14, this, 9, "contact_away_overlay",  i18n( "Stepped out" ),  i18n( "Stepped out" ) ),
-	Invisible( Kopete::OnlineStatus::Invisible,  3, this, 12, "contact_invisible_overlay",  i18n( "Invisible" ), i18n( "Invisible" ), Kopete::OnlineStatusManager::Invisible ),
-	Custom( Kopete::OnlineStatus::Away,         25, this, 99, "contact_busy_overlay", i18n( "Custom" ),       i18n( "Custom" ),	Kopete::OnlineStatusManager::HideFromMenu ),
-	Idle( Kopete::OnlineStatus::Away,           15, this, 999, "yahoo_idle",          i18n( "Idle" ),         i18n( "Idle" ),         Kopete::OnlineStatusManager::Idle ),
+	Offline( Kopete::OnlineStatus::Offline,      0, this, 0x5a55aa56, TQString(),  ( "Offline" ),       i18n( "Offline" ),     Kopete::OnlineStatusManager::Offline ),
+	Online( Kopete::OnlineStatus::Online,       25, this, 0, TQString(),           ( "Online" ),        i18n( "Online" ),      Kopete::OnlineStatusManager::Online  ),
+	BeRightBack( Kopete::OnlineStatus::Away,    22, this, 1, "contact_away_overlay",  ( "Be right back" ), i18n( "Be right back" ) ),
+	Busy( Kopete::OnlineStatus::Away,           20, this, 2, "contact_busy_overlay",  ( "Busy" ),          i18n( "Busy" ),        Kopete::OnlineStatusManager::Busy, Kopete::OnlineStatusManager::HasAwayMessage ),
+	NotAtHome( Kopete::OnlineStatus::Away,      17, this, 3, "contact_xa_overlay",    ( "Not at home" ),   i18n( "Not at home" ), Kopete::OnlineStatusManager::ExtendedAway ),
+	NotAtMyDesk( Kopete::OnlineStatus::Away,    18, this, 4, "contact_xa_overlay",    ( "Not at my desk"), i18n( "Not at my desk"), Kopete::OnlineStatusManager::Away ),
+	NotInTheOffice( Kopete::OnlineStatus::Away, 16, this, 5, "contact_xa_overlay",    ( "Not in the office" ), i18n( "Not in the office" ) ),
+	OnThePhone( Kopete::OnlineStatus::Away,     12, this, 6, "contact_phone_overlay", ( "On the phone" ), i18n( "On the phone" ) ),
+	OnVacation( Kopete::OnlineStatus::Away,      3, this, 7, "contact_xa_overlay",    ( "On vacation" ),  i18n( "On vacation" ) ),
+	OutToLunch( Kopete::OnlineStatus::Away,     10, this, 8, "contact_food_overlay",  ( "Out to lunch" ), i18n( "Out to lunch" ) ),
+	SteppedOut( Kopete::OnlineStatus::Away,     14, this, 9, "contact_away_overlay",  ( "Stepped out" ),  i18n( "Stepped out" ) ),
+	Invisible( Kopete::OnlineStatus::Invisible,  3, this, 12, "contact_invisible_overlay",  ( "Invisible" ), i18n( "Invisible" ), Kopete::OnlineStatusManager::Invisible ),
+	Custom( Kopete::OnlineStatus::Away,         25, this, 99, "contact_busy_overlay", ( "Custom" ),       i18n( "Custom" ),	Kopete::OnlineStatusManager::HideFromMenu ),
+	Idle( Kopete::OnlineStatus::Away,           15, this, 999, "yahoo_idle",          ( "Idle" ),         i18n( "Idle" ),         Kopete::OnlineStatusManager::Idle ),
 	Connecting( Kopete::OnlineStatus::Connecting,2, this, 555, "yahoo_connecting",    i18n( "Connecting" ) ),
 	awayMessage(Kopete::Global::Properties::self()->awayMessage()),
 	iconCheckSum("iconCheckSum", i18n("Buddy Icon Checksum"), TQString(), true, false, true),
