diff -ur tdenetwork-git20111221.orig/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h tdenetwork-git20111221/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h
--- tdenetwork-git20111221.orig/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h	2011-12-21 00:10:19.000000000 -0500
+++ tdenetwork-git20111221/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h	2011-12-21 08:33:52.000000000 -0500
@@ -80,6 +80,7 @@
 
 #include <set>
 #include <list>
+#include <cstdlib>
 
 // On our copy of sigslot.h, we force single threading
 #define SIGSLOT_PURE_ISO

