diff -ur tdenetwork-git20111221.orig/kopete/protocols/sms/services/smsclient.cpp tdenetwork-git20111221/kopete/protocols/sms/services/smsclient.cpp
--- tdenetwork-git20111221.orig/kopete/protocols/sms/services/smsclient.cpp	2011-12-21 00:10:18.000000000 -0500
+++ tdenetwork-git20111221/kopete/protocols/sms/services/smsclient.cpp	2011-12-21 08:17:31.000000000 -0500
@@ -67,7 +67,8 @@
 		return;
 	}
 
-	TQString programName = c->readEntry(TQString("%1:%2").arg("SMSClient").arg("ProgramName"). TQString());
+	TQString programName = c->readEntry(TQString("%1:%2").arg("SMSClient").arg("ProgramName"), TQString());
+
 	if (programName.isNull())
 		programName = "/usr/bin/sms_client";
 
