--- PPI/Changes
+++ PPI/Changes
@@ -705,7 +705,7 @@
 	- Resolved Lexer bug CPAN #13425 ( $p{package} creates a PPI::Statement::Package )
 	  Added smarts to resolve word-started statements as ::Expression in subscripts
 	- Resolved PDOM bug CPAN #13326 ( problems in index_locations )
-	  Patch and comprehensive additional tests provided by johanl[ÄT]DarSerMan.com
+	  Patch and comprehensive additional tests provided by johanl[Ã„T]DarSerMan.com
 	- Removed dependency on Class::Autouse. Just load Tokenizer and Lexer up front.
 	- Removed dependency on File::Slurp. Only use it 3 times and it's not worth it
 	  when almost all the files we will read are under 50k.
