diff -up Fedora-Bugzilla-0.13/Makefile.PL.BAD Fedora-Bugzilla-0.13/Makefile.PL
--- Fedora-Bugzilla-0.13/Makefile.PL.BAD	2010-01-15 11:10:51.202841487 -0500
+++ Fedora-Bugzilla-0.13/Makefile.PL	2010-01-15 11:10:56.327724206 -0500
@@ -22,8 +22,6 @@ requires 'MooseX::AttributeHelpers' => '
 requires 'MooseX::CascadeClearing';
 requires 'MooseX::MultiInitArg';
 requires 'MooseX::StrictConstructor';
-requires 'MooseX::Traits::Attribute::CascadeClear';
-requires 'MooseX::Traits::Attribute::CascadeClearMaster';
 requires 'MooseX::Types';
 requires 'MooseX::Types::DateTime';
 requires 'MooseX::Types::Path::Class';
diff -up Fedora-Bugzilla-0.13/META.yml.BAD Fedora-Bugzilla-0.13/META.yml
--- Fedora-Bugzilla-0.13/META.yml.BAD	2010-01-15 11:11:09.939842426 -0500
+++ Fedora-Bugzilla-0.13/META.yml	2010-01-15 11:11:12.660724153 -0500
@@ -68,8 +68,6 @@ requires:
   MooseX::CascadeClearing: 0
   MooseX::MultiInitArg: 0
   MooseX::StrictConstructor: 0
-  MooseX::Traits::Attribute::CascadeClear: 0
-  MooseX::Traits::Attribute::CascadeClearMaster: 0
   MooseX::Types: 0
   MooseX::Types::DateTime: 0
   MooseX::Types::Path::Class: 0
