--- specifications/gherkin-2.11.6.gemspec.orig	2013-02-18 08:45:39.000000000 +0100
+++ specifications/gherkin-2.11.6.gemspec	2013-02-18 08:46:59.129973696 +0100
@@ -22,7 +22,7 @@
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
       s.add_development_dependency(%q<rake-compiler>, [">= 0.8.2"])
-      s.add_runtime_dependency(%q<json>, [">= 1.7.6"])
+      s.add_runtime_dependency(%q<json>, [">= 1.7.5"])
       s.add_development_dependency(%q<cucumber>, [">= 1.2.1"])
       s.add_development_dependency(%q<rake>, [">= 10.0.3"])
       s.add_development_dependency(%q<bundler>, [">= 1.2.3"])
@@ -35,7 +35,7 @@
       s.add_development_dependency(%q<builder>, [">= 3.1.4"])
     else
       s.add_dependency(%q<rake-compiler>, [">= 0.8.2"])
-      s.add_dependency(%q<json>, [">= 1.7.6"])
+      s.add_dependency(%q<json>, [">= 1.7.5"])
       s.add_dependency(%q<cucumber>, [">= 1.2.1"])
       s.add_dependency(%q<rake>, [">= 10.0.3"])
       s.add_dependency(%q<bundler>, [">= 1.2.3"])
@@ -49,7 +49,7 @@
     end
   else
     s.add_dependency(%q<rake-compiler>, [">= 0.8.2"])
-    s.add_dependency(%q<json>, [">= 1.7.6"])
+    s.add_dependency(%q<json>, [">= 1.7.5"])
     s.add_dependency(%q<cucumber>, [">= 1.2.1"])
     s.add_dependency(%q<rake>, [">= 10.0.3"])
     s.add_dependency(%q<bundler>, [">= 1.2.3"])
