diff -ur libtpcimgio.orig/Makefile libtpcimgio/Makefile
--- libtpcimgio.orig/Makefile	2014-08-01 12:12:23.524103855 +0000
+++ libtpcimgio/Makefile	2014-08-01 12:23:16.937087108 +0000
@@ -83,9 +83,12 @@
 		  img.c sifio.c sif.c weight.c libtpcimgiov.c
 OBJECTS 	= ${SOURCES:.c=.o}
 LIBARCHIVE	= ${LIBRARY}.a
+SHLIBSHORT      = ${LIBRARY}.so
+SHLIBMAJOR      = ${SHLIBSHORT}.0
+SHLIBARCHIVE    = ${SHLIBMAJOR}.0.0
 DOXYGEN		= doxygen
 CC		= gcc
-CFLAGS 		= -std=gnu99 -s -Wall -O3 ${INCLUDE} ${ANSI}
+CFLAGS 		= -std=gnu99 -s -Wall -O3 ${INCLUDE} ${ANSI} -fPIC -DPIC
 LIBS 		= ${LIBPATH} -ltpcimgio -ltpcmisc -lm
 ZIP		= zip
 PRVERS		= progvers -opt=-b
@@ -102,14 +105,20 @@
 LIBEXEC 	= ${LIBRARY}${EXT}
 
 # ------------- Compilation: --------------------------------------------------#
-compile: ${LIBARCHIVE} ${LIBEXEC}
+compile: ${LIBARCHIVE} ${SHLIBARCHIVE} ${LIBEXEC}
 
-# Compile library
+# Compile static library
 ${LIBARCHIVE}: ${OBJECTS}
 	${AR} -rvs ${LIBARCHIVE} ${OBJECTS}
 
+# Compile shared library
+${SHLIBARCHIVE}: ${OBJECTS}
+	${CC} -shared -o $@ ${OBJECTS} -Wl,-soname=${SHLIBMAJOR}
+	ln -s ${SHLIBARCHIVE} ${SHLIBMAJOR}
+	ln -s ${SHLIBARCHIVE} ${SHLIBSHORT}
+
 # Compile executable
-${LIBEXEC}: ${LIBARCHIVE} ${LIBRARY}.c
+${LIBEXEC}: ${SHLIBARCHIVE} ${LIBRARY}.c
 	${CC} ${LIBRARY}.c -o ${LIBEXEC} ${LIBPATH} ${CFLAGS} ${LIBS}
 ifneq ($(OS),Windows_NT)
 	 chmod a+x ${LIBEXEC}
@@ -129,8 +138,9 @@
 # ------------- Installation: -------------------------------------------------#
 install: libversion libinstall execinstall
 
-libinstall: ${LIBARCHIVE}
+libinstall: ${LIBARCHIVE} ${SHAREDLIBARCHIVE}
 	${ADDDATE} -i=${LIBRARY}.version -o=${PET_LIBPATH} -nodate -copy ${LIBARCHIVE}
+	${ADDDATE} -i=${LIBRARY}.version -o=${PET_LIB} -nodate -copy ${SHAREDLIBARCHIVE}
 
 execinstall: ${LIBEXEC}
 	${ADDDATE} -i=${LIBRARY}.version -o=${PET_BIN} -nodate -copy ${LIBEXEC}
