diff -up libtpcimgio/Makefile.shared libtpcimgio/Makefile
--- libtpcimgio/Makefile.shared	2010-09-18 07:05:53.000000000 -0400
+++ libtpcimgio/Makefile	2011-08-08 13:50:36.906876738 -0400
@@ -70,9 +70,12 @@ SOURCES 	= ecat63r.c ecat63w.c ecat63p.c
 		  sifio.c sif.c weight.c libtpcimgiov.c
 OBJECTS 	= ${SOURCES:.c=.o}
 LIBARCHIVE	= ${LIBRARY}.a
+SHLIBSHORT	= ${LIBRARY}.so
+SHLIBMAJOR	= ${SHLIBSHORT}.0
+SHLIBARCHIVE	= ${SHLIBMAJOR}.0.0
 DOXYGEN		= doxygen
 CC		= gcc
-CFLAGS 		= -std=gnu99 -s -Wall ${ANSI} -O2 ${INCLUDE}
+CFLAGS 		= -std=gnu99 -s -Wall ${ANSI} -O2 ${INCLUDE} -fPIC -DPIC
 LDFLAGS         = -L.
 LIBS 		= ${LDFLAGS} -ltpcimgio -ltpcmisc -lm
 ZIP		= zip
@@ -92,14 +95,20 @@ endif
 LIBEXEC 	= ${LIBRARY}${EXT}
 
 # ------------- Compilation: --------------------------------------------------#
-compile: ${LIBARCHIVE} ${LIBEXEC}
+compile: ${LIBARCHIVE} ${SHLIBARCHIVE} ${LIBEXEC}
 
-# Compile library
+# Compile static library
 ${LIBARCHIVE}: ${OBJECTS}
 	${AR} -rvs ${LIBARCHIVE} ${OBJECTS}
 
+# Compile shared library
+${SHLIBARCHIVE}: ${OBJECTS}
+	${CC} -shared -o $@ ${OBJECTS} -Wl,-soname=${SHLIBMAJOR}
+	ln -s ${SHLIBARCHIVE} ${SHLIBMAJOR}
+	ln -s ${SHLIBARCHIVE} ${SHLIBSHORT}
+
 # Compile executable
-${LIBEXEC}: ${LIBARCHIVE} ${LIBRARY}.c
+${LIBEXEC}: ${SHLIBARCHIVE} ${LIBRARY}.c
 	${CC} ${LIBRARY}.c -o ${LIBEXEC} ${CFLAGS} ${LIBS}
 ifneq ($(OS),Windows_NT)
 	 chmod a+x ${LIBEXEC}
@@ -119,8 +128,9 @@ ${LIBRARY}.version: ${LIBEXEC}
 # ------------- Installation: -------------------------------------------------#
 install: libversion libinstall execinstall
 
-libinstall: ${LIBARCHIVE}
+libinstall: ${LIBARCHIVE} ${SHAREDLIBARCHIVE}
 	${ADDDATE} -i=${LIBRARY}.version -o=${PET_LIB} -nodate -copy ${LIBARCHIVE}
+	${ADDDATE} -i=${LIBRARY}.version -o=${PET_LIB} -nodate -copy ${SHAREDLIBARCHIVE}
 
 execinstall: ${LIBEXEC}
 	${ADDDATE} -i=${LIBRARY}.version -o=${PET_BIN} -nodate -copy ${LIBEXEC}
