diff -urp telepathy-gabble-0.18.0.OLD/lib/ext/wocky/examples/Makefile.in telepathy-gabble-0.18.0/lib/ext/wocky/examples/Makefile.in
--- telepathy-gabble-0.18.0.OLD/lib/ext/wocky/examples/Makefile.in	2013-08-09 08:09:16.000000000 -0700
+++ telepathy-gabble-0.18.0/lib/ext/wocky/examples/Makefile.in	2013-08-11 07:54:36.512934714 -0700
@@ -370,6 +370,7 @@ wocky_register_SOURCES = register.c
 wocky_unregister_SOURCES = unregister.c
 LDADD = \
     @GLIB_LIBS@ \
+    @TLS_LIBS@ \
     $(top_builddir)/wocky/libwocky.la
 
 AM_CFLAGS = \
