diff -up wayland-1.2.0/src/connection.c.jx wayland-1.2.0/src/connection.c
--- wayland-1.2.0/src/connection.c.jx	2013-07-12 21:52:33.000000000 -0400
+++ wayland-1.2.0/src/connection.c	2013-10-07 13:30:40.418925754 -0400
@@ -283,8 +283,7 @@ wl_connection_flush(struct wl_connection
 		msg.msg_flags = 0;
 
 		do {
-			len = sendmsg(connection->fd, &msg,
-				      MSG_NOSIGNAL | MSG_DONTWAIT);
+			len = sendmsg(connection->fd, &msg, MSG_NOSIGNAL);
 		} while (len == -1 && errno == EINTR);
 
 		if (len == -1)
