diff -up DirectFB-1.2.8/inputdrivers/keyboard/keyboard.c.tweak DirectFB-1.2.8/inputdrivers/keyboard/keyboard.c
--- DirectFB-1.2.8/inputdrivers/keyboard/keyboard.c.tweak	2009-03-31 14:38:33.000000000 +0200
+++ DirectFB-1.2.8/inputdrivers/keyboard/keyboard.c	2009-05-11 14:41:54.000000000 +0200
@@ -341,7 +341,7 @@ driver_get_available( void )
                return 0;
      }
 
-     fd = open( "/dev/tty0", O_RDWR | O_NOCTTY );
+     fd = open( "/dev/tty1", O_RDWR | O_NOCTTY );
      if (fd < 0)
           return 0;
 
@@ -391,9 +391,9 @@ driver_open_device( CoreInputDevice  *de
           }
      }
      else {
-          fd = open( "/dev/tty0", O_RDWR | O_NOCTTY );
+          fd = open( "/dev/tty1", O_RDWR | O_NOCTTY );
           if (fd < 0) {
-               D_PERROR( "DirectFB/Keyboard: Could not open() /dev/tty0!\n" );
+               D_PERROR( "DirectFB/Keyboard: Could not open() /dev/tty1!\n" );
                return DFB_INIT;
           }
      }
diff -up DirectFB-1.2.8/src/core/core_parts.c.tweak DirectFB-1.2.8/src/core/core_parts.c
diff -up DirectFB-1.2.8/systems/fbdev/vt.c.tweak DirectFB-1.2.8/systems/fbdev/vt.c
--- DirectFB-1.2.8/systems/fbdev/vt.c.tweak	2009-03-31 14:38:33.000000000 +0200
+++ DirectFB-1.2.8/systems/fbdev/vt.c	2009-05-07 15:21:23.000000000 +0200
@@ -109,7 +109,7 @@ dfb_vt_initialize( void )
           return D_OOM();
 
      setsid();
-     dfb_vt->fd0 = open( "/dev/tty0", O_RDONLY | O_NOCTTY );
+     dfb_vt->fd0 = open( "/dev/tty1", O_RDONLY | O_NOCTTY );
      if (dfb_vt->fd0 < 0) {
           if (errno == ENOENT) {
                dfb_vt->fd0 = open( "/dev/vc/0", O_RDONLY | O_NOCTTY );
@@ -130,7 +130,7 @@ dfb_vt_initialize( void )
                }
           }
           else {
-               D_PERROR( "DirectFB/core/vt: Error opening `/dev/tty0'!\n");
+               D_PERROR( "DirectFB/core/vt: Error opening `/dev/tty1'!\n");
 
                D_FREE( dfb_vt );
                dfb_vt = NULL;
