diff -up DirectFB-1.5.3/interfaces/IDirectFBVideoProvider/idirectfbvideoprovider_v4l.c.fix_v4l1 DirectFB-1.5.3/interfaces/IDirectFBVideoProvider/idirectfbvideoprovider_v4l.c
--- DirectFB-1.5.3/interfaces/IDirectFBVideoProvider/idirectfbvideoprovider_v4l.c.fix_v4l1	2011-07-31 20:51:06.000000000 +0200
+++ DirectFB-1.5.3/interfaces/IDirectFBVideoProvider/idirectfbvideoprovider_v4l.c	2011-08-23 22:38:39.154862445 +0200
@@ -49,7 +49,7 @@
 #ifdef HAVE_LINUX_COMPILER_H
 #include <linux/compiler.h>
 #endif
-#include "videodev.h"
+#include <libv4l1-videodev.h>
 
 #include <directfb.h>
 
@@ -78,7 +78,7 @@
 #include <direct/util.h>
 
 #ifdef DFB_HAVE_V4L2
-#include "videodev2.h"
+#include <linux/videodev2.h>
 #endif
 
 static DFBResult
