diff -ur gtk+-3.13.3.orig/testsuite/reftests/Makefile.am gtk+-3.13.3/testsuite/reftests/Makefile.am
--- gtk+-3.13.3.orig/testsuite/reftests/Makefile.am	2014-06-26 20:17:01.878076285 +0000
+++ gtk+-3.13.3/testsuite/reftests/Makefile.am	2014-06-26 20:17:45.683075163 +0000
@@ -386,9 +386,7 @@
 	$(AM_V_GEN) sed $(substitutions) $< > $@.tmp && mv $@.tmp $@
 
 built_tests = 			\
-	reftests.test		\
-	reftests-dark.test	\
-	reftests-hc.test
+	reftests.test		
 
 DISTCLEANFILES = $(built_tests)
 
diff -ur gtk+-3.13.3.orig/testsuite/reftests/Makefile.in gtk+-3.13.3/testsuite/reftests/Makefile.in
--- gtk+-3.13.3.orig/testsuite/reftests/Makefile.in	2014-06-26 20:17:01.882076285 +0000
+++ gtk+-3.13.3/testsuite/reftests/Makefile.in	2014-06-26 20:17:59.788074801 +0000
@@ -888,9 +888,7 @@
 @BUILDOPT_INSTALL_TESTS_TRUE@	$(NULL)
 
 @BUILDOPT_INSTALL_TESTS_TRUE@built_tests = \
-@BUILDOPT_INSTALL_TESTS_TRUE@	reftests.test		\
-@BUILDOPT_INSTALL_TESTS_TRUE@	reftests-dark.test	\
-@BUILDOPT_INSTALL_TESTS_TRUE@	reftests-hc.test
+@BUILDOPT_INSTALL_TESTS_TRUE@	reftests.test		
 
 @BUILDOPT_INSTALL_TESTS_TRUE@DISTCLEANFILES = $(built_tests)
 @BUILDOPT_INSTALL_TESTS_TRUE@testmetadir = $(datadir)/installed-tests/$(PACKAGE)
