diff -Nuarp phonon-4.3.80/gstreamer/mediaobject.cpp phonon-4.3.80.gst_localfile_path/gstreamer/mediaobject.cpp
--- phonon-4.3.80/gstreamer/mediaobject.cpp	2009-12-04 03:29:40.000000000 +0800
+++ phonon-4.3.80.gst_localfile_path/gstreamer/mediaobject.cpp	2010-01-02 20:59:50.000000000 +0800
@@ -408,9 +408,10 @@ bool MediaObject::createPipefromURL(cons
 
     // Create a new datasource based on the input URL
     // add the 'file' scheme if it's missing; the double '/' is needed!
+    // toEncoded() will corrupt the path in non-utf8 locale system
     QByteArray encoded_cstr_url = (url.scheme() == QLatin1String("") ?
-                    "file://" + url.toEncoded() :
-                    url.toEncoded());
+                    QByteArray("file://") + qPrintable(url.toString()) :
+                    QByteArray(qPrintable(url.toString())));
     m_datasource = gst_element_make_from_uri(GST_URI_SRC, encoded_cstr_url.constData(), (const char*)NULL);
     if (!m_datasource)
         return false;
