From 870bcbda2f01f801bbbabfd3c4b2956f79572d9b Mon Sep 17 00:00:00 2001
From: Sean Hefty <sean.hefty@intel.com>
Date: Wed, 14 Dec 2011 16:38:45 -0800
Subject: [PATCH 9/9] librdmacm: Update web site and email addresses

Signed-off-by: Sean Hefty <sean.hefty@intel.com>
---
 configure.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index 03dbed8..8d89741 100644
--- a/configure.in
+++ b/configure.in
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.57)
-AC_INIT(librdmacm, 1.0.15, general@lists.openfabrics.org)
+AC_INIT(librdmacm, 1.0.15, linux-rdma@vger.kernel.org)
 AC_CONFIG_SRCDIR([src/cma.c])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR(config)
-- 
1.7.6.4

