diff -Nur amarok-1.3-orig/amarok/src/engine/gst10/config/gstconfig.kcfg amarok-1.3/amarok/src/engine/gst10/config/gstconfig.kcfg
--- amarok-1.3-orig/amarok/src/engine/gst10/config/gstconfig.kcfg 2005-08-14 17:53:18.000000000 -0500
+++ amarok-1.3/amarok/src/engine/gst10/config/gstconfig.kcfg  2005-08-16 12:15:31.000000000 -0500
@@ -10,7 +10,7 @@
         <entry key="Sound Output" type="String">
             <label>Sound output method to use</label>
             <whatsthis>Select the sound output plugin ("sink"). Their availability depends on the system configuration.</whatsthis>
-            <default>osssink</default>
+            <default>autoaudiosink</default>
         </entry>
         <entry key="Use Custom Sound Device" type="Bool">
             <label>Do not use the autodetected audiosink sound device</label>
