diff -Nur amarok-1.4-beta1/amarok/src/metadata/m4a/mp4audiosampleentry.h amarok-1.4-beta1n/amarok/src/metadata/m4a/mp4audiosampleentry.h
--- amarok-1.4-beta1/amarok/src/metadata/m4a/mp4audiosampleentry.h	2006-02-12 12:49:38.000000000 -0500
+++ amarok-1.4-beta1n/amarok/src/metadata/m4a/mp4audiosampleentry.h	2006-02-28 02:38:45.779083056 -0500
@@ -11,7 +11,7 @@
     class Mp4AudioSampleEntry: public Mp4SampleEntry
     {
     public:
-      Mp4AudioSampleEntry( File* file, MP4::Fourcc fourcc, uint size, long offset );
+      Mp4AudioSampleEntry( TagLib::File* file, MP4::Fourcc fourcc, uint size, long offset );
       ~Mp4AudioSampleEntry();
 
       //! function to get the number of channels
diff -Nur amarok-1.4-beta1/amarok/src/metadata/m4a/mp4sampleentry.h amarok-1.4-beta1n/amarok/src/metadata/m4a/mp4sampleentry.h
--- amarok-1.4-beta1/amarok/src/metadata/m4a/mp4sampleentry.h	2006-02-12 12:49:38.000000000 -0500
+++ amarok-1.4-beta1n/amarok/src/metadata/m4a/mp4sampleentry.h	2006-02-28 02:38:46.174023016 -0500
@@ -11,7 +11,7 @@
     class Mp4SampleEntry: public Mp4IsoBox
     {
     public:
-      Mp4SampleEntry( File* file, MP4::Fourcc fourcc, uint size, long offset );
+      Mp4SampleEntry( TagLib::File* file, MP4::Fourcc fourcc, uint size, long offset );
       ~Mp4SampleEntry();
 
     public:
diff -Nur amarok-1.4-beta1/amarok/src/metadata/m4a/mp4skipbox.h amarok-1.4-beta1n/amarok/src/metadata/m4a/mp4skipbox.h
--- amarok-1.4-beta1/amarok/src/metadata/m4a/mp4skipbox.h	2006-02-12 12:49:38.000000000 -0500
+++ amarok-1.4-beta1n/amarok/src/metadata/m4a/mp4skipbox.h	2006-02-28 02:38:46.573962216 -0500
@@ -11,7 +11,7 @@
     class Mp4SkipBox: public Mp4IsoBox
     {
     public:
-      Mp4SkipBox( File* file, MP4::Fourcc fourcc, uint size, long offset );
+      Mp4SkipBox( TagLib::File* file, MP4::Fourcc fourcc, uint size, long offset );
       ~Mp4SkipBox();
 
     private:
