--- amarok-1.4.7/amarok/src/metabundle.cpp	2007-08-08 01:48:28.000000000 +0800
+++ amarok-1.4.7.local/amarok/src/metabundle.cpp	2007-12-17 00:14:06.000000000 +0800
@@ -511,11 +511,11 @@
         if ( tag )
         {
             #define strip( x ) TStringToQString( x ).stripWhiteSpace()
-            setTitle( strip( tag->title() ) );
-            setArtist( strip( tag->artist() ) );
-            setAlbum( strip( tag->album() ) );
-            setComment( strip( tag->comment() ) );
-            setGenre( strip( tag->genre() ) );
+            setTitle( QString::fromLocal8Bit(strip( tag->title()).latin1()) );
+            setArtist( QString::fromLocal8Bit(strip( tag->artist() ).latin1()) );
+            setAlbum( QString::fromLocal8Bit(strip( tag->album() ).latin1()) );
+            setComment( QString::fromLocal8Bit(strip( tag->comment() ).latin1()) );
+            setGenre( QString::fromLocal8Bit(strip( tag->genre() ).latin1()) );
             setYear( tag->year() );
             setTrack( tag->track() );
             #undef strip
@@ -1471,6 +1471,7 @@
     {
         TagLib::Tag * t = f->tag();
         if ( t ) { // f.tag() can return null if the file couldn't be opened for writing
+            #define QStringToTString(s) TagLib::String(s.local8Bit ().data())
             t->setTitle( QStringToTString( title().stripWhiteSpace() ) );
             t->setArtist( QStringToTString( artist().string().stripWhiteSpace() ) );
             t->setAlbum( QStringToTString( album().string().stripWhiteSpace() ) );
