diff -ur gst-plugins-bad-1.3.3.orig/ext/directfb/dfb-example.c gst-plugins-bad-1.3.3/ext/directfb/dfb-example.c
--- gst-plugins-bad-1.3.3.orig/ext/directfb/dfb-example.c	2014-06-25 08:58:21.815334297 +0000
+++ gst-plugins-bad-1.3.3/ext/directfb/dfb-example.c	2014-06-25 09:00:28.272331056 +0000
@@ -1,12 +1,11 @@
 
-#include <directfb.h>
-
 #ifdef __no_instrument_function__
 #undef __no_instrument_function__
 #endif
 
 #include <stdio.h>
 #include <gst/gst.h>
+#include <directfb.h>
 
 static IDirectFB *dfb = NULL;
 static IDirectFBSurface *primary = NULL;
