--- extlib-1.5.4/install.ml.old	2013-09-14 13:06:41.605862541 +0100
+++ extlib-1.5.4/install.ml	2013-09-14 13:06:48.621872096 +0100
@@ -64,10 +64,18 @@
 		print_endline ("Installing " ^ file);
 		let path = dest ^ file in
 		(try Sys.remove path with _ -> ());
-		try
-			Sys.rename file path;
-		with
-			_ -> failwith "Aborted"
+		let ichan = open_in file in
+		let ochan = open_out path in
+		let buf = String.create 1024 in
+		let rec loop () =
+			let len = input ichan buf 0 1024 in
+			if len > 0 then (
+				output ochan buf 0 len;
+				loop ()
+			)
+		in
+		loop ();
+		Sys.remove file
 	end
 
 let get_version () =
