--- extlib-1.5.4/install.ml.old	2013-09-14 13:08:00.641970190 +0100
+++ extlib-1.5.4/install.ml	2013-09-14 13:08:38.569021874 +0100
@@ -182,16 +182,16 @@
 	  | Dir install_dir ->
 	      sprintf "%sextlib-doc" install_dir in
 	if !autodoc && not (Sys.file_exists doc_dir) then run (sprintf "mkdir %s" doc_dir);
-	run (sprintf "ocamlc -c %s" (m_list ".mli"));
+	run (sprintf "ocamlc -g -c %s" (m_list ".mli"));
 	if !autobyte then begin
-		List.iter (fun m -> run (sprintf "ocamlc -c %s.ml" m)) modules;
-		run (sprintf "ocamlc -a -o extLib.cma %s extLib.ml" (m_list ".cmo"));
+		List.iter (fun m -> run (sprintf "ocamlc -g -c %s.ml" m)) modules;
+		run (sprintf "ocamlc -g -a -o extLib.cma %s extLib.ml" (m_list ".cmo"));
 		List.iter (fun m -> remove (m ^ ".cmo")) modules;
 		remove "extLib.cmo";
 	end;
 	if !autonative then begin
-		List.iter (fun m -> run (sprintf "ocamlopt -c %s.ml" m)) modules;
-		run (sprintf "ocamlopt -a -o extLib.cmxa %s extLib.ml" (m_list ".cmx"));
+		List.iter (fun m -> run (sprintf "ocamlopt -g -c %s.ml" m)) modules;
+		run (sprintf "ocamlopt -g -a -o extLib.cmxa %s extLib.ml" (m_list ".cmx"));
 		List.iter (fun m -> remove (m ^ obj_ext)) modules;
 		remove ("extLib" ^ obj_ext);
 	end;
--- extlib-1.5.4/Makefile.old	2013-09-14 13:08:04.020974793 +0100
+++ extlib-1.5.4/Makefile	2013-09-14 13:08:22.644000168 +0100
@@ -16,13 +16,13 @@
 .PHONY: all opt cmxs doc install uninstall clean release
 
 all: 
-	ocamlc -a -o extLib.cma $(SRC)
+	ocamlc -g -a -o extLib.cma $(SRC)
 opt: 
-	ocamlopt -a -o extLib.cmxa $(SRC)
+	ocamlopt -g -a -o extLib.cmxa $(SRC)
 cmxs: opt
-	ocamlopt -shared -linkall extLib.cmxa -o extLib.cmxs
+	ocamlopt -g -shared -linkall extLib.cmxa -o extLib.cmxs
 doc:
-	ocamlc -c $(MLI)
+	ocamlc -g -c $(MLI)
 	ocamldoc -sort -html -d doc/ $(MLI)
 
 install:
