--- cabal-install-1.18.0.8/cabal-install.cabal~	2015-03-25 18:13:31.448000000 +0900
+++ cabal-install-1.18.0.8/cabal-install.cabal	2015-03-25 18:15:52.029000000 +0900
@@ -144,7 +144,7 @@
     if flag(network-uri)
       build-depends: network-uri >= 2.6, network >= 2.6
     else
-      build-depends: network-uri < 2.6, network < 2.6
+      build-depends: network < 2.6
 
     if os(windows)
       build-depends: Win32 >= 2 && < 3
@@ -198,7 +198,7 @@
   if flag(network-uri)
     build-depends: network-uri >= 2.6, network >= 2.6
   else
-    build-depends: network-uri < 2.6, network < 2.6
+    build-depends: network < 2.6
 
   if os(windows)
     build-depends: Win32
