diff -up Devel-Hide-0.0008/lib/Devel/Hide.pm.orig Devel-Hide-0.0008/lib/Devel/Hide.pm
--- Devel-Hide-0.0008/lib/Devel/Hide.pm.orig	2007-11-15 07:45:02.000000000 -0700
+++ Devel-Hide-0.0008/lib/Devel/Hide.pm	2012-06-28 05:27:24.000000000 -0600
@@ -101,7 +101,7 @@ sub _push_hidden {
 BEGIN {
 
     # unless @HIDDEN was user-defined elsewhere, set default
-    if ( !defined @HIDDEN && $ENV{DEVEL_HIDE_PM} ) {
+    if ( !@HIDDEN && $ENV{DEVEL_HIDE_PM} ) {
         _push_hidden( split q{ }, $ENV{DEVEL_HIDE_PM} );
 
         # NOTE. "split ' ', $s" is special. Read "perldoc -f split".
