From 668a4b58b1cd7d8bf9b449bb66566351158f0efe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 23 Jun 2015 16:27:45 +0200
Subject: [PATCH] Do not install examples into PATH
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

CPAN RT#105426

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 Makefile.PL | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/Makefile.PL b/Makefile.PL
index cb3d8d7..95cd7bb 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -53,12 +53,4 @@ WriteMakefile(
 									'Term::ReadKey' => 0,
 									'Getopt::Long' => 0,
 								   },
-    'EXE_FILES' =>
-                [
-                'examples/flickr_oauth_authentication.pl',
-                'examples/flickr_method_test_echo.pl',
-                'examples/flickr_method_test_login.pl',
-                'examples/flickr_dump_stored_config.pl',
-                'examples/flickr_make_stored_config.pl',
-                ]
 );
-- 
2.1.0

