diff --git a/src/net/sourceforge/cobertura/javancss/FunctionMetric.java b/src/net/sourceforge/cobertura/javancss/FunctionMetric.java
index a0df0b3..9cd08a2 100644
--- a/src/net/sourceforge/cobertura/javancss/FunctionMetric.java
+++ b/src/net/sourceforge/cobertura/javancss/FunctionMetric.java
@@ -63,7 +63,7 @@ package net.sourceforge.cobertura.javancss;
 /**
  * Basic data class to store all metrics attached to a function.
  *
- * @author  Hervé Boutemy
+ * @author  Herve Boutemy
  * @version $Id: FunctionMetric.java 121 2009-01-17 22:19:45Z hboutemy $
  */
 public class FunctionMetric  extends Metric
diff --git a/src/net/sourceforge/cobertura/javancss/Javancss.java b/src/net/sourceforge/cobertura/javancss/Javancss.java
index 5f8e79f..26a615e 100644
--- a/src/net/sourceforge/cobertura/javancss/Javancss.java
+++ b/src/net/sourceforge/cobertura/javancss/Javancss.java
@@ -84,7 +84,7 @@ import net.sourceforge.cobertura.javancss.test.JavancssTest;
  * invokes the Java parser.
  *
  * @author    Chr. Clemens Lee <clemens@kclee.com>
- *            , recursive feature by Pääkö Hannu
+ *            , recursive feature by Paako Hannu
  *            , additional javadoc metrics by Emilio Gongora <emilio@sms.nl>
  *            , and Guillermo Rodriguez <guille@sms.nl>.
  * @version   $Id: Javancss.java 676 2009-09-04 13:42:13Z lewijw $
diff --git a/src/net/sourceforge/cobertura/javancss/Metric.java b/src/net/sourceforge/cobertura/javancss/Metric.java
index 0e73447..0c99a0d 100644
--- a/src/net/sourceforge/cobertura/javancss/Metric.java
+++ b/src/net/sourceforge/cobertura/javancss/Metric.java
@@ -63,7 +63,7 @@ package net.sourceforge.cobertura.javancss;
 /**
  * Base data class to store all metrics common to packages, objects and functions.
  *
- * @author  Hervé Boutemy
+ * @author  Herve Boutemy
  * @version $Id: Metric.java 121 2009-01-17 22:19:45Z hboutemy $
  */
 public abstract class Metric implements Comparable
diff --git a/src/net/sourceforge/cobertura/javancss/ObjectMetric.java b/src/net/sourceforge/cobertura/javancss/ObjectMetric.java
index ea2ff26..4e97d48 100644
--- a/src/net/sourceforge/cobertura/javancss/ObjectMetric.java
+++ b/src/net/sourceforge/cobertura/javancss/ObjectMetric.java
@@ -63,7 +63,7 @@ package net.sourceforge.cobertura.javancss;
 /**
  * Basic data class to store all metrics attached to an object.
  *
- * @author  Hervé Boutemy
+ * @author  Herve Boutemy
  * @version $Id: ObjectMetric.java 121 2009-01-17 22:19:45Z hboutemy $
  */
 public class ObjectMetric extends Metric
