--- gambas-1.0.11/debian/gambas.desktop.DESK	2004-09-22 06:10:47.000000000 -0500
+++ gambas-1.0.11/debian/gambas.desktop	2005-09-05 00:01:06.000000000 -0500
@@ -1,14 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
-Version=1.0
 Exec=gambas
-TryExec=
 Icon=/usr/share/pixmaps/gambas.png
-X-GNOME-DocPath=
 Terminal=false
 Name=Gambas
 Name[es_ES]=Gambas
-GenericName[es_ES]=
+GenericName=Basic IDE
+Comment=IDE based on a Basic interpreter
 Comment[es_ES]=Gambas casi es Visual Basic
 Type=Application
 Categories=GNOME;Application;Development;
diff -up gambas-1.0.19/debian/gambas.desktop.BAD gambas-1.0.19/debian/gambas.desktop
--- gambas-1.0.19/debian/gambas.desktop.BAD	2009-03-04 16:34:30.000000000 -0500
+++ gambas-1.0.19/debian/gambas.desktop	2009-03-04 16:35:32.000000000 -0500
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Exec=gambas
-Icon=/usr/share/pixmaps/gambas.png
+Icon=gambas
 Terminal=false
 Name=Gambas
 Name[es_ES]=Gambas
@@ -9,5 +9,5 @@ GenericName=Basic IDE
 Comment=IDE based on a Basic interpreter
 Comment[es_ES]=Gambas casi es Visual Basic
 Type=Application
-Categories=GNOME;Application;Development;
+Categories=GNOME;GTK;Development;IDE;
 StartupNotify=true
