--- irda-utils-0.9.13/etc/irda.rc.i18n	Thu Feb  8 14:04:06 2001
+++ irda-utils-0.9.13/etc/irda.rc	Thu Feb  8 14:05:26 2001
@@ -13,10 +13,11 @@
 # Source IrDA networking configuration.
 . /etc/sysconfig/irda
 
+prog="IrDA"
 
 start() {
         # Attach irda device 
-        echo -n "Starting IrDA: "
+        echo -n $"Starting $prog: "
         daemon /usr/sbin/irattach ${DEVICE} ${ARGS}
 	touch /var/lock/subsys/irda
         echo
@@ -24,7 +25,7 @@
 
 stop() {
         # Stop service.
-        echo -n "Shutting down IrDA: "
+        echo -n $"Shutting down $prog: "
 	killproc irattach
 	rm -f /var/lock/subsys/irda
         echo
@@ -62,7 +63,7 @@
 	[ -e /var/lock/subsys/irda ] && (stop; start)
 	;;
   *)
-        echo "Usage: irda {start|stop|restart|reload|condrestart|status}"
+        echo $"Usage: $0 {start|stop|restart|reload|condrestart|status}"
         exit 1
 esac
 
