--- irda-utils-0.9.17/etc/irda.rc.rh1	2006-08-01 14:39:44.000000000 +0200
+++ irda-utils-0.9.17/etc/irda.rc	2006-08-01 14:40:02.000000000 +0200
@@ -24,16 +24,15 @@
 	/sbin/modprobe ircomm-tty 2>/dev/null
 	/sbin/modprobe irtty-sir 2>/dev/null
 	/sbin/modprobe irnet 2>/dev/null
-        daemon /usr/sbin/irattach ${DEVICE} ${ARGS}
-	touch /var/lock/subsys/irda
+	daemon /usr/sbin/irattach ${DEVICE} ${ARGS}
+	[ -f /var/run/irattach.pid ] && success $"irattach startup" || failure $"irattach startup"
         echo
 }
 
 stop() {
         # Stop service.
         echo -n $"Shutting down $prog: "
-	killproc irattach
-	rm -f /var/lock/subsys/irda
+        [ -f /var/run/irattach.pid ] && killproc irattach || success $"irattach shutdown"
         echo
 }
 
@@ -66,7 +65,7 @@
 	start
 	;;
   condrestart)
-	[ -e /var/lock/subsys/irda ] && (stop; start)
+	[ -e /var/run/irdattach.pid ] && (stop; start)
 	;;
   *)
         echo $"Usage: $0 {start|stop|restart|reload|condrestart|status}"
