--- irda-utils-0.9.14/etc/irda.rc.chkconfig	Sun Jan 14 23:12:56 2001
+++ irda-utils-0.9.14/etc/irda.rc	Tue Apr 24 15:43:33 2001
@@ -3,8 +3,7 @@
 # irda          This shell script takes care of starting and stopping
 #               IrDA support
 #
-# chkconfig: 2345 45 96
-#
+# chkconfig: - 45 24
 # description: IrDA stack for Linux
 #
 
@@ -14,6 +13,23 @@
 # Source IrDA networking configuration.
 . /etc/sysconfig/irda
 
+
+start() {
+        # Attach irda device 
+        echo -n "Starting IrDA: "
+        daemon /usr/sbin/irattach ${DEVICE} ${ARGS}
+	touch /var/lock/subsys/irda
+        echo
+}
+
+stop() {
+        # Stop service.
+        echo -n "Shutting down IrDA: "
+	killproc irattach
+	rm -f /var/lock/subsys/irda
+        echo
+}
+
 # Check that irda is up.
 [ ${IRDA} = "no" ] && exit 0
 
@@ -30,28 +46,23 @@
 # See how we were called.
 case "$1" in
   start)
-        # Attach irda device 
-        echo -n "Starting IrDA: "
-        daemon /usr/sbin/irattach ${DEVICE} ${ARGS}
-	touch /var/lock/subsys/irda
-        echo
+	start
         ;;
   stop)
-        # Stop service.
-        echo -n "Shutting down IrDA: "
-	killproc irattach
-	rm -f /var/lock/subsys/irda
-        echo
+	stop
         ;;
   status)
 	status irattach
 	;;
   restart|reload)
-	$0 stop
-	$0 start
+	stop
+	start
+	;;
+  condrestart)
+	[ -e /var/lock/subsys/irda ] && (stop; start)
 	;;
   *)
-        echo "Usage: irda {start|stop|restart|reload|status}"
+        echo "Usage: irda {start|stop|restart|reload|condrestart|status}"
         exit 1
 esac
 
--- irda-utils-0.9.14/etc/irda.rc.chkconfig	2002-12-16 14:01:23.000000000 -0500
+++ irda-utils-0.9.14/etc/irda.rc	2002-12-16 13:56:11.000000000 -0500
@@ -4,7 +4,10 @@
 #               IrDA support
 #
 # chkconfig: - 45 24
-# description: IrDA stack for Linux
+# description: IrDA(TM) (Infrared Data Association) is an industry standard  \
+# for wireless, infrared communication between devices. IrDA speeds range \
+# from 9600 bps to 4 Mbps, and IrDA can be used by many modern devices \
+# including laptops, LAN adapters, PDAs, printers, and mobile phones.
 #
 
 # Source function library.
