--- irda-utils-0.9.16/etc/irda.rc.rh2	2004-10-04 12:49:46.016198784 +0200
+++ irda-utils-0.9.16/etc/irda.rc	2004-10-04 12:50:24.128404848 +0200
@@ -21,6 +21,9 @@
 start() {
         # Attach irda device 
         echo -n $"Starting $prog: "
+	/sbin/modprobe ircomm-tty 2>/dev/null
+	/sbin/modprobe irtty-sir 2>/dev/null
+	/sbin/modprobe irnet 2>/dev/null
         daemon /usr/sbin/irattach ${DEVICE} ${ARGS}
 	touch /var/lock/subsys/irda
         echo
