Index: fpcsrc/compiler/utils/samplecfg
===================================================================
--- fpcsrc/compiler/utils/samplecfg	(version 2.4.2)
+++ fpcsrc/compiler/utils/samplecfg	(working copy)
@@ -379,6 +379,11 @@
 #endif
 
 # -----------------------
+# Use BuildId by default
+# -----------------------
+-k--build-id
+
+# -----------------------
 # Set Filenames and Paths
 # -----------------------
 
