From 62ab50f64fa210304338a5bfccdc6a26fac5f0ec Mon Sep 17 00:00:00 2001
From: "Dawn K. Isabel" <dawn.isabel@gmail.com>
Date: Thu, 30 May 2013 23:36:52 -0400
Subject: [PATCH] cython: Explicitly cast ssl_enabled when calling
 lockdownd_start_session


diff --git a/cython/lockdown.pxi b/cython/lockdown.pxi
index 6fb49a3..ee674b4 100644
--- a/cython/lockdown.pxi
+++ b/cython/lockdown.pxi
@@ -221,7 +221,7 @@ cdef class LockdownClient(PropertyListService):
             char* c_session_id = NULL
             bint ssl_enabled
             bytes session_id
-        err = lockdownd_start_session(self._c_client, host_id, &c_session_id, &ssl_enabled)
+        err = lockdownd_start_session(self._c_client, host_id, &c_session_id, <int *>&ssl_enabled)
         try:
             self.handle_error(err)
