From 4b9d4103e5ff91b022ee8e9522040829f009543a Mon Sep 17 00:00:00 2001
From: Alon Levy <alevy@redhat.com>
Date: Tue, 21 Aug 2012 13:51:32 +0300
Subject: [PATCH] qxl: disallow unknown revisions

Signed-off-by: Alon Levy <alevy@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 hw/qxl.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hw/qxl.c b/hw/qxl.c
index 59bf822..71879fe 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -1873,9 +1873,9 @@ static int qxl_init_common(PCIQXLDevice *qxl)
         break;
 #endif
     default:
-        pci_device_rev = QXL_DEFAULT_REVISION;
-        io_size = msb_mask(QXL_IO_RANGE_SIZE * 2 - 1);
-        break;
+        error_report("Invalid revision %d for qxl device (max %d)",
+                     qxl->revision, QXL_DEFAULT_REVISION);
+        return -1;
     }
 
     pci_set_byte(&config[PCI_REVISION_ID], pci_device_rev);
-- 
1.7.12.1

