From c12ef3d896e426505ea3ca4fb2f1d9017f9cf828 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Mon, 3 Sep 2012 13:44:04 +0200
Subject: [PATCH] usb-redir: Enable pipelining for bulk endpoints

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 hw/usb/redirect.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hw/usb/redirect.c b/hw/usb/redirect.c
index f1bb692..f183263 100644
--- a/hw/usb/redirect.c
+++ b/hw/usb/redirect.c
@@ -1219,6 +1219,9 @@ static void usbredir_ep_info(void *priv,
                                      usb_redir_cap_ep_info_max_packet_size)) {
             usb_ep->max_packet_size = ep_info->max_packet_size[i];
         }
+        if (ep_info->type[i] == usb_redir_type_bulk) {
+            usb_ep->pipeline = true;
+        }
     }
 }
 
-- 
1.7.12.1

