From e0575d0a1a14e8f4e8fc11d549cbd07cca433383 Mon Sep 17 00:00:00 2001
From: Alon Levy <alevy@redhat.com>
Date: Tue, 30 Oct 2012 18:00:33 +0200
Subject: [PATCH] wip: hw/qxl: inject interrupts in any state

---
 hw/qxl.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/hw/qxl.c b/hw/qxl.c
index 9389752..b137731 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -1714,7 +1714,6 @@ static void qxl_send_events(PCIQXLDevice *d, uint32_t events)
     uint32_t le_events = cpu_to_le32(events);
 
     trace_qxl_send_events(d->id, events);
-    assert(qemu_spice_display_is_running(&d->ssd));
     old_pending = __sync_fetch_and_or(&d->ram->int_pending, le_events);
     if ((old_pending & le_events) == le_events) {
         return;
-- 
1.7.12.1

