--- src/logrotate.conf	2012-07-03 00:22:59.000000000 -0400
+++ src/logrotate.conf.new	2012-09-09 12:56:09.912929358 -0400
@@ -4,7 +4,7 @@
     compress
     sharedscripts
     postrotate
-        invoke-rc.d ceph reload >/dev/null || service ceph reload >/dev/null
+        service ceph reload >/dev/null 2>/dev/null || true
 
 	# upstart reload isn't very helpful here:
 	#   https://bugs.launchpad.net/upstart/+bug/1012938
