--- ./src/unix/lwt_unix.ml.orig	2012-12-27 05:29:57.000000000 -0700
+++ ./src/unix/lwt_unix.ml	2013-09-18 11:00:00.000000000 -0600
@@ -596,6 +596,9 @@ type open_flag =
 #if ocaml_version >= (3, 13)
   | O_SHARE_DELETE
 #endif
+#if ocaml_version >= (4, 1)
+  | O_CLOEXEC
+#endif
 
 #if windows
 
--- ./src/unix/lwt_unix.mli.orig	2012-12-27 05:29:57.000000000 -0700
+++ ./src/unix/lwt_unix.mli	2013-09-18 11:00:00.000000000 -0600
@@ -315,6 +315,9 @@ type open_flag =
 #if ocaml_version >= (3, 13)
   | O_SHARE_DELETE
 #endif
+#if ocaml_version >= (4, 1)
+  | O_CLOEXEC
+#endif
 
 val openfile : string -> open_flag list -> file_perm -> file_descr Lwt.t
   (** Wrapper for [Unix.openfile]. *)
