diff -ur openjdk.orig/hotspot/src/share/vm/interpreter/abstractInterpreter.hpp openjdk/hotspot/src/share/vm/interpreter/abstractInterpreter.hpp
--- openjdk.orig/hotspot/src/share/vm/interpreter/abstractInterpreter.hpp	2015-10-24 03:41:16.525177435 +0000
+++ openjdk/hotspot/src/share/vm/interpreter/abstractInterpreter.hpp	2015-10-24 03:46:47.855168943 +0000
@@ -34,7 +34,7 @@
 # include INTERP_MASM_MD_HPP
 #elif defined TARGET_ARCH_x86
 # include "interp_masm_x86.hpp"
-#elif TARGET_ARCH_MODEL_aarch64
+#elif defined TARGET_ARCH_MODEL_aarch64
 # include "interp_masm_aarch64.hpp"
 #elif defined TARGET_ARCH_MODEL_sparc
 # include "interp_masm_sparc.hpp"
