diff -ur k9copy-trinity-3.5.13.2.orig/k9author/k9avidecode.h k9copy-trinity-3.5.13.2/k9author/k9avidecode.h
--- k9copy-trinity-3.5.13.2.orig/k9author/k9avidecode.h	2013-07-25 10:24:51.000000000 +0000
+++ k9copy-trinity-3.5.13.2/k9author/k9avidecode.h	2013-07-25 10:25:20.000000000 +0000
@@ -43,7 +43,7 @@
 #else
 typedef int (*av_find_stream_info_t)(AVFormatContext *);
 #endif
-typedef AVCodec* (*avcodec_find_decoder_t)(enum CodecID);
+typedef AVCodec* (*avcodec_find_decoder_t)(enum AVCodecID);
 #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 8, 0)
 typedef int (*avcodec_open2_t)(AVCodecContext *, AVCodec *, AVDictionary **);
 #else
