--- trinity-k9copy-3.5.13.2/src/Makefile.am.ORI	2013-06-17 20:59:49.620929488 +0200
+++ trinity-k9copy-3.5.13.2/src/Makefile.am	2013-06-17 21:00:11.150425671 +0200
@@ -35,7 +35,7 @@
 	$(top_builddir)/k9author/libk9author.la $(top_builddir)/k9decmpeg/libk9decmpeg.la \
 	$(top_builddir)/k9devices/libk9devices.la $(top_builddir)/k9vamps/libk9vamps.la \
 	$(top_builddir)/libdvdnav/libk9dvdnav.la $(top_builddir)/libk9copy/libk9copy.la \
-	$(top_builddir)/dvdread/libdvdread.la $(LIB_KDEUI) -lkmdi $(LIBDL)
+	$(top_builddir)/dvdread/libdvdread.la $(LIB_KDEUI) -lkmdi $(LIBDL) -lk3b -lk3bdevice
 
 # this is where the desktop file will go 
 shelldesktopdir = $(kde_appsdir)/Utilities
