diff -Nuarp kdemultimedia-4.3.95.orig/juk/tag.cpp kdemultimedia-4.3.95/juk/tag.cpp
--- kdemultimedia-4.3.95.orig/juk/tag.cpp	2010-01-29 11:55:49.000000000 +0800
+++ kdemultimedia-4.3.95/juk/tag.cpp	2010-01-29 12:15:16.000000000 +0800
@@ -20,6 +20,7 @@
 
 #include <kdebug.h>
 #include <klocale.h>
+#include <KEncodingProber>
 
 #include <QtCore/QFile>
 
@@ -195,6 +196,33 @@ void Tag::setup(TagLib::File *file)
     m_album   = TStringToQString(file->tag()->album()).trimmed();
     m_genre   = TStringToQString(file->tag()->genre()).trimmed();
     m_comment = TStringToQString(file->tag()->comment()).trimmed();
+    if (QString(file->name()).endsWith(".mp3")) {
+        // detect encoding for mpeg id3v2
+        QString tmp = m_title + m_artist + m_album + m_genre + m_comment;
+        KEncodingProber prober(KEncodingProber::Universal);
+        KEncodingProber::ProberState result = prober.feed(tmp.toAscii());
+        if (result != KEncodingProber::NotMe) {
+            QByteArray encodingname = prober.encoding().toLower();
+            if ( prober.confidence() > 0.47
+                && ( ( encodingname == "gb18030" ) || ( encodingname == "big5" )
+                || ( encodingname == "euc-kr" ) || ( encodingname == "euc-jp" )
+                || ( encodingname == "koi8-r" ) ) ) {
+                m_title     = QTextCodec::codecForName(encodingname)->toUnicode(m_title.toAscii());
+                m_artist    = QTextCodec::codecForName(encodingname)->toUnicode(m_artist.toAscii());
+                m_album     = QTextCodec::codecForName(encodingname)->toUnicode(m_album.toAscii());
+                m_genre     = QTextCodec::codecForName(encodingname)->toUnicode(m_genre.toAscii());
+                m_comment   = QTextCodec::codecForName(encodingname)->toUnicode(m_comment.toAscii());
+            }
+            else if ((prober.confidence() < 0.3 || encodingname != "utf-8")
+                && QTextCodec::codecForLocale()->name().toLower() != "utf-8") {
+                m_title     = QTextCodec::codecForLocale()->toUnicode(m_title.toAscii());
+                m_artist    = QTextCodec::codecForLocale()->toUnicode(m_artist.toAscii());
+                m_album     = QTextCodec::codecForLocale()->toUnicode(m_album.toAscii());
+                m_genre     = QTextCodec::codecForLocale()->toUnicode(m_genre.toAscii());
+                m_comment   = QTextCodec::codecForLocale()->toUnicode(m_comment.toAscii());
+            }
+        }
+    }
 
     m_track = file->tag()->track();
     m_year  = file->tag()->year();
