diff -dur lxpanel-0.5.9.orig/data/default/panels/panel.in lxpanel-0.5.9/data/default/panels/panel.in
--- lxpanel-0.5.9.orig/data/default/panels/panel.in	2012-04-08 01:27:42.000000000 +0200
+++ lxpanel-0.5.9/data/default/panels/panel.in	2012-06-10 19:12:25.864088270 +0200
@@ -14,10 +14,10 @@
     setdocktype=1
     setpartialstrut=1
     usefontcolor=1
-    fontcolor=#ffffff
+    fontcolor=#000000
     usefontsize=0
     fontsize=10    
-    background=1
+    background=0
     backgroundfile=@prefix@/share/lxpanel/images/background.png
 }
 
@@ -37,12 +37,13 @@
         separator {
         }
         item {
+            image=system-run
             command=run
         }
         separator {
         }
         item {
-            image=gnome-logout
+            image=system-log-out
             command=logout
         }
     }
@@ -52,10 +53,10 @@
     type = launchbar
     Config {
         Button {
-            id=pcmanfm.desktop
+            id=fedora-pcmanfm.desktop
         }
         Button {
-            id=gnome-terminal.desktop
+            id=fedora-lxterminal.desktop
         }
         Button {
             id=firefox.desktop
@@ -75,6 +76,7 @@
     Config {
         Button1=iconify
         Button2=shade
+        Toggle=1
     }
 }
 
diff -dur lxpanel-0.5.9.orig/data/two_panels/panels/bottom.in lxpanel-0.5.9/data/two_panels/panels/bottom.in
--- lxpanel-0.5.9.orig/data/two_panels/panels/bottom.in	2012-04-08 01:27:42.000000000 +0200
+++ lxpanel-0.5.9/data/two_panels/panels/bottom.in	2012-06-10 18:49:41.559771662 +0200
@@ -14,7 +14,7 @@
     setdocktype=1
     setpartialstrut=1
     usefontcolor=0
-    fontcolor=#ffffff
+    fontcolor=#000000
     usefontsize=0
     fontsize=10    
     background=0
@@ -33,6 +33,7 @@
     Config {
         Button1=iconify
         Button2=shade
+        Toggle=1
     }
 }
 
diff -dur lxpanel-0.5.9.orig/data/two_panels/panels/top.in lxpanel-0.5.9/data/two_panels/panels/top.in
--- lxpanel-0.5.9.orig/data/two_panels/panels/top.in	2012-04-08 01:27:42.000000000 +0200
+++ lxpanel-0.5.9/data/two_panels/panels/top.in	2012-06-10 19:13:06.529349244 +0200
@@ -14,7 +14,7 @@
     setdocktype=1
     setpartialstrut=1
     usefontcolor=0
-    fontcolor=#ffffff
+    fontcolor=#000000
     usefontsize=0
     fontsize=10    
     background=0
@@ -31,19 +31,20 @@
 Plugin {
     type = menu
     Config {
-        image = @prefix@/share/lxpanel/images/my-computer.svg
+        image=@prefix@/share/lxpanel/images/my-computer.png
         system {
         }
         separator {
         }
         item {
-            command = run
+            image=system-run
+            command=run
         }
         separator {
         }
         item {
-            image = gnome-logout
-            command = logout
+            image=system-log-out
+            command=logout
         }
     }
 }
@@ -52,10 +53,10 @@
     type = launchbar
     Config {
         Button {
-            id=pcmanfm.desktop
+            id=fedora-pcmanfm.desktop
         }
         Button {
-            id=gnome-terminal.desktop
+            id=fedora-lxterminal.desktop
         }
         Button {
             id=firefox.desktop
