diff -Nur kdenetwork-4.0.80/kopete/protocols/msn/msnswitchboardsocket.cpp kdenetwork-4.0.80n/kopete/protocols/msn/msnswitchboardsocket.cpp
--- kdenetwork-4.0.80/kopete/protocols/msn/msnswitchboardsocket.cpp	2008-05-21 16:33:55.000000000 +0800
+++ kdenetwork-4.0.80n/kopete/protocols/msn/msnswitchboardsocket.cpp	2008-05-22 20:09:02.000000000 +0800
@@ -281,6 +281,16 @@
 		// sending a text message. So if it's not supplied, we'll just
 		// assume its that.
 
+		QString P4Name;
+		if( msg.contains( "P4-Context:" ) )
+		{
+		// The P4-Context field can override the default contact name.
+		// It's typically used by plugins of the official client (e.g. Xiaoi's Qun).
+			rx = QRegExp("P4-Context: ([^\r\n]*)");
+			rx.search(msg);
+			P4Name = rx.cap(1);
+		}
+
 		QColor fontColor;
 		QFont font;
 
@@ -350,7 +360,11 @@
 				others.append( m_account->contacts()[ *it2 ] );
 		}
 
-		QString message=msg.right( msg.length() - msg.indexOf("\r\n\r\n") - 4 );
+		QString message;
+		if(P4Name!=QString::null)
+		message=P4Name+" -- \r\n"+msg.right( msg.length() - msg.indexOf("\r\n\r\n") - 4 );
+		else
+		message=msg.right( msg.length() - msg.indexOf("\r\n\r\n") - 4 );
 
 		//Stupid MSN PLUS colors code. message with incorrect charactere are not showed correctly in the chat window.
 		//TODO: parse theses one to show the color too in Kopete
