diff -ur kdenetwork-4.7.0.orig//kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc kdenetwork-4.7.0/kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc
--- kdenetwork-4.7.0.orig//kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc	2011-09-11 22:32:34.000000000 -0400
+++ kdenetwork-4.7.0/kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc	2011-09-11 22:33:48.000000000 -0400
@@ -651,7 +651,7 @@
 #if OPENSSL_VERSION_NUMBER >= 0x1000000fL
       const X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
 #else
-      X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+      const X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
 #endif
       if (!meth)
         break;
