diff -up gnupg-2.0.18/g10/gpg.c.secmem gnupg-2.0.18/g10/gpg.c
--- gnupg-2.0.18/g10/gpg.c.secmem	2011-07-22 14:00:44.000000000 +0200
+++ gnupg-2.0.18/g10/gpg.c	2011-08-17 17:35:14.000000000 +0200
@@ -790,7 +790,7 @@ make_libversion (const char *libname, co
   
   if (maybe_setuid)
     {
-      gcry_control (GCRYCTL_INIT_SECMEM, 0, 0);  /* Drop setuid. */
+      gcry_control (GCRYCTL_INIT_SECMEM, 4096, 0);  /* Drop setuid. */
       maybe_setuid = 0;
     }
   s = getfnc (NULL);
@@ -893,7 +893,7 @@ build_list( const char *text, char lette
     char *list, *p, *line=NULL;
 
     if (maybe_setuid)
-      gcry_control (GCRYCTL_INIT_SECMEM, 0, 0);  /* Drop setuid. */
+      gcry_control (GCRYCTL_INIT_SECMEM, 4096, 0);  /* Drop setuid. */
 
     for(i=0; i <= 110; i++ )
 	if( !chkf(i) && (s=mapf(i)) )
diff -up gnupg-2.0.18/sm/gpgsm.c.secmem gnupg-2.0.18/sm/gpgsm.c
--- gnupg-2.0.18/sm/gpgsm.c.secmem	2011-08-04 11:55:08.000000000 +0200
+++ gnupg-2.0.18/sm/gpgsm.c	2011-08-17 18:20:40.000000000 +0200
@@ -493,7 +493,7 @@ make_libversion (const char *libname, co
 
   if (maybe_setuid)
     {
-      gcry_control (GCRYCTL_INIT_SECMEM, 0, 0);  /* Drop setuid. */
+      gcry_control (GCRYCTL_INIT_SECMEM, 4096, 0);  /* Drop setuid. */
       maybe_setuid = 0;
     }
   s = getfnc (NULL);
