--- dbus-tqt/message.cpp.dbustypes	2011-08-24 20:23:14.000000000 +0200
+++ dbus-tqt/message.cpp	2012-09-06 19:54:47.037175413 +0200
@@ -187,28 +210,28 @@
   case DBUS_TYPE_INT32:
     {
       dbus_int32_t v;
-      dbus_message_iter_get_basic (i, &v);
+      dbus_message_iter_get_int32 (i);
       ret = TQVariant( v );
     }
     break;
   case DBUS_TYPE_UINT32:
     {
       dbus_uint32_t v;
-      dbus_message_iter_get_basic (i, &v);
+      dbus_message_iter_get_uint32 (i);
       ret = TQVariant( v );
     }
     break;
   case DBUS_TYPE_DOUBLE:
     {
       double v;
-      dbus_message_iter_get_basic (i, &v);
+      dbus_message_iter_get_double (i);
       ret = TQVariant( v );
     }
     break;
   case DBUS_TYPE_STRING:
     {
       const char *v;
-      dbus_message_iter_get_basic (i, &v);
+      dbus_message_iter_get_string (i);
       ret = TQVariant( v );
     }
     break;
@@ -241,7 +264,7 @@
       while (dbus_message_iter_get_arg_type (&sub) != DBUS_TYPE_INVALID)
         {
           const char *v;
-          dbus_message_iter_get_basic (&sub, &v);
+          dbus_message_iter_get_string (&sub);
           tempList.append( TQString( v ) );
           dbus_message_iter_next (&sub);
         }
