--- kdebindings-3.5.13.1/dcopjava/binding/Makefile.am.ORI	2012-09-29 19:17:32.829295550 +0200
+++ kdebindings-3.5.13.1/dcopjava/binding/Makefile.am	2012-09-29 19:17:07.309943594 +0200
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = libjavadcop.la
 
 libjavadcop_la_SOURCES = client.cpp 
-libjavadcop_la_LDFLAGS = $(KDE_LDFLAGS) $(KDE_PLUGIN) -lDCOP -lkdecore -lkdeui $(LIB_QT) -lkjs -lkdefx
+libjavadcop_la_LDFLAGS = $(KDE_LDFLAGS) $(KDE_PLUGIN) -lDCOP -lkdecore -lkdeui $(LIB_QT) -lkjs -lkdefx $(QT_LDFLAGS)
 libjavadcop_la_LIBADD = $(LIB_KDECORE)
 
 INCLUDES = $(jni_includes) $(all_includes)
