diff -up local-lib-1.008009/lib/local/lib.pm.orig local-lib-1.008009/lib/local/lib.pm
--- local-lib-1.008009/lib/local/lib.pm.orig	2013-02-24 20:37:08.000000000 -0700
+++ local-lib-1.008009/lib/local/lib.pm	2013-04-19 08:58:58.161710254 -0600
@@ -343,7 +343,7 @@ sub build_bourne_env_declaration {
 sub build_csh_env_declaration {
   my $class = shift;
   my($name, $value) = @_;
-  return defined($value) ? qq{setenv ${name} "${value}"\n} : qq{unsetenv ${name}\n};
+  return defined($value) ? qq{setenv ${name} "${value}";\n} : qq{unsetenv ${name};\n};
 }
 
 sub build_win32_env_declaration {
