diff --git a/configure.ac b/configure.ac
index ff15e98..202a7d2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -331,6 +331,9 @@ TRACKER_MINER_FS_LIBS="$TRACKER_MINER_FS_LIBS -lz -lm"
 TRACKER_MINER_EVOLUTION_REQUIRED="glib-2.0                  >= $GLIB_REQUIRED
                                   evolution-shell-3.0       >= 3.1
                                   evolution-plugin-3.0
+                                  libemail-utils
+                                  libemail-engine
+                                  evolution-plugin-3.0
                                   evolution-data-server-1.2 >= $EDS_REQUIRED
                                   camel-1.2                 >= $CAMEL_REQUIRED"
 
@@ -343,6 +346,7 @@ if test "x$have_tracker_miner_evolution" = "xyes"; then
 
    AC_DEFINE(EVOLUTION_SHELL_3_2, 1, [Use new evolution-shell API])
    AC_DEFINE(EVOLUTION_SHELL_2_91, 1, [Use new evolution-shell API])
+   AC_DEFINE(EVOLUTION_SHELL_3_3_5, 1, [Use new evolution-shell API])
 else
    TRACKER_MINER_EVOLUTION_REQUIRED="glib-2.0                  >= $GLIB_REQUIRED
                                      evolution-shell-3.0
diff --git a/src/plugins/evolution/tracker-evolution-plugin.c b/src/plugins/evolution/tracker-evolution-plugin.c
index fc9923a..15a3977 100644
--- a/src/plugins/evolution/tracker-evolution-plugin.c
+++ b/src/plugins/evolution/tracker-evolution-plugin.c
@@ -40,27 +40,37 @@
 
 #include <sqlite3.h>
 
+#include <libedataserver/eds-version.h>
 #include <camel/camel.h>
 
-#include <mail/mail-config.h>
 #include <mail/em-utils.h>
-#include <mail/mail-ops.h>
 
 #ifdef EVOLUTION_SHELL_3_2
 #include <mail/e-mail.h>
 #endif
 
+#ifdef EVOLUTION_SHELL_3_3_5
+#include <libemail-engine/e-mail-folder-utils.h>
+#include <libemail-engine/e-mail-session.h>
+#include <libemail-engine/mail-config.h>
+#include <libemail-engine/mail-ops.h>
+#include <libemail-utils/e-account-utils.h>
+#else
+#include <mail/e-mail-folder-utils.h>
+#include <mail/mail-config.h>
+#include <mail/mail-ops.h>
+#include <e-util/e-account-utils.h>
 #ifdef EVOLUTION_SHELL_2_91
 #include <mail/e-mail-session.h>
 #else
 #include <mail/mail-session.h>
 #endif
+#endif
 
 #include <mail/e-mail-backend.h>
 #include <shell/e-shell.h>
 
 #include <e-util/e-config.h>
-#include <e-util/e-account-utils.h>
 
 #include <libedataserver/e-account.h>
 #include <libedataserver/e-account-list.h>
@@ -671,7 +681,11 @@ on_folder_summary_changed (CamelFolder           *folder,
 		const CamelFlag *cflags;
 		gchar *full_sparql;
 
+		#if EDS_CHECK_VERSION(3,3,1)
+		linfo = camel_folder_summary_get (summary, merged->pdata[i]);
+		#else
 		linfo = camel_folder_summary_uid (summary, merged->pdata[i]);
+		#endif
 
 		if (linfo) {
 			uid = (gchar *) camel_message_info_uid (linfo);
