diff -up cloog-ppl-0.15.11/configure.in.verfix cloog-ppl-0.15.11/configure.in
--- cloog-ppl-0.15.11/configure.in.verfix	2012-12-03 16:46:17.295807008 -0500
+++ cloog-ppl-0.15.11/configure.in	2012-12-03 16:46:59.010827161 -0500
@@ -338,7 +338,7 @@ if test "x$with_ppl" != "x" -a "x$with_p
 
 	AC_MSG_CHECKING([for version 0.10 (or later revision) of PPL])
 	AC_TRY_COMPILE([#include "ppl_c.h"],[
-	#if PPL_VERSION_MAJOR != 0 || PPL_VERSION_MINOR < 10
+	#if (PPL_VERSION_MAJOR != 1) || (PPL_VERSION_MAJOR == 0 && PPL_VERSION_MINOR < 10)
 	choke me
 	#endif
 	], [AC_MSG_RESULT([yes])], [AC_MSG_ERROR(Can't find correct version of PPL.) ])
diff -up cloog-ppl-0.15.11/configure.verfix cloog-ppl-0.15.11/configure
--- cloog-ppl-0.15.11/configure.verfix	2012-12-03 16:47:21.996837121 -0500
+++ cloog-ppl-0.15.11/configure	2012-12-03 16:47:34.893842170 -0500
@@ -11222,7 +11222,7 @@ int
 main ()
 {
 
-	#if PPL_VERSION_MAJOR != 0 || PPL_VERSION_MINOR < 10
+	#if (PPL_VERSION_MAJOR != 1) || (PPL_VERSION_MAJOR == 0 && PPL_VERSION_MINOR < 10)
 	choke me
 	#endif
 
