--- a/trunk/Xlib/ext/randr.py
+++ b/trunk/Xlib/ext/randr.py
@@ -1139,8 +1139,6 @@ class OutputPropertyNotify(rq.Event):
 # Initialization #
 
 def init(disp, info):
-    print info.__class__
-
     disp.extension_add_method('display', 'xrandr_query_version', query_version)
     disp.extension_add_method('window', 'xrandr_select_input', select_input)
     disp.extension_add_method('window', 'xrandr_get_screen_info', get_screen_info)
