--- ocaml-mysql-1.1.1/configure	2012-05-19 14:13:37.000000000 -0400
+++ ocaml-mysql-1.1.1/configure	2013-02-22 15:06:20.139917364 -0500
@@ -3470,9 +3470,9 @@ 
 	      /usr/local/lib
 	      /usr/lib
 	      /usr/lib/mysql
-	      /sw/lib/mysql"
+	      /usr/lib64/mysql"
 	for d in $dirs; do
-  	  	test -f "$d/libmysqlclient.a" && LDFLAGS="-L$d"
+  	  	test -f "$d/libmysqlclient.so" && LDFLAGS="-L$d"
 	done
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking in more locations..." >&5
 $as_echo "$as_me: checking in more locations..." >&6;}
