diff --git a/preupg-create-group-xml b/preupg-create-group-xml
index e75aba4..aa7a656 100755
--- a/preupg-create-group-xml
+++ b/preupg-create-group-xml
@@ -1,4 +1,5 @@
 #!/usr/bin/python2
+from __future__ import print_function
 import os
 import sys
 import optparse
@@ -24,12 +25,12 @@ def main():
                       )
     opts, args = parser.parse_args()
     if len(args) > 1 or len(args) == 0:
-        print 'Specify just one directory with INI file.'
+        print('Specify just one directory with INI file.')
         parser.print_help()
         sys.exit(1)
 
     if not os.path.exists(args[0]):
-        print 'Dir', args[0], 'does not exists.'
+        print('Dir %s does not exists.' % args[0])
         sys.exit(1)
 
     if args[0].endswith('/'):
@@ -40,12 +41,12 @@ def main():
         if d.endswith(".ini"):
             found = 1
     if not found:
-        print 'In directory %s was not found any INI file.' % args[0]
+        print ('In directory %s was not found any INI file.' % args[0])
         sys.exit(1)
 
     dir_name = utils.get_valid_scenario(args[0])
     if dir_name is None:
-        print 'Dir does not contain proper scenario.'
+        print('Dir does not contain proper scenario.')
         sys.exit(1)
     index = 0
     for i, d in enumerate(args[0].split(os.path.sep)):
@@ -63,7 +64,7 @@ def main():
     try:
         target_tree = ElementTree.parse(template_file).getroot()
     except IOError, e:
-        print 'Problem with reading template.xml file'
+        print ('Problem with reading template.xml file')
         sys.exit(1)
     target_tree = ComposeXML.run_compose(target_tree, os.path.dirname(dir_name), content=os.path.basename(dir_name))
 
@@ -71,9 +72,9 @@ def main():
         # must be encoded by ElementTree!
         data = ElementTree.tostring(target_tree, "utf-8")
         write_to_file(os.path.join(dir_name, filename), "wb", data, False)
-        print 'Generated report file for preupgrade-assistant is:', ''.join(os.path.join(dir_name, filename))
+        print ('Generated report file for preupgrade-assistant is:', ''.join(os.path.join(dir_name, filename)))
     except IOError, e:
-        print "Problem with writing file ", filename
+        print ("Problem with writing file %s." % filename)
 
 
     #oscap_group.write_profile_xml(target_tree)
diff --git a/preupg-xccdf-compose b/preupg-xccdf-compose
index 0da14fe..83529bf 100755
--- a/preupg-xccdf-compose
+++ b/preupg-xccdf-compose
@@ -1,5 +1,5 @@
 #!/usr/bin/python2
-
+from __future__ import print_function
 import sys
 import optparse
 from preuputils.compose import XCCDFCompose
@@ -13,7 +13,7 @@ def main():
     parser = optparse.OptionParser(usage="%prog dirname", description="Create XML files for OpenSCAP")
     opts, args = parser.parse_args()
     if len(args) > 1 or len(args) == 0:
-        print 'Specify just one directory'
+        print ('Specify just one directory')
         parser.print_help()
         sys.exit(0)
 
