diff -ur openjpeg_v1_4_sources_r697.orig/codec/convert.c openjpeg_v1_4_sources_r697/codec/convert.c
--- openjpeg_v1_4_sources_r697.orig/codec/convert.c	2012-01-19 12:00:07.000000000 -0500
+++ openjpeg_v1_4_sources_r697/codec/convert.c	2012-01-19 12:00:58.000000000 -0500
@@ -48,6 +48,7 @@
 #include "../libs/png/png.h"
 #else
 #include <png.h>
+#include <zlib.h>
 #endif /* _WIN32 */
 #endif /* HAVE_LIBPNG */
 
