--- libunique-1.1.6/tests/test-unique.c.orig	2014-05-07 11:05:02.743420865 +0100
+++ libunique-1.1.6/tests/test-unique.c	2014-05-07 11:24:20.163336706 +0100
@@ -82,10 +82,10 @@
                                    GTK_DIALOG_DESTROY_WITH_PARENT,
                                    GTK_MESSAGE_INFO,
                                    GTK_BUTTONS_CLOSE,
-                                   title);
+                                   "%s", title);
   if (message)
     gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog),
-                                              message);
+                                              "%s", message);
 
   gtk_window_set_urgency_hint (GTK_WINDOW (dialog), TRUE);
 
