diff -up foomatic-filters-4.0.8/foomaticrip.c.debug-string foomatic-filters-4.0.8/foomaticrip.c
--- foomatic-filters-4.0.8/foomaticrip.c.debug-string	2011-10-24 13:01:42.000000000 +0200
+++ foomatic-filters-4.0.8/foomaticrip.c	2011-10-24 13:05:55.737030758 +0200
@@ -1651,7 +1651,7 @@ int main(int argc, char** argv)
         if (dontparse == 2) {
             /* Raw queue, simply pass the input into the postpipe (or to STDOUT
                when there is no postpipe) */
-            _log("Raw printing, executing \"cat %s\"\n\n");
+            _log("Raw printing, executing \"cat %s\"\n\n", postpipe->data);
             snprintf(tmp, 1024, "cat %s", postpipe->data);
             run_system_process("raw-printer", tmp);
             continue;
