--- isdn4k-utils/xmonisdn/Makefile.in.man	2013-11-23 12:23:42.000000000 +0100
+++ isdn4k-utils/xmonisdn/Makefile.in	2013-11-23 12:57:21.280407677 +0100
@@ -51,7 +51,7 @@
 	CFLAGS= ; MAKEFLAGS= ; $(MAKE) -f Makefile EXTRA_DEFINES="$(I4LU_DEFINES)"
 	rm -rf X11
 
-xmonisdn.man: xmonisdn.man.in Makefile
+xmonisdn.man: xmonisdn.man.in
 	MANDATE=`grep CHECKIN $< | awk '{print $$4}'`; \
 	XUP=`echo ${CONFIG_XMONISDN_UPCMD}|sed 's/&/\\\&/g'`; \
 	XDOWN=`echo ${CONFIG_XMONISDN_DOWNCMD}|sed 's/&/\\\&/g'`; \
@@ -62,7 +62,7 @@
            -e "s,\@CONFIG_XMONISDN_DOWNCMD\@,$${XDOWN},g" \
 	   < $< > $@
 
-config:
+config: xmonisdn.man
 	@cp Makefile.in GNUmakefile
 	@xmkmf
 
