*** isdn4k-utils/capi20.new/capi20.c.org	2010-05-07 20:56:21.000000000 +0200
--- isdn4k-utils/capi20.new/capi20.c	2010-05-07 22:01:45.000000000 +0200
***************
*** 590,598 ****
  	psDir = opendir( pnModuleDir );
  	if ( psDir != NULL ) {
  		/* read entry by entry */
  		while ( ( psEntry = readdir( psDir ) ) != NULL )  {
! 			/* skip ".", ".." and files which do not end with "so" */
  			nLen = strlen( psEntry -> d_name );
  
  			switch ( nLen ) {
  				case 1:
--- 590,598 ----
  	psDir = opendir( pnModuleDir );
  	if ( psDir != NULL ) {
  		/* read entry by entry */
  		while ( ( psEntry = readdir( psDir ) ) != NULL )  {
! 			/* skip ".", ".." and files which do not end with "so.?" */
  			nLen = strlen( psEntry -> d_name );
  
  			switch ( nLen ) {
  				case 1:
***************
*** 605,613 ****
  						continue;
  					}
  					break;
  				default:
! 					if ( strncmp( psEntry -> d_name + nLen - 3, ".so", nLen ) ) {
  						continue;
  					}
  					break;
  			}
--- 605,613 ----
  						continue;
  					}
  					break;
  				default:
! 					if ( strncmp( psEntry -> d_name + nLen - 5, ".so.0", 4 ) ) {
  						continue;
  					}
  					break;
  			}
