diff -Npru isdn4k-utils-CVS-2010-05-01-patched.org/capi20/Makefile.am isdn4k-utils-CVS-2010-05-01-patched/capi20/Makefile.am
--- isdn4k-utils-CVS-2010-05-01-patched.org/capi20.new/Makefile.am	2009-05-08 23:02:02.000000000 +0200
+++ isdn4k-utils-CVS-2010-05-01-patched/capi20.new/Makefile.am	2012-11-12 00:15:12.237964590 +0100
@@ -19,14 +19,17 @@ modules_LTLIBRARIES = lib_capi_mod_std.l
 lib_capi_mod_std_la_SOURCES = capi_mod_std.c
 lib_capi_mod_std_la_CFLAGS = -fno-strict-aliasing
 lib_capi_mod_std_la_LDFLAGS = -shared
+lib_capi_mod_std_la_LIBADD = libcapi20.la
 
 lib_capi_mod_fritzbox_la_SOURCES = capi_mod_fritzbox.c
 lib_capi_mod_fritzbox_la_CFLAGS = -fno-strict-aliasing
 lib_capi_mod_fritzbox_la_LDFLAGS = -shared
+lib_capi_mod_fritzbox_la_LIBADD = libcapi20.la
 
 lib_capi_mod_rcapi_la_SOURCES = capi_mod_rcapi.c
 lib_capi_mod_rcapi_la_CFLAGS = -fno-strict-aliasing
 lib_capi_mod_rcapi_la_LDFLAGS = -shared
+lib_capi_mod_rcapi_la_LIBADD = libcapi20.la
 
 libcapi20dyn_a_SOURCES = capidyn.c
 libcapi20dyn_a_CFLAGS = -fPIC
