--- isdn4k-utils-CVS-2010-05-01-patched/ipppd/main.c	2014-01-28 15:41:17.902128875 -0500
+++ isdn4k-utils-CVS-2010-05-01-patched.new/ipppd/main.c	2014-01-28 15:40:55.831032101 -0500
@@ -1297,16 +1297,8 @@
         break;
     case 'r':
         f = va_arg(args, char *);
-        /*
-         * XXX We assume a va_list is either a pointer or an array, so
-         * what gets passed for a va_list is like a void * in some sense.
-         */
         a = va_arg(args, void *);
-#if defined(__alpha__) || defined(__sh__)       /* always do this? */
 	n = fmtmsg(buf, buflen + 1, f, a);
-#else
-	n = vfmtmsg(buf, buflen + 1, f, a);
-#endif
         buf += n;
         buflen -= n;
         continue;
