--- python-trinity-3.5.13.1/configure.py.ORI	2012-10-02 19:01:51.164215802 +0000
+++ python-trinity-3.5.13.1/configure.py	2012-10-02 19:02:06.842913909 +0000
@@ -411,6 +411,7 @@
     if not opt_kdeincdir:
         incSearchPaths = [os.path.join (opt_kdebasedir, "include")]
         incSearchPaths.append (os.path.join (opt_kdebasedir, "include", "kde")) # Red Hat
+        incSearchPaths.append (os.path.join (opt_kdebasedir, "include", "tde")) # TDE 3.5.13.1
         opt_kdeincdir = search ("kapplication.h", incSearchPaths)
 
 def check_kde_installation():
