--- Crypt-OpenSSL-PKCS10-0.06/Makefile.PL	2006-02-01 12:44:20.000000000 -0800
+++ Crypt-OpenSSL-PKCS10-0.06.new/Makefile.PL	2007-05-31 14:40:31.000000000 -0700
@@ -9,7 +9,6 @@
     ($] >= 5.005 ?     ## Add these new keywords supported since 5.005
       (ABSTRACT_FROM  => 'PKCS10.pm',
        AUTHOR         => 'JoNO <jonozzz@yahoo.com>') : ()),
-    LIBS              => ['-L/usr/local/ssl/lib -lcrypto'],
+    LIBS              => ['-lcrypto'],
     DEFINE            => '-DPERL5 -Wall',
-    INC               => '-I/usr/local/ssl/include',
 );
