diff -u -r kaffeine-0.8.2.orig/kaffeine/src/input/audiobrowser/playlist.cpp kaffeine-0.8.2/kaffeine/src/input/audiobrowser/playlist.cpp
--- kaffeine-0.8.2.orig/kaffeine/src/input/audiobrowser/playlist.cpp	2006-09-09 09:29:55.000000000 +0800
+++ kaffeine-0.8.2/kaffeine/src/input/audiobrowser/playlist.cpp	2006-09-12 08:55:50.000000000 +0800
@@ -1468,6 +1468,7 @@
 	TQString title;
 	TQString artist;
 	TQString album;
+	TQString ext;
 
 	TQTextCodec *altCodec;
 	TQTextCodec *CodecUtf8;
@@ -1476,6 +1477,9 @@
 	CodecUtf8 = TQTextCodec::codecForName("UTF-8");
 	//kdDebug() << "playlist: locale " << Codec->name << " index: " << m_alternateEncoding << endl;
 
+        ext = mrl.kurl().fileName();
+        ext = ext.remove(0 , ext.findRev('.') +1).lower();
+
 	for (m = 0; m < groups.count(); m++)
 	{
 		//kdDebug() << "Metainfo-Group: " << groups[m] << endl;
@@ -1491,7 +1495,10 @@
 
 			if (keys[n] == "Title")
 			{
-				title = metaGroup.item(keys[n]).value().toString().simplifyWhiteSpace();
+				if (ext=="wma" || ext=="wmv" || ext=="asf" || ext=="rm" || ext=="rmvb" || ext=="rmm" || ext=="ra")
+				        title=mrl.kurl().fileName();
+				else
+				        title = metaGroup.item(keys[n]).value().toString().simplifyWhiteSpace();
 				if ((!title.isEmpty()) && (title.contains(TQRegExp("\\w")) > 2) && (title.left(5).lower() != "track"))
 				{
 					if ((m_useAlternateEncoding) && (CodecUtf8->heuristicContentMatch(title.ascii(), title.length()) < 0))
@@ -1509,7 +1516,10 @@
 
 			if (keys[n] == "Artist")
 			{
-				artist = (metaGroup.item(keys[n]).value().toString().simplifyWhiteSpace());
+				if (ext=="wma" || ext=="wmv" || ext=="asf" || ext=="rm" || ext=="rmvb" || ext=="rmm" || ext=="ra")
+				        artist="";
+				else
+				        artist = (metaGroup.item(keys[n]).value().toString().simplifyWhiteSpace());
 				if (!artist.isEmpty())
 				{
 					if ((m_useAlternateEncoding) && (CodecUtf8->heuristicContentMatch(artist.ascii(),artist.length()) < 0 ))
@@ -1525,7 +1535,10 @@
 
 			if (keys[n] == "Album")
 			{
-				album = (metaGroup.item(keys[n]).value().toString().simplifyWhiteSpace());
+				if (ext=="wma" || ext=="wmv" || ext=="asf" || ext=="rm" || ext=="rmvb" || ext=="rmm" || ext=="ra")
+				         album="";
+				else     
+				         album = (metaGroup.item(keys[n]).value().toString().simplifyWhiteSpace());
 				if (!album.isEmpty())
 				{
 					if ((m_useAlternateEncoding) && (CodecUtf8->heuristicContentMatch(album.ascii(),album.length()) < 0 ))
diff -u -r kaffeine-0.8.2.orig/kaffeine/src/player-parts/xine-part/xine_part.cpp kaffeine-0.8.2/kaffeine/src/player-parts/xine-part/xine_part.cpp
--- kaffeine-0.8.2.orig/kaffeine/src/player-parts/xine-part/xine_part.cpp	2006-09-09 09:29:57.000000000 +0800
+++ kaffeine-0.8.2/kaffeine/src/player-parts/xine-part/xine_part.cpp	2006-09-12 08:53:52.000000000 +0800
@@ -1020,6 +1020,7 @@
 void XinePart::slotTrackPlaying()
 {
 	TQString caption;
+	TQString ext;
 
 	kdDebug() << "XinePart: xine is playing" << endl;
 	m_pauseButton->setChecked(false);
@@ -1038,13 +1039,16 @@
 
 	if (mrl.length().isNull()) /* no meta */
 	{
+		ext = mrl.kurl().fileName();
+                ext = ext.remove(0 , ext.findRev('.') +1).lower();
 		if ((!m_xine->getTitle().isEmpty()) && (!m_xine->getTitle().contains('/'))
 		        && (m_xine->getTitle().contains(TQRegExp("\\w")) > 2) && (m_xine->getTitle().left(5).lower() != "track"))
-			mrl.setTitle(m_xine->getTitle());
+			if (ext!="wma" && ext!="wmv" && ext!="asf" && ext!="rm" && ext!="rmvb" && ext!="rmm" && ext!="ra")					mrl.setTitle(m_xine->getTitle());
 		if ((mrl.artist().isEmpty()) && (!m_xine->getArtist().isEmpty()))
-			mrl.setArtist(m_xine->getArtist());
+			if (ext!="wma" && ext!="wmv" && ext!="asf" && ext!="rm" && ext!="rmvb" && ext!="rmm" && ext!="ra")					mrl.setArtist(m_xine->getArtist());
 		if ((mrl.album().isEmpty()) && (!m_xine->getAlbum().isEmpty()))
-			mrl.setAlbum(m_xine->getAlbum());
+			if (ext!="wma" && ext!="wmv" && ext!="asf" && ext!="rm" && ext!="rmvb" && ext!="rmm" && ext!="ra")
+                            mrl.setAlbum(m_xine->getAlbum());
 		if ((mrl.year().isEmpty()) && (!m_xine->getYear().isEmpty()))
 			mrl.setYear(m_xine->getYear());
 		if ((mrl.genre().isEmpty()) && (!m_xine->getGenre().isEmpty()))
