diff -Nur graphviz-2.30.1-orig/configure.ac graphviz-2.30.1/configure.ac
--- graphviz-2.30.1-orig/configure.ac	2013-08-21 18:13:52.000000000 +0000
+++ graphviz-2.30.1/configure.ac	2013-08-21 18:14:42.000000000 +0000
@@ -3491,7 +3491,7 @@
 case "${host_os}" in
   *linux* )
     case "${host_cpu}" in
-      powerpc64 | s390x | x86_64 | sparc64 )
+      powerpc64 | s390x | x86_64 | sparc64 | mips64* )
         LIBPOSTFIX="64"
         ;;
     esac
