diff -u -r kaffeine-0.8.6.orig/kaffeine/src/pref.cpp kaffeine-0.8.6/kaffeine/src/pref.cpp
--- kaffeine-0.8.6.orig/kaffeine/src/pref.cpp	2008-01-22 18:58:18.000000000 +0800
+++ kaffeine-0.8.6/kaffeine/src/pref.cpp	2008-01-22 19:06:03.000000000 +0800
@@ -42,6 +42,9 @@
 #include <tqlineedit.h>
 
 #include <tqtextcodec.h>
+#include <kprocess.h>
+#include <kstandarddirs.h>
+#include <tqdir.h>
 
 #include <kcharsets.h>
 
@@ -148,6 +151,13 @@
 	grid->addWidget(clearLabel, 0, 1);
 	vb->addWidget( gb );
 
+       gb = new TQGroupBox( "", misc );
+       grid = new TQGridLayout( gb, 1, 1, 20, 6 );
+       m_usePlugin = new TQCheckBox(i18n("Use Plugin in konqueror instead of kaffeine_part to play media streaming"), gb);
+//    connect(m_usePlugin, TQT_SIGNAL(toggled(bool)), this, TQT_SLOT(slotUsePluginToggled(bool)));
+      grid->addWidget(m_usePlugin, 0, 0);
+      vb->addWidget( gb );
+
 	vb->addItem( new TQSpacerItem( 20, 20, TQSizePolicy::Ignored, TQSizePolicy::Ignored ) );
 
 	delete icon;
@@ -174,6 +184,9 @@
 	m_useAlternateEncoding->setChecked(m_useEncoding);
 	m_alternateEncoding->setEnabled(m_useEncoding);
 	m_pauseVideo->setChecked(pauseVideo);
+	TQString homePath= TQDir::homeDirPath();
+        if (TQFileInfo(homePath+"/.kde/share/services/kaffeine_part.desktop").exists())
+              m_usePlugin->setChecked(true);
 }
 
 void KaffeinePreferences::setDvbClient( bool enabled, const TQString &address, int port, int info, const TQString &tspath )
@@ -193,6 +206,7 @@
 	emit signalSetOSDTimeout(m_osdTimeout->value());
 	emit signalPauseVideo(m_pauseVideo->isChecked());
 	emit signalDvbClient(m_dcEnabled->isChecked(), m_dcAddress->text().stripWhiteSpace(), m_dcPort->value(), m_dcInfo->value(), m_shiftDirLe->text() );
+	slotUsePluginToggled(m_usePlugin->isChecked());
 }
 
 void KaffeinePreferences::slotOkPressed()
@@ -217,3 +231,21 @@
 {
 	m_osdTimeout->setEnabled(on);
 }
+
+void KaffeinePreferences::slotUsePluginToggled(bool on)
+{
+  KProcess *proc = new KProcess;
+  TQString homePath= TQDir::homeDirPath();
+  if (on){
+      TQString kpartPath = locate("appdata", "kaffeine_part.desktop");
+      if (!kpartPath)
+              return;
+      *proc << "cp"<<"-f"<<kpartPath<<homePath+"/.kde/share/services/";
+      proc->start();
+  }else{
+      *proc << "rm" <<"-rf" <<homePath+"/.kde/share/services/kaffeine_part.desktop";
+      proc->start();
+  }
+  *proc << "kbuildsycoca";
+  proc->start();
+}
Only in kaffeine-0.8.6/kaffeine/src: pref.cpp.orig
Only in kaffeine-0.8.6/kaffeine/src: pref.cpp.rej
diff -u -r kaffeine-0.8.6.orig/kaffeine/src/pref.h kaffeine-0.8.6/kaffeine/src/pref.h
--- kaffeine-0.8.6.orig/kaffeine/src/pref.h	2008-01-22 18:58:18.000000000 +0800
+++ kaffeine-0.8.6/kaffeine/src/pref.h	2008-01-22 18:58:43.000000000 +0800
@@ -60,6 +60,7 @@
   void slotUseAlternateEncodingToggled(bool);
   void slotEmbedInTrayToggled(bool);
   void setShiftDir();
+ void slotUsePluginToggled(bool);
 
 private:
   TQCheckBox* m_systemTray;
@@ -73,6 +74,7 @@
   TQSpinBox* m_dcInfo;
   TQLineEdit *m_shiftDirLe;
   TQToolButton *m_shiftDirBtn;
+  TQCheckBox* m_usePlugin;
 };
 
 #endif /* PREF_H */
