diff -Naur socat-1.7.2.1-orig/procan-cdefs.c socat-1.7.2.1/procan-cdefs.c
--- socat-1.7.2.1-orig/procan-cdefs.c	2010-10-06 03:25:30.000000000 -0400
+++ socat-1.7.2.1/procan-cdefs.c	2012-11-20 17:15:37.521215330 -0500
@@ -20,7 +20,7 @@
    fprintf(outfile, "#define FD_SETSIZE %u\n", FD_SETSIZE);
 #endif
 #ifdef NFDBITS
-   fprintf(outfile, "#define NFDBITS "F_Zu"\n", NFDBITS);
+   fprintf(outfile, "#define NFDBITS %u\n", NFDBITS);
 #endif
 #ifdef O_RDONLY
    fprintf(outfile, "#define O_RDONLY %u\n", O_RDONLY);
