--- ksensors-0.7.3/admin/cvs.sh.ORI	2012-10-07 11:06:41.616636004 +0200
+++ ksensors-0.7.3/admin/cvs.sh	2012-10-07 11:07:55.975129334 +0200
@@ -64,7 +64,7 @@
     echo "*** KDE requires automake 1.6.1 or newer"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* ) : ;;
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.1[012]* ) : ;;
   "" )
     echo "*** AUTOMAKE NOT FOUND!."
     echo "*** KDE requires automake 1.6.1 or newer"
