--- ksensors-0.7.3/src/sounds/Makefile.am.ORI	2012-10-07 11:24:46.315092603 +0200
+++ ksensors-0.7.3/src/sounds/Makefile.am	2012-10-07 11:25:08.288633760 +0200
@@ -5,6 +5,7 @@
 EXTRA_DIST = ksensors_alert.wav
 
 install-data-local:
+	$(mkinstalldirs) $(kde_sounddir)
 	$(INSTALL_DATA) $(srcdir)/ksensors_alert.wav $(kde_sounddir)/ksensors_alert.wav
 uninstall-local:
 	rm -f $(kde_sounddir)/ksensors_alert.wav
