diff -Naupr p7zip_9.13.orig/makefile.linux_amd64_asm p7zip_9.13/makefile.linux_amd64_asm
--- p7zip_9.13.orig/makefile.linux_amd64_asm	2009-12-22 19:11:03.000000000 +0100
+++ p7zip_9.13/makefile.linux_amd64_asm	2010-07-08 14:12:33.197398954 +0200
@@ -2,7 +2,7 @@
 OPTFLAGS=-O
 
 # use "-m32" to have a 32bits executable
-ALLFLAGS=-m64 ${OPTFLAGS} -pipe -s \
+ALLFLAGS=-m64 ${OPTFLAGS} -pipe \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	-D_7ZIP_LARGE_PAGES \
diff -Naupr p7zip_9.13.orig/makefile.linux_x86_asm_gcc_4.X p7zip_9.13/makefile.linux_x86_asm_gcc_4.X
--- p7zip_9.13.orig/makefile.linux_x86_asm_gcc_4.X	2009-12-22 19:12:54.000000000 +0100
+++ p7zip_9.13/makefile.linux_x86_asm_gcc_4.X	2010-07-08 14:13:36.185397479 +0200
@@ -4,7 +4,7 @@
 
 OPTFLAGS=-O
 
-ALLFLAGS=${OPTFLAGS} -pipe -m32 -s \
+ALLFLAGS=${OPTFLAGS} -pipe -m32 \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	-D_7ZIP_LARGE_PAGES \
