diff --git a/rerun-0.10.0.gemspec b/rerun-0.10.0.gemspec
index db20918..cbb8426 100644
--- a/specifications/rerun-0.10.0.gemspec
+++ b/specifications/rerun-0.10.0.gemspec
@@ -25,11 +25,11 @@ Gem::Specification.new do |s|
     s.specification_version = 2
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
-      s.add_runtime_dependency(%q<listen>, [">= 2.7.3", "~> 2.7"])
+      s.add_runtime_dependency(%q<listen>, ["~> 3.0"])
     else
-      s.add_dependency(%q<listen>, [">= 2.7.3", "~> 2.7"])
+      s.add_dependency(%q<listen>, ["~> 3.0"])
     end
   else
-    s.add_dependency(%q<listen>, [">= 2.7.3", "~> 2.7"])
+    s.add_dependency(%q<listen>, ["~> 3.0"])
   end
 end

