diff -up tsocks-1.8/Makefile.in.soname tsocks-1.8/Makefile.in
--- tsocks-1.8/Makefile.in.soname	2011-02-17 13:05:44.401652007 -0600
+++ tsocks-1.8/Makefile.in	2011-02-17 13:09:32.936663785 -0600
@@ -24,6 +24,7 @@ SCRIPT = tsocks
 SHLIB_MAJOR = 1
 SHLIB_MINOR = 8
 SHLIB = ${LIB_NAME}.so.${SHLIB_MAJOR}.${SHLIB_MINOR}
+SONAME = ${LIB_NAME}.so.${SHLIB_MAJOR}
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
@@ -50,8 +51,9 @@ ${SAVE}: ${SAVE}.c
 	${SHCC} ${CFLAGS} ${INCLUDES} -o ${SAVE} ${SAVE}.c
 
 ${SHLIB}: ${OBJS} ${COMMON}.o ${PARSER}.o
-	${SHCC} ${CFLAGS} ${INCLUDES} -nostdlib -shared -o ${SHLIB} ${OBJS} ${COMMON}.o ${PARSER}.o ${DYNLIB_FLAGS} ${SPECIALLIBS} ${LIBS}
-	ln -sf ${SHLIB} ${LIB_NAME}.so
+	${SHCC} ${CFLAGS} ${INCLUDES} -nostdlib -shared -o ${SHLIB} -Wl,-soname=${SONAME} ${OBJS} ${COMMON}.o ${PARSER}.o ${DYNLIB_FLAGS} ${SPECIALLIBS} ${LIBS}
+	ln -sf ${SHLIB} ${SONAME}
+	ln -sf ${SONAME} ${LIB_NAME}.so
 
 %.so: %.c
 	${SHCC} ${CFLAGS} ${INCLUDES} -c ${CC_SWITCHES} $< -o $@
@@ -68,8 +70,8 @@ installscript:
 installlib:
 	${MKINSTALLDIRS} "${DESTDIR}${libdir}"
 	${INSTALL} ${SHLIB} ${DESTDIR}${libdir}
-	ln -sf ${SHLIB} ${DESTDIR}${libdir}/${LIB_NAME}.so.${SHLIB_MAJOR}
-	ln -sf ${LIB_NAME}.so.${SHLIB_MAJOR} ${DESTDIR}${libdir}/${LIB_NAME}.so
+	ln -sf ${SHLIB} ${DESTDIR}${libdir}/${SONAME}
+	ln -sf ${SONAME} ${DESTDIR}${libdir}/${LIB_NAME}.so
 
 installman:
 	${MKINSTALLDIRS} "${DESTDIR}${mandir}/man1"
